{
 "metadata": {
  "name": ""
 },
 "nbformat": 3,
 "nbformat_minor": 0,
 "worksheets": [
  {
   "cells": [
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "import glob\n",
      "import os\n",
      "\n",
      "import pandas as pd\n",
      "from IPython.core.display import HTML\n",
      "\n",
      "from gscripts.general import parsers\n",
      "\n",
      "es_riboseq = \"/nas3/gpratt/projects/es_riboseq/analysis/es_riboseq_v1/\"\n",
      "public_riboseq = \"/nas3/gpratt/projects/public_riboseq/analysis/public_riboseq_v1/\"\n",
      "es_rnaseq = \"/nas3/gpratt/projects/es_riboseq/analysis/es_rnaseq_v1/\""
     ],
     "language": "python",
     "metadata": {
      "slideshow": {
       "slide_type": "skip"
      }
     },
     "outputs": [],
     "prompt_number": 1
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def percent(x):\n",
      "    return \"{:.2f}%\".format(x * 100)"
     ],
     "language": "python",
     "metadata": {
      "slideshow": {
       "slide_type": "skip"
      }
     },
     "outputs": [],
     "prompt_number": 2
    },
    {
     "cell_type": "markdown",
     "metadata": {
      "slideshow": {
       "slide_type": "skip"
      }
     },
     "source": [
      "#Build Mapping Stats"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "combined_df = pd.concat([parsers.rnaseq_metrics(es_riboseq), \n",
      "])\n",
      "combined_df[\"percent_mapped\"] = combined_df['Uniquely mapped reads number'].astype(float) / combined_df['Number of input reads']\n",
      "combined_df[\"percent_repetative\"] = 1 - (combined_df['Number of input reads'] / combined_df['Processed reads'].astype(float))\n",
      "combined_df = combined_df.drop([\"Finished on\", \n",
      "                  \"Mapping speed, Million of reads per hour\", \n",
      "                  \"Trimmed bases\",\n",
      "                  \"Started job on\",\n",
      "                  \"Started mapping on\"], axis=1)\n",
      "\n"
     ],
     "language": "python",
     "metadata": {
      "slideshow": {
       "slide_type": "skip"
      }
     },
     "outputs": [],
     "prompt_number": 3
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "rnaseq_df = parsers.rnaseq_metrics(es_rnaseq)\n",
      "rnaseq_df[\"percent_mapped\"] = rnaseq_df['Uniquely mapped reads number'].astype(float) / rnaseq_df['Number of input reads']\n",
      "rnaseq_df[\"percent_repetative\"] = 1 - (rnaseq_df['Number of input reads'] / rnaseq_df['Processed reads'].astype(float))\n",
      "rnaseq_df = rnaseq_df.drop([\"Finished on\", \n",
      "                  \"Mapping speed, Million of reads per hour\", \n",
      "                  \"Trimmed bases\",\n",
      "                  \"Started job on\",\n",
      "                  \"Started mapping on\"], axis=1)"
     ],
     "language": "python",
     "metadata": {
      "slideshow": {
       "slide_type": "skip"
      }
     },
     "outputs": [],
     "prompt_number": 4
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "public_df = parsers.rnaseq_metrics(public_riboseq)\n",
      "public_df[\"percent_mapped\"] = public_df['Uniquely mapped reads number'].astype(float) / public_df['Number of input reads']\n",
      "public_df[\"percent_repetative\"] = 1 - (public_df['Number of input reads'] / public_df['Processed reads'].astype(float))\n",
      "public_df = public_df.drop([\"Finished on\", \n",
      "                  \"Mapping speed, Million of reads per hour\", \n",
      "                  \"Trimmed bases\",\n",
      "                  \"Started job on\",\n",
      "                  \"Started mapping on\"], axis=1)"
     ],
     "language": "python",
     "metadata": {
      "slideshow": {
       "slide_type": "skip"
      }
     },
     "outputs": [],
     "prompt_number": 5
    },
    {
     "cell_type": "markdown",
     "metadata": {
      "slideshow": {
       "slide_type": "fragment"
      }
     },
     "source": [
      "#Mapping Stats"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "float(combined_df.sum()[\"Uniquely mapped reads number\"]) / float(combined_df.sum()[\"Processed reads\"])"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 6,
       "text": [
        "0.15548548928921163"
       ]
      }
     ],
     "prompt_number": 6
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "HTML(combined_df[['Processed reads',  \n",
      "                  'Number of input reads', \n",
      "                  'Uniquely mapped reads number', \n",
      "                  '% of reads unmapped: too short', \n",
      "                  'percent_repetative',\n",
      "                  'percent_mapped',\n",
      "                  ]].to_html(formatters={\"Number of input reads\" : parsers.commas,\n",
      "                             \"Uniquely mapped reads number\" : parsers.commas,\n",
      "                             \"Usable Reads\" : parsers.commas,\n",
      "                             \"Num Peaks\": parsers.commas,\n",
      "                             \"Trimmed reads\" : parsers.commas,\n",
      "                             \"Processed reads\" : parsers.commas,\n",
      "                             \"Too short reads\" : parsers.commas,\n",
      "                             \"percent_repetative\" :  percent,\n",
      "                             \"percent_mapped\" : percent,\n",
      "                             }))"
     ],
     "language": "python",
     "metadata": {
      "slideshow": {
       "slide_type": "fragment"
      }
     },
     "outputs": [
      {
       "html": [
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>Processed reads</th>\n",
        "      <th>Number of input reads</th>\n",
        "      <th>Uniquely mapped reads number</th>\n",
        "      <th>% of reads unmapped: too short</th>\n",
        "      <th>percent_repetative</th>\n",
        "      <th>percent_mapped</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>ES_1_GATCAG_L006_R1</th>\n",
        "      <td> 12,250,315</td>\n",
        "      <td> 11,661,617</td>\n",
        "      <td>     62,257</td>\n",
        "      <td> 98.84%</td>\n",
        "      <td> 0.04805574</td>\n",
        "      <td> 0.005338625</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ES_2_ACTTGA_L006_R1</th>\n",
        "      <td> 17,261,048</td>\n",
        "      <td>  8,448,461</td>\n",
        "      <td>  4,361,974</td>\n",
        "      <td> 15.83%</td>\n",
        "      <td>  0.5105476</td>\n",
        "      <td>    0.516304</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ES_3_CAGATC_L006_R1</th>\n",
        "      <td>104,760,758</td>\n",
        "      <td> 45,329,485</td>\n",
        "      <td> 18,259,254</td>\n",
        "      <td> 14.17%</td>\n",
        "      <td>  0.5673047</td>\n",
        "      <td>   0.4028119</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NPC_1_CGATGT_L005_R1</th>\n",
        "      <td>  2,321,435</td>\n",
        "      <td>    667,926</td>\n",
        "      <td>    360,730</td>\n",
        "      <td> 23.74%</td>\n",
        "      <td>  0.7122788</td>\n",
        "      <td>   0.5400748</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NPC_2_ATCACG_L005_R1</th>\n",
        "      <td>101,588,130</td>\n",
        "      <td> 28,117,308</td>\n",
        "      <td> 15,292,287</td>\n",
        "      <td> 21.11%</td>\n",
        "      <td>  0.7232225</td>\n",
        "      <td>   0.5438745</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SOX17_1_TGACCA_L005_R1</th>\n",
        "      <td>  2,905,925</td>\n",
        "      <td>  1,017,382</td>\n",
        "      <td>    277,813</td>\n",
        "      <td> 53.99%</td>\n",
        "      <td>  0.6498939</td>\n",
        "      <td>   0.2730666</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SOX17_2_TTAGGC_L005_R1</th>\n",
        "      <td> 18,112,334</td>\n",
        "      <td>  6,456,528</td>\n",
        "      <td>  1,991,691</td>\n",
        "      <td> 52.75%</td>\n",
        "      <td>  0.6435287</td>\n",
        "      <td>   0.3084771</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>TGFP_1_GCCAAT_L006_R1</th>\n",
        "      <td>  4,468,035</td>\n",
        "      <td>  1,256,328</td>\n",
        "      <td>    360,979</td>\n",
        "      <td> 46.62%</td>\n",
        "      <td>  0.7188187</td>\n",
        "      <td>   0.2873286</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>TGFP_2_ACAGTG_L005_R1</th>\n",
        "      <td>  1,529,539</td>\n",
        "      <td>    668,537</td>\n",
        "      <td>    267,381</td>\n",
        "      <td> 34.70%</td>\n",
        "      <td>   0.562916</td>\n",
        "      <td>   0.3999494</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 7,
       "text": [
        "<IPython.core.display.HTML at 0x5b4e610>"
       ]
      }
     ],
     "prompt_number": 7
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "100000000 - combined_df['Processed reads']"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 8,
       "text": [
        "ES_1_GATCAG_L006_R1       87749685\n",
        "ES_2_ACTTGA_L006_R1       82738952\n",
        "ES_3_CAGATC_L006_R1       -4760758\n",
        "NPC_1_CGATGT_L005_R1      97678565\n",
        "NPC_2_ATCACG_L005_R1      -1588130\n",
        "SOX17_1_TGACCA_L005_R1    97094075\n",
        "SOX17_2_TTAGGC_L005_R1    81887666\n",
        "TGFP_1_GCCAAT_L006_R1     95531965\n",
        "TGFP_2_ACAGTG_L005_R1     98470461\n",
        "Name: Processed reads, dtype: int64"
       ]
      }
     ],
     "prompt_number": 8
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "#Ribo-seq Libraries\n",
      "Analyzable Libraries:\n",
      "----------------\n",
      "+ NPC_2_ATCACG_L005_R1\n",
      "\n",
      "Borderline Libraries:\n",
      "------------------\n",
      "+ ES_2_ACTTGA_L006_R1\n",
      "\n",
      "Libraries to be Resequenced:\n",
      "----------------------------\n",
      "+ NPC_1_CGATGT_L005_R1\n",
      "+ SOX17_1_TGACCA_L005_R1\n",
      "+ SOX17_2_TTAGGC_L005_R1\n",
      "+ TGFP_1_GCCAAT_L006_R1\n",
      "+ TGFP_2_ACAGTG_L005_R1\n",
      "\n",
      "Did Not Work\n",
      "-----------\n",
      "+ ES_3_CAGATC_L006_R1 (faulty, looks like RNA-seq)\n",
      "\n",
      "Looking at percent mapped and precent repetative stats it looks like the lowly sequenced libraries were just miss-quantified because the have similar mapped and repetative removal stats"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "HTML(public_df[['Processed reads',  \n",
      "                  'Number of input reads', \n",
      "                  'Uniquely mapped reads number', \n",
      "                  '% of reads unmapped: too short', \n",
      "                  'percent_repetative',\n",
      "                  'percent_mapped',\n",
      "                  ]].to_html(formatters={\"Number of input reads\" : parsers.commas,\n",
      "                             \"Uniquely mapped reads number\" : parsers.commas,\n",
      "                             \"Usable Reads\" : parsers.commas,\n",
      "                             \"Num Peaks\": parsers.commas,\n",
      "                             \"Trimmed reads\" : parsers.commas,\n",
      "                             \"Processed reads\" : parsers.commas,\n",
      "                             \"Too short reads\" : parsers.commas,\n",
      "                             \"percent_repetative\" :  percent,\n",
      "                             \"percent_mapped\" : percent,\n",
      "                             }))"
     ],
     "language": "python",
     "metadata": {
      "slideshow": {
       "slide_type": "slide"
      }
     },
     "outputs": [
      {
       "html": [
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>Processed reads</th>\n",
        "      <th>Number of input reads</th>\n",
        "      <th>Uniquely mapped reads number</th>\n",
        "      <th>% of reads unmapped: too short</th>\n",
        "      <th>percent_repetative</th>\n",
        "      <th>percent_mapped</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>494_10_NoIndex_L002_R1_</th>\n",
        "      <td>50,303,823</td>\n",
        "      <td> 23,967,437</td>\n",
        "      <td> 14,721,531</td>\n",
        "      <td> 17.33%</td>\n",
        "      <td>   0.5235464</td>\n",
        "      <td>    0.6142305</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>494_11_NoIndex_L003_R1_</th>\n",
        "      <td>44,595,824</td>\n",
        "      <td> 30,145,671</td>\n",
        "      <td> 15,628,433</td>\n",
        "      <td> 25.41%</td>\n",
        "      <td>   0.3240248</td>\n",
        "      <td>    0.5184304</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>494_5_NoIndex_L005_R1_</th>\n",
        "      <td>65,928,633</td>\n",
        "      <td> 36,667,456</td>\n",
        "      <td> 20,789,414</td>\n",
        "      <td> 17.67%</td>\n",
        "      <td>   0.4438311</td>\n",
        "      <td>    0.5669718</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR944653</th>\n",
        "      <td>29,565,498</td>\n",
        "      <td> 29,150,354</td>\n",
        "      <td>     86,793</td>\n",
        "      <td> 99.58%</td>\n",
        "      <td>   0.0140415</td>\n",
        "      <td>  0.002977425</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR944654</th>\n",
        "      <td>62,803,902</td>\n",
        "      <td> 62,367,355</td>\n",
        "      <td>  6,615,231</td>\n",
        "      <td> 88.15%</td>\n",
        "      <td> 0.006950953</td>\n",
        "      <td>    0.1060688</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR944655</th>\n",
        "      <td>26,371,551</td>\n",
        "      <td> 26,063,684</td>\n",
        "      <td>     69,038</td>\n",
        "      <td> 99.62%</td>\n",
        "      <td>  0.01167421</td>\n",
        "      <td>   0.00264882</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR944656</th>\n",
        "      <td>35,941,531</td>\n",
        "      <td> 35,314,908</td>\n",
        "      <td> 30,305,482</td>\n",
        "      <td>  8.18%</td>\n",
        "      <td>  0.01743451</td>\n",
        "      <td>    0.8581498</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR944657</th>\n",
        "      <td>27,324,308</td>\n",
        "      <td> 27,264,470</td>\n",
        "      <td>  1,506,372</td>\n",
        "      <td> 93.21%</td>\n",
        "      <td> 0.002189918</td>\n",
        "      <td>   0.05525037</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR944658</th>\n",
        "      <td>33,031,380</td>\n",
        "      <td> 32,778,204</td>\n",
        "      <td>  3,014,312</td>\n",
        "      <td> 89.78%</td>\n",
        "      <td> 0.007664712</td>\n",
        "      <td>   0.09196087</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR944659</th>\n",
        "      <td>36,965,794</td>\n",
        "      <td> 36,093,673</td>\n",
        "      <td>     31,971</td>\n",
        "      <td> 99.87%</td>\n",
        "      <td>  0.02359265</td>\n",
        "      <td> 0.0008857785</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR944660</th>\n",
        "      <td>57,833,548</td>\n",
        "      <td> 57,609,636</td>\n",
        "      <td>  4,497,568</td>\n",
        "      <td> 90.58%</td>\n",
        "      <td> 0.003871663</td>\n",
        "      <td>   0.07806972</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR944661</th>\n",
        "      <td>29,419,545</td>\n",
        "      <td> 29,280,480</td>\n",
        "      <td>  2,018,046</td>\n",
        "      <td> 92.34%</td>\n",
        "      <td>  0.00472696</td>\n",
        "      <td>   0.06892121</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR944662</th>\n",
        "      <td>39,628,591</td>\n",
        "      <td> 38,825,620</td>\n",
        "      <td>     50,653</td>\n",
        "      <td> 99.83%</td>\n",
        "      <td>  0.02026242</td>\n",
        "      <td>  0.001304628</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 7,
       "text": [
        "<IPython.core.display.HTML at 0x2058c910>"
       ]
      }
     ],
     "prompt_number": 7
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "#Comparison To Public Ribo-seq\n",
      "Gold Standard Libraries have ~15m reads in a sequencing Run, however published data only has between 2-6m reads  \n",
      "Don't know what a good depth is other than gold standard  "
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "HTML(rnaseq_df[['Processed reads',  \n",
      "                  'Number of input reads', \n",
      "                  'Uniquely mapped reads number', \n",
      "                  '% of reads unmapped: too short', \n",
      "                  'percent_repetative',\n",
      "                  'percent_mapped',\n",
      "                  ]].to_html(formatters={\"Number of input reads\" : parsers.commas,\n",
      "                             \"Uniquely mapped reads number\" : parsers.commas,\n",
      "                             \"Usable Reads\" : parsers.commas,\n",
      "                             \"Num Peaks\": parsers.commas,\n",
      "                             \"Trimmed reads\" : parsers.commas,\n",
      "                             \"Processed reads\" : parsers.commas,\n",
      "                             \"Too short reads\" : parsers.commas,\n",
      "                             \"percent_repetative\" :  percent,\n",
      "                             \"percent_mapped\" : percent,\n",
      "                             }))"
     ],
     "language": "python",
     "metadata": {
      "slideshow": {
       "slide_type": "slide"
      }
     },
     "outputs": [
      {
       "html": [
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>Processed reads</th>\n",
        "      <th>Number of input reads</th>\n",
        "      <th>Uniquely mapped reads number</th>\n",
        "      <th>% of reads unmapped: too short</th>\n",
        "      <th>percent_repetative</th>\n",
        "      <th>percent_mapped</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>ES_1_ATCACG_L001_R1</th>\n",
        "      <td> 2,903,303</td>\n",
        "      <td>    951,663</td>\n",
        "      <td>    467,376</td>\n",
        "      <td> 24.12%</td>\n",
        "      <td> 0.6722137</td>\n",
        "      <td>  0.491115</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ES_2_CGATGT_L001_R1</th>\n",
        "      <td>44,534,104</td>\n",
        "      <td> 22,628,405</td>\n",
        "      <td>  7,979,161</td>\n",
        "      <td> 20.65%</td>\n",
        "      <td> 0.4918859</td>\n",
        "      <td>  0.352617</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NPC_1_CAGATC_L002_R1</th>\n",
        "      <td>47,535,698</td>\n",
        "      <td> 25,077,341</td>\n",
        "      <td> 13,118,031</td>\n",
        "      <td> 14.89%</td>\n",
        "      <td> 0.4724525</td>\n",
        "      <td> 0.5231029</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NPC_2_ACTTGA_L002_R1</th>\n",
        "      <td>15,422,702</td>\n",
        "      <td>  8,268,158</td>\n",
        "      <td>  4,213,622</td>\n",
        "      <td> 17.83%</td>\n",
        "      <td> 0.4638969</td>\n",
        "      <td> 0.5096204</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SOX17_1_ACAGTG_L002_R1</th>\n",
        "      <td>   331,623</td>\n",
        "      <td>    232,674</td>\n",
        "      <td>     78,604</td>\n",
        "      <td> 45.11%</td>\n",
        "      <td>  0.298378</td>\n",
        "      <td> 0.3378289</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SOX17_2_GCCAAT_L002_R1</th>\n",
        "      <td> 7,290,168</td>\n",
        "      <td>  5,040,029</td>\n",
        "      <td>  1,525,599</td>\n",
        "      <td> 50.13%</td>\n",
        "      <td> 0.3086539</td>\n",
        "      <td> 0.3026965</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>T-GFP_1_TTAGGC_L001_R1</th>\n",
        "      <td>   719,609</td>\n",
        "      <td>    449,637</td>\n",
        "      <td>    181,229</td>\n",
        "      <td> 31.84%</td>\n",
        "      <td> 0.3751648</td>\n",
        "      <td> 0.4030562</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>T-GFP_2_TGACCA_L001_R1</th>\n",
        "      <td>30,073,843</td>\n",
        "      <td> 18,769,118</td>\n",
        "      <td>  6,948,339</td>\n",
        "      <td> 37.84%</td>\n",
        "      <td> 0.3758989</td>\n",
        "      <td> 0.3702006</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 8,
       "text": [
        "<IPython.core.display.HTML at 0x1fc39090>"
       ]
      }
     ],
     "prompt_number": 8
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "30000000 - rnaseq_df['Processed reads']"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 9,
       "text": [
        "ES_1_ATCACG_L001_R1       27096697\n",
        "ES_2_CGATGT_L001_R1      -14534104\n",
        "NPC_1_CAGATC_L002_R1     -17535698\n",
        "NPC_2_ACTTGA_L002_R1      14577298\n",
        "SOX17_1_ACAGTG_L002_R1    29668377\n",
        "SOX17_2_GCCAAT_L002_R1    22709832\n",
        "T-GFP_1_TTAGGC_L001_R1    29280391\n",
        "T-GFP_2_TGACCA_L001_R1      -73843\n",
        "Name: Processed reads, dtype: int64"
       ]
      }
     ],
     "prompt_number": 9
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "1178977 / float(2903303)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 22,
       "text": [
        "0.40608128052772996"
       ]
      }
     ],
     "prompt_number": 22
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "float(rnaseq_df.sum()[\"Uniquely mapped reads number\"]) / float(rnaseq_df.sum()[\"Processed reads\"])"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 20,
       "text": [
        "0.23191799936899848"
       ]
      }
     ],
     "prompt_number": 20
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "#RNA-seq Libraries\n",
      "Working Libraries\n",
      "---------------\n",
      "+ ES_2_CGATGT_L001_R1\n",
      "+ NPC_1_CAGATC_L002_R1\n",
      "+ T-GFP_2_TGACCA_L001_R1\n",
      "\n",
      "Borderline Libraries\n",
      "------------------\n",
      "+ NPC_2_ACTTGA_L002_R1\n",
      "\n",
      "Libraries to be Resequenced\n",
      "------------------------\n",
      "+ SOX17_1_ACAGTG_L002_R1\n",
      "+ SOX17_2_GCCAAT_L002_R1\n",
      "+ T-GFP_1_TTAGGC_L001_R1\n",
      "\n",
      "Looking at percent mapped and precent repetative stats it looks like the lowly sequenced libraries were just miss-quantified because the have similar mapped and repetative removal stats"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {
      "slideshow": {
       "slide_type": "slide"
      }
     },
     "source": [
      "#Control Read Contamanation Detection"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {
      "slideshow": {
       "slide_type": "skip"
      }
     },
     "source": [
      "On TSCC data:  \n",
      "!zgrep --color GTACACGGAGTCAAGACCCGCAACGC ES_1_GATCAG_L006_R1.fastq.gz | wc -l -> 3623306 + 6509135  / 12250315  \n",
      "!zgrep --color GTACACGGAGTCAAGACCCGCAACGC ES_2_ACTTGA_L006_R1.fastq.gz | wc -l -> 16111 + 23509    / 17261048  \n",
      "!zgrep --color GTACACGGAGTCAAGACCCGCAACGC ES_3_CAGATC_L006_R1.fastq.gz  | wc -l -> 8232 + 14754   / 104760758  \n",
      "!zgrep --color GTACACGGAGTCAAGACCCGCAACGC NPC_1_CGATGT_L005_R1.fastq.gz | wc -l -> 3537 + 2506    / 2321435  \n",
      "!zgrep --color GTACACGGAGTCAAGACCCGCAACGC NPC_2_ATCACG_L005_R1.fastq.gz | wc -l -> 291960 + 217787   / 101588130  \n",
      "!zgrep --color GTACACGGAGTCAAGACCCGCAACGC SOX17_1_TGACCA_L005_R1.fastq.gz | wc -l -> 7838 + 6171 / 2905925  \n",
      "!zgrep --color GTACACGGAGTCAAGACCCGCAACGC SOX17_2_TTAGGC_L005_R1.fastq.gz | wc -l -> 46525 + 43813 / 18112334  \n",
      "!zgrep --color GTACACGGAGTCAAGACCCGCAACGC TGFP_1_GCCAAT_L006_R1.fastq.gz | wc -l -> 24030 + 7769  / 4468035  \n",
      "!zgrep --color GTACACGGAGTCAAGACCCGCAACGC TGFP_2_ACAGTG_L005_R1.fastq.gz | wc -l -> 3859 + 4513  / 1529539  "
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "control_read_counts = pd.DataFrame({\"ES_1_GATCAG_L006_R1\" : {\"short_control\" : 3623306, \"long_control\" : 6509135, \"total\" : 12250315},\n",
      "\"ES_2_ACTTGA_L006_R1\" : {\"short_control\" : 16111, \"long_control\" : 23509 , \"total\" : 17261048},\n",
      "\"ES_3_CAGATC_L006_R1\"  : {\"short_control\" : 8232, \"long_control\" :  14754 , \"total\" : 104760758},\n",
      "\"NPC_1_CGATGT_L005_R1\" : {\"short_control\" : 3537, \"long_control\" : 2506  , \"total\" : 2321435},\n",
      "\"NPC_2_ATCACG_L005_R1\"  : {\"short_control\" : 291960 , \"long_control\" :  217787 , \"total\" : 101588130},\n",
      "\"SOX17_1_TGACCA_L005_R1\" : {\"short_control\" : 7838 , \"long_control\" : 6171, \"total\" :  2905925},\n",
      "\"SOX17_2_TTAGGC_L005_R1\" : {\"short_control\" : 46525, \"long_control\" :  43813 , \"total\" : 18112334},\n",
      "\"TGFP_1_GCCAAT_L006_R1\"  : {\"short_control\" : 24030, \"long_control\" :  7769 , \"total\" :  4468035},\n",
      "\"TGFP_2_ACAGTG_L005_R1\": { \"short_control\" : 3859, \"long_control\" :  4513 , \"total\" :  1529539},}).transpose()\n",
      "\n",
      "control_read_counts[\"total_control\"] = control_read_counts.long_control + control_read_counts.short_control\n",
      "control_read_counts[\"percent_control\"] = control_read_counts.total_control / control_read_counts.total.astype(float)"
     ],
     "language": "python",
     "metadata": {
      "slideshow": {
       "slide_type": "skip"
      }
     },
     "outputs": [],
     "prompt_number": 9
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "control_read_counts"
     ],
     "language": "python",
     "metadata": {
      "slideshow": {
       "slide_type": "slide"
      }
     },
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>long_control</th>\n",
        "      <th>short_control</th>\n",
        "      <th>total</th>\n",
        "      <th>total_control</th>\n",
        "      <th>percent_control</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>ES_1_GATCAG_L006_R1</th>\n",
        "      <td> 6509135</td>\n",
        "      <td> 3623306</td>\n",
        "      <td>  12250315</td>\n",
        "      <td> 10132441</td>\n",
        "      <td> 0.827117</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ES_2_ACTTGA_L006_R1</th>\n",
        "      <td>   23509</td>\n",
        "      <td>   16111</td>\n",
        "      <td>  17261048</td>\n",
        "      <td>    39620</td>\n",
        "      <td> 0.002295</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ES_3_CAGATC_L006_R1</th>\n",
        "      <td>   14754</td>\n",
        "      <td>    8232</td>\n",
        "      <td> 104760758</td>\n",
        "      <td>    22986</td>\n",
        "      <td> 0.000219</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NPC_1_CGATGT_L005_R1</th>\n",
        "      <td>    2506</td>\n",
        "      <td>    3537</td>\n",
        "      <td>   2321435</td>\n",
        "      <td>     6043</td>\n",
        "      <td> 0.002603</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NPC_2_ATCACG_L005_R1</th>\n",
        "      <td>  217787</td>\n",
        "      <td>  291960</td>\n",
        "      <td> 101588130</td>\n",
        "      <td>   509747</td>\n",
        "      <td> 0.005018</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SOX17_1_TGACCA_L005_R1</th>\n",
        "      <td>    6171</td>\n",
        "      <td>    7838</td>\n",
        "      <td>   2905925</td>\n",
        "      <td>    14009</td>\n",
        "      <td> 0.004821</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SOX17_2_TTAGGC_L005_R1</th>\n",
        "      <td>   43813</td>\n",
        "      <td>   46525</td>\n",
        "      <td>  18112334</td>\n",
        "      <td>    90338</td>\n",
        "      <td> 0.004988</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>TGFP_1_GCCAAT_L006_R1</th>\n",
        "      <td>    7769</td>\n",
        "      <td>   24030</td>\n",
        "      <td>   4468035</td>\n",
        "      <td>    31799</td>\n",
        "      <td> 0.007117</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>TGFP_2_ACAGTG_L005_R1</th>\n",
        "      <td>    4513</td>\n",
        "      <td>    3859</td>\n",
        "      <td>   1529539</td>\n",
        "      <td>     8372</td>\n",
        "      <td> 0.005474</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "<p>9 rows \u00d7 5 columns</p>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 10,
       "text": [
        "                        long_control  short_control      total  total_control  \\\n",
        "ES_1_GATCAG_L006_R1          6509135        3623306   12250315       10132441   \n",
        "ES_2_ACTTGA_L006_R1            23509          16111   17261048          39620   \n",
        "ES_3_CAGATC_L006_R1            14754           8232  104760758          22986   \n",
        "NPC_1_CGATGT_L005_R1            2506           3537    2321435           6043   \n",
        "NPC_2_ATCACG_L005_R1          217787         291960  101588130         509747   \n",
        "SOX17_1_TGACCA_L005_R1          6171           7838    2905925          14009   \n",
        "SOX17_2_TTAGGC_L005_R1         43813          46525   18112334          90338   \n",
        "TGFP_1_GCCAAT_L006_R1           7769          24030    4468035          31799   \n",
        "TGFP_2_ACAGTG_L005_R1           4513           3859    1529539           8372   \n",
        "\n",
        "                        percent_control  \n",
        "ES_1_GATCAG_L006_R1            0.827117  \n",
        "ES_2_ACTTGA_L006_R1            0.002295  \n",
        "ES_3_CAGATC_L006_R1            0.000219  \n",
        "NPC_1_CGATGT_L005_R1           0.002603  \n",
        "NPC_2_ATCACG_L005_R1           0.005018  \n",
        "SOX17_1_TGACCA_L005_R1         0.004821  \n",
        "SOX17_2_TTAGGC_L005_R1         0.004988  \n",
        "TGFP_1_GCCAAT_L006_R1          0.007117  \n",
        "TGFP_2_ACAGTG_L005_R1          0.005474  \n",
        "\n",
        "[9 rows x 5 columns]"
       ]
      }
     ],
     "prompt_number": 10
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Control reads come from ES_1 and do not contaminate other datasets\n",
      "=========================================================="
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "#Library Quantification"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "pd.read_excel(\"/nas3/gpratt/Dropbox/riboseq_team/Library_quantification.xls\", \"Final concentrations\", header=2)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>Sample</th>\n",
        "      <th>Unnamed: 1</th>\n",
        "      <th>Unnamed: 2</th>\n",
        "      <th>pM</th>\n",
        "      <th>SD</th>\n",
        "      <th>nM</th>\n",
        "      <th>library</th>\n",
        "      <th>ul library</th>\n",
        "      <th>qs</th>\n",
        "      <th>final vol</th>\n",
        "      <th>pM.1</th>\n",
        "      <th>SD.1</th>\n",
        "      <th>nM.1</th>\n",
        "      <th>Unnamed: 13</th>\n",
        "      <th>Flow Cell</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>0 </th>\n",
        "      <td>  1</td>\n",
        "      <td> Total</td>\n",
        "      <td>    ES_1</td>\n",
        "      <td> 96245.989800</td>\n",
        "      <td> 2735.036898</td>\n",
        "      <td> 96.245990</td>\n",
        "      <td>      NaN</td>\n",
        "      <td>  2</td>\n",
        "      <td> 46.122995</td>\n",
        "      <td> 48.122995</td>\n",
        "      <td>             36354.24</td>\n",
        "      <td> 3757.194290</td>\n",
        "      <td> 36.354239</td>\n",
        "      <td>NaN</td>\n",
        "      <td>  1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1 </th>\n",
        "      <td>  2</td>\n",
        "      <td> Total</td>\n",
        "      <td>    ES_2</td>\n",
        "      <td>  5076.855322</td>\n",
        "      <td>  362.227630</td>\n",
        "      <td>  5.076855</td>\n",
        "      <td>      NaN</td>\n",
        "      <td>  2</td>\n",
        "      <td>  0.538428</td>\n",
        "      <td>  2.538428</td>\n",
        "      <td>             21969.62</td>\n",
        "      <td>  956.497972</td>\n",
        "      <td> 21.969617</td>\n",
        "      <td>NaN</td>\n",
        "      <td>  1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>2 </th>\n",
        "      <td>  3</td>\n",
        "      <td> Total</td>\n",
        "      <td> T-GFP_1</td>\n",
        "      <td> 34433.361752</td>\n",
        "      <td> 2821.767222</td>\n",
        "      <td> 34.433362</td>\n",
        "      <td>      NaN</td>\n",
        "      <td>  2</td>\n",
        "      <td> 15.216681</td>\n",
        "      <td> 17.216681</td>\n",
        "      <td>             2091.555</td>\n",
        "      <td>  205.663720</td>\n",
        "      <td>  2.091555</td>\n",
        "      <td>NaN</td>\n",
        "      <td>  1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>3 </th>\n",
        "      <td>  4</td>\n",
        "      <td> Total</td>\n",
        "      <td> T-GFP_2</td>\n",
        "      <td>  2354.520865</td>\n",
        "      <td>  289.276921</td>\n",
        "      <td>  2.354521</td>\n",
        "      <td> 4.956886</td>\n",
        "      <td>  2</td>\n",
        "      <td> -0.822740</td>\n",
        "      <td>  1.177260</td>\n",
        "      <td>             14623.55</td>\n",
        "      <td> 1652.499050</td>\n",
        "      <td> 14.623545</td>\n",
        "      <td>NaN</td>\n",
        "      <td>  1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>4 </th>\n",
        "      <td>  5</td>\n",
        "      <td> Total</td>\n",
        "      <td> SOX17_1</td>\n",
        "      <td> 68196.355623</td>\n",
        "      <td> 3709.280314</td>\n",
        "      <td> 68.196356</td>\n",
        "      <td>      NaN</td>\n",
        "      <td>  2</td>\n",
        "      <td> 32.098178</td>\n",
        "      <td> 34.098178</td>\n",
        "      <td>              1304.15</td>\n",
        "      <td>  112.562304</td>\n",
        "      <td>  1.304150</td>\n",
        "      <td>NaN</td>\n",
        "      <td>  2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>5 </th>\n",
        "      <td>  6</td>\n",
        "      <td> Total</td>\n",
        "      <td> SOX17_2</td>\n",
        "      <td> 10850.164650</td>\n",
        "      <td>  971.026288</td>\n",
        "      <td> 10.850165</td>\n",
        "      <td>      NaN</td>\n",
        "      <td>  2</td>\n",
        "      <td>  3.425082</td>\n",
        "      <td>  5.425082</td>\n",
        "      <td>             4552.182</td>\n",
        "      <td>  278.370979</td>\n",
        "      <td>  4.552182</td>\n",
        "      <td>NaN</td>\n",
        "      <td>  2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>6 </th>\n",
        "      <td>  7</td>\n",
        "      <td> Total</td>\n",
        "      <td>   NPC_1</td>\n",
        "      <td>  3103.694597</td>\n",
        "      <td>   69.407918</td>\n",
        "      <td>  3.103695</td>\n",
        "      <td> 6.506697</td>\n",
        "      <td>  2</td>\n",
        "      <td> -0.448153</td>\n",
        "      <td>  1.551847</td>\n",
        "      <td>              15171.5</td>\n",
        "      <td> 2288.150641</td>\n",
        "      <td> 15.171502</td>\n",
        "      <td>NaN</td>\n",
        "      <td>  2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>7 </th>\n",
        "      <td>  8</td>\n",
        "      <td> Total</td>\n",
        "      <td>   NPC_2</td>\n",
        "      <td>  4407.666068</td>\n",
        "      <td>  366.876394</td>\n",
        "      <td>  4.407666</td>\n",
        "      <td>      NaN</td>\n",
        "      <td>  2</td>\n",
        "      <td>  0.203833</td>\n",
        "      <td>  2.203833</td>\n",
        "      <td>              3628.47</td>\n",
        "      <td>  195.924421</td>\n",
        "      <td>  3.628470</td>\n",
        "      <td>NaN</td>\n",
        "      <td>  2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>8 </th>\n",
        "      <td>  9</td>\n",
        "      <td>  RPFS</td>\n",
        "      <td>    ES_1</td>\n",
        "      <td>  7113.684256</td>\n",
        "      <td>  579.091563</td>\n",
        "      <td>  7.113684</td>\n",
        "      <td>      NaN</td>\n",
        "      <td>  2</td>\n",
        "      <td>  1.556842</td>\n",
        "      <td>  3.556842</td>\n",
        "      <td>             15499.13</td>\n",
        "      <td>  656.211544</td>\n",
        "      <td> 15.499132</td>\n",
        "      <td>NaN</td>\n",
        "      <td>  3</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>9 </th>\n",
        "      <td> 10</td>\n",
        "      <td>  RPFS</td>\n",
        "      <td>    ES_2</td>\n",
        "      <td> 15416.205833</td>\n",
        "      <td> 1543.465009</td>\n",
        "      <td> 15.416206</td>\n",
        "      <td>      NaN</td>\n",
        "      <td>  2</td>\n",
        "      <td>  5.708103</td>\n",
        "      <td>  7.708103</td>\n",
        "      <td>             1088.654</td>\n",
        "      <td>  103.800542</td>\n",
        "      <td>  1.088654</td>\n",
        "      <td>NaN</td>\n",
        "      <td>  3</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>10</th>\n",
        "      <td> 11</td>\n",
        "      <td>  RPFS</td>\n",
        "      <td>    EX_3</td>\n",
        "      <td>  4318.307129</td>\n",
        "      <td>  325.839383</td>\n",
        "      <td>  4.318307</td>\n",
        "      <td>      NaN</td>\n",
        "      <td>  2</td>\n",
        "      <td>  0.159154</td>\n",
        "      <td>  2.159154</td>\n",
        "      <td>             1355.899</td>\n",
        "      <td>   42.407788</td>\n",
        "      <td>  1.355899</td>\n",
        "      <td>NaN</td>\n",
        "      <td>  3</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>11</th>\n",
        "      <td> 12</td>\n",
        "      <td>  RPFS</td>\n",
        "      <td> T-GFP_1</td>\n",
        "      <td> 63839.852417</td>\n",
        "      <td> 8236.382850</td>\n",
        "      <td> 63.839852</td>\n",
        "      <td>      NaN</td>\n",
        "      <td>  2</td>\n",
        "      <td> 29.919926</td>\n",
        "      <td> 31.919926</td>\n",
        "      <td>             3258.688</td>\n",
        "      <td>  147.999063</td>\n",
        "      <td>  3.258688</td>\n",
        "      <td>NaN</td>\n",
        "      <td>  3</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>12</th>\n",
        "      <td> 13</td>\n",
        "      <td>  RPFS</td>\n",
        "      <td> T-GFP_2</td>\n",
        "      <td> 20386.114582</td>\n",
        "      <td> 1611.670303</td>\n",
        "      <td> 20.386115</td>\n",
        "      <td>      NaN</td>\n",
        "      <td>  2</td>\n",
        "      <td>  8.193057</td>\n",
        "      <td> 10.193057</td>\n",
        "      <td>             756.9283</td>\n",
        "      <td>   69.739683</td>\n",
        "      <td>  0.756928</td>\n",
        "      <td>NaN</td>\n",
        "      <td>  3</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>13</th>\n",
        "      <td> 14</td>\n",
        "      <td>  RPFS</td>\n",
        "      <td> SOX17_1</td>\n",
        "      <td> 11682.315082</td>\n",
        "      <td>  669.999092</td>\n",
        "      <td> 11.682315</td>\n",
        "      <td>      NaN</td>\n",
        "      <td>  2</td>\n",
        "      <td>  3.841158</td>\n",
        "      <td>  5.841158</td>\n",
        "      <td>              1487.32</td>\n",
        "      <td>  146.989391</td>\n",
        "      <td>  1.487320</td>\n",
        "      <td>NaN</td>\n",
        "      <td>  4</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>14</th>\n",
        "      <td> 15</td>\n",
        "      <td>  RPFS</td>\n",
        "      <td> SOX17_2</td>\n",
        "      <td> 11233.670743</td>\n",
        "      <td> 1897.088625</td>\n",
        "      <td> 11.233671</td>\n",
        "      <td>      NaN</td>\n",
        "      <td>  2</td>\n",
        "      <td>  3.616835</td>\n",
        "      <td>  5.616835</td>\n",
        "      <td>             29102.18</td>\n",
        "      <td> 1897.320780</td>\n",
        "      <td> 29.102181</td>\n",
        "      <td>NaN</td>\n",
        "      <td>  4</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>15</th>\n",
        "      <td> 16</td>\n",
        "      <td>  RPFS</td>\n",
        "      <td>   NPC_1</td>\n",
        "      <td> 20081.118954</td>\n",
        "      <td> 4569.467286</td>\n",
        "      <td> 20.081119</td>\n",
        "      <td>      NaN</td>\n",
        "      <td>  2</td>\n",
        "      <td>  8.040559</td>\n",
        "      <td> 10.040559</td>\n",
        "      <td>             8542.764</td>\n",
        "      <td> 1022.370405</td>\n",
        "      <td>  8.542764</td>\n",
        "      <td>NaN</td>\n",
        "      <td>  4</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>16</th>\n",
        "      <td> 17</td>\n",
        "      <td>  RPFS</td>\n",
        "      <td>   NPC_2</td>\n",
        "      <td> 17893.231983</td>\n",
        "      <td> 2800.285732</td>\n",
        "      <td> 17.893232</td>\n",
        "      <td>      NaN</td>\n",
        "      <td>  2</td>\n",
        "      <td>  6.946616</td>\n",
        "      <td>  8.946616</td>\n",
        "      <td>             6897.314</td>\n",
        "      <td>  663.112906</td>\n",
        "      <td>  6.897314</td>\n",
        "      <td>NaN</td>\n",
        "      <td>  4</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>17</th>\n",
        "      <td>NaN</td>\n",
        "      <td>   NaN</td>\n",
        "      <td>     NaN</td>\n",
        "      <td>          NaN</td>\n",
        "      <td>         NaN</td>\n",
        "      <td>       NaN</td>\n",
        "      <td>      NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>       NaN</td>\n",
        "      <td>       NaN</td>\n",
        "      <td> qPCR efficienty: 92%</td>\n",
        "      <td>         NaN</td>\n",
        "      <td>       NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>18</th>\n",
        "      <td>NaN</td>\n",
        "      <td>   NaN</td>\n",
        "      <td>     NaN</td>\n",
        "      <td>          NaN</td>\n",
        "      <td>         NaN</td>\n",
        "      <td>       NaN</td>\n",
        "      <td>      NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>       NaN</td>\n",
        "      <td>       NaN</td>\n",
        "      <td>            R2= 0.994</td>\n",
        "      <td>         NaN</td>\n",
        "      <td>       NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "<p>19 rows \u00d7 15 columns</p>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 18,
       "text": [
        "    Sample Unnamed: 1 Unnamed: 2            pM           SD         nM  \\\n",
        "0        1      Total       ES_1  96245.989800  2735.036898  96.245990   \n",
        "1        2      Total       ES_2   5076.855322   362.227630   5.076855   \n",
        "2        3      Total    T-GFP_1  34433.361752  2821.767222  34.433362   \n",
        "3        4      Total    T-GFP_2   2354.520865   289.276921   2.354521   \n",
        "4        5      Total    SOX17_1  68196.355623  3709.280314  68.196356   \n",
        "5        6      Total    SOX17_2  10850.164650   971.026288  10.850165   \n",
        "6        7      Total      NPC_1   3103.694597    69.407918   3.103695   \n",
        "7        8      Total      NPC_2   4407.666068   366.876394   4.407666   \n",
        "8        9       RPFS       ES_1   7113.684256   579.091563   7.113684   \n",
        "9       10       RPFS       ES_2  15416.205833  1543.465009  15.416206   \n",
        "10      11       RPFS       EX_3   4318.307129   325.839383   4.318307   \n",
        "11      12       RPFS    T-GFP_1  63839.852417  8236.382850  63.839852   \n",
        "12      13       RPFS    T-GFP_2  20386.114582  1611.670303  20.386115   \n",
        "13      14       RPFS    SOX17_1  11682.315082   669.999092  11.682315   \n",
        "14      15       RPFS    SOX17_2  11233.670743  1897.088625  11.233671   \n",
        "15      16       RPFS      NPC_1  20081.118954  4569.467286  20.081119   \n",
        "16      17       RPFS      NPC_2  17893.231983  2800.285732  17.893232   \n",
        "17     NaN        NaN        NaN           NaN          NaN        NaN   \n",
        "18     NaN        NaN        NaN           NaN          NaN        NaN   \n",
        "\n",
        "     library  ul library         qs  final vol                  pM.1  \\\n",
        "0        NaN           2  46.122995  48.122995              36354.24   \n",
        "1        NaN           2   0.538428   2.538428              21969.62   \n",
        "2        NaN           2  15.216681  17.216681              2091.555   \n",
        "3   4.956886           2  -0.822740   1.177260              14623.55   \n",
        "4        NaN           2  32.098178  34.098178               1304.15   \n",
        "5        NaN           2   3.425082   5.425082              4552.182   \n",
        "6   6.506697           2  -0.448153   1.551847               15171.5   \n",
        "7        NaN           2   0.203833   2.203833               3628.47   \n",
        "8        NaN           2   1.556842   3.556842              15499.13   \n",
        "9        NaN           2   5.708103   7.708103              1088.654   \n",
        "10       NaN           2   0.159154   2.159154              1355.899   \n",
        "11       NaN           2  29.919926  31.919926              3258.688   \n",
        "12       NaN           2   8.193057  10.193057              756.9283   \n",
        "13       NaN           2   3.841158   5.841158               1487.32   \n",
        "14       NaN           2   3.616835   5.616835              29102.18   \n",
        "15       NaN           2   8.040559  10.040559              8542.764   \n",
        "16       NaN           2   6.946616   8.946616              6897.314   \n",
        "17       NaN         NaN        NaN        NaN  qPCR efficienty: 92%   \n",
        "18       NaN         NaN        NaN        NaN             R2= 0.994   \n",
        "\n",
        "           SD.1       nM.1  Unnamed: 13  Flow Cell  \n",
        "0   3757.194290  36.354239          NaN          1  \n",
        "1    956.497972  21.969617          NaN          1  \n",
        "2    205.663720   2.091555          NaN          1  \n",
        "3   1652.499050  14.623545          NaN          1  \n",
        "4    112.562304   1.304150          NaN          2  \n",
        "5    278.370979   4.552182          NaN          2  \n",
        "6   2288.150641  15.171502          NaN          2  \n",
        "7    195.924421   3.628470          NaN          2  \n",
        "8    656.211544  15.499132          NaN          3  \n",
        "9    103.800542   1.088654          NaN          3  \n",
        "10    42.407788   1.355899          NaN          3  \n",
        "11   147.999063   3.258688          NaN          3  \n",
        "12    69.739683   0.756928          NaN          3  \n",
        "13   146.989391   1.487320          NaN          4  \n",
        "14  1897.320780  29.102181          NaN          4  \n",
        "15  1022.370405   8.542764          NaN          4  \n",
        "16   663.112906   6.897314          NaN          4  \n",
        "17          NaN        NaN          NaN        NaN  \n",
        "18          NaN        NaN          NaN        NaN  \n",
        "\n",
        "[19 rows x 15 columns]"
       ]
      }
     ],
     "prompt_number": 18
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "pd.read_excel?"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 11
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "#Where we stand\n",
      "+ One fully working ES and NPC pair of Ribo-seq / RNA-seq data\n",
      "+ Other datasets will need to be resequenced before drawing strong conclusions\n",
      "+ 3 Base pair perodicity observed"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "#Next Steps (Computational)\n",
      "Done\n",
      "----\n",
      "+ Computational QC metrics (% binding in genic regions)\n",
      "+ Prototype Translational Efficency Code\n",
      "+ Litature Review for Sequencing Depth of Riboseq -- done 50-100m reads, ~15m mapped usable\n",
      "+ Check if 28s and 5s is present in repDB  -- repbase contains 28s 18s and 5s rRNA, the 5.8S rRNA doesn't appear to be present, didn't check for mt rRNA, lots of rRNAs and tRNAs missed\n",
      "\n",
      "In Progress\n",
      "----------\n",
      "+ Get tRNA database map against them\n",
      "+ Run Ignolia riboseq data\n",
      "+ Identify ribosome pause sites\n",
      "\n",
      "Not Started\n",
      "----------\n",
      "+ Long noncoding RNA check -- need to re-read paper\n",
      "+ pull out rRNA aligning reads and see where they actually map\n",
      "+ Look for mtRNAs\n",
      "+ Re-process Reid and Nichata data\n",
      "\n",
      "3' ORF 5' ORF\n",
      "Translational Frame shifting\n",
      "Codon bias for pausing \n",
      "\n",
      "Look at erics motif data, look and stratify potentally localized mRNAs\n",
      "\n",
      "for CLIP motifs and tranlsation and see if we can seperate TE by those motifs\n",
      "\n",
      "Look at bivalent genes, if there is translational poising \n",
      "\n",
      "Start sites\n",
      "\n",
      "alt 3' ORFs\n",
      "\n",
      "alt 5' start sites...\n",
      "\n",
      "uORFs\n",
      "\n",
      "long non-coding RNA analysis\n",
      "\n",
      "are alt spliced exons getting tranlated?\n",
      "\n",
      "relationship between RO and alt splicing\n",
      "#Next Steps (Biological)\n",
      "+ Resequence\n",
      "+ Figure out why Mesoderm Samples don't appear to be differentiated -- early time point, stuff isn't full down regulated\n",
      "+ Figure out re-do ES_2 because it looks like the riboseq pulldown failled\n",
      "+ Now that we know art-seq works would it be possible to add randomers?  I've found some things that already look like PCR duplicates, and this would give me confidence in peak finding / quantification"
     ]
    }
   ],
   "metadata": {}
  }
 ]
}