{
 "metadata": {
  "name": ""
 },
 "nbformat": 3,
 "nbformat_minor": 0,
 "worksheets": [
  {
   "cells": [
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Initilizaing Analysis \n",
      "-----------\n",
      "\n",
      "Ignore"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "from IPython.core.display import HTML\n",
      "\n",
      "from gscripts.general import parsers"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 7
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "qc_metrics = parsers.clipseq_metrics(\"/nas3/gpratt/projects/tiffany_genentech/analysis/v1/\", iclip=True)\n",
      "qc_metrics = qc_metrics.rename(columns={\"Processed bases\" : \"Input Bases\",\n",
      "           \"Processed reads\" : \"Input Reads\",\n",
      "           \"Number of input reads\" : \"Reads after Quality Filtering\"\n",
      "           })"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 10
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "#Raw QC"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "HTML(qc_metrics.to_html())"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>Processed bases</th>\n",
        "      <th>Processed reads</th>\n",
        "      <th>Quality-trimmed</th>\n",
        "      <th>Too long reads</th>\n",
        "      <th>Too short reads</th>\n",
        "      <th>Trimmed bases</th>\n",
        "      <th>Trimmed reads</th>\n",
        "      <th>% of reads mapped to multiple loci</th>\n",
        "      <th>% of reads mapped to too many loci</th>\n",
        "      <th>% of reads unmapped: other</th>\n",
        "      <th>% of reads unmapped: too many mismatches</th>\n",
        "      <th>% of reads unmapped: too short</th>\n",
        "      <th>Average input read length</th>\n",
        "      <th>Average mapped length</th>\n",
        "      <th>Deletion average length</th>\n",
        "      <th>Deletion rate per base</th>\n",
        "      <th>Finished on</th>\n",
        "      <th>Insertion average length</th>\n",
        "      <th>Insertion rate per base</th>\n",
        "      <th>Mapping speed, Million of reads per hour</th>\n",
        "      <th>Mismatch rate per base, percent</th>\n",
        "      <th>Number of input reads</th>\n",
        "      <th>Number of reads mapped to multiple loci</th>\n",
        "      <th>Number of reads mapped to too many loci</th>\n",
        "      <th>Number of splices: AT/AC</th>\n",
        "      <th>Number of splices: Annotated (sjdb)</th>\n",
        "      <th>Number of splices: GC/AG</th>\n",
        "      <th>Number of splices: GT/AG</th>\n",
        "      <th>Number of splices: Non-canonical</th>\n",
        "      <th>Number of splices: Total</th>\n",
        "      <th>Started job on</th>\n",
        "      <th>Started mapping on</th>\n",
        "      <th>Uniquely mapped reads %</th>\n",
        "      <th>Uniquely mapped reads number</th>\n",
        "      <th>NRF</th>\n",
        "      <th>PCB</th>\n",
        "      <th>Usable Reads</th>\n",
        "      <th>removed_count</th>\n",
        "      <th>total_count</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>LIB19982_SAM18146099_R1.R02</th>\n",
        "      <td> 289550323</td>\n",
        "      <td> 7062203</td>\n",
        "      <td> 223318</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1024</td>\n",
        "      <td>  188140</td>\n",
        "      <td> 13263</td>\n",
        "      <td> 0.00%</td>\n",
        "      <td> 41.27%</td>\n",
        "      <td> 0.00%</td>\n",
        "      <td> 0.00%</td>\n",
        "      <td> 47.41%</td>\n",
        "      <td> 40</td>\n",
        "      <td> 33.05</td>\n",
        "      <td> 1.53</td>\n",
        "      <td> 0.05%</td>\n",
        "      <td> Apr 08 16:17:42</td>\n",
        "      <td> 1.09</td>\n",
        "      <td> 0.00%</td>\n",
        "      <td> 784.78</td>\n",
        "      <td> 1.77%</td>\n",
        "      <td> 6539872</td>\n",
        "      <td> 0</td>\n",
        "      <td> 2699002</td>\n",
        "      <td> 1</td>\n",
        "      <td> 0</td>\n",
        "      <td> 535</td>\n",
        "      <td> 19941</td>\n",
        "      <td> 8264</td>\n",
        "      <td> 28741</td>\n",
        "      <td> Apr 08 16:16:39</td>\n",
        "      <td> Apr 08 16:17:12</td>\n",
        "      <td> 11.32%</td>\n",
        "      <td>  740466</td>\n",
        "      <td> 0.011566</td>\n",
        "      <td> 0.314106</td>\n",
        "      <td> 44267</td>\n",
        "      <td>  696199</td>\n",
        "      <td>  740466</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>LIB19982_SAM18146099_R1.R08</th>\n",
        "      <td> 229813938</td>\n",
        "      <td> 5605218</td>\n",
        "      <td> 220862</td>\n",
        "      <td> 0</td>\n",
        "      <td> 6168</td>\n",
        "      <td> 1418943</td>\n",
        "      <td> 81466</td>\n",
        "      <td> 0.00%</td>\n",
        "      <td> 42.24%</td>\n",
        "      <td> 0.07%</td>\n",
        "      <td> 0.00%</td>\n",
        "      <td> 18.41%</td>\n",
        "      <td> 40</td>\n",
        "      <td> 38.08</td>\n",
        "      <td> 1.41</td>\n",
        "      <td> 0.13%</td>\n",
        "      <td> Apr 08 16:13:17</td>\n",
        "      <td> 1.15</td>\n",
        "      <td> 0.01%</td>\n",
        "      <td> 837.75</td>\n",
        "      <td> 1.29%</td>\n",
        "      <td> 3257933</td>\n",
        "      <td> 0</td>\n",
        "      <td> 1376123</td>\n",
        "      <td> 6</td>\n",
        "      <td> 0</td>\n",
        "      <td> 305</td>\n",
        "      <td>  8930</td>\n",
        "      <td> 3659</td>\n",
        "      <td> 12900</td>\n",
        "      <td> Apr 08 16:10:13</td>\n",
        "      <td> Apr 08 16:13:03</td>\n",
        "      <td> 39.28%</td>\n",
        "      <td> 1279767</td>\n",
        "      <td> 0.028483</td>\n",
        "      <td> 0.152506</td>\n",
        "      <td> 56518</td>\n",
        "      <td> 1223249</td>\n",
        "      <td> 1279767</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>LIB19982_SAM18146099_R1.unassigned</th>\n",
        "      <td>  67410888</td>\n",
        "      <td> 1644168</td>\n",
        "      <td> 132701</td>\n",
        "      <td> 0</td>\n",
        "      <td> 7088</td>\n",
        "      <td>  602317</td>\n",
        "      <td> 24116</td>\n",
        "      <td> 0.00%</td>\n",
        "      <td> 11.07%</td>\n",
        "      <td> 0.02%</td>\n",
        "      <td> 0.00%</td>\n",
        "      <td> 71.91%</td>\n",
        "      <td> 40</td>\n",
        "      <td> 38.96</td>\n",
        "      <td> 1.48</td>\n",
        "      <td> 0.02%</td>\n",
        "      <td> Apr 08 16:15:43</td>\n",
        "      <td> 1.03</td>\n",
        "      <td> 0.00%</td>\n",
        "      <td>  15.99</td>\n",
        "      <td> 0.61%</td>\n",
        "      <td> 1532504</td>\n",
        "      <td> 0</td>\n",
        "      <td>  169702</td>\n",
        "      <td> 0</td>\n",
        "      <td> 0</td>\n",
        "      <td> 135</td>\n",
        "      <td>  7868</td>\n",
        "      <td>  448</td>\n",
        "      <td>  8451</td>\n",
        "      <td> Apr 08 16:07:38</td>\n",
        "      <td> Apr 08 16:09:58</td>\n",
        "      <td> 17.00%</td>\n",
        "      <td>  260540</td>\n",
        "      <td> 0.105189</td>\n",
        "      <td> 0.402722</td>\n",
        "      <td> 34521</td>\n",
        "      <td>  226019</td>\n",
        "      <td>  260540</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 8,
       "text": [
        "<IPython.core.display.HTML at 0x84c41d0>"
       ]
      }
     ],
     "prompt_number": 8
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "#Important QC Metrics"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "qc_metrics[\"Reads After Triming\"] = qc_metrics['Input Reads'] - qc_metrics['Too short reads']\n",
      "\n",
      "filtered_df = qc_metrics[[ \"Input Reads\",\n",
      "                            \"Reads After Triming\",\n",
      "                            \"Reads after Quality Filtering\", \n",
      "                            \"Uniquely mapped reads number\",\n",
      "                            #\"Uniquely mapped reads %\",\n",
      "                            \"Usable Reads\",\n",
      "                            #\"spot\",\n",
      "                            #\"Num Peaks\"\n",
      "                            ]]\n",
      "\n",
      "filtered_df['Percent Usable'] = (filtered_df['Usable Reads'] / filtered_df['Uniquely mapped reads number']) * 100\n",
      "HTML(filtered_df.to_html(formatters={\"Input Reads\" : parsers.commas,\n",
      "                                     \"Reads After Triming\" : parsers.commas,\n",
      "                                     \"Reads after Quality Filtering\" : parsers.commas,\n",
      "                                     \"Uniquely mapped reads number\" : parsers.commas,\n",
      "                                     \"Usable Reads\" : parsers.commas,\n",
      "                                     \"Num Peaks\": parsers.commas\n",
      "                                     } ))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>Input Reads</th>\n",
        "      <th>Reads After Triming</th>\n",
        "      <th>Reads after Quality Filtering</th>\n",
        "      <th>Uniquely mapped reads number</th>\n",
        "      <th>Usable Reads</th>\n",
        "      <th>Percent Usable</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>LIB19982_SAM18146099_R1.R02</th>\n",
        "      <td>7,062,203</td>\n",
        "      <td>7,061,179</td>\n",
        "      <td> 6,539,872</td>\n",
        "      <td>   740,466</td>\n",
        "      <td>44,267</td>\n",
        "      <td> 5.978262</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>LIB19982_SAM18146099_R1.R08</th>\n",
        "      <td>5,605,218</td>\n",
        "      <td>5,599,050</td>\n",
        "      <td> 3,257,933</td>\n",
        "      <td> 1,279,767</td>\n",
        "      <td>56,518</td>\n",
        "      <td> 4.416273</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>LIB19982_SAM18146099_R1.unassigned</th>\n",
        "      <td>1,644,168</td>\n",
        "      <td>1,637,080</td>\n",
        "      <td> 1,532,504</td>\n",
        "      <td>   260,540</td>\n",
        "      <td>34,521</td>\n",
        "      <td> 13.24979</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 18,
       "text": [
        "<IPython.core.display.HTML at 0x86ffe10>"
       ]
      }
     ],
     "prompt_number": 18
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Analysis\n",
      "--------\n",
      "\n",
      "The number of usable (randomer collapsed) reads from this analysis is too low to confidently draw genome wide conclusions.  The Precent Usable Reads is decent when compared to internal datasets from the Yeo Lab.  We see a range of 1%-10% on average.   \n",
      "\n",
      "I recommend much deeper sequencing.  Enough to get up to ~1m usable reads.  For R08 this means sequencing at least 23,000,000 reads and 17,000,000 for R02"
     ]
    }
   ],
   "metadata": {}
  }
 ]
}