{
 "metadata": {
  "name": ""
 },
 "nbformat": 3,
 "nbformat_minor": 0,
 "worksheets": [
  {
   "cells": [
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "import os\n",
      "import glob\n",
      "\n",
      "from collections import defaultdict\n",
      "import gffutils\n",
      "import pandas as pd\n",
      "import numpy as np\n",
      "import numpy\n",
      "\n",
      "from collections import defaultdict\n",
      "import glob\n",
      "import itertools\n",
      "import os\n",
      "\n",
      "from bx.bbi.bigwig_file import BigWigFile\n",
      "from IPython.core.display import HTML\n",
      "from IPython.parallel import Client\n",
      "import pandas as pd\n",
      "import pybedtools\n",
      "from pybedtools.featurefuncs import midpoint\n",
      "from pybedtools import MalformedBedLineError\n",
      "import sqlalchemy\n",
      "import seaborn as sns\n",
      "\n",
      "\n",
      "from gscripts.general import parsers\n",
      "#from gscripts.general import dataviz\n",
      "from gscripts.general.pybedtools_helpers import small_peaks\n",
      "\n",
      "img_dir = \"/nas3/gpratt/Dropbox/Presentations/notebooks/CLIPper_peak_overlap\"\n",
      "analysis_dir = \"/nas3/gpratt/projects/public_clip/analysis/v5\"\n",
      "\n",
      "#engine = sqlalchemy.create_engine('mysql+pymysql://ppliu@sauron.ucsd.edu/rbpdb')\n",
      "#connection = engine.connect()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 1
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "human_db = \"/nas3/gpratt/gencode/gencode.v17.annotation.gtf.db\"\n",
      "db = gffutils.FeatureDB(human_db)\n",
      "genes = db.features_of_type('gene')\n",
      "\n",
      "gene_type_dict = {}\n",
      "gene_name_dict = {}\n",
      "for gene in genes:\n",
      "    try:\n",
      "        gene_type_dict[gene.attributes['gene_id'][0]] = gene.attributes['gene_type'][0]\n",
      "    except KeyError:\n",
      "        pass\n",
      "    \n",
      "    try:\n",
      "        gene_name_dict[gene.attributes['gene_id'][0]] = gene.attributes['gene_name'][0]\n",
      "    except KeyError:\n",
      "        pass"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 2
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "cisbp_db = pd.read_csv(\"/nas3/gpratt/projects/cisbp/analysis/RBP_Information.txt\", sep=\"\\t\")\n",
      " \n",
      "public_clip_list = pd.read_csv(\"/nas3/gpratt/projects/public_clip/v1/public_clip_list.csv\", index_col=0)\n",
      "public_clip_list= public_clip_list[public_clip_list['CLIP-seq?'] == \"Yes\"]\n",
      "public_clip_list['ensembl'] = public_clip_list.Ensemblid.astype(str).apply(lambda x: x.split(\".\")[0])\n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "ename": "AttributeError",
       "evalue": "'DataFrame' object has no attribute 'Ensemblid'",
       "output_type": "pyerr",
       "traceback": [
        "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n\u001b[1;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
        "\u001b[1;32m<ipython-input-3-b8a0c842ff23>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[0;32m      3\u001b[0m \u001b[0mpublic_clip_list\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mread_csv\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"/nas3/gpratt/projects/public_clip/v1/public_clip_list.csv\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mindex_col\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      4\u001b[0m \u001b[0mpublic_clip_list\u001b[0m\u001b[1;33m=\u001b[0m \u001b[0mpublic_clip_list\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mpublic_clip_list\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'CLIP-seq?'\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m==\u001b[0m \u001b[1;34m\"Yes\"\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 5\u001b[1;33m \u001b[0mpublic_clip_list\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'ensembl'\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mpublic_clip_list\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mEnsemblid\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mastype\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mstr\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mapply\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;32mlambda\u001b[0m \u001b[0mx\u001b[0m\u001b[1;33m:\u001b[0m \u001b[0mx\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\".\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[1;32m/nas/nas0/gpratt/virtualenvs/default/lib/python2.7/site-packages/pandas/core/generic.pyc\u001b[0m in \u001b[0;36m__getattr__\u001b[1;34m(self, name)\u001b[0m\n\u001b[0;32m   1934\u001b[0m                 \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mname\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1935\u001b[0m             raise AttributeError(\"'%s' object has no attribute '%s'\" %\n\u001b[1;32m-> 1936\u001b[1;33m                                  (type(self).__name__, name))\n\u001b[0m\u001b[0;32m   1937\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1938\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m__setattr__\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mname\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;31mAttributeError\u001b[0m: 'DataFrame' object has no attribute 'Ensemblid'"
       ]
      }
     ],
     "prompt_number": 3
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "experiment_mapping = {\n",
      "                      \"peaks\" : \"clipper\",\n",
      "                      \"ripseeker\" : \"ripseeker\",\n",
      "                      \"pyicoclip\" : \"pyicoclip\",\n",
      "                      \"piranha\" : \"piranha\",\n",
      "                      \"bed_notrim_ingenes_clusters_hg1950\" : \"kasey\",\n",
      "                      \"bed_notrim_ingenes_clusters_mm950\"  : \"kasey\",\n",
      "                      }\n",
      "\n",
      "peak_files = glob.glob(os.path.join(analysis_dir, \"*.bed\"))\n",
      "samples = itertools.groupby(sorted(peak_files), lambda x: os.path.basename(x).split(\".\")[0])\n",
      "\n",
      "sample_dict = defaultdict(dict)\n",
      "for sample, experiments in samples:\n",
      "    for experiment in experiments:\n",
      "        try:\n",
      "            experiment_type = os.path.basename(experiment).split(\".\")[-2]\n",
      "            sample_dict[sample][experiment_mapping[experiment_type]] = experiment\n",
      "        except:\n",
      "            pass\n",
      "sample_df = pd.DataFrame(sample_dict)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 4
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def gene_name(gencode_id):\n",
      "    try:\n",
      "        return gene_name_dict[gencode_id]\n",
      "    except:\n",
      "        return gene_id\n",
      "    \n",
      "def gene_type(gencode_id):\n",
      "    try:\n",
      "        return gene_type_dict[gencode_id]\n",
      "    except:\n",
      "        return gene_id"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 5
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "riboseq_files = glob.glob(\"/nas3/gpratt/projects/public_riboseq/analysis/public_riboseq_bam_v3/*.rpkm\")\n",
      "riboseq_dict = {\".\".join(os.path.basename(fn).split(\".\")[:1]).replace(\"-\", \"\") : pd.read_csv(fn, sep=\"\\t\", index_col=0) for fn in riboseq_files}\n",
      "riboseq = pd.concat(riboseq_dict).drop(\"flag\", axis=1)\n",
      "\n",
      "rnaseq_files = glob.glob(\"/nas3/gpratt/projects/public_riboseq/analysis/public_rnaseq_bam_v3/*.rpkm\")\n",
      "rnaseq_dict = {\".\".join(os.path.basename(fn).split(\".\")[:1]).replace(\"-\", \"\") : pd.read_csv(fn, sep=\"\\t\", index_col=0) for fn in rnaseq_files}\n",
      "rnaseq = pd.concat(rnaseq_dict).drop(\"flag\", axis=1)\n",
      "\n",
      "riboseq[\"gene_name\"] = [gene_name(gene_id) for gene_id in riboseq.index.get_level_values(1)]\n",
      "riboseq[\"gene_type\"] = [gene_type(gene_id) for gene_id in riboseq.index.get_level_values(1)]\n",
      "riboseq = riboseq.rename({\"SRR334701\" : \"ER\",\n",
      "                \"SRR334703\" : \"Cytosol\"})\n",
      "\n",
      "rnaseq[\"gene_name\"] = [gene_name(gene_id) for gene_id in rnaseq.index.get_level_values(1)]\n",
      "rnaseq[\"gene_type\"] = [gene_type(gene_id) for gene_id in rnaseq.index.get_level_values(1)]\n",
      "\n",
      "rnaseq = rnaseq.rename({\"SRR334700\" : \"ER\",\n",
      "                \"SRR334702\" : \"Cytosol\"})"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 6
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "loglog(rnaseq.xs(\"ER\").RPKM, riboseq.xs(\"ER\").RPKM, 'o', alpha=.1)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "ename": "NameError",
       "evalue": "name 'loglog' is not defined",
       "output_type": "pyerr",
       "traceback": [
        "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[1;32m<ipython-input-7-c5cfbbffcfb4>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mloglog\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mrnaseq\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mxs\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"ER\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mRPKM\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mriboseq\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mxs\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"ER\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mRPKM\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'o'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0malpha\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m.1\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[1;31mNameError\u001b[0m: name 'loglog' is not defined"
       ]
      }
     ],
     "prompt_number": 7
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "loglog(rnaseq.xs(\"Cytosol\").RPKM, riboseq.xs(\"Cytosol\").RPKM, 'o', alpha=.1)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Correlated Enough, I'll use the RNA-seq to define a set of well expressed genes to do translation analysis on.  "
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "#Define set of well expressed genes"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Well expressed is defined in http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3225288/#!po=61.1111 as median footprint coverage excluding 15bp after the start codon and 5bp before the stop codon.  I'll need to re-write read counter to take this into account.  I think the best way is to modify the cds file for reads start and stop cds regions and adjust their size.  "
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "riboseq[(riboseq.RPKM > 30) & (riboseq.gene_type == \"protein_coding\")].xs(\"ER\").sort(\"RPKM\")"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>RPKM</th>\n",
        "      <th>gene_name</th>\n",
        "      <th>gene_type</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>gene</th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>ENSG00000198755.6</th>\n",
        "      <td> 30.036387</td>\n",
        "      <td>         RPL10A</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000104824.11</th>\n",
        "      <td> 30.065847</td>\n",
        "      <td>         HNRNPL</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000229833.4</th>\n",
        "      <td> 30.151987</td>\n",
        "      <td>         PET100</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000134755.10</th>\n",
        "      <td> 30.190126</td>\n",
        "      <td>           DSC2</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000110107.4</th>\n",
        "      <td> 30.198819</td>\n",
        "      <td>         PRPF19</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000115268.5</th>\n",
        "      <td> 30.206131</td>\n",
        "      <td>          RPS15</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000169020.5</th>\n",
        "      <td> 30.219733</td>\n",
        "      <td>          ATP5I</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000151465.9</th>\n",
        "      <td> 30.231000</td>\n",
        "      <td>         CDC123</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000155876.4</th>\n",
        "      <td> 30.254298</td>\n",
        "      <td>          RRAGA</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000138085.11</th>\n",
        "      <td> 30.256726</td>\n",
        "      <td>         ATRAID</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000099995.13</th>\n",
        "      <td> 30.265721</td>\n",
        "      <td>          SF3A1</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000120053.9</th>\n",
        "      <td> 30.364210</td>\n",
        "      <td>           GOT1</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000183527.7</th>\n",
        "      <td> 30.375693</td>\n",
        "      <td>          PSMG1</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000172201.6</th>\n",
        "      <td> 30.376883</td>\n",
        "      <td>            ID4</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000114686.4</th>\n",
        "      <td> 30.406715</td>\n",
        "      <td>          MRPL3</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000090432.5</th>\n",
        "      <td> 30.412271</td>\n",
        "      <td>           MUL1</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000014123.9</th>\n",
        "      <td> 30.454129</td>\n",
        "      <td>           UFL1</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000141543.5</th>\n",
        "      <td> 30.596873</td>\n",
        "      <td>         EIF4A3</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000178982.5</th>\n",
        "      <td> 30.605939</td>\n",
        "      <td>          EIF3K</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000183337.12</th>\n",
        "      <td> 30.610575</td>\n",
        "      <td>           BCOR</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000182054.5</th>\n",
        "      <td> 30.643377</td>\n",
        "      <td>           IDH2</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000101444.7</th>\n",
        "      <td> 30.651087</td>\n",
        "      <td>           AHCY</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000197498.7</th>\n",
        "      <td> 30.659491</td>\n",
        "      <td>           RPF2</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000089248.6</th>\n",
        "      <td> 30.680204</td>\n",
        "      <td>          ERP29</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000144028.10</th>\n",
        "      <td> 30.692755</td>\n",
        "      <td>       SNRNP200</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000000003.10</th>\n",
        "      <td> 30.749206</td>\n",
        "      <td>         TSPAN6</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000182154.7</th>\n",
        "      <td> 30.751339</td>\n",
        "      <td>         MRPL41</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000145386.5</th>\n",
        "      <td> 30.797445</td>\n",
        "      <td>          CCNA2</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000087460.18</th>\n",
        "      <td> 30.822850</td>\n",
        "      <td>           GNAS</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000164366.2</th>\n",
        "      <td> 30.846049</td>\n",
        "      <td>        CCDC127</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000143153.7</th>\n",
        "      <td> 30.951715</td>\n",
        "      <td>         ATP1B1</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000094804.5</th>\n",
        "      <td> 30.977038</td>\n",
        "      <td>           CDC6</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000183048.7</th>\n",
        "      <td> 31.009194</td>\n",
        "      <td>       SLC25A10</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000090863.7</th>\n",
        "      <td> 31.023871</td>\n",
        "      <td>           GLG1</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000145741.11</th>\n",
        "      <td> 31.106795</td>\n",
        "      <td>           BTF3</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000108055.9</th>\n",
        "      <td> 31.196345</td>\n",
        "      <td>           SMC3</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000076716.7</th>\n",
        "      <td> 31.198360</td>\n",
        "      <td>           GPC4</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000266953.1</th>\n",
        "      <td> 31.206850</td>\n",
        "      <td>  RP11-618P17.4</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000101421.3</th>\n",
        "      <td> 31.213672</td>\n",
        "      <td>         CHMP4B</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000179271.1</th>\n",
        "      <td> 31.248454</td>\n",
        "      <td>     GADD45GIP1</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000091483.6</th>\n",
        "      <td> 31.298428</td>\n",
        "      <td>             FH</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000143337.12</th>\n",
        "      <td> 31.300865</td>\n",
        "      <td>       TOR1AIP1</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000135446.12</th>\n",
        "      <td> 31.329108</td>\n",
        "      <td>           CDK4</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000124766.4</th>\n",
        "      <td> 31.343941</td>\n",
        "      <td>           SOX4</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000263809.1</th>\n",
        "      <td> 31.345694</td>\n",
        "      <td>          RPL26</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000054118.9</th>\n",
        "      <td> 31.384905</td>\n",
        "      <td>         THRAP3</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000135052.12</th>\n",
        "      <td> 31.399207</td>\n",
        "      <td>          GOLM1</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000189043.5</th>\n",
        "      <td> 31.534801</td>\n",
        "      <td>         NDUFA4</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000123595.5</th>\n",
        "      <td> 31.541825</td>\n",
        "      <td>          RAB9A</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000006125.12</th>\n",
        "      <td> 31.557887</td>\n",
        "      <td>          AP2B1</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000256646.3</th>\n",
        "      <td> 31.645788</td>\n",
        "      <td> DKFZP686D09174</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000131238.11</th>\n",
        "      <td> 31.657148</td>\n",
        "      <td>           PPT1</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000166441.8</th>\n",
        "      <td> 31.735669</td>\n",
        "      <td>         RPL27A</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000226784.1</th>\n",
        "      <td> 31.749860</td>\n",
        "      <td>          PGAM4</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000254692.1</th>\n",
        "      <td> 31.778749</td>\n",
        "      <td>         TM9SF1</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000089157.11</th>\n",
        "      <td> 31.854211</td>\n",
        "      <td>          RPLP0</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000147383.6</th>\n",
        "      <td> 31.868288</td>\n",
        "      <td>          NSDHL</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000198830.6</th>\n",
        "      <td> 31.910309</td>\n",
        "      <td>          HMGN2</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000149428.14</th>\n",
        "      <td> 31.943861</td>\n",
        "      <td>          HYOU1</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ENSG00000253729.3</th>\n",
        "      <td> 31.944225</td>\n",
        "      <td>          PRKDC</td>\n",
        "      <td> protein_coding</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th></th>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "<p>804 rows \u00d7 3 columns</p>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 11,
       "text": [
        "                         RPKM       gene_name       gene_type\n",
        "gene                                                         \n",
        "ENSG00000198755.6   30.036387          RPL10A  protein_coding\n",
        "ENSG00000104824.11  30.065847          HNRNPL  protein_coding\n",
        "ENSG00000229833.4   30.151987          PET100  protein_coding\n",
        "ENSG00000134755.10  30.190126            DSC2  protein_coding\n",
        "ENSG00000110107.4   30.198819          PRPF19  protein_coding\n",
        "ENSG00000115268.5   30.206131           RPS15  protein_coding\n",
        "ENSG00000169020.5   30.219733           ATP5I  protein_coding\n",
        "ENSG00000151465.9   30.231000          CDC123  protein_coding\n",
        "ENSG00000155876.4   30.254298           RRAGA  protein_coding\n",
        "ENSG00000138085.11  30.256726          ATRAID  protein_coding\n",
        "ENSG00000099995.13  30.265721           SF3A1  protein_coding\n",
        "ENSG00000120053.9   30.364210            GOT1  protein_coding\n",
        "ENSG00000183527.7   30.375693           PSMG1  protein_coding\n",
        "ENSG00000172201.6   30.376883             ID4  protein_coding\n",
        "ENSG00000114686.4   30.406715           MRPL3  protein_coding\n",
        "ENSG00000090432.5   30.412271            MUL1  protein_coding\n",
        "ENSG00000014123.9   30.454129            UFL1  protein_coding\n",
        "ENSG00000141543.5   30.596873          EIF4A3  protein_coding\n",
        "ENSG00000178982.5   30.605939           EIF3K  protein_coding\n",
        "ENSG00000183337.12  30.610575            BCOR  protein_coding\n",
        "ENSG00000182054.5   30.643377            IDH2  protein_coding\n",
        "ENSG00000101444.7   30.651087            AHCY  protein_coding\n",
        "ENSG00000197498.7   30.659491            RPF2  protein_coding\n",
        "ENSG00000089248.6   30.680204           ERP29  protein_coding\n",
        "ENSG00000144028.10  30.692755        SNRNP200  protein_coding\n",
        "ENSG00000000003.10  30.749206          TSPAN6  protein_coding\n",
        "ENSG00000182154.7   30.751339          MRPL41  protein_coding\n",
        "ENSG00000145386.5   30.797445           CCNA2  protein_coding\n",
        "ENSG00000087460.18  30.822850            GNAS  protein_coding\n",
        "ENSG00000164366.2   30.846049         CCDC127  protein_coding\n",
        "ENSG00000143153.7   30.951715          ATP1B1  protein_coding\n",
        "ENSG00000094804.5   30.977038            CDC6  protein_coding\n",
        "ENSG00000183048.7   31.009194        SLC25A10  protein_coding\n",
        "ENSG00000090863.7   31.023871            GLG1  protein_coding\n",
        "ENSG00000145741.11  31.106795            BTF3  protein_coding\n",
        "ENSG00000108055.9   31.196345            SMC3  protein_coding\n",
        "ENSG00000076716.7   31.198360            GPC4  protein_coding\n",
        "ENSG00000266953.1   31.206850   RP11-618P17.4  protein_coding\n",
        "ENSG00000101421.3   31.213672          CHMP4B  protein_coding\n",
        "ENSG00000179271.1   31.248454      GADD45GIP1  protein_coding\n",
        "ENSG00000091483.6   31.298428              FH  protein_coding\n",
        "ENSG00000143337.12  31.300865        TOR1AIP1  protein_coding\n",
        "ENSG00000135446.12  31.329108            CDK4  protein_coding\n",
        "ENSG00000124766.4   31.343941            SOX4  protein_coding\n",
        "ENSG00000263809.1   31.345694           RPL26  protein_coding\n",
        "ENSG00000054118.9   31.384905          THRAP3  protein_coding\n",
        "ENSG00000135052.12  31.399207           GOLM1  protein_coding\n",
        "ENSG00000189043.5   31.534801          NDUFA4  protein_coding\n",
        "ENSG00000123595.5   31.541825           RAB9A  protein_coding\n",
        "ENSG00000006125.12  31.557887           AP2B1  protein_coding\n",
        "ENSG00000256646.3   31.645788  DKFZP686D09174  protein_coding\n",
        "ENSG00000131238.11  31.657148            PPT1  protein_coding\n",
        "ENSG00000166441.8   31.735669          RPL27A  protein_coding\n",
        "ENSG00000226784.1   31.749860           PGAM4  protein_coding\n",
        "ENSG00000254692.1   31.778749          TM9SF1  protein_coding\n",
        "ENSG00000089157.11  31.854211           RPLP0  protein_coding\n",
        "ENSG00000147383.6   31.868288           NSDHL  protein_coding\n",
        "ENSG00000198830.6   31.910309           HMGN2  protein_coding\n",
        "ENSG00000149428.14  31.943861           HYOU1  protein_coding\n",
        "ENSG00000253729.3   31.944225           PRKDC  protein_coding\n",
        "                          ...             ...             ...\n",
        "\n",
        "[804 rows x 3 columns]"
       ]
      }
     ],
     "prompt_number": 11
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Well expressed is hard, I filtered by RPKM but on the reid data there isn't very good coverage anywhere.  Still need to validate pause sites.  Lets plot distribution of reads around peaks..."
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def get_peak_wiggle(tool, data_pos, data_neg, size = 1000, num_peaks=100):\n",
      "    \n",
      "    \"\"\"\n",
      "    \n",
      "    returns matrix of (n,25) of wiggle and peak information\n",
      "    tool - bedtool contains location of peaks\n",
      "    data - bigwig file handle - contains bw data for the peaks\n",
      "\n",
      "    \"\"\"\n",
      "    \n",
      "    results = []\n",
      "    index = []\n",
      "    for line in tool[:num_peaks]:\n",
      "        peak_center = line.start\n",
      "        pad_amt = size / 2\n",
      "        try:\n",
      "            if line.strand == \"+\":\n",
      "                result = data_pos.get_as_array(line.chrom, peak_center - pad_amt , peak_center + pad_amt)\n",
      "            if line.strand == \"-\":\n",
      "                result = data_neg.get_as_array(line.chrom, peak_center - pad_amt , peak_center + pad_amt)[::-1]\n",
      "                \n",
      "            result = [0 if np.isnan(x) else x for x in result]\n",
      "        \n",
      "            results.append(result)\n",
      "            index.append(line.name)\n",
      "            \n",
      "        except:\n",
      "            pass\n",
      "    return pd.DataFrame(results)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "pos = BigWigFile(open(\"/nas3/gpratt/projects/public_riboseq/analysis/public_riboseq_bam_v3/SRR334701.polyATrim.adapterTrim.rmRep.sorted.rg.adj.pos.bw\"))\n",
      "neg = BigWigFile(open(\"/nas3/gpratt/projects/public_riboseq/analysis/public_riboseq_bam_v3/SRR334701.polyATrim.adapterTrim.rmRep.sorted.rg.adj.neg.t.bw\"))\n",
      "bedfile = pybedtools.BedTool(\"/nas3/gpratt/iPython_Notebook/SRR334701.peaks.bed\")\n",
      "results = get_peak_wiggle(bedfile, pos,neg)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 13
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "plot(results.mean())"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 15,
       "text": [
        "[<matplotlib.lines.Line2D at 0x269a9c50>]"
       ]
      },
      {
       "metadata": {},
       "output_type": "display_data",
       "png": "iVBORw0KGgoAAAANSUhEUgAAAfIAAAFXCAYAAABZQMyNAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAIABJREFUeJzsnWmAFNW99p9eZp8BBhhm2GURVFBRUEGDKBoRYRITYxJz\nNQmauMSoQXNjVu+9et/EmBi9YOISvZrFqEmMGvBqjAsQFQQRUIdVZIcZZph967XeD92n+lR1dXXX\nMk739PP7AD3dfaqqaznP+S/nfzyKoigghBBCSE7i7e8DIIQQQoh9KOSEEEJIDkMhJ4QQQnIYCjkh\nhBCSw1DICSGEkByGQk4IIYTkMH6nG5g+fTomT54MABg7diyWL1+OAwcO4Lvf/S66urpw/PHH4557\n7kFBQYHjgyWEEEKIFo/TeeTz58/H66+/rnnv+uuvx2c+8xlccskluOOOOzBx4kR8/etfd7IbQggh\nhBjgums9HA5j/fr1WLBgAQBg8eLFWLVqldu7IYQQQghccK0fO3YMn//85wEA1113HWbMmIGKigr4\nfD4AQHV1Nerr653uhhBCCCEGOBbyN954A0OHDsXevXtx5ZVX4r777nPjuAghhBCSAY5d60OHDgUA\nHHfccZg5cyYOHTqEjo4ORCIRAEBDQwNqampMt8Fy74QQQog9HFnkXV1dKCgoQGFhIZqamrBlyxZc\nf/31OPPMM/Hyyy9j0aJFWLlyJebNm2e6HY/Hg8bGDieHQjKgqqqC57mP4Tnue3iOPxl4nvueqqoK\nV7bjSMgPHjyI22+/HdFoFIFAANdccw1OPPFE/OhHP8J3v/tdPPjgg5g8eTKuvPJKVw6WEEIIIVoc\nTz9zC478+h6OsPsenuO+h+f4k4Hnue9xyyJnZTdCCCEkh6GQE0IIITkMhZwQQgjJYSjkhBBCSA5D\nISeEEEJyGAo5IYQQksNQyAkhhJAchkJOCCGE5DAUckIIISSHoZATQgghOQyFnBBCCMlhKOSEEEJI\nDkMhJyRPURQFdXub0d0b7u9DIYQ4gEJOSJ6ydW8L7n16M/7nr1v6+1AIIQ6gkBOSpxw51gUA2HWw\nrZ+PhBDiBAo5IYQQksNQyAkhhJAchkJOBgz76jvQ3N7b34dBCCGfKBRyMmC456lN+N3LO/r7MAgh\n5BOFQk4GDD2BMHqCnEpFCMkvKORkYKH09wEQQsgnC4WcDAgUJabgCpU8YzweT38fAiHEBSjkJCtQ\nFAW/XbEV67bW22uf9IIQQvIDCjnJCroDYaytq8eGbUftbUDR/EcIIXkDhZxkBYqi/d9y+7iE221P\nCCG5CoWcDAgU+tYJIXkKhZxkBWqymkOTmhY5ISTfoJCTrEDR/W+5PWPkhJA8hUJOsgPHCkwlJ4Tk\nJxRykhWoFrndZDdVx6nkhJD8gkJOsgOHMXLmuhFC8hUKOckKnMbI6VknhOQrFHKSFSgOfeucR04I\nyVco5CSrcJq1TpucEJJvUMhJVpCYR+5wOy4cS77ANVMIGRhQyMmAQHEcZCeEkNyEQk6ygkStddvO\ndelfQgjJHyjkJCtwOv87kStHKSeE5BcUcpIdOF39jPpNCMlTKOQkK2CImxBC7EEhJ1lBYvaYzXnk\nLmW9E0JIrkEhJ9mB4ixZjRY9ISRfcUXIo9EoLrvsMlx11VUAgNbWVixZsgS1tbW4+uqr0d7e7sZu\nyADGrRKtNMkJIfmGK0L+xz/+Eccddxw88QoTy5cvx5w5c7BixQrMnj0by5Ytc2M3ZCDjUMlpkRNC\n8hXHQt7Q0IA33ngDl19+uRqnXL16NWprawEAixYtwurVq53uhgxwEkLseB1TkiEs7EbIwMCxkP/8\n5z/HbbfdBq83san6+npUV1cDAKqrq1FfX+90N2SAozgUYscDAUIIyVEcCfmaNWtQXl6O6dOnsxAH\ncQWni6bwNiSE5Bt+J403bdqE1atXY/78+QgGg2hvb8cNN9yAmpoa1NfXY9SoUWhoaEBNTU3abVVV\nVTg5FJIh2Xqee6Ox//1+r61j9BbGbmWvz157N+nv/WdKeUWx+jpXjlmQa8ebq/A85waOhPyWW27B\nLbfcAgBYv349li9fjgcffBB33XUXVq5ciWuvvRYrV67EvHnz0m6rsbHDyaGQDKiqqsja83ysuQsA\nEApFbB1jc3svACASjvbrb8zmc6yns6NXfZ0rxwzk1jnOZXie+x63BkquzSNXFEXNWr/pppuwdu1a\n1NbW4p133sHNN9/s1m7IQMW1gi70rRNC8gtHFrnMWWedhbPOOgsAMGTIEDz++ONubZrkAU6njzFp\nnRCSr7CyG8kKnK4nLrLVmexGCMk3KOQkK1BrpTudf0YIIXkGhZxkF07nkdMkJ4TkGRRykhU4jXGz\nRKsNPKztRshAgEJOsgrbBjWz3QgheQqFnGQFikObmhY5ISRfoZCTrMCxQc1lTAkheQqFnGQXXMaU\nEEIsQSEnWYHjZDfXKsPlD0x1I2RgQCEnWUGioAuVmBBCrEAhJ9mBQ/1OLGPKgQAhJL+gkJOswGmu\nGuWbEJKvUMhJduA4a50xckJIfkIhJ1mBOo/cphIza50Qkq9QyElW4No8cko5ISTPoJCT7MLxoimu\nHQkhhOQEFHKSFThdxjTRnmQMJ5ITMiCgkJOswrFFTSUnhOQZFHKSFTgV8ESMnUpOCMkvKOQkK3At\nxk0dJ4TkGRRykh0oziaQqSVeXTocQgjJFSjkJCtwOg9ccTYNnRBCchYKOckK3Js+RiUnhOQXFHKS\nHbi2aIrzQyGEkFyCQk6yAqfLmDJbnRCSr1DISXbgUolWWuSEkHyDQk6yAqel0hPJclTyTGFhN0IG\nBhRykhW4tmgKdZwQkmdQyEmW4HQZU84jJ4TkJxRykhU4tcgZGyeE5CsUcpIVuLkMqd3Md0IIyUUo\n5CQ7cDyPPLEBynhmeDxMdyNkIEAhJ1mB03nkhBCSr1DISXbgUmU3N7ZFCCG5BIWcZAVOY+RaHaeS\nE0LyBwo5yQqczyNXjF4SQsiAh0JOsgSn65ETQkh+QiEnWYHT1cs0rnWqOiEkj6CQk6zCcYlWZ1sh\nhJCcg0JOsgLni6YwRk4IyU8o5CQrEPPHbWecK4YvCSFkwEMhJ9mFCzFyKjkhJJ/wO93ANddcg2PH\njqGnpwcnnngifvrTnyIYDGLp0qVoampCVVUV7r//fgwaNMiN4yUDFDcXTeE8ckJIPuHYIn/ggQfw\n/PPP4x//+Aei0SieffZZLF++HHPmzMGKFSswe/ZsLFu2zI1jJQOYRIlW+1tQX1HHCSF5hGMhLykp\nAQCEQiEEg0GMGDECq1evRm1tLQBg0aJFWL16tdPdkIGOw2w3ijchJF9xJUb+zW9+E+eccw4KCwux\nYMEC1NfXo7q6GgBQXV2N+vp6N3ZDBjCulmilqBNC8ghXhPy3v/0t/vWvf6Grqwt/+9vf3NgkyTec\nii+z3QgheYrjZDdBUVERLrjgAmzZsgU1NTWor6/HqFGj0NDQgJqamrTtq6oq3DoUYkK2nufyilYA\nMQm2c4yDGjrV18OGlaO8tNBS++7eEP6xbh8uOms8ykoKLO9fJlvPsZ6KimL1da4csyDXjjdX4XnO\nDRwJeWdnJ0KhECorKxEKhbBmzRrMnTsXfr8fK1euxLXXXouVK1di3rx5abfV2Njh5FBIBlRVVWTt\neW5v7wEQm09u5xjb2nrU141NneixKMZPvLQda7Ycxs59zbj6khMt71+QzedYT0dHr/o6V44ZyK1z\nnMvwPPc9bg2UHAl5W1sbbrrpJoTDYQQCAcydOxdXXHEF2tvbsXTpUqxYsUKdfkZIRrgxj9wG9ce6\nAABNrT1pvjlw8PT3ARBCXMGRkI8ePdowJj5kyBA8/vjjTjZN8gxX55HbyHaLxpt4PJQ3Qkhuwcpu\nJCtwXsRFMXiVOZG4kvu8FHJCSG5BISfZgUOLWtPEhpJH40LupZATQnIMCjnJChzqcMptZUpUyT+L\nnJP0CBkYUMhJ9mFDYbTTyG3EyIVFzhg5ISTHoJCTrEB2p9uJl2vbWydC1zohJEehkJOswM0Sq3ba\nR5nsRgjJUSjkJDvoB/GWETFyWuSEkFyDQk6yAqcWuaJZxtT6BiJ5GCO3c54IIdkHhZxkB07nj9Ei\nJ4TkKRRykhU4t8idtc/HGDntcUIGBhRykhVoSqza2oD80sH0szwSckLIwIBCTrIPhzFyO+3zMUZO\nk5yQgQGFnGQFzueRS69t7D8RI7fRmBBC+hF2WyQrcHUeuY02+ehap0FOyMCAQk6yAzfnkTuYfpZP\nyW6EkIEBhZxkBa7OI7ez/3ij/IqR0yYnZCBAISfZgZvzyB3oUz651gkhAwMKOckKnOqwW8ug5pNr\nnfY4IQMDCjnJCjRZ547j5fY3QIucEJJrUMjJgMCtuuEe5I+QM0ROyMCAQk6yAqeLnrg5fY0QQnIJ\nCjlxlVA4ip5A2HpDV0u02sdOMRpCCOlPKOTEVW594E3ceN8ay+2UlH/YaU8xJoTkDxRy4ipdvTas\ncehLtNragLP2rjTOLbgeOSEDAwo5cY0tHzW5syGHMXJHu3ZpO4QQ8klBISeu8T9/fd92W6eLnqSr\nJxMIRjKK3eeTlZo/v5SQgQ2FnGQFfb1oyg2/Wm0rdk8IIdkOhZxkBw7VWxNjzyOr2hE8TYQMCCjk\nJCvQWuT9GCOnuBFCcgwKOckOnAqoSyVeqeOEkFyDQk5cIRyJOmrvfBlTk8+sbDCPTPL8+aWEDGwo\n5MQVuu1Uc5PQziO3o+Sp24fCzgYZhBCSzVDIiSv0BiOubcupRa5vH7Qg5HllpeaR94GQgQyFnLiC\n00xxV+eR6wiGLAwyqG2EkByDQk7cQSeAVoXdzVrptMgzI59+KyEDGQo5cQW9KDiZQGbLIkfqGLkV\ni5xz0AkhuQaFnLhCkgBa1MN0JVbTbyB1eysWeT7BMQshAwMKOck6XNZxazFyQgjJMSjkxBWSDXKL\nMXJNQRcbjnmTbDkr089opRJCcg0KOXGFpBi5Vde6i6lX+m1Zqgfj2lEQQsgnA4WcuIOLS5bZmkdu\n0t7NQcJAgueFkIEBhZy4gnOLPPW27Ow/8w8dfZkQQvodCjlxhyT9c6LkDku06tpb2Vpexcjz6bcS\nMoDxO2nc2NiIW2+9Fa2trQgGg1i8eDFuuukmtLa2YunSpWhqakJVVRXuv/9+DBo0yK1jJlmImzFy\nty3yvBJnQkje4cgi9/l8+PGPf4wVK1bg+eefx4svvoh3330Xy5cvx5w5c7BixQrMnj0by5Ytc+t4\nSZbixAqOtXfQWNcmWbgz32A+iX4e/VRCBjSOhHzo0KGYOnUqAKCkpARTpkxBY2MjVq9ejdraWgDA\nokWLsHr1audHSnKLT3hNcLNxQD6JMyEk/3AtRn7w4EFs3rwZc+bMQX19PaqrqwEA1dXVqK+vd2s3\nJEvJrnnk9pU7nzK5WY6WkIGBoxi5oKenB7fccgt+/OMfY8iQIba2UVVV4cahkDT01XluC2irpw0b\nVo7S4oKM25eUJL5bWVlm+ThLS4vU14MHl2raVxxuV1+n225pSaHjc5Qr93JZWeKc5coxC3LteHMV\nnufcwLGQRyIRLF26FIsXL8ZFF10EAKipqUF9fT1GjRqFhoYG1NTUpN1OY2OH00Mhaaiqquiz89zS\n0qX5u6mpEyVFmd9e3d1B9XVzcxcqCq05i7q6AoljaevW/M62th71dbrf39UddHSO+vIcu418znLl\nmIHcOse5DM9z3+PWQMmxa/3HP/4xxowZgyVLlqjvzZs3DytXrgQArFy5EvPmzXO6G5LlJLnWHcw+\ns7V/NzdGCCE5hCOLfOPGjXjuuecwdepUXHrppQCAW265BTfddBOWLl2KFStWqNPPSL5hX8ntxalT\nL2NqaVCRR4MAhsgJGRg4EvKZM2di+/bthp89/vjjTjZNcgyHq5hq55E7LNGq37mVgUE+JbsRQgYG\nrOxGXMGRFQx3DWEnRebyyUrNo59KyICGQk7cwcUgt+uLplCxCCEDGAo5cYXkIiwW55FrXtuYR25S\nEkbrtqeqq/BcEDIgoJATd3AYI9ea1M72b2aRp9s0tY0QkmtQyIkrJFnR/biMqZMlTZnsRgjJNSjk\nxBUcZ633oZJrLfK0Sp435NFPJWRAQyEnfYNlH3XqeeCZtTaZR+5wahshhGQzFHLiCm4uY+r2PHJL\n08+s7zp3yasfS8jAhUJO+oRsmkduadsUN0JIjkEhJ67g2GXteBnTzI4l3abzKdktf34pIQMbCjlx\nBefzyJ3Jiuk8cguZdBQ3QkiuQSEn7uDUJHcYIzedR47Un+UzLI5DyMCAQk5cIdkit9/e6ewzsw/T\nbpvaRgjJMSjkxBWSE8wsutadms0mYm1lrXLqOCEk16CQE3dwlCqubWDPIjepp64oht8z3hClnBCS\nW1DIiSskF2Gx2t5+23RtGCM3hueCkIEBhZy4g+NFU5w0Nm9jRbCobYSQXINCTlwhSQA/8WVMM9t1\n2nnkeaTk2fxT27uC+NOrO9HWFezvQyEk66GQE1dwvmiK02VMTWqtW1qRJZvlLX/406s78eq7B/Hk\nKzv6+1AIyXoo5MQlnPrWnTU1y0x36rUfuGTv2WiPW+LttMgJSQuFnLiCm8uYOl00JdnNb75t2WLP\nJ9d6LsDLQUh6KOTEFZyXaDXbmsUtWBxU5KvFns2DFo/HAyC/rgchdqGQE3dwvGiKM6tYa5Gnnkdu\n3DjFa0IIyQEo5MQVkhPMrLY3fm2rvWmt9eSta4rJUMmzC14OQtJCISfukCPzyI2+ls0u5nwl7lnn\nwIqQDKCQE1fo/3nkGbrmOftMJZsHMB7xIouPkZBsgUJOXKH/55HLL/Wpc+Z13J3umvQBHk/67xBC\nAFDIiWtk7zzy9GXf8nP6WS64rbP/CAnpfyjkxBWyeR55uvB7Pol3riDscV4bQtJDISeukuiAnfTA\n7ma7pSv44nwOe46SzT+VQXJCMoZCTlxBdLcem7FNp9XV0k0xy7QxZSM78CB7YuRPv7YLb394pL8P\ng5CU+Pv7AMjAQIinxwNAcTaP3N4BpN6W1m1vPo+cSp4dqNPP+vl6RBUFr2w4AAA4e/rI/j0YQlJA\ni5y4ihvJxn25aIph2zzVcefF+BT8c8MBHG7qMvy8qzfkcA/9fz2i0f4+AkLSQyEnriDEMFEj2+I8\n8jRWc/r2qaeYcVGUvuGjQ2146rVd+Mmj7yR99vI7+3HT/f/C+7uPOdtJP18vZ7kehHwyUMiJKwjh\ntmuRuzoVyiSFPt1+8qrjdvhT27tCKTfzz3dj7uh3dxy1tW0njh1FURAMRRxsIUE06spmCOlTKOTE\nHfQWueX5Z9LLPpx+lq4tyZxwJLXKqTkTNrdt17MDAA//vQ7X37vaFdd+lDcHyQEo5MQVRHfntW2R\ny6/tlGiVXusN8rRB8Px0vTv1gpgKefx/xzkTNg5x/baYF+BoS4/DnQMRxshJDkAhJ+4gLHLYtcid\nlmjNrJF5XTdihZCJkCdOqjMld3Jt3BBh2SI3G7gQ0p9QyIkr6GPklpPdUry21V4n6lZWP8snUXfq\nfQiF+07Y3Jh+9tM/bERnjzP3uiINBgIuxd0JcRsKOXEFfda6VUVUXFRy0xh5OmXIJ9+6QyIRk2p6\n8f+dT0d0dj3Wb2tw1F426gNBCjnJTijkxFW89nRcg9MYucGaqqk/gvnUNZIac9e6O8luTukJhB21\nl+eRB214INq7gjjW1uvoGAhJB4WcuIJzi9xhiVbTeeTyHwZtU/5BzAjHhc1IchMh8v6LkQPO4+Ry\njNyORf6d5W/i3x9829ExEJIOx0L+gx/8AGeffTbmz5+vvtfa2oolS5agtrYWV199Ndrb253uhmQ5\nTmPk7h5Mepdv+jcHPk6jCCL5y+9P7kbUgZ3Nbbu1ZoqZ+z8TmOxGcgHHQn7ZZZfh0Ucf1by3fPly\nzJkzBytWrMDs2bOxbNkyp7sh2Y7DeeR9uoxpmqpxTsPz+YpIdvP7+sCx50KIBnDBIo9SyEmMF9fu\nxe5Dbf19GIY4fgJnzZqFQYMGad5bvXo1amtrAQCLFi3C6tWrne6GZDn9PY9cs62kwm6ZJ7jlU2U3\nt+aRF/hMLrqN+0FRFMeWtCDisDSbPA4Iu3RMJPdoaOnGs6s/xv/7w8b+PhRD+iRGXl9fj+rqagBA\ndXU16uvr+2I3JAtJWOROTHLr+zXdXxprn92zPUKmrnX7yW7/89f3sfmjJs127OJ0QCBPPzNN7ksD\nF1/JbbwuJV/2FVmzjGlVVUV/H0K/09oRwOvvHsCiT03Ahq31mDFlBMpLClzdR1+d5/LyIgCJTn3w\nkFJL+/IX+BLbqii2fJwFhYlbuaysUNO+pKRQfV1ZmXxcXqltYaHf8TnKlXtZPi92jtnnj12zIoNz\nJgZ0pSWFlrctL7Ti83lRVVWBt7YcRmtnAIvOmWB6vHpXekGRH/7iAuw70o4ZU0ZYOg4A6AgmxLu0\nrMj2tR0ytAxF0j2eK+TKvdzXhKQhaTaekz4R8pqaGtTX12PUqFFoaGhATU1N2jaNjR19cSg5xS+e\n2oRt+1qweUcDNu1qwqDSAtx/81zXtl9VVdFn57m9IzbFRlgerS3dlvYlL3LR0dFr+TiDwcQ0o87O\ngKZ9d3dQfd3c3IUSnSu4pSOgvg4Ewo7OUV+eYz2NrT24+8n38PWFJ+DkicMst++RzoudY+7qirX3\nGrQX90F3T9DR+QiHo2hs7MDdv98AADhzynDTc6wv2tLVFcBNv3wDbZ1B/PTa2agZWmpp/8eaO9XX\nzS1dln6LnCjX0NCOkqKssZsy4pO8l7Odo9JSvUePtrs2PdKtQUGfuNbnzZuHlStXAgBWrlyJefPm\n9cVuBhwHG2OdxpFj3QCA9m7niz58YuiT3Wy2B+y5U80XTcl8Hnku8drGg2jpCODB5z+01d7prw6a\nJLuJbbtdqzydi1pfbS4cUdDWGRtwdEgDl8z3J20rbO23hKVjYaJcbhORrl9Xr73aBO3dQWza1ejW\nIWlwLOQ333wzvvzlL+Po0aOYN28ennjiCdx0001Yu3Ytamtr8c477+Dmm29241gHPEJP5M4qV0Qm\nKdnN4nGnE1tL7U1KtKbPe8uN8w0AvvjJdkMs7ZQyFV4Uv0mym+MYNbTXJF2ZVL1gyufG57Xe3Wmm\nn1lMnJNj6kyUy23k+8huaeIfPrwOy5/9AIck694tHPt6Uk0te/zxx51uOu8QHZacaRuJKqYdZdYg\nkptcsMgdF1u32tRZnl2/4YvfF7bFUmq262ArTju+ylJzYZEb7z35XraFomiqs6VbZ1xvscuWlMcT\n82IMLivErBMyi5drpp9Z7MDlDt/xeSD9iizkdq6loijojt/HkT7wzrCyWxYhBCWsuWlyQ1rEUdos\n7Obyoim6z9IIdb8Wr3GAyKR1Y83sxlbrZUSD4ZioGs7Nj7/l9P492tKj8Raks8iThFx+liIKnvzn\nTvzGJBTREwhj/bYGtbOWf5tVq1oj5DYHW5s/asIrGw7YakvcQxZfOzMQ5PK+fbHGPYW8j+joDqK1\nM5D+ixJCUOSHPlemrSQqebmxaIqzijCmMfIBNP/M57AQi3xe7NxnoVA03tZo2zGsClhSWARA3d4W\n9e9AyNya0XeSEYvTxx5duRUPvVCHNZsPAwA27kjENK3GuUMuxMiX/fV9PP3arj7p/EnmRBwaV1rv\nDIU8Z7hl2Zu49YG30G0hMSJhxfTtRe9LvLZLtDqNkct/JFWEybhtLvWXfrvVdwywIxQBE4sc6r1s\nMUHMQPD2NyQyp9NZ5Pr9RSLWxHTngVYAwOGmWMLpqxsPWmov47TzDoUTvzVXBvQDFfn62bkUYYcW\nfToo5H3Mt+9fk/F3VSGXrJi+iKf0BYouRu7EIne9RGuK7xl+nkPmuc+hkMu/1Ja7UFjkupPa1hlQ\nBddqgpiRxS1PD7QcI7dYYlXcv1GD+8BqQRinyW71zT3qa7cq3RF7OPWSyvkVFPIBTiLZzXmMvKUj\ngFA4it2H2nD3HzeizaKb3ypuxshdO5hMN55LZriE102L3IHFqD99Sx94S3394cfNlmYCGAm1Zp5/\nOiHX7UrON8kk21j1KBmcD6ti6nT6mXwucs0zl400tfWoHheryF5SW89KH1vkuVWhYIAjrq9TIW9q\n7cH3HlqLGZOHY8eBFvQEIvjHhgOYPGG4W4eajIiR21w0xfk88gxj5AaqrqT8IzvoCYTx8ZF2nDS+\nUlOIwlUht3jOo1FFtTLTtT1wtBPjqjMrfGEk1JaEXPe8BKWlR9O1BSSL3OAnObHI7TzHTjOliZbv\nPbgWAPDQbfNQaLHKnta1bsMil72sTHYb2BgJ2M4DrXjohQ/TuhRlDscLymz+qEl1fzp1w6ZDHHli\nNxaTnBzGyDXbMkiYSnxmuHPH+1YUpc+Kfix/9n3c+/Rm1O1p1u3T4Yal9laFRhbFdE0zEVCz45Cz\n1oMWk926NVPX0ltFqkdJUZLuI0fTz2zcGxGHAwFiTK+F+1Hg1LhijDyPMOqYH3txG9ZvO4p3tjVk\nuA0Ff131kfq3uOn6WsiT5pE78a07jJGbbjvdxzbV8VfPbMa1v1jVJw/p9v0xd+CR5m7N+3KH8tGh\nNo3lmgkbth9VX1u1MuSBpZEbWsaKAKY7f+kGS/pOVh5EBKXksVRudrkOgj6ubdkil13rDi1yJru5\nh+ylyRSnMXJmrecgdquDmSZaZbjJnkAYBxuTKwf1yZrREm7GyO2cvUx13DDZzcQtnyliipRsAVoh\nHImircu8hKh+KCZ3KD/9w0bc9uu3kCkNzd0aS9dq56S1yM3bhizEltN1cuni1EmudakDlS3yVKLs\nlSxy/aChuc3aXHunFrks/nYGAsQYexa5wxkItMhzD7vlGM36wwKDpSKt7LvPhVwXI+/XeeRJQXLz\nGLmbdNqo5w0Av3xqE5Yuf9O0VKp+oQYncVP9fqxuShbFdAPXVCJ2tLUHz6352JLbMV0nmmmMPFWo\nSvYo6YVchKwyxWnWuvxbcmX2Si4QsGORO42R0yLPPfoiVpppYlMql6HPQplXRVcW0wrZMI88OdnN\n5EM4n/oruCoIAAAgAElEQVQm02lzoZudB9sAxCzlVOgXXHLSH+g7I2cWufl3U1m/v11RhxVv78XL\n7+xX30vXyaV7tvS/S/5LPrff/c3bhve4OMfH2npxy7I3NZ919oTQ1au9vrH8hU2Gx+K0IAxd632D\nLYvc6fQzWuS5h/6hdWMhjkw7glTfs2KR/+WN3bjxvjU42pK5BSKE27ZFLr+2Y5Cb/ZUmmc3Nx8rO\nClua9mYWue5vJx2C/hxbtTKsuNZT3ZM9gdg2duxPVG7TC7k3yQthzSKX2bSrSfP3roPJU5HE/bsj\nxTQlfbLdpl1NmspzMnJBl8de3Ga51KrTZLf3dzfhF09tspRsmA/0Bj75ZDenMxjSQSHvAw4e7dT8\n7UZ5xWCGCUMp58paOIaX18cspEMGsfaUqK51zZ+W29vG1KpOo+QSh5q6HA287Frkgi5LrnUnQq5t\nm2pbqVzQGtd6WitaQWdPCK++e0CzvcFlhQCAgEmxDL0nyWqym1XSrTNtxcWtd6c//douS8fiVDzu\n/8v72LavBZt1A5h8JxCy7m2Uw1jp7ncj5CVwaZHnAAeOduIXT2/WvBcMRdEbNL950k0vC6WZdqN+\nL0VHY6cjKC7MfK5lItnNXta61jVu/VjN2qQ9FukL7V1BRx1fR48zi9x0OVG9a91Bh6BvajTYbGrr\nwfX3rsaf/rkz6TMrrvVwJIrf/2MH/vTqLvzfun3q++KZKJbm9Mr36fDBxUkr/8mfv/7eQazZcjjt\n77BCughWc0fAcA0Fo/3aXe5S4EZhKMAdj+BAwlaMPGLvWjS29iAciWpd65xH3rc0t/di3dZ6R9vQ\nW+MAcON9a/CtX60xtSbadS7Z6qGlmr/lqTNmpOo8bBWksHDDJUq0qu9Y2peSzv+ddgPytlJ+lL4g\nDFK7VTOhx0JtfRlx3kyT3XR/O+kQ9J270aDgz6/HpjHK9cYFspCnE4pwOIoD8efiiJQw1htMrtUu\njmPahKG46xtnJa0hLneof3xlJ554aXva32GFdBb53U++h1sfeCvpN4fCUUQVRXNNHAu57I416Du6\ne0MZJTxSxrX3WK8NIZdnDWT63DU0d+P2h9bif/6yhVnrTglHolhbV59R8tadT2zAI3/fio8Otdne\nn1k/YHYMHTqXrN4azrRTSGWRZ3rzyN+zc8N57c4jl3Ce7KaPkUuJdBkEydP05abYFVeRw2A24NKL\njDOLXDH9u6UjgHellb/0WE12k6d1CUSHamR5njJxGIoKfMmu9TTClepYzj99tGm77t4wfvHUJhxu\nyiyc1NDSo/k7EIrg3qc34xs/f0M9l6mexfauYEZWsplFHghF8O37/4W7//he+oOlkmvOX7ppnsbt\nrce4DzbGBq91e1s0Rlzex8hD4Sj+suojNFhIwnpt40H8dsVW/O7l7Wm/2x4X03YbF1pgNqI3Gwm+\nt1PbaZbYFXKHFrnmhrUwZSaxjKk9tJnjNlzrWiXXfZimreW9pcauuIrzZtZef26duVu1f+v3m24J\n3qClZDfFsPSpyAA3snbELA39Cm/p7klx/142byJmnTBCfb+ipCDpu/KYYN3WemzbZ5y0ZsQPH1mn\nSWwMBiNqe2E9hwy8aGs/rMd3lr+Jd7amL/BklrUucjF2H25Pu51cWgior5D7xeZ2a/UAAHtZ6/LX\nMqkq6IScEvI3Nh3CS+v2Y9lf38+4jRgV7TqYuZVt1yILR6Kmmd5mQr51r7b8ZnGhtgx+Jslur757\nQD03+nnnmXb6YZuxoKQYecYtte3toxi8Sv7baJDgZgzRqrgGQxG8v/uYqtKmQuVijFx/nPptpfNg\nBSzMIw9HoprSp2J/onPTxh/jJYXjlrhYc70oPrAVnxv99t2H2vBR/DmvKC1EofQMGNVhkK2kQaWF\npr9h5LDSpPdkj0XAYJ6w0aD6/96J5Qis33Y06TOztdRZEMYZsnek2WIFRH37TJ87+bmQn6e8r7Xe\n0hEbSR2zMqKycc48Nu3K3720Hc/9a0/Kz806x2A4qnGnFxfpLXLzuM7mXU3406uJrNiSIu1AQO4U\nDjZ2plwNTf5eR3cQj6yoy8zdqIuRWxbHdO7vzJsnXXOr27N7/QHrQv6nV3fh/r9sSYia2cHqrWgH\nHYJ+wKA/7p40U3SERe7zetIWkwlHomrIRXSCsms+YtBJ+uLfF6WFS+P3szhuo5yR//eHjXj9vUMA\nYiEeWcj1sXYgcy/XhbPGYObUEUnvy2WP5UJAZkLe3B577ioHFWne/791+/CNn7+hsRaNzov6GRPY\nLBF2aJFrqvRleO7layb3/XlvkQurtMBGlbJ0VvZrckKPjX5cURS89aF5opxZ5no4omhW5NHXp05n\nkR9q0ibZJQt5Yu3oOx5bry41mbQwhNR5rFy7D+vqGrD82fQeELEV/bzfTDHzjFtvn9omNy7RamOH\nKbD6kG7bp/XEmE1vMrPYrKJPktKfg+5e82l0QsiLC30ZJLspidyJ+HuBFEt0CqEWrnUhlmKQKyzT\ndM+D1wsU+H3S38n3pbbyWurtVVeWJrn45WMDtDFz8RuMYuSiQx+s8wD8ddVuAMAHHx9LbMckRm5l\nGhw1X3st7JRRtrOeeDSVRZ6PQt7c3otH/l6HhuZudQqW1SXogPTa/MzriYVG7Kwv8l9PbEh6r0h3\nnGZWTjgcRYHPgxmTY0uN1uhceevqGtSH3YgyXQywVGfRi5tPtsR+u6IOtz+0VvM9+XMx8NAn4hki\nYuSqRZ6+iVF7e4217c2z1jPAQbKb1YdUP+Ay6ySSpow5EfI0FnlnGiEPSEIud1jyazF1LCS51sXn\n8hQgIc7yNB0hkuKwEhZ5PP4sufaPtfUmxbe9Xg8KC2SLPPmiyuJttiBKgd+btjKi7OFSLXKTKaOp\nvCny29oBjnZbTjPi8w35WgRDUcvPTsiOkEuXqEe63/NyPfL1245i3dYGvLvjqOreyrTuOJC5ded0\ncbD9DcnTzspL/BrLw9Qij0ZRVODDDZdOR3N7r+HCJ/+3bh8+f+5EQ+vCr3Md6mPsEamzFKytiyXc\nRJWExWSUFZyJCzc5Rm7tZnW6jGnGC84YxchdTAay+pCWmoRA0m3bboewfV8Lfrtyq3Zb+uU/00yj\nEzHy4iI/lPYAFCWW0CYfk8/nRTgS0YiQEHAj1/p1v1ylXh5xj4vzUSyEXLXIE+3//cG3k44vybXu\niwVM5F8prKyjrT147d3kKXaCkiK/4fmQf0OX9Ln4PcFwFH6fx7DOeqra6/K7Zha5ldrtXAIVSaV1\ne4MRlBZnLn92YuTyM9XRlRx6cZOst8jFAxuOKKo7z5KQq+fMXKnlbHMr2dpmDCrTxsHSW+ReFPi9\nqB5aqla90pNqPrk+hp4qRm50EwU1nWqyGzojIdfPI7cfIu/jeeRp9g1HBrnlhUz0gzKzpCa5A1EU\nxfZKaw//vc5020CyxRcIRTQJmbJFDiTOq7ydr108FUBM0IQrvK0z1qHJiZ+RqIJAKKK5DsKCFlW0\nxH4ikSg6u4P4z8eTPWAyPq8Hja29mr99upBcKBJFS0cA339oLfYb1H8QlBT6DC162avQrXOd7jzQ\niqMt3Rr3vkxKV750EsyKkFip3d4Xaz/kGvr6DOkKdOmxFSOXviffX3kZI5dLXoqTWWhByAXpQrfy\n56lGTEdbe5IKt5gxtEIr5OYWuaJx3w0uTyHkKdx1+pjhIN1AIGpgkQvkDGT5c6POOR36WKgd3I6R\nu7FMaaZYHW3rLT2zcy1v+9GV22LZ7jYwOsYkIdfdJ4+u3IpfPr0ZG+OZ2sFQBB4PUBgXKtFedF6n\nTBqGUyfFwkShcKKyYXNHzHrXx8j1CUgJi1w88z71uxu3H03rWvZ6PBhTVab+7fcmu8dD4WhGS7+W\nFPkNXevyYEQuBNTRHcLdT76Hrt5wyr4q1fH3aAY4ie9s0k1PlZ/TdOvQ212NcSAh1jAQ4RarRWFk\nAyrjrPUU38tLi1xeREJNdksxyjUi08xe2SJfHS/72NkTwjfveQMr3toDRVHw/YfWYunyN1NtIgl9\nZqqZRR6JRDVJfENSCbmBRd7ZE9LE+I3ai9G9sZAbJx4J1cvkxkssY6ptmyl9Oo8c5p8l7c5JQRiL\nD6neUtDHQt/dnpimJJ+XtXX2KxAaeQ30VoY+2VII+IGjHQBi90xhgU8V3Prmbvx2xVZ1YOLzelTP\nWW8wrFri4UgUqzYd0lizkUg0aUqQsIBFh1tcFLOKMxUlj9eD+TPHJLbn8yQlrGW6reIiv7FFrnGt\nJ66jfE0L/F7D2ylVTP6vq3ariYbyc7dl9zEoioI//GMHfvLoO5qBwG2/fst0YONkuduBgljDYPjg\nEgDWhdxOjDyU4v7KyxKt8rQO8fDr6y+bIdqks8jl57RuTzM6uoPYsb8VkaiC5/61RxVAK9dgaEWx\n5u9UFrmiKAhHFI3rL9Vgxcgi325QyGJIuXYQYepal27qNzYdShxXXPUy+c1OY+TObeXUVre2+qv7\nMXKjEqOZsOdIO5ratJaoXlB/8/yHic9cGskbhY701kMqofn7W3uxevMhBEKxnA7xXC376/tYW1eP\n5+PTL70eD/w+L3xeD5rbA5pjf/r1j5IGjy3tWiEXAwTVhV8QE/JINJrRmu+KomhW/PN6PEli3JOh\nezXmWk9sK2HVJdrLrvXuQEj6rs9wYGjm7hYZ8PrrdKytF29sOoRDTV1JU1nNVjjbsb817VoOA51O\nVchjfXLAiWs9UyHXGV3ivskb13pPIKzemLJFfvhYLAEs07W5Ae0N3t4VxI33rcGqzYeSvqevyBYK\nR9HeFdD8LcikTjUAVOpc6z0pRoHixijIYIBiZJEbnQ99JSsxCjSyQuRz9Ob7R9TX1gaO2hi5o0VT\nbNznSso/zD9L+Z4F5BG2FbE1Wl3OLD/DrZG80THq3zOz8H738g40NHejqMCrhlKERSpCT+KeLCzw\n4WhrTJhOnjgMADChpiJJyPXxfjGPXNyvRYWxcq2RiKJWYDRDf59HFSUpRm62Ut2X509WX+td62KR\nF9mqk0MknT2J1wV+r+GUTLPzK1zl+msiJ8Dq+xIzoX5/9zE8+uK2lJ/nA13xazJ0UEzIHVnkGT6H\n+ntQ3Dd5I+Q33rcGP3n0HQDaEy5OppUTIVwq4YiC93Y2oicQxu9f3pH0Pb0YBsNRzQIPsptE7yJP\nNbrWC3lviuQk8bv0HY0RRha5kfwnVXaLRDX/y4hO1ck66gnXus155MkefYsbkF/qldx8i04fK7Ml\nCiPRKJY+8KbhEpZG943cXu8SdasDMLpEgVAEt/36LVx99+vYfbgtybVuRCicKPQiiujIhWIAaKaA\nTRk7OOYejypaCyeiJHmr9M9jzCL3IhxVMlrzPakOQyiaZJGL7ZQZZC9fdOY49XWRLtlNVJkLpBBy\n+VoX+r2aUrGpjk/mWNxLI67/F86bBABo7kh4b/R9Sbo1x+UQjRUURcHBxk58uMdePoZTunpDeOO9\ng47vfeFhGlQaM3B6gxG8t7Mx46VltVnrme1T/3yLadN9UaUv64RcPBBNbb347Yo6HNUtTgBYS95o\nic/vbOsMauJY+lGVvnMLBCM4ciwxAg5JD0qLripaquOpKE1YxT6vJ6UrL2GRZyDkBha50UOsn2uf\nmH6W2iL/nW4lKTtJMl6bFrkWGzFyk9Zpw+cOn6tfP/eB+lpvRbV3hdDWGcQrGw4ktTOyyuT2+gGj\nU4s8GIrg0ZVbDa/rkWPdqiX4/36/0XReteCyeZNUoRaDJ5HHIgZ0RVKIqLKiCMWFPgSCkSQLp1f3\nW/WV2IoKfSgq8CIQjKgCbCSQAn0nGgpHVDEuj3urhLevUhcCE5x1UjWqK0vibvnE8RQVxIRffu5S\nzSAo9HuxZOEJOOukasPj27SzEXc8tl7zmahzLwbtlfEwmVxhUd+XpBNys3BkKBzFS+v2Ga4x8dfV\nu3HHY+vxq2e2pN2HEYFgBLsP21+EatWmQ/jDKzuxbX/mdfCNEAOn0uLYtQ9Fonjgbx/glQ0HMkpg\nDmtc6+mfjWhUwYtr92neEzOJ+mIWQdbNI29qSwi3mOesJ9PkjVA4qhYziSoKDkkPwo2/WoOx1eWI\nRBR874rTktxfgVAER5oTFrks3vrRsLgwk0YPwu5DsUUMbv/KaWqcetigIvQGI0mdlXycAJIyY/Xz\nXgFji1z2WlxxwfEYObw05fxks2S3dJXpzNBP87M8j9xxiVYTk16TSGfQVtfAaonWD/ckpmXpLQcz\nr4YQy/KSAjWGJwu5XhycxMgDwQhu+NXqjL+fiUU+pqpcXfJVnd8d0hZ0KZAs8sqKYhQX+tArCXks\n7q1oKpoByc9CaXEBSor8aOkIoDvuJpUHyknHb1BARXi8Rg4rxa6DbaprvbKiSF2TQea6z0xTr5/G\ntV6U7FpPhdfrRWGBD9MnDNUslCI8fMv/9kFSGzEYEjFWkbgqGzU9vdp9p1tj22zK7sadR/GXVbvx\nl1W7cf/Nn9LUnX9p3X71dXdvOKnIVToeeO4D1O1pxvf/7XRMGTvEUlsgURfdLAySCeJZK4lfO22o\nNH17uSZAJtOT9fPWgYTnJ9XMIydknUVe35x+ZbNMLUX9Ck57j3SorwOhCD462IY9R9qxdV9zkkXe\n2RNS6yID0CQl6ad6iU6jKp4RCcQ655IiP3567Wzcec1ZKCnyq+5DRVE0Hbxwd/t9yVaIHiOLXO5Q\nzj65BtMnDEsqdmA+jzyWSewEIYaqB9JBjNwpZha50Z7cTCLVDzLNSokKsZQFSbRvaOnW5CsAgGLz\n+VcUBd++f43mvcljBpu2ycQiH1RWqFrconMTHZgwYAt1FnlRYaxIkti+mB9+SFfPXwysb/vSDMyc\nWoUZk4ejuMiPnkBEfY7kwarebX58XDSWLDwBHg8wc+oI9TvCWyAGT1VDjC1y+btyxruIdWZioYrr\nqRfSkOq5SLwnQnEhVcij8AAYFDcIWjsTlqPeIt9zpEN1yRuh71sEz7y+C4/8PVEcSL94k4ydugV1\n8UFuJv26Ee3x35xpYmIqxDkV94w2tGN+r0eVWCio1IJFbbQUdlncG9AXiYdZJ+SZCEqmdYaFq1C4\n0lLdTKFwNKkz17v0//7mHvW1/kKEjYQ4/oTWDC1FSZEfxYV+1VX64PMf4pqfv5F4YK0IuaFFLiXX\nxLehF3IxYjeMkQcjeF76fbZwGCNPZzWnbW6SLefU2jcjaV1v3ek1e2jFdZevlRho/eDhdVj59l7N\n940GYfpVuYzc779duTWpbbrVvvZIg17BhJEVmr8rSgtU17rYvhg8e+NKXiRb5OVFKCrQWuT6CoQC\nIbrTJgzFjZ87GQV+L0qL/IgqCtq7gvB5PRrrUHi/Rg8vwwPfmYsxVeUAgLmnjsKj3zsfwwYXq++N\nGl6mOWbxt+DE8ZUpjwdIPJeZWOTCktSHukSfIb8vikCJcxMMR1Hg96I8fn+0Scm3+qz1P7/xkWaG\ng55UFvk/1mtDPnrLfuKoQSn3aQUrM41kxNrhqbyZmRKOxAZF4tplWmcfSJyT8vgzox/kNrR04xdP\nbdLUQlj+bLKnpUS1yCM43NSFBpuDGyOyQsgDoQiuvvt1PPbiVk01plRYtcj1dcuT9h+MJI3oG+Mu\nfhHXkBdF0Mc2xfH4fYnORZ+BXlzkQ08wDEVR1OUPRawvIrWXua52GoYPLtYkzRkJg+hQLj9vktox\n6F1gnVLSn55AKILqypg3wakOq1nrNtvHXrsbI0/1PfU9vRhbUPtXdB2hFYtc3EdipA6kq7We+Exc\n3x9cOVP7HYPs83UGISo7Hes3a6epolda5Iff500SKHGI4j4Vn/u8HhQV+lBc6EM4kigQo1/lT2A0\nE0NY7y0dvSgs8GnEaUhFrJONRBU1DioQg8t/+/QUfPmC43F5PHlMUFZcgKsvORFnnDACV1x4PG6+\n7JSkfctnVWwvEwv1tClVAIAinZD2BMIIR6KqSBQV+NSkNuFSD8WFXPweec0DI1Hdc6Rdvf76ezrT\nhab0Qi73B+nK9pphZaaRjIjbOxlEADGx9vu9qodI7sNTzfcWiP5VGIR61/pjL27Dtn0tePKfOwGk\nfoYL/F4UFngRCEfx40ffwQ8eWWfvxxiQFULeEh/JvPVBvaG7Si9KorP86GCbaQarcI3XDDUX8vbu\nYNKN1hgX7nEjypO+HwxHsK++AwfiZfdki/yn187GrV88FSMqtfssKfRDUbQWtbhBUlnkJ4yvxD03\nnI3x1QlLSH9+okosGx8AZkrJP3rLOCHkxjHy7kAYpUV+DBuU2s1oimOL3GHauonVbbX8q5UM2X+s\n36/5O2kal3S9Glu1Xh5Di9ygUxkRH2SJ42rpCCAQimDS6EFq52J07NGogkdWxMqxHq9zpcvHmWml\nxJIiv5r1K/abqq0IRQkhFyIsnuW3PqjXvK/HqACLcG22dQZRWODVVC88b8ZoAMDZ02tSHn9psR8X\nnTE2Np1M2v7wwcX41CkjccOl0/HpWWMNPWHyYFoUbEkXl/7J12ah9uzjAGgtbw9i11A8k7OmVuHB\n2+ZhwsiY9ZuwyCMo8HtVAZBJJaoix0g/GM1kRgyg7V8ONHSgoTlxz8rz42U6uoO495nN2BnPlzDC\n7iIvwiJ3w7Ve4POqfaycAyJmSKVCDCLElN5QJIp/bTmMD+N5HaIGQ3t3EA3N3Sk9NQW+2EBiwLrW\n5U7F6Efqk1rCEQVPvrITP/3jRtz1u3dTbldY5Hr3o55jbb1JAifmvk6QXEuCPUc68F9PbFCzlRMW\nuReVFUWYHp8vK1OiJsgkF5GISO2NGFSW+P36TOaOriCa2noxuqoMI4aU6JuqdPWGEI0mRF8mEIqg\nuzeM0mK/4bED6cVNWNF21yM38Yxbbm+m1kbWvr66mhUhnz5hqGnbgNRh/Ofj2uxkfdwOMHaf3/KF\nmIUYjZc2FWVFRWKlZv/SyXvrgyNqRbY502tw/WenJe0byHzcVFzoS3gP4tc51UqEYtqnWK5TCIR+\nqlcq17p+NT8gsXAKEOsUK6WiR2dPr8E918/BojnjM/glgF8agAw3eW4EI4eVSc9wZh3xqOFlqoEg\new/GjihHJKrgYNwQEBa3+I4Y4AXDUdWCLNN5GUQugj4MsGpzrCqlXjiNFggx6mvlcs03/uJ1zbOR\nqjLlc2s+Rt2eZjyhm/Wi2a7FedtArK8U941TizwUUeCPD4qAzAcWm3Y2qpa20KFQOIrHX9qOX/15\nC5rbe9VB4e5D7fjBI+vU6od6CvxeFBV4M1tN0iJZIeRyFriRK1IvCi0dAbz2Xmy1In1lLP33AGDk\n0LKU3wGAdVsbkmIwIkY+1SDTUiQyiO+oFrk/tTUqOiy5kMML8bh0ICzckKmEPGF56EemYjAwebR5\n8pKixJZaXLc12c0aDEXQ1RNCWUkBvjx/Mv7t01OSvxM/xh37WwxrfKurVtmstZ6u+lr6DRhvK/a3\nubX/6+e0sUUrrvV0a4TLHUaq+gMlGiFPvv+FGKzefDhp2dmk44kqiCoK7n16E/78RqJs75DyIpx5\nYrV6n8ideKaDrkK/V7WgRQcrW+TDpJLEYhBdViIShGL7mDpOKzzyIP3bnz9ZfW00t1t+r6mtV1MC\n2ePxYPiQkow9QsLVXFjgVb0M6ZgVX31R7wUxOlZA60mU24ytjnn5fvXnLbH28XPk88bmS4h7JhyO\nqln/+n10xi3yxWcfhzFV5Wq44KODsb5JeP5mTa2Cz+sxvMZG067UXJpoct5QqjXqd8X7w3KT82hn\n6po8Hc5xjDy+TLR+sCQweuY3f9SE5X/7QF0iV/w+OaRy+0Nrk7wdGw2MJQBqKEpvOLhBdgi5Pi6D\nmCtrdHzRg06bsZmWjgC8Hg+qKs1H3L3BSMp415iq8qSHaF99YsQVjSoJIfemPp2iA5TXFK/b04z2\n7qB6k6ayTkSSDpA89U0sn6ifbgYkvACC3zy7xXD7XT1hBMNRlBX7UVjgw+nxuJ5MMByFoij4+Z82\n4f6/GG8HsF9rXcaeRS651pM+M36dCisWuX7gqW9rmuymJnwlrpPRvuWpkXLnduuXTk36blSJJVjV\n7W3RLK0pktuMLJIvzT8+5THKeDwe1QIXQiFb5MfVJLxXY+MhKeHqFoPDaToPxnBppkeVZBkbCbJ+\n3nilrgyxFcR5GD44c/G/4sLj8elZY7Fk4Qma92//t9PTth0qhaxGD9eG66aOjQ1uPB4PCgq86j0V\ns8hF8qpWJIXgVleW4M5rzsTC2eMxuLxQTYiTcxT8Pi/CYQXrtzXgew++ra6d3iVVoFNLl6oWcPp6\nFT2BMFZtPqRWKNTXspfvsYCNKVdtXamz9AFg864mvLYx9fKzTa09uGXZv/D+7mPxGLlPHcDpy6ca\nWej7G7SWdXn8GnRJQhyJKqrQC15NsSRugRSjB4ALZ40x/J4dskLI9W6XmSeMwL03noMZk2OrJw2W\nMmxTjX6BWNzo50++h827mgDEhHxweaFG5PTV1mQmjx6MmVMTIlbg96JyUJGhm0/dZzxpBdC66/QI\nq0vv2t6+rwWBUDzxJ0W88IwTRuCL58dKRuofMBErM3KdydPhAGBdinniYq6mcN8ZxSc7uoJ4b2eT\n+rd+BCv+tOq6SrR3lk6eaRw8k91Yscj1swj0QvzHuFvOCKPMbWHRiytw6xdPNbweADC2Kjl/4+dP\nvoctHzUlvS+sTiEM8gDktOOH43+/Pz/lcZ49vQZXXBgT+/E1sXwN4amSvUjHSVntN1w6HQAworIU\nj91+Pi6IL2Cif/6EgADmzzYAVFeW4ooLEoOOwgIfvnrxVHz3yzNM2xkhhE7efzqKC/244sLjMVg3\ngKjKwDUPAL/69jn40VdnYphun9MmJLwUBT4v9tV3IBqvfieEJ9W5kQdSQ8qK0NYVhKIo6vUtjNeo\n39fQgYdeqENTWy9Wx93vwj1/6dwJ+NFXZwEAeuPnxciwCQS19/oTL23XVMls0xWUkS14O3HhNnm6\nneGNGtQAAB6zSURBVMHAYtmz7+PJf+5Mae2/9t5BdHSHsPzZ9+Pn0qP20foaJUbHp+/DxOI5dl3j\nPYGwpq5CeXFmnqBMyA4hD2lvGpHh+ZlzjsPFZ47DLZfHYoTjRpRj5PDUbvI9R9qx40Arlj37Pv6x\nfj+OtffG5q5KN7s87eZTJ4/EuaeOUv+uKC3AjZ87WY07VVYUwevxmGZ8dvWG1JverFhCSQqRPtbe\nm1jhKcV3PB4PLjpjLIBYwpQcL9q0KzYw0I/YAeDaz0zDmKpyLDxrXNJnMqL0o+gsyksLkjq4h/9e\np6lgpvcMiBKOYjBgVBAhU5xOEUtexlT7qR59SMNK4RX9yF52jR9r6zWNDYYjyRZ5V28YG3c0wuPx\nYPLowZg+cVhKi9EoKetQUxceN4hVVsTv+898agIA4ItSLXEjROIVAJxz8kh8elbs/pt7ykhc/9lp\n+MbikwBohWSslBhaIT1n+uMXg1JA60ky+j169Mv7njdjNE46bmiKb6dGeCusCLkRo4aXGT739930\nqaT3hpQXYdKowZp1EL48f7Kmapw4rr+sioVFCgqMY+QCeXrf4PJCBENR9AYjaiis0O9NEuUP4s+q\n2FdZcYG6HSFoPQZeUL1gfqyr2KavDCd7hMxc6+u3NRgW5JEHBr3BMD4+3I6r735dnZsu0CeRCkRf\nH4l7TQv83pR9uZHhoX/Pg5h7XLjGUw2wU9HY2qOZvWBkfNklK4R82TObNX+LG6/A78MX50/GyGFl\nWHbLXPz4a7PUxRRkhBUki4dY1nPYoGIUFcrlFb244+uz8MMrZ+LqRSfiU6eMVD8TD4t4MEXymZml\n3d0bVqfMmRWWKDFwfQOxUedf3tit2a8R3vi82UNNXbjxvjV4+rVdOHC0Ux1dG7nWRw0vw53XnIlT\n456NVIiRr/A8eD0eLI5n2woO6hb4kLNm27uCal16cQ6tTlVxGiM3s8g1bneD+Hk4rGDS6IRwWXWt\nez0eLJw9Lt428ZlRrExOdhQFP/QJY79+7gNEFUUVtlS5E6LdacebX18gIZITRg7C/35/PqZJ4mf0\ne4+rqcC3Lp2OkycO02S8ezwenHlitdoJTYi708tLCjB9wjBcOGsMvnfFaabHcrE0sBTW7JQxg1FW\nXIBvXTodd11zZurfYbGyWDqGD87Mmk7F13VudiAWlx5clnqevpwXkCquLOZ3Cw+KiKPLCGERiH2+\n8OYeKfSRfO/sPtSO5vZe1UVcVuJX7yUx8JSNBTFrQm+1ysbD8MHF6OoNa5KG5f441YD2aGsPHnqh\nLqlMLaAdGLR3BfHfv48lNj/zurY+upxZLyMnGIYiUfh93pTz6cVAIxKNYl1dPULhiKYPO2XSMJx5\nYrVm2ma6JOpra0/S/D1twlCNN2biKPO8JitkRYlWffBfrmAkEMkiRjd+Z08Ig8oKDV0e584YBZ/X\nC48n1okXFvo0sTx5dCw6J+FaFbHJdBb50ZaYiOmnnMlUS1PgzjqpGpedOxHfe2itpgZ3KotcII9q\nX9lwQNPWrGztmKoyTRnQu6+bjR8+8k6SC1keIZ5zcg3CkSjq9jRj065kV608yt91MDHtRHQ4XVaF\nPOUf1reQ1Nxke73BCKKKop3LbWH/wVAExYU+XH7eZKzZfFhzHd764EjS97/1qzWYNbUKU8YOQSAU\nm16kjy0KiiVr7PrPTsOx9l510Ack7s9ra6fhR4+u01QilPmmrkPRI+6Dy8+bhL/EczhGV5Vh1gkj\nTOuZA7F75r5vn4NAKAKv14OvXJicKGnG+JoK/OirMzFqWMzTlm5/wuo/dZLx7AqrmA2+M0FYZT+7\nbjZ27G9FeUmBYUEZGdlbUV5iXphHfNcoH0CB1tshvBWvbDig1s4oTLEc8uGmLlVoy4sL4PV4UFzo\nU59r8f/is8fjgtPHYOkDbyEQiuBQUxfe392Ei88cp3oZx9dUoLqyBE1tvWjvCqr5AHIMviNFgtcu\nkylrwiIvLPBq+hMF2vDXvoYOTUhUENAkdMYGPanqJ4iBz7OrP8bL7+zHwsZx2HEgEfv+2sUnoKjQ\npzHqRg4rSzJwZp9UrSYUl5cW4PLzJmHTR01YsvAE1AwtxWnHV8X7nOQCS07ICotcT6rsSACGGdVb\n98VcLUZzykUsryiFi0qOfwvXsrBQxEMqLn61QdLcr57Zgi0fHYPP6zF108lux2/WnqR5mAWpkt0y\nYcbk5BtZUFpcgPslV195SSFGDU8edMjnxuf1Yv7pYzQDEBn5wdohPYwJi9yqa12Bkzw5BYm4cpLV\nbdJODG7k327mWg9Horj7jxvVJJtgOKpaPV6vR+1gWjoCOHC0E9MmDMX/3Kx1s767oxF/enUX9jd0\nYuSwspRzfGXL7cwTq7HwLOOpVUWFPpwwLrV4HFdj3mGI87VwdmL7Q1MsJGLE4PIi00GsEUsWnoCr\nLoo9y5NGDU7psdIzdFAxfv8fC/Ctz52c/ssZoI9XZ8p3Lj8V580YpeYMVFeW4txTR+H0KVVpf4uc\nwW5WLx6A6jGUp8jNNEhGBYDBZQmx3x3PJE81PXBtXQNefy9WRVPMihlUWqhmsguLfPjgErWPDIQi\n+Mmj7+Avb+zGjv2taGzrhQfAD6+cqe77nqc2AYiFeJY9+766v7ZO40GmXJpX1OUQCIu8WndvHWrs\nwpoth9W/9a52gT6JusDvTRmm6u4N4bWNB/HyO7G6EC+t268pTibXMBDoa8cvPns8vnrxVPXvipJC\nLJw9Hj+8ciZGDiuDx+NBeUkBrv/sdHzr0un2a24Y0GdCvmbNGixevBgLFy7EI488YqntNYtTWxBD\nyouSYmmP/H0r3v7wiHrhb5QecuF6EhdV7/KSrVDhKjrvtFhxiYvOjMUFC+Kj2uIiP+698ZykY+rs\nCeH0KVUp54EDMdf6BTPH4HPnToTX4zGMB6azyMV6zvo62VdeNCVtvEUueFNS5FMzZfXHqEdM4dOz\n7Nn31YddLgTh83ng8diwyBXJurCh5OGIvN506oow8ifr6urV6VzyQEsxEfKjLT3YebANT/5zJ97+\n8AiOtvSoUwq9Xo9aE+DVjTFvySmThqGitBB3feMsw+1NHTckaYEQwRiDYkQ/u2624XdnnTBCM9CU\np4Wlc0cbJfeZJYW6wdxTR+H80+1l7VYOKjZdBMQKQ20WQDpl0jB89eITDNcaT4fX68HtXzkN5502\nGuOqtdf4hHEJcZg1tUqdLign1KWqky/3bbsOCiH3akIZgrV19WjpCGDCyAr13i8p8qOtM4gVb+9V\nBb2ipAB+nxc+r0cjtPc8tQktHQGMqCyJrbkevxxHW3oQjkTxgW6K6sHGLnxn2b/wq2c2a2rCy8aX\nyAsAYhnje460w+/zaEqfCuQkuz1H2tUa8W99cAT/iou83p0vF9bS84unN6vzxY0wClGUFvnxn0vO\nwPWfnYbf3HouPn/uJI0xZhQO6Sv6ZE/BYBA/+clP8NRTT6GqqgqXXXYZPvWpT+Gkk4wFelBZoSYe\nkm5OtNHD8+jKberrKWMHY8klJ2ge0ovPGoeX39mf5IKRtyWqmp0+pQrLvzNXtdIumDkaH3x8DBed\nMRaVFUW4+7rZ6OoN46V1+9Ryq7N1yxQaofcmzJsxSo1xA+YrFAHAtZ85Ca+/dwgLzhiLv635WHWt\nHz8ms1WF/uvaOTh4uA0ejwefO3cCjrX34uKzxuHuJ98DYBzHSlWIIRCM4Mb71sSmtkhxse7eMMqK\nC/DRoTYcbelOstQaWrrx0cE2TBw1CCOHJRIXY27C2ItUMfJgKILlz76P48cMwZbdx7DwrHGYdcII\nhCNRdHQFMWxwMZraes2nnykKPj7cjlc27Mf6bYk1ms86qRrTJw7FHY+tN7XI5QQccc+J8+b1eNS2\n2/e1wOf14LwZsWTK0cPLsOSSE/D4/2kT0RbOHq/GKosLfZg5pUpdhW7ciOSOJ1WG9IzJwzFj8nDc\n+cQG7K3vwPFjBmNvfQe6esNpPT1GP3fooL4V8v7mzmvORGNLT9qa833F1HGVSXPqAeDfrzgNrZ1B\nvLrxgFoVDkjEqQGo03L1yP2HqHFR5Pfhs+dMUC1NPTdIlqGY+//cmo8x//SYMSM8FoUFPsPQpTAg\nZCNmz5F2tYbBFRcej407GrHzQCvau0P4cE8z1mw5jLNProHX41Er/AHAjv2t6AmE0doZwH8+vgFA\nrODShyksbplfPr0Zv156Lh57MfZMnjJ5OF7VTU1bcKZ2QKPvu/RUVhRh6eWnoqs3ZGg91wwrxbjq\nCoxLMUCoSBM2cZM+EfItW7Zg/PjxGDUq1olddNFFWLVqVUohf+C752PnnmP479+/q9YbNuNrF0/F\nQy/UYeigoqS44HWfmYaK0kLMPWWU5v3PnzsR55822nSqiBzbkl2tp0wajgdvm6daNkKcPn3GWFXI\nTzzOPC5mxFUXTcW2vS042tqDGZOHqws/pKKsuEB9uGdMHo5XNhzAuOpyjTVpxulTR2Ds0NjvLy0u\nwM3ximHfunQ6/v7WHpw2JTlp6ssXHI9AMIJrFp+I7z2YXIxEPAg+rwcnTxyGE8ZVoiwej//+w+uw\n/Dtz8drGg9h1oBVXLpiKHzycqC98/82fQlmxH8FQrOa0CGHIBuKaLYex90g7zp0xSp0fXbc3Frv6\nzfMf4sbPnYwJIyugABhaURQrEBRvrygKXn/vELbuTcS6HnqhzvDcDCkvVD0iwkINBCOo29uMSFTB\niCElGFddrlm4Qo8v7lpv6Qhgz5EOjBhSonpzAGDuKaOw53C7Wn0LiFlRg8sK8e3Pn4zjaiqwv6FT\nFfIJo5I7CBHLTGVdiA41ElXw02tno70rmNZbI3sgLpw1Bu/vPpZU9GSgMaaqXFOfIVvweDyorCjC\n5edpZxUUFfhwwcwx2LD9KCalSJIyEpTCguTyroKFZ43TJPvJxpRwuwtjqKjAi57Utz4umT0eqzYd\nQldvGD/743vq+xecPgZ7j2grEL7y7gGs0C0ItPjs8Vj59j7ceJ92lb4vnDcJHo8naZlbI5ZLrvyl\ny99M+lz/HFQNKVaTdI2465qzUj4740aUpw1ZpTrvfUGfCHl9fT2qqxMWanV1NT74IHk1GEHloGJM\nHDXIdC6rzJknVuOMuCV235+34KND7ag95zjMOak6ZblFv8+bUsR/eNVMBIIR0+kvRu5JsT2Px158\n2+v14O7r51huB8Rcsj+8cqYao3OCWVJTzdBSteDFl+ZPxocfH1OFVObs6TVYcsmJALSZ2Tfd/y/1\ntSziAPCdZW9i0qhBaiLMoLJCtHUGUbenGc+t+RihcBQvx2uZr9p82ND1/9ALH+Kz8SlVlYOKAbSp\nySYHGzuTklFS4fF41JyIXQdb8adXdyYVdigq8KkJNDd+7mScMmkY1m2tVz05Xo8HR1t61BKqRlUH\n5TnI31h8ovpaFOGprCjC1y6eiurKUs20JJkHvnNuysVt5p4yEh8dasPsaTWoKC00zMUQFBZ4EQxF\nNZ3VVy6cgq9cmLIJ6Ue+cuHxuOLC4+H1eHDbl2YkiUxlRRF+vfRcvPDmHtVbV1ToSxmLHatz619+\n/mRNNUBAyhuKj/X0xpPo90qK/PifW+biGz9/Q/3saxdPhdfrSQpfGHn+zjl5JFa+vU/z3r03noPK\niiJc95lpeOuDI/B6PSgr9uORFbFlV39xw9l44c09eDOeVLp9f+rEOTmpUfyGuaeMwuothzWrkN3+\nldPw8z/F4vxmA+AfXDUz5XldOHscOrqMrfi+ok+E/JP4AR6PBwV+H773lfRVldKRzpWfiiHlRbj1\nS6eqGbefJB6PJ+2a0m6z4MxxWHDmOPxtzcfo6A5i445GNVlMtqIvmT0eT726K8VWYklp4uu7DydG\n66dMGg5FUbB682HNiH3ciHLUt3SjJxCG1+PBSRMq8eHHMXdbJKrgb2s+BgDNgi9GpWj1/MfXz8C+\nhg7VMhOuyZ5AxLA6k5wFO3RQEQr8Xo3np6qyRK3RDxivJCcf49nTRyZ97vF4MC++AEgqzFaSmnvq\nKEwdX4mqDJK47r5uDuqPdduOE5NPFo/HoyZ06ivkCUqK/Pj8uRNVIRfW+xUXHo/Nu5pw+fmTcOcT\nsWlck3WW/YIzx+L800Zj3dZ61O1pxqxpI9W+fMEZY7HzQCs+O3eC2n5weSG+uiCR3KUPeYp9nzxx\nGF5cGxPpc08diTVbkmdzVFeW4p4b5uA//ncDBpcV4o6vz1IHCaXFfnw6XkcDAM48qRqe+Pn4yqeP\nx+Qxg3HyxGHqABqIeZb2HGlX1yOQE0X//YrT8OqGg7hg5mh8+owx2LijUR1I+31enDJpWMow581f\nOAVHm7tN80703pRPAo/itKSWAe+++y6WLVuG3//+9wCABx54AD6fDzfccIPbuyL9SDgSxR0Pr8UH\nu5tw0VnjcdMXYxW2FEXBsbZeDC4vwlvvH8asE0agvLQQuw+2ojsQxsmThqO5vRdNrT3YsLUBiqLA\n4/HgsvMnw+/34h/r9uGvr+1ULdo/3bUQPb1hbNvbjGkTh2H4kBL88o8bsXpTQmwHlRXioe9fgK/8\n5CX1vTuuOQt3PpZY2eiJOy5CeWkhPtzdhMljhiRV6AKA/11Rh+dWfYSFZx+HUyYPx/DBJRg9ohwN\nx7qx9P7V6vf+/svPJA1Y99e348ZfJCySn33rHEyfpA1X1H18DN//dcztt+Lez1o+54RkwjOv7kBx\noR+fPTc5VPnMP3fg48Nt+P5Xz7BldC17ZhPe3HIIT965UBM6AoCH/vY+XnxrD0YMLcXD379ADfXs\nOdyGkcPLUOD34Ye/eRNb9zTjwjPG4cIzx6GspADHxQsQNbX2oMDvNXw207FlZyM+PtyG2rkT1f1G\nowoONXZirEmi20CgT4Q8EAjg05/+tJrs9oUvfAE/+9nPMG3atJRtGhuNV4wh7lFVVeH6ee7oDuK5\nNR/j0nMnup44tGN/CyoHFadc1e1/X9ymutWuWXQizjl5JOr2NOPeZzZjfE0F/uPrZ+Dqu18HAHxz\n8UmYY7LEpSAUjqKtM2AYovn4cDv+8I8duPFz0w0/j0YVfOOehJAbhYqa23vx3d+8nfJzkp6+uI9J\nMnbOczgS1cwlNyIaVdTaH/lOVZU7A4w+ca0XFRXhrrvuwnXXXYdIJIJLL73UVMRJ7lJRWoivXpxc\n3coNjLJ6Za5aMFUVchELPum4Stxw6XTNNB4AKMmwHGKB35syz2LiqEH4jyVnpGzr9Xpw27/NxL1P\nbkz5ncqKIpx/2mjb4RxCshm/z5s2VOP1eijiLtNnE93mzZuHefPm9dXmCdHEsURhDY/HgzMMEvdS\n1bp3m5NSxC4FHo8HV0lxRUIIcUpWlGglxCnpiul8UhmkIypL8cXzJ2PiqEHpv0wIIS5AISc5zW1f\nmoFt+1pQk6KU7Jxp1Vhb14BRJqvmuY1RJS1CCOkr+iTZzQ5MXul78jFJKBpVEIlGk7Jr+4p8PMef\nNDzHnww8z31PVie7EZIteL0eeL2fjIgTQkh/kJWrnxFCCCEkMyjkhBBCSA5DISeEEEJyGAo5IYQQ\nksNQyAkhhJAchkJOCCGE5DAUckIIISSHoZATQgghOQyFnBBCCMlhKOSEEEJIDkMhJ4QQQnIYCjkh\nhBCSw1DICSGEkByGQk4IIYTkMBRyQgghJIehkBNCCCE5DIWcEEIIyWEo5IQQQkgOQyEnhBBCchgK\nOSGEEJLDUMgJIYSQHIZCTgghhOQwFHJCCCEkh6GQE0IIITkMhZwQQgjJYSjkhBBCSA5DISeEEEJy\nGAo5IYQQksNQyAkhhJAchkJOCCGE5DAUckIIISSHoZATQgghOQyFnBBCCMlhKOSEEEJIDkMhJ4QQ\nQnIYCjkhhBCSw1DICSGEkByGQk4IIYTkMLaF/KWXXsKiRYtw4oknYv369ZrPHn74YSxcuBCLFy/G\nm2++6fggCSGEEGKMbSGfMmUKHnjgAcyaNQsej0d9/8MPP8SLL76IFStW4OGHH8aPfvQjhEIhVw6W\nEEIIIVpsC/mkSZMwYcKEpPdXrVqFBQsWwO/3Y/To0Rg3bhy2bNni6CAJIYQQYozrMfKjR49ixIgR\n6t81NTVoaGhwezeEEEIIAeA3+3DJkiVoampKen/p0qWYP39+xjuRXe+EEEIIcQ9TIX/88cctb7C6\nulpjgTc0NKC6ujptu6qqCsv7Itbhee57eI77Hp7jTwae59zAFde6oijq63nz5uGVV15BKBTCwYMH\nsXfvXpxyyilu7IYQQgghOjyKrMIWeOmll3D33XejpaUFZWVlGD9+PJ5++mkAwEMPPYQXXngBPp8P\nt99+O+bOnevqQRNCCCEkhm0hJ4QQQkj/w8puhBBCSA5DISeEEEJyGAo5IYQQksP0u5CvWbMGixcv\nxsKFC/HII4/09+HkLI2NjbjqqqtQW1uLBQsWYPny5QCA1tZWLFmyBLW1tbj66qvR3t6utvnv//5v\nXHLJJfjc5z6HrVu39teh5xzRaBSXXXYZrrrqKgA8x31BW1sbvv3tb6O2thaXXHIJtm/fzvPsMvfc\ncw8uuugiXHzxxbjxxhvR2dmJAwcO4Etf+hIWL16MpUuXquW1g8EgvvOd72Dx4sX48pe/jEOHDvXz\n0WcnP/jBD3D22Wdr6qzYuW+fe+45XHLJJbjkkkvw/PPPp9+x0o8EAgHl3HPPVQ4dOqQEg0GltrZW\nqaur689DylmOHTumbN++XVEURenu7lYWLFigbNiwQbnzzjuVhx9+WFEURXn44YeVu+66S1EURXn5\n5ZeVa665RlEURdm0aZNSW1vbPweeg/zud79Tbr31VuWqq65SFEXhOe4DbrrpJuXPf/6zoiiKEg6H\nlc7OTp5nF6mrq1Pmz5+vBAIBRVEU5ZZbblGeeOIJ5brrrlNefPFFRVEU5Sc/+Yny+OOPK4qiKI89\n9phyxx13KIqiKCtXrlSuv/76fjnubGfDhg1KXV2dcv7556vvWb1vGxoalPPOO0/p6OhQ2tvblfPO\nO09pbGw03W+/WuRbtmzB+PHjMWrUKBQUFOCiiy7CqlWr+vOQcpahQ4di6tSpAICSkhJMmTIFjY2N\nWL16NWprawEAixYtwurVqwHEauIvXrwYADBjxgx0dnaylG4GNDQ04I033sDll1+u1k/gOXaXlpYW\nvPfee7j88ssBAD6fD2VlZTzPLlJVVfX/27uXUPj6OI7jb5dJMjPkEiciUsolKckoRhILFmxZKZKN\naVw2spIFkZSQsrK3cSkUNcplyDWLsVCImhGRySUMz0J/j/nn0Z/m6TzH830tzyx+3/PpN79vZ870\n+xEYGMjd3R1PT0/c39+TkJDA2toapaWlAJSXl7+tx++zLykpwW63e+0fIl5lZ2djNBq9rn1l3jqd\nTpaXlzGZTOj1egwGA7m5uSwtLX06rqqN3Ol0eu36Fh0djdPpVLGin+Hk5ITt7W1MJpNXxu/zPTs7\n88o+JiZGsv8D3d3dNDc34+//91dHMvato6MjIiMjaWpqoqysjNbW1rdFTnL2jaioKGprayksLCQ/\nPx+j0UhqaioGg4GAgADAO+P32et0OgwGAxcXF6rVryVfnbcul8vrvJLfd0v9iKqNXPZg9727uzss\nFgvt7e2EhYWpXc6Psri4iF6vJz09XZ5G/kXPz884HA6qqqqYnp4mODiYoaEhtcv6UY6PjxkZGWF+\nfp7FxUUuLy9ZXl5WuyzxTao28t9PRnO5XCiKomJF2ubxeLBarZSXl1NSUgJ4P524XC5iYmKAj/fE\n//WZ+NjW1hY2m42ioiKam5vZ2dmhoaFBMvYxRVEIDQ0lOzsbgOLiYvb391EURXL2kd3dXTIyMggP\nD0en01FUVMT6+jputxuPxwN4r8fv5/jj4yNut5vw8HDV6teSr6wPiqJ8az6r2sgzMjI4PDzk9PSU\nh4cH5ubmKCgoULMkTWtvbycuLo6ampq3a2azmampKQCmpqYwm81v16enp4HXBhUSEvJHh9v8n1ks\nFmw2GwsLC/T19ZGZmcnw8LBk7GOKoqAoCg6HAwC73U5iYiIFBQWSs4/Ex8ezt7fH7e0tLy8vrK6u\nkpSURE5ODjMzM8Brxr/WY7PZzOTkJACzs7Pk5OR4vV4S/+yr64PJZGJlZQW328319TUrKyvk5eV9\nOobqW7TabDZ6enrweDxUVFRQX1+vZjmatbGxQXV1NSkpKW+vLCwWC1lZWVitVs7Pz4mKiqK/v//t\nzxgdHR2srq4SFBREZ2cnaWlpat6CptjtdgYHBxkbG+Pq6koy9jGHw0FbWxv39/fExsbS29vLy8uL\n5OxDAwMDTExM4O/vT2pqKl1dXZydndHS0sLNzQ3Jycn09PSg0+l4eHigtbWVg4MD9Ho9vb29xMXF\nqX0L/zmNjY1sbm5ydXVFREQENTU1VFRUfHnejo+PMzo6ip+fH3V1dVRWVn46ruqNXAghhBDfJ7+N\nCCGEEBomjVwIIYTQMGnkQgghhIZJIxdCCCE0TBq5EEIIoWHSyIUQQggNk0YuhBBCaJg0ciGEEELD\n/gJhM/qc93OLwAAAAABJRU5ErkJggg==\n",
       "text": [
        "<matplotlib.figure.Figure at 0x25ac8610>"
       ]
      }
     ],
     "prompt_number": 15
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "#How correlated are CDS counts and exon counts?"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "updated = pd.read_csv(\"/nas3/gpratt/projects/public_riboseq/analysis/public_riboseq_bam_v3/updated_counts/SRR334701.polyATrim.adapterTrim.rmRep.sorted.rg.rpkm\", \n",
      "            sep=\"\\t\", index_col=0 )\n",
      "original = pd.read_csv(\"/nas3/gpratt/projects/public_riboseq/analysis/public_riboseq_bam_v3/SRR334701.polyATrim.adapterTrim.rmRep.sorted.rg.rpkm\", \n",
      "            sep=\"\\t\", index_col=0)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 16
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "combined_counts = pd.concat({\"updated\" : updated.RPKM, \"original\" : original.RPKM}, axis=1).dropna()\n",
      "#combined_counts = combined_counts[(combined_counts.original >= 1.0) & (combined_counts.updated >= 1.0)]"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 20
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "loglog(combined_counts.original, combined_counts.updated, 'o', alpha=.05)\n",
      "xlabel(\"Original\")\n",
      "ylabel(\"CDS\")"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 21,
       "text": [
        "<matplotlib.text.Text at 0x2a533ad0>"
       ]
      },
      {
       "metadata": {},
       "output_type": "display_data",
       "png": "iVBORw0KGgoAAAANSUhEUgAAAgEAAAFtCAYAAACeOLndAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAIABJREFUeJzsvU2IXOl1//953u5Ldbek0YxmxiZgCFkkGC8CJgvjWMFg\nBwcH7MzaYCfEhmBmGxzwJt6EZOddTEwWCXiXQMgiOOA4DoHghcEYQiAE/Aebn8djz0jdVXXvfV7/\ni3OrVN3q1suo1d2Sng/DqLu6VH1vtehznnO+53tUKaVQqVQqlUrlhUNf9gVUKpVKpVK5HGoSUKlU\nKpXKC0pNAiqVSqVSeUGpSUClUqlUKi8oNQmoVCqVSuUFpSYBlUqlUqm8oNQkoFKpVCqVF5SaBFQq\nlUql8oJypZOAYRh44403+P73v3/Zl1KpVCqVynPHlU4CvvnNb/J7v/d7l30ZlUqlUqk8l1xoEvCV\nr3yFj3zkI3z84x8/9vj3vvc9Pv3pT/OpT32Kb3zjGwD813/9F7/6q7/KzZs3L/ISK5VKpVJ5YbAX\n+c3eeOMNPve5z/HlL395+5j3nq9+9at861vf4tatW7zxxht89KMf5fvf/z7L5ZL/+7//49q1a/zW\nb/3WRV5qpVKpVCrPPReaBHz4wx/mJz/5ybHHfvjDH/KBD3yA97///QB88pOf5Lvf/S5vvvkmAP/4\nj//Ir/zKr1zkZVYqlUql8kJwoUnAafzsZz/jtdde237+2muv8aMf/Wj7+Wc/+9lHep1SCkqpc7++\nSqVSqVSeVy49CTivwK2U4u23j87ltZ5Fbt06qPdf7/+yL+NSeJHvHer91/s/eKK/f+nTAa+//jpv\nvfXW9vO33nqL973vfZd4RZVKpVKpvBhcehLwoQ99iB//+Mf89Kc/xXvPt7/9bT72sY9d9mVVKpVK\npfLcc6HtgDfffJMf/OAH3Llzh9u3b/OFL3yBz3/+83zta1/jS1/6EiklPvOZz/DBD37wIi+rUqlU\nKpUXkgtNAr7+9a+f+vjt27e5ffv2RV5KpVKpVCrPLD4kcikMU6Rv33sov3RhYKVSqVQqlUfHh0TK\nBZDJuCfh0jUBlUqlUqlUHp38hIF/l5oEVCqVSqXyglKTgEqlUqlUniH0ORrj1SSgUqlUKpVniMaZ\nc0sEahJQqVQqlcozhnMardQTJwM1CahUKpVK5RlDK0XbGLonGA+EOiJYqVQqlcqVYuMBABLsG2ee\n2veqSUClUqlUKu+BpxGsdz0AAFIpTD5ty//nTU0CKpVKpfLMMoXE6CNwfyB+mifqpxWsT/MAyKUQ\nQqZtzr8iUDUBlUqlUnkm8SGRUqYUKAVSlkCcS9kG6dO+dh48KFg/S9QkoFKpVCrPJA8KxM9qkD6t\niqCVwrmnE65rO6BSqVQqlTM4q6WglSKdSDTOI1g3zhyrWGymAJ4WtRJQqVQqlWeSB52az+NE/aCW\nwsawJ8TEFCI+JpQ6Hze/XQ+Ap1UB2FCTgEqlUqk8k5x0ztucmjcn9rO+9qg8rKVQKIg2UGGNPjfd\nweZaH/d63wu1HVCpVCqVZ5bdYH/y1Oyc3gbsp3Wibk9MHDxNJf/ToCYBlUqlUnlm0fr+nvl5jQY+\nqO/vQ2L0iVJAa2js8dHEwpN9/4syDKrtgEqlUqk8N5znaOBZLYUYMykXtAIo5FyY5qAdY8YY9UTf\n/2mPN+5Sk4BKpVKpPPP42TRoPYlIb5cnGQ08TaS3CcbO3ksSSimkVHBOo0708R/3+1/keGNtB1Qq\nlUrlmea4e5+cmkcfcVZjtcbNpfrTSuwPK7s/bETPWU2Iefvxs0ZNAiqVSqXyTLN7cpZSvXzsQ0Y5\nBTFjrUKVeyf0VApHK48xanty31j/Fgp+NhzSCrrGHksOdrUCamcSwTkt3/8J/QOelgfBqd/r3F+x\nUqlUKpUz2JTtRx/xIT38Lzwm1mp2q/FKqWOn9V1iLvc9PoXI0dqTcpYSfy6sp8gwxW2y0TjRBYxe\nxIEx5nMdTTyP13hUahJQqVQqlQvhPAVvm2RimCLxRCB3VqO1BOTHLdHnUu67nlJE+LfpyfuQMEah\nFWgFxqhj93EeZj8XZRhU2wGVSqVSeer4kFhP8b6RuvcyV7+rASilYIyc9CUwS2RetMfL960znMw1\nrFYYc/x0vakcPIhcyrYNsPvY5j7Ow+r3adsFb6hJQKVSqVSeKveCdiHERC4w+kjrDK17eBg6Kd47\nrXJgjSKmQuvssa/vBtPTPPmHKZFLJuWEswYF5FSOJQdqTiKetoXvZVCTgEqlUqk8VTaBd1e0VwqM\nPqGVZtGfHYp2T/2SQBSmkGmdPpZASKDWtI05Nk63G7h3HQQLhZzBWcVqjIw+EaLsBMhAjhlrFFpr\n+sYeO5VfpHDvaVOTgEqlUqlcCNZqBh+2Qd0ajbOaFAu2keecdeoP8fgY4OgTCkWek4rdIHxWKX33\n8dFHQJKHUgrWyN9djx6tDTEmjFa8fL2/L7hf9Ka/p0lNAiqVSqXyxGyCtw8ZuLdlb/NnKgUfEzlL\nS0ApaJxmCmnrv3983l9G9nzIOKsYQyTPwwRaA3OCEGLaBuGHzfzvfn1znbkUximjlBj9lgLWSKA3\nWp2pyL+IvQQXQU0CKpVKpfJEbIL37ml99AlnNaVIkCwhE0KmFBEFNjuB31l9pnAQVViNkZyZk4h5\nNr9RFGQccOPlfzKBmHzaquxPfn01elZjxGgtxkJGfASMVUwhiep/p7Vw8qT/LJ/+d6lJQKVSqVSe\niM3peleQV4rM4GunCCHjnDj3pZy2ojulZJxvGCO6d5wlHPQ+E7cJgFQQKDDOQdyHNFcMjp/IpxAZ\ng2z6W60Dxsr3jSmzHmW8EMBoxRQyFLBFNACNM+QCh+uJxpn5+z77Qf8kNQmoVCqVyrmxCeIgwbWd\nrXkLBTYBfMZZ8dnfJAWnCQdjKqBFDBhCpGmtJBeAdZqQ0tZzoMxVBaXUtiqh5ipASJmQ5Xuup8jR\nOoAScaBuDMZohimgtMI5g1Ycu5aNp8GmsvC88Ow2MiqVSqVyJdgExU0Q3xQEtFasxoDS8tim978J\noTFllGLr8HfS7S+ljNEKH+TPprHkJFv7EnmbPPiYCDExTJHVGABJIHyU8b/RJ/R8DSFmvM/4GPFT\nwmpFyXJNrTXs9w0oGCcxBxLbAYWP0q44WoWn4nR4WdRKQKVSqVQem5MiPK0U1mrirN5rnZmV95BS\nwcx9/pQLv3h3TcyFRWcxSk7ey8ETkrgJ5pzF7W9OGsS+N5Hy7NA3n9azaPvEgthnhpAwSqoRKE0/\n9+xTzqSUSamAUqynSGM1paitwDClwv5ey35vRfA3OwGWAmMQrQJF7lWpstU6POtVgSubBPzP//wP\nf/VXf8VyueR3f/d3+cM//MPLvqRKpVKpcLqKH6Bwr2++cd7zJ9bfxpRAKZxROGsIqTD6gNZSmjdG\nY6ymdYZcMjFltFZ0reNwOeITlKJQCvYXjegBfGIKCTIk4Bd3J/pWAw1WK1LOxFRQqtA4jTVQtEFp\nRU5F+v2NpXGaEAshZRRSiXDGMIZEydA1BmcVKRdGH8nF0LdXNow+Ele2HfDrv/7rfPOb3+Tv/u7v\n+Ld/+7fLvpxKpVKpzJzl9W9nY52+sbTzaKDVx214S4a2MSx6d6+NkDIhFtw8lidTBfNegVwYp8gv\n74wMU2IdAj5EysaAKGWmKMJAFISUGafAcpC+/+gT05TwMYr+QCv61uGMJscyWwpbFp2jsUaqDknG\nBbVRpAx+kkSmmasbm/fgtKVEzxoXmgR85Stf4SMf+Qgf//jHjz3+ve99j09/+tN86lOf4hvf+Mb2\n8X/4h3/g05/+NG+88cZFXmalUqlUHoKPabtFz0dpATT2/u13B3sNRkuoCTExxSwmQOl4ADVasd87\n9nuH1bKSt2usbPAbE4ericO1RxWFj5kp5PmEn/FBqgWlgA+i+M85U0omzNdpjEwn+JBlZ4A1dK1l\nv3c4p+kaMycuzIuHDEZJxUFrxcagaHO/j7Jj4FngQu/gjTfe4G/+5m+OPea956tf/Srf+MY3+Kd/\n+if++Z//mf/+7/8G4A/+4A/4l3/5F77zne9c5GVWKpXKC8fDVvzufn09BvK8CwDKbACUMVaduv3O\nOU1MmVyga0ShN06J1VpEdo3V3Dho5367mPw0jSEkUfVPIRKjfHw4TEw+crSeeOdwRGuFUUoEf/MU\nQD+f6pVSWG2kt58LEsoL1hqc0ez1jrYxW0OjjcHQJo9RStE1loN9dyy5KeXZtQk+yYU2Mz784Q/z\nk5/85NhjP/zhD/nABz7A+9//fgA++clP8t3vfpdf/vKX/O3f/i0hBD7xiU9c5GVWKpXKC8WDjHZi\nFHV9ymVr4hNTYTl4lJZNfF1j6Ro72//qrXvfJpnQ86k5hwRFSY9dieBvnCLXbu4RYiHpNAdqmIKc\nultrCM6AUqhcUEURUgKvUEoMf0JIrKYAKLwPOKPpZ9FhyoWumUWKyNhfCIG2s2gNMcqyoCnEeTWw\nXGtMBWfF1lih0L3alv9bZ5hiZLyTaBqN1Yr9RXPhP7fz4NIVDT/72c947bXXtp+/9tpr/OhHP+JP\n/uRP+O3f/u3Heq1btw7O+/KeKer91/t/UXmR7x2e/P6H6V6PfZcQxYBn9IlSCj5mvI+oxrBvpXTe\nOotzhr41LFrHondMITFsqwXStx9m8x/jMnsHHesxEFPm+l4rxjy5EHJGlcJLBz3rKbIcPDFlukVD\n0orJR1KCrrXsdQ3X9i0/f3eF6RwmJ0IouNYxxsy11nHrxoKUCovezclLluQFeQ1rNCkXrFZoo7c2\nwaUUulnwpzfJQxIjI5BthDHnY66B0s5osPcZFqXt+6Bn34SrxKUnAUqd33jF228fndtrPWvcunVQ\n77/e/2VfxqXwIt87nM/9jz5ymtbPhzz7+8e5WiBBbT0GjJaguegsSil+kQt7vYgCf/7uGh/yVtzX\ndZbgM8ZBYyxKiSdASplfjBFnNVpJRWIKmf/vJ3ewxognANJGGKdAjokxZKwqJKt45x2P0pq7hwPD\n3CKQ+ygcHg6sXz2gc5autbh5QVDMGWs04+ApMjF4z79gjkfOatZ6p5UR87FKyd2lRympFCx3Ytgv\nteba3r2KwMkKC9xrI5zXaOGTJoCXngS8/vrrvPXWW9vP33rrLd73vvdd4hVVKpXKi8XZq3HV9uPd\nWOaMJqRMKZo7Rx6toW8d6zGwHiLv3B0Js9HPNO8EaJ2hhExsRJHfJM2ic0wxbROAw0GU/woRHqYs\nS4daa7i+cAyTonHQdzJ9EFNGWwljuWRyVvO4ocJozdE6EBtpS+wtRHCojZzsndXEWHD23j1uxhtP\n7gU4bWvgo+gBTpuiOGsXwWVx6aqGD33oQ/z4xz/mpz/9Kd57vv3tb/Oxj33ssi+rUqlUXhg2wrgN\nmyBotZ5tgGWbX4iJkjMhixp/CoFCRmnwMZALrMbAFBIhZu6uPJNPLMeJn727xKcEBXKR0njMGWfE\nJXA5RFZrz9Faxvu8z0yTlPh9ECfCV24uuPVST2NEd9BYw6KztI2hAKgswkIror9xCqzHJE6EKIzR\nlFxorJZxxvaeIHCj9D8rwBfkOnzIaHN/cDdas+gv/Vz92FzoFb/55pv84Ac/4M6dO9y+fZsvfOEL\nfP7zn+drX/saX/rSl0gp8ZnPfIYPfvCDF3lZlUql8sJz1mrcXEQN3zjNevBYa+gUxFAoWZE3Ty0a\nP0WmkGRsb174I4I76b2vhojRZp7Vt7TWUBQcrfxWN7BcB4YpoSns945u4QAxEYohS4ugFGKUnQE3\nr3esliNv3x0oRaEV7PUNWitxCKSQKaSU0Y2hbS1xx8Fwo3uAszcD+pC27wFA4xqWQ0Br+TvmRBtg\nw9kVlks/f2+50CTg61//+qmP3759m9u3b1/kpVQqlUplh9MCYC7S9w7zzH6hQFHkUNC6YK3CGY1R\nUvYvpaAVoCFnuLvypJxorKFtLEorloOnc5o7R4WYHYvW4UPk3eXIOG/1SzmxDpnlFHifWbA4sIQg\nosC9rkUZ8EHGEn/29orBR4wq9M7Q7jeUIqZEfefYm30H2h1nv8aZrZnRopepBuDM4HxaWb9vDcMk\nExRnVQBOayNclTbAhmevdlGpVCqVx+ak1/9Za3F3n+dDwlmzFc7lklkNkZTBWI1NZd7YlxmnOCvj\nC3uNY/IRZxXFa1qrSDlxeORpW8MwGFwbWY+R/cXGl18RE0whsxojjVU0tpHRwVSYYqa1ShT+60Ca\nHftsaxjHxEvXFkw+SlugFFH4O0vn7LbdkYuMCN44aI61P+x7mO4zWnNtYR4a1M+qsDwuj/rze1xq\nElCpVCrPOQ/yAdgNhqep2ad5Pa9WiuUqcDRGjFEc3hlwzhJjpm0lGI6TfK3vLO1omEwm28ydZcBa\n8Q64s/Ts9S37xpJ05nDpCSkzTJGcC5MP86lfsddarHGkJAuEQoSjdUQpGV8cpoS2MuP/UtfR2IZh\nDBQUqEIpGR/i9h6u77fs944Y82MF1Ccp65/H6f9BP78npSYBlUql8pyyOT0erUTAtzkRO2tOVamf\nLHu7OXCHmIkp4VOmMbJWdwqF1TihTWYKGqU0RkNKarutDyUn5raBlAuFIqd8Z5gmRQxZ2gRKcbT2\ngGz0O+gdKM3hOuCjjAl21tGgCEnuK8YsfoUZpph4++7Afm9pe0fJmWEMJCVVhFzELdCHRCqZe0KG\nsxOiXS67rP+gKYMnpSYBlUqlcgV50vKvD4nBR7yXET2tROQny3nkT5Q6Uf7P9/nhb9zzcpaAjiqs\nxoC1MmoXk5TZnZPtfoVCTplpyqgCIH8np8gUMmRpKYCmayxNgVwSBwvH0TqSU8FoxdF6onVWkphl\nwF23LNeRkmGY/fv3OkfI0pJQClCazmjGKH4EISbGlOg7J18vhWFI7C9OGvpExsBWJ3Dae31eZf2r\nRk0CKpVK5YrxKOX7TfAeJjHyORm4Ri/l9VREGT/lwhgSrTMcLBpiEnHf6EXFDxBzohQjVrlKTvwh\nZIxRYv6TC1orrNG0ToKs9/KXcykyVmikb3+wcLx9GEkxM/nAOEXZKug0KBh9wBpFyprOWZTJtD4T\nsyaEhJqLCbmIXfHRciLmDAWstTLup2R0wWlZVIRSxAyhyJ6DjRNhSonGulPf6zD7ESglC4jOqgxc\npqjvaU4Z1CSgUqlUrhgPM5nZTRJKkd32gxfnPau1BL/5JULIaK0hJXKRHn8XM93CSiKxk2xYrRnn\n5EJrxXIdpP+/jqzXnjFEQiooFClnfCgsesM4Ro4Gjw8GrTV9owmIkO9w6fGpoBSknEFpQowkBatS\nKBniIqMBpTXXesdai3dA01hpMWQoczy2xtI4hcLQzNv/itH4CNf37bwOODNNkZQKfWswRmOMomsd\newt73/sqewUkIQIQs0B7ZZT8T7MdUZOASqVSecbYTRL8jqWtDxnlFJPf3QI4j79Z2eQXY2LtA1OM\nxFDoW1mkU5CxvtUYiDHTWMM0z+KnlEmlsBwjMWbZGKhBofEKjDWYIBMCPiZC59BasR6DOAvmgjaK\nBDRGkbRiPUZUyYQ8UmhAAVnRNBpjoOsci9Yy+EROmZACN/Y7Fr2VE3opHOy3vPrKPuvlhAack2tu\nrOXmdc04JXLOdK1m0Tpu3ejRSh0LqCFmSoE8L0hy1sz+Bg/WCZzF01LxP612RE0CKpVK5YrxOOXf\nnO+vGoxBgrWs+C2kkgkhERO0rWaaElppQhLdQN9KYA1z0O8aw3qKhJQZp0DrLKshEELC+4Rr5DSe\nk4ziaa3pW8svhjVTzCyHwN6ikcqCz8SQKCXPLoEFjUZrhdIQfeYIaQ20zhJTZq9xoAujn0vzVtNa\nTd9bnFM4rWlbx6KxNE7jtVQNYsrzzL+0MNrWELxi0Tlu3ey3AX0TUH2U1x4mSZpShjy3TKxRj23v\n+6hTGO+Fp9WOqElApVKpXDEeVv49LUlQs/WtOPWJyc8YIhkR8hkFsRTKKMG4cQpnFFPI/L9frmid\nzNvv71luuBYKeB+5u/SEIP34FAvOgDZAVmgNIcqUwOFq5GgVUEozek9MiThv6Os6y2rlGeYlRI2W\n9b7LVUBpRTGAMrhSCBEObjZy6lUJYxQlZ16+3onnvzbs77X0jcFqw/VFh82Fw6UHBYvO4mPCao3J\nhRsvNdzY77BaH3v/xGq4YItm8vdGBtX8Xr6X5XYn2zgby+UxKPrGnltV4DypSUClUqlcQR5U/t1N\nErSWgLU19MlsvfAHLwtiSi4oa3AGgpfTvlcJpWA9BFZDoKRMyTKqt1pHcslMIYGCkBIxicAwFmCA\nrrEcrTyowuALh8tAzIWYE5PPDJMnZlHwl1KIGXLOWEQYuJwiISX2jJ3bC2It/OqNltEnjIK+0dxY\nNGhVGIIkL9pobh500sOfIkfDxHIdiCWjisJasSTOGfpOb9+XB53q20ZvWygP2yHwqGwEhxtSPr+q\nwHlSk4BKpVK5RM7qIZ9W/t0814dMLnk7Gre7o15eQxb/wKyuN5r1EFl5j0YRUqFvDNf2W5QCo2E1\nJXLJkEX0F1LEKIW2Bm3AFoUPhfV6IrSOcUqEKEY8k8+sfWYaAz5mYskYpVFz718mBzSNleBcgIgi\njJGRhLUK01pevt6TcqHvFIdjwA8J18h4o3OGfdfwyktS1s+ANUZerEjygyrYeTnR7nKgs9hUVFpn\nUbNHwSahei+l990KzbaysLOc6KptEISaBFQqlcpT42EiscfpIW+ee/KEaU6MsSlVZO7dJ4YxbhcA\nFVUwSrOaAkZphilgnKZvDG2wWJtZD4W7q4mcs5TpS+bWSw3LdSJRUGqeGhg91ui5B59ZTZnlciSr\nIpMAuZBVRumCVQ6fNDYrGqsYQ5b9AjnhnPTdu9ZhjWKcxE1wCoU0FRLAPN2Qc+LVlyxWyz36KaNU\nYdE1rJcTViuaxlAK2+C/qYicdarfrahs/BBaZ95zBeBkG2e3QnNVqUlApVKpPAUeJcCfNgo4hsjo\nFW2jjyUO+cQJE2Q80M8nfj1XBXwQc6AxJGIsaMR85+5ykpl/rRh8RNnC3bcmyKCtxjrNOER8irIu\nVyucVqyGSN8aVuvIevBMQfrnk8+spoBVGqVldG85TVJ5UGV2ELSkVGiBFAsjsGiNuPgB+1aRrIwc\nFiCkgnGKkmEMiVISIYvo78ZBh/eJaUrsdw2rIaE1xJRQWvHSXifb/pDgu3nvHiao27ZdlKJrn7xU\nv3k9o0X8uMtV2yAINQmoVCqVp8JZs/6rIWzLw5sFPRs2oj4tE3D3ecT7mJjmnn7MEmimKXJnOWK0\nZjkGSi4YI5v/VusASkrQfvYJIGdCTuRBTu3WQhwS63c9IRZ8yrRWs987rFWsxogPcLSamIJUAkIs\nrMdEnArZRGxjyOLYTyGTs6JzFmsN1iCrhFOax+8yU8jsLxxaaXKRx0Ms7C8U3stUQ9daQpRqQ2c1\n1shrNc6QcuFg3zKMm7l+td0PYLTenv4fJeiet+p+83ptc/U3CEJNAiqVSuXCELU42J38YHdBz66o\nb8Omj7weAz5kQkr4+aSZdWId5HPnFDEkVqP06o1RhCTViJQLWmmcVcSUmKbM4Woik9nfa7FK4WPh\nzqEX8yGnWQ8Jp0EZ0MbglMIaWE5JRgmBSEZnRRwTWWW61qCwxJgA+f4KRd9rjJIAHXOhdZr1GHHW\nYC2kVLi+aBALIEUqGefEtbBvNV1raaxmf9Gy1zcYrUQnsDD0rWW1VCjAGcO1vfewEvAp8SxYDdck\noFKpVJ4Cp43xFTgW4HcX9Nzzrb8/WCwH2bS3GgMUZNQuBRat4Z13B2KUPvvoE1MQw55xKba5wxix\nRtEZzdEYGSdPKgWlFQbNeh0wRq7LWEUOmRAygQSloI3hoNe4RpESWAUFRVYFqzUpZ6zWKAyuNbRG\n/AZiyKS5mnG08iwai2vBzpWEzkkLIIXCYl9hjLj/KQrWaHzKOKPoW8eiNbx8vedg0dA3hink7Xvr\nrMFoSZxO7j24bK7q6X+XmgRUKpXKObOr4kcVGjv3pp0I13bZCNIkYOj7vh6T9KtDkGmAwUdWQyDl\nxBQ0r/YNKZd5417kcDWJIJBCTImUIBWNVgmtCz5mphhRpWCs3rYcrFHcWFjeWYqwLyUZEew6zRST\nJA6lcHftZTOg00xFKhuRTGM011rLmBIlF/reMXmpSviQIQearOgahyqydKjtDNMUyVnK+esx0HeO\nrrX0VnOw12Ct4dWbPftdi9Ewhcx68uQs710qhYNFc1/p/2k59z1v1CSgUqlUzpFdQaCzcuoNMbPo\nLTlxX3XAaE3XalmNW6Tvv5GTFUQbME6bUbxESAlUFoMbWzhae95+ewlIwrBcToQMhcLkI9Zo+taJ\n+U/IWKNpsmEdAynmrYCwcYbGianNavCkDLpkhnGC5DCAba309ZO4EDbOiNvvvOxnTAlrNOtcUHPr\nYjkmnIFVLPhoUNrS6IKPigOjCMiyozEkVCm0jaWkzGK/5eZBx7X9lr2uAco8HRHRSs9+BNKaCHGj\nMVD3/QzgfJ37njdqElCpVCqPwKOeLHcFgWrneSmWM8ViJxcCHQ2eUmCvd8SUGX1kNXjuDhMpgZ+T\ngqY1xHfXHK4C+wtZuxuB9drjQyQj5jlLK2LE9RAIJWOUIsXCegxzcFQElRnHwt31SMoyTphKIafC\nYRmxRtGPRVoGBZbrKLP6SrPXNRirSDGTkswkHo7TnCBERi/iPZRGkTHGUpCpgEXv8CERk6JpZFbf\nOMN6SNyxE6XAonOojexwfnut1aRU0CjRP+zM3z9sAVPlHjUJqFQqlYdwnifL08Rio4+kDDEmCoWS\nFWNIjH5knBJaS+Adh8SU0iyq06QhspwSw3oiROmdFyCpwljAqszRKoMJdEbT944wZt5dBZoGYi44\npylaz+Y/gYSilEJIkSCyAHQGi2KIHpcMCWkrUApBKUKK7HctShdK0ViloEBCqhkpA6bQkjlaBVgo\nWqcYfObGQUNRCjuv89UKSgbTatgE+CTmQ7sYrVi0lr6zJB/n8cDNIqXjUxdPg+el3VCTgEqlUnkI\nj3OyfNhV3Sw3AAAgAElEQVTyn12xmA+y9349RebFeISQeGc5olBorUgpM0xZvP9nuz2lFMMUJTBq\nNQsOxT44hMRqiEypoK0s7DEF7oyeMRW8j4SYpUyvNSiF0pmSZDzQKEVSMIb5eiK0DnAwRUkOCkCW\nNkDJhWQKTmes01hVZLVvSqSY0VpW85YMURVaXeicoWkUKSbuHnkOes3e/h6pSGKRUsHpgrMKaw3j\nGGgaizMaoxVp3o2glKKxhhhlYdDu2747dXHyZ/CkPE/thpoEVCqVyjnyqLvfN4EkZdBKTsshJsYQ\nSQnkKF1YD54QCkUV9haOYUwiDIyRZUiIxr+wVIHOzYFcK647hY/Ipr4s3v3LpUdryMh6YB8jMWWU\nAq01KWXU3N9nTkqckaU6PmamAYqC1soJfwoS4GOGpk2oDDGBUgU7jxdmNKpkcfNTWl4rZUy29K1h\nr59FfQ30WmyFjYGmcVzb61AoqSTMyc9+32xNgWTbn5Zqxk4CcNrUxXm2AZ6ndkNNAiqVSuUh7J7u\nxdCHrZr/NB5lPnzTAph8Quk5KIfEch1QSm0X9hQFSWdyglYbtIpopZgSkKHpLOM4MU4eVWDRaDRG\nxH4UpqhIqWAoZK2YfGTKiZySeO4XUPNynxjAWjm1pyDXGTJ0jTwvI38ugyQGGflfqxFvAlXIOWO0\npdENvnh616L7gpm7+toYlFJ0rWW/cyilWHSG/c6RUuGll3tiiPSdrCJWSt5Da++ZAO3uVzirDL87\ndXFVZ/SvAjUJqFQqlYewOd2PIc6OfjLPXwqnloEfdvI81k/WhXF2ybPGYIzGKJkIAFi0js5Z3r6z\n5hd3Asv1yJ1VYAyJFAr7gMqFgMGlyEsHC7RS/GKcmOZJg5gSWmv86PFFnP1CkBN/znKaN3PZPkRJ\nAvT8NYDBS8AXjf5OArChSNIwids/SmV8kg79xuWvc4qYZVdB31n61oKWBOCl/ZaD/Z793rFoLYOX\n91kpaBtD19gzT/Raq1NbMJupi6dRnn9Yy+dZoiYBlUql8gg4pxn97FR3iqPfbnA6SzS2efyXh6OM\n72mNNmoraDNa8epLC0LIHC49cQ6EISacMxwuJ+4uPesx4X0i5oS2ihwzXZtItufu2qPn2ULvM9aA\nojBOgSnF7aY8peRUb42U+LO048kRUDDl7YI+NGCQ8n6S7cJbtHwrYhYBodzPxrzHYBvNfuvoe0uj\nLZmMtZrGGQ56xys3FjROc32v5cZBIx4EjWE9J0GNMw9cArR5zskWzGaPwsmfwXnwqC2fZ4GaBFQq\nlcojcJaZzy4+JA7XXhb3aDnFNlYCRqFQCqxGT4gZrTRTzJgs4j8Rus0985AJOZNzxllHCIXgAzGJ\nQ3/MsimwZAgxE0OaRXwDPlhCKpRU6BqDjzJdUABjDCoVYkxoBdqIFsEo8B6J6PNq4TInASCnfqko\n3EsMFPeSA2PmJEKBOBQUGm3YXzhZ72s2A37iR+Aay6K1XN9vcFZz41rHzf1uG+j71tI5O1deIMaC\na3ngqX63BVMo2yoGvHfh3oMmAJ4FS+BHoSYBlUql8og8qAzsQ2LwcQ7UElyHKTJ6GeeLUYJymvvU\nuZSt0l0pGQnslWIcEqNPLDqH97P5jg8s157RB+4ceZajZ4qBkiDNoXrwEWc0cWe3QAyRMEfnmCMp\nFUouaKNAKcosCkzzKCBZgntI9wL95m7Tzj0beao8x0oSYJVMKDDvBkglE2LGakUy0LUOZyTL6J1h\n0Sj2Fg2vXF/w8vUOq48H0kKBwlx5kfdpk0ztvve7H29O46OP9/3sHle497AJgGf59L9LTQIqlUrl\nEXlQGTgXOX2GedUuiIOfNZoQElPMDF5z98jP5jZyetdG0VnDlCLLVSTmRMgJi8KHyP/7ueedo5GV\njyzXiWGKhBhJAcIERQWKOAtjtGIMgcYaxpAJczTf9PeLmvv/quC0Yoj3hIFoyGGeHJjv96yiR5Kn\no4EU5c/OQaMNSRestrMOwGCN4WDPzc6FDdYoqTZYx81r/akJwO77vcsUotgNbwyYSmGc4ry++Hx7\n/8/TBMCDqElApVKp8PA+/ubxB5WBt2Y/c/xIueBD2C638T6zGj2TT7OwUNb+TlMgZ7g7TFilySVz\nZ+nJBVLOHA4TIWbWY4Qc75XkjXwQ49zb14UQC9OUGSeIZQ7U8/VZZgGggtjkrThQaygRPGcH/pNs\nqu2NkekBM/sUOGREsWk01/YajNFYo7Fa0ziNNYr9zvHKSwsWjSXFgn3ExX/yc5BgLxsZC+tJdikc\nLO69yPMk3Hva1CSgUqm88JxV+t0N6JvHS8in9pa1UhRVtjPqZZ63N2YuqhfFapgw8/x9yqLS12p2\n6MsS4GIprMdEiIWY4qzgL6SYZQ1wKZQigsFYwMzXF8Ic3ItUCCLzSN/ONQZk5E8BbgK/80XN8QRA\ncXZCYBEdgXOSADSNxWApuuC0Y3/haOYph4POESlorclFlindOOi4ttc8cOvfaYFcKSWVlXjcZnnT\nKtj8XM5DuPeiJBLP191UKpXKe+Cs0u/k86mPbyoBuzTO0DqLUuJrr2db27YxlCLCvI1AzzWSMByN\ngdUYeffQE3PGWUMImcknUk74kPDzyX45BmKchW9ZyvBkRBcw9/CHCdaTnOjvv8LjnLy1k88/rbiu\nkARAA66ZBYFIINYm0TrD9f2Gvc5ybb/BOUfKiut7Ldf2HF1r2N9v6Vp5n0LKx3r8J9/Pkz3/vrFb\nPcX2muZpjZM/l93e/XsJ3Kd9/7Yx5952uGxqJaBSqVTeA6e1DxTzmF0qNI3moG+4ezRRnHjpy7id\nZjlMhCmjjWJMEW0Vy9Uk2oGQOFpFssqElNFk7qxGQsyzXe892b5S0hKwWcR8G6/+h/Eoz9mE03lg\nYPtxZJ4eKGA19E2Law2d1ex1joM9S984CpquMShd2O+smP9YcfzTWhFjoV+YM70W4H4FvlaKyd+T\nKCql6BrL8pTAfB7CvedlAuBB1CSgUqm88JxV+j1tJFArtR1B27gHAhytPX0ro2+jltG21RDpekMZ\n4O5q2o4QpiSTAylnlFa0VoLYaooMY2DtJQGYpkDMMIZESbK1LyaYJhH0aQ05wZAevZf/uGSgmVsY\nAL2bpwEsGKMpOdNZR9fK6X+/77BaoUyhbx1to+hax2s3FzgnToZQHuq1sHmvTz7mnMZHvd0fsPvc\n8w7Uz8sEwIO4sknAv/7rv/L3f//3rNdrvvjFL/KJT3zisi+pUqk8pzyoh3za46OPcwJwz0p4HEW4\nV1RBZQnqqoFxyGLV6wNTiKyDZ5oiwzwDv9ca1mPGz6N9KWamlBkGzzh5YszMC/sYPahNRUGJQc/0\nsLr/OaCKaABUA06JCNFYR9cYNLIieL+zYnOcM03n2O8bbl7rcEbRtnbeEKhonb7PcOlx0EpxsGie\nG7Oey+bKJgG/8zu/wyc+8Qnu3r3Ln/7pn9YkoFKpPFXOKv0WCj5KCbrdGVnbVAK8z6RcCDmh8+zT\nrxXaKJaHgV8eDRitOFx6csosV5IQyLbdwi+OIt4nUoaiCn5KjD4wxcx6zJQMPkoZ/hhP6+h/Chlw\nVoSAzpm575CJUXNtz2wXAVmj0bMb4I2DhkXnOFi4WduQZSGR1uhGNgBueC+n+M3P63kU610kF5oE\nfOUrX+Hf//3f6bqO73znO9vHv/e97/GXf/mXpJT47Gc/yxe/+EWcc5RS+Ou//mv+6I/+6CIvs1Kp\nvGCc7O9vetMblX8z76YvRcr+KRfePRrJJZMz21NuSoWQEgXDOEggn8bIcowcrkdyhtEHtNIMU2QK\niSkmYixYrQg+swoTORVihOBF5HeZOMRZsGgp/3fGSVkARSkJpWQvgI+Fa3uGl28soChUUbTtHJwV\nNPPq30VvCTFvV/3a2Sb4cdmc/rvWcvScifUukgtNAt544w0+97nP8eUvf3n7mPeer371q3zrW9/i\n1q1bvPHGG3z0ox/l137t1/iLv/gLfv/3f5/f/M3fvMjLrFQqLxAPcoY7OTWwGU1bT5HlOhBSIsbM\n3sLRN4a764lpiuIA6DOL3rBceX5+NHC08qSMuPbNCsJSCslntNGsxonJJ6ZRyvyRCz3sn4oC9lpw\nrQTwRhtcq8lRrn3RNuy1Gts4ri0aGmdpncVqTd9bVJnV+VajlZ6tkeXzEDMpFfr2ap3iH2QV/Dxy\noUnAhz/8YX7yk58ce+yHP/whH/jAB3j/+98PwCc/+Um++93v8p//+Z/8x3/8B//7v//Lb/zGb/Bn\nf/ZnF3mplUrlBeFBznAAq0EW+YA4ADZWM06RKUSO1p7RJ37+7oqutRijyUmc9n1M3F2uubvy3Lkz\nMiXx1Pcho0shlYjRmqIUYYiMPjEFaTOEi3wDHsBBA7Y37DVOPP0biw8ZbKHrLM4YFr2lbyxda1l0\njq6xx/r9zmpI96opIJWTzQjeVRq5e5hV8Hl9j6uUZFy6JuBnP/sZr7322vbz1157jR/96Ef8+Z//\nOX/8x3/8WK9169bBeV/eM0W9/3r/LypPcu/DFCmnJAJaKZnNV3orAFwOniFljkJGW0PXO7Q1/Pzd\ngXUYeeXaHsYZDvZa7i4nxpCxzjCkzHoIYturQKNIRRN8wjWWrAoxg3+KKv9HxSL9/6aBvTnQX7vW\n8vpLC3JR5CI+/loVXrnRif5Ba165tsfLL3dcX3TzPD00zmKNom3sfe/xJgBq/eTB9bz+7T/o30LX\nPnm4nEIipeNKzvN8H94Ll54EqHPMAt9+++jcXutZ49atg3r/9f4v+zIuhdPu/VEtgBtnWK7FqGfz\nmLNmKzb75Z2R9RRkQ1/K3F1NpJh55+4IClZjYPSRkjJDSKzXsizIhzvy9SFwuI6sh5EQCygouRCS\n+PVrAz551uurc/rfnOG1BaPEAMkpxTDK5sGMWCGqUvBTZtFbrnWOxsL6yKNTYTSavd4Rp4hWipWS\nBUdl9vhvnX0sHcCDTs/n+W9/9PHULZHnNX3wNF7/SROgS08CXn/9dd56663t52+99Rbve9/7LvGK\nKpXKs8zjWAAfrWSZD1mse1Mp+CGw6C0lFKaQsFZzuJw4GgIpZHyKrMbIFAOTz4SYiEmW9eQSUUbJ\n+F7MJApx8sSUmQcMxFegiMvfFOTkf//Ou8vBsZkAgP2mY3+vwSmFtbKRsHFWFiWlxMs3F+wvLLde\nWrBoLRQRDjojOwJCTCglK4SdNbPdbyaXednSIy79uYgS/YYXxSp4l0u/sw996EP8+Mc/5qc//Sne\ne7797W/zsY997LIvq1KpPKM8jgVwzIUQ81awFlJGabhzNHHnyHM0TPz07SNClMU/d9eeu8vAL+6u\n+cWdgcPVxDuHIzGWeX1wJqd5tK8UeqvRxuCMJSVYjeA9+CBmP4GrlQA0DtoW+sayv+fonaHvDAbo\ne4drDH3r2F/0tI3j9Zf3eOX6gq51NE40EbIYCUDN2wwjhytZmqTUvTHL06yXT+Nhmo3z5GlbBZ/2\nOpedZFxoJeDNN9/kBz/4AXfu3OH27dt84Qtf4POf/zxf+9rX+NKXvkRKic985jN88IMfvMjLqlQq\nLzjSpy2UXOTEn8v25HrnyBPzSJlNe47WI0rJaVQZOfGvVMRq0EajUITJE1VhmgxDCsQszX7Fzqre\ny27+zyigU1L+dw0YrekbhzOaawcNe11DSQWtFUZrrFX0nWHRuq1botWKZjYE2sQ5Z/XWUbEU8Vpo\nldmOBl4lQeAuT9Mq+DwWG503F5oEfP3rXz/18du3b3P79u2LvJRKpfKcslvS3QQhrRRK3Yu6mzW0\nMWdKls1/ISbGkLi7HDHWMEyRcYxMPvHuas2icTROM4XE6BNKK3SBvb5hPQWWU6Z1ipw1iczgI04n\nhikSg9j7prMu+pLoLTgjVsTayDIi3WhsY+kazbW+YW+vwaEIWdopNw8a+t6hUThzz/3PaM2it8TZ\nFAiY33tIOwlPKeWxRgMvukT/tAPzVdtHcOmagEqlUjlPNqetMURyLrOQTH7ZSk86z2t+pVx9dzVx\ntA6UUhimxNE6sBzXZKQsXsioInP/ID3q0WeMlk2A6yEwhUDMhZw1TmVWUyDETNSJmGTN72Wb/pyk\nN1L6t1bRIZ4IrbXs77Xc2HMc7LU4Z7l5rYMkq4/7ztI1htYZYsp0raVxBqM11/YaALIqx4K2swZi\nnlcqz0HWPXqJ/Sqenp+Eq3b9NQmoVCrPHc5pRq/u86i3RrEcM401FArLdWCYIoeHE2NKtNZwuPbc\nXU3EmGmcmADJCt/Map3IBfZ6y2odGIZAmisKCvAhcDgW0BAjJCVrf69SAmCBtpFuhDWK3jkooLWm\n7xteu9lxvWtpGkvnNLoUXn5pQQiJgoj/UHCw1zKMkSFHXrnZb19/N2hrDRTFfu8I8xrkxpn3bBG8\n+bhyftQkoFKpPHfsbgAUf38JSkqDQtM6wy+PBtnkF+Xkulx77iQIMcw+/wVjI0utWLSOd44mhinQ\nOo2PsJ4CU0qyalcpYkz4WLaKf5D+/xVp/W9xSrYPNgZa57BG01iFaxtuHnTc2OtorMYYzf7C0bcN\n1/dalIHDI88wRZyTZUFdo3HWMI4Ju7jX598E7W6eJoB7orsnsQiunD81CahUKs8Fp/n/DyEy+cjG\nn0Vl+d/oI8MoW/yM0RitmHxmOQRiihgjo2iDz6gigkHvIyjNGECVQEoZHyM5gdFQ5nW7V8Hu9yx6\nRADYOM21vqVpDItW/P61KrTOkUpmmBK3bvQ0TtYD+5gwRclOAKVIsaBsoSiFXweM0xQKfSPtgRgz\nZX4XlAIZFKyn+KtITQIqlcozz2mz5FopcpZRPODYeJqPmRg3veuCtVrWA/uAAvq2IWVPJInLG3Li\nJWUyctqPKZGiiN7CLPkfrpryb6bT0HagtMJqxUHXsb9ncUbzK68eYJTm+n7DGDKHq4m+tbSt5dWX\nenIsrEOU0T8yOYt/Qi6yLbEUMEUSoJTveS/sGsEpxVOZ6688OTUJqFQqzzxnzZKD+P1vEgTpSxf5\nLxdiShijKUDXGZbrjcmPhyKb/UYljf0py2x6jJlhCvdO/lFU8Fdl3v80dJEpAGcdShWclZbIzest\n+13D3p74/y9SxlroG8fBosHHzHoQl7um0XSNRRvZlhhjRjtL46SSstFexCxGQLuuflOIjIGtIPCy\n/fIr96hJQKVSeebxIW3FaEpD5yzOGnKWhT+TT6QMU4qsxkDfGIqGMCVWYyCnQusMfed45+7IlMT0\np28tRhVWQ6TkzBQiQwjkCCiYvIz9nb9tzfnQqLkCgGwmbHVm0bYsWk3XGijQdZaX9ns5pStYdKLy\n7xtHipmiyj2XvzxPReSCcxprJQHY792ZFvCbzYtKSdXgvTr+XbXFO88LNQmoVCpXjsf5hT+FhA/p\n3qa/KTFOiaYxtNZgjeVoHUipcHc9YrUBNJSEAo5WgVwydj7Z55wZfSJTiEkqCYP3hFRIKcm8f4Ew\nXR2//9OwgDXQtzIJYTF0TcOi32z7cyxaiwbuLCcOeoPVBoNCGZnvL0bRKSvl/6yYQsZqxf5eQ+sM\nOcnY4G4CYLXajgOCJA+b9cG7j4WQH1ns9yDr4OeBy0xwahJQqVSuFI/rFZ+z9PRjToSQGMMsSitS\n/m+MRmvRBuRUSGRWYyBESR42C4IUMAwTyzGiKOQE2STWQ2DwkRDF7z+lqy3+AzDIFsBFp+hcI6OS\nWmGdotWGvrU01tK3Dq1FRBkStE5x83rP0RBQbHz/xSnQx9n2t7VYLWuFlYLJZ6Yg6VDrrCQd5V47\nZuPH8CRcpHXwRfMo/96fZpJQk4BKpXKleNAv/AedHJ3V3DmaOBomlFKstWbROkpjxbkOBai5bRC5\neyQmQcthYvKJWAreZ7EQLgofIyoo7g6BOIn6P1zlyD/TAk0Lbas46DtaZ2gbi9EFpw03bnQ4beg7\nS0ibef5MQdM0FqM1BwuHDxmdpL1CAaNlkVDXGJkASBlrRCy4Wb+bSyame6LAjTHQyR/pZfvlXyUe\n9u/9aS9QqklApVJ5plmNgXcORwafeOfOQDP3qrVW5FJ452jAzgY3U0yMU2S1DqxGz+E6sPaRGIss\n/pk8IRYKmWmSXQEhzKK/ZyAB0EgC0HeKtnF0jeXanqVrG9rGcn3RkGbhXs6ZvnN0rcMoxcFeg3MG\nazWNMmgl7ZIxiDCwn3cDbCYsYpFTeCnHT6Yh5vs8AZ7U8e9F3O634b0mxY9KTQIqlcqV4nF+4S/X\nnk6JUG219txdTyg0L19rWHQ9MReGIWKsBKpFa1mtPUerieXkWY+iHxh9YoiBNGVyBp/F6e+ql/13\nsUDfgms11/uG/b2OvrPs9w2v3ujY22+xKFKB9ShOh4vO0TpD11gWncVZWZrUOrMN9ii2wr5NX1+f\n6PE/jCd1/HverIN3uewEpyYBlUrlSvE4v/BjLkxTJBVIFFTR5JK4u/KgFEppxpSxwOgTax+5u5w4\nHCKHQyRF2QMweY+PhcFfXaX/g3CAVdICuN63uNby6isLrnUNSin2Fi17jSEXzYEz3LrecWc5MY5p\nDkDim1BKxpiC2QlCfWOP9fg3Pw8fJDnQmu02wU1ycDKInUfQvizr4Kct2nvYv/ennSTUJKBSqVwa\nZ/2CdU6zWgdyAecUPnDsl+/m700+kbTCx0xrLNcPMutJ7IJ/cTjQOyv9cFVY+sSduwN3lxMxSVtg\n8B7vM9N4tUf9HsTCiWNh34oIsGsdr93suLFoaZtZD2C03F9OLIfI9YOO/v9n7+1CJUvP8u/f87nW\nqqr90T0zmUxe8yZB4Y9IDgLxwIgOKHggCCaDIJg/Rg8iiIoeaYIRoqgQTRCPJORA9MgzCSOISjCB\nKJ4oQQyIYAaMr04mmeneu6rWWs/ne3Cvqr27e3fP9PTu3d0z64Jh9qxdtWpV1Z51f133dXnHsvMY\nDUppUs6UqgBNMgVd1B0B6Hzw2QUvb8UeWPFg0sCvh0dR/T/sefwO90pwHnYXZE4CZsyY8Uhwrxts\nSgV7rt2cy62/y6USUyaVTBqhHwLeCdsdYDsmQirYmulayxAKJ31kMyY2Q+ZkM5JinsR/pO3/pCUA\nDjECWi4MnTW4xmGN4vjQsuo6me1bQ9sYtNLUDK23hJS5eTJy7ahFKQixoFTBWINSIpY0kolZxgK7\nKt9q4Q3ccg1T8NqNDt5qc/qHPY/f4fUC+8PsgsxJwIwZMx4JhpApE7lM2shmf4OtF0ziz/9uJ0Cz\naBxjnXT/hwFnhP0fQhH7WyP2wN++seXGeuDV00QII0PKlFJgp/h31W/+EtA4WC0cz17vCKGwWngO\nV5bGObpWGPxNY6lFoYzCOUUuErCzUZRcaVsrn2kRoyU7+ShUJjJfqfv2fqncUQW/lWbzjzMe5uc8\nJwEzZsy4clzUBagxC9nsgjZrTNIOFea+GPrsHnZw2GGN4rXTkVILtUpCoTSMsXCy3rAZAtuQSTES\nQybERE6T3e9VvelLRKvAWFh2mrb1rBbw/zx9QOMNWiuc0SwWDjIMMaMVFK0pBTJipayUBJfWG3KB\nGDOJgp40Aaii8BdTofNCGryfKvitoPD3qEl7V4E5CZgxY8aVY+c1X8qtx1KutM3U8q87vX9JGFIu\naC269TuVwHUfMScj3znp2Q6JfkwopdCqYq0l5cRpn4ixkGOhouhzYhgg5iczAVhacF5a+xTNM091\n2KJorGbRWlrnaBvDMCZJBAYIQT5DZzWrrsEohbGSRVlrIAm3wlpNOwXrfpTPuJl0AXYIKe87NXcL\n7m9mlr5LGkLMgMJPj32UycNbeSthhzkJmDFjxiOBt4Z1H/bBwmjFonH7G//u5ltqJeZCLZWcJXk4\n3QbWfRQ7YBVYD4kb65GQEyFA4xVKBTSKG+tICBlnIKdMDEwV8ZMHhyQAXeNYdJ6nVp6jRUvnNK2z\nGCN7/CkXFq3DGcPxUvNqGQForKbxohiYciGmglKKRedAJ6zW+9U/Gb1wyyrgLhHbFce5Tq6BVu3t\nm70z9z1L3yUNIckIAipjkKRljBljbj3/VeJRbSVcFeYkYMaMGVcOrRRDTFLZl0pKmWoM2zFRqbTe\n7m++2yGBgr5Pe6bAza2stw1jIpbKpo/EXMhZOgchiE2wVJYJqy3bkBliIKYnQ/nvdjhgtYTGGrw2\nHC4c77i+5PigYeksy4VjGDJQ0UqCudFgreV4BbUUGm9ZdQ6j9cToVzROY7Rm2bpbqt4dCXD3Ue3W\n/87H913QLlGC867iL7VOosNvDLvXvL0ztBki1mhykeTmYbHz74W3YvV/HnMSMGPGjCuHd4Y+pKm6\nk5Z04wwxJTZDwFtpQVutsEYzBEkOhlC4uRkYQoYp0Gz7SD9EhpBIBfoxMcQARZGp1FyoRLH7jRDy\nkyMABOIDsGxh2YnZj3MW7x3vuLbg2qrl+lFHGhLOaHSn0QqGMZFzxVlDypnGarqu2Vv5hpRxRsvJ\nUWitiLGI8c/UInFW3yL/65yeWvVnKBcwKkuVjo09ZyIUU953FW5f97wXroqd/3bGnATMmDHjyhGi\nVJAhFrZDpmk06z6w7ZMQ2YzCKLGzNVaxDZlxiGz6yGabGHPajwaa1lBvCoEtxULOmTCt/lXE8EdV\nCVjhUb/x+8BuGW/RwbNPdRwvGk6HjDGap48sy8ZzdNiyaB19rnTeSlu9irOf6PkrrBKtgOZc4I2x\nYO0uwEuQ11ax6eP+eC6Vg4W75ZruSpS7TT1wZxgkZM5Mqexf//y65/ktg1wvFh663YFwxuViTgJm\nzJhxpdjNf41WoITlH2MhxMzpdgSlcciWwI11IJdMKon1JjOGwsl2QKGIMVGNwjtHKgVVxStAVQks\ntUKIZ1X/k8IBMEBj4WBlaI1lubB0ncN7x6ExHCwcXeN45lqDtZrGWNxSqmvvDKfbiLNqX5F7Z26R\n+w0p3zHXrzEzpkTJkIpCa1i25o5gfTtRzk5kwvM4z56/iFcAd1bzdxMeSlk6QXc7/4wHx5wEzJgx\n44ywZJAAACAASURBVNIQYp72/yspZ7y1d7C8z7d4wyisf6iEXNBGWs4pF7rG0o+RUst+Zz0k0frP\ntcjWQKo4n0gpE5PMj4tMCigT+78ydb2fAGjg+ko6IIeHHYvWonXFYnDOcv3I0TWGtnUcrToab1gt\nHWN/xl5fdIacKlopOn9rS7/xhsrOQOjsexhjIoTKojP7x8dUJEJEbmm9nyfKLReOGMtd2fO717ug\nq38HKtIZqhWUlnMtu3uff8aDY04CZsyYcSmQBCCdY3lDypFczN5OdlfBbfpAKpVcxaO+VkVMmc0Q\nKBVa5yilMMZECoWTfqAfMttYoEgnYdMnlNYUpN3cx0IhUasM/fO5efXjzgFoFFw/Niy8wVjP4cJw\nuOomxj9orSm5crgUVcBl6zBa0TWWg0VDHOI+QKoIWd3ZshdOReJkG6BKoqSU8AZqFTEhdY5sV2sl\npoL1d1bid0sKLqrQbx8hXMQPCDFTK/gLnv9WZ+c/asxJwIwZMy4Fu3U+OJvr1ioBPyRJBELSKCBm\nmRXHLBVpKYVFYzndBFSBTT9ys4gpzmk/EhNsQ2YbIgbYjtIZqKkwRIhDRmlNDokUhQAo/YXp2h7B\n5/FGYIFlB4vG8NRqwXLlsUbjrKL1FqNFwc/ayvXrS7y3Ul2XitGaVefEByFktFIodfFuu1ITJyJJ\nmySX6bOpUFMRVUDNHRX7G2m9v151fv56NkMgZzkWU0FZUXjMtdyxTXB+ZDBX/w8PcxIwY8aMS0VM\nmRil2ku5UGslF0UpYEyZtgCkU+CMJlYJ4ClXjg4a1uuRfqhUBZGK0hpUgSpKgREIYybVgjVGgpmC\nnCLjWBnCRAh8pJ/C60MjpL+DznN40HCwdHSNpfNWdPu9xhpNrQpjwBhF5zQKjXEK32hurEfaRYNW\n4Kzak+6MVWKIALew+ktBKv8o3ROloJ02BqzVt6j8Ga1ZLdylrOLtDKFyqROHoFIKjJNKZEoV765m\n5W/GrZiTgBkzZlwKtFKTuQ/7ynOMSfTojRwzWjGEtA9AcapMT9aBWAqqFsYkAb4kYYuPMRNTkuBU\nKylLZZ9zISN787VWlDJoVW7pADyusEDbQOsN3cLTOsPBouXp444YC1pVlBZRH6NgSFLtt95yvBLm\nv0L0/2uV9yu7ABPnInHX6jmmzDCZJ5lJNtg7CQXO6r2A0GUlAHDWUWiquaXbsBs5mAteZyYAXg3m\nJGDGjBmXAu8MzhlSSVhrqEnY3tYYQsjYTgJMSqI6l1Jm00dO+0AYCwUx+1EKZIJdCaliDQwjpFgo\nuZJSIcWE0zCmDFYzpMg4SFJgebxXARXi/uctLLzFoCiTYFKImXdcWxBjETElKsZpFtqglAZENtla\nRS0V6/Q005dg2txl/343l89FkolaJOBL9S86Cs7qae1SCHmXlQDsugtDEMKn0XdyDF6PYDjj4WFO\nAmbMmHFpcFaTi8x7F51DIUHAmDOlOWs1/ZjJSar+kApNY3j5tS0hpr1jnTWKMRUMsBkzsRZCgYJi\ntWwYxoSpmRTEZ6AqiOHxTgA6A86B9wqnDY13WKc4XDZYK7v8ORW0htXU5lcKXKPRVVFRbMZAky1d\na7FaEVIipryv5i+qoHdzeTt9DzvugMT5SggZzdkGR4xlb9v8ICZA5z0EtJIRQ0xlLwN8fuQwEwAf\nDeYkYMaMGZcGqzVGF9QkFtO2Mt8WUaCBmAvLzlPJfOd0JOZMjoVtriig70XZJyMVaWsN3zkJhBS4\nsY7kyeM+okFVQilUJTrzlMdPDlhxNproDHSdEi6A8zhv8d7QOcOitVQK2zGiNHjjOOoMzmjGmGic\nlfZ/FYZ/jEGSJVc4rOKWmEuiVli29sIq3jnNEBXOGRSiz+CsBN7bH16qSDGf39F/M5K959dBd1wE\naxQpV5w3t4wc5ur/0WBOAmbMmHG5qBBSohQYxszpVkyCcq54bxhjYhwzisI4igZALJl+SCgNqdZJ\n4jeTC1hb2QzCHVcKNmMk10KKdT8IrxVifLy4ALubq9XgHVgLnZuqXqO5vnQYZ1i2dhLwER5FTPD0\noXABjIGFsnvXvlIKi9aSK+RS0FlMmKwRuWCoFyryAZNugJ3OU2857py+UNDndjyoZK+zGpJ0eS6T\nczDjzWNOAmbMmHFpKFVm+VQh7qFklW0cA2aqKlMq5Cy+9cZktNEMQxQ/AEThrmRYj5kQI8OYCFHO\nsx0TKedpZVAY9hn59+OUAIAEfj0lAEZDYz1d53FGsegspcDxwhFzkc2IZYOzhlVnUVrTNJbOWWmn\naxiGJJLARmOmREkrsNqQVMRPLH+4e7D2k15DHxK1ToJCjb1Y9e8SpHpv1whQSu3NoeYE4PHAY5sE\n/Nd//Rd/8id/QimF3//933/UlzNjxozXwXob2AyJMWS0QYx/YiKlQkiFoY94a9Bo2kZjjMHaRC5Z\nkgIF1SiM0pSaaRvDNiTWfSKWNKncZTbbMyXA3Rrg46YDYJHAv+o0CsXBYUc7yff6iYF/uPIopWms\nwhpJcpaNxU4L+ylXqkMUAUOGfbBWUGUlEAVVCSmw9fbCwH1+7W9X9ZcqvA03WQtfRMoLkf08f4f7\nZexfpFkwt/wfLzy27It3v/vd/O7v/u6jvowZM2ZcgJ064BASIWbW20DMBaWkE5BS5bXTgZNNYDNG\ntmNk08vjq8owKdhRoPGWZ651dAuL14aYM8oIaUyc8BRWaXLJey+Ax63qPw8FdB7aVrNatDx1bcFh\n47l+2PL0Uceqa2i8FYU8ozheNRwtG7wVUZy2sSgtrfumkSTCO8OqczgzCQJZPc30xWlR30Vff0fM\nm7YryeWsQ3C49HSN3T9WHB3Pgvz5zsLunI03913BX3TuGY8PrrQT8IlPfIIvf/nLtG3Ll770pf3x\nr3zlK3zmM58h58yHP/xhPv7xj1/lZc2YMeN1cL6aTKnsW/sghLHtkLFWgpJo/8N2iJwMkc4Kqa3x\nmj5ENND4gtcGrQ1do0m5cLzsWPeR2A+ElBmDSMlqrSglk5JIAe+6AI8bphqd1QIOuoaDleOZwwXO\naRSV1aKVjYiSOFw0KA1HC0/TGKiKrpE1QK0VC+94+qgl5bKvxpVSLDpPTIVSC40xlAJHq4YwxMmx\nTzoCu2r7jc7171ahPyhj//YuxDwCePxwpWnZCy+8wBe+8IVbjoUQ+NSnPsXnP/95vvjFL/Liiy/y\n9a9//Sova8aMGffA7dVkKoXx3M0d2AcggK4V3f9UhYAWcyamwjAmnDVoo9BKRIC+c7Pn5qbnle9s\nGUIhxUguipwhp8pmGBmGSB8zOQtp7nFLAFoDhw0cLeD6AXSNZbVwvPNogXWaZ44X/L/vPKT1llzA\nGQO18uy1JW1j8dZyfNBw7bCl8ZqFtzx9XRKAUsVUJ01bEcvGcbT0aKUn+V0tQd+J8Y+65Gp7lxy8\nmQ7ARV2I86OBGY8HrrQT8MEPfpBvfvObtxz72te+xnve8x7e9a53AfBjP/Zj/P3f/z3vete7+Nzn\nPse//Mu/8Ed/9Ef86q/+6lVe6owZMyacv2nHSbkuJtkrX3QercFYce0DCRzGagyKGjM3+8h6K4ZB\nTarQKkoJnGwD2zEwhso4RsxGiQa+Vqw6S5g8APqcSBlS2ivhPhYwSAJgHSxax9FK9vq7xtI2lkXn\nWbYWpaSDYXQFVehay/HSY6wRwqDTrBaew2XD4bKyaOwtHYCdqU5MhUKhFlkzrEoImEOQT6W7gHB3\nOzFvd+wq2vIPY7tgxuXjkRMD//d//5dnn312/9/PPvss//qv/8rx8TG//du/fV/neuaZg8u+vCcK\n8/uf3//DQD8KkzxMrH67DXtp2a6xk2JdIaSMmlq+sSqWh4WTIRDGgLaWFAOmFIzTbMZEKJV+UsaT\nJEFRUqTtPI03FBTaQR0lwYiPEfvPK1gupErumgZnFdevdTz39IqUKprKaulRWrPsHI035AxHR0sq\nlesHLc5qhlBYLiyrzuOc5unDBakUcq6TFDK001pfiJntmPaqgOO0TZFL5fh4wfFBe+G1DmO6pSXf\nNldz29/93dyOh3ENb/f/9x8EjzwJUJc4I3rlldNLO9eThmeeOZjf//z+H/g8t89wdzavuVTGKGI0\nu06AMZrNWiRvF62lVNEC2PaRb9/csBkyN9cDw5DYDolcM0klci7ElNj2omHfD5EhZlKBkiKbMfKd\nG1saZ0khMQ6V7WPSAnAgmw8KtKosGsPxgcdawzOHLSkkKmryO8gsOkfJidEYuokM2MfI/2wDXWPx\nztBvRm7anq5zbE4GusayHTLOyue/nu6RQ8iElCZ/gMm1MVeee/aQfj0Sh3jhNe+qb5C5/ukVzeXP\nqwXusOtC3O81XPR3ucP8//6DJUCPPAl45zvfycsvv7z/75dffpnnnnvuEV7RjBlvT9x+0z6vEFfP\nleFKibb8jgNgRKUGqzXOyOzXGEuuSdb9SkUb0FlTVMVaQ4gS1VOu5AqpiENgrpWSxCRoGIUHMDwm\nCUCroWlk979tPF4pDruWo86x6DyNdygq6z6AguNVQ4yFWkBZUfU7WnmMliRBabWX0M1VpHuD1dSa\npK2PuqO1r5DRAoBCYYysCd6rvf+o1vIuaz3wXn+XM9HwwfHI9zXe//7389JLL/Hf//3fhBD4m7/5\nG374h3/4UV/WjBlvO+xu1mGa+w8hM8S015E3WqOUsPy1UjTO7BMAMaGp+3N4Z0ix8MqNDa+tR7Z9\nYiyFYcysNyNxquxSzqIMmKS1rbQiJ1H/ixHqY0IE9Eyqfwa8VSwbw/Fxx2Lhedc7DjhYNlhdaRqL\nMZqnjzqs0Vin6bzDasOyc9Qq7f1rq0YMl6whl4pWMvPfbCMhZBTqli6piPqYc3r/03GtWHWenCoh\nZk42IzfWIyebsLcPfpS4jPXAe3ELZjw4rrQT8Cu/8iv88z//Mzdu3OD555/n537u5/jYxz7G7/zO\n7/ALv/AL5Jz5yZ/8Sb7v+77vKi9rxowZE9ZDIKWpctPgrRHPd6c5WHjGKTEYQibmjLcaYxS1Vk63\ngSEWFJXtkBhjFIZ7KQxDxjhN2yhyLeQMJ6eBkBOpZKhFyH91WgOc7u+POoxZ4OlrhsbIxgNK0bWG\n64cd1hgOl45cwTtFSpqa4bvesWKzTSilOOga+jHiOkculWVrOVo1rIeEmdYgtWJvmKQVaCMdgqMD\nT50+gF0w7UO61e63c2itpF1O3csBn/SBm2t53qKxHC6be7bUHxbeaPX/KK5thuBKk4A//uM/vvD4\n888/z/PPP3+VlzJjxlsGIWb6UYR4HuQGmlIhp3Nrf0WqU9dqNr1YzQ4xsdkmtBZL21ClSzCGRCmK\nGDO1VrZjYDsmrDZAwTppW4conAGtwDhFTkqshmMmioYQYar+H4c6r7Gw8I7VQUu/iTTe8NRRw1PH\nHVoZqBlnFMZovNfcPB351n9vcNpilOLpa1LBh5BZdYrDA8+q9WLaUyFrsfTzVqOBrnMYo8VQKHNH\nAG0m2V/jRet/V2Ubq0hBvrvNEPeJXAmSRYRYaBuzt/F9nFrqr9fuf5QbDm8HPHJOwIwZM948znax\nZR/7QW7udgoqeV+RieFLPyScM9gK220SrftQKEWU69b9KHr+uUBVhJB49WTglVd7Nn2iaYxUtFox\npERJmVoqpQK1kKlYa4k5UbII7jyqDoBFEhBv4PjQ0ViN9w4KdK3haNWy6GS333tDDpphzPhaWd/Y\n8P99Z0s/FrwVBnwqmWevd7zj2oKnry3RSrHeBrzXlFo5OmjEYClnnHeUXPHt3b+71p+5CcIZ0/78\n40u+s30ec6EMlYOFJ6RMmTIsGxUHS3/ZH+N94fVWCWfp4YeLOQmYMeMJxmXuYssMuZJiQWloWrk9\nVMRMJqRMyoU4BZEhRDYD0gXIhcZ7ToeRWio3TsU2WE1JRIyJXBVGQVGwHiO1lj3TPWcZAYT4aBIA\niyQf1kHj4KhzdIsGZyUtcFrhnKVrHK2zhJTox0gtCkUlZs03v90TgoxDQqpUMiFuUFSeOV7Qj5HW\nO/oxYzQcHTQMY8IoBUpTa2HReajSgVl0F9+eL1Lx00qRSyblypAyChnl+MkdMEyPlwTgvMgTj01H\n4F54UOXCGXfHnATMmDFjTyJzzlAQhbcwSdCuFkaqzyJs/jFkbq5HQJzsQqp4p+hDQFXEF6DUqc0f\nGUNiLCIbpyhy7ijVcoqVcdpsK+XRJABeTax/2DPta9VobWgsKG1onJEgmwon25HOG6yxWAf9WKhZ\nRjIpV4xWNFZTqWhl6PfmPIqYMtYpjFKcrANKSwcGBcbIrL+ZjIZyqtgLivS7VcLWGFJOWKVIRTQG\ndvr/zmjaxhDPjXv05N/Qh8gQxavgUczi30i7f67+Hx7mJGDGjCcYlzUvLbWK6E/Me+LZ+Q2AXCop\nZaxVnNwY6SeVOnH/U6wHcbUbRxlP5CLn7Ico6381s/SezZjJ4oxLiHVyqpMq/FFsARig8ZMqn4LG\nKErVoBXGyKqj0YpaC7UqxphwBpI15JK41ras+4RSoFEYJYZHMBkJNYanDj2rZUPOIq1cijymTByI\nnT8CyM9vBvL9iTugUooxpH1SsWgc1w6E1BmTfG9aiRFROUe82Mn6XnVXYG73P1rMScCMGU8wdjfQ\nHe73BrpjZZ9sg9jTGtnr7yZb2h3RcAxZZGpRWDRay+NyrmQqqRRChO02sg2R7RAYY0YbjabSodhM\njL+cMyEgRMFHxP4z0z/WgnNQqTRWY7TFKc3RymGUkVl6hqoU3sn8vbGGSsFbS6mwah19SBwfeLZj\nRlGxVuO05t3PHPDe7zpkvU4UMt4ZWm8ppRKSJFzGKKyRW7FW7FUX7zeR00qq+c5Dbi3bIU2mQ3Ju\n5zQ2KkplP94Beb2dBfGjkvWd2/2PDnMSMGPGE443u4u9IxWGlOXnVBmUwnmNg1vMaJzTNM4yhMBq\n1bANkVgKzhvGQdrgMRVyrZxuImOSqnIzjGht2KSE0ZpCZb0RD4BHOYHuHHgP1uhJlldRFXht8JMO\nQp10+g+XDZuQpm2JhFKVo2VDrWCtomkcdjQ8ddjyyo0N2zGileF46XjPdx1Nz4too6j1XPWv5OdF\nK8JL8jvpvtwehF9vhe72jpDRmqNls//b2D3fubP1Qjh7vUeNufp/dLivJODVV1/l4OAA59zDup4Z\nM2bcJ3YM8fu9iQ4hU2rhtA8gK/CMKXE6VFqbWC0dy2xlZg2YaZUt54wyGmKmZiGbxVSIOXPzVISA\nQiriflcraUiTuVBB6zMToEfR/vdI8FeAtQZvLYvWghK2fus1IMlKYwwHC0sFFl6TUsVoTWMtY8gY\nrYghM1bQFpyzvPe5A7ah0jrDM9c6rNIMJaN3yn5Gk3PFakXb+LOq32pKgdXkM3Aeb0Qx714t9duf\nryelQqUU1tyaNM6rd28/3PXb/uxnP8t//Md/ABBj5Od//uf50Ic+xA/8wA/w1a9+9coucMaMGZeP\n8zavZWLmjymz6SMhRmLJvHazZz1EtmOSOX+t9EOWgJkmB7sp4ENh00diroRYJ5JdRU22wKWKC2AM\nj+b9KqCzsFgIB8A10FnLc08v6BqHtZpVY3Fa03k9iSBpvLMcrxq6xnN00HG8all1DucNxmhClsSm\ntQZrLYerjv/z7uv8n3dfZ9F4rNWoqcVvtN5zKBadjAS0VoyTtPDBwolOQsy3qP29UcW8u3WEbn/+\nLuloncXqO8l3j/OWwIzLx107AX/7t3/Lr/3arwHwV3/1V7zyyiv8wz/8A//2b//GZz/7WX7wB3/w\nyi5yxowZl4tSK1oLMWzXSg6xUGulbRwaBRrWfcSOmbax5JJpvOY7J4lV53h5K9K0xihevRnIuWKM\nBlUZxswQEjFWrJX296M0AVp6OOgs2misNVitOFxalFZoA6ZWvNcopKuiqHStp/NWPg8VSUUU/hqv\nWVpLiBWloW0s3mq8NeL+h6z4aQ0URecM3hhSzjTOsOwspcDh0pNzBSOchCFkKhWqBOExZJaLN951\nfTMt9XkWP+Ou37pzDj1lif/4j//ICy+8wPXr1/mhH/ohQnhE6fyMGTMuDd6avRfAbk7tnKFUCQrb\nIXG6GVmPkZgSJ5vAaycD637k5RubqZ1fiUk8A/qQ6ftASpkxB2Ks1AJxFA8AzdWvACrOdP+d02gN\ni87xzqc6vPdoNQVxozHG4K1m1Tne8cyKZ447rh11HCwsq6WjaxxHhw3LzpMLGC1Sv85o4TfUOq1X\nSnIgn69cRyoFazTOG6wxLDqL0RrvzH6+P8a0twdmIluutxc7A95P2/6iyn73/F3i8Dh2AEKURHII\n6bHwQXir4q6dgBACr732GoeHh/zTP/0TH/3oR/e/u8gjesaMGU8OdtX/egj0fWaMSXbUvcEohdKK\nlMVEyAyJb726ZhwrqEyIlZwn579JPKhWBbnSx0IfIqVIIByjEOBCunoZYA14C00LnbOkAgtvWXjR\nJ2hbgzIGbyzLxhJixjtN2zg6a/fyvUZJkF+2jhAS3lkU0I/y+FwKerL9XbSORWNRSlT95E4pGv/N\nbmd/0hC4HSGInoCsDEoSUWudzIS45wrd3Sygd8dSKntux5NAwpudA68Od00CXnjhBX7iJ36C5XLJ\nU089xfvf/34AvvGNb7Bara7sAmfMmHH58M7wv99Zk6LslzvryLlwYx1oG812G9hOHr59Tgx9Rhk4\nXY+kDCVntIFaNEYDpRKBfoxkpt1/mQwQhD6AAhxwcW17eVDAspH1P601B61HaYWzhoOlZ+E1sUAI\nmePDhsYZFl3LZhsoRdF5Q+M0VhuZ29vCwarhZB1ovWWIEaUmff+pwtdWtgyOlp7WW2qthFQwStMu\nzF60Z4ecK9ac/XdKMorZBeqdb8OiPVvvu1vbfrwgYJ5uAsacORGayZTIWf1EtP0vUwlzxr1x1yTg\n4x//OB/4wAf49re/fYu1bymFT37yk1dycTNmzLg83F4tgrjWgTDlk9Y4nzjdBHKGcZprn5yO3NyM\nlFq5cRqw1mAUWGdwpuKqJk5iQk5rQkyoCmanxVt3De6H512ukcCvDTgLzni0VjitZU++grEKUwsV\ny2Gn2YTMzXXg3c/IKt3hylOqomssi87jjcFazc3TgTFUFp0jpUKjFSlltAZnxR544S3Lzu0Dfcpl\nUuzT4gyo1K2a/VoIfLvvwzlDhygyyvfDfmPAtfqe1ft5GeAdUpERzW7UIFoAak8cnDFjh3uuCH7/\n938/6/Wa//zP/wTgfe97H9/93d99JRc2Y8aMy8NF7dWQC8aetVdLLRilMMZIANdwcx24uRkZQ+F0\njORYCKmiVMWMCaOhKIVBYbQ4AqqkRTxoEGvg82OAh8ENbBU4D6vOcHDQ0DhHiIVhTBwfeg46Ty6y\nllcrdI2QA30G0yhShX4r1f21I4dCoye/hFwKi4VHTZ9XLhWNoms8jdOUUsQ4SUngT1mkgXeCPGl6\nztgHjFb7lUBjbjUBclbjjGbdR+qk/rfb4b+foC2vXQkxo/XVWPI+DBvg2Tnw6nDXJCDnzO/93u/x\nF3/xFxwfH1Nr5ebNm/z0T/80n/zkJ/ekwRkzZlwd3uwN9/b2akiZkgvjWGgmdUCA1jvaxjIMmVrK\nRHLT1JqpsVIVpJyoCOu9D5UQk8jgFghZFPNCgFGI7w8NBtH9X6xEN/9w1XHYWkqt2MaybC1GQ9da\nSi4SmI1htfAMQ2LZORSVUgreizzyyWlk2Tm2IRFzZtE4vNOkWPDWEGORNruGXApUaJyjDxGtErkU\nhlB46rBBKYW3Rtr1uVJKpb1Nn39X4Yt8cmXZiXAQiG3wGwl6O8nhmM4SPaUVWotI0C6heDNiUq/3\nt/awZvezlPDV4a5JwJ//+Z/zjW98g7/7u7/jne98JwD/8z//w2/+5m/yZ3/2Z3zsYx+7qmucMWMG\n977h3td5Jie5o1XDayejSNjGDLXirWG9HTnZjowxsx4jpcAYRQwoZdELUFRG1ESASwxRU0sllSRS\nwtNlXjanWzMx/o2MG5yBo+USS+WwcyilaBvN8ardt+APFh5nNUPI6J1AjpK2fwiFp44aQqz7Gbo2\nimGMKG8JJuGcBHRnNKoTzQOtFWNKlFoZx4g1Z0Ey58K6j/vxgLOaMQi5b5ds3Y7z8s+76v+NBr3d\n43cBM+aCM0IMzCoBltbfn5jUGw3uD3N2P68vXg3u+sm++OKL/OEf/uE+AQB47rnn+IM/+ANefPHF\nK7m4GTNmnOGNisZchFtu3OXs2PVrDbVUXj0ZefW056WXb/Ctky0n68DJOqDrJJVbs+j+14LTCgqk\nlLAGnLbkUshVzHHKQ9rm0kjV0uzm/lZm6YvG0HUWaxWLVpj545jJtXK08lgjevnvfm6FsfJz10rg\nPFp5coWKEPxWnWXROpwxOGdIqZBimVr7Z0mEd5pl46iT5aIxmpiFgW+Mota6r+i1Uiway2pKUs5/\nJ+eD25uVf949VylFzAWjhYhhp/caUsHY+6vKH+Rv7bLwOK8vvpVw105AKYXr16/fcfz69evkPO9s\nzpjxuOP2FbHzzn7Gyrw5hTKV14W+T2KVu47kksm5YK3DW81gLK1XpD6TEiQKuhRycsSSiSGRgDhI\npVyRdv1l3SkccpmuhUaBmlrch0uPMYrWGZyzpFowWqON2PU23mGNprEaVTVPH7a8ehJQSrNsLVrJ\nrv6oFMZolNJorThctdRaMN7gnMJaTZ3W9w5WnjhxDhaNpU7vN+dKPyauHTS3BC2tFAdLz+k2kKcM\nyWjNweJWn+AHaXnvzIMkwau3HLuXLfGDYp7dP/m4axIwawHMmHF1eCPz1/u54Z5v5+7a/wCbITKE\njM27arWy7ZNsA8TMGApjiIQs8+4x9dQiAW8YIqkI691Qibly0veEVKaKGHIVMuCD3j12CYRB2v+N\nl/Z/O4nreGPQWqG0JgMxV3QqNNPvrVbieliBUqgoSqrUrCT4a41WdU/+s9ZgjYw0VBVnQasdhMDm\n/QAAIABJREFUjddYK3r/rnWiG6A1weRpFDIlT6WiFDTWYCcRoPPfz46ol4skB1orWeM7x9h/UEKd\nd4Z+vByS3hv9W5tn908+7poEvPTSS/zf//t/7/q7GTNmXA7uh1wVYtlL/rbu4jlviJntmPaiM7v2\nf0gS5GPMnGwSWiv0tCbYD4H1NnG6DfQpYXcbA6WynlrrBbDWohXEVKW7UBHhoCSvxSUlAKsGxgTG\nTMI5HlrnWHhLrTK3d1aB0jTeEEPmuPGsuoYQE94bDjpP62XNLsZC68FqTYkViiQAq66RNv5k8FNL\nRWuNNSKaJA6D4M/N/M9X3aUqFIptH7FW07WiwthN4kM7pr4Q9MzesW+XmJUo570sQl3jNWOY1g7P\niQPdb2V+P8F9nt0/2bhrEvCpT30KpdQdHQGlFB/5yEce+oXNmPF2wRshV+0Nfyh7H3jgjqBxZg+c\n9rP5WAqLxhJDYd2Hc2Y2lZgT614kfvtx+jllUqksW8MwFlIQ0xutFbVUtjGRdtc8GRABpPjgK4Aa\nsEI5YNWBdx5nFMaaadYtbobL1rEdE8uFZtF4iku03iN9A1lVNFqzaNzU1VDkLCqGtcBmm/CNpm08\nrbNi6DPN8Xc7+qKaWBhjxmhZ3xNhIYtVGqsVqShab2i92TPwF529I7ErtTJOTH2t1D4xu9d3/mbQ\neotCXUpl/kaD+1z9P9m4axLw7//+77zrXe/iZ3/2Z285/qd/+qe8/PLLD/3CZsx4K+N8+z9MxLN7\nPXY7JkKUEOut3HBLqbx6MrBoLf2Y9ufcDIFxlF11raCi2PSJWkVAJoyJkArOKHKFMFbWIVIp5CKd\nhpILJ9six6qSgIfiNEVSEl7AruofktxIpkNvGq2Rit9qqf4752lah6qK1dLSj4VKRRoAleOVp1I5\n6CyNb4hBzIq89zx10LBo3bSWZ4hRNhtKhJgT2kq6kFKmkjFBqt/WS7WeppXCXTVcpnVJa5Sw+B0y\nJohVlBCtFpvhKRgO6dZ0SCtFLpJkNOdHBff43t8sLqsyn4P72wN3TQK++tWv8pd/+Zd3HP+Zn/kZ\nPvKRj/Drv/7rD/XCZsx4q+L2KrFSb9nnhltnyTvL3zxVj3V67C5Q7dz/cqncOB0YUkahRIp2eoWY\nCmPIbPrEaT9SqNSiiDGRcyJEae2HWCRYZiiI4lzMGe3OOgFVTknIIgtckfn9A/PGp/GF1tBZx6Jz\nlKqwVt7LwoM2ItVrlMYajbGa5dJjChwcynjgcCGaAaWKKVJKBe8Vdcwid6zg+KCl87JJkHJhMyaU\nUqwWjpQLzoqRUs6KnKSz0jV2+rzzft6+a8LEXPb7+hfBWUONeZ8kWa32EsE7XNS2fzO6EHPwnnE/\nuGsSYK3F2jt/fd5dcMaMGfeP29v/O0GZmMpeY353E98HAH3++aLklyfd+vMEwFSKnMdIEAupoIFU\nK+OQeG3TC5EtZ6w1jCnJfn8qhCibAkOfyaXQeSOugLnw6mnEWgNKQr11QgIM6SwJeKO4aGvAMskM\nV7DOcnTQ0npLKhWjNVYjokZO7y2LW2exBnSVoGqsoWsMTx8tUShizkgvQ00rcgprFNo4lq0n5SLO\niVZLdyNXcqqgK40zvHoa2PaJisj5elvFJ2H6rMcovxMGvqFOnYPdiOZ2Yt15TsFy4YgTv2N37CJT\noNlEZ8bDxl2TgBgj4zjSNM0tx8dxJKVHaAw+Y8ZbEG5ioN+NxOWtYTSiPNcP4tLnprZ1jZlvvrzm\nxumARskmgAGqiN9sh8i6z2hT8EbzWkjkCjEFYq44a+jHIpatSSRwq5JNAgWkzKQUmBkHqa6NFS5A\n5WwUoBEFv+F1ZgLnE4DdDWgX0rSDRlvQwtBvlWU7ZkDjrZKVvhaWjSNXxaq1XH96wcmNkdZpVq0n\nlYqeXPdiKiyXE4mvJNqF7Pbv9v3zNJw/XDX7Vb+S4eXXNsIuUEwEwMq6DxzbRjYLtCImcFba/DXm\nKaES3tSisVM34e5B/vXa9rOJzoyrwF2TgB/90R/lt37rt/j0pz9N27YA9H3Ppz/9aX7kR37kyi5w\nxoy3Gi6qEq3WdM2dFd7usTHlaR6dGCcVvBBkDHBzE8goXr25xSmDbzU31j0piyRwjAVvFWOUdn/M\nmX6UkUGlMgyZMWWGIZERdjypkkplDAEmEmHNoC2UdGYNrAGnhcxXCxQDbvrdvboDO2+htpGEQjEJ\n/zhHt3BcW7ZSXSs4biwaxUHncM6hVGG1aCi1smwM15YdTHoHKcMQRlHrQ3N84MgFvFccLDwxFba9\n8BpQlcbJi0vgFnJfTJkQxB/BWU2pBT+17muBppXFRe0AKiFltltZE2y8nkYohTRtIFh9sXPf/bTt\nbzcfmpOAGZeFuyYBv/RLv8Rv/MZv8KEPfYjv+Z7vQSnFf/zHf/D888/zy7/8y1d5jTNmvKWwW78a\nYqJMVWnX3KmKtpsHv3Y6kPJk+esMy6qk4jQwDoWUJBC9+trAonOEk8Ki05ysIxrNqtNsYsYZg9Ya\nY6QhH5Po/PexopFWtzWGdR/Y9oGJVI+uog2QgZplDFCSBG6rJTFAAUV+Zz2M4d5JgJkqfRR4q1k6\nT7cUpb6DhcVYLX3+CoeLhoNlQ4yZw6WdtgQ0XWPIBYaQGWOeNAySsPudRuKurAhWrejHhFKaCmzG\nhNEK58AbUQZcbxPWqTsc/nayvNpI90BNXYSUC0OQ4JyK8C9CKlPXYVpvVI5l+2C7+ud1HnbmQ/NY\nYMZl4a5JgPeez33uc7z00kt8/etfB+B7v/d7ed/73ndlFzdjxlsVdaLWyzqatJTP39h38+CYsrTn\np25A11rM5EOfh4pR4mSXcuVgYdmMot1/cz1Z3TrNZiwoDdtBZH69UagqWvYohVEVbfV+tcwZgzGa\nVAt5ZwQ0tcWNlcCflHATUgFfJ8fg6XFx4gl4ICA3mV1CoJkSAA/eCbHQWctyYfHWcHzgsd7QeY9R\nilIyq9bROs1h5+gaB+xW+OSarBEFvlJkz984TUoFTGXbK4zT4q5XpEtgrWwBlHImFpRKETJlrChn\n0LqKo6JWYjQ0rVUuOkvJQsL0VsR5QD4ASYqk4o8po5RwPUoVzsUbIfjdTgQ8v054/nmXMRZ4GO5/\nM5483NNKGOC9730v733ve6/gUmbMePR4szfG+31eiLLuBpxjo5/d2HfnOt1GtkOkVrnxV0SV7uZm\nnMiElVwQ57u1QauMspoYM9pKgiBzbYU30IfConXknBmTZrOJ5JQJSZFzxml5L94YQijkLAE8TNfd\nFpmTOw8hsGfgKy1kwVrl37pCUWCDbBHYCl0nGw6dV3hrscbgrbD/Mwptxcb4oLXkmHFdI+ZFKdMU\nK1a+fWC1sBwetISQGWNBT1V5nnTzx5DRSmR+DxdG1hhDRqmK1opSRPfAe+kAWKsppe51CJSCa4ct\n2yHRTiOCXCpdI8nFPjhPlr85F5atY4y39j6MVhgDr94YxLJZC7fjPMHvfHKQUsGYs7HB+ZHRZa8T\nzqTDGTu8bhIwY8ZbERcF7Td7Yzxfte/OOca8d5G72+PPv85u7Y+pCzCEzGYIbIeMUuLiV2olbwOl\ntSitcEa0Arpmd+0FpbWQemNhGxKtsxwsHTFkfKMpKFJWWGu5se7ZjpFcQDuFyZoh5in4FOpEEMxn\nl0rMkBU4Je1ulaW63z3EGNBGkUPdW/61ThIDZzTUyrJraLwVfX8UbSss/6cOOlIGowyrI7HUVVUz\nDInX8sDBwqNQdJ1YHTunudY6xkmyV2lIpYIqxJTEfKbR1AIh5Mn9UAkXwcsooVaIqUx2w+IkuFvV\nPD5oyFna8a0X9v/u+9rN6EupeCc+AN+50Z/JB0/GN6VUci0Y6uTGmPfBfNNHcTTcf7aVXG5dFQWw\n0xjiPB5Un38mHc7YYU4CZrztcLdgn2uZQsQZ3siNsUyt+vPnTLmw3kZWizsTgZ3s73nVuFIrKVe0\nrhNPAFlXU5UxZbZ9pJaK85pc4JmjDpQnxMS2T3irsdZSxgHQpBRBaXKtrDeBrrN8+8bAonFs+sB6\niEIGHDNVV2qU66iTCUAo7IPkDvuQU8XBrxSkDW6l8t+R6EqtaCf8gUWn94H36KChZGkbtI3F6sqq\nbVh2nq41WGvQwBhEtc8oxXobUU4q+RAGrh+1DH2iswatHNucWC4bnLWkFKhq9305Fq1jHAveKdCV\nvhe+wMGikXXHlNFGU6us+Xln9s5/zaQBYLSaxILO1gPjNKNXSrHqpAMQYsY4xdhn8TZwes/3cO7s\n+9+5CzbO3D0QT6uiO+zFoWZ9fuDOBH7Gg2FOAma87XDPm699czfX+62sdtoAO1lupcTVbzcicNbs\nq8EQCplK4w05F9CZ034U8Z5aMVZRkSB90Hn++5U1pSpiSowx4Y3htdOBMWReUwO1Vk7XgZubkT6E\n/Wqc1hqlKjVWUoIhCst/d5vd5SxOi1+AUtB6CBHMRPKz2uC0SPAWLe1y7zSNE55Bs9BS6XeGmCuL\nzvKe5w7YDgkQK9xa5WelwDhpxWslhj+pFvqYsFtDKJWFcyKUlDN9SpDBGEPjRD73dBvwk82uMxpn\nDcOYUApWnaNWJS6JUciFWkHXOIxW+6Su1lu7Nbd/185qNkPEKM31w3Z/PKXCsrPEQUiLIMmdd/bC\n9v7tiaE8/qziv0x9/ifV/e+iBH4YkySec0LwpvDYJgHr9Zpf/MVfZL1e89GPfnT2K5jx0HHRTeT1\nbow7cxiZOZ9VbeoeM9zdDdhZfUvluVMI3KHxmj6I0E2DQSsR8Wqcph8z20E6ACi4fuxwVnHzRiCn\nzKunA2Oq6Cqqf6GIWFCplZgLpRTWm0CIEsy1AW0KFPEACPmsxX8+VNjdAT09T4G34L1DAd5orNPE\nJDfltjEoFL7RaBRPX1twvBKN/5QyR6uGPEn7DlMQXnaWFCvOy88n60SpiVIUm01ELz3bIaB0g9Mi\nemSsIicwRnHt0E8jlIzV8OrNgcVCAns/Zlov+/m5gaOlI0RZv4xJlBtzrawHReclmdgF512iKJWo\nfL8hZbw1snkxeQ7svlPnRGvBGk0qWXgd0+fonEYlbglo3konwpoz1cjzCeRlVv9PqvvfPMa4fDy2\nScBf//Vf8+M//uP81E/91F6qeMaMy8DdqqBl9/oqbuexq0qcNZhJm36M0g5up7Wyi2Rgh5DpQ0JR\nab2lOecGeP7aDhYN623CoGgax3oIeKvYDplXbmyIpdI6Mavx3vLya1tevTlwczPQx8IwtdWHVEhB\nvARyrWyHQKqFEGXGrxSoidHf2GkMwMU+AAlp/RszjQf0lEBMq37OiDDPqtN4K6S8MRYOO8dTBy1P\nHy+xVlMpGGXRunLaR7w1pFQ5XHoOlw3bUXb5S8k4m7DV0cdIrWUi3ImN72YI3NyOACxbizEGpYR7\nYJ2h7yNNa3FGPl/vKn1IdNPndrqJ5Cqfk1bSRWCSBhaNwSnYV9FkCFEshDWVGDMxwmhk/XI3y1fq\nLFiFKKqOIvMsSoTO6n3Quj0QryYlQXj4jnyz+98MuOIk4BOf+ARf/vKXaduWL33pS/vjX/nKV/jM\nZz5DzpkPf/jDfPzjH+db3/oWH/jABzDG4Jy7ysuc8RbHvaqg8zfGSmUIaf+Y2xn/56uSRevY9PF1\nZWCHkMil7ivPPmSM0biq96+xu7aYMl1n2I6Rk34kxMK6T7ROS+u+ZE63gWHMGLvmm69s2W5HEcQZ\nJ4/7Iut/JRayqlAqmUqMlZTPgvou6KsqRMC77fgrJAEoCcrEdvfGcnjgCFF4DZRCLmJRrLWm8YqD\nTsyAxlwYs4gXaSM/D0Mk6MLRwuCMsOivHzW8djLCtLaYciXVTONlrBByYdwGVq3jdB2E7NcaKJPw\nkJVzqdZz2g/0fZQ1P6NonKXUjDWeISTGkKlIh0S4Cnov07yr/neKjky3ohDOfABqqWQl0knnuSFq\n2kDYPX/V2Tu6Tbu/txCzjHVivbJ1vSel+j+PJ3WM8TjjSj+5F154gS984Qu3HAsh8KlPfYrPf/7z\nfPGLX+TFF1/k61//Os8++ywvvfQSKSVijFd5mTPeBtgx/m+/gexujErJLHj3z26P/6J25O55q87R\neXtXhbghJPpRgk7OQg7zU/t4l3icv7aKkOMOFp7OO6xW5FzogwSakqEfEpVCHwr9ENmMiW1IbEYZ\nC4w504+BomWW3sdIioXzyt9CBRSEfPcEQAN+WgVIRdTzYs54p9DasOgcjdcSpMdEn6SaPl41oGC7\nTZQspEijhbRIga5rOFg6jLM0jSUmyUpWreNo5Vk0jtYbjhZi+pOydDUUwuyPpbAdZc3Se+mqWKNp\nvMVYYdZX5DvcjomYM1ZL+x4qMQsfIJfKdoiUUgipsg2Jk02Qlr7WHC69vH4souIYC3kK8M5Ipb/7\n+9hxPLzddQHEm2EImZTK/u9DT50DO+k0vJG/tbczdt4aO8jI6c7kasYbx5UmAR/84Ac5PDy85djX\nvvY13vP/s/cuLZJl1933b1/PORGRWVV9eVoSGH8AoZnH7pnwUMZgjwzSxAIPhKcaeGCEEXhio5ER\nHj0fwRgMGhjZH0DGCIwH70Bg633VltRdlZkR55x9fQdrn8isrMy6dWVVdfX5Q0FlRlbkiUvFWnut\n/+V3f5evfe1rOOf45je/yU9/+lP+4A/+gH/+53/mj//4j/mTP/mT13mZK74EWIp955906oOn7x6v\n3sdVKKXovb1REbAwmpd1wGHKXIzhxt9zvDYnWv05ilzw/JBIqRKDZAgoBTEmpjnxyWcXpCwKByHX\nVSmUSpFKYgoZTaEg2v7cdv4v8tHZGVkVJNroXIMxmn3IXBxmITlW8J2h7xy9MygtoTohFuZU+O2j\nicMUhftQIaaKt7qR5TRTkJF/SvWoo1e64r3Fe83GW7a9x2npRpQWHoJvLnq1VlKqeK+5f+LRCjrv\nWvEQMx8FGGeoVTU+gYzxlxDAKWSMEkMlKdKKMSQOc2R/iDJladdn2ph/kRbGVuxLqUe+gHAwZNaS\nciaVyvk+HqdMz/NeW3GJ2xr4FS+HN84J+NWvfsVHH310/Pqjjz7i5z//Odvtlv/7f//vC93Xhx+e\nvOrL+0Jhffyv7vGPczoy969iOXksWJjJN912/f7sIWA7Lwz/hlIqD057ht5hrZYTaSOWGaUYTgpM\nWk7Jw0wIooHPGawz7KfE+SEQQgQtxSxnyQeYyRit6LWlaA1VEVIiR0Ta9wIHTYto/Ze8gKVg1lJQ\n2hISjGNiO1h8p/HWkYoU/q23dN6hqBhn0MbQbzzeW+okMspN5zBWc34RMVZTrcJ1lvu9Y5gKMWZO\nTjxKKR6dTeQqDQhVdPSciKTy/mnH0Fnu7ToG76jacHYxsdl0YpykNQ92Hc5p4SZU2IaOlGvjG2hC\n29cbrTjdSYDawhE4Od3QX4iNs4z7tVy7VnTechrzY6+vUrDZdigtccj6yqpB7KItm+Yt8Kz32tOw\n/t//cj/+z4M33gSoVzjG+fWvz1/ZfX3R8OGHJ+vjf4WP/7oUCS53j+eLdC9eRvcuxMLzW97PU0jt\nTyY0kt4UhDkep4g2wmbPWdj8nRMJ4a8/O8jov8BvH43kUtmHyOBk3E0tlCzks/0UuDgEYoyy78/g\nVOWisf60gnl+dsrfVSyxABlI6dI+GA3OKLQy1FoYQyBnhzEK32x6U6qUkgljpGYp1iYWDmVmHhNd\nJwTKfYYwSjRvLIkYpDlynZzaVYGuM9QsnIAUMykmQq3MsfD++zsefnZgOzjOzipq60mzPNfjJOP/\naU44p8VZ0cAOx8WFxCiL57GMmmvJjIfAPCqG3hBnWUXuD5GqhARaEblerWIUNHWyBjqnUusNOvam\nGplCsxi+8n7yLQnyNjOg295PV7H+318f/+fBG28CvvKVr/DJJ58cv/7kk0/46le/+gavaMWKZ0uo\nliZBoY6ywBjLE+6Cl2uAQki5McMzY/P41wbOx0AFLg4zxsgJdDc4KrDbeP6f/37Eo4uRkAoxJobe\nMeZCDLUl3kWmAmFMzddf/ANcS9UDKeC1wsuwawyXpUub5hSoIJaKUwmrezyw6QxZwxQKu40iK+ja\nuLxXomSIUfIPvNe41jCc72fZiatKiJX9lEg500XD0Flsu485JXZbz/ud5dF5YEoJh0Jr2G29nJyd\nuPRV03bziFVw3zlKKVDEuqiUysngm/VyJtXKfgp01kryn5GsABBnvykl+s4erX1tk39qrZ6Qdz4m\nE7T6iWZyeT8tEtK3zQzoJjOeNWPg3cUbbwK+8Y1v8Itf/IJf/vKXfPjhh/zkJz/hhz/84Zu+rBUr\nniqheh698nU74RAq+2mUE2etlFwJQfz/h85wMSVqCXRWGhClFPvDzMU4MiVpGuaUuXgYMUZjVCVV\nmhVwJquKrnJ6prbdfwHX5HzTSzwHGfn3trTYXyPCuZAKxgBK4bRic7Kl5HL83bko3jvt2gi87d5V\nZXvSHVP4jJECab3E95JViwOOzLE9b7ky9BblFecXEW9nPnpvy27nKOeZYgzbbcc0Bow2zcZXLH5z\nluAkUxVTSHRtBRBTFjJUlZ2Is4YcE7qRFYfOHl0AK/I8nm48uQBGTICWRMHTjWdoI/uFua6uFEoZ\n6cv7aPEb0NduvwszoJfF9QnY2NKgFvnjmjHw7uG1NgHf+973+NnPfsbDhw/5+OOP+c53vsO3v/1t\nfvCDH/Dd736XnDPf+ta3+PrXv/46L2vFihvxeU9k1+2EK4XzfSTECdcJ8awzllQyZ3shyqVcqRT2\n00zKhTFEDqOcVMcpEqKMwHNJhJBxRoFSKKtJpTKHmUMEUgv3KTDnl5sAyDWLl8DJVmKG9ZHJXrHa\nYK3Fd4Z5Cu1kq3FOicVuhQ9OPamKzO9k48ipUpVinhMxV4bBygg9FGpRWFcxymJ0RCkh2tlceNAZ\nvBdTnnFKeKd5cNozh0JnDZ2VLADXdPjjJH4NWsvJ3ygprLkUOi3OjKWKNK9z4iOgr3waLrr+1JIb\nrVaUKIY/xsgkYRjsY++PZ0lPQZqLZ5kBvcl0v+vNrTgYPm5lvJrzvFt4rU3Aj370oxu///HHH/Px\nxx+/zktZseJz4Xn1ysuHaky5+dgbcilyxDaK/RQZx3gsLjlVYkrEVNsJXlNRhFm+N865yf8KVilA\nduXzXJgOEaUgB45WwK8KJYKyhVw1tirubzuqsiglnYaxhs3G0nvD+T7QeUvfGTKaD+71xCYNTKpI\nGBKQSuJ8X8X3wEia4BiKrAqcb9bHYiQUY2Xba+6d+KOz36Z30MiPFTn5ayVBQN6JNG8/RvEuQAKG\nnNVY/bib4+Lnr1Q92vZarei9EbpAqcRSjmN+pRSb3t2oArlterQU/GcZ9KzpfiteN974OmDFircJ\nz3sKex7b1asf2hK5WzEG+l50zRcXM2dTkAyBLA1EKeWonw+x8Ogwo2olJJhi4TAGxiR+/1kZUoyg\noSaRBMrvhfkVNgEGiQUOpeKVoneazjk6b8klM4eCMUrSBKusAZy1jCGw6bfEIg4+WmlCTsQo1xpi\npffNl0FLyFCuMPQaozXTlNn2wgs43Xo2vecwZbQWHoaQiutxbH960mG0IqSM0ooca1MAyPX1nT2u\nC5bXR0FzcIzUImZCWslmogLWyiRhjply9AQwNzYAt70PXuT2N22Le725FSXm403TKs17t7A2AStW\nNDzrFHa9QXBOsz9ESgXnFCHyePpbY/iPc2QOmZhlEpDmRIgSeHO+j21PnYhZHeNsUy1Yrdn1nv/9\ndORimokRQok4ZSjN5IZayQlyEhKgsY24V15NE6CR+9QGdBXDHOMM3okUMIZKt7VUDdOU0WSwQmrc\n9B27waG0RAafjTOHg5DnYjuda12BwuA7Bm+JKWOd4d7W83A/M88J20b8IKqBUgr3dyIXnIMEFO0G\nT+c0hzmRk+jxndP4RtrbDGJn3NyDjyY9zhpIRRqU5u8/9LK2kHVORbV1Rs4VozW7k5sbgHcB15vb\n3kmJeFtIiytePdYmYMWKhptOYVNMTEEKRqUemdy5VuZDxBjVxvKXTm9XR7fOag6z7Mg3g+PsYiKX\nytk+cD5G8Y9vqXaHOTHNUfT0WkhsIRYOIRFDJaUCVZGKyO9AdPsFCElMfHyU/f+rtJkxRh7HrncY\na3BKUcmkJKf7rjOAFPqYKocpcW/joFRSrXQoiUlWC7kM+k6TkhgJgeL9e4Z7O8/FGMm5cJgiJPjg\ndEMomcOcWwSvZrfx5AyllpZk6Djomf0oMsyUpYEKYRndy87f9+LDMMeMbx4BSxzwFPLla5tlghAb\n2387OGqSIKfbQqFeFd4GW9ybVhZvA2lxxd1gbQJWrLgFoeXG6xYIU6to+Z2VIp+KnBavnv6fVAgU\nnJXd/XgRSKUQgnygngyOi30kpEJFomtLFVe7/ZSIpTIeIiFlQJNroRZxn4tRXP80oC04KzLA8Iqf\nA4fIAXcbz8nGU3Pz6K8y+l/kAK6FB11MCaMV242n6xzjlNC9Yze4xvyv7KeItxZnFM7B6cZhtObi\nEMXqt1N8djaLb8B5pmvPb8rgvG5xvvIcKiV8i5SruPTl3E78la4TtcBhSuw2ilol1Mga3dIAC5te\nGpPBG9LR81/+5KSwTh3Z/ous7y5H83ed7vc8666bfud6+n93sTYBK1Y0XD+FLXKuJTIWLklknTOS\nckelUtvPPRkYNIfUkvvEKjeGSipV/Oyb611MGaXEH8BZCS2YQ+HsMEFRxJpQ2pJTYg6VVJr+v8qu\nXlUotxn+f57nA9kJD51Bt6fFO4UqMvWItdIZhTOazhkqig87KeibXvT9U8h0xogngobTjWXohHnv\nncM4w+AMpSDjeMTjoFDovGWcI4eQ+ODeht4Zem+PskulFEMn/vxlWS8gp2ZvDbVIUTVaNbfEfCT+\npfaEHZppkfeGMicqC3lQoyiUKpMDpWDw9pWam92Gu0r3uy5ZBWlqt8O7u95Y8WysTcBZHRRAAAAg\nAElEQVSKFQ03ncL8laCXqw1CTJnavOjnKMVlnicyhU3nj2Pj2iJnUxJp36P9TEgiS1OqHjXzWkPI\nhcFbDlMGJSuBVAsKzRwCIcj+X2mwFqgwJtDp1SoBQMiATkPVIE4ECooi1IqqUoB7q4Xgl8B7Re8s\n2mi2nSWk5n5oFLFknNHsOk+uBZukwBstLPvcvAQ+uxAngxBTi+yVcX/ObR8/KJH2HXcdcvovFYxV\n5CC2v5JJUPBOy1RicBymhG1JCUszsNS9OUjxL1SslgbAO0PMmXEUz3/vNWNITKGwGQzqGv/jaXhR\nyd9d7d2vS1YBUi5cHOJT8y6e97pXfDGxNgErVlzB1VNY5+W0DkIgq42FvqTC9c7w27ORlCrjnChU\nNp07fnhOY8Y4iZ3dj4mLMYj2PIMxhTnI93IulFipQNcbrJFddQViLcSQmAKkKMpC3RSGy0f5HQwB\nMBr6gVYQO7QxYm0cpPB6q5lTYejEBCilQrWw7YTLoLXwBZzhSOwzFhxi5NM5w3bjsEoxt8TGobOE\nIKZItUAsmW3vOBm8pAHqdsKvNFKfNFrGSLG/t+vYj5GYirg5KmH3p1wotRCTvI5LE6GQwh+irHWM\nlSIoz31tSYSyBrrkB2hyqmT1fNK9t03y97zqg7ftulfcHdYmYMWKK7h+Crs6GfDOcJgjYZQkwFIK\noAhZIoJLrdRSsGYgTon9FKmHgnUSTZyKmP+kXDlMlTEEjIKKZpozIRd++2hGaVBVQRFGfi4i+9MK\nYpP+veqT/wIF+JYQmCJYXyg14aylZEhJTI1mYNM38yBj0VpxsrX0zjCmzKazlKJwnaEzpjkmZrab\njvd7jzZQE0xVbHs7b+i9oebKeYlUBd4YNi1YSSuOR/dxitjWDFitON16PvmkchEjU4ic7wPGGuYk\nZkMf3Bvw1jK3CQNV1hzeGfZjc19sdy9rmsoc6rHRiKnI72p/t74pFZ5DuneXkr+l2Ryb2uR5JgzX\nodTNZMc3LVVc8fqwNgErVjwFVycDc0wSYWoVc0iEWDg7zCha+E/JKByP9jObzmKt4jAWYqns58jF\nRSSVTC6iTT+/iLLTL4AqjFPmEOKR/V9yZYpQ2gqgCQLurAEwQOek2fC9NCG5Qm8ttYDtFH11iPdh\nbXbAGm2FD6G1ZkyNoJcqWlViFPXEFBJagzMaVCFlIU0CTFPi1w/3aK3pvcUYjS5wsnUyQdCK3cYB\nipKL7O+bm5/rrWQvqBZZnCrGGsKcGCvkbISz4I2YMeVKrpWu+Qxos5z+LycLpcURd15TlUwtlpqo\nbymarxtXT+q11huVKdexSFZTXkiVknuw6v6/3FibgBUrnoKUCnNMlAoXh4hzYlTjnOF8jExzFm16\nzqQC01wYYm4mKyLVq0XuowJGGS7miWlOjDGJc16z0KVWDrHl/ALjJEU4I/a/dwWFkAB3nfgBYKA3\nlqRLkypaaUJSFWKgUnijMVasj53RnAxCmqu5UEozRrKSyneYhD7ZdZ5pjngnkwOq4mw/kWtTO9RI\nrYqNM2IgpKSg7TYeqzXaiPqhpIJrp17JRpBJg9WarCshJpTWqCyTmmnOKBRKVYyRpqBUGJwhtNe2\nu3KKlohgdVwBlSqkRoXwCxZy4PMUz1cl+bu+n3/Zk/p2cFwcIqXWo8rlpp9/G6SKK14P1iZgxYpb\nEGJmDInSTlypFMrMkVTm2kRgnAND5ykkcoV5zsxzYrfpAcUYEp23pJg4zKHdbwUUNRWmVCiltAkA\nUCspXGr97+rkv6AiU4Ck4MQqtDV0zrEzFecM3loKFaNEvSD5AZK2N3SW3WDZ9J5SKwopqDkVitZo\npcklsxk8RilCqoxzYDMYUiqkJBbCRVXxQMgZvCEV6J24/KkmLay5HsN7UipYK3LBELNwCJLEOpfG\nGfBeTvsxFVKWbINNr+jbiiE2guLV3bfIBe1xArSsAwbvnnDNe96xeIjl2Ej0zr7wOP2m/fwiPX1R\ntYI0Ve6Z6oO7liqueHuwNgErVtyAEHMz78loLUlz1ihSEra/9watJHL2ZPDEXMUsp0pkcK5OQme0\nQlFJseCsfKBOU2aeIhVFjImYkpj/aCle0/zygT8vC+fkwyArg1Wak41tpjwFo5HTvK7EbKil8KBX\nvHfiGXqH0TJSRnGM0PVOVgDWKoZObJJrbZwCpRgn4VAoo2TcUS616PsxcrLzaKPQWpEz5OaOGFKh\nhHpUVChlQEkxK7UScjpmB3TOknPBWrk274RjoJQ6rnWWU/XciH9dm0LIc6IhgvH6pUxzluLtrJgo\nSbKjkBBfhFx306kfJW6GV3kAz3tSf96CfldSxRVvF9YmYMWKa7g8eYkHQC5QYmboLPsSJe0vFfGr\n3/X8v78+pygwSjzvH+x6rNLsp0ihcraXgKApiD2t7wznh0BIiVQLMVXmWfIF0ht6zKWKHDCmxNZ3\nOKtaGqBhjBlrNFPMGJ2pVRFiwVpLbwymNUT7MdB3kgNgjWG3scwh4Z2QCnPN7TaN1i0+OMqzvNu4\ndmrPZMrxJD/NYthkrCQPGqOZozRaxngOU+JrH3kuvCWlgjeW6OR1GzpLAYxSaBR932SbV7weQJqP\npShXKlOQV+HzmuYsxftqtDDwSsh13ppjw3XbtX5erKf/LwfWJmDFlwJP0zzftm/VSh0z4MX4Jzcr\n34SzjvudZX+I2MYyx8AwWELM7LaO/RjJuXJxmDifMrkUdKlse8tnVnbVOUozgbok/r0JlAJUMcQZ\nBofBYCpsN57wcGRKhVrEqMg5MArGOXE6dHhrKBSMUlQNYS64wRyDe2KociJ3il3vjgXYGgnw8VWT\nihRgqHxwf8Nu4xknUWCMc2wcAjgfE53XbLxt5kyV88Ms/gNG413FWU9IElqkkJWGUZI7YK4caJeT\n81LsQsxXPAjeHlncsp+/avJjtOQh5FTXXf2Kz4W1CVjxzuNpmufUNOVXb1v2ra7J0M4OQaxoNWx7\nx7a3gCKkRNcbBi+nsmXH7b3itw8nrNP85uGBKRZiTHx6PuOtYZwy4xQwSpE1kCDcIfHvWdAIOVAB\n1mmM0qBpu3OL0pqqZIyvWhrgfs7oUfbs22TYDt0xX8G6lm+QxXQnlXLU8iuluLfz5FzpneVk45lC\n4jBKDLIe3PGELqz3gtJa1hJGpIK1Vu6ddFgtbou1mQP1TWYoBkyKcU5cjEkcCKsiZpH6eWceG/sv\neNWyuKeR617EiMc7mRwt79PFYCqninOavrOcv8ImZTUJ+nJhbQJWvPN42od7vYl2d2Xf6qzGGrGQ\nHXrbMuoDuchYt+TCduN4tA/EIAZCUyjs54Rt4+1S4HwfmfaJgxKi4ZyL7LgnyQB4k1gk+F1nsEa0\n/vd2HbkqDhcjlYI3mqIqUyrEXBm0wkDLUygoIpvBMYfE0DtAVBGddXiloYgcz2glDZjVeC/PrTOa\nTSfEu8MUGweg4rymhIqpBW8NuYIpipONNBFWQ+fssUjFJGZOXbMhlpXA0lCA1orQyIDPU9SXk7dq\nzc+LFsPbyHUhZqaQHvt+rTx14mCMIpfLUCq4fA+/SqwmQV8+rE3Aii815ENP/p6KsMznmEmxst1Y\nem/Y9PaYMLcfQ/tgh5RFejbPGa0Uw8Yyp8J+nIhFmPL7Q6RvQTqp7ZtLqcyTcA2m19wAKEQNoBH1\ngQa8Be/BGplq5FpJudJrkdTd21rO9xJNVGqiKBgGw27oZL+vhBA5tvCgaU4MvUEZTa4VVWE7yEfN\nwt53zrDtHSFKgZ9iajtuRS6FvhMjopQKvjUVVivYVLS+tHL2zrBpxEMh4YnLY65i09y3fX9Msq5Y\nvr6poF09uS/2usdT93Po8G/CTeS6KTxZaKUpMAzdzR/Jumn67xqrSdCXD2sTsOKdx21j2cspgOxb\npyDudUbLOPxsPzMGGbc6I1bBMVesNe2UqAihkAsYrUgFQox4bzmcTaRS2Y+SEhhzbX76iTlK4l98\nzRyApXRpwGrxHug8dJ1i4x3DIAZHvjkcSjOQJL3PKGKuPDjtmENk2zs2GyurBGWoCOfBGguUZqoj\nJ+97G08uwvCPqbLpzGUx1uLdv7giOqsxZjnxGnYbCUiCyzH+fkpYo446983g2lohU6+Q8TRwGDPe\niZLA28skwJtw9eS+TACuFt6XKYY3ketuLbSpMHS338+q219xF1ibgBXvPG4by04hXRrCFCHH1aLI\nFIw2WGPEM74Wxv0BY1UbK4N3cuLc14oxol9fTFj2ITLHIrn21jCGzBwiJScqCmsqOIivOvf3FmiE\nyNeUeICEA2kP1sBJ3/F/HgxoraiIpM9bg9YVhST5eW9xKLadY7d1mKrlZO3kxD4GhbcipfNonJOJ\nwqYl741BjJUAjBEvfoUU5AlpADpnjja2MUkj8cE9z8PzmVxkdTPOie1giakwhsTQ2aNt7vLvBDIV\nKI3pV5qPwMnGP7WILyf3ZQJwJ69Hey2u4jb73gWvS7e/NhtfPqxNwIp3HiFmci3HwrId3GO3O6uZ\ng4yIrdHEkjFAzM2EJies1oRQQIun/WJ5m0rm/CB77H0IXIwSgJNyIeVMioUxBGIppEAjtsnvvcsN\n69VTv7dAFvlhMwQkF+gMDN7Q9+IJ0DmR96VUxF3PSiRwLrIWONl4Us70XUfJkizovWWcYrtNTsnW\nSJIfqqklmt+CMQraaH6cZSRequz0Q8yNkCkTl8XOtmSR+sVUOEwBUKRcUbrijCHGQs7LCL8cpwNz\nFGmiMULWXF7n21YBC5biqhSPjeyX215FMey95TCnx6YWi0HR0/A6dPurSdCXD2sTsOKdxkJ0Uqjj\nXj/Gctzt5mXs2xlQskeuWT5gQypiEJTBGkXXid3sNAubPaTKGCTHd4yJEBLjJCNpRWGaU1shCCs9\nF5izKAG6O+wAFPIfu179hoIewElWgbVQtSggvDVS6LVCG/HvD0kaICrsBoexphVnzcZZosriHkhl\nGCzeWgYtkr/7uw6FIuSM1ZpiDBAIUeKRjZeJSogFayqDN499EpUio/HNINp/vagStExfSq2EOVMc\nHCWCRTgacDlR8NZQqibqyybgOm5jwt9lMfROnu+bDIqehtdVkFeToC8X1iZgxTuNZxGdlg96bw0K\nRecq54fAHBLOGLo29tft300xE2JiPwXGefENKIwxyqi9FkJM5AxTTGQqNYv//5IBoIG5pQFaXr1B\nkFfge1EdaCXufSULEbGKOy/WKrbOU5Wi62Q3fz5FrLF4W5rXviaVylff33A2JWIsshZwCtc5UszE\nUri/60m54Iym760EATlLjzy/oYjBkFYV5xS9s6RcmoOfxWjVpgKXBddZkcAdX7OFvNnUFqHJMa3R\nzCG3fIf2ujbuQK2PE+qun+SfKR2thRgrWsF28/j06PPiqkHR21Zo19P/lwtrE7DiS42KEPZCzBij\nUTKxJqVCrM1Vv4AyiofnE4c5UmuhKtG8lyzpgrICkMKeKsQiHvfzXIgzWAeUyxjgS0riq4OWqwXE\n04ByuRYoiMlPVVCzhO5Ya9j2Bm/M0SdAK4nq1VrTd4ZSDOdTRiHOe6rWFoQkp/mSxdd/6Jyc8rVu\nksMqaw8FGk3vJeRn07lGCJQI4KGzT+zzr3v0LzvqlIWEqZpNcAiZmJIEF7WC7zq5n8Uf/2kn+dsa\nxP0hYq2W6ZGTZ/Dq9OhVYC20K94WrE3AincazzJsWW5aSGApNVMap/EZDnNinCIlFT69mDBGTse1\nFHKGkgufno8U4PwQSaEQUyaWhFHmWOVzgpAviXkg//lehcrbIA1ARjIHfJX4YWtl8lBKBTHco7eW\nqCudtXgvAUDeOayVtUdp6TuqaQk3vYEWDCSeADI5mUNFG7k9xIzViqwrU4i8fzq0h62ai5/CWnfM\nXlAtxCbnwtk+EFJGA9vBP/EaaSW+AlqLS17VEhwEkHPBKUPOYjO8nPiXicDLjrXLDZ3ZKpNb8a5i\nbQJWvNN42m737BDIuTDHjNaKbe+YQuL8EKlVGP9zyhzmyBwKKVdSycxBNO2HWUbkn53PaA0pFmLN\nQMGgKFSsk0Kc8+NkPaPktL74BIgB7os1BYa2/9dgHKQZtIPOgrGglSGajKkSDWw0hFQ56T2mxf92\nvcUZOUkPXotJkneoikQFa82md8wpQ8lMc2kZCLDtHLUqUi7sJ5ENDt6RWlGmTUt6b1FaCHnRCHlP\ngokUtVac0cwxc34IbHora5grxVYKuGUOBdcrcpZG4sFpx/k+EUs5hgAZrdg0X4Grr/VNu//bG8TV\nFGfFlwdrE7DincdNJ8JPz0YOY6QixjDWGD47m6gKxjGSiqwJxjnjrcaYineWOWQOIXGxj4BiPwVi\nzuRY2zhekZVGq0wtUgy1HKYxjRtQAAfEK0ZByzl1WRUoYOPkRB9vCBbSiMwvFdBWRv0aKf41S9Nh\nncZTMdbQG4t2ir4orFIMg2K3dfTOcbLpxEzHKDa94jBHNr3DGIXVqhnwVLBQs8j9lFHkVCkpYZRm\n6AxaK6wV8mRMpfkNKIbBH5n7g5dCO4ZLdjxwXAeMc2Zz7/H9+1LMT3FMIWPNJdN/DpnBW+aD6C2t\n0eRc6fxlIX/a7r/esDII8VIZEBoHQW57u3b3K1a8CqxNwIp3Htf3ryJHqyjVvAGq4jBF5hjZjxLx\nuxjF7KfIJ59N5Cpxv3Oo5JKBSq6NnIZiygmrDJ01OBRFa3oDCsOcI6GpCHKS/3RKiQnOshJYXPw8\nUtSNhr6XtL2UMnOAmOR6NfLDuYI3cvLvvSerjFARwTlHKS3UpwBeCHmb3qCQff+289w/8XTOCknR\nKIzWbKNDo/BWoa1hN3ge5QlrK/ecRx1kZSAneeg7g9GKvndow2MhPMvzv4TdgDRiY3jyZzpnMPr2\nvXvvLYrHA57u7STxUP6J5D54pxjHxMlW1gtPI4fe1CAu06MpirujTA6EIPqmLXSvTzRAgpymkFaf\n/xUvhbUJWPGlw/Ihaq2hxExMhf0YmNrawDUb25KlyJdS0MriLaQ4czEVcs1oWh690bgYCCVii4TX\n6GZ/G3MkpATa4rokyXYKWlqtpNvVZufb/m41KC0kvSWZz4r9PlrLZEEsdcF1oI1iOxiqs4whkXLG\nNQdD5x0ng5AArTF4Zxm8YdtbPnywxWjFg9Newn+qGP7MMZOK2PmmtsY4GRwhGTnpU5lDamN+6J1h\nt3EYdRlxa81lI7U0YLat/EPMIp1M5ejkB5fJeAtuGuFfL9o1Vk42HZ9+dqAUUT1ApVwp2E/DbQQ9\n5zRTUEcXw6vvnTfFDbg+0RhjayxzvbRKvqJuWEOAVjwP1iZgxZcSxkoRrUCpBa2VEAIRu9tcFO0z\nlo03VCWJdZuNZ46Js1HhvcElUQL4rsNkkax1zpBiRjlHraHJC7PI9ToZ16cMKFENmCbdU63WeAfe\ny0m+s4aSQdmEV+I1sDgWGqXIVDbOY7UjaQkzyskSS+XUW5QFZx2bXorxpvecbjwVkTtuO8PQOe5t\nO4n9TYWQ8jHxbz8GOQ1ri2rxwVtvZXdvNbteHPg6ZwhN078bHClXula0r2IpZNveHZPx5pjpveV0\n65/4uQVXC9zVArwoDZwxYB6XGC4F+2XS/Jbx/w1DhDeG6xMNmbiI5fXVn1nUDQvWEKAVT8PaBKz4\nUmFxprPaMBeZSXsrY+j37hl++3BijJGYCyVLbK7x7ZTaW6aQ4GQglAMpg/eaMhesMyhvsE7S8mIK\naG3IqmKMJqfSdstS7G0b+Vcl+/sSZT3gNC3CuOBUZU7Qe4OtEo1bWiBPbyxdJymHuZ2AVSPibTvN\nPmTuD5bd1qG1jOs7K1TCzmkKCm8UJ7sORSXmgjbSWNzbdMeCM3h73OePIWGbmdBiLmStxlrR+e8G\nd3Tk6zspONeL7NVCtukth0lGIteNfJ43yGYh+C246dT7tDS/pyXmfVEtdFd1w4oXwdoErHgncdMJ\nb/nQd1ZTY6bzhouHk+y1e8M0JXxnOG/2tKc7y9lFZIoy3jZGUbOEBzw48YwXmVlIBRRd2XjHh/c6\nitIiCYwFVQ25iGBfiSkhnXNok+i0WBRnrQiIKY2M5QveWEBhNLjOYVIhlgy1UrVEy24GxzQVnFWc\nbjwxZraDp9TCB60p6Lzj/tYTS2UOhZPBcJgyu63lwwcbrFYYY0hJZH5LAXXu0qxn2eebVjyHznIx\nBrQSe+BSZF1wVda3vAZPRuaqtsMX/sHJxh9ve1l4Z7BaSTrhNY+BpWDftPt/VqPxLNfA26YId4Xr\nTYnWQBXXx6s/s6obVrwI1iZgxTuHm05454dAbH7+qraMgFjEEreZBJ1dBMYp4SzoqgHFB+8P/Oa3\nI3Mqcp9asfUObw01jdQEh2DRlBbDCzEm0efXJoOjUo8mN4bt4NhUS61gSxE2f5nFrEcjq4cifIHe\naPHpV4rBOOgV1uvmOKf58IEhZ0XfG7xVOCupfg92gxgWURlDRinFyWBAaXYbzca7toIQmR5XinCp\nIkdYTsspFULMKNV8BpLcX0xi8mO08Aic1RA5FsqbInNTzpQq04QFN52uX+QUrrXiZOufWrBf1pzn\nNq+BZ00R7gLXm5LeyXPYkpVvVDcs+CJMMFa8GaxNwIp3DtdPeEs2/GGMGKMR4pik2AmRTJwDe2/Y\njxEqOG+pRYh5/aCZzyKpKLYbx8VFpJSK84Y5ZbadxmkLWqOVhM+cbAxqhJIT1hicFg6BNVCRfXMp\nFZ3FQvjBrkc1ktdcJPDopPMYayklk2tFO8uu15z0jqoUMRe2fS8mOcYwEnBWc++k46T3xJw5O0iU\n8Xs7h7NWZH9Gs+0cvRdVQG4hPddH8lcLXS4QUmKcpQjlVHBWHquc8MXv3/rHSXTXYY0mXYnQu604\nv4x3/7PMga6f3K82GhIN/WR88G2/93nXFa8aNz3G5bFcVzesIUArngdvbRPw3//93/z93/89pRR+\n+MMfvunLWfEG8XnHrjd/YIs7oPcaa8Ukx1nN+/c6fns2k1MlpsRhjpwMDk5EopdzZYwRCkwhYazh\n1LvjQVophVHifjenERMU28HTWUvnRFKXURhd0RqmuWB0YegdVsEhFbpQoK0JoJJLxXrLSa95sOvQ\nWqOqQqlCqpUC3N86ai2SEqg0VVVOtx2dMcypsBlci+6VkKBcqqTyVU3n7ZEIuDzHC2lu0clfjAFq\nxVpDirLeCDHj3c3jd/kaphbTDAuhUWSKz+Ob/6KOf0vQUKmVEPNj75WbTu7LNSxN4hI69DZIAW/D\nTQW97+wT31tDgFY8L97aJuB3fud3+Ou//mu+//3vv+lLWfEG8ayQl6XAL4lsIB+UU7zcRUuxsgy9\nI6ZLq2Bl1NHQRjuoVcz1dxsJjrmYJrTSzZZWcXEIaA1Oa86mmRAq1IwZLN46QojEWui8Zbv1bDrH\nr9SeWuWkHdr1qizyO2MURokFsbOGmCv3NobUVcZZCHPWKEIUN8P373dYpZpyQFjhJ51jDoVSKr1T\nGG3oO4Nr645+sPSIHbA1Gmc0SmsiSRIDncVazWFOdKXS+8uCEmKmNNOkeS6y3kgZZw3OCo9geWWM\n1uw27gmSXqmX+QilXhYxq5+vqL/ICfZp75XbTu7QzJmaouDqbU871b/tpMH19L/ieXHnTcD3v/99\n/vVf/5W+7/mXf/mX4/f/7d/+jb/5m78h58wf/uEf8md/9md3fSkrvoC4NeRljEe/fxAf+au68FK5\nLPbLCVdfOtNppTjZOEpB0uKS/LFWMVRLToHON7a70YSQSKWg8iLqh0wVs6FRjFqUkZPj/V1P3xlm\no3gveow2VBTjHJmiRA0bLX4EfSdNxsUoDoPeeJQqGOUoVfTfp4PDefHa3wwDXWewSpErR9c9azSD\n70mpMM2ZGOT074z4A5RaMRoUCuM0vnZoJemJqMUdr9BfqvQARUjpqL8PcTFXAhRsmxTwpgYAwFqR\nS16NzF3SAe1jv+fV4Gkj+tuwmBS9qBRwHbmveFdw523rH/3RH/EP//APj30vhMBf/uVf8uMf/5h/\n/Md/5J/+6Z/4z//8T3784x/zp3/6p/zXf/3XXV/Wii84nvaBL4Y/+iiZ2/QOBa3QVxQSnTtNmTFE\nfv3ZyMUhsh8Dn53NUKFzlvdPek42XsbZFVwbo1sjTn5WgVbNolYhJjyDuNo9Og/sD5mTbYexSnb8\nG8/pxrEbPCVLbl9diuhWs+kN+yXa2MHQWfrOoK3GGYVzDm1E2rfpRIkwdIbN4Hhwr4cqqoG+M/jO\nkkuW6UWz2VUoNhvH4AwaaRxCkiwAKdT1sULuXdMwculNIJMTuT9rDCcbf2MDsKBzhsFbBm8f27W/\nbtx0fdfVDDfd9jRclRK+LROAFSteFHc+Cfi93/s9/ud//uex7/3Hf/wHv/u7v8vXvvY1AL75zW/y\n05/+lD//8z8/TgQePnzI3/7t3/Lv//7v/N3f/R1/8Rd/cdeXuuItxK1j1za6XQhdU8jElOkWxvQ1\ngleltuIupj/jLMY++0lS7A4pooEQE4cpoJ1CI43E0BmM9cwxM0fxvS9Uus7yoBHqSlEMncZouT2X\nyrYz5AK7wdB3lRQr3ioOphBzbidJMKZSS2XoPEYXaoGYM9oUOmPoO0tFcbpxqKoIsTJ0YIzhwwcO\nqzQ4w/58xjnxAkhJVgq7reHiENgNThoAb0ix0ndI8Q8ymVgIfqVWStuXL3KzGOXr0iSM3mm8MY+Z\n+9z22lXFYxyOuyyYTxvRLzkDN53cX/ZUv57+V7wLeCOcgF/96ld89NFHx68/+ugjfv7znz/2M/fv\n3+ev/uqvXuh+P/zw5JVc3xcVX4THP7c9M4i063lOh9P8uNa87yxzzIxzImcZ9dZaefDelt5JYt0c\n0tEGWCtFt/Fse9/IeIlxTlRg/k1lMFIEqIA3+CIrhA8fDOwPGaulMFpnOdtHfns24Zxj6CybwVJz\nmw44xb1Nz2fnB2IU69ptb+mdYZ4j2hh+8/CAdXC660hZoXUlF8ip8v69nlwrhzsLMT0AACAASURB\nVENgjort0OGduPGd7nqsEXOelAvOKYw1PHiw4cP7Aw/PJlLakXKlUtEoKiITPBk8X/1wy9DJiT3G\nTAF+8+lBVhgGTjc9fZs89M6yGSTE59F+5jC157le2hmfDB2bwaH104lzt712z3oPvOj7ZHnv3/T7\nFpQihEFoJkNXrv1ptz0PXuZ9/SrxRfi/f5f4sj/+z4M30gSoW0aHnxe//vX5ndzvFwEffnjy1j/+\n68QtePpIdsHVva5zmvP2s4/285Eg9+C9LRdnEumrEWLgIkVTSpQAxmg6rxnnRIoyHfj/Pr3AGUOM\nmVQqc844o+itYW8V+0NgSkWy662EAnkl+npVM+NeeATeWEpSPHy0Zz9GDmOhUOgn8RvwzmBMIoVE\nRaYMp4PhbB8puWB05TAGQixcTDNWa+YQiSHh73c8OtvTeVlrOKuZJ9BbOH80omLho/9zwie/uWAe\nIzFmjNVYIyfjWcH//m+l84beW0JKpAyfnY3EKM/LeIjc33kutOj+T6+E73z6aBRJZPME6Jzh0ynx\n8OGzzXOuv3a/XfwWnvIeeNH3ydX3/m3vlbvEy76vXxW+CP/37xLr4/98DdAbaQK+8pWv8Mknnxy/\n/uSTT/jqV7/6Ji5lxWvEy2qrbw15sZqLMYpTnDHQiHohFWItnJ1HcikoXTFK03eWnIvIA4tI14zW\njFMilywBNVXTWy1KglzIVeGMbsmCmloyX3m/p1bFo7MAGjadwWiDsYqz/Yw1hgenlsMk16apaCqq\nak53jotDRmuN95b7SpNrodZKyXKS3fYdWhXhEzQy3xQLzlVOBnES7L3FGYnwtVb88+/tOqY5Ya3G\nOXH2AyHyxVRQWtG1lN45JKhKEgxLpmTDo4vIe6fdE657u8ERlsJqbx7lP6/X//O8Bz6PBv9NjOjf\nlGfAihWvAm+EzfKNb3yDX/ziF/zyl78khMBPfvITfv/3f/9NXMqKLzCs1jgj4/JlfJty4RAjv/50\nJNUMqnKYMxdz5LOLiZjzsZApJeN67xS5wH4MzFMk5orWmloa+csoem8opVDaKPxk67l3r2M3eDaD\n497OYZRGodkNDms13lmsURSEWW8MvHey5SvvbRg6A7Vy/9Tz1fd33D/p8F6z2zp2G8u292x7x+mm\nQyG8hJONA6U43XYMvWXo7WNTtcEb7p30WGvovGXohIynlKwRhk4apRALFeEBUCtaaeaUyaWQcn1i\nZ2/b6b+74tN/fbf/Msz8FStWvHnc+STge9/7Hj/72c94+PAhH3/8Md/5znf49re/zQ9+8AO++93v\nknPmW9/6Fl//+tfv+lJWvGG8am21d4Zxzswxcb4PfHo24a2hpAq1cBiryP9iRmthzT/aBzov4TvT\ntOjgEyFJloAziikkfvvoQMqVobd89cH2GI8bY2I/JzHisQU/GKzRzCGRaiHkTCpisZtzwViNUhL6\no5Tc94f3OjHdyYVtb8mlcDJ0dN7y2dl01OCHlKlU3jvtj05/RkmGwa6Z/yyEPmcNnbO8t1NsOitk\nP6AbLDGWozOiboY4OVWsrWy1k/Q/5JTvb5DwvUo53PO8B952Df51fNGud8WKq7jzJuBHP/rRjd//\n+OOP+fjjj+/61694i3AX2mqlKnMsDKVFxhYp/LlIEU5lCQ0Sr31nFDkVQDWZW0dMlVpnUqrEWjkZ\nHHMsTFNAafj0bGS7cXirqdXg275dYnsrOWdSFjLexospkLXCqp/nzG5jsUo1V77Kr89nPjgdqFXi\nfDedJSVpFD56sOHhRYDGHLBK03vJG1BK1gCL4dEUE1ZrvIOUL0lttnkhAE0DX4/j/M5rvNOEqCDJ\nzxutcE6ihm8b9z/NgW5JZsxFXBC9vYzjvf6zz/MeuEsN/l2E/qyeASu+yHhrHQNXvJt4UTvTZ31o\neyfudd5bYb7nQkWRS2npfTLKd05O7JtORupKQU3CdN8OjpQK+xxJJROy5Aj03YYQpYCqMdF3hg/u\nDa1was7HQC2FMYi172Luk3IhJdnvO2tw2pBKwWmNtbJWiEnCi7YbWSM8PJ/QzVxnP0ZKdRIcpMWz\nwFvbGgE5sY+zNA+lwhwTMbf7vKbZD20Kstj7Ltc39JIbUEul8w5rZI1htL45pOeWwnY9mbGUyhyz\nrA9uKYTP8x64C9vbuwz9WW16V3xRsTYBK14rXuSUdHEIxLzIruSEedOHtrMarSS1T2vRt296Byix\nujWazppjYM7pzhPmQi0ZY4A54b1CGcfZhST+1QonG0/0mf0+kkpBaysOeaaiqHij2Y+JaUqUUjhM\nSSJ+vebRIcou3oiR0MnGo7Wm9xqlLM5IVG2MGePk95VSAMNu60nnM86YxlkwbZ8vjn1Waza9EAbn\nmAkxc3GITCFSqcfd/cLOVyCrBQlBxDtpmN47tVyM8eiUGLN4J7xIQbzKBXBXJhBPw/O8B+7iNH2X\nBL719L/ii4q1CVjx2nHT6f769wBSuSwopYgWu1LRQaR+WikOU+QwJVw72RqtUCDmOJ3l0/OZXArO\nKBRV3AOrYruxOKeYQqbvHEorDnNkt5FTceesEA+twmpF3zmsVoQoUsGqxT7YatAtGVBZ8ROeZskG\nuL+zhCiFFaVwRkiFsQUXwWUMbOfNMdLYW82D0641AY6TTTuht6Aj7zTnh8zcTIm0FifjXFrwjdFU\nJX+fmhmQNVKgS7MaNlZjtaZ3st6QhkUcEV/2dKyuTGretuCdFStW3Iy1CVjxWnHTSPZ8HzDmMsUu\ntx22nGKvjrYTpcLgJY3vbJxJzaiHCjHLfn67cWw78d7XWrFvEcJDI9PJ/WoGL1/XmgCNUp7eSjKf\nt1qIdc7x/ukgEjulSKVcSu5QbAZHrXAxRe4NnosxMqdE5zW1woN7PbXIaXwziMWu0YpxTsSU0UaT\ncmbopImJqUik8RTpvagMJOp3Ka7ii1CplwE4FXIW21/Tiv1Ro58zVuvHHBRVcwPsvKFSn/DNf5HT\n8cuS4u5iN/8srAS+FSuexNoErHituGkkm9o+/WohKLWSc5Vx+vK9Im5sC3kt58sC1jmL0/q4DvCN\nEKe1ZjM4Yiocpkjn9JFRv9s4SjUcpoh3hpON/P4xZPaHgPUaY2SH3nsZy0+zKAlEdqeoUTgHFRjn\nSOcN91UHVWyFvTPsBkdnzTGrXinF+RjIFSEpKkkmDFEUCnYxmmkNkDGqJSEathvH+SEcVyNA8/PX\nx+J+FUs07hI0tFzDqzqpvwwp7i5386/6WleseNexNgEr3krkIh/WIeXmkidkvO3wuDY+pqZ517Db\n2GNcbUjCWI+5MM9ZYnStlrVCKFgrq4ahsyjVP1YcvNVsHmyoSANQjqS/Kp4B2lIRm9kYRQ54spGp\nwqwTQwdDJyd+rcRs6GTjcU7kd4sc0aQqiYZa1ghDb4SPoGDX+2aLLOoHYyTmd8lNCO2knotMP1BS\nXI2W2GG4zE/IpR539cKfuDz9Ps/p+Fmn9hclxb1Jc52VwLdixeNYm4AVrxXXi05ImZQLxkBMUuxj\nylhj0FoKb0xiYrPd2ON6IKTMGBIxFlyLgs1tUmCsOp42a66cHwJVFTbO0XeWYeslVfBK0blKajPm\n0njIGUMoEk6km0RPrllkhxVFbaoAazWd908USa1FHmi1xnqxK85FHwmBKWXGOVMLbDqH87IuKKUQ\nFqmf0sxRTstWa5STAn8xBhKPKyCWdcDy2OaQ0e7SJ+BqoX3W6fh5Tu1fpBP1F+laV6x4HVhb4RWv\nFf7KKDokCe3ZDQ6thJQWYibXenSoM1rTe8umtxilj6f8UiqDd/TeklLmbC9pgIM3WGWgNsZ9BW0U\nNSPpfrVy/RyqlTqS2pbri7lg2gndWkWpYkdsrDryFySm2B4DZ7aD5f17A/e2Hb2/3L8PnT0S7p4g\nPyaJ8W2HeZSGEDJnF4FcZJzvrUE12eMcZC2QcmnSwEophb6zTf9fmIIYDS14VuTt026/CyfAl43u\nXbFixavHOglY8dpxHMlWhWtOdstJvMJlhr16Mo3NOc0UFFqBc4pcZS+vEEe/XCoplaYsUKCq8ADk\nKyn4SHFeis7107DRYioElwV78BIZnFN94rqKrigrzUPJFd0eU+/t4z/Xiufi8Fdbal0pYBRCXGy3\njXMmtRO9vhLHG1PBaNH+59IKdyeugKoCSvITzi8iIWa2gxPb35fIZrgrrLv5FSveHqxNwIrXjpuY\n6VcLa7pBa341lW1h3scEWonpzsm2J4VG2HMwj1Wc8BoxTqncpH/iF7Ds1xcsjUlIGWMU45RBleY0\n+Pge/OpKY5lKSGGX+1uaGX+L+95SBJ3VzSug0PemEfvk9q4zqNqaHXu5v3dWX8YjO4NSEoObjj4F\nmpwK2ihSrpwfIrumYFievxdh5t8Vo37dza9Y8XZgbQJWvDHcVmC2G0dsEkFou3kr7PmU5PtTEEmc\n2OHWy+LZClrOlZQvT93eGazWDL38/Xrh0UoR264/5Qq6UIsU9E1vjz+zFNLlJFvKTYW0kFp+wXUb\n3UplCpJ5n0tl04tfwbEgNs3+vW13zKcHaYx8a5z2Y8Todp9aphohZqY54309riqAI6dCt9+hFC/E\nzPfOcL4PpPZvrFacbP0TP/eiWE//K1a8HVibgBVvDE8bC189mWstvgDLqTulZgmMjOxLrUc73eV+\n7p90XBwiSsn9920C8DT726vmRM4YYi2YdqJ25vF/t1zf1QkAQEyXNroKieyd5hZTbBTGtARDxE1Q\nqyUYKLfHIauHzhvGOTVf/iyphI00WUol5SSPtwhHwnmLtfpIDDyEgDEyaQgpH5ujF2XmhyiTkeOq\nxKjXIud7UbwJ34EVK94FrE3AijeK62Ph6x/mzurjymCp0cupVF3ZlXtnucgz/kqxPt36xyYKTxtj\nXy2OUmiFxR8TnGxunhx03jxxsl4KuXAccrv2ClRSkT+LTA/E93+ZhuRS2fb2+Ls6LwRAmXWISVCp\nokoIMRNTkZVAVdw76Tkc5jbJKOQKVqmjMiGmwmaQoKIXwfJ4rns4vA453/NifkO+AytWvAtYm4AV\nbxRXT+Y3ydFCyjf62d+0p78pkvZFCpVWiikm8nEgIKf8ZRWxuBhqxdE86Po0QzXiYUyZOeQ2oTDH\nVcAynu+cOU4NFgMf3OPM+cMYSVmyDFIqKLvYFss1pVyPjRLAtndcjBFrDEYXcpYsApE6XpIa3zXX\nvKtrk+P33rJGZcWKtxVrE7DitY5Sb/tdIWYOc5KQmyt7dCmoUjS1XgJ01FHLv5y6h84+kaD3Ili4\nA/9/e/cWG0X1xwH8O7Oz2wtoECEFNMHwYAoGIxFMxICKKQZTpdKQGkO5WC2BAOFF5RISufiAEsSo\nD2yKJJhIIoZC4cFQg0hDVDQ1SIBGNJJAI/UfkopAu7M7e/4Ps7Od3c62e5ud3T3fz1O77O6cc5bu\n+c25/QZD5loDwMxDMLbKb3boYTMYAQBDAPdC5rn9FQFfwmiGeYb/0HG+UQEgtq5gMGQgbFhz9uZ8\nvWLrxIGhzktRhnInKKqAEQWisdwJ4YiCioCGMbH8CZGIuYBRUZT4IkA9IqDF2jaeEyBsYGy1HyJp\ndGS0nQPlFjQQ0RAGAZIr5BGuqa41tEvA/DcrWZBfU2OHB5mdYSD2c2W1Bj1sQADxI3ArKzT8l0N5\n7Sv2RexY4LGxLXtRIYbNpQthdqpWJ2p1pObCPyN2ZLEZBPg1FQODEWiaGstqqCAaFQgbUYypdA5c\n7NcLaD4zeZKInfuvqvE5flUxF+pptqCoulJDRVRNyAmgKGbgZB6ulP7K/FLYzmeeuJj0GAMVorTw\nr0RybhwGk+m1Qrp5LXtnGF/VriiortLiw/3Wz9ZOgHx+2fv95sK6yoAWDwAADLtbH/V9Yiv8A34t\nPtIRjXXgY2P79lVFQcAh9W7ykb524djpilWVvmHTH9YhR6qiYIyt7Fb57fkC7IFLOoHeaIcNeS05\nF4JVP64HIBodRwKoaFgH6DjddWq2XWla7jvUHFl31cl3vlUBDYOxtL0Wq2N1WjAoFCSsxjcMgUDA\nHhwM3UnbgwB7fe0HCoXC5jRJRSw4sTo3+2hN2DBzKADmWgIhgFDY3KJozyWQTSdejHf/yXjuAFF2\nGARIrpBzvqmuZR3+AwydHKjE7vrzLZ31D8kdiqqYWxSt/f32bXzJkofPNVXFmErfsKkQ6/rVVRqM\n2Ip9e5vbpycGw+aQtz0AsC9808MGDMNKcWxfbGi+n7Wmotg78ly4Gahw+yGVMwYBknNjzjfVl+ZI\n17IeVxQzSY8bX+jprn9wagO/X0VUmGsS/Jo6YpDkdFfqVHdFGTod0RpuT34fhM2dCE47JPSIEV+D\nMCb2Ptb7C9t1MsEOL5FXaY+JCoVBAOV1KHW0L81U1yrEcG4uKWxVxUwEVFUx+nVSBVL2OoZsWxGt\n3RCpsvMpijkKEU4IIJT4IUpCAIYRjXfgVqZF+9RDOh0WO7zhvEx7TFQIDAIor0Opo31pprqWl/PO\ndwZ0DOhm4KGpCsZW53/RwZ17Q0fvRqPRhBXt9t0QCMOxHaKxzh4AzLMSBdRYZ6+q1nMEjIiZ9dC+\nmDHd6R12eETy4QoakobT3ey9wTAQ27MvhLlt7/ZdPeEIYYseNjCoRzCom8f5puvOPd1cuBe7Rigc\nxX93w7Gjj03Wbggn5imBajwngHmKokjYOmlVLRBLv1xh2y3AlfLZc2q3Yt0lQZQN/k+mvCrmL82A\nw1Yyn8NcuxGN4t5AJOExa6g8PvweFQlz/COJOJxoJyAwMJh4jZHaSY3tRrB37gl108zOP+D3xbdR\nmscHR9MOWor5s/OK0/8ZBlVUTjgdQHmVzULDQi5GS157MKCn97p8DpWrqrkd0hg6nxg+VU154qHT\nrooKv5ZQJs2n4n5bdr+oMnTCoRDO8/tO7V7sBwN5gdsPqZwxCKC8c/rSHOm44EIuRkvu2DRVQdhI\n7GB9qprX7YnJ1whoPjMzYTy9rwJfQMTv1pMDoVSds/1Qp+TAabSgJVW7+zQFiA1QsMMzMRiicsa/\ncso7+6p2PWzg9t0QBvWI41B6IU8sdDK2OgCfOvRn4FPNO2pNHX4IULJ0h8qdrnFftR9+1RfLPaDA\niJpb/YyocJxqcDq1L+HkP4ejc0eSqt2NiMjoNEEiKm0MAsgV9jl0I2p2/HrYiO9fL2RHP5rqKg0+\nVR1xBCDXuWGna1hZCPVwFKHYosN7g2EAw9sn06N+Ob9PROngdAClLZO5+5R3+JHosBTAXmep09TE\n+fRUcpkbTr7GvXAYRhSIRAwMxk4GBIaCJb+mAjnciY82v18M7U5E3mMQQCOyOn49bJ5Nb6X4zWTu\n3koBnPBYrMNRFaVkFqMll80eFEUiUWi2ffkjBUhDK/VFrCOOdfx+FQHNnOuPGAKVFWrCazJdPGkP\nWsyTBSMJry+Vdici9zDsp5TsQ/pREUU0lpM+msaQvj0wCMRy3auxPe7JQ+nFnqXOib1tQmEDYSNq\ndqop5vSt1wzqEdwLDXXGgLlTwGof6zRAe2a8bLcn2tdmWK8VAhjQI/j3jo5w1EAklqmxVNqdiPKL\nIwGUklMnYx1SUzHKnWjynaa9U3PKvFcsd6Hp3nHb28Ya5bBPdyRvH7Svxre27AFDI/5VlT5Eo87Z\nCXPdnmh/vR4xAxVAABHmBiCSHYMASks2c8hO2fiKmZvbFe0dsX16pNJv/gkKAP4KFZqquhoQORyE\nGA8oFAUZTzkQUWkr2jHAzs5OrFixAkuXLkVnZ6fXxZGSvePza774ULV9SD8SiaY8SjfTFe1ey2S7\nYsJOAXXosXTO67emR+K/+32oCmjQVDXhXAWrXSMOxwlnMoSfcqeAZltzEMn+REQiKl1FOxLw3HPP\noa6uDv/++y/effdd1NXVeV0k6SQP6QeShvRlzjpnb5uANpRiGHCe3kgeSfFrKgzDzAmQHCQlt6vP\npyIciULzKfG1A5mMFtjLqqoAhDJsh4bPN/zzYvIgovLn+kjA5s2bMXfuXCxYsCDh8bNnz6K+vh6L\nFi1CMBgc9jq/3w8hBPbv34+Wlha3i0kp2BftWXf1Vqfl9UE/+Zbp3np721jn9ad6vhVA6bEtgZGI\nQGXAh6oKbdh1ndrV51MQMUTWi/isslb6tWEBQKmM1BBR/rkeBDQ2NqKtrS3hMV3XsW3bNgSDQXR0\ndODkyZO4fPkygsEgmpub0dPTA13XsXPnTtTV1WHOnDluF5NSsHf+Iw39l4NMDwSyt401lz/S8wUE\nIABVAfyaktGQezyBUJYdtlMQl3z6oNNruGuAqLy5Ph0we/Zs3LhxI+GxCxcuYOrUqZgyZQoAYOHC\nhThz5gzWrl2L1tZWAEAwGERXVxeuXr2K6dOnY8uWLW4XlUbgNPQfiUThiw1RW0q943A7WYzTGf/J\nQ+5uH+TjNJ3AcwOI5OTJmoCbN2+ipqYm/ntNTQ0uXryY8JzW1tZ4QJCuiRPvy0v5SpWb9R8IReJH\n/trpYSPesamKgsoK75aZFNPnHwpbW/EAVTXv4lO1oVO7DYYiCR1yOu2aSf2dyheNDiUxCvgzz0fg\npWL67L3A+std/1x48o2tuDT/+L///efK+5aCiRPvc7X+VgIgJ/bFgv95NLfsdv0zkTxqAphtJGCu\nvg9Hhu64faqKMVX+Ye1mX1uRTrtmUv9U5SvVBZ3F9Nl7gfVn/XPhSRAwadIk9PX1xX/v6+vD5MmT\nvSgKpWmkIepS7DjclGrBpKooCBtDHbCiKNB85vRDcju6ORyf6+FDRFQ+PJm8nTlzJq5du4be3l7o\nuo5Tp05h/vz5XhSF0pRrFr1CsO+td3vhYrbXstZQWOctAKW9o4KISpvrIwEbNmxAd3c3+vv78eyz\nz2LVqlVYuXIldu7cidWrV8MwDDQ0NOCxxx5zuyiUI7cXzeUiVMAzC0Y7H2GkURMRFqMeuew2ZhAk\nIosinFYqlSjZ54Vkrv/Y+6sc6+/GsHqq9RH2a6Vaae/WfHymn3857QSQ/f8+68/654KhP5ELUmVG\nLJZplVLM3EhE+Ve0xwYTZcJpO5tbHVw6w+kj3V0Xw7RKqd/9E1F+MAigsmClKi7EEHc6B+uMlJKY\nHTARFQuOA1LZKOQQ90jXsub9mZGPiIodRwKobBTyDnuka3EfPhGVCo4EEBERSYpBAFGeMSMfEZUK\nfisR5VmxbAMkIhoNgwAiF3AfPhGVAi4MJHIBtwESUSngLQoREZGkGAQQERFJikEAERGRpBgEEBER\nSYpBABERkaQYBBAREUmKQQAREZGkGAQQERFJikEAERGRpBgEEBERSYpBABERkaQYBBAREUmKQQAR\nEZGkGAQQERFJikEAERGRpBgEEBERSYpBABERkaQYBBAREUmKQQAREZGkGAQQERFJikEAERGRpBgE\nEBERSYpBABERkaSKNgjo6elBS0sLmpqa8Pnnn3tdHCIiorJTtEFAbW0tDhw4gC+++ALfffed18Uh\nIiIqO64HAZs3b8bcuXOxYMGChMfPnj2L+vp6LFq0CMFg0PG1R48eRX19PRobG90uJhERkXRcDwIa\nGxvR1taW8Jiu69i2bRuCwSA6Ojpw8uRJXL58GcFgEM3Nzejp6QEALFmyBN988w1Onz7tdjGJiIik\no7l9gdmzZ+PGjRsJj124cAFTp07FlClTAAALFy7EmTNnsHbtWrS2tgIAurq6cPDgQYTDYdTV1bld\nTCIiIum4HgQ4uXnzJmpqauK/19TU4OLFiwnPmTdvHubNm5fR+06ceF9eyleqWH/WX1Yy1x1g/WWv\nfy48WRioKIoXlyUiIiIbT4KASZMmoa+vL/57X18fJk+e7EVRiIiIpOVJEDBz5kxcu3YNvb290HUd\np06dwvz5870oChERkbQUIYRw8wIbNmxAd3c3+vv78eCDD2LVqlVYuXIlvv/+e3z44YcwDAMNDQ1Y\nvXq1m8UgIiKiJK4HAURERFScivbEQCIiInIXgwAiIiJJMQggIiKSlBRBwMDAABobG3H+/Hmvi1Jw\nMmdj7OzsxIoVK7B06VJ0dnZ6XZyCu379OrZu3YrNmzd7XZSCunPnDpYvX44lS5bg6NGjXhen4GT9\n3C0y/91n9X0vJPDJJ5+ItrY28dNPP3ldFM+EQiGxbNkyr4tRULquCyGE6O/vF6tXr/a4NN7ZtGmT\n10UoqCNHjojDhw+LSCQimpqavC6OZ2T73C38u8/s+75kRgKyzUb4448/Ytq0aRg/fnyhiuoKmbMx\nZlt3v98PIQT279+PlpaWQhU373L57MtFJm3wzz//YOrUqfD5fPD7/V4UN+9k/z+QSf3L5e/ekuln\nn/H3vavhSB79/PPP4tKlS+L555+PPxYKhcT8+fNFb2+v0HVdvPzyy+LSpUti//79YtmyZeLKlSvi\n448/Fu+//7544403xMaNGz2sQW6yrb/FMAyxfv16L4qes2zrHgqFxPbt20V3d7eHpc9drp99OdwR\nZtIGX3/9tfjyyy9FOBwum5GATOpvKYfP3ZJJ/cvl796SzWefyfe9JwmEspFtNsLa2loAQHt7Ox5+\n+OHCFjqPZM7GmG3dg8Egurq6cPXqVUyfPh1btmwpeNnzIdv69/f346OPPsKvv/6Kffv2YePGjQUv\ne75k0gYrVqzAmjVrcOTIETQ3N3tR3LzLpP5TpkzB3r17y+Jzt2RS/3PnzpXF370lk7rfunUr4+/7\nkgkCnKSTjdDy6quvFqpYBeNWNsZSkE7dW1tb4x1iuUmn/uPGjcP27dsLXbSCSdUGY8aMwaFDhzws\nWWGkqv+4ceOwY8cOD0tWGKnqv2PHDrz11lselsx9qeq+du3ajL/vS2ZNgBPZsxHKXH+Z6w6w/gDb\ngPWXt/75rHtJBwGyZyOUuf4y1x1g/QG2Aesvb/3zWfeSDgJkz0Yoc/1lrjvA+gNsA9Zf3vrns+4l\nk0BI9myEMtdf5roDrD/ANmD95a2/23UvmSCAiIiI8qukpwOIiIgoewwCo6cPcgAAA0BJREFUiIiI\nJMUggIiISFIMAoiIiCTFIICIiEhSDAKIiIgkxSCAiIhIUgwCiCQ0ODiIXbt24aWXXsLixYvx2muv\n4fz58ymf39DQMOp79vX15ZywadOmTWhvb8/pPYgofSWdRZCIsvPOO+/g/vvvx4kTJ+Dz+XD58mW8\n+eabOHDgAKZPnx5/nmEY8Pl8OHbs2KjvWVNTg2AwmFO5FEWROjEMUaExCCCSzF9//YWuri788MMP\n8Pl8AIAZM2bg9ddfx4EDB/DII4/gjz/+wK1bt3D79m0cP34ctbW16OnpAQAcO3YMn332GR544AE8\n/fTTOHHiBE6fPo0bN25g+fLl8Z+bm5uxYMECdHd34+7du9i9ezdmzZqFP//8E1u3bkUoFMK9e/fQ\n0NCANWvWxMvHQ0yJCofTAUSSuXLlCh599FFUVlYmPP7444/jypUr8ecEg0EcP3484Tl9fX3YvXs3\nDh8+jK+++gq6rqe8c//777/x4osvor29HevWrcO+ffsAAJMnT8ahQ4fQ3t6Ojo4OdHZ2xgMMIios\nBgFEBCDxDvyFF15AVVXVsOd0d3djzpw5mDBhAgBg8eLFKe/cJ0yYgKeeegoA8MQTT6C3txeAOcXw\n3nvvob6+Hk1NTejt7cXVq1fzXR0iSgOnA4gkU1tbi99//x0DAwMJHf1vv/2G2tpaAEB1dXXK16c7\nXB8IBOI/q6qKaDQKAGhra4PP50NHRwdUVcX69ethGEY2VSGiHHEkgEgy06ZNwzPPPINdu3YhEokA\nAC5duoTDhw+jpaVlxNfOmjULv/zyC27dugUA6OjoyPj6d+/exYQJE6CqKq5fv45z585lXgkiyguO\nBBBJaM+ePdizZw9eeeUV+P1+VFZWYu/evZgxYwZOnz497PnWvP+kSZPw9ttvo6mpCePHj8eTTz6Z\nsLbAvj4gea2A9fvy5cuxbt06fPvtt6ipqYlPGaR6HRG5RxFciktEGRgcHIx3/J9++imuX7+O3bt3\ne1wqIsoGRwKIKCMHDx7EqVOncOfOHTz00EP44IMPvC4SEWWJIwFERESS4sJAIiIiSTEIICIikhSD\nACIiIkkxCCAiIpIUgwAiIiJJMQggIiKS1P8B4i9U4fcs+RYAAAAASUVORK5CYII=\n",
       "text": [
        "<matplotlib.figure.Figure at 0x29bd8c10>"
       ]
      }
     ],
     "prompt_number": 21
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "This is useful, its the difference between CDS and transcript level reads.  Looks like on some coding genes there is a large dropout, I wonder what thats caused by.  "
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "#Pause Site Detection Quesetions"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "I'll want to filter by only protein coding genes, ignoring all annotated start and stop codons\n"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def get_length(peak_file):\n",
      "    if type(peak_file) is float:\n",
      "        return np.nan\n",
      "    \n",
      "    try:\n",
      "        with open(peak_file) as peak_handle:\n",
      "            bedtool_size = len(pybedtools.BedTool(peak_handle))\n",
      "        return bedtool_size\n",
      "    except ValueError:\n",
      "        return np.nan\n",
      "    except IOError:\n",
      "        return np.nan\n",
      "    \n",
      "def get_intervals(peak_file):\n",
      "    if type(peak_file) is float:\n",
      "        return np.nan\n",
      "    try:\n",
      "        with open(peak_file) as peak_handle:\n",
      "            bedtool = pybedtools.BedTool(peak_handle)\n",
      "            intervals = [interval for interval in bedtool]\n",
      "            return intervals\n",
      "    except ValueError as e:\n",
      "        print e\n",
      "        return np.nan\n",
      "    except IOError:\n",
      "        return np.nan\n",
      "\n",
      "def memory_efficent_intersect(in_tuple, force=True):\n",
      "    a_bed_file, b_bed_file = in_tuple\n",
      "    \n",
      "    if a_bed_file is numpy.nan or b_bed_file is numpy.nan:\n",
      "        return np.nan\n",
      "    \n",
      "    if os.path.getsize(a_bed_file) == 0 or os.path.getsize(b_bed_file) == 0:\n",
      "        return 0\n",
      "    \n",
      "    out_dir = \"/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/intersect/\"\n",
      "    out_name = os.path.join(out_dir, os.path.basename(a_bed_file) + os.path.basename(b_bed_file))\n",
      "    if os.path.exists(out_name) and not force:\n",
      "        return out_name\n",
      "    a_bed_tool = pybedtools.BedTool(a_bed_file)\n",
      "    b_bed_tool = pybedtools.BedTool(b_bed_file) \n",
      "    a_bed_tool.intersect(b_bed_tool, u=True, stream=True, sorted=True).saveas(out_name)\n",
      "    print out_name\n",
      "\n",
      "def memory_efficent_union(in_tuple, force=True):\n",
      "    a_bed_file, b_bed_file = in_tuple\n",
      "\n",
      "    if a_bed_file is numpy.nan or b_bed_file is numpy.nan:\n",
      "        return np.nan\n",
      "    \n",
      "    if os.path.getsize(a_bed_file) == 0 or os.path.getsize(b_bed_file) == 0:\n",
      "        return 0\n",
      "    else:\n",
      "        \n",
      "        out_dir = \"/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/\"\n",
      "        out_name =  os.path.join(out_dir, os.path.basename(a_bed_file) + os.path.basename(b_bed_file))\n",
      "        if os.path.exists(out_name) and not force:\n",
      "            return out_name\n",
      "        !cat $a_bed_file $b_bed_file | sort -k 1,1 -k 2,2n | cut -f 1-6 | mergeBed > $out_name\n",
      "    print out_name\n",
      "    \n",
      "def make_intersect_file(in_tuple):\n",
      "    a_bed_file, b_bed_file = in_tuple\n",
      "    \n",
      "    if a_bed_file is numpy.nan or b_bed_file is numpy.nan:\n",
      "        return np.nan\n",
      "    out_dir = \"/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/intersect/\"\n",
      "    return os.path.join(out_dir, os.path.basename(a_bed_file) + os.path.basename(b_bed_file))\n",
      "\n",
      "def make_union_file(in_tuple):\n",
      "    a_bed_file, b_bed_file = in_tuple\n",
      "    \n",
      "    if a_bed_file is numpy.nan or b_bed_file is numpy.nan:\n",
      "        return np.nan\n",
      "    out_dir = \"/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/\"\n",
      "    return os.path.join(out_dir, os.path.basename(a_bed_file) + os.path.basename(b_bed_file))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 8
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Filter out known starts and stops"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "pause_sites = [\"/nas3/gpratt/iPython_Notebook/SRR334701.peaks.bed\", \n",
      "               \"/nas3/gpratt/iPython_Notebook/SRR334703.peaks.bed\",\n",
      "               #\"/nas3/gpratt/Dropbox/m6a/B3P27_IP_Day1.peaks.bed\", \n",
      "               #\"/nas3/gpratt/Dropbox/m6a/B3P27_IP_Day23.peaks.bed\",\n",
      "               #\"/nas3/gpratt/Dropbox/m6a/B3P27_IP_Day27.peaks.bed\",\n",
      "               #\"/nas3/gpratt/Dropbox/m6a/B3P27_IP_Day6.peaks.bed\",]"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 9
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "#Get shuffled and true pause sites\n",
      "known_starts = pybedtools.BedTool(\"/nas3/gpratt/clipper/clipper/data/regions/hg19_start_codons.bed\")\n",
      "known_stops = pybedtools.BedTool(\"/nas3/gpratt/clipper/clipper/data/regions/hg19_stop_codons.bed\")\n",
      "for pause_site in pause_sites:\n",
      "    pybedtools.BedTool(pause_site).intersect(known_starts, v=True). \\\n",
      "              intersect(known_stops, v=True). \\\n",
      "              saveas(os.path.splitext(pause_site)[0] + \".true_pause.bed\")\n",
      "\n",
      "for x in range(10):\n",
      "    for pause_site in pause_sites:\n",
      "        try:\n",
      "            pybedtools.BedTool(pause_site). \\\n",
      "            shuffle(g=\"/nas3/yeolab/Genome/ucsc/hg19/hg19.chrom.sizes\", \n",
      "                    incl=\"/nas3/gpratt/clipper/clipper/data/regions/hg19_cds.bed\", stream=True). \\\n",
      "            saveas(os.path.splitext(pause_site)[0] + str(x) + \"_shuffled.bed\") \n",
      "        except:\n",
      "            pass"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "ename": "ValueError",
       "evalue": "File \"/nas3/gpratt/iPython_Notebook/SRR334701.peaks.bed\" does not exist",
       "output_type": "pyerr",
       "traceback": [
        "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n\u001b[1;31mValueError\u001b[0m                                Traceback (most recent call last)",
        "\u001b[1;32m<ipython-input-10-a6045be8975c>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[0;32m      3\u001b[0m \u001b[0mknown_stops\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mpybedtools\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mBedTool\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"/nas3/gpratt/clipper/clipper/data/regions/hg19_stop_codons.bed\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      4\u001b[0m \u001b[1;32mfor\u001b[0m \u001b[0mpause_site\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mpause_sites\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 5\u001b[1;33m     \u001b[0mpybedtools\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mBedTool\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpause_site\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mintersect\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mknown_starts\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mv\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mTrue\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m               \u001b[0mintersect\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mknown_stops\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mv\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mTrue\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m               \u001b[0msaveas\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mos\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msplitext\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpause_site\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m+\u001b[0m \u001b[1;34m\".true_pause.bed\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      6\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      7\u001b[0m \u001b[1;32mfor\u001b[0m \u001b[0mx\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mrange\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;36m10\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;32m/nas/nas0/gpratt/tools/pybedtools/pybedtools/bedtool.pyc\u001b[0m in \u001b[0;36m__init__\u001b[1;34m(self, fn, from_string, remote)\u001b[0m\n\u001b[0;32m    382\u001b[0m                 \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    383\u001b[0m                     \u001b[1;32mif\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[0mos\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mexists\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mfn\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 384\u001b[1;33m                         \u001b[1;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'File \"%s\" does not exist'\u001b[0m \u001b[1;33m%\u001b[0m \u001b[0mfn\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    385\u001b[0m                     \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_isbam\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0misBAM\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mfn\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    386\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;31mValueError\u001b[0m: File \"/nas3/gpratt/iPython_Notebook/SRR334701.peaks.bed\" does not exist"
       ]
      }
     ],
     "prompt_number": 10
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "sample_dict = defaultdict(dict)\n",
      "for pause_site in pause_sites:    \n",
      "    for peak in sample_df.ix['pyicoclip'].dropna():\n",
      "        sample_dict[os.path.basename(pause_site)][os.path.basename(peak).split(\".\")[0]] = (os.path.splitext(pause_site)[0] + \".true_pause.bed\", \n",
      "                                                                             peak)\n",
      "        \n",
      "overlapping_df = pd.DataFrame(sample_dict)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 14
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "sample_dicts = {}\n",
      "for x in range(10):\n",
      "    sample_dict = defaultdict(dict)\n",
      "    for pause_site in pause_sites:    \n",
      "        for peak in sample_df.ix['pyicoclip'].dropna():\n",
      "            sample_dict[os.path.basename(pause_site)][os.path.basename(peak).split(\".\")[0]] = (os.path.splitext(pause_site)[0] + str(x) + \"_shuffled.bed\", peak)\n",
      "    sample_dicts[x] = pd.DataFrame(sample_dict)\n",
      "    \n",
      "pause_shuffled_overlapping_df = pd.concat(sample_dicts)\n",
      "\n",
      "\n",
      "\n",
      "sample_dicts = {}\n",
      "for x in range(10):\n",
      "    sample_dict = defaultdict(dict)\n",
      "    for pause_site in pause_sites:    \n",
      "        for peak in sample_df.ix['pyicoclip'].dropna():            \n",
      "            sample_dict[os.path.basename(pause_site)][os.path.basename(peak).split(\".\")[0]] = (os.path.splitext(pause_site)[0] + \".true_pause.bed\", \n",
      "                                                                                os.path.splitext(peak)[0] + str(x) + \"_shuffled.bed\" )\n",
      "    sample_dicts[x] = pd.DataFrame(sample_dict)\n",
      "    \n",
      "peak_shuffled_overlapping_df = pd.concat(sample_dicts)\n",
      "\n",
      "\n",
      "sample_dicts = {}\n",
      "for x in range(10):\n",
      "    sample_dict = defaultdict(dict)\n",
      "    for pause_site in pause_sites:    \n",
      "        for peak in sample_df.ix['pyicoclip'].dropna():\n",
      "            sample_dict[os.path.basename(pause_site)][os.path.basename(peak).split(\".\")[0]] = (os.path.splitext(pause_site)[0] + str(x) + \"_shuffled.bed\", \n",
      "                                                                             os.path.splitext(peak)[0] + str(x) + \"_shuffled.bed\" )\n",
      "    sample_dicts[x] = pd.DataFrame(sample_dict)\n",
      "    \n",
      "both_shuffled_overlapping_df = pd.concat(sample_dicts)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 19
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "results = map(memory_efficent_union, overlapping_df.stack())\n",
      "results = map(memory_efficent_intersect, overlapping_df.stack())\n",
      "\n",
      "results = map(memory_efficent_union, pause_shuffled_overlapping_df.stack())\n",
      "results = map(memory_efficent_intersect, pause_shuffled_overlapping_df.stack())\n",
      "\n",
      "results = map(memory_efficent_union, peak_shuffled_overlapping_df.stack())\n",
      "results = map(memory_efficent_intersect, peak_shuffled_overlapping_df.stack())\n",
      "\n",
      "results = map(memory_efficent_union, both_shuffled_overlapping_df.stack())\n",
      "results = map(memory_efficent_intersect, both_shuffled_overlapping_df.stack())"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bed293T_hnRNPA1_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bed293T_hnRNPA1_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bed293T_hnRNPA1_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bed293T_hnRNPA1_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bed293T_hnRNPF_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bed293T_hnRNPF_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bed293T_hnRNPF_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bed293T_hnRNPF_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bed293T_hnRNPM_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bed293T_hnRNPM_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bed293T_hnRNPM_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bed293T_hnRNPM_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bed293T_hnRNPU_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bed293T_hnRNPU_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bed293T_hnRNPU_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bed293T_hnRNPU_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedFOX2.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedFOX2.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedFOX2.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedFOX2.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedHEK293FlpIn_FMR1-iso1_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedHEK293FlpIn_FMR1-iso1_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedHEK293FlpIn_FMR1-iso1_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedHEK293FlpIn_FMR1-iso1_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedHEK293FlpIn_FMR1-iso7_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedHEK293FlpIn_FMR1-iso7_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedHEK293FlpIn_FMR1-iso7_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedHEK293FlpIn_FMR1-iso7_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedHEK293_Ago2_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedHEK293_Ago2_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedHEK293_Ago2_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedHEK293_Ago2_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedHEK293_Ago2_WT_rep2.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedHEK293_Ago2_WT_rep2.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedHEK293_Ago2_WT_rep2.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedHEK293_Ago2_WT_rep2.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedHEK293_Ago2_WT_rep3.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedHEK293_Ago2_WT_rep3.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedHEK293_Ago2_WT_rep3.bam.pyicoclip.bed\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedHEK293_Ago2_WT_rep3.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedHEK293_Ago2_WT_rep4.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedHEK293_Ago2_WT_rep4.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedHEK293_Ago2_WT_rep4.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedHEK293_Ago2_WT_rep4.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedHEK293_Ago3_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: "
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedHEK293_Ago3_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedHEK293_Ago3_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedHEK293_Ago3_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedHEK293_Ago4_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedHEK293_Ago4_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedHEK293_Ago4_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedHEK293_Ago4_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedHEK293_DGCR8_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedHEK293_DGCR8_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedHEK293_DGCR8_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedHEK293_DGCR8_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedHEK293_DGCR8_overexpressed_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedHEK293_DGCR8_overexpressed_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedHEK293_DGCR8_overexpressed_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedHEK293_DGCR8_overexpressed_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedHEK293_HuR_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedHEK293_HuR_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedHEK293_HuR_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedHEK293_HuR_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedHEK293_HuR_WT_rep2.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedHEK293_HuR_WT_rep2.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedHEK293_HuR_WT_rep2.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedHEK293_HuR_WT_rep2.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedHEK293_HuR_WT_rep3.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedHEK293_HuR_WT_rep3.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedHEK293_HuR_WT_rep3.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedHEK293_HuR_WT_rep3.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedHEK293_HuR_WT_rep6.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedHEK293_HuR_WT_rep6.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedHEK293_HuR_WT_rep6.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedHEK293_HuR_WT_rep6.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedHEK293_IGF2BP1_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedHEK293_IGF2BP1_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedHEK293_IGF2BP1_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedHEK293_IGF2BP1_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedHEK293_IGF2BP2_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedHEK293_IGF2BP2_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedHEK293_IGF2BP2_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedHEK293_IGF2BP2_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedHEK293_IGF2BP3_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedHEK293_IGF2BP3_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedHEK293_IGF2BP3_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedHEK293_IGF2BP3_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedHEK293_LIN28B_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedHEK293_LIN28B_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedHEK293_LIN28B_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedHEK293_LIN28B_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedHEK293_PAPD5_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedHEK293_PAPD5_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedHEK293_PAPD5_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedHEK293_PAPD5_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedHEK293_Pumilio2_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedHEK293_Pumilio2_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedHEK293_Pumilio2_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedHEK293_Pumilio2_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedHEK293_QIK_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedHEK293_QIK_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedHEK293_QIK_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedHEK293_QIK_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedHEK293_TNRC6A_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedHEK293_TNRC6A_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedHEK293_TNRC6A_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        ": Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedHEK293_TNRC6A_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedHEK293_TNRC6B_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedHEK293_TNRC6B_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedHEK293_TNRC6B_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedHEK293_TNRC6B_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedHEK293_TNRC6C_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedHEK293_TNRC6C_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedHEK293_TNRC6C_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedHEK293_TNRC6C_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedHEK293_hnRNPU_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedHEK293_hnRNPU_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedHEK293_hnRNPU_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedHEK293_hnRNPU_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedHeLa_Ago2_PTB_kd_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedHeLa_Ago2_PTB_kd_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedHeLa_Ago2_PTB_kd_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedHeLa_Ago2_PTB_kd_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedHeLa_Ago2_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedHeLa_Ago2_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedHeLa_Ago2_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedHeLa_Ago2_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedHeLa_eIF4AIII_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedHeLa_eIF4AIII_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedHeLa_eIF4AIII_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedHeLa_eIF4AIII_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedMDAMB231_hnRNPA2B1_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedMDAMB231_hnRNPA2B1_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedMDAMB231_hnRNPA2B1_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedMDAMB231_hnRNPA2B1_WT_rep1.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedSRR034467.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedSRR034467.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedSRR034467.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedSRR034467.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedSRR034468.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedSRR034468.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedSRR034468.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedSRR034468.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedSRR034469.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedSRR034469.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedSRR034469.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedSRR034469.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedSRR048950.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedSRR048950.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedSRR048950.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedSRR048950.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedSRR048951.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedSRR048951.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedSRR048951.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedSRR048951.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedSRR048952.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedSRR048952.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedSRR048952.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedSRR048952.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedSRR048953.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedSRR048953.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day27.peaks.true_pause.bedSRR048953.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day6.peaks.true_pause.bedSRR048953.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n",
        "cut: write error: Broken pipe\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day1.peaks.true_pause.bedSRR048954.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n",
        "\r\n",
        "Tool:    bedtools merge (aka mergeBed)\r\n",
        "Version: v2.20.1-24-gdfabf63\r\n",
        "Summary: Merges overlapping BED/GFF/VCF entries into a single interval.\r\n",
        "\r\n",
        "Usage:   bedtools merge [OPTIONS] -i <bed/gff/vcf>\r\n",
        "\r\n",
        "Options: \r\n",
        "\t-s\tForce strandedness.  That is, only merge features\r\n",
        "\t\tthat are on the same strand.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-S\tForce merge for one specific strand only.\r\n",
        "\t\tFollow with + or - to force merge from only\r\n",
        "\t\tthe forward or reverse strand, respectively.\r\n",
        "\t\t- By default, merging is done without respect to strand.\r\n",
        "\r\n",
        "\t-d\tMaximum distance between features allowed for features\r\n",
        "\t\tto be merged.\r\n",
        "\t\t- Def. 0. That is, overlapping & book-ended features are merged.\r\n",
        "\t\t- (INTEGER)\r\n",
        "\t\t- Note: negative values enforce the number of b.p. required for overlap.\r\n",
        "\r\n",
        "\t-header\tPrint the header from the A file prior to results.\r\n",
        "\r\n",
        "\t-c\tSpecify columns from the input file to operate upon (see -o option, below).\r\n",
        "\t\tMultiple columns can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t-o\tSpecify the operation that should be applied to -c.\r\n",
        "\t\tValid operations:\r\n",
        "\t\t    sum, min, max, absmin, absmax,\r\n",
        "\t\t    mean, median,\r\n",
        "\t\t    collapse (i.e., print a delimited list (duplicates allowed)), \r\n",
        "\t\t    distinct (i.e., print a delimited list (NO duplicates allowed)), \r\n",
        "\t\t    count\r\n",
        "\t\t    count_distinct (i.e., a count of the unique values in the column), \r\n",
        "\t\tDefault: sum\r\n",
        "\t\tMultiple operations can be specified in a comma-delimited list.\r\n",
        "\r\n",
        "\t\tIf there is only column, but multiple operations, all operations will be\r\n",
        "\t\tapplied on that column. Likewise, if there is only one operation, but\r\n",
        "\t\tmultiple columns, that operation will be applied to all columns.\r\n",
        "\t\tOtherwise, the number of columns must match the the number of operations,\r\n",
        "\t\tand will be applied in respective order.\r\n",
        "\t\tE.g., \"-c 5,4,6 -o sum,mean,count\" will give the sum of column 5,\r\n",
        "\t\tthe mean of column 4, and the count of column 6.\r\n",
        "\t\tThe order of output columns will match the ordering given in the command.\r\n",
        "\r\n",
        "\r\n",
        "\t-delim\tSpecify a custom delimiter for the collapse operations.\r\n",
        "\t\t- Example: -delim \"|\"\r\n",
        "\t\t- Default: \",\".\r\n",
        "\r\n",
        "Notes: \r\n",
        "\t(1) The input file (-i) file must be sorted by chrom, then start.\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/union/B3P27_IP_Day23.peaks.true_pause.bedSRR048954.polyATrim.adapterTrim.rmRep.rg.sorted.rmDup.sorted.bam.pyicoclip.bed\n"
       ]
      },
      {
       "ename": "UnboundLocalError",
       "evalue": "local variable 'child' referenced before assignment",
       "output_type": "pyerr",
       "traceback": [
        "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n\u001b[1;31mUnboundLocalError\u001b[0m                         Traceback (most recent call last)",
        "\u001b[1;32m<ipython-input-39-3e6e7fb62df6>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mresults\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mmap\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mmemory_efficent_union\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0moverlapping_df\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mstack\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      2\u001b[0m \u001b[0mresults\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mmap\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mmemory_efficent_intersect\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0moverlapping_df\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mstack\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      3\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      4\u001b[0m \u001b[0mresults\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mmap\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mmemory_efficent_union\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mpause_shuffled_overlapping_df\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mstack\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      5\u001b[0m \u001b[0mresults\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mmap\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mmemory_efficent_intersect\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mpause_shuffled_overlapping_df\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mstack\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;32m<ipython-input-38-6c393812fbc4>\u001b[0m in \u001b[0;36mmemory_efficent_union\u001b[1;34m(in_tuple, force)\u001b[0m\n\u001b[0;32m     58\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mos\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mexists\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mout_name\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mand\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[0mforce\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     59\u001b[0m             \u001b[1;32mreturn\u001b[0m \u001b[0mout_name\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 60\u001b[1;33m         \u001b[0mget_ipython\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msystem\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34mu'cat $a_bed_file $b_bed_file | sort -k 1,1 -k 2,2n | cut -f 1-6 | mergeBed > $out_name'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     61\u001b[0m     \u001b[1;32mprint\u001b[0m \u001b[0mout_name\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     62\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;32m/nas/nas0/yeolab/Software/lib/python2.7/site-packages/IPython/core/interactiveshell.pyc\u001b[0m in \u001b[0;36msystem_piped\u001b[1;34m(self, cmd)\u001b[0m\n\u001b[0;32m   2254\u001b[0m         \u001b[1;31m# a non-None value would trigger :func:`sys.displayhook` calls.\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   2255\u001b[0m         \u001b[1;31m# Instead, we store the exit_code in user_ns.\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 2256\u001b[1;33m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0muser_ns\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'_exit_code'\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0msystem\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mvar_expand\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mcmd\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdepth\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m1\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   2257\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   2258\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0msystem_raw\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mcmd\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;32m/nas/nas0/yeolab/Software/lib/python2.7/site-packages/IPython/utils/_process_posix.pyc\u001b[0m in \u001b[0;36msystem\u001b[1;34m(self, cmd)\u001b[0m\n\u001b[0;32m    169\u001b[0m             \u001b[1;31m# (the character is known as ETX for 'End of Text', see\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    170\u001b[0m             \u001b[1;31m# curses.ascii.ETX).\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 171\u001b[1;33m             \u001b[0mchild\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msendline\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mchr\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;36m3\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    172\u001b[0m             \u001b[1;31m# Read and print any more output the program might produce on its\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    173\u001b[0m             \u001b[1;31m# way out.\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;31mUnboundLocalError\u001b[0m: local variable 'child' referenced before assignment"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stderr",
       "text": [
        "ERROR:tornado.general:Uncaught exception, closing connection.\n",
        "Traceback (most recent call last):\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/zmq/eventloop/zmqstream.py\", line 407, in _run_callback\n",
        "    callback(*args, **kwargs)\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/tornado/stack_context.py\", line 331, in wrapped\n",
        "    raise_exc_info(exc)\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/tornado/stack_context.py\", line 302, in wrapped\n",
        "    ret = fn(*args, **kwargs)\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/IPython/kernel/zmq/ipkernel.py\", line 281, in dispatcher\n",
        "    return self.dispatch_shell(stream, msg)\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/IPython/kernel/zmq/ipkernel.py\", line 245, in dispatch_shell\n",
        "    handler(stream, idents, msg)\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/IPython/kernel/zmq/ipkernel.py\", line 474, in execute_request\n",
        "    self._abort_queues()\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/IPython/kernel/zmq/ipkernel.py\", line 706, in _abort_queues\n",
        "    self._abort_queue(stream)\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/IPython/kernel/zmq/ipkernel.py\", line 725, in _abort_queue\n",
        "    content=status, parent=msg, ident=idents)\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/IPython/kernel/zmq/session.py\", line 635, in send\n",
        "    to_send = self.serialize(msg, ident)\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/IPython/kernel/zmq/session.py\", line 554, in serialize\n",
        "    self.pack(msg['metadata']),\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/IPython/kernel/zmq/session.py\", line 83, in <lambda>\n",
        "    json_packer = lambda obj: jsonapi.dumps(obj, default=date_default)\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/zmq/utils/jsonapi.py\", line 55, in dumps\n",
        "    s = jsonmod.dumps(o, **kwargs)\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/simplejson/__init__.py\", line 369, in dumps\n",
        "    **kw).encode(obj)\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/simplejson/encoder.py\", line 119, in __init__\n",
        "    def __init__(self, skipkeys=False, ensure_ascii=True,\n",
        "KeyboardInterrupt\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stderr",
       "text": [
        "ERROR:tornado.general:Uncaught exception, closing connection.\n",
        "Traceback (most recent call last):\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/zmq/eventloop/zmqstream.py\", line 433, in _handle_events\n",
        "    self._handle_recv()\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/zmq/eventloop/zmqstream.py\", line 465, in _handle_recv\n",
        "    self._run_callback(callback, msg)\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/zmq/eventloop/zmqstream.py\", line 407, in _run_callback\n",
        "    callback(*args, **kwargs)\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/tornado/stack_context.py\", line 331, in wrapped\n",
        "    raise_exc_info(exc)\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/tornado/stack_context.py\", line 302, in wrapped\n",
        "    ret = fn(*args, **kwargs)\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/IPython/kernel/zmq/ipkernel.py\", line 281, in dispatcher\n",
        "    return self.dispatch_shell(stream, msg)\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/IPython/kernel/zmq/ipkernel.py\", line 245, in dispatch_shell\n",
        "    handler(stream, idents, msg)\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/IPython/kernel/zmq/ipkernel.py\", line 474, in execute_request\n",
        "    self._abort_queues()\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/IPython/kernel/zmq/ipkernel.py\", line 706, in _abort_queues\n",
        "    self._abort_queue(stream)\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/IPython/kernel/zmq/ipkernel.py\", line 725, in _abort_queue\n",
        "    content=status, parent=msg, ident=idents)\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/IPython/kernel/zmq/session.py\", line 635, in send\n",
        "    to_send = self.serialize(msg, ident)\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/IPython/kernel/zmq/session.py\", line 554, in serialize\n",
        "    self.pack(msg['metadata']),\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/IPython/kernel/zmq/session.py\", line 83, in <lambda>\n",
        "    json_packer = lambda obj: jsonapi.dumps(obj, default=date_default)\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/zmq/utils/jsonapi.py\", line 55, in dumps\n",
        "    s = jsonmod.dumps(o, **kwargs)\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/simplejson/__init__.py\", line 369, in dumps\n",
        "    **kw).encode(obj)\n",
        "  File \"/nas/nas0/yeolab/Software/lib/python2.7/site-packages/simplejson/encoder.py\", line 119, in __init__\n",
        "    def __init__(self, skipkeys=False, ensure_ascii=True,\n",
        "KeyboardInterrupt\n"
       ]
      }
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "intersect_df = overlapping_df.applymap(make_intersect_file)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 15
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "intersect_df.ix['HEK293FlpIn_FMR1-iso1_WT_rep1']['SRR334701.peaks.bed']"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 16,
       "text": [
        "'/nas3/gpratt/projects/public_riboseq/analysis/peak_cashe/intersect/SRR334701.peaks.true_pause.bedHEK293FlpIn_FMR1-iso1_WT_rep1.bam.pyicoclip.bed'"
       ]
      }
     ],
     "prompt_number": 16
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "intersect_df = overlapping_df.applymap(make_intersect_file).applymap(get_length)\n",
      "union_df = overlapping_df.applymap(make_union_file).applymap(get_length)\n",
      "\n",
      "pause_shuffled_intersect_df = pause_shuffled_overlapping_df.applymap(make_intersect_file).applymap(get_length)\n",
      "pause_shuffled_union_df = pause_shuffled_overlapping_df.applymap(make_union_file).applymap(get_length)\n",
      "\n",
      "peak_shuffled_intersect_df = peak_shuffled_overlapping_df.applymap(make_intersect_file).applymap(get_length)\n",
      "peak_shuffled_union_df = peak_shuffled_overlapping_df.applymap(make_union_file).applymap(get_length)\n",
      "\n",
      "both_shuffled_intersect_df = both_shuffled_overlapping_df.applymap(make_intersect_file).applymap(get_length)\n",
      "both_shuffled_union_df = both_shuffled_overlapping_df.applymap(make_union_file).applymap(get_length)\n",
      "\n",
      "pause_series = pd.Series({ os.path.basename(pause_site) : pause_site for pause_site in pause_sites })\n",
      "clip_size = sample_df.ix['pyicoclip'].apply(get_length)\n",
      "pause_size = pause_series.apply(get_length)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "HTML(intersect_df.to_html())"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>B3P27_IP_Day1.peaks.bed</th>\n",
        "      <th>B3P27_IP_Day23.peaks.bed</th>\n",
        "      <th>B3P27_IP_Day27.peaks.bed</th>\n",
        "      <th>B3P27_IP_Day6.peaks.bed</th>\n",
        "      <th>SRR334701.peaks.bed</th>\n",
        "      <th>SRR334703.peaks.bed</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>293T_hnRNPA1_WT_rep1</th>\n",
        "      <td>  15</td>\n",
        "      <td>   5</td>\n",
        "      <td>   8</td>\n",
        "      <td>   15</td>\n",
        "      <td>  1404</td>\n",
        "      <td>  271</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>293T_hnRNPF_WT_rep1</th>\n",
        "      <td>  58</td>\n",
        "      <td>  19</td>\n",
        "      <td>  77</td>\n",
        "      <td>  149</td>\n",
        "      <td>  2841</td>\n",
        "      <td>  449</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>293T_hnRNPM_WT_rep1</th>\n",
        "      <td>   8</td>\n",
        "      <td>   3</td>\n",
        "      <td>   3</td>\n",
        "      <td>    7</td>\n",
        "      <td>  1015</td>\n",
        "      <td>  219</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>293T_hnRNPU_WT_rep1</th>\n",
        "      <td>  73</td>\n",
        "      <td>  36</td>\n",
        "      <td> 113</td>\n",
        "      <td>  229</td>\n",
        "      <td>  4355</td>\n",
        "      <td>  651</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>FOX2</th>\n",
        "      <td> 137</td>\n",
        "      <td>  67</td>\n",
        "      <td> 211</td>\n",
        "      <td>  428</td>\n",
        "      <td>  8809</td>\n",
        "      <td> 1138</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293FlpIn_FMR1-iso1_WT_rep1</th>\n",
        "      <td> 301</td>\n",
        "      <td> 149</td>\n",
        "      <td> 578</td>\n",
        "      <td> 1085</td>\n",
        "      <td> 13944</td>\n",
        "      <td> 1259</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293FlpIn_FMR1-iso7_WT_rep1</th>\n",
        "      <td> 224</td>\n",
        "      <td> 120</td>\n",
        "      <td> 417</td>\n",
        "      <td>  779</td>\n",
        "      <td>  8371</td>\n",
        "      <td>  921</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago2_WT_rep1</th>\n",
        "      <td>   4</td>\n",
        "      <td>   2</td>\n",
        "      <td>   2</td>\n",
        "      <td>    1</td>\n",
        "      <td>   233</td>\n",
        "      <td>  137</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago2_WT_rep2</th>\n",
        "      <td>  88</td>\n",
        "      <td>  52</td>\n",
        "      <td> 175</td>\n",
        "      <td>  273</td>\n",
        "      <td>  4538</td>\n",
        "      <td>  859</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago2_WT_rep3</th>\n",
        "      <td>  58</td>\n",
        "      <td>  28</td>\n",
        "      <td>  97</td>\n",
        "      <td>  161</td>\n",
        "      <td>  2675</td>\n",
        "      <td>  565</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago2_WT_rep4</th>\n",
        "      <td>  55</td>\n",
        "      <td>  22</td>\n",
        "      <td>  86</td>\n",
        "      <td>  126</td>\n",
        "      <td>  2891</td>\n",
        "      <td>  629</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago3_WT_rep1</th>\n",
        "      <td>   7</td>\n",
        "      <td>   3</td>\n",
        "      <td>   4</td>\n",
        "      <td>    7</td>\n",
        "      <td>   284</td>\n",
        "      <td>  166</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago4_WT_rep1</th>\n",
        "      <td>   5</td>\n",
        "      <td>   4</td>\n",
        "      <td>   5</td>\n",
        "      <td>    2</td>\n",
        "      <td>   776</td>\n",
        "      <td>  284</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_DGCR8_WT_rep1</th>\n",
        "      <td>  11</td>\n",
        "      <td>   9</td>\n",
        "      <td>  13</td>\n",
        "      <td>   23</td>\n",
        "      <td>  2013</td>\n",
        "      <td>  430</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_DGCR8_overexpressed_rep1</th>\n",
        "      <td>   8</td>\n",
        "      <td>   6</td>\n",
        "      <td>   5</td>\n",
        "      <td>   10</td>\n",
        "      <td>  1274</td>\n",
        "      <td>  289</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_HuR_WT_rep1</th>\n",
        "      <td>  18</td>\n",
        "      <td>  13</td>\n",
        "      <td>  26</td>\n",
        "      <td>   31</td>\n",
        "      <td>   290</td>\n",
        "      <td>  111</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_HuR_WT_rep2</th>\n",
        "      <td>  21</td>\n",
        "      <td>  14</td>\n",
        "      <td>  27</td>\n",
        "      <td>   39</td>\n",
        "      <td>   954</td>\n",
        "      <td>  271</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_HuR_WT_rep3</th>\n",
        "      <td> 289</td>\n",
        "      <td> 145</td>\n",
        "      <td> 534</td>\n",
        "      <td>  980</td>\n",
        "      <td> 11666</td>\n",
        "      <td> 1173</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_HuR_WT_rep6</th>\n",
        "      <td>  17</td>\n",
        "      <td>  12</td>\n",
        "      <td>  20</td>\n",
        "      <td>   33</td>\n",
        "      <td>   268</td>\n",
        "      <td>  136</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_IGF2BP1_WT_rep1</th>\n",
        "      <td> 244</td>\n",
        "      <td> 121</td>\n",
        "      <td> 457</td>\n",
        "      <td>  778</td>\n",
        "      <td> 11249</td>\n",
        "      <td> 1000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_IGF2BP2_WT_rep1</th>\n",
        "      <td> 182</td>\n",
        "      <td>  96</td>\n",
        "      <td> 365</td>\n",
        "      <td>  584</td>\n",
        "      <td>  7040</td>\n",
        "      <td>  810</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_IGF2BP3_WT_rep1</th>\n",
        "      <td> 241</td>\n",
        "      <td> 123</td>\n",
        "      <td> 480</td>\n",
        "      <td>  769</td>\n",
        "      <td> 11459</td>\n",
        "      <td> 1058</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_LIN28B_WT_rep1</th>\n",
        "      <td>   5</td>\n",
        "      <td>   3</td>\n",
        "      <td>   5</td>\n",
        "      <td>   10</td>\n",
        "      <td>   840</td>\n",
        "      <td>  302</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_PAPD5_WT_rep1</th>\n",
        "      <td>   7</td>\n",
        "      <td>   2</td>\n",
        "      <td>   3</td>\n",
        "      <td>    5</td>\n",
        "      <td>  1458</td>\n",
        "      <td>  320</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Pumilio2_WT_rep1</th>\n",
        "      <td>   8</td>\n",
        "      <td>   3</td>\n",
        "      <td>   8</td>\n",
        "      <td>   19</td>\n",
        "      <td>   389</td>\n",
        "      <td>  198</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_QIK_WT_rep1</th>\n",
        "      <td>  11</td>\n",
        "      <td>   5</td>\n",
        "      <td>  12</td>\n",
        "      <td>   16</td>\n",
        "      <td>   632</td>\n",
        "      <td>  258</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_TNRC6A_WT_rep1</th>\n",
        "      <td>   5</td>\n",
        "      <td>   2</td>\n",
        "      <td>   0</td>\n",
        "      <td>    1</td>\n",
        "      <td>   284</td>\n",
        "      <td>  163</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_TNRC6B_WT_rep1</th>\n",
        "      <td>   4</td>\n",
        "      <td>   1</td>\n",
        "      <td>   0</td>\n",
        "      <td>    1</td>\n",
        "      <td>   212</td>\n",
        "      <td>  173</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_TNRC6C_WT_rep1</th>\n",
        "      <td>   5</td>\n",
        "      <td>   2</td>\n",
        "      <td>   0</td>\n",
        "      <td>    0</td>\n",
        "      <td>   675</td>\n",
        "      <td>  294</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_hnRNPU_WT_rep1</th>\n",
        "      <td>  45</td>\n",
        "      <td>  17</td>\n",
        "      <td>  72</td>\n",
        "      <td>  145</td>\n",
        "      <td>  3076</td>\n",
        "      <td>  249</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HeLa_Ago2_PTB_kd_rep1</th>\n",
        "      <td> 183</td>\n",
        "      <td> 113</td>\n",
        "      <td> 365</td>\n",
        "      <td>  587</td>\n",
        "      <td> 12549</td>\n",
        "      <td> 1226</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HeLa_Ago2_WT_rep1</th>\n",
        "      <td> 193</td>\n",
        "      <td>  99</td>\n",
        "      <td> 340</td>\n",
        "      <td>  580</td>\n",
        "      <td> 12561</td>\n",
        "      <td> 1148</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HeLa_eIF4AIII_WT_rep1</th>\n",
        "      <td> 267</td>\n",
        "      <td> 115</td>\n",
        "      <td> 455</td>\n",
        "      <td> 1006</td>\n",
        "      <td> 21070</td>\n",
        "      <td> 1569</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MDAMB231_hnRNPA2B1_WT_rep1</th>\n",
        "      <td>   0</td>\n",
        "      <td>   0</td>\n",
        "      <td>   0</td>\n",
        "      <td>    0</td>\n",
        "      <td>     0</td>\n",
        "      <td>    0</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR034467</th>\n",
        "      <td>   5</td>\n",
        "      <td>   3</td>\n",
        "      <td>   4</td>\n",
        "      <td>    3</td>\n",
        "      <td>   305</td>\n",
        "      <td>   88</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR034468</th>\n",
        "      <td>   3</td>\n",
        "      <td>   2</td>\n",
        "      <td>   1</td>\n",
        "      <td>    0</td>\n",
        "      <td>   169</td>\n",
        "      <td>   68</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR034469</th>\n",
        "      <td>   5</td>\n",
        "      <td>   3</td>\n",
        "      <td>   2</td>\n",
        "      <td>    0</td>\n",
        "      <td>   250</td>\n",
        "      <td>   70</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048950</th>\n",
        "      <td>  25</td>\n",
        "      <td>  12</td>\n",
        "      <td>  29</td>\n",
        "      <td>   51</td>\n",
        "      <td>   574</td>\n",
        "      <td>  114</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048951</th>\n",
        "      <td> 109</td>\n",
        "      <td>  53</td>\n",
        "      <td> 195</td>\n",
        "      <td>  313</td>\n",
        "      <td>  3774</td>\n",
        "      <td>  339</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048952</th>\n",
        "      <td>  67</td>\n",
        "      <td>  33</td>\n",
        "      <td> 101</td>\n",
        "      <td>  157</td>\n",
        "      <td>  2786</td>\n",
        "      <td>  373</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048953</th>\n",
        "      <td>  18</td>\n",
        "      <td>   8</td>\n",
        "      <td>  17</td>\n",
        "      <td>   25</td>\n",
        "      <td>   756</td>\n",
        "      <td>  145</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048954</th>\n",
        "      <td>   5</td>\n",
        "      <td>   2</td>\n",
        "      <td>   0</td>\n",
        "      <td>    1</td>\n",
        "      <td>   135</td>\n",
        "      <td>   93</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048955</th>\n",
        "      <td>   4</td>\n",
        "      <td>   0</td>\n",
        "      <td>   5</td>\n",
        "      <td>    2</td>\n",
        "      <td>    96</td>\n",
        "      <td>   19</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048956</th>\n",
        "      <td>  61</td>\n",
        "      <td>  34</td>\n",
        "      <td> 121</td>\n",
        "      <td>  192</td>\n",
        "      <td>  1933</td>\n",
        "      <td>  125</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048957</th>\n",
        "      <td>  40</td>\n",
        "      <td>  22</td>\n",
        "      <td>  67</td>\n",
        "      <td>   97</td>\n",
        "      <td>   976</td>\n",
        "      <td>  211</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048958</th>\n",
        "      <td> 116</td>\n",
        "      <td>  58</td>\n",
        "      <td> 209</td>\n",
        "      <td>  317</td>\n",
        "      <td>  3190</td>\n",
        "      <td>  338</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048959</th>\n",
        "      <td>  38</td>\n",
        "      <td>  23</td>\n",
        "      <td>  57</td>\n",
        "      <td>   90</td>\n",
        "      <td>   976</td>\n",
        "      <td>  178</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048960</th>\n",
        "      <td>  11</td>\n",
        "      <td>   8</td>\n",
        "      <td>   8</td>\n",
        "      <td>   14</td>\n",
        "      <td>   610</td>\n",
        "      <td>  160</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048961</th>\n",
        "      <td>   4</td>\n",
        "      <td>   1</td>\n",
        "      <td>   0</td>\n",
        "      <td>    0</td>\n",
        "      <td>   238</td>\n",
        "      <td>  105</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048962</th>\n",
        "      <td>  28</td>\n",
        "      <td>   9</td>\n",
        "      <td>  33</td>\n",
        "      <td>   64</td>\n",
        "      <td>   502</td>\n",
        "      <td>  122</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048963</th>\n",
        "      <td> 118</td>\n",
        "      <td>  56</td>\n",
        "      <td> 206</td>\n",
        "      <td>  341</td>\n",
        "      <td>  3511</td>\n",
        "      <td>  373</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048964</th>\n",
        "      <td>  92</td>\n",
        "      <td>  47</td>\n",
        "      <td> 148</td>\n",
        "      <td>  259</td>\n",
        "      <td>  3641</td>\n",
        "      <td>  427</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048965</th>\n",
        "      <td>  23</td>\n",
        "      <td>  17</td>\n",
        "      <td>  42</td>\n",
        "      <td>   58</td>\n",
        "      <td>  2034</td>\n",
        "      <td>  242</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048966</th>\n",
        "      <td>   7</td>\n",
        "      <td>   3</td>\n",
        "      <td>   7</td>\n",
        "      <td>   13</td>\n",
        "      <td>   561</td>\n",
        "      <td>  105</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048967</th>\n",
        "      <td>   6</td>\n",
        "      <td>   1</td>\n",
        "      <td>   5</td>\n",
        "      <td>   12</td>\n",
        "      <td>   178</td>\n",
        "      <td>  115</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048968</th>\n",
        "      <td>   6</td>\n",
        "      <td>   2</td>\n",
        "      <td>   4</td>\n",
        "      <td>    9</td>\n",
        "      <td>   144</td>\n",
        "      <td>  124</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048969</th>\n",
        "      <td>   6</td>\n",
        "      <td>   2</td>\n",
        "      <td>   2</td>\n",
        "      <td>    4</td>\n",
        "      <td>   191</td>\n",
        "      <td>  109</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048970</th>\n",
        "      <td>   5</td>\n",
        "      <td>   2</td>\n",
        "      <td>   2</td>\n",
        "      <td>    4</td>\n",
        "      <td>   208</td>\n",
        "      <td>   92</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048971</th>\n",
        "      <td>   7</td>\n",
        "      <td>   2</td>\n",
        "      <td>   2</td>\n",
        "      <td>    4</td>\n",
        "      <td>   301</td>\n",
        "      <td>  132</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048972</th>\n",
        "      <td>   6</td>\n",
        "      <td>   2</td>\n",
        "      <td>   1</td>\n",
        "      <td>    3</td>\n",
        "      <td>    58</td>\n",
        "      <td>   63</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048973</th>\n",
        "      <td>  11</td>\n",
        "      <td>   4</td>\n",
        "      <td>  10</td>\n",
        "      <td>    8</td>\n",
        "      <td>   317</td>\n",
        "      <td>  139</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048974</th>\n",
        "      <td>   3</td>\n",
        "      <td>   1</td>\n",
        "      <td>   0</td>\n",
        "      <td>    1</td>\n",
        "      <td>    40</td>\n",
        "      <td>   53</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048975</th>\n",
        "      <td>   4</td>\n",
        "      <td>   1</td>\n",
        "      <td>   0</td>\n",
        "      <td>    0</td>\n",
        "      <td>   190</td>\n",
        "      <td>  102</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048976</th>\n",
        "      <td>   7</td>\n",
        "      <td>   2</td>\n",
        "      <td>   5</td>\n",
        "      <td>    6</td>\n",
        "      <td>   320</td>\n",
        "      <td>  160</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048977</th>\n",
        "      <td>   1</td>\n",
        "      <td>   0</td>\n",
        "      <td>   0</td>\n",
        "      <td>    0</td>\n",
        "      <td>    17</td>\n",
        "      <td>   29</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048978</th>\n",
        "      <td>   5</td>\n",
        "      <td>   2</td>\n",
        "      <td>   1</td>\n",
        "      <td>    0</td>\n",
        "      <td>   146</td>\n",
        "      <td>  110</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048979</th>\n",
        "      <td>   5</td>\n",
        "      <td>   2</td>\n",
        "      <td>   1</td>\n",
        "      <td>    1</td>\n",
        "      <td>   504</td>\n",
        "      <td>  202</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048980</th>\n",
        "      <td>  32</td>\n",
        "      <td>  16</td>\n",
        "      <td>  58</td>\n",
        "      <td>   85</td>\n",
        "      <td>   730</td>\n",
        "      <td>  152</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048981</th>\n",
        "      <td>   9</td>\n",
        "      <td>   8</td>\n",
        "      <td>  20</td>\n",
        "      <td>   26</td>\n",
        "      <td>   333</td>\n",
        "      <td>  121</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048982</th>\n",
        "      <td>   5</td>\n",
        "      <td>   2</td>\n",
        "      <td>   0</td>\n",
        "      <td>    0</td>\n",
        "      <td>    68</td>\n",
        "      <td>   69</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048983</th>\n",
        "      <td>   5</td>\n",
        "      <td>   2</td>\n",
        "      <td>   0</td>\n",
        "      <td>    1</td>\n",
        "      <td>   117</td>\n",
        "      <td>   96</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048984</th>\n",
        "      <td>   4</td>\n",
        "      <td>   1</td>\n",
        "      <td>   0</td>\n",
        "      <td>    0</td>\n",
        "      <td>    96</td>\n",
        "      <td>  111</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048985</th>\n",
        "      <td>   2</td>\n",
        "      <td>   1</td>\n",
        "      <td>   0</td>\n",
        "      <td>    0</td>\n",
        "      <td>    30</td>\n",
        "      <td>   51</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048986</th>\n",
        "      <td>   4</td>\n",
        "      <td>   2</td>\n",
        "      <td>   0</td>\n",
        "      <td>    0</td>\n",
        "      <td>   419</td>\n",
        "      <td>  151</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048987</th>\n",
        "      <td>   4</td>\n",
        "      <td>   1</td>\n",
        "      <td>   0</td>\n",
        "      <td>    0</td>\n",
        "      <td>   192</td>\n",
        "      <td>  136</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189775</th>\n",
        "      <td>  10</td>\n",
        "      <td>   8</td>\n",
        "      <td>  16</td>\n",
        "      <td>   18</td>\n",
        "      <td>   100</td>\n",
        "      <td>   38</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189776</th>\n",
        "      <td>  12</td>\n",
        "      <td>  10</td>\n",
        "      <td>  20</td>\n",
        "      <td>   23</td>\n",
        "      <td>   178</td>\n",
        "      <td>   64</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189777</th>\n",
        "      <td>  12</td>\n",
        "      <td>  11</td>\n",
        "      <td>  15</td>\n",
        "      <td>   20</td>\n",
        "      <td>   573</td>\n",
        "      <td>  189</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189778</th>\n",
        "      <td>  12</td>\n",
        "      <td>   9</td>\n",
        "      <td>  18</td>\n",
        "      <td>   19</td>\n",
        "      <td>   525</td>\n",
        "      <td>  200</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189779</th>\n",
        "      <td> 226</td>\n",
        "      <td> 104</td>\n",
        "      <td> 431</td>\n",
        "      <td>  744</td>\n",
        "      <td>  9790</td>\n",
        "      <td> 1017</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189780</th>\n",
        "      <td> 144</td>\n",
        "      <td>  68</td>\n",
        "      <td> 203</td>\n",
        "      <td>  394</td>\n",
        "      <td>  3890</td>\n",
        "      <td>  567</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189781</th>\n",
        "      <td>  91</td>\n",
        "      <td>  41</td>\n",
        "      <td> 149</td>\n",
        "      <td>  268</td>\n",
        "      <td>  3674</td>\n",
        "      <td>  586</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189782</th>\n",
        "      <td>  16</td>\n",
        "      <td>   6</td>\n",
        "      <td>  25</td>\n",
        "      <td>   50</td>\n",
        "      <td>  1816</td>\n",
        "      <td>  456</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189783</th>\n",
        "      <td>  11</td>\n",
        "      <td>   5</td>\n",
        "      <td>   9</td>\n",
        "      <td>   16</td>\n",
        "      <td>  2435</td>\n",
        "      <td>  597</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189784</th>\n",
        "      <td>  49</td>\n",
        "      <td>  26</td>\n",
        "      <td>  76</td>\n",
        "      <td>  121</td>\n",
        "      <td>  2372</td>\n",
        "      <td>  501</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189785</th>\n",
        "      <td>  14</td>\n",
        "      <td>   6</td>\n",
        "      <td>  18</td>\n",
        "      <td>   20</td>\n",
        "      <td>   557</td>\n",
        "      <td>  178</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189786</th>\n",
        "      <td>  39</td>\n",
        "      <td>  15</td>\n",
        "      <td>  58</td>\n",
        "      <td>   90</td>\n",
        "      <td>  1870</td>\n",
        "      <td>  474</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189787</th>\n",
        "      <td>  15</td>\n",
        "      <td>  10</td>\n",
        "      <td>  22</td>\n",
        "      <td>   30</td>\n",
        "      <td>   806</td>\n",
        "      <td>  285</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR248532</th>\n",
        "      <td>  14</td>\n",
        "      <td>  11</td>\n",
        "      <td>  20</td>\n",
        "      <td>   28</td>\n",
        "      <td>   315</td>\n",
        "      <td>  109</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR309285</th>\n",
        "      <td>   6</td>\n",
        "      <td>   3</td>\n",
        "      <td>   2</td>\n",
        "      <td>    4</td>\n",
        "      <td>   148</td>\n",
        "      <td>   76</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR309286</th>\n",
        "      <td>   6</td>\n",
        "      <td>   3</td>\n",
        "      <td>   3</td>\n",
        "      <td>    7</td>\n",
        "      <td>   109</td>\n",
        "      <td>   78</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR309287</th>\n",
        "      <td>  11</td>\n",
        "      <td>   9</td>\n",
        "      <td>  14</td>\n",
        "      <td>   22</td>\n",
        "      <td>   141</td>\n",
        "      <td>   69</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR390466</th>\n",
        "      <td>  13</td>\n",
        "      <td>   4</td>\n",
        "      <td>  20</td>\n",
        "      <td>   37</td>\n",
        "      <td>  1377</td>\n",
        "      <td>  130</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR390467</th>\n",
        "      <td>  38</td>\n",
        "      <td>  17</td>\n",
        "      <td>  60</td>\n",
        "      <td>  120</td>\n",
        "      <td>  2004</td>\n",
        "      <td>  134</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398195</th>\n",
        "      <td>   4</td>\n",
        "      <td>   2</td>\n",
        "      <td>   0</td>\n",
        "      <td>    0</td>\n",
        "      <td>   210</td>\n",
        "      <td>   14</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398196</th>\n",
        "      <td>   5</td>\n",
        "      <td>   3</td>\n",
        "      <td>   4</td>\n",
        "      <td>    4</td>\n",
        "      <td>   405</td>\n",
        "      <td>   82</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398197</th>\n",
        "      <td>   8</td>\n",
        "      <td>   5</td>\n",
        "      <td>   6</td>\n",
        "      <td>   10</td>\n",
        "      <td>   962</td>\n",
        "      <td>  189</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398198</th>\n",
        "      <td>   0</td>\n",
        "      <td>   0</td>\n",
        "      <td>   0</td>\n",
        "      <td>    0</td>\n",
        "      <td>     0</td>\n",
        "      <td>    0</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398199</th>\n",
        "      <td>   5</td>\n",
        "      <td>   2</td>\n",
        "      <td>   3</td>\n",
        "      <td>    8</td>\n",
        "      <td>   485</td>\n",
        "      <td>   89</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398200</th>\n",
        "      <td>  10</td>\n",
        "      <td>   2</td>\n",
        "      <td>   5</td>\n",
        "      <td>   12</td>\n",
        "      <td>   387</td>\n",
        "      <td>   94</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398201</th>\n",
        "      <td>  12</td>\n",
        "      <td>   6</td>\n",
        "      <td>  11</td>\n",
        "      <td>   20</td>\n",
        "      <td>  1016</td>\n",
        "      <td>  190</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398202</th>\n",
        "      <td>   9</td>\n",
        "      <td>   3</td>\n",
        "      <td>   5</td>\n",
        "      <td>    5</td>\n",
        "      <td>   691</td>\n",
        "      <td>  139</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398203</th>\n",
        "      <td>  13</td>\n",
        "      <td>   5</td>\n",
        "      <td>  20</td>\n",
        "      <td>   47</td>\n",
        "      <td>   552</td>\n",
        "      <td>  101</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398204</th>\n",
        "      <td>   3</td>\n",
        "      <td>   1</td>\n",
        "      <td>   0</td>\n",
        "      <td>    1</td>\n",
        "      <td>   207</td>\n",
        "      <td>   61</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398205</th>\n",
        "      <td>   7</td>\n",
        "      <td>   3</td>\n",
        "      <td>   1</td>\n",
        "      <td>    6</td>\n",
        "      <td>   815</td>\n",
        "      <td>  164</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398206</th>\n",
        "      <td>  10</td>\n",
        "      <td>   3</td>\n",
        "      <td>   7</td>\n",
        "      <td>   16</td>\n",
        "      <td>   532</td>\n",
        "      <td>  100</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398207</th>\n",
        "      <td>  10</td>\n",
        "      <td>   4</td>\n",
        "      <td>  11</td>\n",
        "      <td>   16</td>\n",
        "      <td>  1326</td>\n",
        "      <td>  260</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398208</th>\n",
        "      <td>   8</td>\n",
        "      <td>   3</td>\n",
        "      <td>   5</td>\n",
        "      <td>    6</td>\n",
        "      <td>   983</td>\n",
        "      <td>  177</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398209</th>\n",
        "      <td>  10</td>\n",
        "      <td>   4</td>\n",
        "      <td>   4</td>\n",
        "      <td>   11</td>\n",
        "      <td>  1222</td>\n",
        "      <td>  225</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR408236</th>\n",
        "      <td> 219</td>\n",
        "      <td>  92</td>\n",
        "      <td> 362</td>\n",
        "      <td>  698</td>\n",
        "      <td> 11924</td>\n",
        "      <td> 1086</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR408237</th>\n",
        "      <td> 227</td>\n",
        "      <td>  97</td>\n",
        "      <td> 391</td>\n",
        "      <td>  782</td>\n",
        "      <td> 12646</td>\n",
        "      <td> 1181</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR427115</th>\n",
        "      <td>   0</td>\n",
        "      <td>   0</td>\n",
        "      <td>   0</td>\n",
        "      <td>    0</td>\n",
        "      <td>     0</td>\n",
        "      <td>    0</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR427116</th>\n",
        "      <td>   0</td>\n",
        "      <td>   0</td>\n",
        "      <td>   0</td>\n",
        "      <td>    0</td>\n",
        "      <td>     0</td>\n",
        "      <td>    0</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR452496</th>\n",
        "      <td>   7</td>\n",
        "      <td>   2</td>\n",
        "      <td>   2</td>\n",
        "      <td>    6</td>\n",
        "      <td>  1345</td>\n",
        "      <td>  261</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR452497</th>\n",
        "      <td>   4</td>\n",
        "      <td>   1</td>\n",
        "      <td>   1</td>\n",
        "      <td>    1</td>\n",
        "      <td>   273</td>\n",
        "      <td>  117</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR490650</th>\n",
        "      <td>  33</td>\n",
        "      <td>  14</td>\n",
        "      <td>  47</td>\n",
        "      <td>   70</td>\n",
        "      <td>   352</td>\n",
        "      <td>  113</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR518495</th>\n",
        "      <td>   5</td>\n",
        "      <td>   2</td>\n",
        "      <td>   1</td>\n",
        "      <td>    2</td>\n",
        "      <td>   313</td>\n",
        "      <td>  127</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR518496</th>\n",
        "      <td>   5</td>\n",
        "      <td>   2</td>\n",
        "      <td>   0</td>\n",
        "      <td>    1</td>\n",
        "      <td>    97</td>\n",
        "      <td>   58</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR518497</th>\n",
        "      <td>   7</td>\n",
        "      <td>   4</td>\n",
        "      <td>   4</td>\n",
        "      <td>    9</td>\n",
        "      <td>  1245</td>\n",
        "      <td>  290</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR518498</th>\n",
        "      <td>  10</td>\n",
        "      <td>  10</td>\n",
        "      <td>  11</td>\n",
        "      <td>   22</td>\n",
        "      <td>  1829</td>\n",
        "      <td>  372</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527727</th>\n",
        "      <td> 207</td>\n",
        "      <td>  91</td>\n",
        "      <td> 395</td>\n",
        "      <td>  755</td>\n",
        "      <td> 10628</td>\n",
        "      <td>  989</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527728</th>\n",
        "      <td>  66</td>\n",
        "      <td>  38</td>\n",
        "      <td> 119</td>\n",
        "      <td>  197</td>\n",
        "      <td>  2999</td>\n",
        "      <td>  493</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527729</th>\n",
        "      <td> 165</td>\n",
        "      <td>  83</td>\n",
        "      <td> 295</td>\n",
        "      <td>  576</td>\n",
        "      <td>  5990</td>\n",
        "      <td>  652</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527730</th>\n",
        "      <td> 197</td>\n",
        "      <td> 103</td>\n",
        "      <td> 332</td>\n",
        "      <td>  639</td>\n",
        "      <td>  6811</td>\n",
        "      <td>  719</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527731</th>\n",
        "      <td>   5</td>\n",
        "      <td>   1</td>\n",
        "      <td>   4</td>\n",
        "      <td>    5</td>\n",
        "      <td>   292</td>\n",
        "      <td>  122</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527732</th>\n",
        "      <td>  18</td>\n",
        "      <td>   6</td>\n",
        "      <td>  25</td>\n",
        "      <td>   45</td>\n",
        "      <td>  1216</td>\n",
        "      <td>  183</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR567525</th>\n",
        "      <td> 115</td>\n",
        "      <td>  22</td>\n",
        "      <td> 136</td>\n",
        "      <td>  370</td>\n",
        "      <td>  9914</td>\n",
        "      <td>  699</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR567526</th>\n",
        "      <td> 252</td>\n",
        "      <td> 106</td>\n",
        "      <td> 403</td>\n",
        "      <td>  926</td>\n",
        "      <td> 19619</td>\n",
        "      <td> 1506</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628446</th>\n",
        "      <td> 252</td>\n",
        "      <td> 145</td>\n",
        "      <td> 516</td>\n",
        "      <td>  878</td>\n",
        "      <td> 15364</td>\n",
        "      <td>  979</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628447</th>\n",
        "      <td>  39</td>\n",
        "      <td>  18</td>\n",
        "      <td>  46</td>\n",
        "      <td>  100</td>\n",
        "      <td>  1338</td>\n",
        "      <td>  201</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628448</th>\n",
        "      <td>  31</td>\n",
        "      <td>   9</td>\n",
        "      <td>  47</td>\n",
        "      <td>   68</td>\n",
        "      <td>  1491</td>\n",
        "      <td>  194</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628449</th>\n",
        "      <td>  34</td>\n",
        "      <td>  10</td>\n",
        "      <td>  50</td>\n",
        "      <td>   73</td>\n",
        "      <td>  1427</td>\n",
        "      <td>  215</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628450</th>\n",
        "      <td> 168</td>\n",
        "      <td>  81</td>\n",
        "      <td> 295</td>\n",
        "      <td>  471</td>\n",
        "      <td> 10885</td>\n",
        "      <td> 1005</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628451</th>\n",
        "      <td>  37</td>\n",
        "      <td>  17</td>\n",
        "      <td>  47</td>\n",
        "      <td>   84</td>\n",
        "      <td>  1599</td>\n",
        "      <td>  224</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628452</th>\n",
        "      <td>  53</td>\n",
        "      <td>  22</td>\n",
        "      <td> 102</td>\n",
        "      <td>  156</td>\n",
        "      <td>  2492</td>\n",
        "      <td>  350</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628453</th>\n",
        "      <td>  58</td>\n",
        "      <td>  29</td>\n",
        "      <td> 111</td>\n",
        "      <td>  173</td>\n",
        "      <td>  2556</td>\n",
        "      <td>  346</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628454</th>\n",
        "      <td> 106</td>\n",
        "      <td>  61</td>\n",
        "      <td> 185</td>\n",
        "      <td>  305</td>\n",
        "      <td>  7641</td>\n",
        "      <td>  828</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR764666</th>\n",
        "      <td>   5</td>\n",
        "      <td>   3</td>\n",
        "      <td>   2</td>\n",
        "      <td>    6</td>\n",
        "      <td>   903</td>\n",
        "      <td>  283</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR764667</th>\n",
        "      <td>   7</td>\n",
        "      <td>   4</td>\n",
        "      <td>   4</td>\n",
        "      <td>    7</td>\n",
        "      <td>    82</td>\n",
        "      <td>   42</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR764668</th>\n",
        "      <td>   6</td>\n",
        "      <td>   2</td>\n",
        "      <td>   3</td>\n",
        "      <td>    2</td>\n",
        "      <td>    99</td>\n",
        "      <td>   33</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR764669</th>\n",
        "      <td>   5</td>\n",
        "      <td>   3</td>\n",
        "      <td>   1</td>\n",
        "      <td>    4</td>\n",
        "      <td>   633</td>\n",
        "      <td>  234</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 24,
       "text": [
        "<IPython.core.display.HTML at 0x13703dd0>"
       ]
      }
     ],
     "prompt_number": 24
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "overlaps.columns"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 29,
       "text": [
        "Index([u'B3P27_IP_Day1.peaks.bed', u'B3P27_IP_Day23.peaks.bed', u'B3P27_IP_Day27.peaks.bed', u'B3P27_IP_Day6.peaks.bed', u'SRR334701.peaks.bed', u'SRR334703.peaks.bed'], dtype='object')"
       ]
      }
     ],
     "prompt_number": 29
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "overlaps = intersect_df / union_df\n",
      "HTML((overlaps).sort(\"SRR334701.peaks.bed\", ascending=False).to_html())"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>B3P27_IP_Day1.peaks.bed</th>\n",
        "      <th>B3P27_IP_Day23.peaks.bed</th>\n",
        "      <th>B3P27_IP_Day27.peaks.bed</th>\n",
        "      <th>B3P27_IP_Day6.peaks.bed</th>\n",
        "      <th>SRR334701.peaks.bed</th>\n",
        "      <th>SRR334703.peaks.bed</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>MDAMB231_hnRNPA2B1_WT_rep1</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td>      inf</td>\n",
        "      <td>      inf</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628450</th>\n",
        "      <td> 0.005094</td>\n",
        "      <td> 0.002473</td>\n",
        "      <td> 0.008801</td>\n",
        "      <td> 0.013730</td>\n",
        "      <td> 0.189783</td>\n",
        "      <td> 0.029976</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HeLa_Ago2_WT_rep1</th>\n",
        "      <td> 0.004221</td>\n",
        "      <td> 0.002175</td>\n",
        "      <td> 0.007355</td>\n",
        "      <td> 0.012351</td>\n",
        "      <td> 0.183998</td>\n",
        "      <td> 0.025204</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628454</th>\n",
        "      <td> 0.005002</td>\n",
        "      <td> 0.002915</td>\n",
        "      <td> 0.008492</td>\n",
        "      <td> 0.013481</td>\n",
        "      <td> 0.157520</td>\n",
        "      <td> 0.037055</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HeLa_Ago2_PTB_kd_rep1</th>\n",
        "      <td> 0.003006</td>\n",
        "      <td> 0.001863</td>\n",
        "      <td> 0.005948</td>\n",
        "      <td> 0.009455</td>\n",
        "      <td> 0.149841</td>\n",
        "      <td> 0.020138</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527727</th>\n",
        "      <td> 0.003396</td>\n",
        "      <td> 0.001497</td>\n",
        "      <td> 0.006435</td>\n",
        "      <td> 0.012193</td>\n",
        "      <td> 0.124831</td>\n",
        "      <td> 0.016152</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293FlpIn_FMR1-iso1_WT_rep1</th>\n",
        "      <td> 0.002940</td>\n",
        "      <td> 0.001457</td>\n",
        "      <td> 0.005629</td>\n",
        "      <td> 0.010531</td>\n",
        "      <td> 0.116535</td>\n",
        "      <td> 0.012818</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_IGF2BP1_WT_rep1</th>\n",
        "      <td> 0.003182</td>\n",
        "      <td> 0.001582</td>\n",
        "      <td> 0.005929</td>\n",
        "      <td> 0.010016</td>\n",
        "      <td> 0.113316</td>\n",
        "      <td> 0.013137</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR567525</th>\n",
        "      <td> 0.001795</td>\n",
        "      <td> 0.000345</td>\n",
        "      <td> 0.002102</td>\n",
        "      <td> 0.005652</td>\n",
        "      <td> 0.111764</td>\n",
        "      <td> 0.010742</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_IGF2BP3_WT_rep1</th>\n",
        "      <td> 0.002990</td>\n",
        "      <td> 0.001529</td>\n",
        "      <td> 0.005926</td>\n",
        "      <td> 0.009423</td>\n",
        "      <td> 0.111378</td>\n",
        "      <td> 0.013268</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR567526</th>\n",
        "      <td> 0.001424</td>\n",
        "      <td> 0.000600</td>\n",
        "      <td> 0.002271</td>\n",
        "      <td> 0.005206</td>\n",
        "      <td> 0.103955</td>\n",
        "      <td> 0.008763</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HeLa_eIF4AIII_WT_rep1</th>\n",
        "      <td> 0.001333</td>\n",
        "      <td> 0.000575</td>\n",
        "      <td> 0.002266</td>\n",
        "      <td> 0.005000</td>\n",
        "      <td> 0.100907</td>\n",
        "      <td> 0.008127</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628446</th>\n",
        "      <td> 0.001649</td>\n",
        "      <td> 0.000950</td>\n",
        "      <td> 0.003368</td>\n",
        "      <td> 0.005707</td>\n",
        "      <td> 0.093368</td>\n",
        "      <td> 0.006777</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_IGF2BP2_WT_rep1</th>\n",
        "      <td> 0.003532</td>\n",
        "      <td> 0.001871</td>\n",
        "      <td> 0.007023</td>\n",
        "      <td> 0.011085</td>\n",
        "      <td> 0.090244</td>\n",
        "      <td> 0.015724</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293FlpIn_FMR1-iso7_WT_rep1</th>\n",
        "      <td> 0.003023</td>\n",
        "      <td> 0.001624</td>\n",
        "      <td> 0.005596</td>\n",
        "      <td> 0.010383</td>\n",
        "      <td> 0.089293</td>\n",
        "      <td> 0.013557</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527729</th>\n",
        "      <td> 0.003663</td>\n",
        "      <td> 0.001852</td>\n",
        "      <td> 0.006478</td>\n",
        "      <td> 0.012473</td>\n",
        "      <td> 0.084197</td>\n",
        "      <td> 0.014846</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527730</th>\n",
        "      <td> 0.003352</td>\n",
        "      <td> 0.001759</td>\n",
        "      <td> 0.005603</td>\n",
        "      <td> 0.010676</td>\n",
        "      <td> 0.084039</td>\n",
        "      <td> 0.013240</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago2_WT_rep2</th>\n",
        "      <td> 0.003054</td>\n",
        "      <td> 0.001822</td>\n",
        "      <td> 0.005954</td>\n",
        "      <td> 0.009023</td>\n",
        "      <td> 0.077378</td>\n",
        "      <td> 0.028961</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048951</th>\n",
        "      <td> 0.005649</td>\n",
        "      <td> 0.002782</td>\n",
        "      <td> 0.009818</td>\n",
        "      <td> 0.015118</td>\n",
        "      <td> 0.076654</td>\n",
        "      <td> 0.016198</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048964</th>\n",
        "      <td> 0.004772</td>\n",
        "      <td> 0.002471</td>\n",
        "      <td> 0.007441</td>\n",
        "      <td> 0.012491</td>\n",
        "      <td> 0.073489</td>\n",
        "      <td> 0.020521</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527728</th>\n",
        "      <td> 0.005193</td>\n",
        "      <td> 0.003057</td>\n",
        "      <td> 0.008938</td>\n",
        "      <td> 0.013883</td>\n",
        "      <td> 0.068416</td>\n",
        "      <td> 0.034957</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048952</th>\n",
        "      <td> 0.006876</td>\n",
        "      <td> 0.003484</td>\n",
        "      <td> 0.009737</td>\n",
        "      <td> 0.013920</td>\n",
        "      <td> 0.068301</td>\n",
        "      <td> 0.032568</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189783</th>\n",
        "      <td> 0.002745</td>\n",
        "      <td> 0.001350</td>\n",
        "      <td> 0.001927</td>\n",
        "      <td> 0.002843</td>\n",
        "      <td> 0.067618</td>\n",
        "      <td> 0.106550</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago2_WT_rep4</th>\n",
        "      <td> 0.003871</td>\n",
        "      <td> 0.001579</td>\n",
        "      <td> 0.005795</td>\n",
        "      <td> 0.007988</td>\n",
        "      <td> 0.063740</td>\n",
        "      <td> 0.040455</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR408236</th>\n",
        "      <td> 0.001234</td>\n",
        "      <td> 0.000519</td>\n",
        "      <td> 0.002034</td>\n",
        "      <td> 0.003909</td>\n",
        "      <td> 0.063387</td>\n",
        "      <td> 0.006569</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048958</th>\n",
        "      <td> 0.005053</td>\n",
        "      <td> 0.002554</td>\n",
        "      <td> 0.008891</td>\n",
        "      <td> 0.013014</td>\n",
        "      <td> 0.059920</td>\n",
        "      <td> 0.013893</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR408237</th>\n",
        "      <td> 0.001024</td>\n",
        "      <td> 0.000438</td>\n",
        "      <td> 0.001760</td>\n",
        "      <td> 0.003511</td>\n",
        "      <td> 0.056641</td>\n",
        "      <td> 0.005879</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048963</th>\n",
        "      <td> 0.003631</td>\n",
        "      <td> 0.001736</td>\n",
        "      <td> 0.006232</td>\n",
        "      <td> 0.010071</td>\n",
        "      <td> 0.056251</td>\n",
        "      <td> 0.011061</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_DGCR8_WT_rep1</th>\n",
        "      <td> 0.002802</td>\n",
        "      <td> 0.002485</td>\n",
        "      <td> 0.002835</td>\n",
        "      <td> 0.004149</td>\n",
        "      <td> 0.056044</td>\n",
        "      <td> 0.076704</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048965</th>\n",
        "      <td> 0.004591</td>\n",
        "      <td> 0.003610</td>\n",
        "      <td> 0.007427</td>\n",
        "      <td> 0.008787</td>\n",
        "      <td> 0.055711</td>\n",
        "      <td> 0.035677</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189784</th>\n",
        "      <td> 0.003755</td>\n",
        "      <td> 0.002036</td>\n",
        "      <td> 0.005554</td>\n",
        "      <td> 0.008281</td>\n",
        "      <td> 0.052897</td>\n",
        "      <td> 0.034149</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR518498</th>\n",
        "      <td> 0.004044</td>\n",
        "      <td> 0.004619</td>\n",
        "      <td> 0.003512</td>\n",
        "      <td> 0.005382</td>\n",
        "      <td> 0.052797</td>\n",
        "      <td> 0.086916</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago2_WT_rep3</th>\n",
        "      <td> 0.002991</td>\n",
        "      <td> 0.001465</td>\n",
        "      <td> 0.004847</td>\n",
        "      <td> 0.007693</td>\n",
        "      <td> 0.052651</td>\n",
        "      <td> 0.027179</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628453</th>\n",
        "      <td> 0.002983</td>\n",
        "      <td> 0.001513</td>\n",
        "      <td> 0.005534</td>\n",
        "      <td> 0.008257</td>\n",
        "      <td> 0.050533</td>\n",
        "      <td> 0.016383</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628452</th>\n",
        "      <td> 0.002892</td>\n",
        "      <td> 0.001219</td>\n",
        "      <td> 0.005383</td>\n",
        "      <td> 0.007859</td>\n",
        "      <td> 0.050396</td>\n",
        "      <td> 0.017534</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189786</th>\n",
        "      <td> 0.004739</td>\n",
        "      <td> 0.001887</td>\n",
        "      <td> 0.006537</td>\n",
        "      <td> 0.009170</td>\n",
        "      <td> 0.046573</td>\n",
        "      <td> 0.048234</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189782</th>\n",
        "      <td> 0.002140</td>\n",
        "      <td> 0.000836</td>\n",
        "      <td> 0.003074</td>\n",
        "      <td> 0.005509</td>\n",
        "      <td> 0.045810</td>\n",
        "      <td> 0.049874</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189779</th>\n",
        "      <td> 0.001087</td>\n",
        "      <td> 0.000501</td>\n",
        "      <td> 0.002069</td>\n",
        "      <td> 0.003560</td>\n",
        "      <td> 0.044649</td>\n",
        "      <td> 0.005222</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048956</th>\n",
        "      <td> 0.003734</td>\n",
        "      <td> 0.002117</td>\n",
        "      <td> 0.007147</td>\n",
        "      <td> 0.010766</td>\n",
        "      <td> 0.040522</td>\n",
        "      <td> 0.006894</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_PAPD5_WT_rep1</th>\n",
        "      <td> 0.001878</td>\n",
        "      <td> 0.000584</td>\n",
        "      <td> 0.000683</td>\n",
        "      <td> 0.000934</td>\n",
        "      <td> 0.040396</td>\n",
        "      <td> 0.058770</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628451</th>\n",
        "      <td> 0.004072</td>\n",
        "      <td> 0.001931</td>\n",
        "      <td> 0.004823</td>\n",
        "      <td> 0.007873</td>\n",
        "      <td> 0.039061</td>\n",
        "      <td> 0.020588</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR452496</th>\n",
        "      <td> 0.002649</td>\n",
        "      <td> 0.000854</td>\n",
        "      <td> 0.000604</td>\n",
        "      <td> 0.001405</td>\n",
        "      <td> 0.038351</td>\n",
        "      <td> 0.059010</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_hnRNPU_WT_rep1</th>\n",
        "      <td> 0.000794</td>\n",
        "      <td> 0.000301</td>\n",
        "      <td> 0.001256</td>\n",
        "      <td> 0.002492</td>\n",
        "      <td> 0.036539</td>\n",
        "      <td> 0.004505</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628448</th>\n",
        "      <td> 0.003411</td>\n",
        "      <td> 0.001022</td>\n",
        "      <td> 0.004827</td>\n",
        "      <td> 0.006366</td>\n",
        "      <td> 0.036311</td>\n",
        "      <td> 0.017833</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR518497</th>\n",
        "      <td> 0.003497</td>\n",
        "      <td> 0.002354</td>\n",
        "      <td> 0.001500</td>\n",
        "      <td> 0.002482</td>\n",
        "      <td> 0.036188</td>\n",
        "      <td> 0.075130</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_DGCR8_overexpressed_rep1</th>\n",
        "      <td> 0.002191</td>\n",
        "      <td> 0.001793</td>\n",
        "      <td> 0.001159</td>\n",
        "      <td> 0.001896</td>\n",
        "      <td> 0.035378</td>\n",
        "      <td> 0.052911</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628449</th>\n",
        "      <td> 0.003626</td>\n",
        "      <td> 0.001100</td>\n",
        "      <td> 0.004987</td>\n",
        "      <td> 0.006656</td>\n",
        "      <td> 0.034459</td>\n",
        "      <td> 0.019248</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398209</th>\n",
        "      <td> 0.002728</td>\n",
        "      <td> 0.001188</td>\n",
        "      <td> 0.000923</td>\n",
        "      <td> 0.002079</td>\n",
        "      <td> 0.033888</td>\n",
        "      <td> 0.040717</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527732</th>\n",
        "      <td> 0.004048</td>\n",
        "      <td> 0.001445</td>\n",
        "      <td> 0.004896</td>\n",
        "      <td> 0.007436</td>\n",
        "      <td> 0.033076</td>\n",
        "      <td> 0.028947</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>293T_hnRNPA1_WT_rep1</th>\n",
        "      <td> 0.001406</td>\n",
        "      <td> 0.000482</td>\n",
        "      <td> 0.000706</td>\n",
        "      <td> 0.001220</td>\n",
        "      <td> 0.032743</td>\n",
        "      <td> 0.021878</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628447</th>\n",
        "      <td> 0.003996</td>\n",
        "      <td> 0.001900</td>\n",
        "      <td> 0.004414</td>\n",
        "      <td> 0.008821</td>\n",
        "      <td> 0.032026</td>\n",
        "      <td> 0.017388</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398207</th>\n",
        "      <td> 0.001025</td>\n",
        "      <td> 0.000423</td>\n",
        "      <td> 0.001056</td>\n",
        "      <td> 0.001407</td>\n",
        "      <td> 0.031553</td>\n",
        "      <td> 0.022470</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR390466</th>\n",
        "      <td> 0.000973</td>\n",
        "      <td> 0.000306</td>\n",
        "      <td> 0.001427</td>\n",
        "      <td> 0.002472</td>\n",
        "      <td> 0.030661</td>\n",
        "      <td> 0.008794</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>293T_hnRNPF_WT_rep1</th>\n",
        "      <td> 0.000915</td>\n",
        "      <td> 0.000301</td>\n",
        "      <td> 0.001203</td>\n",
        "      <td> 0.002295</td>\n",
        "      <td> 0.030082</td>\n",
        "      <td> 0.006932</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189781</th>\n",
        "      <td> 0.000924</td>\n",
        "      <td> 0.000417</td>\n",
        "      <td> 0.001503</td>\n",
        "      <td> 0.002680</td>\n",
        "      <td> 0.029384</td>\n",
        "      <td> 0.006089</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398208</th>\n",
        "      <td> 0.004665</td>\n",
        "      <td> 0.002123</td>\n",
        "      <td> 0.002100</td>\n",
        "      <td> 0.001795</td>\n",
        "      <td> 0.028705</td>\n",
        "      <td> 0.049003</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_HuR_WT_rep3</th>\n",
        "      <td> 0.000612</td>\n",
        "      <td> 0.000307</td>\n",
        "      <td> 0.001130</td>\n",
        "      <td> 0.002072</td>\n",
        "      <td> 0.026327</td>\n",
        "      <td> 0.002793</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048959</th>\n",
        "      <td> 0.006704</td>\n",
        "      <td> 0.004279</td>\n",
        "      <td> 0.009033</td>\n",
        "      <td> 0.012428</td>\n",
        "      <td> 0.025739</td>\n",
        "      <td> 0.023730</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398197</th>\n",
        "      <td> 0.001355</td>\n",
        "      <td> 0.000892</td>\n",
        "      <td> 0.000913</td>\n",
        "      <td> 0.001328</td>\n",
        "      <td> 0.025062</td>\n",
        "      <td> 0.024331</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048957</th>\n",
        "      <td> 0.005432</td>\n",
        "      <td> 0.003108</td>\n",
        "      <td> 0.008377</td>\n",
        "      <td> 0.010855</td>\n",
        "      <td> 0.024631</td>\n",
        "      <td> 0.023058</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>293T_hnRNPU_WT_rep1</th>\n",
        "      <td> 0.000496</td>\n",
        "      <td> 0.000245</td>\n",
        "      <td> 0.000765</td>\n",
        "      <td> 0.001542</td>\n",
        "      <td> 0.024595</td>\n",
        "      <td> 0.004388</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398201</th>\n",
        "      <td> 0.001188</td>\n",
        "      <td> 0.000612</td>\n",
        "      <td> 0.001022</td>\n",
        "      <td> 0.001707</td>\n",
        "      <td> 0.023887</td>\n",
        "      <td> 0.015912</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR764666</th>\n",
        "      <td> 0.000690</td>\n",
        "      <td> 0.000432</td>\n",
        "      <td> 0.000253</td>\n",
        "      <td> 0.000676</td>\n",
        "      <td> 0.022920</td>\n",
        "      <td> 0.032857</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>FOX2</th>\n",
        "      <td> 0.000356</td>\n",
        "      <td> 0.000174</td>\n",
        "      <td> 0.000547</td>\n",
        "      <td> 0.001108</td>\n",
        "      <td> 0.022550</td>\n",
        "      <td> 0.003121</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago4_WT_rep1</th>\n",
        "      <td> 0.001601</td>\n",
        "      <td> 0.001418</td>\n",
        "      <td> 0.001320</td>\n",
        "      <td> 0.000421</td>\n",
        "      <td> 0.021753</td>\n",
        "      <td> 0.059340</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048953</th>\n",
        "      <td> 0.007171</td>\n",
        "      <td> 0.003617</td>\n",
        "      <td> 0.005361</td>\n",
        "      <td> 0.006055</td>\n",
        "      <td> 0.021661</td>\n",
        "      <td> 0.033349</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189787</th>\n",
        "      <td> 0.003149</td>\n",
        "      <td> 0.002241</td>\n",
        "      <td> 0.004058</td>\n",
        "      <td> 0.004703</td>\n",
        "      <td> 0.021659</td>\n",
        "      <td> 0.043779</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR390467</th>\n",
        "      <td> 0.000599</td>\n",
        "      <td> 0.000269</td>\n",
        "      <td> 0.000936</td>\n",
        "      <td> 0.001847</td>\n",
        "      <td> 0.021569</td>\n",
        "      <td> 0.002115</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189780</th>\n",
        "      <td> 0.000801</td>\n",
        "      <td> 0.000379</td>\n",
        "      <td> 0.001125</td>\n",
        "      <td> 0.002175</td>\n",
        "      <td> 0.020002</td>\n",
        "      <td> 0.003422</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_LIN28B_WT_rep1</th>\n",
        "      <td> 0.000470</td>\n",
        "      <td> 0.000291</td>\n",
        "      <td> 0.000443</td>\n",
        "      <td> 0.000817</td>\n",
        "      <td> 0.019713</td>\n",
        "      <td> 0.025513</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398202</th>\n",
        "      <td> 0.003400</td>\n",
        "      <td> 0.001277</td>\n",
        "      <td> 0.001508</td>\n",
        "      <td> 0.001169</td>\n",
        "      <td> 0.019584</td>\n",
        "      <td> 0.030396</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_TNRC6C_WT_rep1</th>\n",
        "      <td> 0.001565</td>\n",
        "      <td> 0.000692</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.018971</td>\n",
        "      <td> 0.062593</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>293T_hnRNPM_WT_rep1</th>\n",
        "      <td> 0.000374</td>\n",
        "      <td> 0.000142</td>\n",
        "      <td> 0.000136</td>\n",
        "      <td> 0.000304</td>\n",
        "      <td> 0.018910</td>\n",
        "      <td> 0.009500</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048980</th>\n",
        "      <td> 0.004545</td>\n",
        "      <td> 0.002369</td>\n",
        "      <td> 0.007553</td>\n",
        "      <td> 0.009857</td>\n",
        "      <td> 0.018525</td>\n",
        "      <td> 0.017197</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048960</th>\n",
        "      <td> 0.006037</td>\n",
        "      <td> 0.005270</td>\n",
        "      <td> 0.003215</td>\n",
        "      <td> 0.004065</td>\n",
        "      <td> 0.017750</td>\n",
        "      <td> 0.043561</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398205</th>\n",
        "      <td> 0.000509</td>\n",
        "      <td> 0.000223</td>\n",
        "      <td> 0.000069</td>\n",
        "      <td> 0.000390</td>\n",
        "      <td> 0.017645</td>\n",
        "      <td> 0.010570</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048966</th>\n",
        "      <td> 0.004464</td>\n",
        "      <td> 0.002368</td>\n",
        "      <td> 0.003138</td>\n",
        "      <td> 0.004078</td>\n",
        "      <td> 0.016433</td>\n",
        "      <td> 0.030382</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048950</th>\n",
        "      <td> 0.006702</td>\n",
        "      <td> 0.003488</td>\n",
        "      <td> 0.006601</td>\n",
        "      <td> 0.009565</td>\n",
        "      <td> 0.015824</td>\n",
        "      <td> 0.020209</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR764669</th>\n",
        "      <td> 0.000614</td>\n",
        "      <td> 0.000383</td>\n",
        "      <td> 0.000114</td>\n",
        "      <td> 0.000410</td>\n",
        "      <td> 0.015673</td>\n",
        "      <td> 0.024503</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189785</th>\n",
        "      <td> 0.003373</td>\n",
        "      <td> 0.001557</td>\n",
        "      <td> 0.003743</td>\n",
        "      <td> 0.003465</td>\n",
        "      <td> 0.015121</td>\n",
        "      <td> 0.029637</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048979</th>\n",
        "      <td> 0.002865</td>\n",
        "      <td> 0.001386</td>\n",
        "      <td> 0.000415</td>\n",
        "      <td> 0.000296</td>\n",
        "      <td> 0.014618</td>\n",
        "      <td> 0.056409</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_QIK_WT_rep1</th>\n",
        "      <td> 0.000914</td>\n",
        "      <td> 0.000426</td>\n",
        "      <td> 0.000945</td>\n",
        "      <td> 0.001172</td>\n",
        "      <td> 0.014207</td>\n",
        "      <td> 0.018898</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398206</th>\n",
        "      <td> 0.001946</td>\n",
        "      <td> 0.000620</td>\n",
        "      <td> 0.001206</td>\n",
        "      <td> 0.002367</td>\n",
        "      <td> 0.014063</td>\n",
        "      <td> 0.014108</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398203</th>\n",
        "      <td> 0.001639</td>\n",
        "      <td> 0.000655</td>\n",
        "      <td> 0.002330</td>\n",
        "      <td> 0.004936</td>\n",
        "      <td> 0.013606</td>\n",
        "      <td> 0.010243</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048962</th>\n",
        "      <td> 0.004700</td>\n",
        "      <td> 0.001587</td>\n",
        "      <td> 0.004992</td>\n",
        "      <td> 0.008489</td>\n",
        "      <td> 0.013020</td>\n",
        "      <td> 0.015623</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398199</th>\n",
        "      <td> 0.000943</td>\n",
        "      <td> 0.000400</td>\n",
        "      <td> 0.000503</td>\n",
        "      <td> 0.001156</td>\n",
        "      <td> 0.012767</td>\n",
        "      <td> 0.012346</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048986</th>\n",
        "      <td> 0.001873</td>\n",
        "      <td> 0.001092</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.012081</td>\n",
        "      <td> 0.039695</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398196</th>\n",
        "      <td> 0.001943</td>\n",
        "      <td> 0.001323</td>\n",
        "      <td> 0.001237</td>\n",
        "      <td> 0.000953</td>\n",
        "      <td> 0.011469</td>\n",
        "      <td> 0.018170</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398200</th>\n",
        "      <td> 0.001672</td>\n",
        "      <td> 0.000352</td>\n",
        "      <td> 0.000752</td>\n",
        "      <td> 0.001578</td>\n",
        "      <td> 0.009992</td>\n",
        "      <td> 0.011870</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Pumilio2_WT_rep1</th>\n",
        "      <td> 0.001130</td>\n",
        "      <td> 0.000443</td>\n",
        "      <td> 0.001033</td>\n",
        "      <td> 0.002185</td>\n",
        "      <td> 0.009885</td>\n",
        "      <td> 0.023507</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048981</th>\n",
        "      <td> 0.003002</td>\n",
        "      <td> 0.002972</td>\n",
        "      <td> 0.005478</td>\n",
        "      <td> 0.005644</td>\n",
        "      <td> 0.009328</td>\n",
        "      <td> 0.024765</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048973</th>\n",
        "      <td> 0.004499</td>\n",
        "      <td> 0.001863</td>\n",
        "      <td> 0.003215</td>\n",
        "      <td> 0.001964</td>\n",
        "      <td> 0.009021</td>\n",
        "      <td> 0.032228</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048976</th>\n",
        "      <td> 0.002470</td>\n",
        "      <td> 0.000789</td>\n",
        "      <td> 0.001428</td>\n",
        "      <td> 0.001345</td>\n",
        "      <td> 0.009002</td>\n",
        "      <td> 0.034357</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR518495</th>\n",
        "      <td> 0.001691</td>\n",
        "      <td> 0.000754</td>\n",
        "      <td> 0.000276</td>\n",
        "      <td> 0.000436</td>\n",
        "      <td> 0.008763</td>\n",
        "      <td> 0.026343</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048971</th>\n",
        "      <td> 0.003149</td>\n",
        "      <td> 0.001040</td>\n",
        "      <td> 0.000692</td>\n",
        "      <td> 0.001038</td>\n",
        "      <td> 0.008615</td>\n",
        "      <td> 0.032649</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527731</th>\n",
        "      <td> 0.001716</td>\n",
        "      <td> 0.000383</td>\n",
        "      <td> 0.001119</td>\n",
        "      <td> 0.001102</td>\n",
        "      <td> 0.008180</td>\n",
        "      <td> 0.025374</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_TNRC6A_WT_rep1</th>\n",
        "      <td> 0.001913</td>\n",
        "      <td> 0.000865</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000236</td>\n",
        "      <td> 0.008040</td>\n",
        "      <td> 0.037223</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago3_WT_rep1</th>\n",
        "      <td> 0.002262</td>\n",
        "      <td> 0.001074</td>\n",
        "      <td> 0.001063</td>\n",
        "      <td> 0.001482</td>\n",
        "      <td> 0.007932</td>\n",
        "      <td> 0.033843</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR452497</th>\n",
        "      <td> 0.002328</td>\n",
        "      <td> 0.000707</td>\n",
        "      <td> 0.000420</td>\n",
        "      <td> 0.000299</td>\n",
        "      <td> 0.007903</td>\n",
        "      <td> 0.032249</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR034467</th>\n",
        "      <td> 0.000759</td>\n",
        "      <td> 0.000477</td>\n",
        "      <td> 0.000551</td>\n",
        "      <td> 0.000365</td>\n",
        "      <td> 0.007739</td>\n",
        "      <td> 0.010292</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR490650</th>\n",
        "      <td> 0.002423</td>\n",
        "      <td> 0.001050</td>\n",
        "      <td> 0.003295</td>\n",
        "      <td> 0.004605</td>\n",
        "      <td> 0.007645</td>\n",
        "      <td> 0.007419</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048961</th>\n",
        "      <td> 0.003439</td>\n",
        "      <td> 0.001164</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.007009</td>\n",
        "      <td> 0.034113</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago2_WT_rep1</th>\n",
        "      <td> 0.002040</td>\n",
        "      <td> 0.001206</td>\n",
        "      <td> 0.000762</td>\n",
        "      <td> 0.000279</td>\n",
        "      <td> 0.006702</td>\n",
        "      <td> 0.035901</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR034469</th>\n",
        "      <td> 0.000665</td>\n",
        "      <td> 0.000416</td>\n",
        "      <td> 0.000244</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.006194</td>\n",
        "      <td> 0.007391</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_TNRC6B_WT_rep1</th>\n",
        "      <td> 0.001677</td>\n",
        "      <td> 0.000480</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000249</td>\n",
        "      <td> 0.006045</td>\n",
        "      <td> 0.042011</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_HuR_WT_rep6</th>\n",
        "      <td> 0.001439</td>\n",
        "      <td> 0.001042</td>\n",
        "      <td> 0.001604</td>\n",
        "      <td> 0.002459</td>\n",
        "      <td> 0.006037</td>\n",
        "      <td> 0.010079</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398195</th>\n",
        "      <td> 0.001770</td>\n",
        "      <td> 0.001023</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.005987</td>\n",
        "      <td> 0.003314</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048970</th>\n",
        "      <td> 0.002018</td>\n",
        "      <td> 0.000919</td>\n",
        "      <td> 0.000636</td>\n",
        "      <td> 0.000975</td>\n",
        "      <td> 0.005896</td>\n",
        "      <td> 0.020985</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398204</th>\n",
        "      <td> 0.001016</td>\n",
        "      <td> 0.000378</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000218</td>\n",
        "      <td> 0.005783</td>\n",
        "      <td> 0.012416</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048987</th>\n",
        "      <td> 0.002714</td>\n",
        "      <td> 0.000854</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.005604</td>\n",
        "      <td> 0.040915</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048975</th>\n",
        "      <td> 0.003037</td>\n",
        "      <td> 0.000986</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.005564</td>\n",
        "      <td> 0.031825</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189778</th>\n",
        "      <td> 0.000193</td>\n",
        "      <td> 0.000145</td>\n",
        "      <td> 0.000286</td>\n",
        "      <td> 0.000297</td>\n",
        "      <td> 0.005531</td>\n",
        "      <td> 0.003122</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048969</th>\n",
        "      <td> 0.002337</td>\n",
        "      <td> 0.000883</td>\n",
        "      <td> 0.000618</td>\n",
        "      <td> 0.000954</td>\n",
        "      <td> 0.005402</td>\n",
        "      <td> 0.024483</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189777</th>\n",
        "      <td> 0.000151</td>\n",
        "      <td> 0.000139</td>\n",
        "      <td> 0.000187</td>\n",
        "      <td> 0.000246</td>\n",
        "      <td> 0.005110</td>\n",
        "      <td> 0.002325</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048967</th>\n",
        "      <td> 0.001676</td>\n",
        "      <td> 0.000305</td>\n",
        "      <td> 0.001178</td>\n",
        "      <td> 0.002308</td>\n",
        "      <td> 0.004913</td>\n",
        "      <td> 0.021686</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR248532</th>\n",
        "      <td> 0.000440</td>\n",
        "      <td> 0.000349</td>\n",
        "      <td> 0.000616</td>\n",
        "      <td> 0.000838</td>\n",
        "      <td> 0.004895</td>\n",
        "      <td> 0.003253</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_HuR_WT_rep2</th>\n",
        "      <td> 0.000126</td>\n",
        "      <td> 0.000084</td>\n",
        "      <td> 0.000161</td>\n",
        "      <td> 0.000231</td>\n",
        "      <td> 0.004804</td>\n",
        "      <td> 0.001614</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048978</th>\n",
        "      <td> 0.003494</td>\n",
        "      <td> 0.001773</td>\n",
        "      <td> 0.000477</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.004264</td>\n",
        "      <td> 0.033557</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR034468</th>\n",
        "      <td> 0.000423</td>\n",
        "      <td> 0.000295</td>\n",
        "      <td> 0.000129</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.004230</td>\n",
        "      <td> 0.007520</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR309285</th>\n",
        "      <td> 0.002779</td>\n",
        "      <td> 0.001616</td>\n",
        "      <td> 0.000708</td>\n",
        "      <td> 0.001057</td>\n",
        "      <td> 0.004226</td>\n",
        "      <td> 0.018627</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048954</th>\n",
        "      <td> 0.004864</td>\n",
        "      <td> 0.002759</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000376</td>\n",
        "      <td> 0.003981</td>\n",
        "      <td> 0.031377</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048968</th>\n",
        "      <td> 0.001524</td>\n",
        "      <td> 0.000550</td>\n",
        "      <td> 0.000869</td>\n",
        "      <td> 0.001618</td>\n",
        "      <td> 0.003941</td>\n",
        "      <td> 0.022242</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR309287</th>\n",
        "      <td> 0.001998</td>\n",
        "      <td> 0.001730</td>\n",
        "      <td> 0.002272</td>\n",
        "      <td> 0.003090</td>\n",
        "      <td> 0.003683</td>\n",
        "      <td> 0.009360</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048983</th>\n",
        "      <td> 0.003453</td>\n",
        "      <td> 0.001747</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000325</td>\n",
        "      <td> 0.003418</td>\n",
        "      <td> 0.029188</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR309286</th>\n",
        "      <td> 0.002300</td>\n",
        "      <td> 0.001300</td>\n",
        "      <td> 0.000916</td>\n",
        "      <td> 0.001654</td>\n",
        "      <td> 0.003072</td>\n",
        "      <td> 0.017184</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR764668</th>\n",
        "      <td> 0.003595</td>\n",
        "      <td> 0.001464</td>\n",
        "      <td> 0.001285</td>\n",
        "      <td> 0.000606</td>\n",
        "      <td> 0.002872</td>\n",
        "      <td> 0.009272</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189776</th>\n",
        "      <td> 0.000396</td>\n",
        "      <td> 0.000334</td>\n",
        "      <td> 0.000646</td>\n",
        "      <td> 0.000721</td>\n",
        "      <td> 0.002821</td>\n",
        "      <td> 0.001988</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048984</th>\n",
        "      <td> 0.002869</td>\n",
        "      <td> 0.000916</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.002805</td>\n",
        "      <td> 0.034028</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048955</th>\n",
        "      <td> 0.002708</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.002340</td>\n",
        "      <td> 0.000645</td>\n",
        "      <td> 0.002795</td>\n",
        "      <td> 0.005493</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR518496</th>\n",
        "      <td> 0.001638</td>\n",
        "      <td> 0.000728</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000214</td>\n",
        "      <td> 0.002699</td>\n",
        "      <td> 0.011600</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_HuR_WT_rep1</th>\n",
        "      <td> 0.000224</td>\n",
        "      <td> 0.000162</td>\n",
        "      <td> 0.000321</td>\n",
        "      <td> 0.000378</td>\n",
        "      <td> 0.002569</td>\n",
        "      <td> 0.001353</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR764667</th>\n",
        "      <td> 0.003037</td>\n",
        "      <td> 0.001998</td>\n",
        "      <td> 0.001346</td>\n",
        "      <td> 0.001781</td>\n",
        "      <td> 0.002333</td>\n",
        "      <td> 0.009913</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048982</th>\n",
        "      <td> 0.003676</td>\n",
        "      <td> 0.001892</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.001986</td>\n",
        "      <td> 0.021005</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048972</th>\n",
        "      <td> 0.004115</td>\n",
        "      <td> 0.001727</td>\n",
        "      <td> 0.000470</td>\n",
        "      <td> 0.000972</td>\n",
        "      <td> 0.001689</td>\n",
        "      <td> 0.018573</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189775</th>\n",
        "      <td> 0.000298</td>\n",
        "      <td> 0.000241</td>\n",
        "      <td> 0.000467</td>\n",
        "      <td> 0.000512</td>\n",
        "      <td> 0.001506</td>\n",
        "      <td> 0.001071</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048974</th>\n",
        "      <td> 0.003055</td>\n",
        "      <td> 0.001473</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000383</td>\n",
        "      <td> 0.001180</td>\n",
        "      <td> 0.017984</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048985</th>\n",
        "      <td> 0.001730</td>\n",
        "      <td> 0.001178</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000883</td>\n",
        "      <td> 0.016893</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048977</th>\n",
        "      <td> 0.001274</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000504</td>\n",
        "      <td> 0.010515</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR427115</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398198</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR427116</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 26,
       "text": [
        "<IPython.core.display.HTML at 0x2aaaaecc1c10>"
       ]
      }
     ],
     "prompt_number": 26
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "result = overlaps.sort(\"SRR334701.peaks.bed\", ascending=False)['SRR334701.peaks.bed'][1:24]"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 48
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "from gscripts.general import dataviz\n",
      "import seaborn as sns"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 54
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "sns.barplot?"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 65
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "dataviz.Figure??"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 61
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "named_result"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 69,
       "text": [
        "HeLa_Ago2_WT_rep1                0.183998\n",
        "HeLa_Ago2_PTB_kd_rep1            0.149841\n",
        "HEK293FlpIn_FMR1-iso1_WT_rep1    0.116535\n",
        "HEK293_IGF2BP1_WT_rep1           0.113316\n",
        "HEK293_IGF2BP3_WT_rep1           0.111378\n",
        "HeLa_eIF4AIII_WT_rep1            0.100907\n",
        "HEK293_IGF2BP2_WT_rep1           0.090244\n",
        "HEK293FlpIn_FMR1-iso7_WT_rep1    0.089293\n",
        "HEK293_Ago2_WT_rep2              0.077378\n",
        "HEK293_Ago2_WT_rep4              0.063740\n",
        "Name: SRR334701.peaks.bed, dtype: float64"
       ]
      }
     ],
     "prompt_number": 69
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "with dataviz.Figure(\"peak_distributions.svg\", figsize=(6,6)) as fig:\n",
      "    ax = fig.add_subplot(1,1,1)\n",
      "    named_result = result[[not item.startswith(\"SRR\") for item in result.index]]\n",
      "    sns.barplot(named_result.index, named_result.values, ax=ax)\n",
      "    ax.set_ylabel(\"Jaccard Index\")\n",
      "    ax.set_title(\"Jaccard Index of Top 10 overlapping RBPs\")\n",
      "    ax.set_xlabel(\"Top 10 Datasets\")\n",
      "    [tick.set_rotation(90) for tick in ax.get_xticklabels()]"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "display_data",
       "png": "iVBORw0KGgoAAAANSUhEUgAAAacAAAGrCAYAAABkLVjjAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAIABJREFUeJzs3XdUFOf+BvBnUUARY0FA0URjvAQFy6KgxoKioigoaGIn\nxkQxJhaumoi9x97AaCRer9HYrrFEQCMRJYkNUCn2gmIEZWkWFHUp8/uDH3NZKUty3ZlRns85nrNT\ndt/H2WW/OzPvvKMSBEEAERGRghjJHYCIiOhlLE5ERKQ4LE5ERKQ4LE5ERKQ4LE5ERKQ4LE5ERKQ4\nLE5kEJGRkXB1dTXIa/v4+GD//v0Gee3SREREoFOnTlCr1Th37pykbb+uDPkZKOrevXtQq9UGb4ek\nxeL0GnN1dUVUVJTcMf4WOzs73Lt3728/X6VSvcI0+m3YsAFjx45FTEwMWrduLc4v/GIs/GdnZ4eW\nLVuK06+qkAUFBaFv376wt7fHunXrii0PDg6Gq6srHB0dMX78eDx+/PiVtPs6sLGxQUxMjEFe29/f\nHw4ODuL7OWTIEFy+fFlcvm/fPjRt2hRqtRqtWrVChw4dsHr1anF5UlIS7OzsxOWdOnVCQECAQbK+\naVicXnNSf0mXJC8vT+4IBpeRkYEmTZoUm1/4xVj4DwA2bdokThctZP+LBg0aYPLkySXuidy4cQPT\np0/H4sWLERUVhVq1amHBggWvpN2/Kzc3V9b2XxWVSoXPP/8cMTExOHPmDBo2bIipU6fqrOPk5ISY\nmBjExsbi+++/x9atW3HkyBGddQqXBwYGYvPmzTh69KiU/43XEovTGyA2Nhb9+/dH69at0bZtW8ya\nNQtarVZcfvnyZfj4+MDR0RFt27bFihUrxGVbt25Ft27doFar4ebmhrNnzwIA9uzZg549e0KtVsPF\nxQXbtm0Tn1N4uGb16tXo0KEDZs2ahWfPnmHChAlQq9Xw8PDAhQsXyp3f398fc+fOxSeffAK1Wg0v\nLy/8+eef4vLw8HB07doVzs7O4pdu0YFNtm3bJu41DB8+XHxuUFAQBg4cKBbPHTt2wMPDQ2fbFLVl\nyxZ07twZjo6O+PTTT5GcnAwA6NOnD5KTkzF69Gg4OjqW+//18OFDTJgwAa1bt8YHH3yA1atXi7n3\n7duHYcOGYcaMGXBycoKrqysiIiJKfa3evXvDxcUF5ubmxZYFBwfD2dkZbdu2ReXKlTFu3Dj88ssv\nePHiRYmvdfr0aXh4eECtVsPT0xOnT58GABw6dAgDBgwotk3Gjh0LAHj+/DnmzJmD9u3bw8nJCVOn\nTsXz588BFP9MzJ49u9gPp/Xr18PV1RVqtRo9evTAoUOHxGX6toePjw9WrFgBLy8vqNVqfPrpp8jM\nzATw372TouuuXr0aH374IVq1agUfHx88ePBAXL5r1y506NABH3zwATZu3FjuIxCmpqZwd3fHnTt3\ndOYX/Sw2a9YMTZo00fn8FtWqVSs0adIEt27dws2bNzFw4ECo1Wq0bdsW06ZN05uhImFxegOYmJhg\nwYIFOHfuHIKDgxETEyMWk4cPH2LkyJFwd3fH2bNn8fvvv8PNzQ0AcODAAWzZsgUbNmwQn2NtbQ0A\nqFu3LrZs2YKYmBisXbsWK1eu1Ck4KSkpMDExwYkTJzB37lysXLkST548wYkTJ/DDDz8gODj4L+3V\n/fLLL5g+fTrOnj2LJk2aYNWqVQAAjUaDKVOmYM6cOYiKisJ7772H8+fPi68dHByM7du3Y+vWrTh/\n/jy6du2KCRMmAABGjx4NExMTbNiwAYmJiVizZg1WrFgBExOTYu0fP34cmzZtwpYtWxAVFYUmTZpg\n4sSJAIDQ0FDUq1cPmzZtwvnz58v9f5o9ezaMjY1x+vRp7N+/H0eOHMGOHTvE5bGxsXBwcEB0dDRm\nzZqFSZMmIT09vdyvX+jGjRuwtbUVp62srGBmZobExMRi66alpWHcuHGYOHEiYmJi8OWXX2LcuHHI\nyMhA165dcfv2bZ0v3+DgYHh6egIAFi1ahIcPH+Lo0aP4448/8OzZM6xcuVJc9+XPxMsjo9na2uKn\nn35CTEwMpk6dCn9/f6SlpZW5PTIyMsTloaGhCAwMRGRkJCwsLDBnzpxSt8kvv/yCwMBAnDlzBkDB\n3iwAXLp0CStWrMD333+PEydOIDs7G6mpqeX6rD59+hTBwcFo1apVqevExcXh5s2bJZ4DEwQB586d\nw82bN/H+++9j7dq16NatG2JiYnDy5EkMHz5cb4aKhMXpDdCsWTPY29sDKPhi+uijjxAdHQ2gYK+j\ncePGGDp0KIyMjGBqaooWLVoAAH766SeMHTtW/GKztrbG22+/DQDo1KkT6tWrB6Dg117Hjh3FvSqg\n4Ffk2LFjoVKpYGJigiNHjmDs2LGoVq0aLCws8MknnxT7ciqNSqVCr169YGtri0qVKsHT0xNXr14F\nUNARoXnz5ujSpQsAYOjQoWIBBYD//Oc/+Pzzz9GgQQMAwKeffop79+4hMTERKpUKS5cuxbZt2/DF\nF19g1KhROr+wiwoJCcHgwYPRuHFjVK5cGZMmTcK1a9dw9+7dcv0fXvb8+XMcO3YMkyZNgomJCayt\nrTFq1CiEhoaK61hbW2PIkCEAgK5du8Le3h7Hjx//y209e/YMZmZmOvPMzMzw9OnTYuseP34czZo1\nQ48ePQAAvXr1gp2dHY4dO4aqVauiW7duCAkJAQAkJibi9u3b6NatG7RaLX7++WdMnToV1apVQ5Uq\nVfD555/j8OHD4mu//Jl4Wffu3VG7dm3x8XvvvYfY2Ngyt8exY8fE5QMGDMDbb78NExMT+Pn54dix\nY6XuBX/44YeoV68eqlSpAnd3d/HzdOTIEfTs2RPNmjWDkZERvvzyS1SqVKnUbSsIAoKCguDk5ITW\nrVsjNjYWa9as0Vnn/PnzcHJygqOjIwYNGgQPDw+0adNGZx0nJyc4OTnB398f48aNg4uLC6pUqYK7\nd+8iKSkJlStXFv+GqQCL0xvg5s2b+Oyzz9CuXTu0adMGq1evxrNnzwAAqampYsF5WVpaWqnLwsPD\n0b9/f/GPKiIiAtnZ2eLyWrVqwcjovx+fzMxM1K1bV5wuWkDKo06dOuLjKlWqiF86GRkZOq8LQGda\no9Fg3rx5Yk5nZ2dotVrxkE/9+vXh7OyMe/fuYdiwYaW2n5mZKRbjwgy1a9f+W3syQMEea25uLmxs\nbMR5NjY2OnsCRdsr/H/9nfbMzMx03hsAyM7ORrVq1Yqtm56eXqxdGxsbcXt5eHiIBTQkJAQ9evSA\nqakpHjx4AK1Wi379+onb+uOPP9YpgC9/Jl62Z88eeHp6ok2bNnBycsL169d1nl/S9ihte1lbWyMv\nL0/ncF1RlpaW4uOin6fMzExYWVmJy0xMTFCrVq1SM6tUKvj6+iI6OhoREREwMzPDvn37dNZxdHRE\ndHQ0zp8/jzNnzuDu3btYvny5zjrR0dE4e/Ysfv31V4waNQoAMHnyZDx//hwDBw5E9+7ddfaqicXp\ntScIAmbPno0WLVrgt99+w9mzZzFlyhTk5+cDKPgjLu3Xv5WVVYnLsrOz8c9//hN+fn6IjIxEdHQ0\nXF1dy9wTql27NlJSUsTpoo//F3Xq1Cn2WkWnLS0tsXz5ckRHR4v/YmNjxXNDERERiI2NRbt27bB0\n6dJS27GwsNDpPfj8+XNkZmbCwsLib+WuWbMmKlWqJJ63Agp69hV9vZL+X0WLdHn94x//wPXr18Vp\njUaD7OxsNGrUqNi6lpaWuH//vs68e/fuiXs0H3zwATIzM3H16lWEhobCw8ND/P9UrlwZ4eHh4nY+\ne/ZsuXvJ3blzB9988w0WL16Ms2fPIjo6uthebEnbo+j2Kpo7JSUFlSpVKrOwlMTCwgKpqanitFar\nLbXAvaxu3bqYNWsWNmzYgCdPnpS4Ts2aNdGjR48yzx8Wfb0VK1bg1KlTWLJkCZYsWYJbt26VK0tF\nwOL0Bnj27BmqVKkCExMT3L17Fzt37hSXubq64vbt29i1axdyc3Px/PlzxMfHAwD69++P7777Djdu\n3ABQ8Ad/9+5d5OTkICcnB+bm5jAyMsLp06dx4sSJMjP07NkTGzZswNOnT5GRkaHTgUKfsoqei4sL\nLly4gN9++w0AsHPnTmg0GnH5wIEDsX79evEE9NOnTxEWFgag4FfyzJkzsWjRIixZsgTHjx8XX+dl\nvXv3xu7du3H79m3k5uZizZo1sLW1xTvvvFPu/0dRVapUgaurK9asWQOtVguNRoPNmzejT58+4jop\nKSnYvXs3gILDbZcuXULXrl1LfL3c3Fy8ePECubm54uPCHyCenp6Ijo5GVFQUcnNzsW7dOri7u8PU\n1LTY63Tp0gWXLl0Se4sdOXIEV65cEXsBGhsbo1evXli6dCkeP36MDh06ACg4ZNevXz8sXrxY7Kae\nmpqKkydPlmt7PHv2DIIgoHr16hAEAcHBwbh27ZrOOvq2x759+3D37l1otVoEBgbC1dW1xMOHQOmf\nqR49eiAsLAxXrlxBXl4eNmzY8Jd6mzo5OeG9997D9u3bS1z++PFjHD16tMQfBi8LDw8X95Rr1KgB\nIyOjMvc8K5rKcgeg/42RkRG+/vprzJo1C9999x2aNGmCbt26icfya9asiU2bNuGbb77B0qVLYWpq\nioEDB6JFixbw9vbGo0ePMGbMGDx48ABWVlb45ptv8Pbbb+Prr7/G2LFjkZ+fj/bt28PFxUWn3ZdP\nIE+aNAn+/v7o2LEjrKys0Ldv32KHP0p7fkknowvnWVtbY/ny5Zg7dy6ysrLQvXt3nR5z/fr1Q3Z2\nNkaNGoX09HSYmZmhXbt2cHNzw5w5c9C9e3d07twZQMEJ/RkzZiAkJAQ1atTQac/V1RV37tzBxx9/\njCdPnqBVq1b/8/Uo8+fPx+zZs9G+fXuYmJjgww8/1Dm06OjoiOvXr8PZ2Rnm5uZYuXJlqXtOM2fO\nxIEDB8Tp7777DkuWLIGXlxeaNGmCb775Bv7+/nj48CE6dOiAb775psTXsbS0RGBgIBYvXoyvvvoK\nDRo0wLfffquzh+Lh4YHhw4dj2LBhOl+Ws2fPxooVK9C7d29kZ2fD0tISgwcPFgtYWe+jnZ0dhgwZ\nAm9vb5iYmMDNza1YpwF926Nv376YPHkybty4AbVardPr9OW2X/58FU47ODhg0qRJ+OyzzwAU9Oyz\ntLT8S0Vh5MiRWLRoEUaOHAmVSoVz586J/5fKlSujXbt2mD17dqnZCsXExGDOnDl48uQJatasiSlT\nppSrqFUUKkPebPD333/HsmXLkJeXB29vb/j6+uos/+GHH7Bnzx4AgLm5ORYtWoT33nsPALB//358\n//33AABfX194eXkZKuZrq0OHDvj+++/RrFkzuaPQX7Rv3z7s37//L+1hvsn0bQ8fHx8MGDDglX8P\nPH/+HE5OTggNDf3be8lkGAbbh9RqtZg1axaCgoJw8OBBhISE6FxZDQDNmzfHvn37EBISggEDBojX\nsKSmpiIgIAD/+c9/sHv3bqxdu/Zvn5h+U509exZPnz5F48aN5Y5CJIlX9Tv66NGjeP78ObRaLVau\nXImGDRuyMCmQwYpTXFwcGjZsCBsbGxgbG8PNza3YSUJHR0fxmHGLFi3Eax5OnTqF9u3bw9zcHNWr\nV0e7du3KfWy7Ipg9ezb8/Pwwd+5cVKlSRe449DcUPdRE5dser2p7HTx4EB06dEDbtm1x5cqVYl3D\nSRkMds4pJSVFpzuxtbV1maMG7Nq1S7w4NCUlRae7p7W1tc5J8Ipu/vz5mD9/vtwx6H/g7e0Nb29v\nuWMohr7t8SoPf3Jsu9eDwYrTX/mVU3jIr/AD+Fd/IT1//hwXL16EpaVlmRfUERGRMuTl5SEtLQ0O\nDg4lHgEyWHGqW7euzt6ORqMpdpEdAERFRWHDhg3Ytm2beIjP2toakZGROs9t3759qW1dvHixzAss\niYhImbZv315sRA3AgMWpefPmSExMRHJyMiwtLREWFobFixfrrHP16lXMnDkT33//vXgRIFBwIeCa\nNWuQlZUFQRBw+vRpTJ48udS2Cq8G3759e7HRBIiISHlSUlIwbNgwndE8ijJYcTI1NcWCBQswZswY\n5OXlwcvLC/b29ggICEDz5s3RtWtXLFu2DE+fPhUH2LS2tsbGjRthZWWF8ePHY+DAgVCpVJg4cWKZ\nV84XHsqrW7euOMYaEREpX2mnYgx6nZNUkpKS0K1bN4SHh7M4ERG9BvR9b3OsDCIiUhwWJyIiUhwW\nJyIiUhwWJyIiUhwWJyIiUhwWJyIiUhwWJyIiUhwWJyIiUhwWJyIiUhwWJyIiUhwWJyIiUhwWJyIi\nUhyDjUpOREQFtFotEhMTJW+3UaNG4n3yXjcsTkREBpaYmAifaTtgVsNKsjazH6Vi2+KhsLW1lazN\nV4nFiYhIAmY1rGBeq77cMV4bPOdERESKw+JERESKw+JERESKw+JERESKw+JERESKw+JERESKw+JE\nRESKw+JERESKw+JERESKw+JERESKw+JERESKw+JERESKw+JERESKw+JERESKw+JERESKw+JERESK\nw+JERESKw+JERESKw+JERESKw+JERESKw+JERESKw+JERESKw+JERESKw+JERESKw+JERESKw+JE\nRESKw+JERESKw+JERESKY9Di9Pvvv8PDwwPu7u4ICgoqtjw6Ohre3t6wt7fH/v37xflJSUlQq9Xw\n8vKCl5cXFixYYMiYRESkMJUN9cJarRazZs3Czp07YWlpiQEDBqBjx45o1qyZuI6NjQ2WLFmCzZs3\nF3u+g4MDtm3bZqh4RESkYAYrTnFxcWjYsCFsbGwAAG5uboiIiNApTvXr1wcAGBnx6CIREf2XwapC\nSkoKrK2txWlra2ukpKSU+/lXr15Fv379MGTIEERFRRkiIhERKZTB9pxUKtXffq6VlRWOHz8Oc3Nz\nxMbGYuzYsQgLC0P16tVfYUIiehNptVokJiZK3m6jRo1gYmIiebtvKoMVp7p160Kj0YjTGo0G9erV\nK3X9osXMxMREfJNbtWqF+vXrIyEhAa1atTJUXCJ6QyQmJsJn2g6Y1bCSrM3sR6nYtngobG1tJWvz\nTWew4tS8eXMkJiYiOTkZlpaWCAsLw+LFi0tcVxAECIIgTj9+/Bjm5uYwMjLC7du3cffuXbzzzjuG\nikpEbxizGlYwr1Vf7hj0PzBYcTI1NcWCBQswZswY5OXlwcvLC/b29ggICICDgwNcXV0RExMDPz8/\nPH78GMePH8fatWsRERGBuLg4rFixAvn5+cjLy8OcOXNQu3ZtQ0UlIiKFMVhxAgAXFxe4uLjozJsw\nYYL4WK1W47fffiv2vE6dOqFTp06GjEZERArGPtxERKQ4LE5ERKQ4LE5ERKQ4LE5ERKQ4LE5ERKQ4\nLE5ERKQ4LE5ERKQ4LE5ERKQ4LE5ERKQ4LE5ERKQ4LE5ERKQ4LE5ERKQ4LE5ERKQ4f6s45efnv+oc\nREREIr3FacKECcjKyhKn7927h2HDhhk0FBERVWx6i1OLFi3w4YcfIjY2FmFhYRg+fDiGDx8uRTYi\nIqqg9N5scNSoUWjRogV8fHxQs2ZN7NixA2+//bYU2YiIqILSu+d0/fp1LFiwAEOGDEHjxo2xcuVK\nPHnyRIpsRERUQendc/r8888xb948dOrUCfn5+QgKCkL//v0RFhYmRT4iIqqA9Ban3bt3w9LSEgBg\nZGSEzz//HO3btzd4MCIiqrj0HtarWrUqlixZAj8/PwBAYmIi0tLSDB6MiIgqLr3Fafr06ahRowZu\n3LgBALC2tkZAQIDBgxERUcWltzjdunULY8eOhbGxMYCCPSmVSmXwYEREVHHpLU5GRrqrvHjxAjk5\nOQYLREREpLdDRNeuXbF06VJkZ2cjPDwc27dvR+/evaXIRkREFZTePSc/Pz80bNgQjRs3xp49e+Dm\n5oZx48ZJkY2IiCoovXtOKpUKgwcPxuDBg6XIQ0REVHpxmjZtWplPXLx48SsPQ0REBJRxWM/JyQlO\nTk4wMTFBQkICmjZtCjs7O9y6dQuVK+vd4SIiIvrbSq0y/fv3BwDs2bMH27dvF7uSDx06lLfMICIi\ng9K7C5Senq7TnVylUuHBgwcGDfW/0mq1SExMlLzdRo0awcTERPJ2iYjeNHqLk4uLC4YNG4Y+ffoA\nAA4fPowuXboYOtf/JDExET7TdsCshpVkbWY/SsW2xUNha2srWZtERG8qvcVpxowZOHz4MKKjo6FS\nqfDxxx+jV69eUmT7n5jVsIJ5rfpyxyAior+hXF3Je/fuzQtvqcLh4WEi+egtTjExMVizZg2Sk5OR\nn58PoKBghYeHGzwckZx4eJhIPnqL09SpUzFu3Di0atWq2Dh7RG86Hh4um1x7lwD3MN90eotTtWrV\n0LdvXymyENFrRo69S4B7mBWB3uLk6OiIqKgoODs7S5GHiF4z3LskQ9BbnI4dO4bt27ejTp064i40\nzzkREZEh6S1O27ZtkyIHERGRSG9xatCggRQ5iIiIRKUWJ1dX11KfxMN6RERkSKUWJ94Sg4iI5FJq\ncWrbtq2UOYiIiEQGvar2999/h4eHB9zd3REUFFRseXR0NLy9vWFvb4/9+/frLNu/f784bNKBAwcM\nGZOIiBTGYHcN1Gq1mDVrFnbu3AlLS0sMGDAAHTt2RLNmzcR1bGxssGTJEmzevFnnuampqQgICEBw\ncDAEQUDfvn3RsWNH1KlTx1BxiYhIQQxWnOLi4tCwYUPY2NgAANzc3BAREaFTnOrXL7hw7+VhkU6d\nOoX27dvD3NwcANCuXTucPHkS/fr1M1RcIqIK43UYdqrU4rR//36oVCoIggCVSlVsuZeXV5kvnJKS\nAmtra3Ha2toaFy5c0BsIADQaDays/jscirW1NTQaTbmeS0REZXsdhp0qtThFRkZCpVIhMzMT0dHR\n+OCDDwAAZ86cgaOjo97iVFJBIyIiZVD6sFOlFqclS5YAAHx9fXH48GFxL0ij0cDf31/vC9etW1dn\nb0ej0aBevXqlrl+0mFlbWyMyMlLnue3bt9fbJhERvRn09tZLTk4udnguNTVV7ws3b94ciYmJSE5O\nhlarRVhYGDp37lziuoIgQBAEcfqDDz7A6dOnkZWVhcePH+P06dPinhsREb359HaIqFu3LpYtWyZ2\nRggODkbdunX1vrCpqSkWLFiAMWPGIC8vD15eXrC3t0dAQAAcHBzg6uqKmJgY+Pn54fHjxzh+/DjW\nrl2LiIgIWFlZYfz48Rg4cCBUKhUmTpzInnpERBWI3uK0atUqrF69GpMnTwYAODs7Y9WqVeV6cRcX\nF7i4uOjMmzBhgvhYrVbjt99+K/G5AwYMwIABA8rVDhERvVnKLE55eXlYsmQJhzIiIiJJlXnOqVKl\nSkhISJAqCxEREYByHNZr3rw5pkyZAk9PT5iZmYnXPTk5OUmRj4iIKiC9xen69esAgE2bNunM500I\niYjIUHgnXCIiUpxyja334MED3L59G3l5eeI8HtYjkp5cY6KVdzw0oldFb3H66aefsH79ejx48ADv\nvvsurl69ihYtWmDXrl1S5HsjvA6DLNLrQY4x0f7KeGhEr4re4rRp0ybs378fI0aMwL59+xAXF1fs\nFhdUttdhkEV6fSh9TDSiV0FvcapSpQpq1KiB3NxcAEDLli1x584dgwd70/ALhYio/PQWJ2NjY+Tn\n5+O9997D+vXrYWVlhWfPnkmRjYiIKii9A79+/fXXyM7OxvTp03Hr1i2Eh4dj+fLlUmQjIqIKSu+e\nU2GvPHNzc6xYscLggYiIiPTuOQ0aNAhZWVni9KNHjzBkyBCDhiIioopNb3F68eIFqlevLk7XqFGD\n55yIiMig9BYnrVaLp0+fitNPnz7FixcvDBqKiIgqNr3nnD766CP4+PhgwIABEAQB+/btw6BBg6TI\nRkREFZTe4jRy5Eg0atQIf/zxB1QqFfz8/Eq93ToREdGroLc4PXjwAB06dEDXrl0BFBzme/jwIWrW\nrGnwcEREVDHpLU6fffYZdu7cKU7n5eVh1KhR+OmnnwwajAyLA4gSkZLpLU75+fkwNTUVp6tWrSoO\nZUSvLw4gSkRKprc45eXlIT09HXXq1AEApKWlsTi9ITjeHxEpld7iNHr0aAwcOBA9e/YEABw5cgST\nJk0yeDAiIqq49Banvn37wtbWFidPnoRKpcJ3333HwzJERGRQ5boTrp2dHezs7AydhYiICEA5itOd\nO3ewbNky3Lx5Ezk5OQAAlUqF8PBwg4cjIqKKSe/wRVOnTkWfPn1QuXJlrF+/Hl26dIGHh4cU2YiI\nqILSW5yePn2K3r17w8jICHZ2dpg9ezZOnDghRTYiIqqg9BYnY2NjAICZmRkSExPx5MkTZGRkGDwY\nERFVXHrPOXXo0AEPHz7E6NGj8dFHHyE/Px+ffPKJBNGIiKiiKrU4RUVFQaVSoVOnTrhx4wZq1KiB\n5cuXQ6vVok6dOjoX5hIREb1KpRYnf3//Up8kCALS09Mxbtw4jBkzxiDBiIio4iq1OB07dqzMJ6an\np2PgwIEsTkRE9MqV6yLcktSpUwejR49+lVmoguNI6URU6G8XJwAYMmTIq8pBxJHSiUj0PxUnoleN\nI6UTEVCO65yIiIikxuJERESKw+JERESKw+JERESKw+JERESKw+JERESKw+JERESKw+JERESKY/Di\n9Pvvv8PDwwPu7u4ICgoqtlyr1cLPzw8eHh4YPHgwkpOTAQCRkZFwdnaGl5cXvLy8sHHjRkNHJSIi\nhTDoCBFarRazZs3Czp07YWlpiQEDBqBjx45o1qyZuM6PP/6IGjVqICQkBKGhoVi4cCE2bNgAAOjW\nrRsWL15syIhERKRABt1ziouLQ8OGDWFjYwNjY2O4ubkhIiJCZ53ffvsNnp6eAAA3NzdERkZCEARD\nxiIiIoUzaHFKSUmBtbW1OG1tbY2UlJRS1zE2Nkb16tXF28D/8ccf8PT0xMiRI3Ht2jVDRiUiIgUx\n6GE9lUr1t5/r4OCA8PBwmJqa4ujRoxg7dqzee0wREdGbwaB7TnXr1oVGoxGnNRoN6tWrV2ydwr2p\nnJwcZGUdowLlAAAgAElEQVRloXbt2qhWrRpMTU0BAN27d8ezZ8+QlpZmyLhERKQQBi1OzZs3R2Ji\nIpKTk6HVahEWFobOnTvrrOPi4oLg4GAAwJEjR+Ds7AwjIyM8fPhQXOfs2bNQqVSwsLAwZFwiIlII\ngx7WMzU1xYIFCzBmzBjk5eXBy8sL9vb2CAgIgIODA1xdXTF8+HB89dVX8PDwgLm5OVasWAEAOH78\nOLZs2YK8vDxUrlwZq1evhpERL8siIqoIDH6zQRcXF7i4uOjMmzBhgvjYxMQEa9euLfY8b29veHt7\nGzoeEREpEHdFiIhIcViciIhIcViciIhIcViciIhIcViciIhIcViciIhIcViciIhIcViciIhIcVic\niIhIcViciIhIcViciIhIcViciIhIcViciIhIcViciIhIcViciIhIcViciIhIcViciIhIcViciIhI\ncViciIhIcViciIhIcViciIhIcViciIhIcViciIhIcViciIhIcViciIhIcViciIhIcViciIhIcVic\niIhIcViciIhIcViciIhIcViciIhIcViciIhIcViciIhIcViciIhIcViciIhIcViciIhIcViciIhI\ncViciIhIcViciIhIcViciIhIcViciIhIcViciIhIcViciIhIcQxanH7//Xd4eHjA3d0dQUFBxZZr\ntVr4+fnBw8MDgwcPRnJysrhs48aNcHd3h4eHB06cOGHImEREpDAGK05arRazZs1CUFAQDh48iJCQ\nEFy+fFlnnR9//BE1atRASEgIfHx8sHDhQgDAxYsXERoaiuDgYGzcuBEzZsxATk6OoaISEZHCGKw4\nxcXFoWHDhrCxsYGxsTHc3NwQERGhs85vv/0GT09PAICbmxsiIyORn5+PiIgI9OzZE5UrV0b9+vXx\nzjvvIC4uzlBRiYhIYQxWnFJSUmBtbS1OW1tbIyUlpdR1jI2NUb16dWRkZCA1NRVWVlbienXr1oVG\nozFUVCIiUpjKhnphlUol2evl5eUBgFj8NBoNstJuIff541eaoSzPstKh0WhgZmZWbJkceZSYqaw8\nSsyktDxKzMTPtv48SsykhPet8Pu68Pv7ZQYrTi/v7Wg0GtSrV6/YOikpKXj77beRk5ODrKws1K5d\nG9bW1sWeW3Qv7GVpaWkAgGHDhunMz3oV/5G/YNSo4DKXS50HUF4mfXkA5WVSWh5AeZn42eb7Vl4v\nZ0pLS0PDhg2LrWew4tS8eXMkJiYiOTkZlpaWCAsLw+LFi3XWcXFxQXBwMJycnHDkyBE4OzujUqVK\ncHFxwfTp0zFmzBhoNBokJiaiRYsWpbbl4OCA7du3w9LSEpUqVTLUf4mIiF6RvLw8pKWlwcHBocTl\nKkEQBEM1/ttvv2H58uXIy8uDl5cXxowZg4CAADg4OMDV1RVarRZfffUVEhISYG5ujhUrVqBBgwYA\ngO+++w4///wzKlWqhKlTp6JTp06GiklERApj0OJERET0d3CECCIiUhwWJyIiUhwWJyIiUhwWp9eI\nHBci5+Xl4eeff8a//vUv3Lp1S2fZ7t27Jc/zulHiyCbMpJ/S8gDKzGRILE6l6NKli9wRihk0aJDk\nbS5YsAB79+5Feno6Ro0ahS1btojLdu7cKXme0vTv31/uCCWaMGGC3BGKYSb9lJYHUF4mPz8/g76+\nwa5zeh3s378fKpUKRTssFk4/e/ZMlkzr1q0rdVlWlvSXzJ0+fRqhoaGoXLkyxo4diylTpiApKQkz\nZsyQPEshV1fXYu+bRqMR54eHh0uaZ9q0aaUue/xY2ivwCzGTfkrLAygv071790qcLwgCzp49a9C2\nK3RxmjlzJjw8PGBkpLsDKQgCtFqtLJk2btyITz/9FMbGxsUyyaFy5cqoXLngY/LWW29hw4YNmD9/\nPsaNGyfbNmratCmqVKmCL774AlWqVIEgCBg2bBi2bdsmy3b65ZdfMHXqVJiYmIjzCovnsWPHJM/D\nTK9nHiVmcnV1hY2NTYnLHj58aNjGhQrMy8tLuHr1aonLOnfuLHGaAgMHDhTi4+NLXCZHplGjRgnR\n0dHF5gcGBgrvv/++5HkKhYeHC0OHDhXCwsIEQRCErl27ypZl+PDhwtmzZ0tc1qlTJ4nTFGAm/ZSW\nRxCUl6l79+5CUlJSicsM/X1UoS/CjY6Oho2NDerXr19sWUxMDNRqteSZEhISUKtWLdSuXbvYspSU\nFNStW1fSPIX30Xp5T06uPEVlZ2cjMDAQt27dwuXLl/HHH3/IkuPhw4cwNTVF1apVZWm/JMykn9Ly\nAMrL9OOPP6J169Zo2rRpsWX//ve/MXLkSIO1XaGL0+vgwYMHAIBatWrJmkMQBBw6dAjnzp0DALRp\n0wa9e/eWNVOh69evIy4uDh999JHcUfDixQtcv34dKpUKtra2OodnmEm5mZSWR6mZpMTiBODOnTuY\nP38+bt68CQCwtbXFzJkzSxwpVyqxsbGYMWOG+IHMycnBokWL0LJlS1nyzJo1C1lZWfD09IQgCAgN\nDYW5uTkWLFggSx5AeQUzPDwcixYtQrNmzQAAly9fxsyZM+Hq6spMCs6ktDxKzPTixQv88MMPOn9r\nH3/8MUxNTQ3XqEEPGr4mvL29hYiICHE6IiJC8Pb2ljGRIPTp00fnfNi1a9eEPn36yJanV69e5Zon\npZkzZwoTJ04Ujh49Kvz666+Cn5+fMHPmTNnyuLm5CampqeJ0amqq4ObmJlseQWCm1zGPICgv09ix\nY4WlS5cKV65cEa5cuSIsW7ZM+OKLLwzaZoXurVdIEAS4uLiI0y4uLli7dq2MiQATExO8//774rSt\nra1hf6XoYW1tjczMTPFcWGZmZqm9eKRy9uxZHD58WJzu3r073N3dZctjbm4OS0tLcdrS0hJvvfWW\nbHkAZnod8wDKy3T37l2sX79enLazs4Onp6dB26w0d+7cuQZt4TWQnJyMu3fv4t1334VWq8WBAwdQ\nu3ZttGvXTrZMt2/fRlRUFCwsLJCeno5t27ahXr16ePfdd5GVlYXq1atLmickJAQrVqzAhQsXcPjw\nYSxduhQ2NjY4deoUwsPD0b17d0nzAMDRo0fRuXNn8eRxZmYmzpw5g379+kmeBQCuXbuGgwcPwsTE\nBLdv38a3336LRo0aoWrVqrh3716JHW+YSf5MSsujxEwnTpyAnZ2d+OM0ISEB165dM+iPQZ5zAso8\njivHRZ0A4OPjU+bybdu2SZSkQGRkJADoXPxa+FilUsHZ2VnSPAAwatQoxMXFiT8iIiMj0apVK1hY\nWABAsZtbGpq/vz+Agu0CQNw2haTOw0yvZx4lZvrwww9x5coV8WjOtWvXYG9vD1NTU6hUKmzduvWV\nt8niROV2+fJlJCUlwc3NDZmZmXj8+DEaNWokWx4lFkyiN1FSUhKAkv/WAIg3iX2VOLYeCrprL1y4\nUBwrKiEhATt27JA1059//omxY8di8ODBAIAbN27Ieh5sw4YNCAgIwMqVKwEA+fn5+Oqrr2TLAwBt\n27ZF9erV8fDhQ7Rt2xb/+Mc/YGVlhbZt28pSmOLj4/HRRx+Je+LXrl3D9OnTJc/BTK93HiVmatCg\nAa5cuYKjR4+iQYMGMDIyQmpqKho0aGCQwgSwOAEApkyZgtatWyMhIQEA0LBhQ9mLk7+/P8aOHYvn\nz58DAJo0aYKwsDDZ8oSGhmL9+vXi+Z06derINnxRIaUVzHnz5iEwMFA8cf3+++8jPj5etjzM9Hrm\nUWKm+fPn4/Tp09i1axcAoHr16pg3b55B22RxApCamgp3d3dUqlQJQMF4ci+Ptye1rKwstGjRQpxW\nqVSyZhIEQaf97Oxs2YuT0gqmVqstNmKG3EfNmUk/peUBlJcpOjoas2fPFnsMV69e3eB52JUcgKmp\nKdLT08Xp8+fPy9ptGygYZLVwTw4o+CKuWbOmbHnc3d3x9ddf49GjR9i6dSsOHDgAb29v2fIAyiuY\nVlZWOHnyJADgyZMn2LFjB9555x3Z8jDT65lHiZkEQUBubq44nZmZqTNtCOwQgYJiNGfOHCQlJaFR\no0bIyMjAunXrdPZcpJaQkAB/f39cuXIF1apVQ/369bF69WpZRq0QBAG3bt1CYmKi+AfTsWNHWa+g\nBwpuL/Lnn38iOjoaI0eOxIEDB9CrVy/4+vrKkic9PR1z587V2Ubz5s0rcZxEZlJOJqXlUWKm3bt3\n49ChQ0hISECfPn3w66+/Yvz48Yb9gWrQS3xfA3l5ecKGDRsErVYrxMfHC3FxccKLFy9kzzR//nxB\nEAQhIyNDyMjIkDWPIAhC37595Y6gIz8/X7h586Zw9OhRYd68ecK8efOE8PBw2fLk5uYK48aNk639\nkjCTfkrLIwjKy5SXlyecOHFCuHLlirBp0yZh06ZNwpUrVwzeboU/rGdkZITw8HB8/vnnaN68udxx\nABRkunjxIgRBkPXXW1F2dnY4e/Ys2rRpI3cUAAXn4CZNmoSff/4Z3bp1kzsOKlWqhIyMDGi1WsUM\n0MlMr18eQHmZjIyMsHLlSuzbtw92dnaStVvhixNQcPO66dOno2fPnuLJdZVKBScnJ9kyNWrUCJ99\n9hm6d++uk8nLy0uWPJGRkfj5559Rq1YtnTxyXKBcSGkFs06dOvD29oaLiwvMzMzE+ePGjWMmBWdS\nWh4lZmrbtq142LxKlSqStMnihIKeMSqVCr/88ovOfDmLU+H1A5mZmbJlKCoiIqLM5ZcuXYK9vb00\nYf6f0gpmly5d0KVLF50r+eXGTPopLQ+gvExHjhzBkSNHEBAQIM4z+N+awQ8cvgECAgLkjlCMnKNv\nl6Rfv35yRyjm4sWLckfQ4evrK3eEYphJP6XlEQTlZTp27Ngrf01e51QOch66Ks2FCxfkjqB4M2bM\nkDuCDo1GI3eEYphJP6XlAZSXyRCj17A4ERGR4rA4ERGR4rA4UZnS0tLkjkBEFVCFLk67d+8u13pS\njoQQGBhYrvUKRys3tNGjR5drvQMHDhg4yX8prWDOmjWrXOuNHz/ewEn+i5lenYq8jS5dulSu9ays\nrF552xV6+CIvLy9Jv1TLQ2mZlJYHUF4mpeUBmKk8lHiTUaVtIznz8DonhcnJycG9e/dKXW5jYyNh\nGuD+/fuYNm1aqcvluEuo0mRnZyM6OrrY3UoLyXG9HDPpd+zYMQDA6tWrYWVlhX79+kEQBAQHB4s3\n15Oa0raRnCr0nlNZQ3GoVCpcuXJFwjQFHBwcytxFLvyDkkrXrl0xYcKEYn8shdNyjEzetm3bMn/1\nSl0w1Wo1HBwcSl2+bds2CdMUYKby69+/P/bt26czz9vbG/v375c8i9K2UVl5DHV79kIVes/Jzs5O\nUbvQQMFNBZWUqUaNGrLfGuNlZmZmcHZ2LrVgSq1hw4ayfbGWhpn+mhMnTqBjx44AgJMnT8o2MoPS\ntpGlpSXGjx9f4r2bDL2NKnRxoteTEgsmvb5WrlyJhQsXihdtN2nSBKtWrZI5lTIU/hCUQ4UuTnLf\nj6gkvXr1kjuCjjZt2uDQoUNo3bo1rK2t5Y6jSD169JBtr600zFR+1tbW+Ne//qUzLzU1VZYsSttG\nhuiFV16V5s6dO1e21mU2fvx4REVFITk5GXl5ebCwsICxsbGsmcaNG4e9e/fi6tWrePDgAapWrYpa\ntWrJlufOnTs4ffo01q1bhy1btiA+Ph5paWmoXLkyLCwsZPkjunnzJp48eYK33noL5ubmkrf/siVL\nlmDlypU4deoU/vzzT+Tk5KB27dqy3u6Amcpv1KhR6N27NypXLvitnpSUBF9fXwwdOlTyLErbRqGh\nobh//z6AgpHSK1WqJFnbFbpDRFZWFmJjYxETE4OYmBhcvnwZ9evXh6OjIxwdHdG7d29Zct26dUvM\nFBsbi4yMDLRq1QpqtVq2u7wCBeN5FeYKDw9HZmYmzp8/L3mObdu2ie+bIAhQq9Xie2ZnZ6dz63ap\nZGdnIz4+XnzP4uPjYWlpCbVajXnz5kmeh5nKLzQ0FPv27cOGDRuQlJSECRMmYN68eWjdurUseZS0\njY4dOybmuHr1Kho3bgy1Wo3WrVtDrVajTp06Bmu7Qhenl2VnZ2Pv3r344YcfkJSUhKtXr8odCX/+\n+SciIiKwdetWaDQaWQZ8FQQB165dw/nz5xETE4OEhATUqlULarVa1nveAMopmIWePn2K+Ph4nDt3\nDgcOHEB+fr7kPSyZ6a/bu3cvgoODkZGRgSVLlkh++5eSKG0b5eXl4fLly4iKisKuXbuQlJRk0B7N\nFbo4aTQa8Qu38M6z9vb2UKvVaNmyJRo0aCB5pnPnzom/VO7fv4+3334brVq1QqtWrdCsWTPJd+9H\njhyJJ0+eoGnTpmjRogXUajUaN24s+zFxJRXMgwcPIiYmBleuXIGJiQmaN28uvmeWlpaSZmGm8lu3\nbp3OdFhYGBo3bowmTZoAkOfGfkrbRgDEH3wxMTGIj4/HixcvYGdnB7VabdCOSRW6ONnZ2aFZs2YY\nMWIE3N3dZT/2/XKmHj166NwFUw6zZ8/G1atXUaVKFbRs2RJqtRqtWrWS9fbxSiuYarUa7777LgYP\nHow2bdqgcePGsuRgpr9m3759ZV6KIEePUKVtIzc3N5ibm6Nnz55o0aIFWrRogWrVqknSdoUuTkXP\n69y9exf169cXv3ybN28uS7FKTU3VOdacm5sLe3t78ZzT22+/LXkm4L/n5wr/PXjwAE2aNMGyZcsk\nz6K0gpmbm4urV6+K58Fu3bolniNo1aoV2rdvz0wKzVQeY8aMwcaNGyVpS2nbaOPGjYiJiUFqaioa\nNWok5mjWrJnBO0dU6OL0sqSkJBw/fhxbt25FSkqKIm7o9+zZM53zYHKMWgEU3Mo+Pj4e58+fx/nz\n5xEXFwcLCwuEhITIkgdQVsEsKj09HYcPH8YPP/yA5ORk2d4zZno15BxfTknb6NatW2LRPHfuHGrV\nqoXt27cbrL0KfZ0TACQkJIjHU2NiYpCVlYWWLVtKNur3y7KysnTyXLlyBQ0bNkTXrl3h6OgoeZ5v\nvvkGMTExSExMRNOmTaFWqzFkyBAsW7YMb731luR5ijI1NUXVqlVhamoKY2Nj3L9/H1qtVvIcV69e\n1XnPcnJyoFar4ePjA7VaLXkeZnp9KXUb3b17F/Hx8YiLi0NcXBwyMjIMfhSnQu85tW3bFlZWVmJX\nZLVajYYNG8qaqV27duIhPEdHRzg4OKBq1aqy5fnhhx/QunVr2NnZideBlOTSpUuS9XAqqWAWvn9y\nFEwvLy+xa61arUb9+vUlz8BMhiPlnpPSttGXX36JuLg4VKtWTbxcw9HRUZpzvEIF9ujRo3KtFxAQ\nYOAk/5Wfn1+u9WbOnGngJH9Nv379JGtry5YtwoULF4ScnJwy17t48aJEicrH19dX7gjFMJN+Un62\ny0uqbfTrr78KGRkZetc7duzYK2+7Qt9ssLy/sqW8r0t5f40o4XyYXEaMGAEHB4cy9+QAiGOlKYVG\no5E7QjHMpJ/Sbn4ISLeNunfvXq6ORmvXrn3lbVfo4kREFdelS5fwxRdfYPbs2bh79y5GjhyJFi1a\noFevXjh37py4Xrdu3WRMWXGxOBFRhTRjxgy4uLjgnXfewZAhQzBw4EBERUVh9uzZWLBggdzxKrwK\n31uPiCqmvLw8DBo0CADwn//8B+7u7gCADz74AHl5eXJGI3DPqVyUeGsNKbu6a7VaZGZmFpuflpYm\nPlbSDRKVSonnLipypqIXkb48EouUo2//HUp73wxxa40K3ZW8LEuXLsXUqVMlb3fcuHHo2bMnunfv\nLmsX8kJHjx7FtGnTUKlSJbz33ntYvHgx3nnnHQDyXpyo1Wrx5MmTYidr09LSZBuDrCSDBw/Grl27\nJG83LS0Na9asQVpaGtzc3PDhhx+Ky2bNmiXLYavz589j4cKFsLKywvTp0zFjxgzExcXB1tYWy5Yt\nk3yoHgcHB/FLNTU1VecLNjU1FRcvXpQ0D6C89y0qKqrMTlpOTk4Ga5uH9Upx6NAhWYpTbGws8vLy\nsGDBAnTq1Al9+/ZFx44dZfslt3r1auzevRuNGzfG4cOH8dlnn2Hp0qWyXBBcqKyCOXr0aMkLZml/\nwIIg4M6dO5JmKTR16lS0aNEC3bp1w65du3D8+HGsWrUKpqamsvX0nD9/PiZMmIDs7GwMGTIEkyZN\nQlBQEMLDwzFnzhzJb09+5MiREm8/LielvW+BgYFlLjfke1ahi5OdnV2py+QaRLROnTrYsGEDHj16\nhCNHjmDTpk2YNm0aevbsCQ8PD8nvMWNkZCT+onV3d4etrS3Gjx+PiRMnSpqjKKUVzBEjRqBNmzYl\nLnv69KnEaQpoNBr4+fkBKDgsHRQUhBEjRmD9+vWy5AEKzvEUHiJfu3YtBgwYAADw8PDAhg0bJM8z\nf/58ycbMKy+lvW9S/2AoqkIXp7p162LPnj3FDgMJgoAuXbrIE+r/1ahRAwMHDsTAgQOh0Whw6NAh\nLF68GD/99JOkOYyNjZGRkQELCwsAwHvvvYcffvgBvr6+SExMlDRLIaUVzMaNG2P+/Pl49913iy1z\ncXGRIVGB/Px88caLvr6+sLGxgY+Pj2wFs+gPvs8++0xnmRw3iFTa9VSFlPS+RUdHl7mch/UMpF+/\nfrh//36x4qRSqdCrVy+ZUhVnbW2NkSNHYuTIkZK3/fXXXyMzM1MsTgBgaWmJbdu2yfarSmkFc9y4\nccjPzy9xmb+/v8RpCnTt2hVnz56Fs7OzOM/DwwO1atXC/PnzZck0ePBgZGdnw8zMTKdDz59//okW\nLVpInufRo0c4cOBAiYf2VCoVvLy8JM+ktPctICCgzOWG/A5ghwgqN61Wi2vXrkGlUsHW1la2+1+d\nOXMGFhYW+Mc//qEz/8mTJ9i2bRvGjh0rSy56vbRt27bMnriLFy+WMA29jMXp/x06dAhnz54FULCr\nWnjNg5yUlOmPP/7A119/jXr16gEA7t+/j+XLl6Njx46yZQKUUTALL9xMSkoSO2g0a9ZM8hzM9NfI\n2dtUn4cPH2L16tXiSBVOTk7w8/NDjRo1JM0h52E9FicAixYtwqVLl+Du7g5BEBAWFoamTZvKOjab\n0jJ5eHhgzZo14i2sExISMHHiRFnv56SUgunp6Ql/f3+0adMGx44dw48//mjQ+9ww06uh5OI0atQo\nNG3aFB4eHhAEAYcOHcLly5exadMmSXP4+PiUudygh/Zf+VCyr6FevXoJubm54nRubq7Qs2dPGRMp\nL5O7u3uxeb1795YhyX/16dNHuHHjhjh98+ZNoU+fPpLn8PT0LHNaDsyk34ULF4TY2Fjh4MGDwuXL\nl2XN8jIPD49yzZPTixcvDPr6FbpDRFGPHj0SL+jMysqSrSt5UUrKZGdnhylTpqBPnz7intz7778v\nWx6goFdT4Z4cUNAxQpDhQEBWVpbOifWi03KdWGcm/Y4cOYKwsDDY29tj1apVGDFiBD755BNJM5TG\nyMgIJ0+eRIcOHQAUnGfVNwq/FARBwOnTpxESEoKIiAicOnXKYG3xsB6AvXv3Yvny5XB2doYgCDh3\n7hwmT54sXofBTMDz58+xZcsWnD9/HgDQpk0bjBgxAqamprLkAYBJkybByMhIp2BqtVqsWrVK0hz+\n/v46PxwKv2wLyXFinZn0c3Nzw8GDB1GlShU8ePAAH3/8MYKDgyXNUJr4+Hh89dVXePjwIQCgVq1a\nWLZsmSy9GgEgJiYGoaGhOHr0KB49eoRZs2bB1dUVNWvWNFibLE7/7969e4iNjYVKpYJarUbdunXl\njqTITEqixIJJrw9vb2/s379fnFbiOaiMjAyoVKpy3VPJEFauXImwsDA0aNAA7u7u6N69O/r3749j\nx44ZvG0WJwCnT59Gy5YtYWZmhr179+LKlSvw8fGR9ZbtSsvk6uoKlUqlc9jM1NQU9vb2mDx5stgp\noSLSarX4/fffYWFhAbVajYMHD+LcuXOoX78+hg8fXmxQUWZSRia1Wg0HBwdx+uLFi+K0SqXC1q1b\nJc1T1Lp164rNMzExgYODAz744APJcrRr1w729vYYMmQIOnfuDBMTE7i6urI4ScXDwwMhISGIj4/H\nwoULMWLECOzcuRM//vgjM/2/gIAAWFpawtPTEwAQEhKCP//8E02bNsWuXbtk6XWllII5ZcoU3Lt3\nDy9evECDBg2Qk5ODTp064fz583jx4oXeCxmZSZ5MkZGRpS5TqVQ6F8JKbfbs2bh586Y4GMAvv/yC\nd955Bw8ePEDjxo0lG/czNzcXJ0+exKFDh3DmzBk4Ozvj1KlTiIiIgLGxsWEbN2h3i9dEv379BEEQ\nhLVr1wo//fSTIAiC4O3tLWckxWUqqe3+/fsLglByTz4prF27VtixY4eQlZUlZGVlCTt37hSWLl0q\nHDx4UBg6dKhkOXr06CHk5+cLz549E5ydnYWcnBxxmZubm2Q5mMkwfH19JW9z8ODBOr11c3JyhEGD\nBglarVbo3r275HkEQRCeP38uHD58WBg3bpzQvn17YdKkSQZtj/dzAlCtWjV8//33CA0NhYuLC/Ly\n8pCTk8NMReTn5+v0zDl9+rR4Qza5RoqIiIjAkCFDYG5uDnNzcwwePBiRkZHw9PTEgwcPJMthZmYG\nlUqFKlWqoF69ejq9quS67QkzvTpyjMGXkZGh8/eem5uLzMxMGBsbw9zcXPI8QMFRiV69eiEwMBBh\nYWEGv55Q/r6JCrBy5UoEBwdj0aJFqFOnDjQaDXx9fZnppTwLFy4Ue1w1btwYK1euxPPnzzF58mRZ\nMhUWzMJj8HIVzPT0dPEcQdHHhdNyYKbX24gRI9C/f3+4uLhAEAT88ccf+Pjjj/H8+XO0bNlS1mxf\nfPEF1q9fD29vb4O2w3NO9LfFx8fL1rUVKBilYuHChUhISBAL5syZM1G/fn1ER0ejU6dOkuQIDAwU\nu0QLL3WPBgoGhpUaM706cvXiS05ORlxcHFQqFVq2bAkbGxvJM/j4+BQ7rxsXF4eWLVsavNNIhS5O\nvva2nGgAAB2+SURBVL6+CAoKKnHwR5VKhfDwcGZ6yc2bNxEcHIxDhw6hevXq2Ldvn6x5XiZ3waQ3\nj9xdzO/cuYOQkBAcOnQIoaGhkrY9YMAANGrUCB999BGMjIwgCAImT56MVatWQRAEtG3b1mBtV+ji\nROVz9+5dHDp0CCEhITA2NkZycjL27t2LBg0ayB0NgPwFMy8vDyEhIUhPT0fXrl11bje+e/duDBo0\nSNI8zPRqhYeHo1u3bpK2WXgPt5CQEFy/fh2+vr5wc3OTfFSW/Px8/Pjjjzh+/DgmT54MBwcHybqS\ns0PES/TdllgOcmYaNGgQJk6cCJVKhfXr12Pfvn2oVq2a7IXp7t272LhxIzw9PfH1119j165d+Pe/\n/y3LntyCBQuwd+9epKWlYdSoUdiyZYu4bOfOnZLnYabySUtLw4wZM+Dr61vsJp6zZs0SH0tZmHbt\n2gUfHx+MHDkSjx8/xjfffANLS0uMHz9eluHCjIyM8PHHH2PJkiXYvHkzZs+eLZ7XNXjbkrTyGpH7\nsFlJ5MxkYWGBrKwspKenIyMjQ7YcRSmtYJ4+fRqbN2+Gv78/Dhw4gFOnTmHhwoWyjPPHTOU3depU\nWFpaYvDgwQgLC8OXX36JFy9eAAAuXLggS6YFCxagWrVqWLVqFSZOnCj7+JWFrK2tsWrVKnGECCmw\nOL1EiUc55cy0fv167N27F7a2tggICEC3bt3w+PFjxMXFyZZJaQWzcuXKYrfot956Cxs2bEBOTg7G\njRsHrVbLTArNpNFo4OfnB1dXVwQFBaFly5YYMWIEMjMzJc9S6I8//kDnzp2xcOFCuLu7Y82aNcjN\nzZUtz88//yw+1mg06Ny5MyZOnChN4wa9ioreOGlpacLWrVuFQYMGCZ07d5Ytx6NHj4Q9e/YII0eO\nFFxdXYXWrVsLsbGxsmQZNWqUEB0dXWx+YGCg8P7778uQiJnKo3fv3kJeXp7OvODgYKF3796Ci4uL\n5Hledu/ePWHTpk2Ct7e30LNnT2HlypWSZygcDODlx1JgcSrFkiVL5I5QjNIy3b17V3w8c+ZM2XLI\nXTC1Wq2g1WpLXHb//n2J0xRgJv2WL18uREZGFpt/4sQJxY1YcevWLSEwMFCcPnbsmCTtsjgpkJx7\nBaVRYqZCUn9wSyNHwQwKChIfx8XFSdKmPsz0ZpPq761Hjx5CVFSUEBkZqfM4KipKiIqKMmjbFXqE\nCDs7u1KXyXVjPyVmep0U7RQh1Unt0NBQjB49GkDBgJ1KuO0CM5Xfw4cPsXr1apw7dw4A4OTkBD8/\nP9SoUUPmZPKztrYWB+Qt+riQIW/TXqGLU926dbFnzx5YWlrqzBcEAV26dGEmogpgypQpaNq0KVau\nXAlBEHDo0CFMnjwZmzZtkjua7AxZfPSp0MWpX79+uH//frFCoFKpxKHqmYn0efTokXi78aKPAch2\nS3RmKj+NRqNTiOzs7MRbw1R00dHRZS53cnIyWNsVujj985//LHXZtGnTJEzyX0rMVB6DBw+WO4Js\n2rZtK94bqOjjQnJ86TJT+RkZGeHkyZPo0KEDAODMmTM6I6YrkZWVlSTt6LvHliH3rDh8UQnu37+P\njRs3Yu7cuXJHEcmVKSEhAWvWrIG1tTXGjx+PJUuWIDY2Fra2tpg5c2axPTwl2bVrV4UumlQ+8fHx\n+Oqrr/Dw4UMAQK1atbBs2TLZx2i8ePEibt26hby8PPF8s1wFXA4VujhdvXoVy5cvR3JyMt5//334\n+/tj8+bNOHbsGHx8fPDJJ59U+EyDBw9Gnz598OzZM+zduxd9+vRB7969ER4ejpiYGHz33XeS5gGU\nVzDT09Oh0Whgb28PANi8eTOys7MBAL1799YZQ46ZlJWpqIyMDKhUKtSuXVvWHACwYsUK3L59G3Fx\ncfD09MSRI0fg6OiIFStWyJLnwYMH+Pbbb5Geno41a9YgISEBkZGRGDp0qMHarNAjREybNg19+vTB\nt99+C0dHR/Tt2xfm5uY4fPiwLIVJiZmePHkCHx8f+Pr64sWLF5gwYQKaNGmCMWPG4M6dO5LnAYAZ\nM2bA2dkZVlZWGDhwIOrVq4fAwEA0a9ZMZ0w0qSxYsABJSUni9K5duwAAz58/x5o1ayTPw0x/zbp1\n67Bu3Trs3LkTO3bswLp16xAUFKRzc02pHT16FIGBgbCwsMDUqVNx4MAB3Lt3T7Y8U6ZMQevWrZGQ\nkAAAaNiwIXbs2PF/7d17UFTXHQfw70UBpYYafMWGMdMoAnFsNelC/wBWBV10NwhSceKwPGIpIT4y\ndnxEkRjDqKgZK0FxYmrpaCCgbRKVLIj4SCxJDJOaqkQIkgShEsGlVnkuK6d/EK4sb6N7z6H395lx\nZu9e5H59wI9z7zm/Y9drin1j1c6sVqvcJ2ry5Mk4fPiwcq05hkimYcOGya+7by7G6758Z8EEgKys\nLKxatQoAMGXKFMyfP1/xPOXl5dDpdPLxyJEj5b2JwsPDFc9DmR5MbW0trl27Jk84ys/Px6RJk/Dl\nl1/i/PnzWL9+veKZXFxc4ODQMXZoaWmBq6sr11ZdtbW1mD9/Pt5++20AHV/7nfnsRdXFqbW1VZ6N\nwhiDg4MDiouL5Y3Q7DkTZahkmjVrFpqamuDi4mJTJK9fv67YQ9nuRCuYjo6ONsd/+ctf5Ne8+qJR\npsErLy/Hu+++K/+/Wrp0KSIjI3H48GEsWLCAS3Hy8PDA3bt3ERYWhhdeeAGjRo3iugOus7OzzW7F\n//znP+Hs7GzXa6q6OHVfVNb9mMccf9Ey9TV7cNKkSTh48KCiWTqJVjCdnJxw48YNeafSMWPGAABu\n3Lhh9y9gyvTwzGYz2tra5OJktVpRX18PR0dHjBo1ikumHTt2AABiYmIQEBCA5uZmrs/kNm7ciNjY\nWFRXVyMsLAxmsxl79+616zVVPSGCDE59fT2KiopQU1MDoON2o7+/P5ycnDgnE8O5c+ewY8cOrFy5\nEt7e3pAkCV9//TX27duHdevWQavVUiZBMwFAZmYmMjMzodVqwRjD+fPn8cILL+B3v/sdUlJShJm1\nO2vWLJw7d47b9S0WC8rKysAYg5eXl83X/9mzZzF79uxHej1VFyeeC8z6Ilomk8mEtLQ0eHh44OLF\ni5g5cyYkSUJpaSlSU1P7bbdkT6IVzC+//BKHDh1CeXk5AGDq1KmIjo7GzJkzueShTA/m3//+N/71\nr39BkiT8+te/lkd3ItFqtfj44495x+iVPbayV3Vx8vLygre3d59Ddx639UTLZDAYkJ2djVGjRuHW\nrVtYvXo1Dh8+jLKyMrz22mvIyclRNA8gbsEkQ19lZSVyc3NhMpnw0Ucf8Y5jQ23FSdXPnDZu3IiT\nJ0/i8ccfR0hICLRabY+HtmrP1NbWhpEjRwLomEHUuRGbp6cnWlpauGRKT0/H0aNHhSmYA3Xu2L59\nu0JJ7qNMg3fz5k2YTCbk5ubim2++wR/+8Afs3r2bS5b+nuM0NDQomIQ/VRenqKgoREVF4fvvv8eJ\nEydw4MABeHt7Y8uWLZTpRwEBAYiMjMRzzz2HTz/9FEFBQQA6Ojm3t7dzySRawdRoNJAkqdcdi3l1\nkqdMA8vOzsZHH30Es9kMnU6Hbdu2ISEhAStXrlQ8S6fOWbm9HcfGxvKKxYWqi1OncePGwd3dHY89\n9pjNIkGeRMm0ceNGFBQU4Ntvv8Xy5csRGBgIABg9ejROnDjBJZNoBbNzXdpAkpKSkJycbOc0HSjT\nwJKTk+Hv74/ExERhbgXzLIwPwx6zZFX9zOns2bM4ceIEbt68iblz50Kv13PvFSdiJtEwxuSCOXXq\nVLlgis4e9+Uflpoz1dfXIz8/HyaTSR49vf/++/jkk0/sfu2BXLp0CcnJyTCbzThz5gxKS0tx6NAh\nbNu2jVsmpXv9qXrklJCQgOnTp2PKlCkoKytDWVmZzXke98BFy1RRUYHk5GTU1dVh3rx5SEhIkGfE\nxcfHyyvGlSRJkk2nAUJ+Cjc3NyxduhRLly5FTU0NTCYTxo4di+DgYMybNw9//OMfuWXbsmUL9u3b\nh5deeglAx0SpS5cuccvTV68/Kk52sm3bth73urvuL8ODaJkSExMRHh6OX/3qV8jOzobRaMT+/fvh\n5uaGmzdvKp4HELNgkqFt4sSJWLZsGZYtW4bvvvvOZqaePdbwDMRiseCJJ56weY/nTa7CwkKYTCaE\nhYVh/fr1SEhIkAunvai6OM2dOxePPfYY7xg2RMvU0NCAxYsXAwA2b96M3NxcGI1GpKenc8skYsEk\n/z9++ctfyj3/ACA1NVXx4jRhwgQUFRUB6PgazMrKwqRJkxTN0BWPXn+q7koeFRUlv964cSPHJPeJ\nmMliscivDQYDNm/ejLi4ONTV1XHJ01kwPT09sXnzZhiNRhiNRm5d0gdLxL2lKJOYtm/fjvfeew+V\nlZXw9/fH5cuXsXXrVm55uvf6MxqNdu/1p+qRU9dh8tdff80xyX2iZVq8eDGuXLmCZ599Vn7Px8cH\nb731Fnbu3Mktl8VikW/lGQwGjB8/HnFxcWhsbFQ8y2D3l1Lymy5lGtrGjRtn9951D4JHrz9VFycy\nMIPBIDfo7MrLywsJCQkcEolXMBMTE+UNGSMiIqDX67Fs2TKcPn0aSUlJXDZkpExD00AFqevtRl46\ni5K9e/2pujhdu3YNc+bMAdCxX0nna6Bj8sHp06dVnyk6OhoZGRk9prMXFhYiOTmZSzsV0QqmaPtL\nUaZHS8lO952LbhljyMnJEXoUae8JGqouTvn5+bwj9CBapvj4eERFRSEjI0OePXTkyBEcOHAAGRkZ\nXDKJVjBF218KoEwPqr81PAcOHFAsR9dFuKdPnxZipMSLqouTu7v7oD4uLCwMH3zwgZ3TdBAt0/PP\nPw8nJydER0fj4MGDOH78OPLy8pCZmYkJEybY/fq9Ea1gira/FGV6MDzW8AwVPHv9qbpDxGCpeRV9\npzNnzmDDhg3w8PBAeno6XF1dFbt2b06ePIndu3fbFMw///nP3AomGbqCg4PlNTzHjh3DnTt38NJL\nLyErK4trLhG+76SlpfXZ6w+w7zMwVY+cyMC6PvNqaWlBZWWl/BMlr+dyAKDT6eDo6Ijw8HB4eHgg\nMzOTa8EUbX8pyjR4PNbw9KV7j7+ux5Ik4erVq4rm4dnrj4oT6dehQ4d4R+hBtILZ2/5SJSUl2Llz\nJ7f9pSjT4HVfwzNq1Ci7r+HpS2lpKZfrDoRLrz9GBrRw4ULeEXpQKlNtbW2f57755htFMnRXVVXV\n7y+l6fV6dvfuXcYYY3V1dSwyMpIxxlhpaSmLiIhQPA9l+ukqKirYlStXWFNTE+8oQlm0aBGrqamx\n+b6j1+vtek1Vd4gYLKW2FHgQSmWKi4uTXy9ZssTm3Nq1axXJ0J2zszPc3d17/dXc3Kx4HtH2l6JM\nP93TTz+NadOmCT21nQcevf7oth46bjfs2bMHVqtVfq/r7aHp06dTJgCtra2KX7M3cXFx8oPiJUuW\n2Ox8u3btWsUfIou2vxRlenj2/sY71HDp9WfXcdkQMXv2bFZTU8M7hg1RMnUdxne/lcjrdqdomdrb\n21l+fj5LT09nhYWFil+/N5Tp4QQEBPCOIJTa2lq2fPlyNmPGDDZjxgy2YsUKZjab7XpNGjkBGDt2\nbI8hK2+iZLp165a81qHr685jIub+UpRpYDzX8Aw1PHr9UXECMGPGDCQlJUGn08lTWiVJgkajUX2m\nrs+Zur5mjHFrrSJawRRxfynKNDDWbc1O1+PY2FhFs4iKZ68/Kk4A7ty5A0mSbDYYA8C1OImSiec6\nh76IVjBF3F+KMg1MxP/bomEce/1RhwjSr6HQJZk3g8GA3Nxc+Tg3Nxf79+9Heno6XnnlFS6r/CnT\n4HFZwzMEKd2xQtUjJxG/8YqWiedPTn0R7e8IEGt/Kcr0YLZs2YJ9+/bJ2457eXnh0qVL3PKQDqou\nTl3vMXcfQHa9F60k0TKJ2CVZtIIp2v5SlOnB8FjDQwZGt/XIoInQiLI7ETKZzeZe95cCgOLiYi7P\nLinT4P3+979HbGwsdu3ahXfffRdZWVm4ePEi9u/fzyWPSPprKWXvXn/UIQJAZWUlli1bBq1WC61W\ni7i4OFRWVlImMijR0dGoq6vr8X5hYSHWrFnDIRFlehDbt2/He++9h8rKSvj7++Py5cvYunUrtzwi\nKS0t7fOX3ZvQ2nUV1RARFhbGzp07Jx+fO3eOhYWFcUwkTiZPT88+f3l5eSmepzsR+h4eP36cBQcH\n2yyazsnJYYGBgayiooIyCZyJiEvVz5w6Mcag1WrlY61Wi9TUVI6JxMkkYpdk0bYVEHFDRso0MBEn\n1pD7qDgB8PPzQ05ODgwGAxhjMJlMCAgIoEyCErFgira/FGUaGBNsYg2xRRMiYLs/UHe8NtQTMRPp\nXdd/K7PZDFdXVzg6OgIQ4/8PZRqYCBNriC0qToQ8pOrq6n7Pu7u7K5TkPsr0YKg4iUfVt/W++OIL\nAH2vH+IxtVXETKR/zs7OGDduXK/nysvLFU7TgTKRoU7VI6dXX32134Wt27dvVzBNBxEzkf51/am7\n+/5SvH4ip0wD47mGhwxM1SOnlJQU3hF6EDETGTxRNmTsijL1TsSJNeQ+VS/CfeWVV+TXWVlZHJPc\nJ2ImQghRmqpHTl07Lhw5cgRLly7lmKaDiJlI/0TbX4oykf8Hqh45EfIoiLa/VPcclIkMRaqeEOHj\n44PAwEAAwJkzZ3qsLeIx+UDETIQQojRVF6f333+/z3OSJCEsLEzBNB1EzET6J2IbHMpEhjpVP3Na\ntGgR7wg9iJiJ9E/ENjiUiQx1qh45VVRUYM+ePZgwYQJWrlyJlJQUfPXVV5g6dSo2bdrU54JBtWUi\ngydipwHKRIYiVU+ISExMhI+PD8aPH4+IiAhMnDgRaWlpeOaZZ5CUlESZCCGEE1Xf1mtoaIDRaATQ\nsaZo1apVAIApU6Zg/vz5lIkQQjhRdXEaNmyY/Lr7RIPhw/n81YiYifRPtP2lumfofkyZyFCg6u92\ns2bNQlNTE1xcXGw6M1y/fh3jx4+nTGRQRGyDQ5nIUKfqCRGEEELEpOqREwDU19ejqKgINTU1AIDJ\nkyfD398fTk5OlIkQQjhR9cjJZDIhLS0NHh4euHjxImbOnAlJklBaWorU1NR+W+qrKRMhhCiOqZhe\nr2d3795ljDFWV1fHIiMjGWOMlZaWsoiICMpECCGcqHqdU1tbG0aOHAkAcHFxQX19PQDA09MTLS0t\nlIkQQjhR9TOngIAAREZG4rnnnsOnn36KoKAgAMDt27fR3t5OmQghhBNVP3NijKGgoADffvstpk6d\nKncDp0yEEMKXqosTIYQQMan6mVNFRQViYmKg1+uRmpoKi8Uin4uPj6dMhBDCiaqLU2JiIvR6PXbv\n3o3bt2/DaDTKExBu3rxJmQghhBNVF6eGhgYsXrwYnp6e2Lx5M4xGI4xGIyorKykTIYRwpOrZegBg\nsVjkzgsGgwHjx49HXFwcGhsbKRMhhHAy7PXXX3+ddwhe2traIEkSJk6cKL/35JNPQqPRoKysDAsX\nLqRMhBDCgapn65nNZowZM6bXc8XFxdBoNAonEjMTIYQoTdXPnKKjo1FXV9fj/cLCQqxZs4ZDIjEz\nEUKI0lRdnOLj4xEVFYUffvhBfu/IkSNISUlBRkYGZSKEEE5UPSHi+eefh5OTE6Kjo3Hw4EEcP34c\neXl5yMzMxIQJEygTIYRwouriBAA6nQ6Ojo4IDw+Hh4cHMjMz4erqSpkIIYQjVU+ImDNnjvzabDbD\n1dUVjo6OAABJknD69GnKRAghHKi6OFVXV/d73t3dXaEk94mYiRBClKbq23rOzs4YN25cr+fKy8sV\nTtNBxEyEEKI0Vc/Wi4uLk18vWbLE5tzatWuVjgNAzEyEEKI0VRenrlpbW3lH6EHETIQQogQqToQQ\nQoSj6mdOt27dwt69e3u87jymTIQQwoeqZ+ulpaVBkiQAHdujd3+9YsUKykQIIRyoujgRQggRk6pv\n63W9ZdYbHqMUETMRQojSVD0honPQyBhDdnY25zQdRMxECCFKo9t6PwoNDcWHH37IO4YNETMRQogS\nVD1yIoQQIiYqToQQQoSj6tt6Xl5efZ6TJAlXr15VME0HETMRQojSVF2cCCGEiIlu6xFCCBEOFSdC\nCCHCoeJECCFEOKruEEHUKzQ0FADQ1NSEGzduYMqUKQCAwMBArFy58id/3vr6eqxevRolJSXw9vbG\n4cOHbc7v27cPJ06cAACEhITg5Zdf7vE5qqurERQUBC8vLzg4OODevXvQaDRYsWIFRo8e3e/1q6qq\n8NlnnyEiIuIn/xn6cufOHeTk5NjsOUaI3TBCVOzChQts9uzZj+zzNTQ0sOLiYnb27FkWGRlpc+7j\njz9mOp2OtbS0sObmZqbT6dj58+d7fI6qqirm6ekpH7e2trJdu3axhQsXMqvV2u/1P//88x7XfVSq\nqqoe6d8VIf2hkRNRNdZtsuru3btx6tQpAIBGo8Frr72G4cOH49VXX8WwYcNw7do11NfXY86cOdiw\nYUOPz/ezn/0Mv/nNb3DhwoUe506dOgW9Xg9nZ2cAwIIFC1BQUAA/P79+Mzo5OWHNmjUoKCjAP/7x\nD2i1Wqxbtw7Xrl1Da2srJk6ciJ07d8LNzQ2bNm3CrVu3EBoaisDAQBiNRqxYsQJNTU1obGyEn58f\nNm3aBEmSYDKZsHfvXjg6OoIxhj179uDpp59GaWkptm7div/+979gjGH58uUIDg7Gpk2bUFtbi9DQ\nUEyfPh0bNmzA6tWrUV1djeHDh8PHxweJiYk/9Z+CEFuciyMhXH3++efyaMBkMrHw8HBmsVhYe3s7\nS0hIYBkZGYwxxtavX8+WLFnC2tramMViYYsXL2YnT57s9/N2H8G8+OKL7OjRo/JxTk4Oe/HFF3v8\n3u4jp06rV69m77zzDmOMsf/85z/y+wcOHGBbt25ljHWMBLte12KxsMbGRsYYY1arlcXHx7NTp04x\nxhjT6/Wsrq6OMcbYvXv3WEtLC7NYLCwkJIRVV1czxhirr69ns2fPZvX19ay6utpm5GQymVhSUpJ8\n3HkdQh4FGjkR8qMvvvgCISEhcHR0BAAsWrQIf/vb3xATEwNJkqDX6zF8eMeXjMFgwIULFzBv3jzF\n8rW3t8v7exUUFCA7OxtWqxUtLS1wd3cH0HMkyBjDW2+9haKiIjg4OMBsNmPGjBkICgrCs88+i3Xr\n1kGr1cLPzw+TJ0/GlStXcP36dSxfvtzm81RVVcHNzc3mvWeeeQa7du3Cjh07MHPmTGi1Wjv+6Yna\nUHEi5EeSJPX45t5V13P9fVzn5+ruF7/4BW7cuCEf19TU4Mknnxx0vqtXr2LRokX4/vvvkZaWhg8/\n/BBjxozBmTNnkJGR0evvOXbsGCoqKvD3v/8dTk5OSElJgdVqBQC88cYbKCkpQXFxMV5++WWsWrUK\nTz31FJ566qleGw5XV1fbHHd+3GeffYZPPvkEaWlpOHbsGBwcaBIweXj0v4iQH/n6+iI3NxcWiwXt\n7e344IMP8Nvf/hZARzEymUywWq1oa2tDXl6efK43vRWvoKAg5OXloaWlBc3NzcjLy8PcuXMHzGWx\nWPCnP/0JLi4u8PPzQ2NjI0aMGAFXV1fcu3cPR48elT92xIgRaGpqko8bGxvx+OOPw8nJCbdv30Z+\nfr58rqqqCtOmTUNMTAzCwsLw1VdfwcvLC3fu3EFRUZH8cZ0ts0aMGIHm5mb5/bq6Ojg7O0On0yEx\nMRE//PAD7t69O+Cfh5DBoJETUb3OUY5Op0NJSQlCQkIgSRI0Gg2MRqP8MZ6enoiMjITZbEZgYGCf\nhSUgIABtbW1oaGiAVqtFbGwsYmJioNVqcfnyZXkau8FggL+/f5+5QkND4eDgAKvVCl9fX/z1r3+F\ng4MDpk2bBl9fX+h0Ori5uUGj0aCkpAQA4O3tDRcXF+j1egQHB8NoNKKgoADz58/H2LFj4ePjI3/+\nN998E9999x0YYxg9ejRSUlIwfPhwvP3223jjjTeQkpICAHjiiSfwzjvvYOzYsfD19UVwcDA0Gg0W\nLFiAHTt2AABaW1sRHx+Pn//85w/5r0FIB+qtR8ggbNiwAb6+vnJhIYTYF93WI4QQIhwaORFCCBEO\njZwIIYQIh4oTIYQQ4VBxIoQQIhwqToQQQoRDxYkQQohwqDgRQggRzv8Ai5ABCjv5LhIAAAAASUVO\nRK5CYII=\n",
       "text": [
        "<matplotlib.figure.Figure at 0x2aaaeb84ba50>"
       ]
      }
     ],
     "prompt_number": 71
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "pause_shuffled_intersect_df / pause_shuffled_union_df"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th>B3P27_IP_Day1.peaks.bed</th>\n",
        "      <th>B3P27_IP_Day23.peaks.bed</th>\n",
        "      <th>B3P27_IP_Day27.peaks.bed</th>\n",
        "      <th>B3P27_IP_Day6.peaks.bed</th>\n",
        "      <th>SRR334701.peaks.bed</th>\n",
        "      <th>SRR334703.peaks.bed</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th rowspan=\"30\" valign=\"top\">0</th>\n",
        "      <th>293T_hnRNPA1_WT_rep1</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000087</td>\n",
        "      <td> 0.000397</td>\n",
        "      <td> 0.000359</td>\n",
        "      <td> 0.000226</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>293T_hnRNPF_WT_rep1</th>\n",
        "      <td> 0.000252</td>\n",
        "      <td> 0.000063</td>\n",
        "      <td> 0.000436</td>\n",
        "      <td> 0.000643</td>\n",
        "      <td> 0.001509</td>\n",
        "      <td> 0.000137</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>293T_hnRNPM_WT_rep1</th>\n",
        "      <td> 0.000047</td>\n",
        "      <td> 0.000047</td>\n",
        "      <td> 0.000090</td>\n",
        "      <td> 0.000086</td>\n",
        "      <td> 0.000157</td>\n",
        "      <td> 0.000125</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>293T_hnRNPU_WT_rep1</th>\n",
        "      <td> 0.000143</td>\n",
        "      <td> 0.000109</td>\n",
        "      <td> 0.000433</td>\n",
        "      <td> 0.000718</td>\n",
        "      <td> 0.001630</td>\n",
        "      <td> 0.000134</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>FOX2</th>\n",
        "      <td> 0.000184</td>\n",
        "      <td> 0.000109</td>\n",
        "      <td> 0.000456</td>\n",
        "      <td> 0.000665</td>\n",
        "      <td> 0.002121</td>\n",
        "      <td> 0.000197</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293FlpIn_FMR1-iso1_WT_rep1</th>\n",
        "      <td> 0.001042</td>\n",
        "      <td> 0.000459</td>\n",
        "      <td> 0.002372</td>\n",
        "      <td> 0.003678</td>\n",
        "      <td> 0.015676</td>\n",
        "      <td> 0.001705</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293FlpIn_FMR1-iso7_WT_rep1</th>\n",
        "      <td> 0.001022</td>\n",
        "      <td> 0.000567</td>\n",
        "      <td> 0.002387</td>\n",
        "      <td> 0.003701</td>\n",
        "      <td> 0.012330</td>\n",
        "      <td> 0.001517</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago2_WT_rep1</th>\n",
        "      <td> 0.000487</td>\n",
        "      <td> 0.000585</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000135</td>\n",
        "      <td> 0.000214</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago2_WT_rep2</th>\n",
        "      <td> 0.000552</td>\n",
        "      <td> 0.000210</td>\n",
        "      <td> 0.000707</td>\n",
        "      <td> 0.001105</td>\n",
        "      <td> 0.001560</td>\n",
        "      <td> 0.000323</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago2_WT_rep3</th>\n",
        "      <td> 0.000614</td>\n",
        "      <td> 0.000104</td>\n",
        "      <td> 0.000543</td>\n",
        "      <td> 0.000843</td>\n",
        "      <td> 0.000895</td>\n",
        "      <td> 0.000229</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago2_WT_rep4</th>\n",
        "      <td> 0.000488</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000929</td>\n",
        "      <td> 0.001301</td>\n",
        "      <td> 0.001332</td>\n",
        "      <td> 0.000299</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago3_WT_rep1</th>\n",
        "      <td> 0.000627</td>\n",
        "      <td> 0.000351</td>\n",
        "      <td> 0.000509</td>\n",
        "      <td> 0.000598</td>\n",
        "      <td> 0.000418</td>\n",
        "      <td> 0.000346</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago4_WT_rep1</th>\n",
        "      <td> 0.000311</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000506</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000088</td>\n",
        "      <td> 0.000175</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_DGCR8_WT_rep1</th>\n",
        "      <td> 0.000497</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000632</td>\n",
        "      <td> 0.000685</td>\n",
        "      <td> 0.000865</td>\n",
        "      <td> 0.000759</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_DGCR8_overexpressed_rep1</th>\n",
        "      <td> 0.000801</td>\n",
        "      <td> 0.000294</td>\n",
        "      <td> 0.000446</td>\n",
        "      <td> 0.000180</td>\n",
        "      <td> 0.000456</td>\n",
        "      <td> 0.000157</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_HuR_WT_rep1</th>\n",
        "      <td> 0.000062</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000049</td>\n",
        "      <td> 0.000243</td>\n",
        "      <td> 0.000147</td>\n",
        "      <td> 0.000012</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_HuR_WT_rep2</th>\n",
        "      <td> 0.000036</td>\n",
        "      <td> 0.000030</td>\n",
        "      <td> 0.000119</td>\n",
        "      <td> 0.000178</td>\n",
        "      <td> 0.000082</td>\n",
        "      <td> 0.000006</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_HuR_WT_rep3</th>\n",
        "      <td> 0.000214</td>\n",
        "      <td> 0.000108</td>\n",
        "      <td> 0.000421</td>\n",
        "      <td> 0.000688</td>\n",
        "      <td> 0.003269</td>\n",
        "      <td> 0.000256</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_HuR_WT_rep6</th>\n",
        "      <td> 0.000084</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000316</td>\n",
        "      <td> 0.000146</td>\n",
        "      <td> 0.000222</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_IGF2BP1_WT_rep1</th>\n",
        "      <td> 0.000753</td>\n",
        "      <td> 0.000365</td>\n",
        "      <td> 0.001623</td>\n",
        "      <td> 0.002314</td>\n",
        "      <td> 0.008096</td>\n",
        "      <td> 0.001083</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_IGF2BP2_WT_rep1</th>\n",
        "      <td> 0.000618</td>\n",
        "      <td> 0.000389</td>\n",
        "      <td> 0.001391</td>\n",
        "      <td> 0.001477</td>\n",
        "      <td> 0.004767</td>\n",
        "      <td> 0.000701</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_IGF2BP3_WT_rep1</th>\n",
        "      <td> 0.000717</td>\n",
        "      <td> 0.000310</td>\n",
        "      <td> 0.001471</td>\n",
        "      <td> 0.002386</td>\n",
        "      <td> 0.008917</td>\n",
        "      <td> 0.000948</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_LIN28B_WT_rep1</th>\n",
        "      <td> 0.000373</td>\n",
        "      <td> 0.000193</td>\n",
        "      <td> 0.000087</td>\n",
        "      <td> 0.000319</td>\n",
        "      <td> 0.000324</td>\n",
        "      <td> 0.000157</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_PAPD5_WT_rep1</th>\n",
        "      <td> 0.000523</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000219</td>\n",
        "      <td> 0.000177</td>\n",
        "      <td> 0.000109</td>\n",
        "      <td> 0.000157</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Pumilio2_WT_rep1</th>\n",
        "      <td> 0.000279</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000253</td>\n",
        "      <td> 0.000222</td>\n",
        "      <td> 0.000123</td>\n",
        "      <td> 0.000108</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_QIK_WT_rep1</th>\n",
        "      <td> 0.000247</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000233</td>\n",
        "      <td> 0.000215</td>\n",
        "      <td> 0.001015</td>\n",
        "      <td> 0.000206</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_TNRC6A_WT_rep1</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000089</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_TNRC6B_WT_rep1</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000311</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000045</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_TNRC6C_WT_rep1</th>\n",
        "      <td> 0.000304</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000022</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_hnRNPU_WT_rep1</th>\n",
        "      <td> 0.000123</td>\n",
        "      <td> 0.000035</td>\n",
        "      <td> 0.000452</td>\n",
        "      <td> 0.000905</td>\n",
        "      <td> 0.001776</td>\n",
        "      <td> 0.000303</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>...</th>\n",
        "      <th>...</th>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "      <td>...</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th rowspan=\"30\" valign=\"top\">9</th>\n",
        "      <th>SRR427115</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR427116</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR452496</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000418</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.001096</td>\n",
        "      <td> 0.000111</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR452497</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000275</td>\n",
        "      <td> 0.000023</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR490650</th>\n",
        "      <td> 0.000146</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000483</td>\n",
        "      <td> 0.000193</td>\n",
        "      <td> 0.000287</td>\n",
        "      <td> 0.000062</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR518495</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000616</td>\n",
        "      <td> 0.000022</td>\n",
        "      <td> 0.000177</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR518496</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000022</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR518497</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.001414</td>\n",
        "      <td> 0.001275</td>\n",
        "      <td> 0.000406</td>\n",
        "      <td> 0.000210</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR518498</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.001213</td>\n",
        "      <td> 0.001824</td>\n",
        "      <td> 0.000624</td>\n",
        "      <td> 0.000765</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527727</th>\n",
        "      <td> 0.001062</td>\n",
        "      <td> 0.000526</td>\n",
        "      <td> 0.001924</td>\n",
        "      <td> 0.003515</td>\n",
        "      <td> 0.011563</td>\n",
        "      <td> 0.001534</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527728</th>\n",
        "      <td> 0.001011</td>\n",
        "      <td> 0.000480</td>\n",
        "      <td> 0.001768</td>\n",
        "      <td> 0.003208</td>\n",
        "      <td> 0.003445</td>\n",
        "      <td> 0.000860</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527729</th>\n",
        "      <td> 0.001061</td>\n",
        "      <td> 0.000490</td>\n",
        "      <td> 0.001982</td>\n",
        "      <td> 0.004476</td>\n",
        "      <td> 0.010775</td>\n",
        "      <td> 0.001221</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527730</th>\n",
        "      <td> 0.001152</td>\n",
        "      <td> 0.000511</td>\n",
        "      <td> 0.002297</td>\n",
        "      <td> 0.004355</td>\n",
        "      <td> 0.011749</td>\n",
        "      <td> 0.001280</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527731</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000802</td>\n",
        "      <td> 0.000828</td>\n",
        "      <td> 0.000243</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527732</th>\n",
        "      <td> 0.000440</td>\n",
        "      <td> 0.000713</td>\n",
        "      <td> 0.000947</td>\n",
        "      <td> 0.002356</td>\n",
        "      <td> 0.001219</td>\n",
        "      <td> 0.000278</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR567525</th>\n",
        "      <td> 0.002074</td>\n",
        "      <td> 0.000987</td>\n",
        "      <td> 0.004310</td>\n",
        "      <td> 0.007482</td>\n",
        "      <td> 0.031601</td>\n",
        "      <td> 0.003614</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR567526</th>\n",
        "      <td> 0.001328</td>\n",
        "      <td> 0.000605</td>\n",
        "      <td> 0.002586</td>\n",
        "      <td> 0.004681</td>\n",
        "      <td> 0.025696</td>\n",
        "      <td> 0.002401</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628446</th>\n",
        "      <td> 0.000418</td>\n",
        "      <td> 0.000236</td>\n",
        "      <td> 0.001152</td>\n",
        "      <td> 0.001978</td>\n",
        "      <td> 0.009236</td>\n",
        "      <td> 0.000871</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628447</th>\n",
        "      <td> 0.000506</td>\n",
        "      <td> 0.000210</td>\n",
        "      <td> 0.001225</td>\n",
        "      <td> 0.001454</td>\n",
        "      <td> 0.002231</td>\n",
        "      <td> 0.000321</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628448</th>\n",
        "      <td> 0.000435</td>\n",
        "      <td> 0.000226</td>\n",
        "      <td> 0.001209</td>\n",
        "      <td> 0.001090</td>\n",
        "      <td> 0.003767</td>\n",
        "      <td> 0.000679</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628449</th>\n",
        "      <td> 0.000421</td>\n",
        "      <td> 0.000109</td>\n",
        "      <td> 0.001077</td>\n",
        "      <td> 0.001150</td>\n",
        "      <td> 0.002908</td>\n",
        "      <td> 0.000413</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628450</th>\n",
        "      <td> 0.000783</td>\n",
        "      <td> 0.000335</td>\n",
        "      <td> 0.001505</td>\n",
        "      <td> 0.003379</td>\n",
        "      <td> 0.008603</td>\n",
        "      <td> 0.001514</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628451</th>\n",
        "      <td> 0.000434</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.001609</td>\n",
        "      <td> 0.001088</td>\n",
        "      <td> 0.002065</td>\n",
        "      <td> 0.000424</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628452</th>\n",
        "      <td> 0.000596</td>\n",
        "      <td> 0.000110</td>\n",
        "      <td> 0.001251</td>\n",
        "      <td> 0.001926</td>\n",
        "      <td> 0.003293</td>\n",
        "      <td> 0.000954</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628453</th>\n",
        "      <td> 0.000613</td>\n",
        "      <td> 0.000208</td>\n",
        "      <td> 0.001330</td>\n",
        "      <td> 0.002105</td>\n",
        "      <td> 0.003639</td>\n",
        "      <td> 0.000950</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628454</th>\n",
        "      <td> 0.000890</td>\n",
        "      <td> 0.000238</td>\n",
        "      <td> 0.001314</td>\n",
        "      <td> 0.002938</td>\n",
        "      <td> 0.007440</td>\n",
        "      <td> 0.001688</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR764666</th>\n",
        "      <td> 0.000136</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000546</td>\n",
        "      <td> 0.000163</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR764667</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000237</td>\n",
        "      <td> 0.000022</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR764668</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000023</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR764669</th>\n",
        "      <td> 0.000122</td>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000112</td>\n",
        "      <td> 0.000697</td>\n",
        "      <td> 0.000120</td>\n",
        "      <td> 0.000192</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "<p>1410 rows \u00d7 6 columns</p>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 27,
       "text": [
        "                                   B3P27_IP_Day1.peaks.bed  \\\n",
        "0 293T_hnRNPA1_WT_rep1                            0.000000   \n",
        "  293T_hnRNPF_WT_rep1                             0.000252   \n",
        "  293T_hnRNPM_WT_rep1                             0.000047   \n",
        "  293T_hnRNPU_WT_rep1                             0.000143   \n",
        "  FOX2                                            0.000184   \n",
        "  HEK293FlpIn_FMR1-iso1_WT_rep1                   0.001042   \n",
        "  HEK293FlpIn_FMR1-iso7_WT_rep1                   0.001022   \n",
        "  HEK293_Ago2_WT_rep1                             0.000487   \n",
        "  HEK293_Ago2_WT_rep2                             0.000552   \n",
        "  HEK293_Ago2_WT_rep3                             0.000614   \n",
        "  HEK293_Ago2_WT_rep4                             0.000488   \n",
        "  HEK293_Ago3_WT_rep1                             0.000627   \n",
        "  HEK293_Ago4_WT_rep1                             0.000311   \n",
        "  HEK293_DGCR8_WT_rep1                            0.000497   \n",
        "  HEK293_DGCR8_overexpressed_rep1                 0.000801   \n",
        "  HEK293_HuR_WT_rep1                              0.000062   \n",
        "  HEK293_HuR_WT_rep2                              0.000036   \n",
        "  HEK293_HuR_WT_rep3                              0.000214   \n",
        "  HEK293_HuR_WT_rep6                              0.000084   \n",
        "  HEK293_IGF2BP1_WT_rep1                          0.000753   \n",
        "  HEK293_IGF2BP2_WT_rep1                          0.000618   \n",
        "  HEK293_IGF2BP3_WT_rep1                          0.000717   \n",
        "  HEK293_LIN28B_WT_rep1                           0.000373   \n",
        "  HEK293_PAPD5_WT_rep1                            0.000523   \n",
        "  HEK293_Pumilio2_WT_rep1                         0.000279   \n",
        "  HEK293_QIK_WT_rep1                              0.000247   \n",
        "  HEK293_TNRC6A_WT_rep1                           0.000000   \n",
        "  HEK293_TNRC6B_WT_rep1                           0.000000   \n",
        "  HEK293_TNRC6C_WT_rep1                           0.000304   \n",
        "  HEK293_hnRNPU_WT_rep1                           0.000123   \n",
        "...                                                    ...   \n",
        "9 SRR427115                                       0.000000   \n",
        "  SRR427116                                       0.000000   \n",
        "  SRR452496                                       0.000000   \n",
        "  SRR452497                                       0.000000   \n",
        "  SRR490650                                       0.000146   \n",
        "  SRR518495                                       0.000000   \n",
        "  SRR518496                                       0.000000   \n",
        "  SRR518497                                       0.000000   \n",
        "  SRR518498                                       0.000000   \n",
        "  SRR527727                                       0.001062   \n",
        "  SRR527728                                       0.001011   \n",
        "  SRR527729                                       0.001061   \n",
        "  SRR527730                                       0.001152   \n",
        "  SRR527731                                       0.000000   \n",
        "  SRR527732                                       0.000440   \n",
        "  SRR567525                                       0.002074   \n",
        "  SRR567526                                       0.001328   \n",
        "  SRR628446                                       0.000418   \n",
        "  SRR628447                                       0.000506   \n",
        "  SRR628448                                       0.000435   \n",
        "  SRR628449                                       0.000421   \n",
        "  SRR628450                                       0.000783   \n",
        "  SRR628451                                       0.000434   \n",
        "  SRR628452                                       0.000596   \n",
        "  SRR628453                                       0.000613   \n",
        "  SRR628454                                       0.000890   \n",
        "  SRR764666                                       0.000136   \n",
        "  SRR764667                                       0.000000   \n",
        "  SRR764668                                       0.000000   \n",
        "  SRR764669                                       0.000122   \n",
        "\n",
        "                                   B3P27_IP_Day23.peaks.bed  \\\n",
        "0 293T_hnRNPA1_WT_rep1                             0.000000   \n",
        "  293T_hnRNPF_WT_rep1                              0.000063   \n",
        "  293T_hnRNPM_WT_rep1                              0.000047   \n",
        "  293T_hnRNPU_WT_rep1                              0.000109   \n",
        "  FOX2                                             0.000109   \n",
        "  HEK293FlpIn_FMR1-iso1_WT_rep1                    0.000459   \n",
        "  HEK293FlpIn_FMR1-iso7_WT_rep1                    0.000567   \n",
        "  HEK293_Ago2_WT_rep1                              0.000585   \n",
        "  HEK293_Ago2_WT_rep2                              0.000210   \n",
        "  HEK293_Ago2_WT_rep3                              0.000104   \n",
        "  HEK293_Ago2_WT_rep4                              0.000000   \n",
        "  HEK293_Ago3_WT_rep1                              0.000351   \n",
        "  HEK293_Ago4_WT_rep1                              0.000000   \n",
        "  HEK293_DGCR8_WT_rep1                             0.000000   \n",
        "  HEK293_DGCR8_overexpressed_rep1                  0.000294   \n",
        "  HEK293_HuR_WT_rep1                               0.000000   \n",
        "  HEK293_HuR_WT_rep2                               0.000030   \n",
        "  HEK293_HuR_WT_rep3                               0.000108   \n",
        "  HEK293_HuR_WT_rep6                               0.000000   \n",
        "  HEK293_IGF2BP1_WT_rep1                           0.000365   \n",
        "  HEK293_IGF2BP2_WT_rep1                           0.000389   \n",
        "  HEK293_IGF2BP3_WT_rep1                           0.000310   \n",
        "  HEK293_LIN28B_WT_rep1                            0.000193   \n",
        "  HEK293_PAPD5_WT_rep1                             0.000000   \n",
        "  HEK293_Pumilio2_WT_rep1                          0.000000   \n",
        "  HEK293_QIK_WT_rep1                               0.000000   \n",
        "  HEK293_TNRC6A_WT_rep1                            0.000000   \n",
        "  HEK293_TNRC6B_WT_rep1                            0.000000   \n",
        "  HEK293_TNRC6C_WT_rep1                            0.000000   \n",
        "  HEK293_hnRNPU_WT_rep1                            0.000035   \n",
        "...                                                     ...   \n",
        "9 SRR427115                                        0.000000   \n",
        "  SRR427116                                        0.000000   \n",
        "  SRR452496                                        0.000418   \n",
        "  SRR452497                                        0.000000   \n",
        "  SRR490650                                        0.000000   \n",
        "  SRR518495                                        0.000000   \n",
        "  SRR518496                                        0.000000   \n",
        "  SRR518497                                        0.000000   \n",
        "  SRR518498                                        0.000000   \n",
        "  SRR527727                                        0.000526   \n",
        "  SRR527728                                        0.000480   \n",
        "  SRR527729                                        0.000490   \n",
        "  SRR527730                                        0.000511   \n",
        "  SRR527731                                        0.000000   \n",
        "  SRR527732                                        0.000713   \n",
        "  SRR567525                                        0.000987   \n",
        "  SRR567526                                        0.000605   \n",
        "  SRR628446                                        0.000236   \n",
        "  SRR628447                                        0.000210   \n",
        "  SRR628448                                        0.000226   \n",
        "  SRR628449                                        0.000109   \n",
        "  SRR628450                                        0.000335   \n",
        "  SRR628451                                        0.000000   \n",
        "  SRR628452                                        0.000110   \n",
        "  SRR628453                                        0.000208   \n",
        "  SRR628454                                        0.000238   \n",
        "  SRR764666                                        0.000000   \n",
        "  SRR764667                                        0.000000   \n",
        "  SRR764668                                        0.000000   \n",
        "  SRR764669                                        0.000000   \n",
        "\n",
        "                                   B3P27_IP_Day27.peaks.bed  \\\n",
        "0 293T_hnRNPA1_WT_rep1                             0.000087   \n",
        "  293T_hnRNPF_WT_rep1                              0.000436   \n",
        "  293T_hnRNPM_WT_rep1                              0.000090   \n",
        "  293T_hnRNPU_WT_rep1                              0.000433   \n",
        "  FOX2                                             0.000456   \n",
        "  HEK293FlpIn_FMR1-iso1_WT_rep1                    0.002372   \n",
        "  HEK293FlpIn_FMR1-iso7_WT_rep1                    0.002387   \n",
        "  HEK293_Ago2_WT_rep1                              0.000000   \n",
        "  HEK293_Ago2_WT_rep2                              0.000707   \n",
        "  HEK293_Ago2_WT_rep3                              0.000543   \n",
        "  HEK293_Ago2_WT_rep4                              0.000929   \n",
        "  HEK293_Ago3_WT_rep1                              0.000509   \n",
        "  HEK293_Ago4_WT_rep1                              0.000506   \n",
        "  HEK293_DGCR8_WT_rep1                             0.000632   \n",
        "  HEK293_DGCR8_overexpressed_rep1                  0.000446   \n",
        "  HEK293_HuR_WT_rep1                               0.000049   \n",
        "  HEK293_HuR_WT_rep2                               0.000119   \n",
        "  HEK293_HuR_WT_rep3                               0.000421   \n",
        "  HEK293_HuR_WT_rep6                               0.000316   \n",
        "  HEK293_IGF2BP1_WT_rep1                           0.001623   \n",
        "  HEK293_IGF2BP2_WT_rep1                           0.001391   \n",
        "  HEK293_IGF2BP3_WT_rep1                           0.001471   \n",
        "  HEK293_LIN28B_WT_rep1                            0.000087   \n",
        "  HEK293_PAPD5_WT_rep1                             0.000219   \n",
        "  HEK293_Pumilio2_WT_rep1                          0.000253   \n",
        "  HEK293_QIK_WT_rep1                               0.000233   \n",
        "  HEK293_TNRC6A_WT_rep1                            0.000000   \n",
        "  HEK293_TNRC6B_WT_rep1                            0.000311   \n",
        "  HEK293_TNRC6C_WT_rep1                            0.000000   \n",
        "  HEK293_hnRNPU_WT_rep1                            0.000452   \n",
        "...                                                     ...   \n",
        "9 SRR427115                                        0.000000   \n",
        "  SRR427116                                        0.000000   \n",
        "  SRR452496                                        0.000000   \n",
        "  SRR452497                                        0.000000   \n",
        "  SRR490650                                        0.000483   \n",
        "  SRR518495                                        0.000000   \n",
        "  SRR518496                                        0.000000   \n",
        "  SRR518497                                        0.001414   \n",
        "  SRR518498                                        0.001213   \n",
        "  SRR527727                                        0.001924   \n",
        "  SRR527728                                        0.001768   \n",
        "  SRR527729                                        0.001982   \n",
        "  SRR527730                                        0.002297   \n",
        "  SRR527731                                        0.000802   \n",
        "  SRR527732                                        0.000947   \n",
        "  SRR567525                                        0.004310   \n",
        "  SRR567526                                        0.002586   \n",
        "  SRR628446                                        0.001152   \n",
        "  SRR628447                                        0.001225   \n",
        "  SRR628448                                        0.001209   \n",
        "  SRR628449                                        0.001077   \n",
        "  SRR628450                                        0.001505   \n",
        "  SRR628451                                        0.001609   \n",
        "  SRR628452                                        0.001251   \n",
        "  SRR628453                                        0.001330   \n",
        "  SRR628454                                        0.001314   \n",
        "  SRR764666                                        0.000000   \n",
        "  SRR764667                                        0.000000   \n",
        "  SRR764668                                        0.000000   \n",
        "  SRR764669                                        0.000112   \n",
        "\n",
        "                                   B3P27_IP_Day6.peaks.bed  \\\n",
        "0 293T_hnRNPA1_WT_rep1                            0.000397   \n",
        "  293T_hnRNPF_WT_rep1                             0.000643   \n",
        "  293T_hnRNPM_WT_rep1                             0.000086   \n",
        "  293T_hnRNPU_WT_rep1                             0.000718   \n",
        "  FOX2                                            0.000665   \n",
        "  HEK293FlpIn_FMR1-iso1_WT_rep1                   0.003678   \n",
        "  HEK293FlpIn_FMR1-iso7_WT_rep1                   0.003701   \n",
        "  HEK293_Ago2_WT_rep1                             0.000000   \n",
        "  HEK293_Ago2_WT_rep2                             0.001105   \n",
        "  HEK293_Ago2_WT_rep3                             0.000843   \n",
        "  HEK293_Ago2_WT_rep4                             0.001301   \n",
        "  HEK293_Ago3_WT_rep1                             0.000598   \n",
        "  HEK293_Ago4_WT_rep1                             0.000000   \n",
        "  HEK293_DGCR8_WT_rep1                            0.000685   \n",
        "  HEK293_DGCR8_overexpressed_rep1                 0.000180   \n",
        "  HEK293_HuR_WT_rep1                              0.000243   \n",
        "  HEK293_HuR_WT_rep2                              0.000178   \n",
        "  HEK293_HuR_WT_rep3                              0.000688   \n",
        "  HEK293_HuR_WT_rep6                              0.000146   \n",
        "  HEK293_IGF2BP1_WT_rep1                          0.002314   \n",
        "  HEK293_IGF2BP2_WT_rep1                          0.001477   \n",
        "  HEK293_IGF2BP3_WT_rep1                          0.002386   \n",
        "  HEK293_LIN28B_WT_rep1                           0.000319   \n",
        "  HEK293_PAPD5_WT_rep1                            0.000177   \n",
        "  HEK293_Pumilio2_WT_rep1                         0.000222   \n",
        "  HEK293_QIK_WT_rep1                              0.000215   \n",
        "  HEK293_TNRC6A_WT_rep1                           0.000000   \n",
        "  HEK293_TNRC6B_WT_rep1                           0.000000   \n",
        "  HEK293_TNRC6C_WT_rep1                           0.000000   \n",
        "  HEK293_hnRNPU_WT_rep1                           0.000905   \n",
        "...                                                    ...   \n",
        "9 SRR427115                                       0.000000   \n",
        "  SRR427116                                       0.000000   \n",
        "  SRR452496                                       0.001096   \n",
        "  SRR452497                                       0.000275   \n",
        "  SRR490650                                       0.000193   \n",
        "  SRR518495                                       0.000616   \n",
        "  SRR518496                                       0.000000   \n",
        "  SRR518497                                       0.001275   \n",
        "  SRR518498                                       0.001824   \n",
        "  SRR527727                                       0.003515   \n",
        "  SRR527728                                       0.003208   \n",
        "  SRR527729                                       0.004476   \n",
        "  SRR527730                                       0.004355   \n",
        "  SRR527731                                       0.000828   \n",
        "  SRR527732                                       0.002356   \n",
        "  SRR567525                                       0.007482   \n",
        "  SRR567526                                       0.004681   \n",
        "  SRR628446                                       0.001978   \n",
        "  SRR628447                                       0.001454   \n",
        "  SRR628448                                       0.001090   \n",
        "  SRR628449                                       0.001150   \n",
        "  SRR628450                                       0.003379   \n",
        "  SRR628451                                       0.001088   \n",
        "  SRR628452                                       0.001926   \n",
        "  SRR628453                                       0.002105   \n",
        "  SRR628454                                       0.002938   \n",
        "  SRR764666                                       0.000546   \n",
        "  SRR764667                                       0.000237   \n",
        "  SRR764668                                       0.000000   \n",
        "  SRR764669                                       0.000697   \n",
        "\n",
        "                                   SRR334701.peaks.bed  SRR334703.peaks.bed  \n",
        "0 293T_hnRNPA1_WT_rep1                        0.000359             0.000226  \n",
        "  293T_hnRNPF_WT_rep1                         0.001509             0.000137  \n",
        "  293T_hnRNPM_WT_rep1                         0.000157             0.000125  \n",
        "  293T_hnRNPU_WT_rep1                         0.001630             0.000134  \n",
        "  FOX2                                        0.002121             0.000197  \n",
        "  HEK293FlpIn_FMR1-iso1_WT_rep1               0.015676             0.001705  \n",
        "  HEK293FlpIn_FMR1-iso7_WT_rep1               0.012330             0.001517  \n",
        "  HEK293_Ago2_WT_rep1                         0.000135             0.000214  \n",
        "  HEK293_Ago2_WT_rep2                         0.001560             0.000323  \n",
        "  HEK293_Ago2_WT_rep3                         0.000895             0.000229  \n",
        "  HEK293_Ago2_WT_rep4                         0.001332             0.000299  \n",
        "  HEK293_Ago3_WT_rep1                         0.000418             0.000346  \n",
        "  HEK293_Ago4_WT_rep1                         0.000088             0.000175  \n",
        "  HEK293_DGCR8_WT_rep1                        0.000865             0.000759  \n",
        "  HEK293_DGCR8_overexpressed_rep1             0.000456             0.000157  \n",
        "  HEK293_HuR_WT_rep1                          0.000147             0.000012  \n",
        "  HEK293_HuR_WT_rep2                          0.000082             0.000006  \n",
        "  HEK293_HuR_WT_rep3                          0.003269             0.000256  \n",
        "  HEK293_HuR_WT_rep6                          0.000222             0.000000  \n",
        "  HEK293_IGF2BP1_WT_rep1                      0.008096             0.001083  \n",
        "  HEK293_IGF2BP2_WT_rep1                      0.004767             0.000701  \n",
        "  HEK293_IGF2BP3_WT_rep1                      0.008917             0.000948  \n",
        "  HEK293_LIN28B_WT_rep1                       0.000324             0.000157  \n",
        "  HEK293_PAPD5_WT_rep1                        0.000109             0.000157  \n",
        "  HEK293_Pumilio2_WT_rep1                     0.000123             0.000108  \n",
        "  HEK293_QIK_WT_rep1                          0.001015             0.000206  \n",
        "  HEK293_TNRC6A_WT_rep1                       0.000089             0.000000  \n",
        "  HEK293_TNRC6B_WT_rep1                       0.000045             0.000000  \n",
        "  HEK293_TNRC6C_WT_rep1                       0.000022             0.000000  \n",
        "  HEK293_hnRNPU_WT_rep1                       0.001776             0.000303  \n",
        "...                                                ...                  ...  \n",
        "9 SRR427115                                   0.000000             0.000000  \n",
        "  SRR427116                                   0.000000             0.000000  \n",
        "  SRR452496                                   0.000111             0.000000  \n",
        "  SRR452497                                   0.000023             0.000000  \n",
        "  SRR490650                                   0.000287             0.000062  \n",
        "  SRR518495                                   0.000022             0.000177  \n",
        "  SRR518496                                   0.000022             0.000000  \n",
        "  SRR518497                                   0.000406             0.000210  \n",
        "  SRR518498                                   0.000624             0.000765  \n",
        "  SRR527727                                   0.011563             0.001534  \n",
        "  SRR527728                                   0.003445             0.000860  \n",
        "  SRR527729                                   0.010775             0.001221  \n",
        "  SRR527730                                   0.011749             0.001280  \n",
        "  SRR527731                                   0.000243             0.000000  \n",
        "  SRR527732                                   0.001219             0.000278  \n",
        "  SRR567525                                   0.031601             0.003614  \n",
        "  SRR567526                                   0.025696             0.002401  \n",
        "  SRR628446                                   0.009236             0.000871  \n",
        "  SRR628447                                   0.002231             0.000321  \n",
        "  SRR628448                                   0.003767             0.000679  \n",
        "  SRR628449                                   0.002908             0.000413  \n",
        "  SRR628450                                   0.008603             0.001514  \n",
        "  SRR628451                                   0.002065             0.000424  \n",
        "  SRR628452                                   0.003293             0.000954  \n",
        "  SRR628453                                   0.003639             0.000950  \n",
        "  SRR628454                                   0.007440             0.001688  \n",
        "  SRR764666                                   0.000163             0.000000  \n",
        "  SRR764667                                   0.000022             0.000000  \n",
        "  SRR764668                                   0.000023             0.000000  \n",
        "  SRR764669                                   0.000120             0.000192  \n",
        "\n",
        "[1410 rows x 6 columns]"
       ]
      }
     ],
     "prompt_number": 27
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "shuffled_overlaps = (pause_shuffled_intersect_df / pause_shuffled_union_df)\n",
      "bg = x.groupby(level=1)\n",
      "mean = bg.mean()\n",
      "std = bg.std()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "ename": "AttributeError",
       "evalue": "'int' object has no attribute 'groupby'",
       "output_type": "pyerr",
       "traceback": [
        "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n\u001b[1;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
        "\u001b[1;32m<ipython-input-28-1bd23cbf70e8>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[0mshuffled_overlaps\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m(\u001b[0m\u001b[0mpause_shuffled_intersect_df\u001b[0m \u001b[1;33m/\u001b[0m \u001b[0mpause_shuffled_union_df\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 2\u001b[1;33m \u001b[0mbg\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mx\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mgroupby\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mlevel\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m1\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      3\u001b[0m \u001b[0mmean\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mbg\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mmean\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      4\u001b[0m \u001b[0mstd\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mbg\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mstd\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;31mAttributeError\u001b[0m: 'int' object has no attribute 'groupby'"
       ]
      }
     ],
     "prompt_number": 28
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "hist(shuffled_overlaps['SRR334701.peaks.bed'][~isinf(shuffled_overlaps['SRR334701.peaks.bed'])].values)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 77,
       "text": [
        "(array([ 1140.,    91.,    78.,    48.,     9.,     2.,     0.,     2.,\n",
        "          18.,    10.]),\n",
        " array([ 0.        ,  0.0031601 ,  0.00632019,  0.00948029,  0.01264039,\n",
        "        0.01580049,  0.01896058,  0.02212068,  0.02528078,  0.02844088,\n",
        "        0.03160097]),\n",
        " <a list of 10 Patch objects>)"
       ]
      },
      {
       "metadata": {},
       "output_type": "display_data",
       "png": "iVBORw0KGgoAAAANSUhEUgAAAfgAAAFXCAYAAABOYlxEAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAGYRJREFUeJzt3X9s1fW9x/HXgZY1IKswkZZzTgWlrD09HGj5VWBMfjbj\nuLmthdQ2m6nNWJZFO4PTzSxaGW5orv8MCZqxpcybHTukRF1O08m0c5NRHIxRkBCaSbudrwWGtVRg\n0Np+7h/eHUFKB6enPe2nz0fiH35PzzmfNx/rk2/POd+6jDFGAADAKqMSvQAAABB/BB4AAAsReAAA\nLETgAQCwEIEHAMBCBB4AAAv1Gfjy8nJNnjxZ06ZNix774Q9/qJycHOXk5KigoEAnT56M3rZp0yZl\nZ2fL7/frtddeix6vq6uT3+9Xdna2nn766QEYAwAAXM7V1+fg//SnP+mmm25SYWGhTpw4IUl68803\ntWTJEo0aNUqPPfaYWltb9Ytf/EIHDhxQeXm5Dhw4IMdx9IUvfEHvvvuuenp6NH36dO3Zs0fp6ema\nO3eutm/frtzc3EEbEgCAkabPM/glS5ZowoQJVxy78847NWrUx3ebP3++WltbJUnhcFhr1qxRUlKS\nbrvtNk2fPl379u3Tvn37lJmZqYyMDCUnJ6uwsFDhcHiAxgEAAFI/X4N//vnnVVRUJElyHEdTpkyJ\n3ubxeBSJROQ4jtxu91XHAQDAwIk58E899ZTGjBmj8vLyeK4HAADEQVIsd3rxxRf16quv6o033oge\n83g8chwn+u+O48jr9aqnp+eK45FIRF6v978+h8vlimVpAAAMW/H89TA3HPjdu3frySef1B/+8Ael\npKREjweDQZWXl+vRRx+V4zg6fvy45s+fr+7ubh0/flwtLS1KT0/Xrl27VFVVdV3Pxe/BAQCMFPE+\nse0z8GvXrtWePXt05swZeb1erV+/Xs8995wuXryogoICSdK8efP085//XHPmzFFxcbECgYBGjx6t\nX/7yl0pOTlZycrK2bdumu+66S93d3br33nuVl5cX1yEAAMCV+vyYXCK5XC7O4AEAI0a8u8eV7AAA\nsBCBBwDAQgQeAAALEXgAACxE4AEAsBCBBwDAQgQeAAALEXgAACxE4AEAsBCBBwDAQgQeAAALxfTr\nYgfLmu88negl9Or8h23a/MS3lJmZmeilAADQqyEd+EvjshK9hF5d7HxPH330UaKXAQDANfEjegAA\nLETgAQCwEIEHAMBCBB4AAAsReAAALETgAQCwEIEHAMBCBB4AAAsReAAALETgAQCwEIEHAMBCBB4A\nAAsReAAALETgAQCwEIEHAMBCBB4AAAsReAAALETgAQCwEIEHAMBCBB4AAAsReAAALETgAQCwEIEH\nAMBCBB4AAAsReAAALETgAQCwEIEHAMBCBB4AAAsReAAALNRn4MvLyzV58mRNmzYteqytrU2rVq1S\nIBBQQUGB2tvbo7dVVFTI5/MpLy9PBw8ejB7/1a9+JZ/PJ5/PpxdeeGEAxgAAAJfrM/D33Xef6urq\nrjhWWVmplStXqrGxUcuXL1dlZaUkqaamRk1NTTp69Ki2bt2qsrIySVJra6sef/xxNTQ0aO/evXrs\nscd06tSpgZkGAABI+i+BX7JkiSZMmHDFsdraWpWWlkqSSkpKFA6HJUnhcDh6PD8/Xx0dHYpEItq9\ne7dWrlypz372s0pNTdXy5cu1e/fugZgFAAD8vxt+DT4SicjtdkuS3G63IpGIJMlxnOhxSfJ4PIpE\nInIcR1OmTLnqOAAAGDi8yQ4AAAvdcOAvPwN3HEcej+eq4/+5zev1yuPxyHGc6PFIJCKv19vfdQMA\ngD7ccOCDwaBCoZAkKRQKKRgMRo9XV1dLkvbu3avx48fL7XZrxYoV+v3vf6+zZ8+qvb1dr7/+ulau\nXBnHEQAAwKcl9XXj2rVrtWfPHp05c0Zer1cPPfSQNmzYoOLiYoVCIaWlpWnHjh2SpKKiItXX18vn\n8yklJUVVVVWSpClTpmjDhg3Kz8+Xy+XSxo0bNXny5IGfDACAEcxljDGJXkRvXC6Xvrz+5UQvo1fn\nPnhPWx9equzs7EQvBQBgCZfLpXgmmTfZAQBgIQIPAICFCDwAABYi8AAAWIjAAwBgIQIPAICFCDwA\nABYi8AAAWIjAAwBgIQIPAICFCDwAABYi8AAAWIjAAwBgIQIPAICFCDwAABYi8AAAWIjAAwBgIQIP\nAICFCDwAABYi8AAAWIjAAwBgIQIPAICFCDwAABYi8AAAWIjAAwBgIQIPAICFCDwAABYi8AAAWIjA\nAwBgIQIPAICFCDwAABYi8AAAWIjAAwBgIQIPAICFCDwAABYi8AAAWIjAAwBgIQIPAICFCDwAABYi\n8AAAWIjAAwBgIQIPAICFCDwAABaKOfCPPPKIMjMzlZWVpa9//evq6OjQiRMntHDhQvn9ft1zzz3q\n6uqSJF26dEnFxcXy+/1atGiRWlpa4jYAAAC4WkyBP3jwoHbu3Kl33nlHx44d05gxY1RVVaWKigo9\n+OCDOnLkiG6++WZt2bJFkrRlyxZNnDhRR44cUUVFhSoqKuI6BAAAuFJMgU9PT1dSUpLOnz+vjz76\nSBcuXND06dP15ptvas2aNZKkkpIShcNhSVJtba1KS0slSUVFRaqvr5cxJk4jAACAT4sp8GlpaXrk\nkUeUkZGhKVOmaMKECcrLy1NqaqpGjx4tSXK73YpEIpKkSCQit9stSUpOTlZqaqpOnz4dpxEAAMCn\nxRT4v//979q0aZPeffddOY6jM2fOaPfu3fFeGwAAiFFMgX/77bc1b948TZo0ScnJybr77rv1xz/+\nUWfPnlV3d7ckyXEceb1eSZLH44mezXd1dens2bOaNGlSnEYAAACfFlPgp0+frv379+v8+fMyxuiN\nN95QVlaW7rzzTr300kuSpFAopNWrV0uSgsGgQqGQJGnnzp1aunSpRo3iE3oAAAyUmCo7b948feMb\n39CsWbOUlZWlUaNG6f7779fmzZv1s5/9TH6/X+3t7XrggQckSffff7/a2trk9/v17LPPavPmzXEd\nAgAAXMllhujb2V0ul768/uVEL6NX5z54T1sfXqrs7OxELwUAYAmXyxXXT5jxc3IAACxE4AEAsBCB\nBwDAQgQeAAALEXgAACxE4AEAsBCBBwDAQgQeAAALEXgAACxE4AEAsBCBBwDAQgQeAAALEXgAACxE\n4AEAsBCBBwDAQgQeAAALEXgAACxE4AEAsBCBBwDAQgQeAAALEXgAACxE4AEAsBCBBwDAQgQeAAAL\nEXgAACxE4AEAsBCBBwDAQgQeAAALEXgAACxE4AEAsBCBBwDAQgQeAAALEXgAACxE4AEAsBCBBwDA\nQgQeAAALEXgAACxE4AEAsBCBBwDAQgQeAAALEXgAACxE4AEAsFDMgf/ggw9UWFioQCAgn8+nxsZG\ntbW1adWqVQoEAiooKFB7e3v06ysqKuTz+ZSXl6eDBw/GZfEAAKB3MQf+29/+toLBoBobG3X48GHd\nfvvtqqys1MqVK9XY2Kjly5ersrJSklRTU6OmpiYdPXpUW7duVVlZWbzWDwAAehFT4N9//33t2bNH\n3/rWtyRJo0eP1k033aTa2lqVlpZKkkpKShQOhyVJ4XA4ejw/P18dHR1yHCce6wcAAL2IKfBNTU2a\nPHmySkpKlJOTo29+85v68MMPFYlE5Ha7JUlut1uRSESS5DhO9LgkeTye6G0AACD+Ygp8T0+PDh06\npO9+97t65513NG7cOG3cuDHeawMAADGKKfBer1cTJ07UkiVLJElf+9rXdOjQIXm93ivO2j0ej6Sr\nz9gvvw0AAMRfzIH3er1qbGyUJNXX1ysrK0urV69WKBSSJIVCIQWDQUlSMBhUdXW1JGnv3r0aP378\nFT+yBwAA8ZUU6x23b9+u8vJyXbhwQVOnTtWvf/1rGWNUXFysUCiktLQ07dixQ5JUVFSk+vp6+Xw+\npaSkqKqqKm4DAACAq7mMMSbRi+iNy+XSl9e/nOhl9OrcB+9p68NLlZ2dneilAAAs4XK5FM8kcyU7\nAAAsROABALAQgQcAwEIEHgAACxF4AAAsROABALAQgQcAwEIEHgAACxF4AAAsROABALAQgQcAwEIE\nHgAACxF4AAAsROABALAQgQcAwEIEHgAACxF4AAAsROABALAQgQcAwEIEHgAACxF4AAAsROABALAQ\ngQcAwEIEHgAACxF4AAAsROABALAQgQcAwEIEHgAACxF4AAAsROABALAQgQcAwEIEHgAACxF4AAAs\nROABALAQgQcAwEIEHgAACxF4AAAsROABALAQgQcAwEIEHgAACxF4AAAsROABALBQvwLf09OjefPm\nadmyZZKktrY2rVq1SoFAQAUFBWpvb49+bUVFhXw+n/Ly8nTw4MH+rRoAAPSpX4HfsmWLMjMz5XK5\nJEmVlZVauXKlGhsbtXz5clVWVkqSampq1NTUpKNHj2rr1q0qKyvr98IBAMC1xRz49957T7/97W+1\nbt06GWMkSbW1tSotLZUklZSUKBwOS5LC4XD0eH5+vjo6OuQ4Tn/XDgAAriHmwD/00EN66qmnNGrU\nJw8RiUTkdrslSW63W5FIRJLkOE70uCR5PJ7obQAAIP5iCnxdXZ1SU1M1Z86c6Nk7AAAYOpJiudOf\n//xnhcNhTZs2TRcvXlR7e7u++tWvRs/MMzIy5DiOPB6PpKvP2C+/DQAAxF9MZ/A//vGP9c9//lMn\nTpxQdXW18vPz9corrygYDCoUCkmSQqGQgsGgJCkYDKq6ulqStHfvXo0fP/6KH9kDAID4iukM/nLG\nmOi76Dds2KDi4mKFQiGlpaVpx44dkqSioiLV19fL5/MpJSVFVVVV/X1aAADQB5cZoi+iu1wufXn9\ny4leRq/OffCetj68VNnZ2YleCgDAEi6XK67va+NKdgAAWIjAAwBgIQIPAICFCDwAABYi8AAAWIjA\nAwBgIQIPAICFCDwAABYi8AAAWIjAAwBgIQIPAICFCDwAABYi8AAAWIjAAwBgIQIPAICFCDwAABYi\n8AAAWIjAAwBgIQIPAICFCDwAABYi8AAAWIjAAwBgIQIPAICFCDwAABYi8AAAWIjAAwBgIQIPAICF\nCDwAABYi8AAAWIjAAwBgIQIPAICFCDwAABYi8AAAWIjAAwBgIQIPAICFCDwAABYi8AAAWIjAAwBg\nIQIPAICFCDwAABYi8AAAWIjAAwBgoZgCf/LkSS1dulQzZ87UjBkz9MQTT0iS2tratGrVKgUCARUU\nFKi9vT16n4qKCvl8PuXl5engwYNxWTwAAOhdTIFPSkrSs88+q8OHD+tvf/ubqqur9dZbb6myslIr\nV65UY2Ojli9frsrKSklSTU2NmpqadPToUW3dulVlZWXxnAEAAHxKTIG/5ZZbNHPmTEnS2LFj5ff7\n1draqtraWpWWlkqSSkpKFA6HJUnhcDh6PD8/Xx0dHXIcJx7rBwAAvej3a/DNzc1qaGjQihUrFIlE\n5Ha7JUlut1uRSESS5DhO9LgkeTye6G0AACD++hX4CxcuaO3atdq8ebMmTpwYrzUBAIB+ijnw3d3d\nKi4uVklJiQoLCyVdeWbuOI48Hs9Vxz99GwAAiL+YA79u3TpNmzZN69evjx4LBoMKhUKSpFAopGAw\nGD1eXV0tSdq7d6/Gjx9/xY/sAQBAfCXFcqc9e/Zo+/btCgQCys3NlSRt3LhRGzZsUHFxsUKhkNLS\n0rRjxw5JUlFRkerr6+Xz+ZSSkqKqqqr4TQAAAK4SU+AXL16snp6eXm/bvXt3r8e3bNkSy1MBAIAY\ncCU7AAAsROABALAQgQcAwEIEHgAACxF4AAAsROABALAQgQcAwEIEHgAACxF4AAAsROABALAQgQcA\nwEIEHgAACxF4AAAsROABALAQgQcAwEIEHgAACxF4AAAsROABALAQgQcAwEIEHgAACxF4AAAsROAB\nALAQgQcAwEJJiV7AcNTT/ZGam5s1evToRC/lmqZOnaoxY8YkehkAgAQh8DG4eO59PfGL9zU29USi\nl9KrC2dP6383lWrGjBmJXgoAIEEIfIzGpt6qmya4E70MAAB6ReAt1NP9kU6cGJo/XbgcLyMAwMAh\n8Ba6eO59Pf7zvRqb+vdEL+WaeBkBAAYWgbcULyEAwMjGx+QAALAQgQcAwEIEHgAACxF4AAAsROAB\nALAQ76JHQgyHz+rzOX0AwxmBR0IM9c/q8zl9AMMdgUfC8Fl9ABg4vAYPAICFCDwAABYi8AAAWIjA\nAwBgId5kBwBDUGdnp5qbmxO9jD7xUdKhbVADX1dXp+9///vq7u5WWVmZfvCDHwzm0wPAsNHc3Kxv\nPhrS2NRbE72UXvFR0qFv0AJ/6dIlrVu3Tnv27FF6errmzp2rgoIC5ebmDtYSgOvGhXgwFAzlj5IO\nh+8RaWR/nwxa4Pft26fMzExlZGRIkgoLCxUOhwk8hiQuxAP0bah/j0h8nwxa4CORiNzuT/4m6vF4\n9Je//GWwnh64YZw99d9QPnsa6q9xD4f9HcrfIxjEwLtcrhu/z9l3BmAl/Tf6wj90IdGL6MO/P2yT\ndON/3oNpqK9xqK/vg9YmPfzMMaXcNDHRS7mmi+fa9D/f/6qmTZuW6KX06sSJE3r4mVeG7J/h2VPv\n6ub0oXvmOdS/R6SPz+BHskELvMfjkeM40X+PRCLyer193ue3v/zRQC8LwAD60u+fS/QSMMJ9fteG\nRC8hYQYt8PPmzdPx48fV0tKi9PR07dq1S1VVVdf8emPMYC0NAADrDFrgU1JStG3bNt11113q7u7W\nvffeq7y8vMF6egAARhSX4VQZAADrcKlaAAAsROABALAQgQcAwEKDEvi6ujr5/X5lZ2fr6aefvur2\nS5cuqbi4WH6/X4sWLVJLS0v0tk2bNik7O1t+v1+vvfbadT/mUDIQ848ZM0a5ubnKzc1VUVHRoMwR\nq1jnb2tr07JlyzR+/Hjdd999V9znwIEDys3Nlc/n0/e+971BmSNWAzH/1KlTo/u/cOHCQZkjFrHO\nXl9frzlz5igQCCgnJ0cvv/xy9D4jYe/7mn+47L0U+/wtLS2aM2eOcnNzlZmZqQ0bPvmo20jY/77m\nv6H9NwPs4sWLxuPxmJaWFtPZ2WkCgYD561//esXXPPPMM+Y73/mOMcaYF1980dx9993GGGP2799v\nAoGA6erqMs3Nzcbj8ZjOzs7resyhYiDmN8aYqVOnDu4gMerP/OfPnzdvvfWWef75501ZWdkV9/H7\n/aahocEYY8yXvvQls2vXrkGY5sYN1PzDYf/7M/uRI0fMqVOnjDHGHDt2zNx8883m0qVLxpiRsfd9\nzT8c9t6Y/s3f2dlpurq6jDHGnDt3zmRkZJjDhw8bY0bG/vc1/43s/4CfwV9+Dfrk5OToNegvV1tb\nq9LSUklSUVGR6uvr1dPTo3A4rDVr1igpKUm33Xabpk+frn379l3XYw4V8Z7/7bffTsQYMYt1fmOM\nxo4dq8WLF+szn/nMFV//j3/8Q+fPn9eCBQskSffcc491+9/X/MNFf2bPycnRrbd+/FvUPv/5zysp\nKUkdHR0jZu+vNf9w0p/5k5OTlZT08ae4//3vfyspKUmf+9znRsz+X2v+GzXgge/tGvSRSOSaX5Oc\nnKzU1FSdPn1ajuNoypQpV93XcZz/+phDxUDML0mnTp3SnDlzNHfuXNXU1AzCJLHpz/zX+5hut9vK\n/e+Ly+XSggULNGvWLD333NC8Wly8Zn/ppZeUnZ2tW265ZUTu/eXzS8Nj76X+z9/a2qpAIKCMjAw9\n+OCDSk9PH1H739v80o3t/4Bf6CaWa9DbZKDmb2lp0aRJk9TU1KQvfvGLmjlz5pD8jUns/8DMv2/f\nPk2aNEknT57UsmXLNGPGDK1YsWJAnitW8Zj92LFjevTRR/W73/0uDisaXAM1/3DYe6n/86enp6ux\nsVGO42jx4sUKBoNxWtngGIj577jjDjU0NOjWW2+9rv0f8DP467kG/eV/s+nq6tLZs2c1adKkq+7r\nOI68Xm9M17VPlHjP7/F4JEmTJk2SJGVmZmrJkiU6cODAQI8Sk/7M/x+f/kbp689lqBmI+aVP9j8t\nLU1f+cpXhuRLN/2d/eTJkyosLNQLL7ygO+64o9fHtHnve5tfGh57L8Xnv33p47P0BQsW6MCBA/J6\nvSNm///jP/Pv379fkqIv3VzP/g944C+/Bn1nZ6d27dql1atXX/E1wWBQoVBIkrRz504tXbpUo0eP\nVjAYVE1Njbq6utTc3Kzjx49r/vz51/WYQ8VAzH/u3Dl1dnZK+vhH9Q0NDfL5fIM+2/WIdf5Roz75\nT9N86mKLGRkZGjdunBoaGmSM0W9+85sh+7f7gZi/s7NT58+flySdO3dOr7/++pDc//7M3tHRoWAw\nqI0bN2rRokXRrx8pe3+t+YfL3kv9m99xnOj/4/71r3+poaFBOTk58nq9I2L/e5vf7/ff+P7H/BbB\nG1BbW2tycnJMVlaW+elPf2qMMebxxx83r776qjHm43cbrl271uTk5JiFCxeaEydORO/7k5/8xGRl\nZZmcnBxTV1fX52MOVfGe/9ChQ2b27NkmEAiYzMxMs3nz5kGf6Ub0Z363220mTpxoxo0bZzweT/Td\ns/v37zezZ8822dnZ5oEHHhj0mW5EvOc/ffq0ycvLM7NmzTK33367+dGPfpSIsa5LrLM/+eSTZuzY\nsWb27NnRf1pbW40xI2PvrzX/qVOnhs3eGxP7/K+88orx+/0mEAgYn89ntm3bFn3MkbD/15r/Rvef\na9EDAGAhrmQHAICFCDwAABYi8AAAWIjAAwBgIQIPAICFCDwAABYi8AAAWIjAAwBgof8Dg53n4W78\nhTkAAAAASUVORK5CYII=\n",
       "text": [
        "<matplotlib.figure.Figure at 0x31c62c10>"
       ]
      }
     ],
     "prompt_number": 77
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "(overlaps - mean) / std"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "ename": "NameError",
       "evalue": "name 'mean' is not defined",
       "output_type": "pyerr",
       "traceback": [
        "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[1;32m<ipython-input-33-405fe17bb9ed>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[1;33m(\u001b[0m\u001b[0moverlaps\u001b[0m \u001b[1;33m-\u001b[0m \u001b[0mmean\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m/\u001b[0m \u001b[0mstd\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[1;31mNameError\u001b[0m: name 'mean' is not defined"
       ]
      }
     ],
     "prompt_number": 33
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "HTML(intersect_df.div(clip_size, axis=0).to_html())"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>SRR334701.peaks.bed</th>\n",
        "      <th>SRR334703.peaks.bed</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>293T_hnRNPA1_WT_rep1</th>\n",
        "      <td> 0.126978</td>\n",
        "      <td> 0.024509</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>293T_hnRNPF_WT_rep1</th>\n",
        "      <td> 0.041421</td>\n",
        "      <td> 0.006546</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>293T_hnRNPM_WT_rep1</th>\n",
        "      <td> 0.046341</td>\n",
        "      <td> 0.009999</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>293T_hnRNPU_WT_rep1</th>\n",
        "      <td> 0.026957</td>\n",
        "      <td> 0.004030</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>C2C12_Cstf64_WT_rep1</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>FOX2</th>\n",
        "      <td> 0.021213</td>\n",
        "      <td> 0.002740</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293FlpIn_FMR1-iso1_WT_rep1</th>\n",
        "      <td> 0.123066</td>\n",
        "      <td> 0.011112</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293FlpIn_FMR1-iso7_WT_rep1</th>\n",
        "      <td> 0.101175</td>\n",
        "      <td> 0.011132</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago2_WT_rep1</th>\n",
        "      <td> 0.152288</td>\n",
        "      <td> 0.089542</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago2_WT_rep2</th>\n",
        "      <td> 0.140062</td>\n",
        "      <td> 0.026512</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago2_WT_rep3</th>\n",
        "      <td> 0.123848</td>\n",
        "      <td> 0.026159</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago2_WT_rep4</th>\n",
        "      <td> 0.185892</td>\n",
        "      <td> 0.040445</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago3_WT_rep1</th>\n",
        "      <td> 0.102305</td>\n",
        "      <td> 0.059798</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago4_WT_rep1</th>\n",
        "      <td> 0.274205</td>\n",
        "      <td> 0.100353</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_DGCR8_WT_rep1</th>\n",
        "      <td> 0.535942</td>\n",
        "      <td> 0.114483</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_DGCR8_overexpressed_rep1</th>\n",
        "      <td> 0.373279</td>\n",
        "      <td> 0.084676</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_HuR_WT_rep1</th>\n",
        "      <td> 0.003301</td>\n",
        "      <td> 0.001264</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_HuR_WT_rep2</th>\n",
        "      <td> 0.005235</td>\n",
        "      <td> 0.001487</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_HuR_WT_rep3</th>\n",
        "      <td> 0.022572</td>\n",
        "      <td> 0.002270</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_HuR_WT_rep6</th>\n",
        "      <td> 0.021273</td>\n",
        "      <td> 0.010795</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_IGF2BP1_WT_rep1</th>\n",
        "      <td> 0.130260</td>\n",
        "      <td> 0.011580</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_IGF2BP2_WT_rep1</th>\n",
        "      <td> 0.121782</td>\n",
        "      <td> 0.014012</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_IGF2BP3_WT_rep1</th>\n",
        "      <td> 0.126321</td>\n",
        "      <td> 0.011663</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_LIN28B_WT_rep1</th>\n",
        "      <td> 0.075621</td>\n",
        "      <td> 0.027188</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_PAPD5_WT_rep1</th>\n",
        "      <td> 0.418605</td>\n",
        "      <td> 0.091875</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Pumilio2_WT_rep1</th>\n",
        "      <td> 0.052868</td>\n",
        "      <td> 0.026909</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_QIK_WT_rep1</th>\n",
        "      <td> 0.049776</td>\n",
        "      <td> 0.020320</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_TNRC6A_WT_rep1</th>\n",
        "      <td> 0.125442</td>\n",
        "      <td> 0.071996</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_TNRC6B_WT_rep1</th>\n",
        "      <td> 0.106159</td>\n",
        "      <td> 0.086630</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_TNRC6C_WT_rep1</th>\n",
        "      <td> 0.234375</td>\n",
        "      <td> 0.102083</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_hnRNPU_WT_rep1</th>\n",
        "      <td> 0.050440</td>\n",
        "      <td> 0.004083</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HeLa_Ago2_PTB_kd_rep1</th>\n",
        "      <td> 0.183962</td>\n",
        "      <td> 0.017973</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HeLa_Ago2_WT_rep1</th>\n",
        "      <td> 0.246029</td>\n",
        "      <td> 0.022486</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HeLa_eIF4AIII_WT_rep1</th>\n",
        "      <td> 0.095455</td>\n",
        "      <td> 0.007108</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MDAMB231_hnRNPA2B1_WT_rep1</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEF_SRSF1_WT_rep1</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MEF_SRSF1_siSRSF2_rep1</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MouseCortex_Nova_WT_rep1</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MouseHippocampus_Nova_WT_rep1</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>NIH3T3_CIRBP_WT_rep1</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>P16Brain_Nova_WT_rep1</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR034467</th>\n",
        "      <td> 0.046551</td>\n",
        "      <td> 0.013431</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR034468</th>\n",
        "      <td> 0.024040</td>\n",
        "      <td> 0.009673</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR034469</th>\n",
        "      <td> 0.033320</td>\n",
        "      <td> 0.009330</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048950</th>\n",
        "      <td> 0.164611</td>\n",
        "      <td> 0.032693</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048951</th>\n",
        "      <td> 0.180410</td>\n",
        "      <td> 0.016205</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048952</th>\n",
        "      <td> 0.267730</td>\n",
        "      <td> 0.035845</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048953</th>\n",
        "      <td> 0.351955</td>\n",
        "      <td> 0.067505</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048954</th>\n",
        "      <td> 0.277778</td>\n",
        "      <td> 0.191358</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048955</th>\n",
        "      <td> 0.099585</td>\n",
        "      <td> 0.019710</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048956</th>\n",
        "      <td> 0.109998</td>\n",
        "      <td> 0.007113</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048957</th>\n",
        "      <td> 0.128523</td>\n",
        "      <td> 0.027785</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048958</th>\n",
        "      <td> 0.126267</td>\n",
        "      <td> 0.013379</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048959</th>\n",
        "      <td> 0.168887</td>\n",
        "      <td> 0.030801</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048960</th>\n",
        "      <td> 0.443959</td>\n",
        "      <td> 0.116448</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048961</th>\n",
        "      <td> 0.382022</td>\n",
        "      <td> 0.168539</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048962</th>\n",
        "      <td> 0.083030</td>\n",
        "      <td> 0.020179</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048963</th>\n",
        "      <td> 0.097832</td>\n",
        "      <td> 0.010393</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048964</th>\n",
        "      <td> 0.172494</td>\n",
        "      <td> 0.020229</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048965</th>\n",
        "      <td> 0.414510</td>\n",
        "      <td> 0.049317</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048966</th>\n",
        "      <td> 0.521375</td>\n",
        "      <td> 0.097584</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048967</th>\n",
        "      <td> 0.053024</td>\n",
        "      <td> 0.034257</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048968</th>\n",
        "      <td> 0.038035</td>\n",
        "      <td> 0.032752</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048969</th>\n",
        "      <td> 0.087856</td>\n",
        "      <td> 0.050138</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048970</th>\n",
        "      <td> 0.100678</td>\n",
        "      <td> 0.044530</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048971</th>\n",
        "      <td> 0.165385</td>\n",
        "      <td> 0.072527</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048972</th>\n",
        "      <td> 0.060354</td>\n",
        "      <td> 0.065557</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048973</th>\n",
        "      <td> 0.155697</td>\n",
        "      <td> 0.068271</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048974</th>\n",
        "      <td> 0.095465</td>\n",
        "      <td> 0.126492</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048975</th>\n",
        "      <td> 0.234279</td>\n",
        "      <td> 0.125771</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048976</th>\n",
        "      <td> 0.129189</td>\n",
        "      <td> 0.064594</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048977</th>\n",
        "      <td> 0.086735</td>\n",
        "      <td> 0.147959</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048978</th>\n",
        "      <td> 0.154825</td>\n",
        "      <td> 0.116649</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048979</th>\n",
        "      <td> 0.396226</td>\n",
        "      <td> 0.158805</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048980</th>\n",
        "      <td> 0.099918</td>\n",
        "      <td> 0.020805</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048981</th>\n",
        "      <td> 0.123838</td>\n",
        "      <td> 0.044998</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048982</th>\n",
        "      <td> 0.078886</td>\n",
        "      <td> 0.080046</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048983</th>\n",
        "      <td> 0.126761</td>\n",
        "      <td> 0.104009</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048984</th>\n",
        "      <td> 0.108720</td>\n",
        "      <td> 0.125708</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048985</th>\n",
        "      <td> 0.048465</td>\n",
        "      <td> 0.082391</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048986</th>\n",
        "      <td> 0.250299</td>\n",
        "      <td> 0.090203</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048987</th>\n",
        "      <td> 0.193939</td>\n",
        "      <td> 0.137374</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR053640</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR053641</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR053642</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR053643</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR053644</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR053645</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR053646</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR053647</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR053648</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR053649</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR053650</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR053651</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR053652</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR053653</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR053654</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR053655</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR053657</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR053658</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR053659</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR053660</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR053661</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR053662</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR053663</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR053664</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR072951</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR072952</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR072953</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR072954</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR072955</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR072956</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR107031</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR107032</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189775</th>\n",
        "      <td> 0.002751</td>\n",
        "      <td> 0.001046</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189776</th>\n",
        "      <td> 0.005447</td>\n",
        "      <td> 0.001958</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189777</th>\n",
        "      <td> 0.006615</td>\n",
        "      <td> 0.002182</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189778</th>\n",
        "      <td> 0.007789</td>\n",
        "      <td> 0.002967</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189779</th>\n",
        "      <td> 0.043031</td>\n",
        "      <td> 0.004470</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189780</th>\n",
        "      <td> 0.019885</td>\n",
        "      <td> 0.002898</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189781</th>\n",
        "      <td> 0.034221</td>\n",
        "      <td> 0.005458</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189782</th>\n",
        "      <td> 0.230603</td>\n",
        "      <td> 0.057905</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189783</th>\n",
        "      <td> 0.600049</td>\n",
        "      <td> 0.147117</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189784</th>\n",
        "      <td> 0.165112</td>\n",
        "      <td> 0.034874</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189785</th>\n",
        "      <td> 0.139529</td>\n",
        "      <td> 0.044589</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189786</th>\n",
        "      <td> 0.212548</td>\n",
        "      <td> 0.053876</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189787</th>\n",
        "      <td> 0.171928</td>\n",
        "      <td> 0.060794</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR248532</th>\n",
        "      <td> 0.009124</td>\n",
        "      <td> 0.003157</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR309285</th>\n",
        "      <td> 0.086702</td>\n",
        "      <td> 0.044523</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR309286</th>\n",
        "      <td> 0.049277</td>\n",
        "      <td> 0.035262</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR309287</th>\n",
        "      <td> 0.025739</td>\n",
        "      <td> 0.012596</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR390466</th>\n",
        "      <td> 0.098788</td>\n",
        "      <td> 0.009326</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR390467</th>\n",
        "      <td> 0.029376</td>\n",
        "      <td> 0.001964</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398195</th>\n",
        "      <td> 0.116796</td>\n",
        "      <td> 0.007786</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398196</th>\n",
        "      <td> 0.186722</td>\n",
        "      <td> 0.037805</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398197</th>\n",
        "      <td> 0.164501</td>\n",
        "      <td> 0.032319</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398198</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398199</th>\n",
        "      <td> 0.093829</td>\n",
        "      <td> 0.017218</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398200</th>\n",
        "      <td> 0.065839</td>\n",
        "      <td> 0.015992</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398201</th>\n",
        "      <td> 0.097956</td>\n",
        "      <td> 0.018319</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398202</th>\n",
        "      <td> 0.304405</td>\n",
        "      <td> 0.061233</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398203</th>\n",
        "      <td> 0.068469</td>\n",
        "      <td> 0.012528</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398204</th>\n",
        "      <td> 0.082866</td>\n",
        "      <td> 0.024420</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398205</th>\n",
        "      <td> 0.058356</td>\n",
        "      <td> 0.011743</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398206</th>\n",
        "      <td> 0.107236</td>\n",
        "      <td> 0.020157</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398207</th>\n",
        "      <td> 0.133750</td>\n",
        "      <td> 0.026226</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398208</th>\n",
        "      <td> 0.797242</td>\n",
        "      <td> 0.143552</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398209</th>\n",
        "      <td> 0.361967</td>\n",
        "      <td> 0.066647</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR408236</th>\n",
        "      <td> 0.061919</td>\n",
        "      <td> 0.005639</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR408237</th>\n",
        "      <td> 0.052488</td>\n",
        "      <td> 0.004902</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR427115</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR427116</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR452496</th>\n",
        "      <td> 0.586824</td>\n",
        "      <td> 0.113874</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR452497</th>\n",
        "      <td> 0.221053</td>\n",
        "      <td> 0.094737</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR458758</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR458759</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR458760</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR490650</th>\n",
        "      <td> 0.024438</td>\n",
        "      <td> 0.007845</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR492994</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR492995</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR518495</th>\n",
        "      <td> 0.121553</td>\n",
        "      <td> 0.049320</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR518496</th>\n",
        "      <td> 0.036343</td>\n",
        "      <td> 0.021731</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR518497</th>\n",
        "      <td> 0.781054</td>\n",
        "      <td> 0.181932</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR518498</th>\n",
        "      <td> 0.857076</td>\n",
        "      <td> 0.174321</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527727</th>\n",
        "      <td> 0.159465</td>\n",
        "      <td> 0.014839</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527728</th>\n",
        "      <td> 0.223639</td>\n",
        "      <td> 0.036764</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527729</th>\n",
        "      <td> 0.121142</td>\n",
        "      <td> 0.013186</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527730</th>\n",
        "      <td> 0.103996</td>\n",
        "      <td> 0.010978</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527731</th>\n",
        "      <td> 0.116196</td>\n",
        "      <td> 0.048548</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527732</th>\n",
        "      <td> 0.287267</td>\n",
        "      <td> 0.043232</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR533660</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR533661</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR533662</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR533663</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR533664</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR556762</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR556763</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR556766</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR567525</th>\n",
        "      <td> 0.142053</td>\n",
        "      <td> 0.010016</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR567526</th>\n",
        "      <td> 0.101041</td>\n",
        "      <td> 0.007756</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR618490</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR618491</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628446</th>\n",
        "      <td> 0.092950</td>\n",
        "      <td> 0.005923</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628447</th>\n",
        "      <td> 0.131009</td>\n",
        "      <td> 0.019681</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628448</th>\n",
        "      <td> 0.152423</td>\n",
        "      <td> 0.019832</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628449</th>\n",
        "      <td> 0.141750</td>\n",
        "      <td> 0.021357</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628450</th>\n",
        "      <td> 0.298899</td>\n",
        "      <td> 0.027597</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628451</th>\n",
        "      <td> 0.168760</td>\n",
        "      <td> 0.023641</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628452</th>\n",
        "      <td> 0.124476</td>\n",
        "      <td> 0.017483</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628453</th>\n",
        "      <td> 0.120464</td>\n",
        "      <td> 0.016307</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628454</th>\n",
        "      <td> 0.327715</td>\n",
        "      <td> 0.035512</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR764666</th>\n",
        "      <td> 0.123226</td>\n",
        "      <td> 0.038619</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR764667</th>\n",
        "      <td> 0.042620</td>\n",
        "      <td> 0.021830</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR764668</th>\n",
        "      <td> 0.085789</td>\n",
        "      <td> 0.028596</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR764669</th>\n",
        "      <td> 0.076737</td>\n",
        "      <td> 0.028367</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR765756</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR765757</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR765758</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR765759</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR765760</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR765761</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>WholeBrain_Mbnl_WT_rep1</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>mESC_Ago2_WT_rep1</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>mESC_Ago2_dicerNull_rep1</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>mESC_LIN28A_WT_rep1</th>\n",
        "      <td>      NaN</td>\n",
        "      <td>      NaN</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 38,
       "text": [
        "<IPython.core.display.HTML at 0x2aaac9073190>"
       ]
      }
     ],
     "prompt_number": 38
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "HTML(intersect_df.div(pause_size, axis=1).to_html())"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>SRR334701.peaks.bed</th>\n",
        "      <th>SRR334703.peaks.bed</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>293T_hnRNPA1_WT_rep1</th>\n",
        "      <td> 0.032314</td>\n",
        "      <td> 0.080225</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>293T_hnRNPF_WT_rep1</th>\n",
        "      <td> 0.065389</td>\n",
        "      <td> 0.132919</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>293T_hnRNPM_WT_rep1</th>\n",
        "      <td> 0.023361</td>\n",
        "      <td> 0.064831</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>293T_hnRNPU_WT_rep1</th>\n",
        "      <td> 0.100235</td>\n",
        "      <td> 0.192718</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>FOX2</th>\n",
        "      <td> 0.202748</td>\n",
        "      <td> 0.336886</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293FlpIn_FMR1-iso1_WT_rep1</th>\n",
        "      <td> 0.320935</td>\n",
        "      <td> 0.372706</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293FlpIn_FMR1-iso7_WT_rep1</th>\n",
        "      <td> 0.192667</td>\n",
        "      <td> 0.272647</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago2_WT_rep1</th>\n",
        "      <td> 0.005363</td>\n",
        "      <td> 0.040557</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago2_WT_rep2</th>\n",
        "      <td> 0.104447</td>\n",
        "      <td> 0.254292</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago2_WT_rep3</th>\n",
        "      <td> 0.061568</td>\n",
        "      <td> 0.167259</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago2_WT_rep4</th>\n",
        "      <td> 0.066539</td>\n",
        "      <td> 0.186205</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago3_WT_rep1</th>\n",
        "      <td> 0.006537</td>\n",
        "      <td> 0.049142</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago4_WT_rep1</th>\n",
        "      <td> 0.017860</td>\n",
        "      <td> 0.084073</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_DGCR8_WT_rep1</th>\n",
        "      <td> 0.046331</td>\n",
        "      <td> 0.127294</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_DGCR8_overexpressed_rep1</th>\n",
        "      <td> 0.029322</td>\n",
        "      <td> 0.085554</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_HuR_WT_rep1</th>\n",
        "      <td> 0.006675</td>\n",
        "      <td> 0.032860</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_HuR_WT_rep2</th>\n",
        "      <td> 0.021957</td>\n",
        "      <td> 0.080225</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_HuR_WT_rep3</th>\n",
        "      <td> 0.268505</td>\n",
        "      <td> 0.347247</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_HuR_WT_rep6</th>\n",
        "      <td> 0.006168</td>\n",
        "      <td> 0.040261</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_IGF2BP1_WT_rep1</th>\n",
        "      <td> 0.258907</td>\n",
        "      <td> 0.296033</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_IGF2BP2_WT_rep1</th>\n",
        "      <td> 0.162033</td>\n",
        "      <td> 0.239787</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_IGF2BP3_WT_rep1</th>\n",
        "      <td> 0.263741</td>\n",
        "      <td> 0.313203</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_LIN28B_WT_rep1</th>\n",
        "      <td> 0.019333</td>\n",
        "      <td> 0.089402</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_PAPD5_WT_rep1</th>\n",
        "      <td> 0.033557</td>\n",
        "      <td> 0.094731</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Pumilio2_WT_rep1</th>\n",
        "      <td> 0.008953</td>\n",
        "      <td> 0.058615</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_QIK_WT_rep1</th>\n",
        "      <td> 0.014546</td>\n",
        "      <td> 0.076377</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_TNRC6A_WT_rep1</th>\n",
        "      <td> 0.006537</td>\n",
        "      <td> 0.048253</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_TNRC6B_WT_rep1</th>\n",
        "      <td> 0.004879</td>\n",
        "      <td> 0.051214</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_TNRC6C_WT_rep1</th>\n",
        "      <td> 0.015536</td>\n",
        "      <td> 0.087034</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_hnRNPU_WT_rep1</th>\n",
        "      <td> 0.070797</td>\n",
        "      <td> 0.073712</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HeLa_Ago2_PTB_kd_rep1</th>\n",
        "      <td> 0.288828</td>\n",
        "      <td> 0.362937</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HeLa_Ago2_WT_rep1</th>\n",
        "      <td> 0.289104</td>\n",
        "      <td> 0.339846</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HeLa_eIF4AIII_WT_rep1</th>\n",
        "      <td> 0.484948</td>\n",
        "      <td> 0.464476</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MDAMB231_hnRNPA2B1_WT_rep1</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR034467</th>\n",
        "      <td> 0.007020</td>\n",
        "      <td> 0.026051</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR034468</th>\n",
        "      <td> 0.003890</td>\n",
        "      <td> 0.020130</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR034469</th>\n",
        "      <td> 0.005754</td>\n",
        "      <td> 0.020722</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048950</th>\n",
        "      <td> 0.013211</td>\n",
        "      <td> 0.033748</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048951</th>\n",
        "      <td> 0.086862</td>\n",
        "      <td> 0.100355</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048952</th>\n",
        "      <td> 0.064123</td>\n",
        "      <td> 0.110420</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048953</th>\n",
        "      <td> 0.017400</td>\n",
        "      <td> 0.042925</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048954</th>\n",
        "      <td> 0.003107</td>\n",
        "      <td> 0.027531</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048955</th>\n",
        "      <td> 0.002210</td>\n",
        "      <td> 0.005625</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048956</th>\n",
        "      <td> 0.044490</td>\n",
        "      <td> 0.037004</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048957</th>\n",
        "      <td> 0.022464</td>\n",
        "      <td> 0.062463</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048958</th>\n",
        "      <td> 0.073421</td>\n",
        "      <td> 0.100059</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048959</th>\n",
        "      <td> 0.022464</td>\n",
        "      <td> 0.052694</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048960</th>\n",
        "      <td> 0.014040</td>\n",
        "      <td> 0.047365</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048961</th>\n",
        "      <td> 0.005478</td>\n",
        "      <td> 0.031083</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048962</th>\n",
        "      <td> 0.011554</td>\n",
        "      <td> 0.036116</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048963</th>\n",
        "      <td> 0.080809</td>\n",
        "      <td> 0.110420</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048964</th>\n",
        "      <td> 0.083801</td>\n",
        "      <td> 0.126406</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048965</th>\n",
        "      <td> 0.046815</td>\n",
        "      <td> 0.071640</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048966</th>\n",
        "      <td> 0.012912</td>\n",
        "      <td> 0.031083</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048967</th>\n",
        "      <td> 0.004097</td>\n",
        "      <td> 0.034044</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048968</th>\n",
        "      <td> 0.003314</td>\n",
        "      <td> 0.036708</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048969</th>\n",
        "      <td> 0.004396</td>\n",
        "      <td> 0.032268</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048970</th>\n",
        "      <td> 0.004787</td>\n",
        "      <td> 0.027235</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048971</th>\n",
        "      <td> 0.006928</td>\n",
        "      <td> 0.039076</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048972</th>\n",
        "      <td> 0.001335</td>\n",
        "      <td> 0.018650</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048973</th>\n",
        "      <td> 0.007296</td>\n",
        "      <td> 0.041149</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048974</th>\n",
        "      <td> 0.000921</td>\n",
        "      <td> 0.015690</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048975</th>\n",
        "      <td> 0.004373</td>\n",
        "      <td> 0.030195</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048976</th>\n",
        "      <td> 0.007365</td>\n",
        "      <td> 0.047365</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048977</th>\n",
        "      <td> 0.000391</td>\n",
        "      <td> 0.008585</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048978</th>\n",
        "      <td> 0.003360</td>\n",
        "      <td> 0.032564</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048979</th>\n",
        "      <td> 0.011600</td>\n",
        "      <td> 0.059799</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048980</th>\n",
        "      <td> 0.016802</td>\n",
        "      <td> 0.044997</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048981</th>\n",
        "      <td> 0.007664</td>\n",
        "      <td> 0.035820</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048982</th>\n",
        "      <td> 0.001565</td>\n",
        "      <td> 0.020426</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048983</th>\n",
        "      <td> 0.002693</td>\n",
        "      <td> 0.028419</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048984</th>\n",
        "      <td> 0.002210</td>\n",
        "      <td> 0.032860</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048985</th>\n",
        "      <td> 0.000690</td>\n",
        "      <td> 0.015098</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048986</th>\n",
        "      <td> 0.009644</td>\n",
        "      <td> 0.044701</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048987</th>\n",
        "      <td> 0.004419</td>\n",
        "      <td> 0.040261</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189775</th>\n",
        "      <td> 0.002302</td>\n",
        "      <td> 0.011249</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189776</th>\n",
        "      <td> 0.004097</td>\n",
        "      <td> 0.018946</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189777</th>\n",
        "      <td> 0.013188</td>\n",
        "      <td> 0.055950</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189778</th>\n",
        "      <td> 0.012083</td>\n",
        "      <td> 0.059207</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189779</th>\n",
        "      <td> 0.225327</td>\n",
        "      <td> 0.301066</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189780</th>\n",
        "      <td> 0.089532</td>\n",
        "      <td> 0.167851</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189781</th>\n",
        "      <td> 0.084561</td>\n",
        "      <td> 0.173475</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189782</th>\n",
        "      <td> 0.041797</td>\n",
        "      <td> 0.134991</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189783</th>\n",
        "      <td> 0.056044</td>\n",
        "      <td> 0.176732</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189784</th>\n",
        "      <td> 0.054594</td>\n",
        "      <td> 0.148313</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189785</th>\n",
        "      <td> 0.012820</td>\n",
        "      <td> 0.052694</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189786</th>\n",
        "      <td> 0.043040</td>\n",
        "      <td> 0.140320</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189787</th>\n",
        "      <td> 0.018551</td>\n",
        "      <td> 0.084369</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR248532</th>\n",
        "      <td> 0.007250</td>\n",
        "      <td> 0.032268</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR309285</th>\n",
        "      <td> 0.003406</td>\n",
        "      <td> 0.022499</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR309286</th>\n",
        "      <td> 0.002509</td>\n",
        "      <td> 0.023091</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR309287</th>\n",
        "      <td> 0.003245</td>\n",
        "      <td> 0.020426</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR390466</th>\n",
        "      <td> 0.031693</td>\n",
        "      <td> 0.038484</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR390467</th>\n",
        "      <td> 0.046124</td>\n",
        "      <td> 0.039668</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398195</th>\n",
        "      <td> 0.004833</td>\n",
        "      <td> 0.004144</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398196</th>\n",
        "      <td> 0.009321</td>\n",
        "      <td> 0.024275</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398197</th>\n",
        "      <td> 0.022141</td>\n",
        "      <td> 0.055950</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398198</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398199</th>\n",
        "      <td> 0.011163</td>\n",
        "      <td> 0.026347</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398200</th>\n",
        "      <td> 0.008907</td>\n",
        "      <td> 0.027827</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398201</th>\n",
        "      <td> 0.023384</td>\n",
        "      <td> 0.056246</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398202</th>\n",
        "      <td> 0.015904</td>\n",
        "      <td> 0.041149</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398203</th>\n",
        "      <td> 0.012705</td>\n",
        "      <td> 0.029899</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398204</th>\n",
        "      <td> 0.004764</td>\n",
        "      <td> 0.018058</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398205</th>\n",
        "      <td> 0.018758</td>\n",
        "      <td> 0.048549</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398206</th>\n",
        "      <td> 0.012245</td>\n",
        "      <td> 0.029603</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398207</th>\n",
        "      <td> 0.030519</td>\n",
        "      <td> 0.076969</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398208</th>\n",
        "      <td> 0.022625</td>\n",
        "      <td> 0.052398</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398209</th>\n",
        "      <td> 0.028126</td>\n",
        "      <td> 0.066607</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR408236</th>\n",
        "      <td> 0.274443</td>\n",
        "      <td> 0.321492</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR408237</th>\n",
        "      <td> 0.291061</td>\n",
        "      <td> 0.349615</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR427115</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR427116</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR452496</th>\n",
        "      <td> 0.030957</td>\n",
        "      <td> 0.077265</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR452497</th>\n",
        "      <td> 0.006283</td>\n",
        "      <td> 0.034636</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR490650</th>\n",
        "      <td> 0.008102</td>\n",
        "      <td> 0.033452</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR518495</th>\n",
        "      <td> 0.007204</td>\n",
        "      <td> 0.037596</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR518496</th>\n",
        "      <td> 0.002233</td>\n",
        "      <td> 0.017170</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR518497</th>\n",
        "      <td> 0.028655</td>\n",
        "      <td> 0.085850</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR518498</th>\n",
        "      <td> 0.042096</td>\n",
        "      <td> 0.110124</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527727</th>\n",
        "      <td> 0.244614</td>\n",
        "      <td> 0.292777</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527728</th>\n",
        "      <td> 0.069025</td>\n",
        "      <td> 0.145944</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527729</th>\n",
        "      <td> 0.137866</td>\n",
        "      <td> 0.193014</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527730</th>\n",
        "      <td> 0.156762</td>\n",
        "      <td> 0.212848</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527731</th>\n",
        "      <td> 0.006721</td>\n",
        "      <td> 0.036116</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527732</th>\n",
        "      <td> 0.027987</td>\n",
        "      <td> 0.054174</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR567525</th>\n",
        "      <td> 0.228181</td>\n",
        "      <td> 0.206927</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR567526</th>\n",
        "      <td> 0.451551</td>\n",
        "      <td> 0.445826</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628446</th>\n",
        "      <td> 0.353618</td>\n",
        "      <td> 0.289816</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628447</th>\n",
        "      <td> 0.030795</td>\n",
        "      <td> 0.059503</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628448</th>\n",
        "      <td> 0.034317</td>\n",
        "      <td> 0.057430</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628449</th>\n",
        "      <td> 0.032844</td>\n",
        "      <td> 0.063647</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628450</th>\n",
        "      <td> 0.250529</td>\n",
        "      <td> 0.297513</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628451</th>\n",
        "      <td> 0.036803</td>\n",
        "      <td> 0.066311</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628452</th>\n",
        "      <td> 0.057356</td>\n",
        "      <td> 0.103612</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628453</th>\n",
        "      <td> 0.058829</td>\n",
        "      <td> 0.102427</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628454</th>\n",
        "      <td> 0.175865</td>\n",
        "      <td> 0.245115</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR764666</th>\n",
        "      <td> 0.020783</td>\n",
        "      <td> 0.083777</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR764667</th>\n",
        "      <td> 0.001887</td>\n",
        "      <td> 0.012433</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR764668</th>\n",
        "      <td> 0.002279</td>\n",
        "      <td> 0.009769</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR764669</th>\n",
        "      <td> 0.014569</td>\n",
        "      <td> 0.069272</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 39,
       "text": [
        "<IPython.core.display.HTML at 0x2aaac9073310>"
       ]
      }
     ],
     "prompt_number": 39
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "HTML((intersect_df.div(pause_size, axis=1)).sort(\"SRR334701.peaks.bed\", ascending=False).to_html())"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>SRR334701.peaks.bed</th>\n",
        "      <th>SRR334703.peaks.bed</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>HeLa_eIF4AIII_WT_rep1</th>\n",
        "      <td> 0.484948</td>\n",
        "      <td> 0.464476</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR567526</th>\n",
        "      <td> 0.451551</td>\n",
        "      <td> 0.445826</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628446</th>\n",
        "      <td> 0.353618</td>\n",
        "      <td> 0.289816</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293FlpIn_FMR1-iso1_WT_rep1</th>\n",
        "      <td> 0.320935</td>\n",
        "      <td> 0.372706</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR408237</th>\n",
        "      <td> 0.291061</td>\n",
        "      <td> 0.349615</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HeLa_Ago2_WT_rep1</th>\n",
        "      <td> 0.289104</td>\n",
        "      <td> 0.339846</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HeLa_Ago2_PTB_kd_rep1</th>\n",
        "      <td> 0.288828</td>\n",
        "      <td> 0.362937</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR408236</th>\n",
        "      <td> 0.274443</td>\n",
        "      <td> 0.321492</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_HuR_WT_rep3</th>\n",
        "      <td> 0.268505</td>\n",
        "      <td> 0.347247</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_IGF2BP3_WT_rep1</th>\n",
        "      <td> 0.263741</td>\n",
        "      <td> 0.313203</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_IGF2BP1_WT_rep1</th>\n",
        "      <td> 0.258907</td>\n",
        "      <td> 0.296033</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628450</th>\n",
        "      <td> 0.250529</td>\n",
        "      <td> 0.297513</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527727</th>\n",
        "      <td> 0.244614</td>\n",
        "      <td> 0.292777</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR567525</th>\n",
        "      <td> 0.228181</td>\n",
        "      <td> 0.206927</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189779</th>\n",
        "      <td> 0.225327</td>\n",
        "      <td> 0.301066</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>FOX2</th>\n",
        "      <td> 0.202748</td>\n",
        "      <td> 0.336886</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293FlpIn_FMR1-iso7_WT_rep1</th>\n",
        "      <td> 0.192667</td>\n",
        "      <td> 0.272647</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628454</th>\n",
        "      <td> 0.175865</td>\n",
        "      <td> 0.245115</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_IGF2BP2_WT_rep1</th>\n",
        "      <td> 0.162033</td>\n",
        "      <td> 0.239787</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527730</th>\n",
        "      <td> 0.156762</td>\n",
        "      <td> 0.212848</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527729</th>\n",
        "      <td> 0.137866</td>\n",
        "      <td> 0.193014</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago2_WT_rep2</th>\n",
        "      <td> 0.104447</td>\n",
        "      <td> 0.254292</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>293T_hnRNPU_WT_rep1</th>\n",
        "      <td> 0.100235</td>\n",
        "      <td> 0.192718</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189780</th>\n",
        "      <td> 0.089532</td>\n",
        "      <td> 0.167851</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048951</th>\n",
        "      <td> 0.086862</td>\n",
        "      <td> 0.100355</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189781</th>\n",
        "      <td> 0.084561</td>\n",
        "      <td> 0.173475</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048964</th>\n",
        "      <td> 0.083801</td>\n",
        "      <td> 0.126406</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048963</th>\n",
        "      <td> 0.080809</td>\n",
        "      <td> 0.110420</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048958</th>\n",
        "      <td> 0.073421</td>\n",
        "      <td> 0.100059</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_hnRNPU_WT_rep1</th>\n",
        "      <td> 0.070797</td>\n",
        "      <td> 0.073712</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527728</th>\n",
        "      <td> 0.069025</td>\n",
        "      <td> 0.145944</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago2_WT_rep4</th>\n",
        "      <td> 0.066539</td>\n",
        "      <td> 0.186205</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>293T_hnRNPF_WT_rep1</th>\n",
        "      <td> 0.065389</td>\n",
        "      <td> 0.132919</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048952</th>\n",
        "      <td> 0.064123</td>\n",
        "      <td> 0.110420</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago2_WT_rep3</th>\n",
        "      <td> 0.061568</td>\n",
        "      <td> 0.167259</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628453</th>\n",
        "      <td> 0.058829</td>\n",
        "      <td> 0.102427</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628452</th>\n",
        "      <td> 0.057356</td>\n",
        "      <td> 0.103612</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189783</th>\n",
        "      <td> 0.056044</td>\n",
        "      <td> 0.176732</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189784</th>\n",
        "      <td> 0.054594</td>\n",
        "      <td> 0.148313</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048965</th>\n",
        "      <td> 0.046815</td>\n",
        "      <td> 0.071640</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_DGCR8_WT_rep1</th>\n",
        "      <td> 0.046331</td>\n",
        "      <td> 0.127294</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR390467</th>\n",
        "      <td> 0.046124</td>\n",
        "      <td> 0.039668</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048956</th>\n",
        "      <td> 0.044490</td>\n",
        "      <td> 0.037004</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189786</th>\n",
        "      <td> 0.043040</td>\n",
        "      <td> 0.140320</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR518498</th>\n",
        "      <td> 0.042096</td>\n",
        "      <td> 0.110124</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189782</th>\n",
        "      <td> 0.041797</td>\n",
        "      <td> 0.134991</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628451</th>\n",
        "      <td> 0.036803</td>\n",
        "      <td> 0.066311</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628448</th>\n",
        "      <td> 0.034317</td>\n",
        "      <td> 0.057430</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_PAPD5_WT_rep1</th>\n",
        "      <td> 0.033557</td>\n",
        "      <td> 0.094731</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628449</th>\n",
        "      <td> 0.032844</td>\n",
        "      <td> 0.063647</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>293T_hnRNPA1_WT_rep1</th>\n",
        "      <td> 0.032314</td>\n",
        "      <td> 0.080225</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR390466</th>\n",
        "      <td> 0.031693</td>\n",
        "      <td> 0.038484</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR452496</th>\n",
        "      <td> 0.030957</td>\n",
        "      <td> 0.077265</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628447</th>\n",
        "      <td> 0.030795</td>\n",
        "      <td> 0.059503</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398207</th>\n",
        "      <td> 0.030519</td>\n",
        "      <td> 0.076969</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_DGCR8_overexpressed_rep1</th>\n",
        "      <td> 0.029322</td>\n",
        "      <td> 0.085554</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR518497</th>\n",
        "      <td> 0.028655</td>\n",
        "      <td> 0.085850</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398209</th>\n",
        "      <td> 0.028126</td>\n",
        "      <td> 0.066607</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527732</th>\n",
        "      <td> 0.027987</td>\n",
        "      <td> 0.054174</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398201</th>\n",
        "      <td> 0.023384</td>\n",
        "      <td> 0.056246</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>293T_hnRNPM_WT_rep1</th>\n",
        "      <td> 0.023361</td>\n",
        "      <td> 0.064831</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398208</th>\n",
        "      <td> 0.022625</td>\n",
        "      <td> 0.052398</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048959</th>\n",
        "      <td> 0.022464</td>\n",
        "      <td> 0.052694</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048957</th>\n",
        "      <td> 0.022464</td>\n",
        "      <td> 0.062463</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398197</th>\n",
        "      <td> 0.022141</td>\n",
        "      <td> 0.055950</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_HuR_WT_rep2</th>\n",
        "      <td> 0.021957</td>\n",
        "      <td> 0.080225</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR764666</th>\n",
        "      <td> 0.020783</td>\n",
        "      <td> 0.083777</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_LIN28B_WT_rep1</th>\n",
        "      <td> 0.019333</td>\n",
        "      <td> 0.089402</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398205</th>\n",
        "      <td> 0.018758</td>\n",
        "      <td> 0.048549</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189787</th>\n",
        "      <td> 0.018551</td>\n",
        "      <td> 0.084369</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago4_WT_rep1</th>\n",
        "      <td> 0.017860</td>\n",
        "      <td> 0.084073</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048953</th>\n",
        "      <td> 0.017400</td>\n",
        "      <td> 0.042925</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048980</th>\n",
        "      <td> 0.016802</td>\n",
        "      <td> 0.044997</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398202</th>\n",
        "      <td> 0.015904</td>\n",
        "      <td> 0.041149</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_TNRC6C_WT_rep1</th>\n",
        "      <td> 0.015536</td>\n",
        "      <td> 0.087034</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR764669</th>\n",
        "      <td> 0.014569</td>\n",
        "      <td> 0.069272</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_QIK_WT_rep1</th>\n",
        "      <td> 0.014546</td>\n",
        "      <td> 0.076377</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048960</th>\n",
        "      <td> 0.014040</td>\n",
        "      <td> 0.047365</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048950</th>\n",
        "      <td> 0.013211</td>\n",
        "      <td> 0.033748</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189777</th>\n",
        "      <td> 0.013188</td>\n",
        "      <td> 0.055950</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048966</th>\n",
        "      <td> 0.012912</td>\n",
        "      <td> 0.031083</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189785</th>\n",
        "      <td> 0.012820</td>\n",
        "      <td> 0.052694</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398203</th>\n",
        "      <td> 0.012705</td>\n",
        "      <td> 0.029899</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398206</th>\n",
        "      <td> 0.012245</td>\n",
        "      <td> 0.029603</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189778</th>\n",
        "      <td> 0.012083</td>\n",
        "      <td> 0.059207</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048979</th>\n",
        "      <td> 0.011600</td>\n",
        "      <td> 0.059799</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048962</th>\n",
        "      <td> 0.011554</td>\n",
        "      <td> 0.036116</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398199</th>\n",
        "      <td> 0.011163</td>\n",
        "      <td> 0.026347</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048986</th>\n",
        "      <td> 0.009644</td>\n",
        "      <td> 0.044701</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398196</th>\n",
        "      <td> 0.009321</td>\n",
        "      <td> 0.024275</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Pumilio2_WT_rep1</th>\n",
        "      <td> 0.008953</td>\n",
        "      <td> 0.058615</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398200</th>\n",
        "      <td> 0.008907</td>\n",
        "      <td> 0.027827</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR490650</th>\n",
        "      <td> 0.008102</td>\n",
        "      <td> 0.033452</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048981</th>\n",
        "      <td> 0.007664</td>\n",
        "      <td> 0.035820</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048976</th>\n",
        "      <td> 0.007365</td>\n",
        "      <td> 0.047365</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048973</th>\n",
        "      <td> 0.007296</td>\n",
        "      <td> 0.041149</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR248532</th>\n",
        "      <td> 0.007250</td>\n",
        "      <td> 0.032268</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR518495</th>\n",
        "      <td> 0.007204</td>\n",
        "      <td> 0.037596</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR034467</th>\n",
        "      <td> 0.007020</td>\n",
        "      <td> 0.026051</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048971</th>\n",
        "      <td> 0.006928</td>\n",
        "      <td> 0.039076</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527731</th>\n",
        "      <td> 0.006721</td>\n",
        "      <td> 0.036116</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_HuR_WT_rep1</th>\n",
        "      <td> 0.006675</td>\n",
        "      <td> 0.032860</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_TNRC6A_WT_rep1</th>\n",
        "      <td> 0.006537</td>\n",
        "      <td> 0.048253</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago3_WT_rep1</th>\n",
        "      <td> 0.006537</td>\n",
        "      <td> 0.049142</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR452497</th>\n",
        "      <td> 0.006283</td>\n",
        "      <td> 0.034636</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_HuR_WT_rep6</th>\n",
        "      <td> 0.006168</td>\n",
        "      <td> 0.040261</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR034469</th>\n",
        "      <td> 0.005754</td>\n",
        "      <td> 0.020722</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048961</th>\n",
        "      <td> 0.005478</td>\n",
        "      <td> 0.031083</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_Ago2_WT_rep1</th>\n",
        "      <td> 0.005363</td>\n",
        "      <td> 0.040557</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_TNRC6B_WT_rep1</th>\n",
        "      <td> 0.004879</td>\n",
        "      <td> 0.051214</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398195</th>\n",
        "      <td> 0.004833</td>\n",
        "      <td> 0.004144</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048970</th>\n",
        "      <td> 0.004787</td>\n",
        "      <td> 0.027235</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398204</th>\n",
        "      <td> 0.004764</td>\n",
        "      <td> 0.018058</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048987</th>\n",
        "      <td> 0.004419</td>\n",
        "      <td> 0.040261</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048969</th>\n",
        "      <td> 0.004396</td>\n",
        "      <td> 0.032268</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048975</th>\n",
        "      <td> 0.004373</td>\n",
        "      <td> 0.030195</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048967</th>\n",
        "      <td> 0.004097</td>\n",
        "      <td> 0.034044</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189776</th>\n",
        "      <td> 0.004097</td>\n",
        "      <td> 0.018946</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR034468</th>\n",
        "      <td> 0.003890</td>\n",
        "      <td> 0.020130</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR309285</th>\n",
        "      <td> 0.003406</td>\n",
        "      <td> 0.022499</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048978</th>\n",
        "      <td> 0.003360</td>\n",
        "      <td> 0.032564</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048968</th>\n",
        "      <td> 0.003314</td>\n",
        "      <td> 0.036708</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR309287</th>\n",
        "      <td> 0.003245</td>\n",
        "      <td> 0.020426</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048954</th>\n",
        "      <td> 0.003107</td>\n",
        "      <td> 0.027531</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048983</th>\n",
        "      <td> 0.002693</td>\n",
        "      <td> 0.028419</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR309286</th>\n",
        "      <td> 0.002509</td>\n",
        "      <td> 0.023091</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189775</th>\n",
        "      <td> 0.002302</td>\n",
        "      <td> 0.011249</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR764668</th>\n",
        "      <td> 0.002279</td>\n",
        "      <td> 0.009769</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR518496</th>\n",
        "      <td> 0.002233</td>\n",
        "      <td> 0.017170</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048955</th>\n",
        "      <td> 0.002210</td>\n",
        "      <td> 0.005625</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048984</th>\n",
        "      <td> 0.002210</td>\n",
        "      <td> 0.032860</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR764667</th>\n",
        "      <td> 0.001887</td>\n",
        "      <td> 0.012433</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048982</th>\n",
        "      <td> 0.001565</td>\n",
        "      <td> 0.020426</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048972</th>\n",
        "      <td> 0.001335</td>\n",
        "      <td> 0.018650</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048974</th>\n",
        "      <td> 0.000921</td>\n",
        "      <td> 0.015690</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048985</th>\n",
        "      <td> 0.000690</td>\n",
        "      <td> 0.015098</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR048977</th>\n",
        "      <td> 0.000391</td>\n",
        "      <td> 0.008585</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR398198</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>MDAMB231_hnRNPA2B1_WT_rep1</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR427115</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR427116</th>\n",
        "      <td> 0.000000</td>\n",
        "      <td> 0.000000</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 40,
       "text": [
        "<IPython.core.display.HTML at 0x2aaac9073650>"
       ]
      }
     ],
     "prompt_number": 40
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "(intersect_df.div(pause_size, axis=1)).sort(\"SRR334701.peaks.bed\", ascending=False).head(15)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>SRR334701.peaks.bed</th>\n",
        "      <th>SRR334703.peaks.bed</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>HeLa_eIF4AIII_WT_rep1</th>\n",
        "      <td> 0.484948</td>\n",
        "      <td> 0.464476</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR567526</th>\n",
        "      <td> 0.451551</td>\n",
        "      <td> 0.445826</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628446</th>\n",
        "      <td> 0.353618</td>\n",
        "      <td> 0.289816</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293FlpIn_FMR1-iso1_WT_rep1</th>\n",
        "      <td> 0.320935</td>\n",
        "      <td> 0.372706</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR408237</th>\n",
        "      <td> 0.291061</td>\n",
        "      <td> 0.349615</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HeLa_Ago2_WT_rep1</th>\n",
        "      <td> 0.289104</td>\n",
        "      <td> 0.339846</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HeLa_Ago2_PTB_kd_rep1</th>\n",
        "      <td> 0.288828</td>\n",
        "      <td> 0.362937</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR408236</th>\n",
        "      <td> 0.274443</td>\n",
        "      <td> 0.321492</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_HuR_WT_rep3</th>\n",
        "      <td> 0.268505</td>\n",
        "      <td> 0.347247</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_IGF2BP3_WT_rep1</th>\n",
        "      <td> 0.263741</td>\n",
        "      <td> 0.313203</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>HEK293_IGF2BP1_WT_rep1</th>\n",
        "      <td> 0.258907</td>\n",
        "      <td> 0.296033</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR628450</th>\n",
        "      <td> 0.250529</td>\n",
        "      <td> 0.297513</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR527727</th>\n",
        "      <td> 0.244614</td>\n",
        "      <td> 0.292777</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR567525</th>\n",
        "      <td> 0.228181</td>\n",
        "      <td> 0.206927</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>SRR189779</th>\n",
        "      <td> 0.225327</td>\n",
        "      <td> 0.301066</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "<p>15 rows \u00d7 2 columns</p>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 81,
       "text": [
        "                               SRR334701.peaks.bed  SRR334703.peaks.bed\n",
        "HeLa_eIF4AIII_WT_rep1                     0.484948             0.464476\n",
        "SRR567526                                 0.451551             0.445826\n",
        "SRR628446                                 0.353618             0.289816\n",
        "HEK293FlpIn_FMR1-iso1_WT_rep1             0.320935             0.372706\n",
        "SRR408237                                 0.291061             0.349615\n",
        "HeLa_Ago2_WT_rep1                         0.289104             0.339846\n",
        "HeLa_Ago2_PTB_kd_rep1                     0.288828             0.362937\n",
        "SRR408236                                 0.274443             0.321492\n",
        "HEK293_HuR_WT_rep3                        0.268505             0.347247\n",
        "HEK293_IGF2BP3_WT_rep1                    0.263741             0.313203\n",
        "HEK293_IGF2BP1_WT_rep1                    0.258907             0.296033\n",
        "SRR628450                                 0.250529             0.297513\n",
        "SRR527727                                 0.244614             0.292777\n",
        "SRR567525                                 0.228181             0.206927\n",
        "SRR189779                                 0.225327             0.301066\n",
        "\n",
        "[15 rows x 2 columns]"
       ]
      }
     ],
     "prompt_number": 81
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "HEK293FlpIn_FMR1-iso1_WT_rep1"
     ],
     "language": "python",
     "metadata": {},
     "outputs": []
    }
   ],
   "metadata": {}
  }
 ]
}