{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/nas3/gpratt/anaconda/lib/python2.7/site-packages/matplotlib/__init__.py:1318: UserWarning:  This call to matplotlib.use() has no effect\n",
      "because the backend has already been chosen;\n",
      "matplotlib.use() must be called *before* pylab, matplotlib.pyplot,\n",
      "or matplotlib.backends is imported for the first time.\n",
      "\n",
      "  warnings.warn(_use_error_msg)\n"
     ]
    }
   ],
   "source": [
    "%matplotlib inline\n",
    "from collections import Counter\n",
    "from collections import defaultdict\n",
    "from collections import OrderedDict\n",
    "from itertools import izip\n",
    "import os\n",
    "\n",
    "from IPython.core.display import HTML\n",
    "import matplotlib as mpl\n",
    "from matplotlib import pyplot as plt\n",
    "from matplotlib import gridspec\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import pybedtools\n",
    "from scipy.stats import hypergeom\n",
    "import seaborn as sns\n",
    "import scipy.stats\n",
    "from matplotlib_venn import venn3\n",
    "\n",
    "from clipper.src import CLIP_analysis\n",
    "from gscripts.general import dataviz\n",
    "from gscripts import GO\n",
    "from gscripts.general import region_helpers\n",
    "\n",
    "img_dir = \"/nas3/gpratt/Dropbox/TAF15/Data/SplicingAnalysis_array/figures\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "legend = None\n",
    "\n",
    "OUTSIDE_LEGEND_SAVEFIG_KWS = dict(bbox_extra_artists=(legend,),\n",
    "                                  bbox_inches='tight')\n",
    "from matplotlib import rc\n",
    "\n",
    "mpl.rcParams['svg.fonttype'] = 'none'\n",
    "\n",
    "rc('text', usetex=False)\n",
    "rc('font',**{'family':'sans-serif','sans-serif':['Helvetica']})\n",
    "\n",
    "rbp_name_dict = {\"taf15\": \"TAF15\",\n",
    "                 \"tls\": \"FUS/TLS\",\n",
    "                 \"tdp43\": \"TDP43\"}\n",
    "cell_type_dict = {\"whole brain\": \"Whole Brain\",\n",
    "                  \"spinal cord\": \"Spinal Cord\",\n",
    "                  \"striatum\": \"Striatum\"}\n",
    "\n",
    "color_map = {\"fus\": '#43A247',\n",
    "              'taf15': '#DB1D3D',\n",
    "              'tdp43': '#712878',\n",
    "              'fus_taf15': 'c',\n",
    "              'taf15_tdp43': 'c'}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Get Gene ID and Name mapping\n",
    "-----"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def counts_to_rpkm(featureCountsTable):\n",
    "    counts = featureCountsTable.ix[:,5:]\n",
    "    lengths = featureCountsTable['Length']\n",
    "    mapped_reads = counts.sum()\n",
    "    return (counts * pow(10,9)).div(mapped_reads, axis=1).div(lengths, axis=0)\n",
    "\n",
    "def gencode_to_ensembl(gene_list):\n",
    "    for gene in gene_list:\n",
    "        yield gene.split(\".\")[0]\n",
    "        \n",
    "def plot_go_enrichment(df, filter_value=None, max_terms=None, **kwargs):\n",
    "    df = df.copy()\n",
    "    new_index = []\n",
    "    for index, description in izip(df.index, df['GO Term Description']):\n",
    "        new_index.append(list(index[:-1]) + [description])\n",
    "    df.index = pd.MultiIndex.from_tuples(new_index)\n",
    "\n",
    "    go_matrix = df['Bonferroni-corrected Hypergeometric p-Value'].apply(lambda x: -1 * np.log10(x))\n",
    "    go_matrix = go_matrix.unstack(range(len(go_matrix.index.levels) - 1))\n",
    "    go_matrix = go_matrix.fillna(0)\n",
    "    \n",
    "    #Set cutoff on values \n",
    "    if filter_value is not None:\n",
    "        go_matrix = go_matrix[go_matrix.apply(max, axis=1) > filter_value]\n",
    "    \n",
    "    #Set cutoff on number of go terms to show\n",
    "    if max_terms is not None:\n",
    "        go_matrix = go_matrix.ix[go_matrix.max(axis=1).order(ascending=False).index].ix[:max_terms]\n",
    "    sns.clustermap(go_matrix, robust=True, **kwargs)\n",
    "    \n",
    "def get_expressed_genes(rbp, cell_type):\n",
    "    name = (rbp, cell_type)\n",
    "    mapping = mapper[name]\n",
    "    indexer = mapping[:-1]\n",
    "    analysis_group = mapping[-1]\n",
    "    control_rpkms = rpkm.xs(analysis_group, level=\"analysis_group\").ix[indexer]\n",
    "    return list(gencode_to_ensembl(control_rpkms[control_rpkms > 1].dropna(axis=1).columns))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "gene_id_to_name = region_helpers.gene_id_to_name(\"/nas3/gpratt/gencode/gencode.vM1.annotation.gtf.db\")\n",
    "name_to_gene_id = {value: key for key, value in gene_id_to_name.items()}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "ensembl_to_gencode = {key.split(\".\")[0]: key for key in gene_id_to_name.keys()}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def open_splicing_matrix(fn):\n",
    "    df = pd.read_table(fn, index_col=0)\n",
    "    new_index = []\n",
    "    for item in df.index:\n",
    "        item = item.split(\"|\")\n",
    "        new_index.append(item)\n",
    "\n",
    "    df.index = pd.MultiIndex.from_tuples(new_index, names=[\"gene\", \"splice_type\", \"location\"])\n",
    "    return df\n",
    "\n",
    "def open_omniviewer(fn):\n",
    "    #Think about adding in / flitering on q-values\n",
    "    df = pd.read_table(fn).sort(columns=['event name', 'path'])\n",
    "    df = df[(df.path != \"sk\") &  ~(df.notes.str.contains(\"Ncomplex\"))]\n",
    "    df.index = pd.MultiIndex.from_tuples(list(izip(df['event name'], df['notes'], df['event position'])), \n",
    "                                         names=[\"gene\", \"splice_type\", \"location\"])\n",
    "   \n",
    "    num_sepscores = np.arange((len(df.columns) - 5) / 7)\n",
    "    sepscores = np.array(num_sepscores) * 7 + 6\n",
    "    qvalues = sepscores + 1\n",
    "    return df.iloc[:,sepscores], df.iloc[:,qvalues]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Load All Data from Steph's Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "index_names = [\"rbp\", \"cell_type\"]\n",
    "\n",
    "ews_names = [[\"ews\", \"spinal cord\"]]\n",
    "ews_full, ews_qvalue = open_omniviewer(\"/nas3/shuelga/projects/ews_splicing/ews_taf15_omniviewer.txt\")\n",
    "ews_full = ews_full.drop(\"mjay_sepscore_yeo_TAF15_spcord_sepscore\", axis=1)\n",
    "ews_qvalue = ews_qvalue.drop(\"mjay_sepscore_yeo_TAF15_spcord_q-value\", axis=1)\n",
    "\n",
    "ews_full.columns = pd.MultiIndex.from_tuples(ews_names, names=index_names)\n",
    "ews_qvalue.columns = pd.MultiIndex.from_tuples(ews_names, names=index_names)\n",
    "\n",
    "taf15_names = [[\"taf15\", \"striatum\"], [\"taf15\", \"whole brain\"], [\"taf15\", \"spinal cord\"]]\n",
    "taf15_full, taf15_qvalue = open_omniviewer(\"/nas3/shuelga/projects/taf15_splicing/striatumTAF15_brainTAF15_spinalTAF15_omniviewer.txt\")\n",
    "\n",
    "taf15_full.columns = pd.MultiIndex.from_tuples(taf15_names, names=index_names)\n",
    "taf15_qvalue.columns = pd.MultiIndex.from_tuples(taf15_names, names=index_names)\n",
    "\n",
    "tia1_elav_names = [[\"tia1\", \"whole brain\"], [\"elav\", \"spinal cord\"], [\"tia1\", \"spinal cord\"]]\n",
    "tia1_elav_full, tia1_elav_qvalue = open_omniviewer(\"/nas3/gpratt/projects/kris/data/elav_something_corrected.csv\")\n",
    "\n",
    "tia1_elav_full.columns = pd.MultiIndex.from_tuples(tia1_elav_names, names=index_names)\n",
    "tia1_elav_qvalue.columns = pd.MultiIndex.from_tuples(tia1_elav_names, names=index_names)\n",
    "\n",
    "a2b1_names = [[\"hnrnpa2b1\", \"spinal cord\"]]\n",
    "a2b1_full, a2b1_qvalue = open_omniviewer(\"/nas3/shuelga/projects/a2b1_splicing/a2b1sc_omniviewer.txt\")\n",
    "\n",
    "a2b1_full.columns = pd.MultiIndex.from_tuples(a2b1_names, names=index_names)\n",
    "a2b1_qvalue.columns = pd.MultiIndex.from_tuples(a2b1_names, names=index_names)\n",
    "\n",
    "tdp43tls_names = [[\"tdp43\", \"striatum\"], [\"tls\", \"striatum\"]]\n",
    "tdp43tls_full, tdp43tls_qvalue = open_omniviewer(\"/nas3/shuelga/old_projects/TLS/jxnarray/TDP43KD_TLSKD_TLSKO.txt\")\n",
    "\n",
    "tdp43tls_full = tdp43tls_full.drop(\"mjay_sepscore_yeo_august_ko56_sepscore\", axis=1)\n",
    "tdp43tls_qvalue = tdp43tls_qvalue.drop(\"mjay_sepscore_yeo_august_ko56_q-value\", axis=1)\n",
    "\n",
    "tdp43tls_full.columns = pd.MultiIndex.from_tuples(tdp43tls_names, names=index_names)\n",
    "tdp43tls_qvalue.columns = pd.MultiIndex.from_tuples(tdp43tls_names, names=index_names)\n",
    "\n",
    "\n",
    "tia1_names = [[\"tia1\", \"striatum\"]]\n",
    "tia1_full, tia1_qvalue = open_omniviewer(\"/nas3/shuelga/projects/taf15_splicing/taf15_tia1_omniviewer.txt\")\n",
    "tia1_full = tia1_full.drop(\"mjay_sepscore_yeo_stTAF15_sepscore\", axis=1)\n",
    "tia1_qvalue = tia1_qvalue.drop(\"mjay_sepscore_yeo_stTAF15_q-value\", axis=1)\n",
    "\n",
    "tia1_full.columns = pd.MultiIndex.from_tuples(tia1_names, names=index_names)\n",
    "tia1_qvalue.columns = pd.MultiIndex.from_tuples(tia1_names, names=index_names)\n",
    "\n",
    "all_events_full = pd.concat([ews_full, taf15_full, tia1_elav_full, a2b1_full, tdp43tls_full, tia1_full], axis=1)\n",
    "all_events_full = all_events_full.fillna(0.)\n",
    "all_events_full = all_events_full.groupby(level=all_events_full.index.names)  \n",
    "all_events_full = all_events_full.last()\n",
    "\n",
    "all_events_qvalue = pd.concat([ews_qvalue, taf15_qvalue, tia1_elav_qvalue, a2b1_qvalue, tdp43tls_qvalue, tia1_qvalue], axis=1)\n",
    "all_events_qvalue = all_events_qvalue.fillna(1.)\n",
    "all_events_qvalue = all_events_qvalue.groupby(level=all_events_qvalue.index.names)  \n",
    "all_events_qvalue = all_events_qvalue.last()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "all_events_full.to_csv(\"/nas3/gpratt/Dropbox/TAF15/Data/SplicingAnalysis_array/splicing_matrix_sepscores.csv\")\n",
    "all_events_qvalue.to_csv(\"/nas3/gpratt/Dropbox/TAF15/Data/SplicingAnalysis_array/splicing_matrix_qvalue.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'ENSMUSG00000037266.11'"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ensembl_to_gencode['ENSMUSG00000037266']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>sep_score</th>\n",
       "      <th>q_value</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>old_splice_annotation</th>\n",
       "      <th>splice_type</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rbp</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>location</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"30\" valign=\"top\">ews</th>\n",
       "      <th rowspan=\"30\" valign=\"top\">spinal cord</th>\n",
       "      <th>chr13:63970487-64001507:+</th>\n",
       "      <td>0.750</td>\n",
       "      <td>0.000</td>\n",
       "      <td>ENSMUSG00000021470.10</td>\n",
       "      <td>0610007P08Rik_1_s28e41</td>\n",
       "      <td>alt_end</td>\n",
       "      <td>0610007P08Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr17:25377608-25378008:+</th>\n",
       "      <td>0.416</td>\n",
       "      <td>0.037</td>\n",
       "      <td>ENSMUSG00000021470.10</td>\n",
       "      <td>0610007P22Rik_(398)</td>\n",
       "      <td>retained_int</td>\n",
       "      <td>0610007P08Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr2:163322367-163367238:-</th>\n",
       "      <td>-0.608</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000046840.10</td>\n",
       "      <td>uc008nsx-1_(204)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>0610008F07Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr18:38412619-38413561:+</th>\n",
       "      <td>-0.391</td>\n",
       "      <td>0.000</td>\n",
       "      <td>ENSMUSG00000046840.10</td>\n",
       "      <td>0610009O20Rik_(940)</td>\n",
       "      <td>retained_int</td>\n",
       "      <td>0610008F07Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr11:23520467-23524935:-</th>\n",
       "      <td>-0.544</td>\n",
       "      <td>0.000</td>\n",
       "      <td>ENSMUSG00000042208.8</td>\n",
       "      <td>0610010F05Rik_(94)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>0610010F05Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr11:23531334-23533631:-</th>\n",
       "      <td>-0.286</td>\n",
       "      <td>0.116</td>\n",
       "      <td>ENSMUSG00000042208.8</td>\n",
       "      <td>0610010F05Rik_1_s0e7</td>\n",
       "      <td>alt_start</td>\n",
       "      <td>0610010F05Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr11:23464960-23476730:-</th>\n",
       "      <td>-0.102</td>\n",
       "      <td>0.159</td>\n",
       "      <td>ENSMUSG00000042208.8</td>\n",
       "      <td>0610010F05Rik_4_s46e53</td>\n",
       "      <td>alt_end</td>\n",
       "      <td>0610010F05Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr11:70049713-70051032:-</th>\n",
       "      <td>-0.206</td>\n",
       "      <td>0.000</td>\n",
       "      <td>ENSMUSG00000042208.8</td>\n",
       "      <td>0610010K14Rik_(102)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>0610010F05Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr18:36526395-36562632:+</th>\n",
       "      <td>-0.024</td>\n",
       "      <td>0.161</td>\n",
       "      <td>ENSMUSG00000042208.8</td>\n",
       "      <td>0610010O12Rik_1_s3e8</td>\n",
       "      <td>alt_end</td>\n",
       "      <td>0610010F05Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr17:26012671-26013165:+</th>\n",
       "      <td>0.272</td>\n",
       "      <td>0.179</td>\n",
       "      <td>ENSMUSG00000042208.8</td>\n",
       "      <td>0610011F06Rik_(163)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>0610010F05Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr2:156373619-156376384:+</th>\n",
       "      <td>0.434</td>\n",
       "      <td>0.000</td>\n",
       "      <td>ENSMUSG00000027628.4</td>\n",
       "      <td>0610011L14Rik_(143)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>0610011L14Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr11:69716954-69717367:+</th>\n",
       "      <td>0.175</td>\n",
       "      <td>0.030</td>\n",
       "      <td>ENSMUSG00000047284.7</td>\n",
       "      <td>0610025P10Rik_(66)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Neurl4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr4:116381699-116382042:+</th>\n",
       "      <td>0.139</td>\n",
       "      <td>0.116</td>\n",
       "      <td>ENSMUSG00000047284.7</td>\n",
       "      <td>0610037D15Rik_(12)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Neurl4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr4:116381200-116381697:+</th>\n",
       "      <td>0.457</td>\n",
       "      <td>0.000</td>\n",
       "      <td>ENSMUSG00000028689.8</td>\n",
       "      <td>0610037D15Rik_1_s0e3</td>\n",
       "      <td>alt_start</td>\n",
       "      <td>Ccdc163</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr4:107567543-107569475:+</th>\n",
       "      <td>-0.029</td>\n",
       "      <td>0.509</td>\n",
       "      <td>ENSMUSG00000028608.9</td>\n",
       "      <td>0610037L13Rik_2_s14e19</td>\n",
       "      <td>complex</td>\n",
       "      <td>0610037L13Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr16:14304680-14317386:-</th>\n",
       "      <td>-0.578</td>\n",
       "      <td>0.000</td>\n",
       "      <td>ENSMUSG00000022677.8</td>\n",
       "      <td>0610037P05Rik_(198_85)</td>\n",
       "      <td>twin_cassett</td>\n",
       "      <td>0610037P05Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr19:6985163-6985241:+</th>\n",
       "      <td>-1.502</td>\n",
       "      <td>0.000</td>\n",
       "      <td>ENSMUSG00000038812.10</td>\n",
       "      <td>0610038D11Rik_(76)</td>\n",
       "      <td>retained_int</td>\n",
       "      <td>Trmt112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr19:6984345-6984443:+</th>\n",
       "      <td>-0.326</td>\n",
       "      <td>0.113</td>\n",
       "      <td>ENSMUSG00000038812.10</td>\n",
       "      <td>0610038D11Rik_(96)</td>\n",
       "      <td>retained_int</td>\n",
       "      <td>Trmt112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr19:10591579-10591809:-</th>\n",
       "      <td>-0.455</td>\n",
       "      <td>0.114</td>\n",
       "      <td>ENSMUSG00000038812.10</td>\n",
       "      <td>0610038F07Rik_(150)</td>\n",
       "      <td>alt_5</td>\n",
       "      <td>Trmt112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr19:10576129-10591469:-</th>\n",
       "      <td>0.138</td>\n",
       "      <td>0.572</td>\n",
       "      <td>ENSMUSG00000038812.10</td>\n",
       "      <td>0610038F07Rik_2_s6e10</td>\n",
       "      <td>alt_end</td>\n",
       "      <td>Trmt112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr15:63802523-63892010:-</th>\n",
       "      <td>0.238</td>\n",
       "      <td>0.294</td>\n",
       "      <td>ENSMUSG00000038812.10</td>\n",
       "      <td>0910001A06Rik_1_s0e7</td>\n",
       "      <td>alt_start</td>\n",
       "      <td>Trmt112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr15:63781662-63802505:-</th>\n",
       "      <td>-0.674</td>\n",
       "      <td>0.000</td>\n",
       "      <td>ENSMUSG00000022378.7</td>\n",
       "      <td>0910001A06Rik_1_s5e11</td>\n",
       "      <td>complex</td>\n",
       "      <td>Fam49b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr15:63781662-63789160:-</th>\n",
       "      <td>0.067</td>\n",
       "      <td>0.082</td>\n",
       "      <td>ENSMUSG00000022378.7</td>\n",
       "      <td>0910001A06Rik_1_s7e11</td>\n",
       "      <td>complex</td>\n",
       "      <td>Fam49b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr11:87551775-87556297:+</th>\n",
       "      <td>0.000</td>\n",
       "      <td>0.038</td>\n",
       "      <td>ENSMUSG00000038954.8</td>\n",
       "      <td>Supt4h1_(142)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Supt3h</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr5:136637594-136646407:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>0.301</td>\n",
       "      <td>ENSMUSG00000027584.7</td>\n",
       "      <td>Orai2_(109_63)</td>\n",
       "      <td>twin_cassett</td>\n",
       "      <td>Oprl1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr11:86922806-86929646:+</th>\n",
       "      <td>0.205</td>\n",
       "      <td>0.468</td>\n",
       "      <td>ENSMUSG00000020492.5</td>\n",
       "      <td>1110001A07Rik_(40)</td>\n",
       "      <td>alt_3</td>\n",
       "      <td>Fam33a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr12:3406711-3407700:-</th>\n",
       "      <td>-0.073</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000071456.1</td>\n",
       "      <td>uc007mwp-1_(56)</td>\n",
       "      <td>alt_5</td>\n",
       "      <td>1110002L01Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:128288238-128290238:+</th>\n",
       "      <td>0.061</td>\n",
       "      <td>0.048</td>\n",
       "      <td>ENSMUSG00000071456.1</td>\n",
       "      <td>1110005A23Rik_(195)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>1110002L01Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:128259027-128270367:+</th>\n",
       "      <td>-0.772</td>\n",
       "      <td>0.000</td>\n",
       "      <td>ENSMUSG00000078427.2</td>\n",
       "      <td>1110005A23Rik_(53)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Sarnp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:128290239-128309477:+</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000078427.2</td>\n",
       "      <td>1110005A23Rik_(69)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Sarnp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"30\" valign=\"top\">tia1</th>\n",
       "      <th rowspan=\"30\" valign=\"top\">striatum</th>\n",
       "      <th>chr9:101009091-101027352:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000085832.1</td>\n",
       "      <td>uc009rfk-1_2_s28e32</td>\n",
       "      <td>alt_end</td>\n",
       "      <td>D430036J16Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:112136861-112181204:+</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000087038.3</td>\n",
       "      <td>uc009rwj-1_1_s1e10</td>\n",
       "      <td>alt_end</td>\n",
       "      <td>2900079G21Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:117873696-117922991:-</th>\n",
       "      <td>0.084</td>\n",
       "      <td>0.612</td>\n",
       "      <td>ENSMUSG00000087038.3</td>\n",
       "      <td>uc009rzh-1_(160)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>2900079G21Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:117821156-117869084:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000087038.3</td>\n",
       "      <td>uc009rzh-1_(59_98)</td>\n",
       "      <td>twin_cassett</td>\n",
       "      <td>2900079G21Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:117785145-117922797:-</th>\n",
       "      <td>0.083</td>\n",
       "      <td>0.214</td>\n",
       "      <td>ENSMUSG00000087038.3</td>\n",
       "      <td>uc009rzh-1_1_s3e23</td>\n",
       "      <td>alt_end</td>\n",
       "      <td>2900079G21Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:119043720-119053199:+</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000087038.3</td>\n",
       "      <td>uc009sah-1_1_s40e68</td>\n",
       "      <td>alt_end</td>\n",
       "      <td>2900079G21Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:121012770-121186299:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000087038.3</td>\n",
       "      <td>uc009scx-1_1_s0e69</td>\n",
       "      <td>alt_start</td>\n",
       "      <td>2900079G21Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:120983035-121066090:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000087038.3</td>\n",
       "      <td>uc009scx-1_1_s52e83</td>\n",
       "      <td>alt_end</td>\n",
       "      <td>2900079G21Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:120869385-120983035:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000087038.3</td>\n",
       "      <td>uc009scx-1_1_s73e83</td>\n",
       "      <td>alt_end</td>\n",
       "      <td>2900079G21Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:121849908-121855654:+</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000087038.3</td>\n",
       "      <td>uc009seh-1_1_s5e10</td>\n",
       "      <td>alt_end</td>\n",
       "      <td>2900079G21Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:20538809-20539662:+</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000086877.1</td>\n",
       "      <td>uc009suh-1_1_s0e6</td>\n",
       "      <td>alt_start</td>\n",
       "      <td>A230072C01Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:47911561-47915856:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000085396.1</td>\n",
       "      <td>uc009tdq-1_3_s45e49</td>\n",
       "      <td>alt_end</td>\n",
       "      <td>6720401G13Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:47908921-47911407:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000085396.1</td>\n",
       "      <td>uc009tdq-1_3_s49e53</td>\n",
       "      <td>alt_end</td>\n",
       "      <td>6720401G13Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:71536389-71540421:+</th>\n",
       "      <td>-0.283</td>\n",
       "      <td>0.043</td>\n",
       "      <td>error</td>\n",
       "      <td>uc009tok-1_1_d1s0e4</td>\n",
       "      <td>complex</td>\n",
       "      <td>error</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:75697015-75729089:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>error</td>\n",
       "      <td>uc009tqy-1_3_s11e18</td>\n",
       "      <td>alt_end</td>\n",
       "      <td>error</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:83067283-83133167:+</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>error</td>\n",
       "      <td>uc009tsa-1_1_s0e5</td>\n",
       "      <td>alt_start</td>\n",
       "      <td>error</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:100688896-100691776:+</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>error</td>\n",
       "      <td>uc009tzr-1_1_s0e5</td>\n",
       "      <td>alt_start</td>\n",
       "      <td>error</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:102239450-102262097:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>error</td>\n",
       "      <td>uc009uap-1_(102)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>error</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:102262216-102263909:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>error</td>\n",
       "      <td>uc009uap-1_(106)</td>\n",
       "      <td>alt_5</td>\n",
       "      <td>error</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:102235151-102239340:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>error</td>\n",
       "      <td>uc009uap-1_(180)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>error</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:102265014-102265403:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>error</td>\n",
       "      <td>uc009uap-1_(64)</td>\n",
       "      <td>alt_5</td>\n",
       "      <td>error</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:102238643-102239340:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>error</td>\n",
       "      <td>uc009uap-1_(695)</td>\n",
       "      <td>retained_int</td>\n",
       "      <td>error</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:102583275-102586814:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000087620.1</td>\n",
       "      <td>uc009uax-1_(74)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>5330434G04Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:106031674-106082068:+</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000087620.1</td>\n",
       "      <td>uc009ucp-1_2_s5e16</td>\n",
       "      <td>alt_end</td>\n",
       "      <td>5330434G04Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:131564563-131594518:-</th>\n",
       "      <td>0.216</td>\n",
       "      <td>0.188</td>\n",
       "      <td>ENSMUSG00000087620.1</td>\n",
       "      <td>uc009uha-1_1_s0e7</td>\n",
       "      <td>alt_start</td>\n",
       "      <td>5330434G04Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:133125470-133127444:+</th>\n",
       "      <td>-0.062</td>\n",
       "      <td>0.159</td>\n",
       "      <td>ENSMUSG00000087620.1</td>\n",
       "      <td>uc009uip-1_(46)</td>\n",
       "      <td>alt_3</td>\n",
       "      <td>5330434G04Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:146854984-146893694:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000087620.1</td>\n",
       "      <td>uc009uoh-1_1_s0e5</td>\n",
       "      <td>alt_start</td>\n",
       "      <td>5330434G04Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:146832317-146854884:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000090483.1</td>\n",
       "      <td>uc009uoh-1_2_s5e10</td>\n",
       "      <td>alt_end</td>\n",
       "      <td>Tmem29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:52699256-52702104:+</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000090483.1</td>\n",
       "      <td>uc009uzw-1_(69)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Tmem29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:52694324-52702104:+</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000090483.1</td>\n",
       "      <td>uc009uzw-1_1_s3e13</td>\n",
       "      <td>complex</td>\n",
       "      <td>Tmem29</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>148423 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              sep_score  q_value  \\\n",
       "rbp  cell_type   location                                          \n",
       "ews  spinal cord chr13:63970487-64001507:+        0.750    0.000   \n",
       "                 chr17:25377608-25378008:+        0.416    0.037   \n",
       "                 chr2:163322367-163367238:-      -0.608    1.000   \n",
       "                 chr18:38412619-38413561:+       -0.391    0.000   \n",
       "                 chr11:23520467-23524935:-       -0.544    0.000   \n",
       "                 chr11:23531334-23533631:-       -0.286    0.116   \n",
       "                 chr11:23464960-23476730:-       -0.102    0.159   \n",
       "                 chr11:70049713-70051032:-       -0.206    0.000   \n",
       "                 chr18:36526395-36562632:+       -0.024    0.161   \n",
       "                 chr17:26012671-26013165:+        0.272    0.179   \n",
       "                 chr2:156373619-156376384:+       0.434    0.000   \n",
       "                 chr11:69716954-69717367:+        0.175    0.030   \n",
       "                 chr4:116381699-116382042:+       0.139    0.116   \n",
       "                 chr4:116381200-116381697:+       0.457    0.000   \n",
       "                 chr4:107567543-107569475:+      -0.029    0.509   \n",
       "                 chr16:14304680-14317386:-       -0.578    0.000   \n",
       "                 chr19:6985163-6985241:+         -1.502    0.000   \n",
       "                 chr19:6984345-6984443:+         -0.326    0.113   \n",
       "                 chr19:10591579-10591809:-       -0.455    0.114   \n",
       "                 chr19:10576129-10591469:-        0.138    0.572   \n",
       "                 chr15:63802523-63892010:-        0.238    0.294   \n",
       "                 chr15:63781662-63802505:-       -0.674    0.000   \n",
       "                 chr15:63781662-63789160:-        0.067    0.082   \n",
       "                 chr11:87551775-87556297:+        0.000    0.038   \n",
       "                 chr5:136637594-136646407:-       0.000    0.301   \n",
       "                 chr11:86922806-86929646:+        0.205    0.468   \n",
       "                 chr12:3406711-3407700:-         -0.073    1.000   \n",
       "                 chr10:128288238-128290238:+      0.061    0.048   \n",
       "                 chr10:128259027-128270367:+     -0.772    0.000   \n",
       "                 chr10:128290239-128309477:+      0.000    1.000   \n",
       "...                                                 ...      ...   \n",
       "tia1 striatum    chr9:101009091-101027352:-       0.000    1.000   \n",
       "                 chr9:112136861-112181204:+       0.000    1.000   \n",
       "                 chr9:117873696-117922991:-       0.084    0.612   \n",
       "                 chr9:117821156-117869084:-       0.000    1.000   \n",
       "                 chr9:117785145-117922797:-       0.083    0.214   \n",
       "                 chr9:119043720-119053199:+       0.000    1.000   \n",
       "                 chr9:121012770-121186299:-       0.000    1.000   \n",
       "                 chr9:120983035-121066090:-       0.000    1.000   \n",
       "                 chr9:120869385-120983035:-       0.000    1.000   \n",
       "                 chr9:121849908-121855654:+       0.000    1.000   \n",
       "                 chrX:20538809-20539662:+         0.000    1.000   \n",
       "                 chrX:47911561-47915856:-         0.000    1.000   \n",
       "                 chrX:47908921-47911407:-         0.000    1.000   \n",
       "                 chrX:71536389-71540421:+        -0.283    0.043   \n",
       "                 chrX:75697015-75729089:-         0.000    1.000   \n",
       "                 chrX:83067283-83133167:+         0.000    1.000   \n",
       "                 chrX:100688896-100691776:+       0.000    1.000   \n",
       "                 chrX:102239450-102262097:-       0.000    1.000   \n",
       "                 chrX:102262216-102263909:-       0.000    1.000   \n",
       "                 chrX:102235151-102239340:-       0.000    1.000   \n",
       "                 chrX:102265014-102265403:-       0.000    1.000   \n",
       "                 chrX:102238643-102239340:-       0.000    1.000   \n",
       "                 chrX:102583275-102586814:-       0.000    1.000   \n",
       "                 chrX:106031674-106082068:+       0.000    1.000   \n",
       "                 chrX:131564563-131594518:-       0.216    0.188   \n",
       "                 chrX:133125470-133127444:+      -0.062    0.159   \n",
       "                 chrX:146854984-146893694:-       0.000    1.000   \n",
       "                 chrX:146832317-146854884:-       0.000    1.000   \n",
       "                 chr1:52699256-52702104:+         0.000    1.000   \n",
       "                 chr1:52694324-52702104:+         0.000    1.000   \n",
       "\n",
       "                                                            gene_id  \\\n",
       "rbp  cell_type   location                                             \n",
       "ews  spinal cord chr13:63970487-64001507:+    ENSMUSG00000021470.10   \n",
       "                 chr17:25377608-25378008:+    ENSMUSG00000021470.10   \n",
       "                 chr2:163322367-163367238:-   ENSMUSG00000046840.10   \n",
       "                 chr18:38412619-38413561:+    ENSMUSG00000046840.10   \n",
       "                 chr11:23520467-23524935:-     ENSMUSG00000042208.8   \n",
       "                 chr11:23531334-23533631:-     ENSMUSG00000042208.8   \n",
       "                 chr11:23464960-23476730:-     ENSMUSG00000042208.8   \n",
       "                 chr11:70049713-70051032:-     ENSMUSG00000042208.8   \n",
       "                 chr18:36526395-36562632:+     ENSMUSG00000042208.8   \n",
       "                 chr17:26012671-26013165:+     ENSMUSG00000042208.8   \n",
       "                 chr2:156373619-156376384:+    ENSMUSG00000027628.4   \n",
       "                 chr11:69716954-69717367:+     ENSMUSG00000047284.7   \n",
       "                 chr4:116381699-116382042:+    ENSMUSG00000047284.7   \n",
       "                 chr4:116381200-116381697:+    ENSMUSG00000028689.8   \n",
       "                 chr4:107567543-107569475:+    ENSMUSG00000028608.9   \n",
       "                 chr16:14304680-14317386:-     ENSMUSG00000022677.8   \n",
       "                 chr19:6985163-6985241:+      ENSMUSG00000038812.10   \n",
       "                 chr19:6984345-6984443:+      ENSMUSG00000038812.10   \n",
       "                 chr19:10591579-10591809:-    ENSMUSG00000038812.10   \n",
       "                 chr19:10576129-10591469:-    ENSMUSG00000038812.10   \n",
       "                 chr15:63802523-63892010:-    ENSMUSG00000038812.10   \n",
       "                 chr15:63781662-63802505:-     ENSMUSG00000022378.7   \n",
       "                 chr15:63781662-63789160:-     ENSMUSG00000022378.7   \n",
       "                 chr11:87551775-87556297:+     ENSMUSG00000038954.8   \n",
       "                 chr5:136637594-136646407:-    ENSMUSG00000027584.7   \n",
       "                 chr11:86922806-86929646:+     ENSMUSG00000020492.5   \n",
       "                 chr12:3406711-3407700:-       ENSMUSG00000071456.1   \n",
       "                 chr10:128288238-128290238:+   ENSMUSG00000071456.1   \n",
       "                 chr10:128259027-128270367:+   ENSMUSG00000078427.2   \n",
       "                 chr10:128290239-128309477:+   ENSMUSG00000078427.2   \n",
       "...                                                             ...   \n",
       "tia1 striatum    chr9:101009091-101027352:-    ENSMUSG00000085832.1   \n",
       "                 chr9:112136861-112181204:+    ENSMUSG00000087038.3   \n",
       "                 chr9:117873696-117922991:-    ENSMUSG00000087038.3   \n",
       "                 chr9:117821156-117869084:-    ENSMUSG00000087038.3   \n",
       "                 chr9:117785145-117922797:-    ENSMUSG00000087038.3   \n",
       "                 chr9:119043720-119053199:+    ENSMUSG00000087038.3   \n",
       "                 chr9:121012770-121186299:-    ENSMUSG00000087038.3   \n",
       "                 chr9:120983035-121066090:-    ENSMUSG00000087038.3   \n",
       "                 chr9:120869385-120983035:-    ENSMUSG00000087038.3   \n",
       "                 chr9:121849908-121855654:+    ENSMUSG00000087038.3   \n",
       "                 chrX:20538809-20539662:+      ENSMUSG00000086877.1   \n",
       "                 chrX:47911561-47915856:-      ENSMUSG00000085396.1   \n",
       "                 chrX:47908921-47911407:-      ENSMUSG00000085396.1   \n",
       "                 chrX:71536389-71540421:+                     error   \n",
       "                 chrX:75697015-75729089:-                     error   \n",
       "                 chrX:83067283-83133167:+                     error   \n",
       "                 chrX:100688896-100691776:+                   error   \n",
       "                 chrX:102239450-102262097:-                   error   \n",
       "                 chrX:102262216-102263909:-                   error   \n",
       "                 chrX:102235151-102239340:-                   error   \n",
       "                 chrX:102265014-102265403:-                   error   \n",
       "                 chrX:102238643-102239340:-                   error   \n",
       "                 chrX:102583275-102586814:-    ENSMUSG00000087620.1   \n",
       "                 chrX:106031674-106082068:+    ENSMUSG00000087620.1   \n",
       "                 chrX:131564563-131594518:-    ENSMUSG00000087620.1   \n",
       "                 chrX:133125470-133127444:+    ENSMUSG00000087620.1   \n",
       "                 chrX:146854984-146893694:-    ENSMUSG00000087620.1   \n",
       "                 chrX:146832317-146854884:-    ENSMUSG00000090483.1   \n",
       "                 chr1:52699256-52702104:+      ENSMUSG00000090483.1   \n",
       "                 chr1:52694324-52702104:+      ENSMUSG00000090483.1   \n",
       "\n",
       "                                               old_splice_annotation  \\\n",
       "rbp  cell_type   location                                              \n",
       "ews  spinal cord chr13:63970487-64001507:+    0610007P08Rik_1_s28e41   \n",
       "                 chr17:25377608-25378008:+       0610007P22Rik_(398)   \n",
       "                 chr2:163322367-163367238:-         uc008nsx-1_(204)   \n",
       "                 chr18:38412619-38413561:+       0610009O20Rik_(940)   \n",
       "                 chr11:23520467-23524935:-        0610010F05Rik_(94)   \n",
       "                 chr11:23531334-23533631:-      0610010F05Rik_1_s0e7   \n",
       "                 chr11:23464960-23476730:-    0610010F05Rik_4_s46e53   \n",
       "                 chr11:70049713-70051032:-       0610010K14Rik_(102)   \n",
       "                 chr18:36526395-36562632:+      0610010O12Rik_1_s3e8   \n",
       "                 chr17:26012671-26013165:+       0610011F06Rik_(163)   \n",
       "                 chr2:156373619-156376384:+      0610011L14Rik_(143)   \n",
       "                 chr11:69716954-69717367:+        0610025P10Rik_(66)   \n",
       "                 chr4:116381699-116382042:+       0610037D15Rik_(12)   \n",
       "                 chr4:116381200-116381697:+     0610037D15Rik_1_s0e3   \n",
       "                 chr4:107567543-107569475:+   0610037L13Rik_2_s14e19   \n",
       "                 chr16:14304680-14317386:-    0610037P05Rik_(198_85)   \n",
       "                 chr19:6985163-6985241:+          0610038D11Rik_(76)   \n",
       "                 chr19:6984345-6984443:+          0610038D11Rik_(96)   \n",
       "                 chr19:10591579-10591809:-       0610038F07Rik_(150)   \n",
       "                 chr19:10576129-10591469:-     0610038F07Rik_2_s6e10   \n",
       "                 chr15:63802523-63892010:-      0910001A06Rik_1_s0e7   \n",
       "                 chr15:63781662-63802505:-     0910001A06Rik_1_s5e11   \n",
       "                 chr15:63781662-63789160:-     0910001A06Rik_1_s7e11   \n",
       "                 chr11:87551775-87556297:+             Supt4h1_(142)   \n",
       "                 chr5:136637594-136646407:-           Orai2_(109_63)   \n",
       "                 chr11:86922806-86929646:+        1110001A07Rik_(40)   \n",
       "                 chr12:3406711-3407700:-             uc007mwp-1_(56)   \n",
       "                 chr10:128288238-128290238:+     1110005A23Rik_(195)   \n",
       "                 chr10:128259027-128270367:+      1110005A23Rik_(53)   \n",
       "                 chr10:128290239-128309477:+      1110005A23Rik_(69)   \n",
       "...                                                              ...   \n",
       "tia1 striatum    chr9:101009091-101027352:-      uc009rfk-1_2_s28e32   \n",
       "                 chr9:112136861-112181204:+       uc009rwj-1_1_s1e10   \n",
       "                 chr9:117873696-117922991:-         uc009rzh-1_(160)   \n",
       "                 chr9:117821156-117869084:-       uc009rzh-1_(59_98)   \n",
       "                 chr9:117785145-117922797:-       uc009rzh-1_1_s3e23   \n",
       "                 chr9:119043720-119053199:+      uc009sah-1_1_s40e68   \n",
       "                 chr9:121012770-121186299:-       uc009scx-1_1_s0e69   \n",
       "                 chr9:120983035-121066090:-      uc009scx-1_1_s52e83   \n",
       "                 chr9:120869385-120983035:-      uc009scx-1_1_s73e83   \n",
       "                 chr9:121849908-121855654:+       uc009seh-1_1_s5e10   \n",
       "                 chrX:20538809-20539662:+          uc009suh-1_1_s0e6   \n",
       "                 chrX:47911561-47915856:-        uc009tdq-1_3_s45e49   \n",
       "                 chrX:47908921-47911407:-        uc009tdq-1_3_s49e53   \n",
       "                 chrX:71536389-71540421:+        uc009tok-1_1_d1s0e4   \n",
       "                 chrX:75697015-75729089:-        uc009tqy-1_3_s11e18   \n",
       "                 chrX:83067283-83133167:+          uc009tsa-1_1_s0e5   \n",
       "                 chrX:100688896-100691776:+        uc009tzr-1_1_s0e5   \n",
       "                 chrX:102239450-102262097:-         uc009uap-1_(102)   \n",
       "                 chrX:102262216-102263909:-         uc009uap-1_(106)   \n",
       "                 chrX:102235151-102239340:-         uc009uap-1_(180)   \n",
       "                 chrX:102265014-102265403:-          uc009uap-1_(64)   \n",
       "                 chrX:102238643-102239340:-         uc009uap-1_(695)   \n",
       "                 chrX:102583275-102586814:-          uc009uax-1_(74)   \n",
       "                 chrX:106031674-106082068:+       uc009ucp-1_2_s5e16   \n",
       "                 chrX:131564563-131594518:-        uc009uha-1_1_s0e7   \n",
       "                 chrX:133125470-133127444:+          uc009uip-1_(46)   \n",
       "                 chrX:146854984-146893694:-        uc009uoh-1_1_s0e5   \n",
       "                 chrX:146832317-146854884:-       uc009uoh-1_2_s5e10   \n",
       "                 chr1:52699256-52702104:+            uc009uzw-1_(69)   \n",
       "                 chr1:52694324-52702104:+         uc009uzw-1_1_s3e13   \n",
       "\n",
       "                                               splice_type      gene_name  \n",
       "rbp  cell_type   location                                                  \n",
       "ews  spinal cord chr13:63970487-64001507:+         alt_end  0610007P08Rik  \n",
       "                 chr17:25377608-25378008:+    retained_int  0610007P08Rik  \n",
       "                 chr2:163322367-163367238:-   alt_cassette  0610008F07Rik  \n",
       "                 chr18:38412619-38413561:+    retained_int  0610008F07Rik  \n",
       "                 chr11:23520467-23524935:-    alt_cassette  0610010F05Rik  \n",
       "                 chr11:23531334-23533631:-       alt_start  0610010F05Rik  \n",
       "                 chr11:23464960-23476730:-         alt_end  0610010F05Rik  \n",
       "                 chr11:70049713-70051032:-    alt_cassette  0610010F05Rik  \n",
       "                 chr18:36526395-36562632:+         alt_end  0610010F05Rik  \n",
       "                 chr17:26012671-26013165:+    alt_cassette  0610010F05Rik  \n",
       "                 chr2:156373619-156376384:+   alt_cassette  0610011L14Rik  \n",
       "                 chr11:69716954-69717367:+    alt_cassette         Neurl4  \n",
       "                 chr4:116381699-116382042:+   alt_cassette         Neurl4  \n",
       "                 chr4:116381200-116381697:+      alt_start        Ccdc163  \n",
       "                 chr4:107567543-107569475:+        complex  0610037L13Rik  \n",
       "                 chr16:14304680-14317386:-    twin_cassett  0610037P05Rik  \n",
       "                 chr19:6985163-6985241:+      retained_int        Trmt112  \n",
       "                 chr19:6984345-6984443:+      retained_int        Trmt112  \n",
       "                 chr19:10591579-10591809:-           alt_5        Trmt112  \n",
       "                 chr19:10576129-10591469:-         alt_end        Trmt112  \n",
       "                 chr15:63802523-63892010:-       alt_start        Trmt112  \n",
       "                 chr15:63781662-63802505:-         complex         Fam49b  \n",
       "                 chr15:63781662-63789160:-         complex         Fam49b  \n",
       "                 chr11:87551775-87556297:+    alt_cassette         Supt3h  \n",
       "                 chr5:136637594-136646407:-   twin_cassett          Oprl1  \n",
       "                 chr11:86922806-86929646:+           alt_3         Fam33a  \n",
       "                 chr12:3406711-3407700:-             alt_5  1110002L01Rik  \n",
       "                 chr10:128288238-128290238:+  alt_cassette  1110002L01Rik  \n",
       "                 chr10:128259027-128270367:+  alt_cassette          Sarnp  \n",
       "                 chr10:128290239-128309477:+  alt_cassette          Sarnp  \n",
       "...                                                    ...            ...  \n",
       "tia1 striatum    chr9:101009091-101027352:-        alt_end  D430036J16Rik  \n",
       "                 chr9:112136861-112181204:+        alt_end  2900079G21Rik  \n",
       "                 chr9:117873696-117922991:-   alt_cassette  2900079G21Rik  \n",
       "                 chr9:117821156-117869084:-   twin_cassett  2900079G21Rik  \n",
       "                 chr9:117785145-117922797:-        alt_end  2900079G21Rik  \n",
       "                 chr9:119043720-119053199:+        alt_end  2900079G21Rik  \n",
       "                 chr9:121012770-121186299:-      alt_start  2900079G21Rik  \n",
       "                 chr9:120983035-121066090:-        alt_end  2900079G21Rik  \n",
       "                 chr9:120869385-120983035:-        alt_end  2900079G21Rik  \n",
       "                 chr9:121849908-121855654:+        alt_end  2900079G21Rik  \n",
       "                 chrX:20538809-20539662:+        alt_start  A230072C01Rik  \n",
       "                 chrX:47911561-47915856:-          alt_end  6720401G13Rik  \n",
       "                 chrX:47908921-47911407:-          alt_end  6720401G13Rik  \n",
       "                 chrX:71536389-71540421:+          complex          error  \n",
       "                 chrX:75697015-75729089:-          alt_end          error  \n",
       "                 chrX:83067283-83133167:+        alt_start          error  \n",
       "                 chrX:100688896-100691776:+      alt_start          error  \n",
       "                 chrX:102239450-102262097:-   alt_cassette          error  \n",
       "                 chrX:102262216-102263909:-          alt_5          error  \n",
       "                 chrX:102235151-102239340:-   alt_cassette          error  \n",
       "                 chrX:102265014-102265403:-          alt_5          error  \n",
       "                 chrX:102238643-102239340:-   retained_int          error  \n",
       "                 chrX:102583275-102586814:-   alt_cassette  5330434G04Rik  \n",
       "                 chrX:106031674-106082068:+        alt_end  5330434G04Rik  \n",
       "                 chrX:131564563-131594518:-      alt_start  5330434G04Rik  \n",
       "                 chrX:133125470-133127444:+          alt_3  5330434G04Rik  \n",
       "                 chrX:146854984-146893694:-      alt_start  5330434G04Rik  \n",
       "                 chrX:146832317-146854884:-        alt_end         Tmem29  \n",
       "                 chr1:52699256-52702104:+     alt_cassette         Tmem29  \n",
       "                 chr1:52694324-52702104:+          complex         Tmem29  \n",
       "\n",
       "[148423 rows x 6 columns]"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_events_full"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>sep_score</th>\n",
       "      <th>q_value</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>old_splice_annotation</th>\n",
       "      <th>splice_type</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rbp</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>location</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [sep_score, q_value, gene_id, old_splice_annotation, splice_type, gene_name]\n",
       "Index: []"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_events_full[all_events_full.gene_name == 'ENSMUSG00000037266.11']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "ews = open_splicing_matrix(\"/nas3/shuelga/projects/ews_splicing/analysis/all_events_sep_matrix.txt\")\n",
    "ews = ews.drop(\"taf15\", axis=1)\n",
    "ews.columns = pd.MultiIndex.from_tuples([[\"ews\", \"spinal cord\"]], names=[\"rbp\", \"cell_type\"])\n",
    "\n",
    "taf15 = open_splicing_matrix(\"/nas3/shuelga/projects/taf15_splicing/analysis_allTAF15/all_events_sep_matrix.txt\")\n",
    "taf15.columns = pd.MultiIndex.from_tuples(\n",
    "[[\"taf15\", \"striatum\"],\n",
    "[\"taf15\", \"whole brain\"],\n",
    "[\"taf15\", \"spinal cord\"]]\n",
    "\n",
    ", names=[\"rbp\", \"cell_type\"])\n",
    "\n",
    "tia1_elav = open_splicing_matrix(\"/nas3/gpratt/projects/kris/analysis/all_events_sep_matrix.txt\")\n",
    "tia1_elav.columns = pd.MultiIndex.from_tuples(\n",
    "[[\"tia1\", \"whole brain\"],\n",
    "[\"elav\", \"spinal cord\"],\n",
    "[\"tia1\", \"spinal cord\"]]\n",
    ", names=[\"rbp\", \"cell_type\"])\n",
    "\n",
    "a2b1 = open_splicing_matrix(\"/nas3/shuelga/projects/a2b1_splicing/analysis_a2b1_sc/all_events_sep_matrix.txt\")\n",
    "a2b1.columns = pd.MultiIndex.from_tuples(\n",
    "[[\"hnrnpa2b1\", \"spinal cord\"]]\n",
    ", names=[\"rbp\", \"cell_type\"])\n",
    "\n",
    "tdp43tls = open_splicing_matrix(\"/nas3/shuelga/old_projects/TLS/jxnarray/analysis_new2/all_events_sep_matrix.txt\")\n",
    "tdp43tls = tdp43tls.drop(\"TLSKO\", axis=1)\n",
    "\n",
    "tdp43tls.columns = pd.MultiIndex.from_tuples(\n",
    "[[\"tdp43\", \"striatum\"],\n",
    "[\"tls\", \"striatum\"]], names=[\"rbp\", \"cell_type\"])\n",
    "\n",
    "tia1_striatum = open_splicing_matrix(\"/nas3/shuelga/projects/taf15_splicing/analysis/all_events_sep_matrix.txt\")\n",
    "tia1_striatum = tia1_striatum.drop(\"taf15\", axis=1)\n",
    "\n",
    "tia1_striatum.columns = pd.MultiIndex.from_tuples(\n",
    "[[\"tia1\", \"striatum\"],]\n",
    ", names=[\"rbp\", \"cell_type\"])\n",
    "\n",
    "all_events = pd.concat([ews, taf15, tia1_elav, a2b1, tdp43tls, tia1_striatum], axis=1)\n",
    "all_events = all_events.fillna(0.)\n",
    "all_events = all_events[all_events.apply(lambda x: any(x != 0), axis=1)]\n",
    "all_events = all_events.groupby(level=all_events.index.names)  \n",
    "all_events = all_events.last()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Store Old Index in dict to pull out later"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "genes = pybedtools.BedTool(\"/nas3/gpratt/clipper/clipper/data/regions/mm9_genes.bed\")\n",
    "splice_annotation_dict = dict(zip(all_events.index.get_level_values(level=\"location\"), \n",
    "                                  all_events.index.get_level_values(level=\"gene\")))\n",
    "splice_type_dict = dict(zip(all_events.index.get_level_values(level=\"location\"), \n",
    "                            all_events.index.get_level_values(level=\"splice_type\")))\n",
    "\n",
    "splice_annotation_dict_full = dict(zip(all_events_full.index.get_level_values(level=\"location\"), \n",
    "                                  all_events_full.index.get_level_values(level=\"gene\")))\n",
    "splice_type_dict_full = dict(zip(all_events_full.index.get_level_values(level=\"location\"), \n",
    "                            map(lambda x: x.strip(), all_events_full.index.get_level_values(level=\"splice_type\"))))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Annotate with Ensembl Gene IDs and Names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Should think about how to better assign these, possibly based off the assigned gene name... \n",
    "intervals = []\n",
    "for interval in all_events.index.get_level_values(level=\"location\"):\n",
    "    chrom, loc, strand = interval.split(\":\")\n",
    "    start, stop = loc.split(\"-\")\n",
    "    intervals.append(pybedtools.interval_constructor([chrom, start, stop, interval, \"0\", strand]))\n",
    "locations = pybedtools.BedTool(intervals)\n",
    "\n",
    "locations_and_names = locations.intersect(genes, loj=True, s=True)\n",
    "\n",
    "splice_location_to_gene_id = defaultdict(list)\n",
    "for location in locations_and_names:\n",
    "    splice_location_to_gene_id[location.name].append(location[9])\n",
    "    \n",
    "#Should think about how to better assign these, possibly based off the assigned gene name... \n",
    "intervals = []\n",
    "for interval in all_events_full.index.get_level_values(level=\"location\"):\n",
    "    chrom, loc, strand = interval.split(\":\")\n",
    "    start, stop = loc.split(\"-\")\n",
    "    intervals.append(pybedtools.interval_constructor([chrom, start, stop, interval, \"0\", strand]))\n",
    "locations = pybedtools.BedTool(intervals)\n",
    "\n",
    "locations_and_names = locations.intersect(genes, loj=True, s=True)\n",
    "\n",
    "splice_location_to_gene_id_full = defaultdict(list)\n",
    "for location in locations_and_names:\n",
    "    splice_location_to_gene_id_full[location.name].append(location[9])\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Tidy Data\n",
    "all_events.index = all_events.index.get_level_values(level=\"location\")\n",
    "all_events = all_events.T\n",
    "all_events = all_events.stack()\n",
    "all_events = all_events[all_events != 0]\n",
    "all_events = pd.DataFrame(all_events, columns=[\"sep_score\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "all_events_full.index = all_events_full.index.get_level_values(level=\"location\")\n",
    "all_events_full = all_events_full.T\n",
    "all_events_full = all_events_full.stack()\n",
    "all_events_full = pd.DataFrame(all_events_full, columns=[\"sep_score\"])\n",
    "\n",
    "all_events_qvalue.index = all_events_qvalue.index.get_level_values(level=\"location\")\n",
    "all_events_qvalue = all_events_qvalue.T\n",
    "all_events_qvalue = all_events_qvalue.stack()\n",
    "all_events_qvalue = pd.DataFrame(all_events_qvalue, columns=[\"q_value\"])\n",
    "\n",
    "all_events_full = all_events_full.join(all_events_qvalue)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "accurate_location_to_gene_id = {}\n",
    "for splice_location in all_events.index.get_level_values(level=\"location\"):\n",
    "    annotated_gene_name = splice_annotation_dict[splice_location].split(\"_\")[0]\n",
    "    for gene_id in splice_location_to_gene_id[splice_location]:\n",
    "        if gene_id == \".\":\n",
    "            gene_id = \"error\"\n",
    "        elif gene_id_to_name[gene_id] == annotated_gene_name:\n",
    "            break\n",
    "    \n",
    "    accurate_location_to_gene_id[splice_location] = gene_id\n",
    "    \n",
    "accurate_location_to_gene_id_full = {}\n",
    "for splice_location in all_events_full.index.get_level_values(level=\"location\"):\n",
    "    annotated_gene_name = splice_annotation_dict_full[splice_location].split(\"_\")[0]\n",
    "    for gene_id in splice_location_to_gene_id[splice_location]:\n",
    "        if gene_id == \".\":\n",
    "            gene_id = \"error\"\n",
    "        elif gene_id_to_name[gene_id] == annotated_gene_name:\n",
    "            break\n",
    "    \n",
    "    accurate_location_to_gene_id_full[splice_location] = gene_id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "all_events['gene_id'] = [accurate_location_to_gene_id[splice_location] for splice_location in all_events.index.get_level_values(level=\"location\")]\n",
    "all_events['old_splice_annotation'] = [splice_annotation_dict[splice_location] for splice_location in all_events.index.get_level_values(level=\"location\")]\n",
    "all_events['splice_type'] = [splice_type_dict[splice_location] for splice_location in all_events.index.get_level_values(level=\"location\")]\n",
    "\n",
    "all_events_full['gene_id'] = [accurate_location_to_gene_id_full[splice_location] for splice_location in all_events_full.index.get_level_values(level=\"location\")]\n",
    "all_events_full['old_splice_annotation'] = [splice_annotation_dict_full[splice_location] for splice_location in all_events_full.index.get_level_values(level=\"location\")]\n",
    "all_events_full['splice_type'] = [splice_type_dict_full[splice_location] for splice_location in all_events_full.index.get_level_values(level=\"location\")]\n",
    "\n",
    "\n",
    "\n",
    "gene_names = []\n",
    "for splice_location in all_events['gene_id']:\n",
    "    try:\n",
    "        gene_names.append(gene_id_to_name[splice_location])\n",
    "    except:\n",
    "        gene_names.append(\"error\")\n",
    "        \n",
    "all_events['gene_name'] = gene_names\n",
    "\n",
    "gene_names = []\n",
    "for splice_location in all_events_full['gene_id']:\n",
    "    try:\n",
    "        gene_names.append(gene_id_to_name[splice_location])\n",
    "    except:\n",
    "        gene_names.append(\"error\")\n",
    "        \n",
    "all_events_full['gene_name'] = gene_names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "alt_cassette_events = all_events[all_events.splice_type == \"alt_cassette\"]\n",
    "alt_cassette_events_full = all_events_full[all_events_full.splice_type == \"alt_cassette\"]\n",
    "\n",
    "#Get central exons\n",
    "\n",
    "#Make sure events in stephs list are overlap all exons I'm interested in\n",
    "alt_cassette_exons = pd.read_table(\"/nas3/shuelga/old_projects/TLS/jxnarray/alt_cassette_exons.BED\", header=None,\n",
    "              names=[\"chrom_internal\", \"start_internal\", \"stop_internal\", \"event_id\", \"strand_internal\"])\n",
    "\n",
    "alt_cassette_events_full = pd.merge(left=alt_cassette_events_full.reset_index(), \n",
    "                               right=alt_cassette_exons, \n",
    "                               left_on=\"old_splice_annotation\", \n",
    "                               right_on=\"event_id\").set_index(['rbp', 'cell_type', 'location']).sort_index()\n",
    "\n",
    "#I'm missing 23 events, I think this is an alright loss\n",
    "\n",
    "alt_cassette_event_locations = alt_cassette_events_full.reset_index().set_index(\"location\").groupby(level=0).first()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#After assigning ensembl IDs how many Old and new gene names mismatch?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(510, 5905, 0.0863674851820491)"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(alt_cassette_exons.event_id)), len(set(alt_cassette_events.old_splice_annotation)), len(set(alt_cassette_exons.event_id) & set(alt_cassette_events.old_splice_annotation))\n",
    "\n",
    "all_events['only_name'] = all_events.old_splice_annotation.apply(lambda x: x.split(\"_\")[0])\n",
    "mismatched_events = all_events[all_events.only_name != all_events.gene_name]\n",
    "len(mismatched_events), len(all_events), len(mismatched_events) / (len(all_events) * 1.0)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "8% of events have different gene names, I think this is alright, gene names change stuff looks alright still"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#What is the total number of alt-spliced events?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Total number of events for TAF15, FUS and TDP-43\n",
    "for figure, total anlaysis is below"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "9794"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(alt_cassette_events_full[alt_cassette_events_full.q_value < .05])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "included = alt_cassette_events_full[(alt_cassette_events_full.sep_score > .5) & (alt_cassette_events_full.q_value < .05)]\n",
    "excluded = alt_cassette_events_full[(alt_cassette_events_full.sep_score < -.5) & (alt_cassette_events_full.q_value < .05)]\n",
    "split_alt_cassette_events = pd.concat({\"Included\": included, \"Excluded\": excluded}, names=[\"direction\", \"rbp\", \"cell_type\", \"location\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "direction  rbp        cell_type  \n",
       "Excluded   elav       spinal cord    103\n",
       "           ews        spinal cord    409\n",
       "           hnrnpa2b1  spinal cord     83\n",
       "           taf15      spinal cord     82\n",
       "                      striatum        20\n",
       "                      whole brain     16\n",
       "           tdp43      striatum       145\n",
       "           tia1       spinal cord     14\n",
       "                      striatum        13\n",
       "                      whole brain      7\n",
       "           tls        striatum        64\n",
       "Included   elav       spinal cord     97\n",
       "           ews        spinal cord    207\n",
       "           hnrnpa2b1  spinal cord     51\n",
       "           taf15      spinal cord     95\n",
       "                      striatum        31\n",
       "                      whole brain     29\n",
       "           tdp43      striatum       185\n",
       "           tia1       spinal cord     25\n",
       "                      striatum        36\n",
       "                      whole brain      8\n",
       "           tls        striatum       113\n",
       "Name: sep_score, dtype: int64"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "split_alt_cassette_events.groupby(level=[\"direction\", \"rbp\", \"cell_type\"]).count().sep_score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "rbp        cell_type  \n",
       "elav       spinal cord    200\n",
       "ews        spinal cord    616\n",
       "hnrnpa2b1  spinal cord    134\n",
       "taf15      spinal cord    177\n",
       "           striatum        51\n",
       "           whole brain     45\n",
       "tdp43      striatum       330\n",
       "tia1       spinal cord     39\n",
       "           striatum        49\n",
       "           whole brain     15\n",
       "tls        striatum       177\n",
       "Name: sep_score, dtype: int64"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "split_alt_cassette_events.groupby(level=[\"rbp\", \"cell_type\"]).count().sep_score"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Figure Legend: Number of significant included and excluded alt cassette events in TAF15, TDP43 and FUS/TLS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "significant_events = all_events_full[(all_events_full.sep_score.abs() > .5) & (all_events_full.q_value < .05)]\n",
    "significant_alt_cassette_events = alt_cassette_events_full[(alt_cassette_events_full.sep_score.abs() > .5) & (alt_cassette_events_full.q_value < .05)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "items = alt_cassette_events.groupby(level=[\"rbp\", \"cell_type\"]).count().sep_score\n",
    "with dataviz.Figure(os.path.join(img_dir, \"total_number_of_events.svg\"), figsize=(12,3), tight_layout=True) as fig:\n",
    "    ax = fig.add_subplot(1,2,1)\n",
    "    items = significant_events.ix[['taf15', 'tdp43', 'tls']].groupby(level=[\"rbp\", \"cell_type\"]).count().sep_score\n",
    "    names = np.array([\" \".join([rbp_name_dict[item[0]], cell_type_dict[item[1]]]) for item in items.index])\n",
    "    sns.barplot(names,\n",
    "                items.values,\n",
    "                ci=None,\n",
    "                palette=\"Paired\",\n",
    "                hline=.1,\n",
    "                ax=ax,\n",
    "                x_order=names\n",
    "                )\n",
    "    [tick.set_rotation(45) for tick in ax.get_xticklabels()]\n",
    "    ax.set_title(\"Number of Alt Splicing Events\", fontsize=18)\n",
    "    ax.set_ylabel(\"Number of Significantly\\nChanging Events\", fontsize=14)\n",
    "    [tick.set_fontsize(14) for tick in ax.get_xticklabels()]\n",
    "\n",
    "    ax = fig.add_subplot(1,2,2)\n",
    "    items = significant_alt_cassette_events.ix[['taf15', 'tdp43', 'tls']].groupby(level=[\"rbp\", \"cell_type\"]).count().sep_score\n",
    "    names = np.array([\" \".join([rbp_name_dict[item[0]], cell_type_dict[item[1]]]) for item in items.index])\n",
    "    sns.barplot(names,\n",
    "                items.values,\n",
    "                ci=None,\n",
    "                palette=\"Paired\",\n",
    "                hline=.1, \n",
    "                ax=ax,\n",
    "                x_order=names)\n",
    "    [tick.set_rotation(45) for tick in ax.get_xticklabels()]\n",
    "    [tick.set_fontsize(12) for tick in ax.get_xticklabels()]\n",
    "\n",
    "    ax.set_title(\"Number of Alt Cassette Events\")\n",
    "    #ax.set_ylabel(\"Number of Events\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Figure Legend: Figure Legend: Number of significant alt splicing and alt casssete events in TAF15, TDP43 and FUS/TLS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "items = alt_cassette_events.groupby(level=[\"rbp\", \"cell_type\"]).count().sep_score\n",
    "with dataviz.Figure(os.path.join(img_dir, \"foo.svg\"), figsize=(6,3), tight_layout=True) as fig:\n",
    "    ax = fig.add_subplot(1,2,1)\n",
    "    items = significant_events.groupby(level=[\"rbp\", \"cell_type\"]).count().sep_score\n",
    "    sns.barplot(np.array([\" \".join(item) for item in items.index]), items.values, ci=None, palette=\"Paired\", hline=.1, ax=ax)\n",
    "    [tick.set_rotation(90) for tick in ax.get_xticklabels()]\n",
    "    ax.set_title(\"Total Number of Events\")\n",
    "    \n",
    "    ax = fig.add_subplot(1,2,2)\n",
    "    items = significant_alt_cassette_events.groupby(level=[\"rbp\", \"cell_type\"]).count().sep_score\n",
    "    sns.barplot(np.array([\" \".join(item) for item in items.index]), items.values, ci=None, palette=\"Paired\", hline=.1, ax=ax)\n",
    "    [tick.set_rotation(90) for tick in ax.get_xticklabels()]\n",
    "    ax.set_title(\"Number of Alt Splcing Events\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Figure Legend: Figure Legend: Number of significant alt splicing and alt casssete events in for all knockdowns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#What is the fraction of all events?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "num_events = {}\n",
    "for name, df in significant_events.ix[['taf15', 'tdp43', 'tls']].groupby(level=['rbp', 'cell_type']):\n",
    "    num_events[name] = df.groupby(\"splice_type\").count().sep_score\n",
    "    \n",
    "num_events = pd.DataFrame(num_events)\n",
    "num_events.columns = pd.MultiIndex.from_tuples(num_events.columns)\n",
    "num_events = num_events.fillna(0)\n",
    "\n",
    "fraction_events = num_events / num_events.sum() \n",
    "cumsum_events = fraction_events.cumsum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "\n",
    "num_rows = 1\n",
    "num_cols = 1\n",
    "with dataviz.Figure(os.path.join(img_dir, \"fraction_binding_striatum.svg\"), figsize=(6, 4)) as fig:\n",
    "    ax = fig.add_subplot(num_rows,num_cols,1)\n",
    "    \n",
    "    legend_builder = []\n",
    "    legend_labels = []\n",
    "    for splice_type, color in izip(reversed(cumsum_events.index), sns.color_palette(\"Set2\", len(cumsum_events.index))):\n",
    "        names = np.array([\" \".join([rbp_name_dict[item[0]], cell_type_dict[item[1]]]) for item in cumsum_events.columns])\n",
    "\n",
    "        sns.barplot(names, \n",
    "                    y=cumsum_events.ix[splice_type], color=color, ax=ax)\n",
    "        \n",
    "        legend_builder.append(plt.Rectangle((0,0),.25,.25, fc=color, edgecolor = 'none'))\n",
    "        legend_labels.append(splice_type)\n",
    "\n",
    "    sns.despine(ax=ax, left=True)\n",
    "    \n",
    "   \n",
    "\n",
    "    l = ax.legend(legend_builder, \n",
    "                  legend_labels, loc=1, ncol = 1, \n",
    "                  prop={'size':10}, \n",
    "                  bbox_to_anchor=(1.5, 1))\n",
    "    l.draw_frame(False)\n",
    "    [tick.set_rotation(45) for tick in ax.get_xticklabels()]\n",
    "    #Need to change to percent\n",
    "    ax.set_ylabel(\"Fraction of Events\", fontsize=14)\n",
    "    [tick.set_fontsize(12) for tick in ax.get_xticklabels()]\n",
    "    \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Figure Legend: Fraction of significant alt splcicing events belonging to given group for TAF15, TDP and FUS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "num_rows = 1\n",
    "num_cols = 1\n",
    "cumsum_striatum_events = cumsum_events.T.xs(\"striatum\", level=1).T\n",
    "with dataviz.Figure(os.path.join(img_dir, \"fraction_binding_striatum.svg\"), figsize=(6, 4)) as fig:\n",
    "    ax = fig.add_subplot(num_rows,num_cols,1)\n",
    "    \n",
    "    legend_builder = []\n",
    "    legend_labels = []\n",
    "    for splice_type, color in izip(reversed(cumsum_striatum_events.index), sns.color_palette(\"Set2\", len(cumsum_striatum_events.index))):\n",
    "        names = np.array([rbp_name_dict[item] for item in cumsum_striatum_events.columns])\n",
    "        sns.barplot(names, \n",
    "                    cumsum_striatum_events.ix[splice_type], \n",
    "                    color=color, \n",
    "                    ax=ax)\n",
    "        \n",
    "        legend_builder.append(plt.Rectangle((0,0),.25,.25, fc=color, edgecolor = 'none'))\n",
    "        legend_labels.append(splice_type)\n",
    "\n",
    "    sns.despine(ax=ax, left=True)\n",
    "    \n",
    "   \n",
    "\n",
    "    l = ax.legend(legend_builder, \n",
    "                  legend_labels, loc=1, ncol = 1, \n",
    "                  prop={'size':10}, \n",
    "                  bbox_to_anchor=(1.5, 1))\n",
    "    l.draw_frame(False)\n",
    "    [tick.set_rotation(0) for tick in ax.get_xticklabels()]\n",
    "    #Need to change to percent\n",
    "    ax.set_ylabel(\"Fraction of Events\", fontsize=14)\n",
    "    [tick.set_fontsize(12) for tick in ax.get_xticklabels()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/nas3/gpratt/Dropbox/TAF15/Data/SplicingAnalysis_array/figures/num_splicing_events.csv'"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.path.join(img_dir, \"num_splicing_events.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "num_events.to_csv(os.path.join(img_dir, \"num_splicing_events.csv\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "num_rows = 1\n",
    "num_cols = 1\n",
    "cumsum_striatum_events = num_events.cumsum().T.xs(\"striatum\", level=1).T\n",
    "with dataviz.Figure(os.path.join(img_dir, \"total_binding_striatum.svg\"), figsize=(4, 2)) as fig:\n",
    "    ax = fig.add_subplot(num_rows,num_cols,1)\n",
    "    \n",
    "    legend_builder = []\n",
    "    legend_labels = []\n",
    "    for splice_type, color in izip(reversed(cumsum_striatum_events.index), sns.color_palette(\"Set2\", len(cumsum_striatum_events.index))):\n",
    "        names = np.array([rbp_name_dict[item] for item in cumsum_striatum_events.columns])\n",
    "        sns.barplot(names, \n",
    "                    cumsum_striatum_events.ix[splice_type], \n",
    "                    color=color, \n",
    "                    ax=ax)\n",
    "        \n",
    "        legend_builder.append(plt.Rectangle((0,0),.25,.25, fc=color, edgecolor = 'none'))\n",
    "        legend_labels.append(splice_type)\n",
    "\n",
    "    sns.despine(ax=ax, left=True)\n",
    "    \n",
    "   \n",
    "\n",
    "    l = ax.legend(legend_builder, \n",
    "                  legend_labels, loc=1, ncol = 2, \n",
    "                  prop={'size':8}, \n",
    "                  bbox_to_anchor=(1.5, 1))\n",
    "    l.draw_frame(False)\n",
    "    [tick.set_rotation(0) for tick in ax.get_xticklabels()]\n",
    "    #Need to change to percent\n",
    "    ax.set_ylabel(\"Total Number of Events\", fontsize=10)\n",
    "    [tick.set_fontsize(12) for tick in ax.get_xticklabels()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th colspan=\"3\" halign=\"left\">taf15</th>\n",
       "      <th>tdp43</th>\n",
       "      <th>tls</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>spinal cord</th>\n",
       "      <th>striatum</th>\n",
       "      <th>whole brain</th>\n",
       "      <th>striatum</th>\n",
       "      <th>striatum</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>alt_3</th>\n",
       "      <td>34</td>\n",
       "      <td>13</td>\n",
       "      <td>8</td>\n",
       "      <td>45</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>alt_5</th>\n",
       "      <td>24</td>\n",
       "      <td>8</td>\n",
       "      <td>12</td>\n",
       "      <td>37</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>alt_cassette</th>\n",
       "      <td>177</td>\n",
       "      <td>51</td>\n",
       "      <td>45</td>\n",
       "      <td>330</td>\n",
       "      <td>177</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>alt_end</th>\n",
       "      <td>119</td>\n",
       "      <td>49</td>\n",
       "      <td>20</td>\n",
       "      <td>105</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>alt_start</th>\n",
       "      <td>90</td>\n",
       "      <td>43</td>\n",
       "      <td>19</td>\n",
       "      <td>101</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>complex</th>\n",
       "      <td>44</td>\n",
       "      <td>15</td>\n",
       "      <td>10</td>\n",
       "      <td>69</td>\n",
       "      <td>40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mutually_exc</th>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>retained_int</th>\n",
       "      <td>38</td>\n",
       "      <td>15</td>\n",
       "      <td>12</td>\n",
       "      <td>52</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>twin_cassett</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   taf15                         tdp43      tls\n",
       "             spinal cord striatum whole brain striatum striatum\n",
       "alt_3                 34       13           8       45       20\n",
       "alt_5                 24        8          12       37       27\n",
       "alt_cassette         177       51          45      330      177\n",
       "alt_end              119       49          20      105       34\n",
       "alt_start             90       43          19      101       22\n",
       "complex               44       15          10       69       40\n",
       "mutually_exc           3        0           0        5        1\n",
       "retained_int          38       15          12       52       38\n",
       "twin_cassett           3        3           1       12        8"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "num_events"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "num_rows = 1\n",
    "num_cols = 1\n",
    "cumsum_striatum_events = num_events.cumsum().T.ix[\"taf15\"].T\n",
    "with dataviz.Figure(os.path.join(img_dir, \"total_binding_taf15.svg\"), figsize=(4, 2)) as fig:\n",
    "    ax = fig.add_subplot(num_rows,num_cols,1)\n",
    "    \n",
    "    legend_builder = []\n",
    "    legend_labels = []\n",
    "    for splice_type, color in izip(reversed(cumsum_striatum_events.index), sns.color_palette(\"Set2\", len(cumsum_striatum_events.index))):\n",
    "        names = np.array([cell_type_dict[item] for item in cumsum_striatum_events.columns])\n",
    "        sns.barplot(names, \n",
    "                    cumsum_striatum_events.ix[splice_type], \n",
    "                    color=color, \n",
    "                    ax=ax)\n",
    "        \n",
    "        legend_builder.append(plt.Rectangle((0,0),.25,.25, fc=color, edgecolor = 'none'))\n",
    "        legend_labels.append(splice_type)\n",
    "\n",
    "    sns.despine(ax=ax, left=True)\n",
    "    \n",
    "   \n",
    "\n",
    "    l = ax.legend(legend_builder, \n",
    "                  legend_labels, loc=1, ncol = 2, \n",
    "                  prop={'size':8}, \n",
    "                  bbox_to_anchor=(1.5, 1))\n",
    "    l.draw_frame(False)\n",
    "    [tick.set_rotation(0) for tick in ax.get_xticklabels()]\n",
    "    #Need to change to percent\n",
    "    ax.set_ylabel(\"Total Number of Events\", fontsize=10)\n",
    "    [tick.set_fontsize(12) for tick in ax.get_xticklabels()]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Conclusions\n",
    "---\n",
    "\n",
    "1. There are about as many splicing events in TAF15 as there are in TDP43 and TLS under some conditions\n",
    "2. The majority of splicing events are alt cassette, however there appears to be an icrease of alt start and alt end events in TAF15 as compared to other RPBs.  Unknown if this is important"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#How do splicing events overlap?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "concordent = defaultdict(dict)\n",
    "discordent = defaultdict(dict)\n",
    "\n",
    "for rbp_1, df_1 in alt_cassette_events.groupby(level=[\"rbp\", \"cell_type\"]):\n",
    "    for rbp_2, df_2 in alt_cassette_events.groupby(level=[\"rbp\", \"cell_type\"]):\n",
    "        name1 = \"_\".join(rbp_1)\n",
    "        name2 = \"_\".join(rbp_2)\n",
    "        \n",
    "        df_1.index = df_1.index.get_level_values(level=\"location\")\n",
    "        df_2.index = df_2.index.get_level_values(level=\"location\")\n",
    "        combined_df = pd.concat({\"df1\": df_1.sep_score, \"df2\": df_2.sep_score}, axis=1)\n",
    "        combined_df = combined_df.dropna()\n",
    "        if len(combined_df) == 0:\n",
    "            concordent[name1][name2] = 0\n",
    "            discordent[name1][name2] = 0\n",
    "            continue\n",
    "        concordent_changes = combined_df.apply(lambda x: all(x > 0) or all(x < 0), axis=1)\n",
    "        discordent_changes = combined_df.apply(lambda x: (x[0] > 0 and x[1] < 0) or (x[0] < 0 and x[1] > 0), axis=1)\n",
    "\n",
    "        concordent[name1][name2] = sum(concordent_changes)\n",
    "        discordent[name1][name2] = sum(discordent_changes)\n",
    "\n",
    "concordent = pd.DataFrame(concordent)\n",
    "discordent = pd.DataFrame(discordent)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "num_rows = len(discordent)\n",
    "num_cols = len(discordent)\n",
    "width=.35\n",
    "\n",
    "with dataviz.Figure(os.path.join(img_dir, \"replicate_plotting.svg\"), figsize=(2.5 * num_cols,2.5*num_rows), tight_layout=False) as fig:\n",
    "    gs = gridspec.GridSpec(num_rows, num_cols)\n",
    "\n",
    "    for x, ((name_1, concordent_row), (name_1, discordent_row)) in enumerate(zip(concordent.iterrows(), discordent.iterrows())):\n",
    "        for y, ((name_2, concordent_value), (name_2, discordent_value)) in enumerate(zip(concordent_row.iteritems(), discordent_row.iteritems())):\n",
    "            ax = fig.add_subplot(gs[x,y])\n",
    "            \n",
    "            concordent_value = concordent_value + 1 if concordent_value == 0 else concordent_value\n",
    "            discordent_value = discordent_value + 1 if discordent_value == 0 else discordent_value\n",
    "\n",
    "            ax.bar([0], [concordent_value], width, color='r')\n",
    "            ax.bar([width], [discordent_value], width, color='b')\n",
    "            \n",
    "            sns.despine(ax=ax, left=True, bottom=True)\n",
    "            ax.set_xticklabels([])\n",
    "\n",
    "            ax.set_ylim(0, max(concordent_row.max(), concordent_row.max()))\n",
    "            if ax.is_first_col():\n",
    "                ax.set_ylabel(name_1)\n",
    "            else:\n",
    "                ax.set_yticklabels([])\n",
    "            if ax.is_last_row():\n",
    "                ax.set_xlabel(name_2)\n",
    "                \n",
    "            if ax.is_last_col() and ax.is_first_row():\n",
    "                ax.legend([\"Concordent\", \"Discordent\"])\n",
    "            #ax.set_title(\"{} vs\\n{}\".format(name1, name2), fontsize=6)\n",
    "            #[tick.set_fontsize(10) for tick in ax.get_xticklabels()]\n",
    "            #[tick.set_fontsize(10) for tick in ax.get_yticklabels()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "interesing_kds = [u'taf15_spinal cord', u'taf15_striatum', u'taf15_whole brain', u'tdp43_striatum', u'tls_striatum']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "interesting_concordent = concordent.ix[interesing_kds, interesing_kds]\n",
    "interesting_discordent = discordent.ix[interesing_kds, interesing_kds]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "num_rows = len(interesting_concordent)\n",
    "num_cols = len(interesting_concordent)\n",
    "width=.35\n",
    "\n",
    "with dataviz.Figure(os.path.join(img_dir, \"replicate_plotting.svg\"), figsize=(1.5 * num_cols,1.5*num_rows)) as fig:\n",
    "    gs = gridspec.GridSpec(num_rows, num_cols)\n",
    "\n",
    "    for x, ((name_1, concordent_row), (name_1, discordent_row)) in enumerate(zip(interesting_concordent.iterrows(), interesting_discordent.iterrows())):\n",
    "        for y, ((name_2, concordent_value), (name_2, discordent_value)) in enumerate(zip(concordent_row.iteritems(), discordent_row.iteritems())):\n",
    "        \n",
    "            ax = fig.add_subplot(gs[x,y])\n",
    "            ax.set_ylim(0, 50)\n",
    "\n",
    "            if ax.is_first_col():\n",
    "                ax.set_ylabel(name_1)\n",
    "            else:\n",
    "                ax.set_yticklabels([])\n",
    "            if ax.is_last_row():\n",
    "                ax.set_xlabel(name_2)\n",
    "            sns.despine(ax=ax, left=True, bottom=True)\n",
    "            ax.set_xticklabels([])\n",
    "\n",
    "            if name_1 == name_2:\n",
    "                continue\n",
    "            concordent_value = concordent_value + 1 if concordent_value == 0 else concordent_value\n",
    "            discordent_value = discordent_value + 1 if discordent_value == 0 else discordent_value\n",
    "\n",
    "            ax.bar([0], [concordent_value], width, color='r')\n",
    "            ax.bar([width], [discordent_value], width, color='b')\n",
    "            \n",
    "            sns.despine(ax=ax, left=True, bottom=True)\n",
    "            ax.set_xticklabels([])\n",
    "\n",
    "            #ax.set_ylim(0, max(concordent_row.drop(name_1).max(), discordent_row.drop(name_1).max()))\n",
    "            ax.set_ylim(0, 50)\n",
    "\n",
    "            if ax.is_first_col():\n",
    "                ax.set_ylabel(name_1)\n",
    "            else:\n",
    "                ax.set_yticklabels([])\n",
    "            if ax.is_last_row():\n",
    "                ax.set_xlabel(name_2)\n",
    "                \n",
    "            if ax.is_last_col() and ax.is_first_row():\n",
    "                ax.legend([\"Concordent\", \"Discordent\"])\n",
    "            #ax.set_title(\"{} vs\\n{}\".format(name1, name2), fontsize=6)\n",
    "            #[tick.set_fontsize(10) for tick in ax.get_xticklabels()]\n",
    "            #[tick.set_fontsize(10) for tick in ax.get_yticklabels()]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Figure Legend: Number of Concordent and discordent overlapping events between TAF15, FUs and TLS for all tissue types and knockdowns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Clustering Actual Events"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<seaborn.matrix.ClusterGrid at 0x25b59fd0>"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "heatmap_prep = all_events.sep_score.unstack()\n",
    "heatmap_prep = heatmap_prep.fillna(0)\n",
    "sns.clustermap(heatmap_prep, linewidths=0)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Figure Legend: Clustergram of all significant alt-splcing events, in all mouse knockdowns, blue is excluded splicing event, red is included"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#sns.clustermap(heatmap_prep.ix[['taf15', 'tdp43', 'tls']], linewidths=0, figsize=(20,10))\n",
    "#plt.savefig(os.path.join(img_dir, \"splicing_events.pdf\"))\n",
    "#plt.savefig(os.path.join(img_dir, \"splicing_events.svg\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<seaborn.matrix.ClusterGrid at 0x20638510>"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "heatmap_prep = alt_cassette_events.sep_score.unstack()\n",
    "heatmap_prep = heatmap_prep.fillna(0)\n",
    "sns.clustermap(heatmap_prep, linewidths=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<seaborn.matrix.ClusterGrid at 0x2aaad7d69c10>"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "heatmap_prep = all_events.ix[['taf15', 'tdp43', 'tls']] \n",
    "heatmap_prep = heatmap_prep.sep_score.unstack()\n",
    "heatmap_prep = heatmap_prep.fillna(0)\n",
    "sns.clustermap(heatmap_prep, linewidths=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "heatmap_prep = alt_cassette_events.ix[['taf15', 'tdp43', 'tls']] \n",
    "heatmap_prep = heatmap_prep.sep_score.unstack()\n",
    "heatmap_prep = heatmap_prep.fillna(0)\n",
    "sns.clustermap(heatmap_prep, linewidths=0, xticklabels=False, yticklabels=names)\n",
    "plt.savefig(os.path.join(img_dir, \"splicing_events.pdf\"))\n",
    "plt.savefig(os.path.join(img_dir, \"splicing_events.svg\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "idx = pd.IndexSlice"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "heatmap_prep = alt_cassette_events.xs(\"striatum\", level=\"cell_type\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>sep_score</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>old_splice_annotation</th>\n",
       "      <th>splice_type</th>\n",
       "      <th>gene_name</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rbp</th>\n",
       "      <th>location</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"30\" valign=\"top\">taf15</th>\n",
       "      <th>chr10:106364918-106368179:+</th>\n",
       "      <td>-0.911</td>\n",
       "      <td>ENSMUSG00000053825.7</td>\n",
       "      <td>Ppfia2_(92)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Ppfia2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:119487172-119491888:+</th>\n",
       "      <td>-0.535</td>\n",
       "      <td>ENSMUSG00000034813.10</td>\n",
       "      <td>Grip1_(45)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Grip1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:5299010-5302787:+</th>\n",
       "      <td>0.776</td>\n",
       "      <td>ENSMUSG00000019769.9</td>\n",
       "      <td>uc007egt-1_(69)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Syne1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr11:106616365-106622496:+</th>\n",
       "      <td>-0.677</td>\n",
       "      <td>ENSMUSG00000040528.7</td>\n",
       "      <td>Gm885_(112)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Gm885</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr11:106732924-106781494:-</th>\n",
       "      <td>0.582</td>\n",
       "      <td>ENSMUSG00000018363.10</td>\n",
       "      <td>Smurf2_(39)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Smurf2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr11:115865028-115866865:+</th>\n",
       "      <td>-0.696</td>\n",
       "      <td>ENSMUSG00000020758.9</td>\n",
       "      <td>Itgb4_(195)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Itgb4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr11:115867081-115867774:+</th>\n",
       "      <td>-0.844</td>\n",
       "      <td>ENSMUSG00000020758.9</td>\n",
       "      <td>Itgb4_(159)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Itgb4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr11:4602223-4604160:-</th>\n",
       "      <td>-0.777</td>\n",
       "      <td>ENSMUSG00000059534.8</td>\n",
       "      <td>1110020P15Rik_(86)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Uqcr10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr11:5879724-5882503:-</th>\n",
       "      <td>-0.548</td>\n",
       "      <td>ENSMUSG00000057897.7</td>\n",
       "      <td>Camk2b_(45)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Camk2b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr11:78781535-78784818:-</th>\n",
       "      <td>-0.643</td>\n",
       "      <td>ENSMUSG00000001123.9</td>\n",
       "      <td>Lgals9_(93)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Lgals9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr12:88781364-88782592:-</th>\n",
       "      <td>-0.730</td>\n",
       "      <td>ENSMUSG00000079036.3</td>\n",
       "      <td>Alkbh1_(73)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Alkbh1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr13:103541690-103548581:-</th>\n",
       "      <td>1.072</td>\n",
       "      <td>ENSMUSG00000034751.8</td>\n",
       "      <td>Mast4_(201)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Mast4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr13:46848241-46856641:-</th>\n",
       "      <td>0.556</td>\n",
       "      <td>ENSMUSG00000021375.8</td>\n",
       "      <td>Kif13a_(105)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Kif13a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr13:59634580-59637630:-</th>\n",
       "      <td>0.933</td>\n",
       "      <td>ENSMUSG00000021557.7</td>\n",
       "      <td>Agtpbp1_(150)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Agtpbp1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr14:13324377-13338297:-</th>\n",
       "      <td>0.754</td>\n",
       "      <td>ENSMUSG00000054423.6</td>\n",
       "      <td>Cadps_(159)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Cadps</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr14:52530460-52531140:-</th>\n",
       "      <td>-0.678</td>\n",
       "      <td>ENSMUSG00000004558.8</td>\n",
       "      <td>Ndrg2_(42)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Ndrg2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr14:58465899-58475446:-</th>\n",
       "      <td>0.627</td>\n",
       "      <td>ENSMUSG00000021969.6</td>\n",
       "      <td>Zdhhc20_(36)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Zdhhc20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr14:69984705-69991069:-</th>\n",
       "      <td>0.585</td>\n",
       "      <td>ENSMUSG00000085092.1</td>\n",
       "      <td>uc007uml-1_(153)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Gm16867</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr15:82203007-82203541:-</th>\n",
       "      <td>-0.574</td>\n",
       "      <td>ENSMUSG00000061740.6</td>\n",
       "      <td>Cyp2d22_(142)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Cyp2d22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr17:66729776-66735199:-</th>\n",
       "      <td>0.884</td>\n",
       "      <td>ENSMUSG00000052105.8</td>\n",
       "      <td>1110012J17Rik_(75)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>1110012J17Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr18:60970072-60971512:+</th>\n",
       "      <td>-0.875</td>\n",
       "      <td>ENSMUSG00000024610.8</td>\n",
       "      <td>Cd74_(192)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Cd74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr19:3986611-3987050:+</th>\n",
       "      <td>0.705</td>\n",
       "      <td>ENSMUSG00000024866.9</td>\n",
       "      <td>Acy3_(129)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Acy3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr19:46443823-46446671:+</th>\n",
       "      <td>-0.790</td>\n",
       "      <td>ENSMUSG00000025227.8</td>\n",
       "      <td>Tmem180_(172)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Tmem180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr19:46943121-46951697:+</th>\n",
       "      <td>0.706</td>\n",
       "      <td>ENSMUSG00000064105.5</td>\n",
       "      <td>Cnnm2_(66)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Cnnm2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr19:7062745-7064191:-</th>\n",
       "      <td>-0.689</td>\n",
       "      <td>ENSMUSG00000024963.4</td>\n",
       "      <td>Dnajc4_(162)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Dnajc4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:120218554-120224222:+</th>\n",
       "      <td>0.658</td>\n",
       "      <td>ENSMUSG00000026377.6</td>\n",
       "      <td>Mki67ip_(144)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Mki67ip</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:14273279-14294505:-</th>\n",
       "      <td>0.634</td>\n",
       "      <td>ENSMUSG00000025932.7</td>\n",
       "      <td>Eya1_(128)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Eya1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:34306483-34307454:+</th>\n",
       "      <td>0.757</td>\n",
       "      <td>ENSMUSG00000026131.9</td>\n",
       "      <td>Dst_(327)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Dst</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:58476577-58477926:-</th>\n",
       "      <td>0.789</td>\n",
       "      <td>ENSMUSG00000026034.10</td>\n",
       "      <td>Clk1_(91)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Clk1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1:68089312-68118091:-</th>\n",
       "      <td>1.176</td>\n",
       "      <td>ENSMUSG00000062209.8</td>\n",
       "      <td>Erbb4_(48)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Erbb4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"30\" valign=\"top\">tls</th>\n",
       "      <th>chr7:97540963-97543672:+</th>\n",
       "      <td>-0.508</td>\n",
       "      <td>ENSMUSG00000030616.7</td>\n",
       "      <td>Sytl2_(93)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Sytl2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr8:108223391-108223760:-</th>\n",
       "      <td>0.670</td>\n",
       "      <td>ENSMUSG00000038000.8</td>\n",
       "      <td>Acd_(152)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Acd</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr8:124963652-124964081:-</th>\n",
       "      <td>-0.665</td>\n",
       "      <td>ENSMUSG00000006517.4</td>\n",
       "      <td>Mvd_(103)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Mvd</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr8:23584052-23586977:-</th>\n",
       "      <td>-0.505</td>\n",
       "      <td>ENSMUSG00000031534.8</td>\n",
       "      <td>AI316807_(59)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>AI316807</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr8:3634123-3635446:+</th>\n",
       "      <td>0.524</td>\n",
       "      <td>ENSMUSG00000004626.8</td>\n",
       "      <td>Stxbp2_(149)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Stxbp2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr8:46821556-46830773:+</th>\n",
       "      <td>0.707</td>\n",
       "      <td>ENSMUSG00000031626.9</td>\n",
       "      <td>Sorbs2_(93)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Sorbs2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr8:63372657-63375534:+</th>\n",
       "      <td>-0.641</td>\n",
       "      <td>ENSMUSG00000038005.8</td>\n",
       "      <td>2700029M09Rik_(190)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>2700029M09Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr8:72828575-72830418:+</th>\n",
       "      <td>0.543</td>\n",
       "      <td>ENSMUSG00000055681.8</td>\n",
       "      <td>Cope_(101)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Cope</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr8:86097770-86098377:-</th>\n",
       "      <td>-0.613</td>\n",
       "      <td>ENSMUSG00000031708.9</td>\n",
       "      <td>Gpsn2_(52)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Tecr</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr8:88420436-88425775:+</th>\n",
       "      <td>0.685</td>\n",
       "      <td>ENSMUSG00000036879.8</td>\n",
       "      <td>Phkb_(89)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Phkb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:107659323-107662652:-</th>\n",
       "      <td>0.582</td>\n",
       "      <td>ENSMUSG00000032580.5</td>\n",
       "      <td>Rbm5_(74)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Rbm5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:112088273-112090093:-</th>\n",
       "      <td>0.617</td>\n",
       "      <td>ENSMUSG00000032503.10</td>\n",
       "      <td>Arpp21_(38)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Arpp21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:118446041-118448015:+</th>\n",
       "      <td>0.574</td>\n",
       "      <td>ENSMUSG00000038708.8</td>\n",
       "      <td>Golga4_(84)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Golga4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:44495781-44496699:-</th>\n",
       "      <td>-0.543</td>\n",
       "      <td>ENSMUSG00000048537.8</td>\n",
       "      <td>Phldb1_(33)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Phldb1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:44581867-44590183:+</th>\n",
       "      <td>0.742</td>\n",
       "      <td>ENSMUSG00000002031.4</td>\n",
       "      <td>1500035H01Rik_(77)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Ift46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:45734911-45735657:+</th>\n",
       "      <td>-0.542</td>\n",
       "      <td>ENSMUSG00000035382.7</td>\n",
       "      <td>Pcsk7_(82)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Pcsk7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:57106764-57110236:-</th>\n",
       "      <td>0.598</td>\n",
       "      <td>ENSMUSG00000032300.5</td>\n",
       "      <td>1700017B05Rik_(184)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>1700017B05Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:58047770-58049107:-</th>\n",
       "      <td>0.643</td>\n",
       "      <td>ENSMUSG00000051243.6</td>\n",
       "      <td>Islr2_(190)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Islr2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:59488903-59496636:+</th>\n",
       "      <td>0.571</td>\n",
       "      <td>ENSMUSG00000025237.8</td>\n",
       "      <td>Parp6_(70)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Parp6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:69851474-69854870:+</th>\n",
       "      <td>-0.798</td>\n",
       "      <td>ENSMUSG00000011958.11</td>\n",
       "      <td>Bnip2_(36)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Bnip2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:70324475-70351265:-</th>\n",
       "      <td>0.543</td>\n",
       "      <td>ENSMUSG00000032217.6</td>\n",
       "      <td>Rnf111_(63)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Rnf111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:75041944-75045449:+</th>\n",
       "      <td>-0.633</td>\n",
       "      <td>ENSMUSG00000034593.10</td>\n",
       "      <td>Myo5a_(75)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Myo5a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:96188141-96195383:+</th>\n",
       "      <td>-0.551</td>\n",
       "      <td>ENSMUSG00000032411.8</td>\n",
       "      <td>Tfdp2_(48)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Tfdp2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr9:99438178-99469020:-</th>\n",
       "      <td>0.644</td>\n",
       "      <td>ENSMUSG00000032468.5</td>\n",
       "      <td>Armc8_(77)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Armc8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:102583275-102586814:-</th>\n",
       "      <td>1.045</td>\n",
       "      <td>ENSMUSG00000087620.1</td>\n",
       "      <td>uc009uax-1_(74)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>5330434G04Rik</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:45976985-45982449:+</th>\n",
       "      <td>0.630</td>\n",
       "      <td>ENSMUSG00000031105.9</td>\n",
       "      <td>Slc25a14_(236)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Slc25a14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:61526284-61529974:-</th>\n",
       "      <td>0.671</td>\n",
       "      <td>ENSMUSG00000046699.8</td>\n",
       "      <td>Slitrk4_(72)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Slitrk4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:71533852-71534443:+</th>\n",
       "      <td>-1.009</td>\n",
       "      <td>ENSMUSG00000009995.10</td>\n",
       "      <td>Taz_(53)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Taz</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:7780941-7782711:-</th>\n",
       "      <td>0.512</td>\n",
       "      <td>ENSMUSG00000031169.7</td>\n",
       "      <td>Porcn_(122)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Porcn</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:98497770-98502414:+</th>\n",
       "      <td>-0.514</td>\n",
       "      <td>ENSMUSG00000031302.9</td>\n",
       "      <td>Nlgn3_(60)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Nlgn3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>557 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   sep_score                gene_id  \\\n",
       "rbp   location                                                        \n",
       "taf15 chr10:106364918-106368179:+     -0.911   ENSMUSG00000053825.7   \n",
       "      chr10:119487172-119491888:+     -0.535  ENSMUSG00000034813.10   \n",
       "      chr10:5299010-5302787:+          0.776   ENSMUSG00000019769.9   \n",
       "      chr11:106616365-106622496:+     -0.677   ENSMUSG00000040528.7   \n",
       "      chr11:106732924-106781494:-      0.582  ENSMUSG00000018363.10   \n",
       "      chr11:115865028-115866865:+     -0.696   ENSMUSG00000020758.9   \n",
       "      chr11:115867081-115867774:+     -0.844   ENSMUSG00000020758.9   \n",
       "      chr11:4602223-4604160:-         -0.777   ENSMUSG00000059534.8   \n",
       "      chr11:5879724-5882503:-         -0.548   ENSMUSG00000057897.7   \n",
       "      chr11:78781535-78784818:-       -0.643   ENSMUSG00000001123.9   \n",
       "      chr12:88781364-88782592:-       -0.730   ENSMUSG00000079036.3   \n",
       "      chr13:103541690-103548581:-      1.072   ENSMUSG00000034751.8   \n",
       "      chr13:46848241-46856641:-        0.556   ENSMUSG00000021375.8   \n",
       "      chr13:59634580-59637630:-        0.933   ENSMUSG00000021557.7   \n",
       "      chr14:13324377-13338297:-        0.754   ENSMUSG00000054423.6   \n",
       "      chr14:52530460-52531140:-       -0.678   ENSMUSG00000004558.8   \n",
       "      chr14:58465899-58475446:-        0.627   ENSMUSG00000021969.6   \n",
       "      chr14:69984705-69991069:-        0.585   ENSMUSG00000085092.1   \n",
       "      chr15:82203007-82203541:-       -0.574   ENSMUSG00000061740.6   \n",
       "      chr17:66729776-66735199:-        0.884   ENSMUSG00000052105.8   \n",
       "      chr18:60970072-60971512:+       -0.875   ENSMUSG00000024610.8   \n",
       "      chr19:3986611-3987050:+          0.705   ENSMUSG00000024866.9   \n",
       "      chr19:46443823-46446671:+       -0.790   ENSMUSG00000025227.8   \n",
       "      chr19:46943121-46951697:+        0.706   ENSMUSG00000064105.5   \n",
       "      chr19:7062745-7064191:-         -0.689   ENSMUSG00000024963.4   \n",
       "      chr1:120218554-120224222:+       0.658   ENSMUSG00000026377.6   \n",
       "      chr1:14273279-14294505:-         0.634   ENSMUSG00000025932.7   \n",
       "      chr1:34306483-34307454:+         0.757   ENSMUSG00000026131.9   \n",
       "      chr1:58476577-58477926:-         0.789  ENSMUSG00000026034.10   \n",
       "      chr1:68089312-68118091:-         1.176   ENSMUSG00000062209.8   \n",
       "...                                      ...                    ...   \n",
       "tls   chr7:97540963-97543672:+        -0.508   ENSMUSG00000030616.7   \n",
       "      chr8:108223391-108223760:-       0.670   ENSMUSG00000038000.8   \n",
       "      chr8:124963652-124964081:-      -0.665   ENSMUSG00000006517.4   \n",
       "      chr8:23584052-23586977:-        -0.505   ENSMUSG00000031534.8   \n",
       "      chr8:3634123-3635446:+           0.524   ENSMUSG00000004626.8   \n",
       "      chr8:46821556-46830773:+         0.707   ENSMUSG00000031626.9   \n",
       "      chr8:63372657-63375534:+        -0.641   ENSMUSG00000038005.8   \n",
       "      chr8:72828575-72830418:+         0.543   ENSMUSG00000055681.8   \n",
       "      chr8:86097770-86098377:-        -0.613   ENSMUSG00000031708.9   \n",
       "      chr8:88420436-88425775:+         0.685   ENSMUSG00000036879.8   \n",
       "      chr9:107659323-107662652:-       0.582   ENSMUSG00000032580.5   \n",
       "      chr9:112088273-112090093:-       0.617  ENSMUSG00000032503.10   \n",
       "      chr9:118446041-118448015:+       0.574   ENSMUSG00000038708.8   \n",
       "      chr9:44495781-44496699:-        -0.543   ENSMUSG00000048537.8   \n",
       "      chr9:44581867-44590183:+         0.742   ENSMUSG00000002031.4   \n",
       "      chr9:45734911-45735657:+        -0.542   ENSMUSG00000035382.7   \n",
       "      chr9:57106764-57110236:-         0.598   ENSMUSG00000032300.5   \n",
       "      chr9:58047770-58049107:-         0.643   ENSMUSG00000051243.6   \n",
       "      chr9:59488903-59496636:+         0.571   ENSMUSG00000025237.8   \n",
       "      chr9:69851474-69854870:+        -0.798  ENSMUSG00000011958.11   \n",
       "      chr9:70324475-70351265:-         0.543   ENSMUSG00000032217.6   \n",
       "      chr9:75041944-75045449:+        -0.633  ENSMUSG00000034593.10   \n",
       "      chr9:96188141-96195383:+        -0.551   ENSMUSG00000032411.8   \n",
       "      chr9:99438178-99469020:-         0.644   ENSMUSG00000032468.5   \n",
       "      chrX:102583275-102586814:-       1.045   ENSMUSG00000087620.1   \n",
       "      chrX:45976985-45982449:+         0.630   ENSMUSG00000031105.9   \n",
       "      chrX:61526284-61529974:-         0.671   ENSMUSG00000046699.8   \n",
       "      chrX:71533852-71534443:+        -1.009  ENSMUSG00000009995.10   \n",
       "      chrX:7780941-7782711:-           0.512   ENSMUSG00000031169.7   \n",
       "      chrX:98497770-98502414:+        -0.514   ENSMUSG00000031302.9   \n",
       "\n",
       "                                  old_splice_annotation   splice_type  \\\n",
       "rbp   location                                                          \n",
       "taf15 chr10:106364918-106368179:+           Ppfia2_(92)  alt_cassette   \n",
       "      chr10:119487172-119491888:+            Grip1_(45)  alt_cassette   \n",
       "      chr10:5299010-5302787:+           uc007egt-1_(69)  alt_cassette   \n",
       "      chr11:106616365-106622496:+           Gm885_(112)  alt_cassette   \n",
       "      chr11:106732924-106781494:-           Smurf2_(39)  alt_cassette   \n",
       "      chr11:115865028-115866865:+           Itgb4_(195)  alt_cassette   \n",
       "      chr11:115867081-115867774:+           Itgb4_(159)  alt_cassette   \n",
       "      chr11:4602223-4604160:-        1110020P15Rik_(86)  alt_cassette   \n",
       "      chr11:5879724-5882503:-               Camk2b_(45)  alt_cassette   \n",
       "      chr11:78781535-78784818:-             Lgals9_(93)  alt_cassette   \n",
       "      chr12:88781364-88782592:-             Alkbh1_(73)  alt_cassette   \n",
       "      chr13:103541690-103548581:-           Mast4_(201)  alt_cassette   \n",
       "      chr13:46848241-46856641:-            Kif13a_(105)  alt_cassette   \n",
       "      chr13:59634580-59637630:-           Agtpbp1_(150)  alt_cassette   \n",
       "      chr14:13324377-13338297:-             Cadps_(159)  alt_cassette   \n",
       "      chr14:52530460-52531140:-              Ndrg2_(42)  alt_cassette   \n",
       "      chr14:58465899-58475446:-            Zdhhc20_(36)  alt_cassette   \n",
       "      chr14:69984705-69991069:-        uc007uml-1_(153)  alt_cassette   \n",
       "      chr15:82203007-82203541:-           Cyp2d22_(142)  alt_cassette   \n",
       "      chr17:66729776-66735199:-      1110012J17Rik_(75)  alt_cassette   \n",
       "      chr18:60970072-60971512:+              Cd74_(192)  alt_cassette   \n",
       "      chr19:3986611-3987050:+                Acy3_(129)  alt_cassette   \n",
       "      chr19:46443823-46446671:+           Tmem180_(172)  alt_cassette   \n",
       "      chr19:46943121-46951697:+              Cnnm2_(66)  alt_cassette   \n",
       "      chr19:7062745-7064191:-              Dnajc4_(162)  alt_cassette   \n",
       "      chr1:120218554-120224222:+          Mki67ip_(144)  alt_cassette   \n",
       "      chr1:14273279-14294505:-               Eya1_(128)  alt_cassette   \n",
       "      chr1:34306483-34307454:+                Dst_(327)  alt_cassette   \n",
       "      chr1:58476577-58477926:-                Clk1_(91)  alt_cassette   \n",
       "      chr1:68089312-68118091:-               Erbb4_(48)  alt_cassette   \n",
       "...                                                 ...           ...   \n",
       "tls   chr7:97540963-97543672:+               Sytl2_(93)  alt_cassette   \n",
       "      chr8:108223391-108223760:-              Acd_(152)  alt_cassette   \n",
       "      chr8:124963652-124964081:-              Mvd_(103)  alt_cassette   \n",
       "      chr8:23584052-23586977:-            AI316807_(59)  alt_cassette   \n",
       "      chr8:3634123-3635446:+               Stxbp2_(149)  alt_cassette   \n",
       "      chr8:46821556-46830773:+              Sorbs2_(93)  alt_cassette   \n",
       "      chr8:63372657-63375534:+      2700029M09Rik_(190)  alt_cassette   \n",
       "      chr8:72828575-72830418:+               Cope_(101)  alt_cassette   \n",
       "      chr8:86097770-86098377:-               Gpsn2_(52)  alt_cassette   \n",
       "      chr8:88420436-88425775:+                Phkb_(89)  alt_cassette   \n",
       "      chr9:107659323-107662652:-              Rbm5_(74)  alt_cassette   \n",
       "      chr9:112088273-112090093:-            Arpp21_(38)  alt_cassette   \n",
       "      chr9:118446041-118448015:+            Golga4_(84)  alt_cassette   \n",
       "      chr9:44495781-44496699:-              Phldb1_(33)  alt_cassette   \n",
       "      chr9:44581867-44590183:+       1500035H01Rik_(77)  alt_cassette   \n",
       "      chr9:45734911-45735657:+               Pcsk7_(82)  alt_cassette   \n",
       "      chr9:57106764-57110236:-      1700017B05Rik_(184)  alt_cassette   \n",
       "      chr9:58047770-58049107:-              Islr2_(190)  alt_cassette   \n",
       "      chr9:59488903-59496636:+               Parp6_(70)  alt_cassette   \n",
       "      chr9:69851474-69854870:+               Bnip2_(36)  alt_cassette   \n",
       "      chr9:70324475-70351265:-              Rnf111_(63)  alt_cassette   \n",
       "      chr9:75041944-75045449:+               Myo5a_(75)  alt_cassette   \n",
       "      chr9:96188141-96195383:+               Tfdp2_(48)  alt_cassette   \n",
       "      chr9:99438178-99469020:-               Armc8_(77)  alt_cassette   \n",
       "      chrX:102583275-102586814:-        uc009uax-1_(74)  alt_cassette   \n",
       "      chrX:45976985-45982449:+           Slc25a14_(236)  alt_cassette   \n",
       "      chrX:61526284-61529974:-             Slitrk4_(72)  alt_cassette   \n",
       "      chrX:71533852-71534443:+                 Taz_(53)  alt_cassette   \n",
       "      chrX:7780941-7782711:-                Porcn_(122)  alt_cassette   \n",
       "      chrX:98497770-98502414:+               Nlgn3_(60)  alt_cassette   \n",
       "\n",
       "                                       gene_name  \n",
       "rbp   location                                    \n",
       "taf15 chr10:106364918-106368179:+         Ppfia2  \n",
       "      chr10:119487172-119491888:+          Grip1  \n",
       "      chr10:5299010-5302787:+              Syne1  \n",
       "      chr11:106616365-106622496:+          Gm885  \n",
       "      chr11:106732924-106781494:-         Smurf2  \n",
       "      chr11:115865028-115866865:+          Itgb4  \n",
       "      chr11:115867081-115867774:+          Itgb4  \n",
       "      chr11:4602223-4604160:-             Uqcr10  \n",
       "      chr11:5879724-5882503:-             Camk2b  \n",
       "      chr11:78781535-78784818:-           Lgals9  \n",
       "      chr12:88781364-88782592:-           Alkbh1  \n",
       "      chr13:103541690-103548581:-          Mast4  \n",
       "      chr13:46848241-46856641:-           Kif13a  \n",
       "      chr13:59634580-59637630:-          Agtpbp1  \n",
       "      chr14:13324377-13338297:-            Cadps  \n",
       "      chr14:52530460-52531140:-            Ndrg2  \n",
       "      chr14:58465899-58475446:-          Zdhhc20  \n",
       "      chr14:69984705-69991069:-          Gm16867  \n",
       "      chr15:82203007-82203541:-          Cyp2d22  \n",
       "      chr17:66729776-66735199:-    1110012J17Rik  \n",
       "      chr18:60970072-60971512:+             Cd74  \n",
       "      chr19:3986611-3987050:+               Acy3  \n",
       "      chr19:46443823-46446671:+          Tmem180  \n",
       "      chr19:46943121-46951697:+            Cnnm2  \n",
       "      chr19:7062745-7064191:-             Dnajc4  \n",
       "      chr1:120218554-120224222:+         Mki67ip  \n",
       "      chr1:14273279-14294505:-              Eya1  \n",
       "      chr1:34306483-34307454:+               Dst  \n",
       "      chr1:58476577-58477926:-              Clk1  \n",
       "      chr1:68089312-68118091:-             Erbb4  \n",
       "...                                          ...  \n",
       "tls   chr7:97540963-97543672:+             Sytl2  \n",
       "      chr8:108223391-108223760:-             Acd  \n",
       "      chr8:124963652-124964081:-             Mvd  \n",
       "      chr8:23584052-23586977:-          AI316807  \n",
       "      chr8:3634123-3635446:+              Stxbp2  \n",
       "      chr8:46821556-46830773:+            Sorbs2  \n",
       "      chr8:63372657-63375534:+     2700029M09Rik  \n",
       "      chr8:72828575-72830418:+              Cope  \n",
       "      chr8:86097770-86098377:-              Tecr  \n",
       "      chr8:88420436-88425775:+              Phkb  \n",
       "      chr9:107659323-107662652:-            Rbm5  \n",
       "      chr9:112088273-112090093:-          Arpp21  \n",
       "      chr9:118446041-118448015:+          Golga4  \n",
       "      chr9:44495781-44496699:-            Phldb1  \n",
       "      chr9:44581867-44590183:+             Ift46  \n",
       "      chr9:45734911-45735657:+             Pcsk7  \n",
       "      chr9:57106764-57110236:-     1700017B05Rik  \n",
       "      chr9:58047770-58049107:-             Islr2  \n",
       "      chr9:59488903-59496636:+             Parp6  \n",
       "      chr9:69851474-69854870:+             Bnip2  \n",
       "      chr9:70324475-70351265:-            Rnf111  \n",
       "      chr9:75041944-75045449:+             Myo5a  \n",
       "      chr9:96188141-96195383:+             Tfdp2  \n",
       "      chr9:99438178-99469020:-             Armc8  \n",
       "      chrX:102583275-102586814:-   5330434G04Rik  \n",
       "      chrX:45976985-45982449:+          Slc25a14  \n",
       "      chrX:61526284-61529974:-           Slitrk4  \n",
       "      chrX:71533852-71534443:+               Taz  \n",
       "      chrX:7780941-7782711:-               Porcn  \n",
       "      chrX:98497770-98502414:+             Nlgn3  \n",
       "\n",
       "[557 rows x 5 columns]"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'MultiIndex Slicing requires the index to be fully lexsorted tuple len (1), lexsort depth (0)'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-72-d91fb486fb86>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mheatmap_prep\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0maxis\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'taf15'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'tdp43'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'tls'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;32m/nas3/gpratt/anaconda/lib/python2.7/site-packages/pandas/core/indexing.pyc\u001b[0m in \u001b[0;36m__getitem__\u001b[1;34m(self, key)\u001b[0m\n\u001b[0;32m   1215\u001b[0m             \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_getitem_tuple\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1216\u001b[0m         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1217\u001b[1;33m             \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_getitem_axis\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1218\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1219\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m_getitem_axis\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m/nas3/gpratt/anaconda/lib/python2.7/site-packages/pandas/core/indexing.pyc\u001b[0m in \u001b[0;36m_getitem_axis\u001b[1;34m(self, key, axis)\u001b[0m\n\u001b[0;32m   1353\u001b[0m             \u001b[1;31m# nested tuple slicing\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1354\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mis_nested_tuple\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mlabels\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1355\u001b[1;33m                 \u001b[0mlocs\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mlabels\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget_locs\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1356\u001b[0m                 \u001b[0mindexer\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m[\u001b[0m \u001b[0mslice\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mNone\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m]\u001b[0m \u001b[1;33m*\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mndim\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1357\u001b[0m                 \u001b[0mindexer\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0maxis\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mlocs\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m/nas3/gpratt/anaconda/lib/python2.7/site-packages/pandas/core/index.pyc\u001b[0m in \u001b[0;36mget_locs\u001b[1;34m(self, tup)\u001b[0m\n\u001b[0;32m   4532\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mis_lexsorted_for_tuple\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mtup\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   4533\u001b[0m             raise KeyError('MultiIndex Slicing requires the index to be fully lexsorted'\n\u001b[1;32m-> 4534\u001b[1;33m                            ' tuple len ({0}), lexsort depth ({1})'.format(len(tup), self.lexsort_depth))\n\u001b[0m\u001b[0;32m   4535\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   4536\u001b[0m         \u001b[1;32mdef\u001b[0m \u001b[0m_convert_indexer\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mr\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mKeyError\u001b[0m: 'MultiIndex Slicing requires the index to be fully lexsorted tuple len (1), lexsort depth (0)'"
     ]
    }
   ],
   "source": [
    "foo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "heatmap_prep = alt_cassette_events.xs(\"striatum\", level=\"cell_type\").sort_index().ix[['taf15', 'tdp43', 'tls']] \n",
    "\n",
    "heatmap_prep = heatmap_prep.sep_score.unstack()\n",
    "heatmap_prep = heatmap_prep.fillna(0)\n",
    "heatmap_prep.index = np.array([rbp_name_dict[item] for item in heatmap_prep.index])\n",
    "sns.clustermap(heatmap_prep, linewidths=0, xticklabels=False, figsize=(4,4))\n",
    "plt.savefig(os.path.join(img_dir, \"splicing_events_straitum_significant.pdf\"))\n",
    "plt.savefig(os.path.join(img_dir, \"splicing_events_straitum_significant.svg\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "sep_scores = alt_cassette_events_full.xs(\"striatum\", level=\"cell_type\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>sep_score</th>\n",
       "      <th>q_value</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>old_splice_annotation</th>\n",
       "      <th>splice_type</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>chrom_internal</th>\n",
       "      <th>start_internal</th>\n",
       "      <th>stop_internal</th>\n",
       "      <th>event_id</th>\n",
       "      <th>strand_internal</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rbp</th>\n",
       "      <th>location</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"30\" valign=\"top\">taf15</th>\n",
       "      <th>chr10:100041065-100046548:-</th>\n",
       "      <td>0.082</td>\n",
       "      <td>0.345</td>\n",
       "      <td>error</td>\n",
       "      <td>4930430F08Rik_(60)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>error</td>\n",
       "      <td>chr10</td>\n",
       "      <td>100044896</td>\n",
       "      <td>100044956</td>\n",
       "      <td>4930430F08Rik_(60)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:10409700-10439020:-</th>\n",
       "      <td>0.376</td>\n",
       "      <td>0.379</td>\n",
       "      <td>ENSMUSG00000031153.9</td>\n",
       "      <td>Grm1_(85)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Gripap1</td>\n",
       "      <td>chr10</td>\n",
       "      <td>10417845</td>\n",
       "      <td>10417930</td>\n",
       "      <td>Grm1_(85)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:106199203-106237793:+</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000037519.8</td>\n",
       "      <td>Ppfia2_(75)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Ppfia1</td>\n",
       "      <td>chr10</td>\n",
       "      <td>106204529</td>\n",
       "      <td>106204604</td>\n",
       "      <td>Ppfia2_(75)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:106364918-106368179:+</th>\n",
       "      <td>-0.911</td>\n",
       "      <td>0.000</td>\n",
       "      <td>ENSMUSG00000053825.7</td>\n",
       "      <td>Ppfia2_(92)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Ppfia2</td>\n",
       "      <td>chr10</td>\n",
       "      <td>106366109</td>\n",
       "      <td>106366201</td>\n",
       "      <td>Ppfia2_(92)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:106708998-106817135:+</th>\n",
       "      <td>0.174</td>\n",
       "      <td>0.011</td>\n",
       "      <td>ENSMUSG00000019920.10</td>\n",
       "      <td>Lin7a_(119)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Lims1</td>\n",
       "      <td>chr10</td>\n",
       "      <td>106760966</td>\n",
       "      <td>106761085</td>\n",
       "      <td>Lin7a_(119)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:106817207-106819559:+</th>\n",
       "      <td>0.107</td>\n",
       "      <td>0.415</td>\n",
       "      <td>ENSMUSG00000019920.10</td>\n",
       "      <td>Lin7a_(77)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Lims1</td>\n",
       "      <td>chr10</td>\n",
       "      <td>106817282</td>\n",
       "      <td>106817359</td>\n",
       "      <td>Lin7a_(77)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:107689980-107696811:+</th>\n",
       "      <td>0.136</td>\n",
       "      <td>0.541</td>\n",
       "      <td>ENSMUSG00000019907.8</td>\n",
       "      <td>Ppp1r12a_(171)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Ppp1r12a</td>\n",
       "      <td>chr10</td>\n",
       "      <td>107690380</td>\n",
       "      <td>107690551</td>\n",
       "      <td>Ppp1r12a_(171)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:107708825-107711193:+</th>\n",
       "      <td>0.119</td>\n",
       "      <td>0.291</td>\n",
       "      <td>ENSMUSG00000019907.8</td>\n",
       "      <td>Ppp1r12a_(31)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Ppp1r12a</td>\n",
       "      <td>chr10</td>\n",
       "      <td>107710450</td>\n",
       "      <td>107710481</td>\n",
       "      <td>Ppp1r12a_(31)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:108339371-108447631:-</th>\n",
       "      <td>0.002</td>\n",
       "      <td>0.461</td>\n",
       "      <td>ENSMUSG00000058420.5</td>\n",
       "      <td>Syt1_(80)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Syt17</td>\n",
       "      <td>chr10</td>\n",
       "      <td>108430893</td>\n",
       "      <td>108430973</td>\n",
       "      <td>Syt1_(80)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:110641896-110670608:+</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000042359.9</td>\n",
       "      <td>Osbpl8_(37)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Osbpl6</td>\n",
       "      <td>chr10</td>\n",
       "      <td>110661727</td>\n",
       "      <td>110661764</td>\n",
       "      <td>Osbpl8_(37)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:111893591-111895546:+</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000018470.2</td>\n",
       "      <td>Kcnc2_(104)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Kcnab3</td>\n",
       "      <td>chr10</td>\n",
       "      <td>111893984</td>\n",
       "      <td>111894088</td>\n",
       "      <td>Kcnc2_(104)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:114810002-114821260:-</th>\n",
       "      <td>0.120</td>\n",
       "      <td>0.563</td>\n",
       "      <td>ENSMUSG00000048852.7</td>\n",
       "      <td>Thap2_(196)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Gm12185</td>\n",
       "      <td>chr10</td>\n",
       "      <td>114813407</td>\n",
       "      <td>114813603</td>\n",
       "      <td>Thap2_(196)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:115944244-115986088:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000056167.5</td>\n",
       "      <td>Cnot2_(67)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Cnot10</td>\n",
       "      <td>chr10</td>\n",
       "      <td>115954372</td>\n",
       "      <td>115954439</td>\n",
       "      <td>Cnot2_(67)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:115954439-115986088:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000056167.5</td>\n",
       "      <td>Cnot2_(123)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Cnot10</td>\n",
       "      <td>chr10</td>\n",
       "      <td>115964882</td>\n",
       "      <td>115965005</td>\n",
       "      <td>Cnot2_(123)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:115964985-116018529:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000020166.7</td>\n",
       "      <td>Cnot2_(73)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Cnot2</td>\n",
       "      <td>chr10</td>\n",
       "      <td>115986088</td>\n",
       "      <td>115986161</td>\n",
       "      <td>Cnot2_(73)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:115965005-115986088:-</th>\n",
       "      <td>-0.060</td>\n",
       "      <td>0.561</td>\n",
       "      <td>ENSMUSG00000056167.5</td>\n",
       "      <td>Cnot2_(228)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Cnot10</td>\n",
       "      <td>chr10</td>\n",
       "      <td>115974276</td>\n",
       "      <td>115974504</td>\n",
       "      <td>Cnot2_(228)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:116356707-116374452:-</th>\n",
       "      <td>0.254</td>\n",
       "      <td>0.031</td>\n",
       "      <td>ENSMUSG00000064181.4</td>\n",
       "      <td>Rab3ip_(96)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Rab3ip</td>\n",
       "      <td>chr10</td>\n",
       "      <td>116365244</td>\n",
       "      <td>116365340</td>\n",
       "      <td>Rab3ip_(96)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:116387748-116392583:+</th>\n",
       "      <td>0.047</td>\n",
       "      <td>0.417</td>\n",
       "      <td>error</td>\n",
       "      <td>uc007hcn-1_(213)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>error</td>\n",
       "      <td>chr10</td>\n",
       "      <td>116391568</td>\n",
       "      <td>116391781</td>\n",
       "      <td>uc007hcn-1_(213)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:116671537-116674673:-</th>\n",
       "      <td>0.327</td>\n",
       "      <td>0.102</td>\n",
       "      <td>ENSMUSG00000020177.6</td>\n",
       "      <td>9530003J23Rik_(79)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>9530003J23Rik</td>\n",
       "      <td>chr10</td>\n",
       "      <td>116672639</td>\n",
       "      <td>116672718</td>\n",
       "      <td>9530003J23Rik_(79)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:116798416-116799070:-</th>\n",
       "      <td>0.018</td>\n",
       "      <td>0.554</td>\n",
       "      <td>ENSMUSG00000018727.12</td>\n",
       "      <td>Cpsf6_(111)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Cpsf4l</td>\n",
       "      <td>chr10</td>\n",
       "      <td>116798829</td>\n",
       "      <td>116798940</td>\n",
       "      <td>Cpsf6_(111)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:117589330-117594251:+</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000034912.8</td>\n",
       "      <td>Mdm1_(30)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Mdga2</td>\n",
       "      <td>chr10</td>\n",
       "      <td>117593889</td>\n",
       "      <td>117593919</td>\n",
       "      <td>Mdm1_(30)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:119415663-119423387:+</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000026959.7</td>\n",
       "      <td>Grip1_(156)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Grin1</td>\n",
       "      <td>chr10</td>\n",
       "      <td>119422529</td>\n",
       "      <td>119422685</td>\n",
       "      <td>Grip1_(156)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:119487172-119491888:+</th>\n",
       "      <td>-0.535</td>\n",
       "      <td>0.041</td>\n",
       "      <td>ENSMUSG00000034813.10</td>\n",
       "      <td>Grip1_(45)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Grip1</td>\n",
       "      <td>chr10</td>\n",
       "      <td>119487287</td>\n",
       "      <td>119487332</td>\n",
       "      <td>Grip1_(45)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:119807080-119819120:+</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>error</td>\n",
       "      <td>uc007hfc-1_(48)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>error</td>\n",
       "      <td>chr10</td>\n",
       "      <td>119808312</td>\n",
       "      <td>119808360</td>\n",
       "      <td>uc007hfc-1_(48)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:121533150-121533956:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000034620.10</td>\n",
       "      <td>Tmem5_(114)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Tmem5</td>\n",
       "      <td>chr10</td>\n",
       "      <td>121533403</td>\n",
       "      <td>121533517</td>\n",
       "      <td>Tmem5_(114)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:122702597-123030549:+</th>\n",
       "      <td>0.084</td>\n",
       "      <td>0.213</td>\n",
       "      <td>ENSMUSG00000044071.6</td>\n",
       "      <td>Fam19a2_(212)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Fam19a2</td>\n",
       "      <td>chr10</td>\n",
       "      <td>122867767</td>\n",
       "      <td>122867979</td>\n",
       "      <td>Fam19a2_(212)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:124731900-124739012:-</th>\n",
       "      <td>-0.164</td>\n",
       "      <td>0.108</td>\n",
       "      <td>ENSMUSG00000041920.8</td>\n",
       "      <td>Slc16a7_(81)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Slc16a6</td>\n",
       "      <td>chr10</td>\n",
       "      <td>124735677</td>\n",
       "      <td>124735758</td>\n",
       "      <td>Slc16a7_(81)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:126501276-126501641:+</th>\n",
       "      <td>-0.127</td>\n",
       "      <td>0.221</td>\n",
       "      <td>ENSMUSG00000012819.8</td>\n",
       "      <td>Cdk4_(234)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Cdh23</td>\n",
       "      <td>chr10</td>\n",
       "      <td>126501275</td>\n",
       "      <td>126501509</td>\n",
       "      <td>Cdk4_(234)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:126526414-126527482:+</th>\n",
       "      <td>-0.401</td>\n",
       "      <td>0.000</td>\n",
       "      <td>ENSMUSG00000025422.7</td>\n",
       "      <td>Agap2_(60)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Agap2</td>\n",
       "      <td>chr10</td>\n",
       "      <td>126527197</td>\n",
       "      <td>126527257</td>\n",
       "      <td>Agap2_(60)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:126533798-126534987:-</th>\n",
       "      <td>0.095</td>\n",
       "      <td>0.539</td>\n",
       "      <td>ENSMUSG00000026761.6</td>\n",
       "      <td>Os9_(165)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Orc4</td>\n",
       "      <td>chr10</td>\n",
       "      <td>126533970</td>\n",
       "      <td>126534135</td>\n",
       "      <td>Os9_(165)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"30\" valign=\"top\">tls</th>\n",
       "      <th>chrX:70590251-70597147:+</th>\n",
       "      <td>-0.190</td>\n",
       "      <td>0.085</td>\n",
       "      <td>ENSMUSG00000031365.6</td>\n",
       "      <td>Zfp275_(92)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Zfp275</td>\n",
       "      <td>chrX</td>\n",
       "      <td>70596607</td>\n",
       "      <td>70596699</td>\n",
       "      <td>Zfp275_(92)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:70729101-70736118:+</th>\n",
       "      <td>-0.266</td>\n",
       "      <td>0.043</td>\n",
       "      <td>ENSMUSG00000050071.8</td>\n",
       "      <td>Bgn_(64)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Bex1</td>\n",
       "      <td>chrX</td>\n",
       "      <td>70735206</td>\n",
       "      <td>70735270</td>\n",
       "      <td>Bgn_(64)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:71004795-71005233:+</th>\n",
       "      <td>0.025</td>\n",
       "      <td>0.312</td>\n",
       "      <td>ENSMUSG00000016382.9</td>\n",
       "      <td>Plxnb3_(128)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Pls3</td>\n",
       "      <td>chrX</td>\n",
       "      <td>71004794</td>\n",
       "      <td>71004922</td>\n",
       "      <td>Plxnb3_(128)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:71034292-71035139:+</th>\n",
       "      <td>-0.263</td>\n",
       "      <td>0.137</td>\n",
       "      <td>ENSMUSG00000002014.6</td>\n",
       "      <td>Ssr4_(99)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Ssr4</td>\n",
       "      <td>chrX</td>\n",
       "      <td>71034393</td>\n",
       "      <td>71034492</td>\n",
       "      <td>Ssr4_(99)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:71151202-71151767:-</th>\n",
       "      <td>0.022</td>\n",
       "      <td>0.555</td>\n",
       "      <td>ENSMUSG00000039831.9</td>\n",
       "      <td>Arhgap4_(63)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Arhgap29</td>\n",
       "      <td>chrX</td>\n",
       "      <td>71151495</td>\n",
       "      <td>71151558</td>\n",
       "      <td>Arhgap4_(63)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:71262850-71263468:-</th>\n",
       "      <td>-0.071</td>\n",
       "      <td>0.196</td>\n",
       "      <td>ENSMUSG00000031392.11</td>\n",
       "      <td>Irak1_(117)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Irak1</td>\n",
       "      <td>chrX</td>\n",
       "      <td>71263239</td>\n",
       "      <td>71263356</td>\n",
       "      <td>Irak1_(117)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:71282671-71330848:-</th>\n",
       "      <td>-0.259</td>\n",
       "      <td>0.415</td>\n",
       "      <td>ENSMUSG00000024556.3</td>\n",
       "      <td>Mecp2_(124)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Me2</td>\n",
       "      <td>chrX</td>\n",
       "      <td>71325247</td>\n",
       "      <td>71325371</td>\n",
       "      <td>Mecp2_(124)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:71383926-71388785:+</th>\n",
       "      <td>0.201</td>\n",
       "      <td>0.108</td>\n",
       "      <td>ENSMUSG00000062257.6</td>\n",
       "      <td>Opn1mw_(166)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Opcml</td>\n",
       "      <td>chrX</td>\n",
       "      <td>71385141</td>\n",
       "      <td>71385307</td>\n",
       "      <td>Opn1mw_(166)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:71533852-71534443:+</th>\n",
       "      <td>-1.009</td>\n",
       "      <td>0.000</td>\n",
       "      <td>ENSMUSG00000009995.10</td>\n",
       "      <td>Taz_(53)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Taz</td>\n",
       "      <td>chrX</td>\n",
       "      <td>71534000</td>\n",
       "      <td>71534053</td>\n",
       "      <td>Taz_(53)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:7156972-7162937:+</th>\n",
       "      <td>0.021</td>\n",
       "      <td>0.272</td>\n",
       "      <td>ENSMUSG00000030067.9</td>\n",
       "      <td>Foxp3_(67)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Foxp1</td>\n",
       "      <td>chrX</td>\n",
       "      <td>7162696</td>\n",
       "      <td>7162763</td>\n",
       "      <td>Foxp3_(67)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:71613248-71613755:-</th>\n",
       "      <td>0.435</td>\n",
       "      <td>0.046</td>\n",
       "      <td>ENSMUSG00000028960.11</td>\n",
       "      <td>Ubl4_(106)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Ube4b</td>\n",
       "      <td>chrX</td>\n",
       "      <td>71613491</td>\n",
       "      <td>71613597</td>\n",
       "      <td>Ubl4_(106)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:73044829-73045841:-</th>\n",
       "      <td>0.213</td>\n",
       "      <td>0.064</td>\n",
       "      <td>ENSMUSG00000016382.9</td>\n",
       "      <td>Pls3_(27)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Pls3</td>\n",
       "      <td>chrX</td>\n",
       "      <td>73045055</td>\n",
       "      <td>73045082</td>\n",
       "      <td>Pls3_(27)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:7344810-7346555:+</th>\n",
       "      <td>0.098</td>\n",
       "      <td>0.036</td>\n",
       "      <td>ENSMUSG00000020167.8</td>\n",
       "      <td>Tcfe3_(105)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Tcf3</td>\n",
       "      <td>chrX</td>\n",
       "      <td>7345018</td>\n",
       "      <td>7345123</td>\n",
       "      <td>Tcfe3_(105)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:7380526-7381486:+</th>\n",
       "      <td>0.002</td>\n",
       "      <td>0.811</td>\n",
       "      <td>ENSMUSG00000031153.9</td>\n",
       "      <td>Gripap1_(93)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Gripap1</td>\n",
       "      <td>chrX</td>\n",
       "      <td>7380719</td>\n",
       "      <td>7380812</td>\n",
       "      <td>Gripap1_(93)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:7718334-7720376:-</th>\n",
       "      <td>-0.195</td>\n",
       "      <td>0.484</td>\n",
       "      <td>ENSMUSG00000027620.9</td>\n",
       "      <td>Rbm3_(129)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Rbm39</td>\n",
       "      <td>chrX</td>\n",
       "      <td>7719952</td>\n",
       "      <td>7720081</td>\n",
       "      <td>Rbm3_(129)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:7721053-7722101:-</th>\n",
       "      <td>0.297</td>\n",
       "      <td>0.042</td>\n",
       "      <td>ENSMUSG00000027620.9</td>\n",
       "      <td>Rbm3_(107)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Rbm39</td>\n",
       "      <td>chrX</td>\n",
       "      <td>7721698</td>\n",
       "      <td>7721805</td>\n",
       "      <td>Rbm3_(107)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:7780941-7782711:-</th>\n",
       "      <td>0.512</td>\n",
       "      <td>0.000</td>\n",
       "      <td>ENSMUSG00000031169.7</td>\n",
       "      <td>Porcn_(122)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Porcn</td>\n",
       "      <td>chrX</td>\n",
       "      <td>7781321</td>\n",
       "      <td>7781443</td>\n",
       "      <td>Porcn_(122)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:82991640-82999670:-</th>\n",
       "      <td>-0.168</td>\n",
       "      <td>0.000</td>\n",
       "      <td>ENSMUSG00000025534.10</td>\n",
       "      <td>Gyk_(84)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Gusb</td>\n",
       "      <td>chrX</td>\n",
       "      <td>82994950</td>\n",
       "      <td>82995034</td>\n",
       "      <td>Gyk_(84)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:90869254-90871826:-</th>\n",
       "      <td>0.138</td>\n",
       "      <td>0.160</td>\n",
       "      <td>ENSMUSG00000020260.8</td>\n",
       "      <td>Pola1_(112)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Pofut2</td>\n",
       "      <td>chrX</td>\n",
       "      <td>90869724</td>\n",
       "      <td>90869836</td>\n",
       "      <td>Pola1_(112)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:91347719-91368406:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000024695.9</td>\n",
       "      <td>Zfx_(68)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Zfp91</td>\n",
       "      <td>chrX</td>\n",
       "      <td>91359534</td>\n",
       "      <td>91359602</td>\n",
       "      <td>Zfx_(68)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:91785886-91787231:-</th>\n",
       "      <td>0.169</td>\n",
       "      <td>0.046</td>\n",
       "      <td>ENSMUSG00000036634.8</td>\n",
       "      <td>Maged1_(74)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Mag</td>\n",
       "      <td>chrX</td>\n",
       "      <td>91786589</td>\n",
       "      <td>91786663</td>\n",
       "      <td>Maged1_(74)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:9646690-9694569:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000000751.7</td>\n",
       "      <td>Rpgr_(60)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Rpa1</td>\n",
       "      <td>chrX</td>\n",
       "      <td>9650894</td>\n",
       "      <td>9650954</td>\n",
       "      <td>Rpgr_(60)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:9743402-9755396:-</th>\n",
       "      <td>-0.086</td>\n",
       "      <td>0.201</td>\n",
       "      <td>ENSMUSG00000000751.7</td>\n",
       "      <td>Rpgr_(879)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Rpa1</td>\n",
       "      <td>chrX</td>\n",
       "      <td>9753128</td>\n",
       "      <td>9754007</td>\n",
       "      <td>Rpgr_(879)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:97963767-97966953:+</th>\n",
       "      <td>-0.320</td>\n",
       "      <td>0.223</td>\n",
       "      <td>ENSMUSG00000000881.6</td>\n",
       "      <td>Dlg3_(54)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Dlg3</td>\n",
       "      <td>chrX</td>\n",
       "      <td>97966645</td>\n",
       "      <td>97966699</td>\n",
       "      <td>Dlg3_(54)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:98005246-98008064:+</th>\n",
       "      <td>0.210</td>\n",
       "      <td>0.000</td>\n",
       "      <td>ENSMUSG00000000881.6</td>\n",
       "      <td>Dlg3_(42)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Dlg3</td>\n",
       "      <td>chrX</td>\n",
       "      <td>98006840</td>\n",
       "      <td>98006882</td>\n",
       "      <td>Dlg3_(42)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:98279613-98280610:-</th>\n",
       "      <td>-0.253</td>\n",
       "      <td>0.187</td>\n",
       "      <td>ENSMUSG00000041313.8</td>\n",
       "      <td>Slc7a3_(158)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Slc7a1</td>\n",
       "      <td>chrX</td>\n",
       "      <td>98280159</td>\n",
       "      <td>98280317</td>\n",
       "      <td>Slc7a3_(158)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:98497770-98502414:+</th>\n",
       "      <td>-0.514</td>\n",
       "      <td>0.000</td>\n",
       "      <td>ENSMUSG00000031302.9</td>\n",
       "      <td>Nlgn3_(60)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Nlgn3</td>\n",
       "      <td>chrX</td>\n",
       "      <td>98498410</td>\n",
       "      <td>98498470</td>\n",
       "      <td>Nlgn3_(60)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:98601358-98603815:-</th>\n",
       "      <td>-0.423</td>\n",
       "      <td>0.000</td>\n",
       "      <td>ENSMUSG00000021945.7</td>\n",
       "      <td>Zmym3_(255)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Zmym2</td>\n",
       "      <td>chrX</td>\n",
       "      <td>98602543</td>\n",
       "      <td>98602798</td>\n",
       "      <td>Zmym3_(255)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:98611277-98612247:-</th>\n",
       "      <td>-0.382</td>\n",
       "      <td>0.000</td>\n",
       "      <td>ENSMUSG00000021945.7</td>\n",
       "      <td>Zmym3_(295)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Zmym2</td>\n",
       "      <td>chrX</td>\n",
       "      <td>98611505</td>\n",
       "      <td>98611800</td>\n",
       "      <td>Zmym3_(295)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:99750956-99754903:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000034055.10</td>\n",
       "      <td>Phka1_(177)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Phka1</td>\n",
       "      <td>chrX</td>\n",
       "      <td>99752434</td>\n",
       "      <td>99752611</td>\n",
       "      <td>Phka1_(177)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11979 rows Ã— 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   sep_score  q_value                gene_id  \\\n",
       "rbp   location                                                                 \n",
       "taf15 chr10:100041065-100046548:-      0.082    0.345                  error   \n",
       "      chr10:10409700-10439020:-        0.376    0.379   ENSMUSG00000031153.9   \n",
       "      chr10:106199203-106237793:+      0.000    1.000   ENSMUSG00000037519.8   \n",
       "      chr10:106364918-106368179:+     -0.911    0.000   ENSMUSG00000053825.7   \n",
       "      chr10:106708998-106817135:+      0.174    0.011  ENSMUSG00000019920.10   \n",
       "      chr10:106817207-106819559:+      0.107    0.415  ENSMUSG00000019920.10   \n",
       "      chr10:107689980-107696811:+      0.136    0.541   ENSMUSG00000019907.8   \n",
       "      chr10:107708825-107711193:+      0.119    0.291   ENSMUSG00000019907.8   \n",
       "      chr10:108339371-108447631:-      0.002    0.461   ENSMUSG00000058420.5   \n",
       "      chr10:110641896-110670608:+      0.000    1.000   ENSMUSG00000042359.9   \n",
       "      chr10:111893591-111895546:+      0.000    1.000   ENSMUSG00000018470.2   \n",
       "      chr10:114810002-114821260:-      0.120    0.563   ENSMUSG00000048852.7   \n",
       "      chr10:115944244-115986088:-      0.000    1.000   ENSMUSG00000056167.5   \n",
       "      chr10:115954439-115986088:-      0.000    1.000   ENSMUSG00000056167.5   \n",
       "      chr10:115964985-116018529:-      0.000    1.000   ENSMUSG00000020166.7   \n",
       "      chr10:115965005-115986088:-     -0.060    0.561   ENSMUSG00000056167.5   \n",
       "      chr10:116356707-116374452:-      0.254    0.031   ENSMUSG00000064181.4   \n",
       "      chr10:116387748-116392583:+      0.047    0.417                  error   \n",
       "      chr10:116671537-116674673:-      0.327    0.102   ENSMUSG00000020177.6   \n",
       "      chr10:116798416-116799070:-      0.018    0.554  ENSMUSG00000018727.12   \n",
       "      chr10:117589330-117594251:+      0.000    1.000   ENSMUSG00000034912.8   \n",
       "      chr10:119415663-119423387:+      0.000    1.000   ENSMUSG00000026959.7   \n",
       "      chr10:119487172-119491888:+     -0.535    0.041  ENSMUSG00000034813.10   \n",
       "      chr10:119807080-119819120:+      0.000    1.000                  error   \n",
       "      chr10:121533150-121533956:-      0.000    1.000  ENSMUSG00000034620.10   \n",
       "      chr10:122702597-123030549:+      0.084    0.213   ENSMUSG00000044071.6   \n",
       "      chr10:124731900-124739012:-     -0.164    0.108   ENSMUSG00000041920.8   \n",
       "      chr10:126501276-126501641:+     -0.127    0.221   ENSMUSG00000012819.8   \n",
       "      chr10:126526414-126527482:+     -0.401    0.000   ENSMUSG00000025422.7   \n",
       "      chr10:126533798-126534987:-      0.095    0.539   ENSMUSG00000026761.6   \n",
       "...                                      ...      ...                    ...   \n",
       "tls   chrX:70590251-70597147:+        -0.190    0.085   ENSMUSG00000031365.6   \n",
       "      chrX:70729101-70736118:+        -0.266    0.043   ENSMUSG00000050071.8   \n",
       "      chrX:71004795-71005233:+         0.025    0.312   ENSMUSG00000016382.9   \n",
       "      chrX:71034292-71035139:+        -0.263    0.137   ENSMUSG00000002014.6   \n",
       "      chrX:71151202-71151767:-         0.022    0.555   ENSMUSG00000039831.9   \n",
       "      chrX:71262850-71263468:-        -0.071    0.196  ENSMUSG00000031392.11   \n",
       "      chrX:71282671-71330848:-        -0.259    0.415   ENSMUSG00000024556.3   \n",
       "      chrX:71383926-71388785:+         0.201    0.108   ENSMUSG00000062257.6   \n",
       "      chrX:71533852-71534443:+        -1.009    0.000  ENSMUSG00000009995.10   \n",
       "      chrX:7156972-7162937:+           0.021    0.272   ENSMUSG00000030067.9   \n",
       "      chrX:71613248-71613755:-         0.435    0.046  ENSMUSG00000028960.11   \n",
       "      chrX:73044829-73045841:-         0.213    0.064   ENSMUSG00000016382.9   \n",
       "      chrX:7344810-7346555:+           0.098    0.036   ENSMUSG00000020167.8   \n",
       "      chrX:7380526-7381486:+           0.002    0.811   ENSMUSG00000031153.9   \n",
       "      chrX:7718334-7720376:-          -0.195    0.484   ENSMUSG00000027620.9   \n",
       "      chrX:7721053-7722101:-           0.297    0.042   ENSMUSG00000027620.9   \n",
       "      chrX:7780941-7782711:-           0.512    0.000   ENSMUSG00000031169.7   \n",
       "      chrX:82991640-82999670:-        -0.168    0.000  ENSMUSG00000025534.10   \n",
       "      chrX:90869254-90871826:-         0.138    0.160   ENSMUSG00000020260.8   \n",
       "      chrX:91347719-91368406:-         0.000    1.000   ENSMUSG00000024695.9   \n",
       "      chrX:91785886-91787231:-         0.169    0.046   ENSMUSG00000036634.8   \n",
       "      chrX:9646690-9694569:-           0.000    1.000   ENSMUSG00000000751.7   \n",
       "      chrX:9743402-9755396:-          -0.086    0.201   ENSMUSG00000000751.7   \n",
       "      chrX:97963767-97966953:+        -0.320    0.223   ENSMUSG00000000881.6   \n",
       "      chrX:98005246-98008064:+         0.210    0.000   ENSMUSG00000000881.6   \n",
       "      chrX:98279613-98280610:-        -0.253    0.187   ENSMUSG00000041313.8   \n",
       "      chrX:98497770-98502414:+        -0.514    0.000   ENSMUSG00000031302.9   \n",
       "      chrX:98601358-98603815:-        -0.423    0.000   ENSMUSG00000021945.7   \n",
       "      chrX:98611277-98612247:-        -0.382    0.000   ENSMUSG00000021945.7   \n",
       "      chrX:99750956-99754903:-         0.000    1.000  ENSMUSG00000034055.10   \n",
       "\n",
       "                                  old_splice_annotation   splice_type  \\\n",
       "rbp   location                                                          \n",
       "taf15 chr10:100041065-100046548:-    4930430F08Rik_(60)  alt_cassette   \n",
       "      chr10:10409700-10439020:-               Grm1_(85)  alt_cassette   \n",
       "      chr10:106199203-106237793:+           Ppfia2_(75)  alt_cassette   \n",
       "      chr10:106364918-106368179:+           Ppfia2_(92)  alt_cassette   \n",
       "      chr10:106708998-106817135:+           Lin7a_(119)  alt_cassette   \n",
       "      chr10:106817207-106819559:+            Lin7a_(77)  alt_cassette   \n",
       "      chr10:107689980-107696811:+        Ppp1r12a_(171)  alt_cassette   \n",
       "      chr10:107708825-107711193:+         Ppp1r12a_(31)  alt_cassette   \n",
       "      chr10:108339371-108447631:-             Syt1_(80)  alt_cassette   \n",
       "      chr10:110641896-110670608:+           Osbpl8_(37)  alt_cassette   \n",
       "      chr10:111893591-111895546:+           Kcnc2_(104)  alt_cassette   \n",
       "      chr10:114810002-114821260:-           Thap2_(196)  alt_cassette   \n",
       "      chr10:115944244-115986088:-            Cnot2_(67)  alt_cassette   \n",
       "      chr10:115954439-115986088:-           Cnot2_(123)  alt_cassette   \n",
       "      chr10:115964985-116018529:-            Cnot2_(73)  alt_cassette   \n",
       "      chr10:115965005-115986088:-           Cnot2_(228)  alt_cassette   \n",
       "      chr10:116356707-116374452:-           Rab3ip_(96)  alt_cassette   \n",
       "      chr10:116387748-116392583:+      uc007hcn-1_(213)  alt_cassette   \n",
       "      chr10:116671537-116674673:-    9530003J23Rik_(79)  alt_cassette   \n",
       "      chr10:116798416-116799070:-           Cpsf6_(111)  alt_cassette   \n",
       "      chr10:117589330-117594251:+             Mdm1_(30)  alt_cassette   \n",
       "      chr10:119415663-119423387:+           Grip1_(156)  alt_cassette   \n",
       "      chr10:119487172-119491888:+            Grip1_(45)  alt_cassette   \n",
       "      chr10:119807080-119819120:+       uc007hfc-1_(48)  alt_cassette   \n",
       "      chr10:121533150-121533956:-           Tmem5_(114)  alt_cassette   \n",
       "      chr10:122702597-123030549:+         Fam19a2_(212)  alt_cassette   \n",
       "      chr10:124731900-124739012:-          Slc16a7_(81)  alt_cassette   \n",
       "      chr10:126501276-126501641:+            Cdk4_(234)  alt_cassette   \n",
       "      chr10:126526414-126527482:+            Agap2_(60)  alt_cassette   \n",
       "      chr10:126533798-126534987:-             Os9_(165)  alt_cassette   \n",
       "...                                                 ...           ...   \n",
       "tls   chrX:70590251-70597147:+              Zfp275_(92)  alt_cassette   \n",
       "      chrX:70729101-70736118:+                 Bgn_(64)  alt_cassette   \n",
       "      chrX:71004795-71005233:+             Plxnb3_(128)  alt_cassette   \n",
       "      chrX:71034292-71035139:+                Ssr4_(99)  alt_cassette   \n",
       "      chrX:71151202-71151767:-             Arhgap4_(63)  alt_cassette   \n",
       "      chrX:71262850-71263468:-              Irak1_(117)  alt_cassette   \n",
       "      chrX:71282671-71330848:-              Mecp2_(124)  alt_cassette   \n",
       "      chrX:71383926-71388785:+             Opn1mw_(166)  alt_cassette   \n",
       "      chrX:71533852-71534443:+                 Taz_(53)  alt_cassette   \n",
       "      chrX:7156972-7162937:+                 Foxp3_(67)  alt_cassette   \n",
       "      chrX:71613248-71613755:-               Ubl4_(106)  alt_cassette   \n",
       "      chrX:73044829-73045841:-                Pls3_(27)  alt_cassette   \n",
       "      chrX:7344810-7346555:+                Tcfe3_(105)  alt_cassette   \n",
       "      chrX:7380526-7381486:+               Gripap1_(93)  alt_cassette   \n",
       "      chrX:7718334-7720376:-                 Rbm3_(129)  alt_cassette   \n",
       "      chrX:7721053-7722101:-                 Rbm3_(107)  alt_cassette   \n",
       "      chrX:7780941-7782711:-                Porcn_(122)  alt_cassette   \n",
       "      chrX:82991640-82999670:-                 Gyk_(84)  alt_cassette   \n",
       "      chrX:90869254-90871826:-              Pola1_(112)  alt_cassette   \n",
       "      chrX:91347719-91368406:-                 Zfx_(68)  alt_cassette   \n",
       "      chrX:91785886-91787231:-              Maged1_(74)  alt_cassette   \n",
       "      chrX:9646690-9694569:-                  Rpgr_(60)  alt_cassette   \n",
       "      chrX:9743402-9755396:-                 Rpgr_(879)  alt_cassette   \n",
       "      chrX:97963767-97966953:+                Dlg3_(54)  alt_cassette   \n",
       "      chrX:98005246-98008064:+                Dlg3_(42)  alt_cassette   \n",
       "      chrX:98279613-98280610:-             Slc7a3_(158)  alt_cassette   \n",
       "      chrX:98497770-98502414:+               Nlgn3_(60)  alt_cassette   \n",
       "      chrX:98601358-98603815:-              Zmym3_(255)  alt_cassette   \n",
       "      chrX:98611277-98612247:-              Zmym3_(295)  alt_cassette   \n",
       "      chrX:99750956-99754903:-              Phka1_(177)  alt_cassette   \n",
       "\n",
       "                                       gene_name chrom_internal  \\\n",
       "rbp   location                                                    \n",
       "taf15 chr10:100041065-100046548:-          error          chr10   \n",
       "      chr10:10409700-10439020:-          Gripap1          chr10   \n",
       "      chr10:106199203-106237793:+         Ppfia1          chr10   \n",
       "      chr10:106364918-106368179:+         Ppfia2          chr10   \n",
       "      chr10:106708998-106817135:+          Lims1          chr10   \n",
       "      chr10:106817207-106819559:+          Lims1          chr10   \n",
       "      chr10:107689980-107696811:+       Ppp1r12a          chr10   \n",
       "      chr10:107708825-107711193:+       Ppp1r12a          chr10   \n",
       "      chr10:108339371-108447631:-          Syt17          chr10   \n",
       "      chr10:110641896-110670608:+         Osbpl6          chr10   \n",
       "      chr10:111893591-111895546:+         Kcnab3          chr10   \n",
       "      chr10:114810002-114821260:-        Gm12185          chr10   \n",
       "      chr10:115944244-115986088:-         Cnot10          chr10   \n",
       "      chr10:115954439-115986088:-         Cnot10          chr10   \n",
       "      chr10:115964985-116018529:-          Cnot2          chr10   \n",
       "      chr10:115965005-115986088:-         Cnot10          chr10   \n",
       "      chr10:116356707-116374452:-         Rab3ip          chr10   \n",
       "      chr10:116387748-116392583:+          error          chr10   \n",
       "      chr10:116671537-116674673:-  9530003J23Rik          chr10   \n",
       "      chr10:116798416-116799070:-         Cpsf4l          chr10   \n",
       "      chr10:117589330-117594251:+          Mdga2          chr10   \n",
       "      chr10:119415663-119423387:+          Grin1          chr10   \n",
       "      chr10:119487172-119491888:+          Grip1          chr10   \n",
       "      chr10:119807080-119819120:+          error          chr10   \n",
       "      chr10:121533150-121533956:-          Tmem5          chr10   \n",
       "      chr10:122702597-123030549:+        Fam19a2          chr10   \n",
       "      chr10:124731900-124739012:-        Slc16a6          chr10   \n",
       "      chr10:126501276-126501641:+          Cdh23          chr10   \n",
       "      chr10:126526414-126527482:+          Agap2          chr10   \n",
       "      chr10:126533798-126534987:-           Orc4          chr10   \n",
       "...                                          ...            ...   \n",
       "tls   chrX:70590251-70597147:+            Zfp275           chrX   \n",
       "      chrX:70729101-70736118:+              Bex1           chrX   \n",
       "      chrX:71004795-71005233:+              Pls3           chrX   \n",
       "      chrX:71034292-71035139:+              Ssr4           chrX   \n",
       "      chrX:71151202-71151767:-          Arhgap29           chrX   \n",
       "      chrX:71262850-71263468:-             Irak1           chrX   \n",
       "      chrX:71282671-71330848:-               Me2           chrX   \n",
       "      chrX:71383926-71388785:+             Opcml           chrX   \n",
       "      chrX:71533852-71534443:+               Taz           chrX   \n",
       "      chrX:7156972-7162937:+               Foxp1           chrX   \n",
       "      chrX:71613248-71613755:-             Ube4b           chrX   \n",
       "      chrX:73044829-73045841:-              Pls3           chrX   \n",
       "      chrX:7344810-7346555:+                Tcf3           chrX   \n",
       "      chrX:7380526-7381486:+             Gripap1           chrX   \n",
       "      chrX:7718334-7720376:-               Rbm39           chrX   \n",
       "      chrX:7721053-7722101:-               Rbm39           chrX   \n",
       "      chrX:7780941-7782711:-               Porcn           chrX   \n",
       "      chrX:82991640-82999670:-              Gusb           chrX   \n",
       "      chrX:90869254-90871826:-            Pofut2           chrX   \n",
       "      chrX:91347719-91368406:-             Zfp91           chrX   \n",
       "      chrX:91785886-91787231:-               Mag           chrX   \n",
       "      chrX:9646690-9694569:-                Rpa1           chrX   \n",
       "      chrX:9743402-9755396:-                Rpa1           chrX   \n",
       "      chrX:97963767-97966953:+              Dlg3           chrX   \n",
       "      chrX:98005246-98008064:+              Dlg3           chrX   \n",
       "      chrX:98279613-98280610:-            Slc7a1           chrX   \n",
       "      chrX:98497770-98502414:+             Nlgn3           chrX   \n",
       "      chrX:98601358-98603815:-             Zmym2           chrX   \n",
       "      chrX:98611277-98612247:-             Zmym2           chrX   \n",
       "      chrX:99750956-99754903:-             Phka1           chrX   \n",
       "\n",
       "                                   start_internal  stop_internal  \\\n",
       "rbp   location                                                     \n",
       "taf15 chr10:100041065-100046548:-       100044896      100044956   \n",
       "      chr10:10409700-10439020:-          10417845       10417930   \n",
       "      chr10:106199203-106237793:+       106204529      106204604   \n",
       "      chr10:106364918-106368179:+       106366109      106366201   \n",
       "      chr10:106708998-106817135:+       106760966      106761085   \n",
       "      chr10:106817207-106819559:+       106817282      106817359   \n",
       "      chr10:107689980-107696811:+       107690380      107690551   \n",
       "      chr10:107708825-107711193:+       107710450      107710481   \n",
       "      chr10:108339371-108447631:-       108430893      108430973   \n",
       "      chr10:110641896-110670608:+       110661727      110661764   \n",
       "      chr10:111893591-111895546:+       111893984      111894088   \n",
       "      chr10:114810002-114821260:-       114813407      114813603   \n",
       "      chr10:115944244-115986088:-       115954372      115954439   \n",
       "      chr10:115954439-115986088:-       115964882      115965005   \n",
       "      chr10:115964985-116018529:-       115986088      115986161   \n",
       "      chr10:115965005-115986088:-       115974276      115974504   \n",
       "      chr10:116356707-116374452:-       116365244      116365340   \n",
       "      chr10:116387748-116392583:+       116391568      116391781   \n",
       "      chr10:116671537-116674673:-       116672639      116672718   \n",
       "      chr10:116798416-116799070:-       116798829      116798940   \n",
       "      chr10:117589330-117594251:+       117593889      117593919   \n",
       "      chr10:119415663-119423387:+       119422529      119422685   \n",
       "      chr10:119487172-119491888:+       119487287      119487332   \n",
       "      chr10:119807080-119819120:+       119808312      119808360   \n",
       "      chr10:121533150-121533956:-       121533403      121533517   \n",
       "      chr10:122702597-123030549:+       122867767      122867979   \n",
       "      chr10:124731900-124739012:-       124735677      124735758   \n",
       "      chr10:126501276-126501641:+       126501275      126501509   \n",
       "      chr10:126526414-126527482:+       126527197      126527257   \n",
       "      chr10:126533798-126534987:-       126533970      126534135   \n",
       "...                                           ...            ...   \n",
       "tls   chrX:70590251-70597147:+           70596607       70596699   \n",
       "      chrX:70729101-70736118:+           70735206       70735270   \n",
       "      chrX:71004795-71005233:+           71004794       71004922   \n",
       "      chrX:71034292-71035139:+           71034393       71034492   \n",
       "      chrX:71151202-71151767:-           71151495       71151558   \n",
       "      chrX:71262850-71263468:-           71263239       71263356   \n",
       "      chrX:71282671-71330848:-           71325247       71325371   \n",
       "      chrX:71383926-71388785:+           71385141       71385307   \n",
       "      chrX:71533852-71534443:+           71534000       71534053   \n",
       "      chrX:7156972-7162937:+              7162696        7162763   \n",
       "      chrX:71613248-71613755:-           71613491       71613597   \n",
       "      chrX:73044829-73045841:-           73045055       73045082   \n",
       "      chrX:7344810-7346555:+              7345018        7345123   \n",
       "      chrX:7380526-7381486:+              7380719        7380812   \n",
       "      chrX:7718334-7720376:-              7719952        7720081   \n",
       "      chrX:7721053-7722101:-              7721698        7721805   \n",
       "      chrX:7780941-7782711:-              7781321        7781443   \n",
       "      chrX:82991640-82999670:-           82994950       82995034   \n",
       "      chrX:90869254-90871826:-           90869724       90869836   \n",
       "      chrX:91347719-91368406:-           91359534       91359602   \n",
       "      chrX:91785886-91787231:-           91786589       91786663   \n",
       "      chrX:9646690-9694569:-              9650894        9650954   \n",
       "      chrX:9743402-9755396:-              9753128        9754007   \n",
       "      chrX:97963767-97966953:+           97966645       97966699   \n",
       "      chrX:98005246-98008064:+           98006840       98006882   \n",
       "      chrX:98279613-98280610:-           98280159       98280317   \n",
       "      chrX:98497770-98502414:+           98498410       98498470   \n",
       "      chrX:98601358-98603815:-           98602543       98602798   \n",
       "      chrX:98611277-98612247:-           98611505       98611800   \n",
       "      chrX:99750956-99754903:-           99752434       99752611   \n",
       "\n",
       "                                             event_id strand_internal  \n",
       "rbp   location                                                         \n",
       "taf15 chr10:100041065-100046548:-  4930430F08Rik_(60)               -  \n",
       "      chr10:10409700-10439020:-             Grm1_(85)               -  \n",
       "      chr10:106199203-106237793:+         Ppfia2_(75)               +  \n",
       "      chr10:106364918-106368179:+         Ppfia2_(92)               +  \n",
       "      chr10:106708998-106817135:+         Lin7a_(119)               +  \n",
       "      chr10:106817207-106819559:+          Lin7a_(77)               +  \n",
       "      chr10:107689980-107696811:+      Ppp1r12a_(171)               +  \n",
       "      chr10:107708825-107711193:+       Ppp1r12a_(31)               +  \n",
       "      chr10:108339371-108447631:-           Syt1_(80)               -  \n",
       "      chr10:110641896-110670608:+         Osbpl8_(37)               +  \n",
       "      chr10:111893591-111895546:+         Kcnc2_(104)               +  \n",
       "      chr10:114810002-114821260:-         Thap2_(196)               -  \n",
       "      chr10:115944244-115986088:-          Cnot2_(67)               -  \n",
       "      chr10:115954439-115986088:-         Cnot2_(123)               -  \n",
       "      chr10:115964985-116018529:-          Cnot2_(73)               -  \n",
       "      chr10:115965005-115986088:-         Cnot2_(228)               -  \n",
       "      chr10:116356707-116374452:-         Rab3ip_(96)               -  \n",
       "      chr10:116387748-116392583:+    uc007hcn-1_(213)               +  \n",
       "      chr10:116671537-116674673:-  9530003J23Rik_(79)               -  \n",
       "      chr10:116798416-116799070:-         Cpsf6_(111)               -  \n",
       "      chr10:117589330-117594251:+           Mdm1_(30)               +  \n",
       "      chr10:119415663-119423387:+         Grip1_(156)               +  \n",
       "      chr10:119487172-119491888:+          Grip1_(45)               +  \n",
       "      chr10:119807080-119819120:+     uc007hfc-1_(48)               +  \n",
       "      chr10:121533150-121533956:-         Tmem5_(114)               -  \n",
       "      chr10:122702597-123030549:+       Fam19a2_(212)               +  \n",
       "      chr10:124731900-124739012:-        Slc16a7_(81)               -  \n",
       "      chr10:126501276-126501641:+          Cdk4_(234)               +  \n",
       "      chr10:126526414-126527482:+          Agap2_(60)               +  \n",
       "      chr10:126533798-126534987:-           Os9_(165)               -  \n",
       "...                                               ...             ...  \n",
       "tls   chrX:70590251-70597147:+            Zfp275_(92)               +  \n",
       "      chrX:70729101-70736118:+               Bgn_(64)               +  \n",
       "      chrX:71004795-71005233:+           Plxnb3_(128)               +  \n",
       "      chrX:71034292-71035139:+              Ssr4_(99)               +  \n",
       "      chrX:71151202-71151767:-           Arhgap4_(63)               -  \n",
       "      chrX:71262850-71263468:-            Irak1_(117)               -  \n",
       "      chrX:71282671-71330848:-            Mecp2_(124)               -  \n",
       "      chrX:71383926-71388785:+           Opn1mw_(166)               +  \n",
       "      chrX:71533852-71534443:+               Taz_(53)               +  \n",
       "      chrX:7156972-7162937:+               Foxp3_(67)               +  \n",
       "      chrX:71613248-71613755:-             Ubl4_(106)               -  \n",
       "      chrX:73044829-73045841:-              Pls3_(27)               -  \n",
       "      chrX:7344810-7346555:+              Tcfe3_(105)               +  \n",
       "      chrX:7380526-7381486:+             Gripap1_(93)               +  \n",
       "      chrX:7718334-7720376:-               Rbm3_(129)               -  \n",
       "      chrX:7721053-7722101:-               Rbm3_(107)               -  \n",
       "      chrX:7780941-7782711:-              Porcn_(122)               -  \n",
       "      chrX:82991640-82999670:-               Gyk_(84)               -  \n",
       "      chrX:90869254-90871826:-            Pola1_(112)               -  \n",
       "      chrX:91347719-91368406:-               Zfx_(68)               -  \n",
       "      chrX:91785886-91787231:-            Maged1_(74)               -  \n",
       "      chrX:9646690-9694569:-                Rpgr_(60)               -  \n",
       "      chrX:9743402-9755396:-               Rpgr_(879)               -  \n",
       "      chrX:97963767-97966953:+              Dlg3_(54)               +  \n",
       "      chrX:98005246-98008064:+              Dlg3_(42)               +  \n",
       "      chrX:98279613-98280610:-           Slc7a3_(158)               -  \n",
       "      chrX:98497770-98502414:+             Nlgn3_(60)               +  \n",
       "      chrX:98601358-98603815:-            Zmym3_(255)               -  \n",
       "      chrX:98611277-98612247:-            Zmym3_(295)               -  \n",
       "      chrX:99750956-99754903:-            Phka1_(177)               -  \n",
       "\n",
       "[11979 rows x 11 columns]"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sep_scores.sort_index().loc(axis=0)[['taf15', 'tdp43', 'tls']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "sep_scores = alt_cassette_events_full.xs(\"striatum\", level=\"cell_type\").sort_index().loc(axis=0)[['taf15', 'tdp43', 'tls']].sep_score.unstack()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>sep_score</th>\n",
       "      <th>q_value</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>old_splice_annotation</th>\n",
       "      <th>splice_type</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>chrom_internal</th>\n",
       "      <th>start_internal</th>\n",
       "      <th>stop_internal</th>\n",
       "      <th>event_id</th>\n",
       "      <th>strand_internal</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rbp</th>\n",
       "      <th>location</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"30\" valign=\"top\">taf15</th>\n",
       "      <th>chr10:100041065-100046548:-</th>\n",
       "      <td>0.082</td>\n",
       "      <td>0.345</td>\n",
       "      <td>error</td>\n",
       "      <td>4930430F08Rik_(60)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>error</td>\n",
       "      <td>chr10</td>\n",
       "      <td>100044896</td>\n",
       "      <td>100044956</td>\n",
       "      <td>4930430F08Rik_(60)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:10409700-10439020:-</th>\n",
       "      <td>0.376</td>\n",
       "      <td>0.379</td>\n",
       "      <td>ENSMUSG00000031153.9</td>\n",
       "      <td>Grm1_(85)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Gripap1</td>\n",
       "      <td>chr10</td>\n",
       "      <td>10417845</td>\n",
       "      <td>10417930</td>\n",
       "      <td>Grm1_(85)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:106199203-106237793:+</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000037519.8</td>\n",
       "      <td>Ppfia2_(75)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Ppfia1</td>\n",
       "      <td>chr10</td>\n",
       "      <td>106204529</td>\n",
       "      <td>106204604</td>\n",
       "      <td>Ppfia2_(75)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:106364918-106368179:+</th>\n",
       "      <td>-0.911</td>\n",
       "      <td>0.000</td>\n",
       "      <td>ENSMUSG00000053825.7</td>\n",
       "      <td>Ppfia2_(92)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Ppfia2</td>\n",
       "      <td>chr10</td>\n",
       "      <td>106366109</td>\n",
       "      <td>106366201</td>\n",
       "      <td>Ppfia2_(92)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:106708998-106817135:+</th>\n",
       "      <td>0.174</td>\n",
       "      <td>0.011</td>\n",
       "      <td>ENSMUSG00000019920.10</td>\n",
       "      <td>Lin7a_(119)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Lims1</td>\n",
       "      <td>chr10</td>\n",
       "      <td>106760966</td>\n",
       "      <td>106761085</td>\n",
       "      <td>Lin7a_(119)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:106817207-106819559:+</th>\n",
       "      <td>0.107</td>\n",
       "      <td>0.415</td>\n",
       "      <td>ENSMUSG00000019920.10</td>\n",
       "      <td>Lin7a_(77)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Lims1</td>\n",
       "      <td>chr10</td>\n",
       "      <td>106817282</td>\n",
       "      <td>106817359</td>\n",
       "      <td>Lin7a_(77)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:107689980-107696811:+</th>\n",
       "      <td>0.136</td>\n",
       "      <td>0.541</td>\n",
       "      <td>ENSMUSG00000019907.8</td>\n",
       "      <td>Ppp1r12a_(171)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Ppp1r12a</td>\n",
       "      <td>chr10</td>\n",
       "      <td>107690380</td>\n",
       "      <td>107690551</td>\n",
       "      <td>Ppp1r12a_(171)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:107708825-107711193:+</th>\n",
       "      <td>0.119</td>\n",
       "      <td>0.291</td>\n",
       "      <td>ENSMUSG00000019907.8</td>\n",
       "      <td>Ppp1r12a_(31)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Ppp1r12a</td>\n",
       "      <td>chr10</td>\n",
       "      <td>107710450</td>\n",
       "      <td>107710481</td>\n",
       "      <td>Ppp1r12a_(31)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:108339371-108447631:-</th>\n",
       "      <td>0.002</td>\n",
       "      <td>0.461</td>\n",
       "      <td>ENSMUSG00000058420.5</td>\n",
       "      <td>Syt1_(80)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Syt17</td>\n",
       "      <td>chr10</td>\n",
       "      <td>108430893</td>\n",
       "      <td>108430973</td>\n",
       "      <td>Syt1_(80)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:110641896-110670608:+</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000042359.9</td>\n",
       "      <td>Osbpl8_(37)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Osbpl6</td>\n",
       "      <td>chr10</td>\n",
       "      <td>110661727</td>\n",
       "      <td>110661764</td>\n",
       "      <td>Osbpl8_(37)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:111893591-111895546:+</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000018470.2</td>\n",
       "      <td>Kcnc2_(104)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Kcnab3</td>\n",
       "      <td>chr10</td>\n",
       "      <td>111893984</td>\n",
       "      <td>111894088</td>\n",
       "      <td>Kcnc2_(104)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:114810002-114821260:-</th>\n",
       "      <td>0.120</td>\n",
       "      <td>0.563</td>\n",
       "      <td>ENSMUSG00000048852.7</td>\n",
       "      <td>Thap2_(196)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Gm12185</td>\n",
       "      <td>chr10</td>\n",
       "      <td>114813407</td>\n",
       "      <td>114813603</td>\n",
       "      <td>Thap2_(196)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:115944244-115986088:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000056167.5</td>\n",
       "      <td>Cnot2_(67)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Cnot10</td>\n",
       "      <td>chr10</td>\n",
       "      <td>115954372</td>\n",
       "      <td>115954439</td>\n",
       "      <td>Cnot2_(67)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:115954439-115986088:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000056167.5</td>\n",
       "      <td>Cnot2_(123)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Cnot10</td>\n",
       "      <td>chr10</td>\n",
       "      <td>115964882</td>\n",
       "      <td>115965005</td>\n",
       "      <td>Cnot2_(123)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:115964985-116018529:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000020166.7</td>\n",
       "      <td>Cnot2_(73)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Cnot2</td>\n",
       "      <td>chr10</td>\n",
       "      <td>115986088</td>\n",
       "      <td>115986161</td>\n",
       "      <td>Cnot2_(73)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:115965005-115986088:-</th>\n",
       "      <td>-0.060</td>\n",
       "      <td>0.561</td>\n",
       "      <td>ENSMUSG00000056167.5</td>\n",
       "      <td>Cnot2_(228)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Cnot10</td>\n",
       "      <td>chr10</td>\n",
       "      <td>115974276</td>\n",
       "      <td>115974504</td>\n",
       "      <td>Cnot2_(228)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:116356707-116374452:-</th>\n",
       "      <td>0.254</td>\n",
       "      <td>0.031</td>\n",
       "      <td>ENSMUSG00000064181.4</td>\n",
       "      <td>Rab3ip_(96)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Rab3ip</td>\n",
       "      <td>chr10</td>\n",
       "      <td>116365244</td>\n",
       "      <td>116365340</td>\n",
       "      <td>Rab3ip_(96)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:116387748-116392583:+</th>\n",
       "      <td>0.047</td>\n",
       "      <td>0.417</td>\n",
       "      <td>error</td>\n",
       "      <td>uc007hcn-1_(213)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>error</td>\n",
       "      <td>chr10</td>\n",
       "      <td>116391568</td>\n",
       "      <td>116391781</td>\n",
       "      <td>uc007hcn-1_(213)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:116671537-116674673:-</th>\n",
       "      <td>0.327</td>\n",
       "      <td>0.102</td>\n",
       "      <td>ENSMUSG00000020177.6</td>\n",
       "      <td>9530003J23Rik_(79)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>9530003J23Rik</td>\n",
       "      <td>chr10</td>\n",
       "      <td>116672639</td>\n",
       "      <td>116672718</td>\n",
       "      <td>9530003J23Rik_(79)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:116798416-116799070:-</th>\n",
       "      <td>0.018</td>\n",
       "      <td>0.554</td>\n",
       "      <td>ENSMUSG00000018727.12</td>\n",
       "      <td>Cpsf6_(111)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Cpsf4l</td>\n",
       "      <td>chr10</td>\n",
       "      <td>116798829</td>\n",
       "      <td>116798940</td>\n",
       "      <td>Cpsf6_(111)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:117589330-117594251:+</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000034912.8</td>\n",
       "      <td>Mdm1_(30)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Mdga2</td>\n",
       "      <td>chr10</td>\n",
       "      <td>117593889</td>\n",
       "      <td>117593919</td>\n",
       "      <td>Mdm1_(30)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:119415663-119423387:+</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000026959.7</td>\n",
       "      <td>Grip1_(156)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Grin1</td>\n",
       "      <td>chr10</td>\n",
       "      <td>119422529</td>\n",
       "      <td>119422685</td>\n",
       "      <td>Grip1_(156)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:119487172-119491888:+</th>\n",
       "      <td>-0.535</td>\n",
       "      <td>0.041</td>\n",
       "      <td>ENSMUSG00000034813.10</td>\n",
       "      <td>Grip1_(45)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Grip1</td>\n",
       "      <td>chr10</td>\n",
       "      <td>119487287</td>\n",
       "      <td>119487332</td>\n",
       "      <td>Grip1_(45)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:119807080-119819120:+</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>error</td>\n",
       "      <td>uc007hfc-1_(48)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>error</td>\n",
       "      <td>chr10</td>\n",
       "      <td>119808312</td>\n",
       "      <td>119808360</td>\n",
       "      <td>uc007hfc-1_(48)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:121533150-121533956:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000034620.10</td>\n",
       "      <td>Tmem5_(114)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Tmem5</td>\n",
       "      <td>chr10</td>\n",
       "      <td>121533403</td>\n",
       "      <td>121533517</td>\n",
       "      <td>Tmem5_(114)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:122702597-123030549:+</th>\n",
       "      <td>0.084</td>\n",
       "      <td>0.213</td>\n",
       "      <td>ENSMUSG00000044071.6</td>\n",
       "      <td>Fam19a2_(212)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Fam19a2</td>\n",
       "      <td>chr10</td>\n",
       "      <td>122867767</td>\n",
       "      <td>122867979</td>\n",
       "      <td>Fam19a2_(212)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:124731900-124739012:-</th>\n",
       "      <td>-0.164</td>\n",
       "      <td>0.108</td>\n",
       "      <td>ENSMUSG00000041920.8</td>\n",
       "      <td>Slc16a7_(81)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Slc16a6</td>\n",
       "      <td>chr10</td>\n",
       "      <td>124735677</td>\n",
       "      <td>124735758</td>\n",
       "      <td>Slc16a7_(81)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:126501276-126501641:+</th>\n",
       "      <td>-0.127</td>\n",
       "      <td>0.221</td>\n",
       "      <td>ENSMUSG00000012819.8</td>\n",
       "      <td>Cdk4_(234)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Cdh23</td>\n",
       "      <td>chr10</td>\n",
       "      <td>126501275</td>\n",
       "      <td>126501509</td>\n",
       "      <td>Cdk4_(234)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:126526414-126527482:+</th>\n",
       "      <td>-0.401</td>\n",
       "      <td>0.000</td>\n",
       "      <td>ENSMUSG00000025422.7</td>\n",
       "      <td>Agap2_(60)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Agap2</td>\n",
       "      <td>chr10</td>\n",
       "      <td>126527197</td>\n",
       "      <td>126527257</td>\n",
       "      <td>Agap2_(60)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr10:126533798-126534987:-</th>\n",
       "      <td>0.095</td>\n",
       "      <td>0.539</td>\n",
       "      <td>ENSMUSG00000026761.6</td>\n",
       "      <td>Os9_(165)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Orc4</td>\n",
       "      <td>chr10</td>\n",
       "      <td>126533970</td>\n",
       "      <td>126534135</td>\n",
       "      <td>Os9_(165)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"30\" valign=\"top\">tls</th>\n",
       "      <th>chrX:70590251-70597147:+</th>\n",
       "      <td>-0.190</td>\n",
       "      <td>0.085</td>\n",
       "      <td>ENSMUSG00000031365.6</td>\n",
       "      <td>Zfp275_(92)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Zfp275</td>\n",
       "      <td>chrX</td>\n",
       "      <td>70596607</td>\n",
       "      <td>70596699</td>\n",
       "      <td>Zfp275_(92)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:70729101-70736118:+</th>\n",
       "      <td>-0.266</td>\n",
       "      <td>0.043</td>\n",
       "      <td>ENSMUSG00000050071.8</td>\n",
       "      <td>Bgn_(64)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Bex1</td>\n",
       "      <td>chrX</td>\n",
       "      <td>70735206</td>\n",
       "      <td>70735270</td>\n",
       "      <td>Bgn_(64)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:71004795-71005233:+</th>\n",
       "      <td>0.025</td>\n",
       "      <td>0.312</td>\n",
       "      <td>ENSMUSG00000016382.9</td>\n",
       "      <td>Plxnb3_(128)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Pls3</td>\n",
       "      <td>chrX</td>\n",
       "      <td>71004794</td>\n",
       "      <td>71004922</td>\n",
       "      <td>Plxnb3_(128)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:71034292-71035139:+</th>\n",
       "      <td>-0.263</td>\n",
       "      <td>0.137</td>\n",
       "      <td>ENSMUSG00000002014.6</td>\n",
       "      <td>Ssr4_(99)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Ssr4</td>\n",
       "      <td>chrX</td>\n",
       "      <td>71034393</td>\n",
       "      <td>71034492</td>\n",
       "      <td>Ssr4_(99)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:71151202-71151767:-</th>\n",
       "      <td>0.022</td>\n",
       "      <td>0.555</td>\n",
       "      <td>ENSMUSG00000039831.9</td>\n",
       "      <td>Arhgap4_(63)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Arhgap29</td>\n",
       "      <td>chrX</td>\n",
       "      <td>71151495</td>\n",
       "      <td>71151558</td>\n",
       "      <td>Arhgap4_(63)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:71262850-71263468:-</th>\n",
       "      <td>-0.071</td>\n",
       "      <td>0.196</td>\n",
       "      <td>ENSMUSG00000031392.11</td>\n",
       "      <td>Irak1_(117)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Irak1</td>\n",
       "      <td>chrX</td>\n",
       "      <td>71263239</td>\n",
       "      <td>71263356</td>\n",
       "      <td>Irak1_(117)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:71282671-71330848:-</th>\n",
       "      <td>-0.259</td>\n",
       "      <td>0.415</td>\n",
       "      <td>ENSMUSG00000024556.3</td>\n",
       "      <td>Mecp2_(124)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Me2</td>\n",
       "      <td>chrX</td>\n",
       "      <td>71325247</td>\n",
       "      <td>71325371</td>\n",
       "      <td>Mecp2_(124)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:71383926-71388785:+</th>\n",
       "      <td>0.201</td>\n",
       "      <td>0.108</td>\n",
       "      <td>ENSMUSG00000062257.6</td>\n",
       "      <td>Opn1mw_(166)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Opcml</td>\n",
       "      <td>chrX</td>\n",
       "      <td>71385141</td>\n",
       "      <td>71385307</td>\n",
       "      <td>Opn1mw_(166)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:71533852-71534443:+</th>\n",
       "      <td>-1.009</td>\n",
       "      <td>0.000</td>\n",
       "      <td>ENSMUSG00000009995.10</td>\n",
       "      <td>Taz_(53)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Taz</td>\n",
       "      <td>chrX</td>\n",
       "      <td>71534000</td>\n",
       "      <td>71534053</td>\n",
       "      <td>Taz_(53)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:7156972-7162937:+</th>\n",
       "      <td>0.021</td>\n",
       "      <td>0.272</td>\n",
       "      <td>ENSMUSG00000030067.9</td>\n",
       "      <td>Foxp3_(67)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Foxp1</td>\n",
       "      <td>chrX</td>\n",
       "      <td>7162696</td>\n",
       "      <td>7162763</td>\n",
       "      <td>Foxp3_(67)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:71613248-71613755:-</th>\n",
       "      <td>0.435</td>\n",
       "      <td>0.046</td>\n",
       "      <td>ENSMUSG00000028960.11</td>\n",
       "      <td>Ubl4_(106)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Ube4b</td>\n",
       "      <td>chrX</td>\n",
       "      <td>71613491</td>\n",
       "      <td>71613597</td>\n",
       "      <td>Ubl4_(106)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:73044829-73045841:-</th>\n",
       "      <td>0.213</td>\n",
       "      <td>0.064</td>\n",
       "      <td>ENSMUSG00000016382.9</td>\n",
       "      <td>Pls3_(27)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Pls3</td>\n",
       "      <td>chrX</td>\n",
       "      <td>73045055</td>\n",
       "      <td>73045082</td>\n",
       "      <td>Pls3_(27)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:7344810-7346555:+</th>\n",
       "      <td>0.098</td>\n",
       "      <td>0.036</td>\n",
       "      <td>ENSMUSG00000020167.8</td>\n",
       "      <td>Tcfe3_(105)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Tcf3</td>\n",
       "      <td>chrX</td>\n",
       "      <td>7345018</td>\n",
       "      <td>7345123</td>\n",
       "      <td>Tcfe3_(105)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:7380526-7381486:+</th>\n",
       "      <td>0.002</td>\n",
       "      <td>0.811</td>\n",
       "      <td>ENSMUSG00000031153.9</td>\n",
       "      <td>Gripap1_(93)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Gripap1</td>\n",
       "      <td>chrX</td>\n",
       "      <td>7380719</td>\n",
       "      <td>7380812</td>\n",
       "      <td>Gripap1_(93)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:7718334-7720376:-</th>\n",
       "      <td>-0.195</td>\n",
       "      <td>0.484</td>\n",
       "      <td>ENSMUSG00000027620.9</td>\n",
       "      <td>Rbm3_(129)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Rbm39</td>\n",
       "      <td>chrX</td>\n",
       "      <td>7719952</td>\n",
       "      <td>7720081</td>\n",
       "      <td>Rbm3_(129)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:7721053-7722101:-</th>\n",
       "      <td>0.297</td>\n",
       "      <td>0.042</td>\n",
       "      <td>ENSMUSG00000027620.9</td>\n",
       "      <td>Rbm3_(107)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Rbm39</td>\n",
       "      <td>chrX</td>\n",
       "      <td>7721698</td>\n",
       "      <td>7721805</td>\n",
       "      <td>Rbm3_(107)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:7780941-7782711:-</th>\n",
       "      <td>0.512</td>\n",
       "      <td>0.000</td>\n",
       "      <td>ENSMUSG00000031169.7</td>\n",
       "      <td>Porcn_(122)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Porcn</td>\n",
       "      <td>chrX</td>\n",
       "      <td>7781321</td>\n",
       "      <td>7781443</td>\n",
       "      <td>Porcn_(122)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:82991640-82999670:-</th>\n",
       "      <td>-0.168</td>\n",
       "      <td>0.000</td>\n",
       "      <td>ENSMUSG00000025534.10</td>\n",
       "      <td>Gyk_(84)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Gusb</td>\n",
       "      <td>chrX</td>\n",
       "      <td>82994950</td>\n",
       "      <td>82995034</td>\n",
       "      <td>Gyk_(84)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:90869254-90871826:-</th>\n",
       "      <td>0.138</td>\n",
       "      <td>0.160</td>\n",
       "      <td>ENSMUSG00000020260.8</td>\n",
       "      <td>Pola1_(112)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Pofut2</td>\n",
       "      <td>chrX</td>\n",
       "      <td>90869724</td>\n",
       "      <td>90869836</td>\n",
       "      <td>Pola1_(112)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:91347719-91368406:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000024695.9</td>\n",
       "      <td>Zfx_(68)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Zfp91</td>\n",
       "      <td>chrX</td>\n",
       "      <td>91359534</td>\n",
       "      <td>91359602</td>\n",
       "      <td>Zfx_(68)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:91785886-91787231:-</th>\n",
       "      <td>0.169</td>\n",
       "      <td>0.046</td>\n",
       "      <td>ENSMUSG00000036634.8</td>\n",
       "      <td>Maged1_(74)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Mag</td>\n",
       "      <td>chrX</td>\n",
       "      <td>91786589</td>\n",
       "      <td>91786663</td>\n",
       "      <td>Maged1_(74)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:9646690-9694569:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000000751.7</td>\n",
       "      <td>Rpgr_(60)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Rpa1</td>\n",
       "      <td>chrX</td>\n",
       "      <td>9650894</td>\n",
       "      <td>9650954</td>\n",
       "      <td>Rpgr_(60)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:9743402-9755396:-</th>\n",
       "      <td>-0.086</td>\n",
       "      <td>0.201</td>\n",
       "      <td>ENSMUSG00000000751.7</td>\n",
       "      <td>Rpgr_(879)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Rpa1</td>\n",
       "      <td>chrX</td>\n",
       "      <td>9753128</td>\n",
       "      <td>9754007</td>\n",
       "      <td>Rpgr_(879)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:97963767-97966953:+</th>\n",
       "      <td>-0.320</td>\n",
       "      <td>0.223</td>\n",
       "      <td>ENSMUSG00000000881.6</td>\n",
       "      <td>Dlg3_(54)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Dlg3</td>\n",
       "      <td>chrX</td>\n",
       "      <td>97966645</td>\n",
       "      <td>97966699</td>\n",
       "      <td>Dlg3_(54)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:98005246-98008064:+</th>\n",
       "      <td>0.210</td>\n",
       "      <td>0.000</td>\n",
       "      <td>ENSMUSG00000000881.6</td>\n",
       "      <td>Dlg3_(42)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Dlg3</td>\n",
       "      <td>chrX</td>\n",
       "      <td>98006840</td>\n",
       "      <td>98006882</td>\n",
       "      <td>Dlg3_(42)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:98279613-98280610:-</th>\n",
       "      <td>-0.253</td>\n",
       "      <td>0.187</td>\n",
       "      <td>ENSMUSG00000041313.8</td>\n",
       "      <td>Slc7a3_(158)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Slc7a1</td>\n",
       "      <td>chrX</td>\n",
       "      <td>98280159</td>\n",
       "      <td>98280317</td>\n",
       "      <td>Slc7a3_(158)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:98497770-98502414:+</th>\n",
       "      <td>-0.514</td>\n",
       "      <td>0.000</td>\n",
       "      <td>ENSMUSG00000031302.9</td>\n",
       "      <td>Nlgn3_(60)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Nlgn3</td>\n",
       "      <td>chrX</td>\n",
       "      <td>98498410</td>\n",
       "      <td>98498470</td>\n",
       "      <td>Nlgn3_(60)</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:98601358-98603815:-</th>\n",
       "      <td>-0.423</td>\n",
       "      <td>0.000</td>\n",
       "      <td>ENSMUSG00000021945.7</td>\n",
       "      <td>Zmym3_(255)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Zmym2</td>\n",
       "      <td>chrX</td>\n",
       "      <td>98602543</td>\n",
       "      <td>98602798</td>\n",
       "      <td>Zmym3_(255)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:98611277-98612247:-</th>\n",
       "      <td>-0.382</td>\n",
       "      <td>0.000</td>\n",
       "      <td>ENSMUSG00000021945.7</td>\n",
       "      <td>Zmym3_(295)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Zmym2</td>\n",
       "      <td>chrX</td>\n",
       "      <td>98611505</td>\n",
       "      <td>98611800</td>\n",
       "      <td>Zmym3_(295)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX:99750956-99754903:-</th>\n",
       "      <td>0.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>ENSMUSG00000034055.10</td>\n",
       "      <td>Phka1_(177)</td>\n",
       "      <td>alt_cassette</td>\n",
       "      <td>Phka1</td>\n",
       "      <td>chrX</td>\n",
       "      <td>99752434</td>\n",
       "      <td>99752611</td>\n",
       "      <td>Phka1_(177)</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11979 rows Ã— 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   sep_score  q_value                gene_id  \\\n",
       "rbp   location                                                                 \n",
       "taf15 chr10:100041065-100046548:-      0.082    0.345                  error   \n",
       "      chr10:10409700-10439020:-        0.376    0.379   ENSMUSG00000031153.9   \n",
       "      chr10:106199203-106237793:+      0.000    1.000   ENSMUSG00000037519.8   \n",
       "      chr10:106364918-106368179:+     -0.911    0.000   ENSMUSG00000053825.7   \n",
       "      chr10:106708998-106817135:+      0.174    0.011  ENSMUSG00000019920.10   \n",
       "      chr10:106817207-106819559:+      0.107    0.415  ENSMUSG00000019920.10   \n",
       "      chr10:107689980-107696811:+      0.136    0.541   ENSMUSG00000019907.8   \n",
       "      chr10:107708825-107711193:+      0.119    0.291   ENSMUSG00000019907.8   \n",
       "      chr10:108339371-108447631:-      0.002    0.461   ENSMUSG00000058420.5   \n",
       "      chr10:110641896-110670608:+      0.000    1.000   ENSMUSG00000042359.9   \n",
       "      chr10:111893591-111895546:+      0.000    1.000   ENSMUSG00000018470.2   \n",
       "      chr10:114810002-114821260:-      0.120    0.563   ENSMUSG00000048852.7   \n",
       "      chr10:115944244-115986088:-      0.000    1.000   ENSMUSG00000056167.5   \n",
       "      chr10:115954439-115986088:-      0.000    1.000   ENSMUSG00000056167.5   \n",
       "      chr10:115964985-116018529:-      0.000    1.000   ENSMUSG00000020166.7   \n",
       "      chr10:115965005-115986088:-     -0.060    0.561   ENSMUSG00000056167.5   \n",
       "      chr10:116356707-116374452:-      0.254    0.031   ENSMUSG00000064181.4   \n",
       "      chr10:116387748-116392583:+      0.047    0.417                  error   \n",
       "      chr10:116671537-116674673:-      0.327    0.102   ENSMUSG00000020177.6   \n",
       "      chr10:116798416-116799070:-      0.018    0.554  ENSMUSG00000018727.12   \n",
       "      chr10:117589330-117594251:+      0.000    1.000   ENSMUSG00000034912.8   \n",
       "      chr10:119415663-119423387:+      0.000    1.000   ENSMUSG00000026959.7   \n",
       "      chr10:119487172-119491888:+     -0.535    0.041  ENSMUSG00000034813.10   \n",
       "      chr10:119807080-119819120:+      0.000    1.000                  error   \n",
       "      chr10:121533150-121533956:-      0.000    1.000  ENSMUSG00000034620.10   \n",
       "      chr10:122702597-123030549:+      0.084    0.213   ENSMUSG00000044071.6   \n",
       "      chr10:124731900-124739012:-     -0.164    0.108   ENSMUSG00000041920.8   \n",
       "      chr10:126501276-126501641:+     -0.127    0.221   ENSMUSG00000012819.8   \n",
       "      chr10:126526414-126527482:+     -0.401    0.000   ENSMUSG00000025422.7   \n",
       "      chr10:126533798-126534987:-      0.095    0.539   ENSMUSG00000026761.6   \n",
       "...                                      ...      ...                    ...   \n",
       "tls   chrX:70590251-70597147:+        -0.190    0.085   ENSMUSG00000031365.6   \n",
       "      chrX:70729101-70736118:+        -0.266    0.043   ENSMUSG00000050071.8   \n",
       "      chrX:71004795-71005233:+         0.025    0.312   ENSMUSG00000016382.9   \n",
       "      chrX:71034292-71035139:+        -0.263    0.137   ENSMUSG00000002014.6   \n",
       "      chrX:71151202-71151767:-         0.022    0.555   ENSMUSG00000039831.9   \n",
       "      chrX:71262850-71263468:-        -0.071    0.196  ENSMUSG00000031392.11   \n",
       "      chrX:71282671-71330848:-        -0.259    0.415   ENSMUSG00000024556.3   \n",
       "      chrX:71383926-71388785:+         0.201    0.108   ENSMUSG00000062257.6   \n",
       "      chrX:71533852-71534443:+        -1.009    0.000  ENSMUSG00000009995.10   \n",
       "      chrX:7156972-7162937:+           0.021    0.272   ENSMUSG00000030067.9   \n",
       "      chrX:71613248-71613755:-         0.435    0.046  ENSMUSG00000028960.11   \n",
       "      chrX:73044829-73045841:-         0.213    0.064   ENSMUSG00000016382.9   \n",
       "      chrX:7344810-7346555:+           0.098    0.036   ENSMUSG00000020167.8   \n",
       "      chrX:7380526-7381486:+           0.002    0.811   ENSMUSG00000031153.9   \n",
       "      chrX:7718334-7720376:-          -0.195    0.484   ENSMUSG00000027620.9   \n",
       "      chrX:7721053-7722101:-           0.297    0.042   ENSMUSG00000027620.9   \n",
       "      chrX:7780941-7782711:-           0.512    0.000   ENSMUSG00000031169.7   \n",
       "      chrX:82991640-82999670:-        -0.168    0.000  ENSMUSG00000025534.10   \n",
       "      chrX:90869254-90871826:-         0.138    0.160   ENSMUSG00000020260.8   \n",
       "      chrX:91347719-91368406:-         0.000    1.000   ENSMUSG00000024695.9   \n",
       "      chrX:91785886-91787231:-         0.169    0.046   ENSMUSG00000036634.8   \n",
       "      chrX:9646690-9694569:-           0.000    1.000   ENSMUSG00000000751.7   \n",
       "      chrX:9743402-9755396:-          -0.086    0.201   ENSMUSG00000000751.7   \n",
       "      chrX:97963767-97966953:+        -0.320    0.223   ENSMUSG00000000881.6   \n",
       "      chrX:98005246-98008064:+         0.210    0.000   ENSMUSG00000000881.6   \n",
       "      chrX:98279613-98280610:-        -0.253    0.187   ENSMUSG00000041313.8   \n",
       "      chrX:98497770-98502414:+        -0.514    0.000   ENSMUSG00000031302.9   \n",
       "      chrX:98601358-98603815:-        -0.423    0.000   ENSMUSG00000021945.7   \n",
       "      chrX:98611277-98612247:-        -0.382    0.000   ENSMUSG00000021945.7   \n",
       "      chrX:99750956-99754903:-         0.000    1.000  ENSMUSG00000034055.10   \n",
       "\n",
       "                                  old_splice_annotation   splice_type  \\\n",
       "rbp   location                                                          \n",
       "taf15 chr10:100041065-100046548:-    4930430F08Rik_(60)  alt_cassette   \n",
       "      chr10:10409700-10439020:-               Grm1_(85)  alt_cassette   \n",
       "      chr10:106199203-106237793:+           Ppfia2_(75)  alt_cassette   \n",
       "      chr10:106364918-106368179:+           Ppfia2_(92)  alt_cassette   \n",
       "      chr10:106708998-106817135:+           Lin7a_(119)  alt_cassette   \n",
       "      chr10:106817207-106819559:+            Lin7a_(77)  alt_cassette   \n",
       "      chr10:107689980-107696811:+        Ppp1r12a_(171)  alt_cassette   \n",
       "      chr10:107708825-107711193:+         Ppp1r12a_(31)  alt_cassette   \n",
       "      chr10:108339371-108447631:-             Syt1_(80)  alt_cassette   \n",
       "      chr10:110641896-110670608:+           Osbpl8_(37)  alt_cassette   \n",
       "      chr10:111893591-111895546:+           Kcnc2_(104)  alt_cassette   \n",
       "      chr10:114810002-114821260:-           Thap2_(196)  alt_cassette   \n",
       "      chr10:115944244-115986088:-            Cnot2_(67)  alt_cassette   \n",
       "      chr10:115954439-115986088:-           Cnot2_(123)  alt_cassette   \n",
       "      chr10:115964985-116018529:-            Cnot2_(73)  alt_cassette   \n",
       "      chr10:115965005-115986088:-           Cnot2_(228)  alt_cassette   \n",
       "      chr10:116356707-116374452:-           Rab3ip_(96)  alt_cassette   \n",
       "      chr10:116387748-116392583:+      uc007hcn-1_(213)  alt_cassette   \n",
       "      chr10:116671537-116674673:-    9530003J23Rik_(79)  alt_cassette   \n",
       "      chr10:116798416-116799070:-           Cpsf6_(111)  alt_cassette   \n",
       "      chr10:117589330-117594251:+             Mdm1_(30)  alt_cassette   \n",
       "      chr10:119415663-119423387:+           Grip1_(156)  alt_cassette   \n",
       "      chr10:119487172-119491888:+            Grip1_(45)  alt_cassette   \n",
       "      chr10:119807080-119819120:+       uc007hfc-1_(48)  alt_cassette   \n",
       "      chr10:121533150-121533956:-           Tmem5_(114)  alt_cassette   \n",
       "      chr10:122702597-123030549:+         Fam19a2_(212)  alt_cassette   \n",
       "      chr10:124731900-124739012:-          Slc16a7_(81)  alt_cassette   \n",
       "      chr10:126501276-126501641:+            Cdk4_(234)  alt_cassette   \n",
       "      chr10:126526414-126527482:+            Agap2_(60)  alt_cassette   \n",
       "      chr10:126533798-126534987:-             Os9_(165)  alt_cassette   \n",
       "...                                                 ...           ...   \n",
       "tls   chrX:70590251-70597147:+              Zfp275_(92)  alt_cassette   \n",
       "      chrX:70729101-70736118:+                 Bgn_(64)  alt_cassette   \n",
       "      chrX:71004795-71005233:+             Plxnb3_(128)  alt_cassette   \n",
       "      chrX:71034292-71035139:+                Ssr4_(99)  alt_cassette   \n",
       "      chrX:71151202-71151767:-             Arhgap4_(63)  alt_cassette   \n",
       "      chrX:71262850-71263468:-              Irak1_(117)  alt_cassette   \n",
       "      chrX:71282671-71330848:-              Mecp2_(124)  alt_cassette   \n",
       "      chrX:71383926-71388785:+             Opn1mw_(166)  alt_cassette   \n",
       "      chrX:71533852-71534443:+                 Taz_(53)  alt_cassette   \n",
       "      chrX:7156972-7162937:+                 Foxp3_(67)  alt_cassette   \n",
       "      chrX:71613248-71613755:-               Ubl4_(106)  alt_cassette   \n",
       "      chrX:73044829-73045841:-                Pls3_(27)  alt_cassette   \n",
       "      chrX:7344810-7346555:+                Tcfe3_(105)  alt_cassette   \n",
       "      chrX:7380526-7381486:+               Gripap1_(93)  alt_cassette   \n",
       "      chrX:7718334-7720376:-                 Rbm3_(129)  alt_cassette   \n",
       "      chrX:7721053-7722101:-                 Rbm3_(107)  alt_cassette   \n",
       "      chrX:7780941-7782711:-                Porcn_(122)  alt_cassette   \n",
       "      chrX:82991640-82999670:-                 Gyk_(84)  alt_cassette   \n",
       "      chrX:90869254-90871826:-              Pola1_(112)  alt_cassette   \n",
       "      chrX:91347719-91368406:-                 Zfx_(68)  alt_cassette   \n",
       "      chrX:91785886-91787231:-              Maged1_(74)  alt_cassette   \n",
       "      chrX:9646690-9694569:-                  Rpgr_(60)  alt_cassette   \n",
       "      chrX:9743402-9755396:-                 Rpgr_(879)  alt_cassette   \n",
       "      chrX:97963767-97966953:+                Dlg3_(54)  alt_cassette   \n",
       "      chrX:98005246-98008064:+                Dlg3_(42)  alt_cassette   \n",
       "      chrX:98279613-98280610:-             Slc7a3_(158)  alt_cassette   \n",
       "      chrX:98497770-98502414:+               Nlgn3_(60)  alt_cassette   \n",
       "      chrX:98601358-98603815:-              Zmym3_(255)  alt_cassette   \n",
       "      chrX:98611277-98612247:-              Zmym3_(295)  alt_cassette   \n",
       "      chrX:99750956-99754903:-              Phka1_(177)  alt_cassette   \n",
       "\n",
       "                                       gene_name chrom_internal  \\\n",
       "rbp   location                                                    \n",
       "taf15 chr10:100041065-100046548:-          error          chr10   \n",
       "      chr10:10409700-10439020:-          Gripap1          chr10   \n",
       "      chr10:106199203-106237793:+         Ppfia1          chr10   \n",
       "      chr10:106364918-106368179:+         Ppfia2          chr10   \n",
       "      chr10:106708998-106817135:+          Lims1          chr10   \n",
       "      chr10:106817207-106819559:+          Lims1          chr10   \n",
       "      chr10:107689980-107696811:+       Ppp1r12a          chr10   \n",
       "      chr10:107708825-107711193:+       Ppp1r12a          chr10   \n",
       "      chr10:108339371-108447631:-          Syt17          chr10   \n",
       "      chr10:110641896-110670608:+         Osbpl6          chr10   \n",
       "      chr10:111893591-111895546:+         Kcnab3          chr10   \n",
       "      chr10:114810002-114821260:-        Gm12185          chr10   \n",
       "      chr10:115944244-115986088:-         Cnot10          chr10   \n",
       "      chr10:115954439-115986088:-         Cnot10          chr10   \n",
       "      chr10:115964985-116018529:-          Cnot2          chr10   \n",
       "      chr10:115965005-115986088:-         Cnot10          chr10   \n",
       "      chr10:116356707-116374452:-         Rab3ip          chr10   \n",
       "      chr10:116387748-116392583:+          error          chr10   \n",
       "      chr10:116671537-116674673:-  9530003J23Rik          chr10   \n",
       "      chr10:116798416-116799070:-         Cpsf4l          chr10   \n",
       "      chr10:117589330-117594251:+          Mdga2          chr10   \n",
       "      chr10:119415663-119423387:+          Grin1          chr10   \n",
       "      chr10:119487172-119491888:+          Grip1          chr10   \n",
       "      chr10:119807080-119819120:+          error          chr10   \n",
       "      chr10:121533150-121533956:-          Tmem5          chr10   \n",
       "      chr10:122702597-123030549:+        Fam19a2          chr10   \n",
       "      chr10:124731900-124739012:-        Slc16a6          chr10   \n",
       "      chr10:126501276-126501641:+          Cdh23          chr10   \n",
       "      chr10:126526414-126527482:+          Agap2          chr10   \n",
       "      chr10:126533798-126534987:-           Orc4          chr10   \n",
       "...                                          ...            ...   \n",
       "tls   chrX:70590251-70597147:+            Zfp275           chrX   \n",
       "      chrX:70729101-70736118:+              Bex1           chrX   \n",
       "      chrX:71004795-71005233:+              Pls3           chrX   \n",
       "      chrX:71034292-71035139:+              Ssr4           chrX   \n",
       "      chrX:71151202-71151767:-          Arhgap29           chrX   \n",
       "      chrX:71262850-71263468:-             Irak1           chrX   \n",
       "      chrX:71282671-71330848:-               Me2           chrX   \n",
       "      chrX:71383926-71388785:+             Opcml           chrX   \n",
       "      chrX:71533852-71534443:+               Taz           chrX   \n",
       "      chrX:7156972-7162937:+               Foxp1           chrX   \n",
       "      chrX:71613248-71613755:-             Ube4b           chrX   \n",
       "      chrX:73044829-73045841:-              Pls3           chrX   \n",
       "      chrX:7344810-7346555:+                Tcf3           chrX   \n",
       "      chrX:7380526-7381486:+             Gripap1           chrX   \n",
       "      chrX:7718334-7720376:-               Rbm39           chrX   \n",
       "      chrX:7721053-7722101:-               Rbm39           chrX   \n",
       "      chrX:7780941-7782711:-               Porcn           chrX   \n",
       "      chrX:82991640-82999670:-              Gusb           chrX   \n",
       "      chrX:90869254-90871826:-            Pofut2           chrX   \n",
       "      chrX:91347719-91368406:-             Zfp91           chrX   \n",
       "      chrX:91785886-91787231:-               Mag           chrX   \n",
       "      chrX:9646690-9694569:-                Rpa1           chrX   \n",
       "      chrX:9743402-9755396:-                Rpa1           chrX   \n",
       "      chrX:97963767-97966953:+              Dlg3           chrX   \n",
       "      chrX:98005246-98008064:+              Dlg3           chrX   \n",
       "      chrX:98279613-98280610:-            Slc7a1           chrX   \n",
       "      chrX:98497770-98502414:+             Nlgn3           chrX   \n",
       "      chrX:98601358-98603815:-             Zmym2           chrX   \n",
       "      chrX:98611277-98612247:-             Zmym2           chrX   \n",
       "      chrX:99750956-99754903:-             Phka1           chrX   \n",
       "\n",
       "                                   start_internal  stop_internal  \\\n",
       "rbp   location                                                     \n",
       "taf15 chr10:100041065-100046548:-       100044896      100044956   \n",
       "      chr10:10409700-10439020:-          10417845       10417930   \n",
       "      chr10:106199203-106237793:+       106204529      106204604   \n",
       "      chr10:106364918-106368179:+       106366109      106366201   \n",
       "      chr10:106708998-106817135:+       106760966      106761085   \n",
       "      chr10:106817207-106819559:+       106817282      106817359   \n",
       "      chr10:107689980-107696811:+       107690380      107690551   \n",
       "      chr10:107708825-107711193:+       107710450      107710481   \n",
       "      chr10:108339371-108447631:-       108430893      108430973   \n",
       "      chr10:110641896-110670608:+       110661727      110661764   \n",
       "      chr10:111893591-111895546:+       111893984      111894088   \n",
       "      chr10:114810002-114821260:-       114813407      114813603   \n",
       "      chr10:115944244-115986088:-       115954372      115954439   \n",
       "      chr10:115954439-115986088:-       115964882      115965005   \n",
       "      chr10:115964985-116018529:-       115986088      115986161   \n",
       "      chr10:115965005-115986088:-       115974276      115974504   \n",
       "      chr10:116356707-116374452:-       116365244      116365340   \n",
       "      chr10:116387748-116392583:+       116391568      116391781   \n",
       "      chr10:116671537-116674673:-       116672639      116672718   \n",
       "      chr10:116798416-116799070:-       116798829      116798940   \n",
       "      chr10:117589330-117594251:+       117593889      117593919   \n",
       "      chr10:119415663-119423387:+       119422529      119422685   \n",
       "      chr10:119487172-119491888:+       119487287      119487332   \n",
       "      chr10:119807080-119819120:+       119808312      119808360   \n",
       "      chr10:121533150-121533956:-       121533403      121533517   \n",
       "      chr10:122702597-123030549:+       122867767      122867979   \n",
       "      chr10:124731900-124739012:-       124735677      124735758   \n",
       "      chr10:126501276-126501641:+       126501275      126501509   \n",
       "      chr10:126526414-126527482:+       126527197      126527257   \n",
       "      chr10:126533798-126534987:-       126533970      126534135   \n",
       "...                                           ...            ...   \n",
       "tls   chrX:70590251-70597147:+           70596607       70596699   \n",
       "      chrX:70729101-70736118:+           70735206       70735270   \n",
       "      chrX:71004795-71005233:+           71004794       71004922   \n",
       "      chrX:71034292-71035139:+           71034393       71034492   \n",
       "      chrX:71151202-71151767:-           71151495       71151558   \n",
       "      chrX:71262850-71263468:-           71263239       71263356   \n",
       "      chrX:71282671-71330848:-           71325247       71325371   \n",
       "      chrX:71383926-71388785:+           71385141       71385307   \n",
       "      chrX:71533852-71534443:+           71534000       71534053   \n",
       "      chrX:7156972-7162937:+              7162696        7162763   \n",
       "      chrX:71613248-71613755:-           71613491       71613597   \n",
       "      chrX:73044829-73045841:-           73045055       73045082   \n",
       "      chrX:7344810-7346555:+              7345018        7345123   \n",
       "      chrX:7380526-7381486:+              7380719        7380812   \n",
       "      chrX:7718334-7720376:-              7719952        7720081   \n",
       "      chrX:7721053-7722101:-              7721698        7721805   \n",
       "      chrX:7780941-7782711:-              7781321        7781443   \n",
       "      chrX:82991640-82999670:-           82994950       82995034   \n",
       "      chrX:90869254-90871826:-           90869724       90869836   \n",
       "      chrX:91347719-91368406:-           91359534       91359602   \n",
       "      chrX:91785886-91787231:-           91786589       91786663   \n",
       "      chrX:9646690-9694569:-              9650894        9650954   \n",
       "      chrX:9743402-9755396:-              9753128        9754007   \n",
       "      chrX:97963767-97966953:+           97966645       97966699   \n",
       "      chrX:98005246-98008064:+           98006840       98006882   \n",
       "      chrX:98279613-98280610:-           98280159       98280317   \n",
       "      chrX:98497770-98502414:+           98498410       98498470   \n",
       "      chrX:98601358-98603815:-           98602543       98602798   \n",
       "      chrX:98611277-98612247:-           98611505       98611800   \n",
       "      chrX:99750956-99754903:-           99752434       99752611   \n",
       "\n",
       "                                             event_id strand_internal  \n",
       "rbp   location                                                         \n",
       "taf15 chr10:100041065-100046548:-  4930430F08Rik_(60)               -  \n",
       "      chr10:10409700-10439020:-             Grm1_(85)               -  \n",
       "      chr10:106199203-106237793:+         Ppfia2_(75)               +  \n",
       "      chr10:106364918-106368179:+         Ppfia2_(92)               +  \n",
       "      chr10:106708998-106817135:+         Lin7a_(119)               +  \n",
       "      chr10:106817207-106819559:+          Lin7a_(77)               +  \n",
       "      chr10:107689980-107696811:+      Ppp1r12a_(171)               +  \n",
       "      chr10:107708825-107711193:+       Ppp1r12a_(31)               +  \n",
       "      chr10:108339371-108447631:-           Syt1_(80)               -  \n",
       "      chr10:110641896-110670608:+         Osbpl8_(37)               +  \n",
       "      chr10:111893591-111895546:+         Kcnc2_(104)               +  \n",
       "      chr10:114810002-114821260:-         Thap2_(196)               -  \n",
       "      chr10:115944244-115986088:-          Cnot2_(67)               -  \n",
       "      chr10:115954439-115986088:-         Cnot2_(123)               -  \n",
       "      chr10:115964985-116018529:-          Cnot2_(73)               -  \n",
       "      chr10:115965005-115986088:-         Cnot2_(228)               -  \n",
       "      chr10:116356707-116374452:-         Rab3ip_(96)               -  \n",
       "      chr10:116387748-116392583:+    uc007hcn-1_(213)               +  \n",
       "      chr10:116671537-116674673:-  9530003J23Rik_(79)               -  \n",
       "      chr10:116798416-116799070:-         Cpsf6_(111)               -  \n",
       "      chr10:117589330-117594251:+           Mdm1_(30)               +  \n",
       "      chr10:119415663-119423387:+         Grip1_(156)               +  \n",
       "      chr10:119487172-119491888:+          Grip1_(45)               +  \n",
       "      chr10:119807080-119819120:+     uc007hfc-1_(48)               +  \n",
       "      chr10:121533150-121533956:-         Tmem5_(114)               -  \n",
       "      chr10:122702597-123030549:+       Fam19a2_(212)               +  \n",
       "      chr10:124731900-124739012:-        Slc16a7_(81)               -  \n",
       "      chr10:126501276-126501641:+          Cdk4_(234)               +  \n",
       "      chr10:126526414-126527482:+          Agap2_(60)               +  \n",
       "      chr10:126533798-126534987:-           Os9_(165)               -  \n",
       "...                                               ...             ...  \n",
       "tls   chrX:70590251-70597147:+            Zfp275_(92)               +  \n",
       "      chrX:70729101-70736118:+               Bgn_(64)               +  \n",
       "      chrX:71004795-71005233:+           Plxnb3_(128)               +  \n",
       "      chrX:71034292-71035139:+              Ssr4_(99)               +  \n",
       "      chrX:71151202-71151767:-           Arhgap4_(63)               -  \n",
       "      chrX:71262850-71263468:-            Irak1_(117)               -  \n",
       "      chrX:71282671-71330848:-            Mecp2_(124)               -  \n",
       "      chrX:71383926-71388785:+           Opn1mw_(166)               +  \n",
       "      chrX:71533852-71534443:+               Taz_(53)               +  \n",
       "      chrX:7156972-7162937:+               Foxp3_(67)               +  \n",
       "      chrX:71613248-71613755:-             Ubl4_(106)               -  \n",
       "      chrX:73044829-73045841:-              Pls3_(27)               -  \n",
       "      chrX:7344810-7346555:+              Tcfe3_(105)               +  \n",
       "      chrX:7380526-7381486:+             Gripap1_(93)               +  \n",
       "      chrX:7718334-7720376:-               Rbm3_(129)               -  \n",
       "      chrX:7721053-7722101:-               Rbm3_(107)               -  \n",
       "      chrX:7780941-7782711:-              Porcn_(122)               -  \n",
       "      chrX:82991640-82999670:-               Gyk_(84)               -  \n",
       "      chrX:90869254-90871826:-            Pola1_(112)               -  \n",
       "      chrX:91347719-91368406:-               Zfx_(68)               -  \n",
       "      chrX:91785886-91787231:-            Maged1_(74)               -  \n",
       "      chrX:9646690-9694569:-                Rpgr_(60)               -  \n",
       "      chrX:9743402-9755396:-               Rpgr_(879)               -  \n",
       "      chrX:97963767-97966953:+              Dlg3_(54)               +  \n",
       "      chrX:98005246-98008064:+              Dlg3_(42)               +  \n",
       "      chrX:98279613-98280610:-           Slc7a3_(158)               -  \n",
       "      chrX:98497770-98502414:+             Nlgn3_(60)               +  \n",
       "      chrX:98601358-98603815:-            Zmym3_(255)               -  \n",
       "      chrX:98611277-98612247:-            Zmym3_(295)               -  \n",
       "      chrX:99750956-99754903:-            Phka1_(177)               -  \n",
       "\n",
       "[11979 rows x 11 columns]"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sep_scores"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "Index contains duplicate entries, cannot reshape",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-84-04f53e00bd7b>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0msep_scores\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0malt_cassette_events_full\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mxs\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"striatum\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mlevel\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;34m\"cell_type\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msort_index\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0maxis\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'taf15'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'tdp43'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'tls'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msep_score\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0munstack\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      2\u001b[0m \u001b[0msep_scores\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0msep_scores\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfillna\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      3\u001b[0m \u001b[0msep_scores\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0msep_scores\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mT\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mgroupby\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mlevel\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;34m\"location\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mlast\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      4\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      5\u001b[0m \u001b[0mq_values\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0malt_cassette_events_full\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mxs\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"striatum\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mlevel\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;34m\"cell_type\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msort_index\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0maxis\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'taf15'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'tdp43'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'tls'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mq_value\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0munstack\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m/nas3/gpratt/anaconda/lib/python2.7/site-packages/pandas/core/series.pyc\u001b[0m in \u001b[0;36munstack\u001b[1;34m(self, level)\u001b[0m\n\u001b[0;32m   1944\u001b[0m         \"\"\"\n\u001b[0;32m   1945\u001b[0m         \u001b[1;32mfrom\u001b[0m \u001b[0mpandas\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcore\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mreshape\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0munstack\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1946\u001b[1;33m         \u001b[1;32mreturn\u001b[0m \u001b[0munstack\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mlevel\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1947\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1948\u001b[0m     \u001b[1;31m#----------------------------------------------------------------------\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m/nas3/gpratt/anaconda/lib/python2.7/site-packages/pandas/core/reshape.pyc\u001b[0m in \u001b[0;36munstack\u001b[1;34m(obj, level)\u001b[0m\n\u001b[0;32m    397\u001b[0m             \u001b[1;32mreturn\u001b[0m \u001b[0mobj\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mT\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mstack\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mdropna\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mFalse\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    398\u001b[0m     \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 399\u001b[1;33m         \u001b[0munstacker\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0m_Unstacker\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mobj\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mobj\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mlevel\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mlevel\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    400\u001b[0m         \u001b[1;32mreturn\u001b[0m \u001b[0munstacker\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget_result\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    401\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m/nas3/gpratt/anaconda/lib/python2.7/site-packages/pandas/core/reshape.pyc\u001b[0m in \u001b[0;36m__init__\u001b[1;34m(self, values, index, level, value_columns)\u001b[0m\n\u001b[0;32m     93\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     94\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_make_sorted_values_labels\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 95\u001b[1;33m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_make_selectors\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     96\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     97\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m_make_sorted_values_labels\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m/nas3/gpratt/anaconda/lib/python2.7/site-packages/pandas/core/reshape.pyc\u001b[0m in \u001b[0;36m_make_selectors\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    131\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    132\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mmask\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msum\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m<\u001b[0m \u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 133\u001b[1;33m             raise ValueError('Index contains duplicate entries, '\n\u001b[0m\u001b[0;32m    134\u001b[0m                              'cannot reshape')\n\u001b[0;32m    135\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mValueError\u001b[0m: Index contains duplicate entries, cannot reshape"
     ]
    }
   ],
   "source": [
    "sep_scores = alt_cassette_events_full.xs(\"striatum\", level=\"cell_type\").sort_index().loc(axis=0)[['taf15', 'tdp43', 'tls']].sep_score.unstack()\n",
    "sep_scores = sep_scores.fillna(0)\n",
    "sep_scores = sep_scores.T.groupby(level=\"location\").last()\n",
    "\n",
    "q_values = alt_cassette_events_full.xs(\"striatum\", level=\"cell_type\").sort_index().loc(axis=0)[['taf15', 'tdp43', 'tls']].q_value.unstack()\n",
    "q_values = q_values.fillna(1)\n",
    "q_values = q_values.T.groupby(level=\"location\").last()\n",
    "\n",
    "sep_scores = sep_scores[(sep_scores.abs().max(axis=1) > .5) & (q_values.min(axis=1) <= .05)].T\n",
    "\n",
    "\n",
    "sep_scores.index = np.array([rbp_name_dict[item] for item in sep_scores.index])\n",
    "\n",
    "sns.clustermap(sep_scores, linewidths=0, xticklabels=False, figsize=(4,4))\n",
    "plt.savefig(os.path.join(img_dir, \"splicing_events_straitum.pdf\"))\n",
    "plt.savefig(os.path.join(img_dir, \"splicing_events_straitum.svg\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "excluded = sep_scores[sep_scores < -.5]\n",
    "included = sep_scores[sep_scores > .5]\n",
    "\n",
    "num_rows = 1\n",
    "num_cols = 2\n",
    "with dataviz.Figure(os.path.join(img_dir, \"included_excluded_pie_charts.svg\"), figsize=(1.5 * num_cols,1.5*num_rows)) as fig:\n",
    "    ax = fig.add_subplot(num_rows, num_cols, 1)\n",
    "\n",
    "    venn3([included.ix['TAF15'].dropna().keys(),\n",
    "           included.ix['TDP43'].dropna().keys(),\n",
    "           included.ix['FUS/TLS'].dropna().keys()],\n",
    "         set_labels=[\"TAF15\", \"TDP43\", \"FUS/TLS\"],\n",
    "         set_colors=[color_map['taf15'], color_map['tdp43'], color_map['fus']],\n",
    "         ax=ax)\n",
    "    ax.set_title(\"Included\")\n",
    "    \n",
    "    ax = fig.add_subplot(num_rows, num_cols, 2)\n",
    "\n",
    "    venn3([excluded.ix['TAF15'].dropna().keys(),\n",
    "           excluded.ix['TDP43'].dropna().keys(),\n",
    "           excluded.ix['FUS/TLS'].dropna().keys()],\n",
    "         set_labels=[\"TAF15\", \"TDP43\", \"FUS/TLS\"],\n",
    "         set_colors=[color_map['taf15'], color_map['tdp43'], color_map['fus']],\n",
    "         ax=ax)\n",
    "    ax.set_title(\"Excluded\")\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Figure Legend: Clustergram of all alt cassette events for TAF15, FUS and TDP43 in all tissue types "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Set up things I need for GO analysis, background expression and GO object"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "metadata = pd.read_csv(\"/nas3/gpratt/Dropbox/TAF15/unix_mouse_fet_data_locations_and_annitations.csv\")\n",
    "metadata = metadata[metadata.geo_name != \"not_submitted\"]\n",
    "null_bioreps = metadata[metadata['Biological Replicate'].isnull()]\n",
    "metadata.loc[null_bioreps.index,'Biological Replicate'] = null_bioreps.Location.apply(lambda x: os.path.basename(x).split(\".\")[0]).values\n",
    "biorep_info = metadata.groupby('Biological Replicate').first()\n",
    "\n",
    "all_counts = pd.read_table(\"/nas3/gpratt/Dropbox/TAF15/Data/mouse_integration/all_counts.txt\", index_col=0, skiprows=1)\n",
    "all_counts.columns = [item.split(\".\")[0] for item in all_counts.columns]\n",
    "rpkm = counts_to_rpkm(all_counts)\n",
    "rpkm = rpkm.T\n",
    "rpkm = rpkm.ix[biorep_info.index]\n",
    "flotilla_rpkm = rpkm.copy()\n",
    "\n",
    "result = []\n",
    "for index in rpkm.index:\n",
    "    exp_info = biorep_info.ix[index] \n",
    "    result.append([exp_info['rbp'], exp_info['Cell Type'], exp_info['bio_rep'], exp_info['analysis_group'], biorep_info.ix[index].name])\n",
    "rpkm.index = pd.MultiIndex.from_tuples(result, names=['knockdown', 'cell_type', 'rep', 'analysis_group', 'exp'])\n",
    "rpkm = rpkm.sortlevel()\n",
    "\n",
    "#Executing\n",
    "rpkm = rpkm.drop(\"PBS\", level=0)\n",
    "rpkm = rpkm.drop(\"TAF15_ASO_3\", level=\"exp\")\n",
    "rpkm = rpkm.drop(\"TIA1_Br_1\", level=\"exp\")\n",
    "\n",
    "mapper = {\n",
    "('ews', 'spinal cord'): (\"HTT\", \"Spinal Cord\", 4),\n",
    "('fus', 'spinal cord'): (\"HTT\", \"Spinal Cord\", 6),\n",
    "('tls', 'striatum'): (\"Control Oligo\", \"Striatum\", 1),\n",
    "('hnrnpa2b1', 'spinal cord'): (\"HTT\", \"Spinal Cord\", 7),\n",
    "('elav', 'spinal cord'): (\"HTT\", \"Spinal Cord\", 5),\n",
    "('taf15', 'spinal cord'): (\"HTT\", \"Spinal Cord\", 4),\n",
    "('taf15', 'striatum'): (\"HTT\", \"Striatum\", 2),\n",
    "('taf15', 'whole brain'): (\"HTT\", \"Whole Brain\", 3),\n",
    "('tdp43', 'striatum'): (\"Control Oligo\", \"Striatum\", 1),\n",
    "('tia1', 'spinal cord'): (\"HTT\", \"Spinal Cord\", 5),\n",
    "('tia1', 'striatum'): (\"HTT\", \"Striatum\", 2),\n",
    "('tia1', 'whole brain'): (\"HTT\", \"Whole Brain\", 5),\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "mm9go = GO.mm9GO()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#This takes a few minutes...\n",
    "all_alt_cassete_events_go = {}\n",
    "for name, df in significant_alt_cassette_events.groupby(level=['rbp', 'cell_type']):\n",
    "\n",
    "    expressed_genes = get_expressed_genes(*name)\n",
    "    all_alt_cassete_events_go[name] = mm9go.enrichment(gencode_to_ensembl(df.gene_id),\n",
    "                                               background=expressed_genes\n",
    "                                               )\n",
    "    \n",
    "all_alt_cassete_events_go = pd.concat(all_alt_cassete_events_go, names=['rbp', 'cell_type', 'go_term_id'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "sig_changing = pd.concat({\"included\": significant_alt_cassette_events[significant_alt_cassette_events.sep_score > 0], \n",
    "                          \"excluded\": significant_alt_cassette_events[significant_alt_cassette_events.sep_score < 0]}, names=[\"change\", 'rbp', 'cell_type', \"location\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#This takes a few minutes...\n",
    "sig_change_go = {}\n",
    "for name, df in sig_changing.groupby(level=['change', 'rbp', 'cell_type']):\n",
    "    expressed_genes = get_expressed_genes(*name[1:])\n",
    "    sig_change_go[name] = mm9go.enrichment(gencode_to_ensembl(df.gene_id),\n",
    "                                               background=expressed_genes\n",
    "                                               )\n",
    "    \n",
    "sig_change_go = pd.concat(sig_change_go, names=['change', 'rbp', 'cell_type', 'go_term_id'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#All Alt-Cassete Events"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "plot_go_enrichment(all_alt_cassete_events_go, figsize=(7,55))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "plot_go_enrichment(all_alt_cassete_events_go.ix['taf15'], figsize=(7,55))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "plot_go_enrichment(sig_change_go, figsize=(15,40))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "plot_go_enrichment(sig_change_go.xs(\"striatum\", level=\"cell_type\"), figsize=(10,20))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "plot_go_enrichment(sig_change_go.xs(\"taf15\", level=\"rbp\"), figsize=(10,20))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "sig_change_go.xs(\"GO:0005524\", level=\"go_term_id\").xs(\"taf15\", level=\"rbp\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#What do the distribution of sep scores look like for significant changes?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "with dataviz.Figure(os.path.join(img_dir, \"sep_score_distributions.svg\"), figsize=(8,4)) as fig:\n",
    "    ax = fig.add_subplot(1,1,1)\n",
    "    g = [\" \".join(item[:-1]) for item in all_events.index]\n",
    "    sns.violinplot(all_events.sep_score, groupby=g, ax=ax)\n",
    "    [tick.set_rotation(90) for tick in ax.get_xticklabels()]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Figure Legend: Violin Plot of distributon of sep scores for all knockdown experiments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "with dataviz.Figure(os.path.join(img_dir, \"sep_score_distributions_pdf.svg\"), figsize=(8,4)) as fig:\n",
    "    ax = fig.add_subplot(1,1,1)\n",
    "    for name, df in all_events.sep_score.groupby(level=['rbp', 'cell_type']):\n",
    "        dataviz.plot_pdf(df, label=\" \".join(name), ax=ax, alpha=.7, linewidth=2)\n",
    "    ax.legend()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Figure Legend: PDF Plot of distributon of sep scores for all knockdown experiments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "with dataviz.Figure(os.path.join(img_dir, \"sep_score_distributions_cdf.svg\"), figsize=(8,4)) as fig:\n",
    "    ax = fig.add_subplot(1,1,1)\n",
    "    for name, df in all_events.sep_score.groupby(level=['rbp', 'cell_type']):\n",
    "        dataviz.plot_cdf(df, label=\" \".join(name), ax=ax, alpha=.7, linewidth=2)\n",
    "    ax.legend()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Figure Legend: CDF Plot of distributon of sep scores for all knockdown experiments"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Conclusions\n",
    "---\n",
    "1. Total Distribution of splicing events isn't very different, except for ews spinal cord, which has a few more excluded events"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#How many specific event locations (give or take a few hundred base pairs have taf15 binding sites?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#How many events overlap binding sites in TAF15?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "regions = OrderedDict()\n",
    "regions[\"cds\"] = \"CDS\"\n",
    "regions[\"three_prime_utrs\"] = \"3' UTR\"\n",
    "regions[\"five_prime_utrs\"] = \"5' UTR\"\n",
    "regions[\"proxintron500\"] = \"Proximal\\nIntron\"\n",
    "regions[\"distintron500\"] = \"Distal\\nIntron\"\n",
    "\n",
    "def move_name(interval):\n",
    "    interval.name = interval[12]\n",
    "    return interval"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "genes = pybedtools.BedTool(\"/nas3/gpratt/clipper/clipper/data/regions/mm9_genes.bed\")\n",
    "rbps = {\"taf15\": pybedtools.BedTool(\"/nas3/gpratt/projects/fet_family/data/stuff_for_KK/TAF15_combined_notrim_ingenes_clusters_mm950.bed\"),\n",
    "        \"tls\": pybedtools.BedTool(\"/nas3/gpratt/projects/fet_family/data/stuff_for_KK/TLS_hiseq_notrim_ingenes_clusters_mm950.bed\"),\n",
    "        \"tdp43\": pybedtools.BedTool(\"/nas3/gpratt/projects/fet_family/data/stuff_for_KK/TDP43brainclip_MP41_kcomb_notrim_ingenes_clusters_mm950.bed\"),\n",
    " }\n",
    "\n",
    "binding_df = []\n",
    "assigned_peaks = {}\n",
    "\n",
    "for name, rbp in rbps.items():\n",
    "    overlapping_genes = rbp.intersect(genes, wo=True, s=True).each(move_name).saveas()\n",
    "\n",
    "    cluster_regions = CLIP_analysis.assign_to_regions(overlapping_genes, \"overlapping\", \n",
    "                                    regions, \"assigned\", nrand=0, species=\"mm9\")\n",
    "    assigned_peaks[rbp] = cluster_regions \n",
    "    for region in cluster_regions.keys():\n",
    "        gene_names = pd.Series(Counter([interval.name.split(\",\")[0] for interval in cluster_regions[region]['real']]),\n",
    "                                name=name + \"_\" + region)\n",
    "        gene_names.name = (name, region)\n",
    "        binding_df.append(gene_names)\n",
    "binding_df = pd.concat(binding_df,axis=1)\n",
    "binding_df.columns = pd.MultiIndex.from_tuples(binding_df.columns)\n",
    "binding_df = binding_df.T\n",
    "binding_df = binding_df.stack()\n",
    "binding_df.index.names = ['rbp', 'region', 'gene_id']\n",
    "binding_df = binding_df.sort_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "all_events_significance = defaultdict(dict)\n",
    "for (spliced_rbp, cell_type), splicng_df in all_events.groupby(level=['rbp', 'cell_type']):\n",
    "    expressed_genes = get_expressed_genes(spliced_rbp, cell_type)\n",
    "    for (bound_rbp, region), region_df in binding_df.groupby(level=['rbp', 'region']):\n",
    "        total_bound = len(region_df)\n",
    "        total_spliced = len(splicng_df)\n",
    "        both = len(region_df.index.get_level_values(level=\"gene_id\") & splicng_df.gene_id)\n",
    "        total = len(expressed_genes)\n",
    "        test_result = hypergeom.sf(both, total, total_bound, total_spliced)\n",
    "        all_events_significance[(spliced_rbp, cell_type)][(bound_rbp, region)] = test_result\n",
    "        \n",
    "all_events_significance = pd.DataFrame(all_events_significance).dropna(axis=1)\n",
    "all_events_significance[all_events_significance > .05 / (len(all_events_significance.columns) * len(all_events_significance.index))] = 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "len(all_events_significance.columns), len(all_events_significance.index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "sns.heatmap(-1 * np.log10(all_events_significance))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Figure Legend: -1*log10(hypergeometric distribution) of genes that have an alt splicing event, and genes that are bound by the given RBP in the specificed region"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "alt_cassette_events_significance = defaultdict(dict)\n",
    "for (spliced_rbp, cell_type), splicng_df in alt_cassette_events.groupby(level=['rbp', 'cell_type']):\n",
    "    expressed_genes = get_expressed_genes(spliced_rbp, cell_type)\n",
    "    for (bound_rbp, region), region_df in binding_df.groupby(level=['rbp', 'region']):\n",
    "        total_bound = len(region_df)\n",
    "        total_spliced = len(splicng_df)\n",
    "        both = len(region_df.index.get_level_values(level=\"gene_id\") & splicng_df.gene_id)\n",
    "        total = len(expressed_genes)\n",
    "        test_result = hypergeom.sf(both, total, total_bound, total_spliced)\n",
    "        alt_cassette_events_significance[(spliced_rbp, cell_type)][(bound_rbp, region)] = test_result\n",
    "        \n",
    "alt_cassette_events_significance = pd.DataFrame(alt_cassette_events_significance).dropna(axis=1)\n",
    "alt_cassette_events_significance[alt_cassette_events_significance > .05 / (21 * 10)] = 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "sns.heatmap(-1 * np.log10(alt_cassette_events_significance))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Figure Legend: -1*log10(hypergeometric distribution) of genes that have an alt cassette event, and genes that are bound by the given RBP in the specificed region"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def format_bed_from_location(array_interval):\n",
    "    chrom, loc, strand = array_interval.split(\":\")\n",
    "    start, stop = loc.split(\"-\")\n",
    "    return map(str, [chrom, start, stop, 0, 0, strand, chrom])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# alt_cassette_events_significance = defaultdict(dict)\n",
    "\n",
    "# for name, df in alt_cassette_events.groupby(level=[\"rbp\", \"cell_type\"]):\n",
    "#     array_regions = pybedtools.BedTool([pybedtools.interval_constructor(format_bed_from_location(array_interval)) for array_interval in df.index.get_level_values(level=\"location\")]).sort().saveas()\n",
    "#     for rbp in rbps:\n",
    "#         total_bound = len(rbps[rbp])\n",
    "#         total_spliced = len(array_regions)\n",
    "#         both = len(array_regions.intersect(rbps[rbp], s=True, u=True))\n",
    "#         total = len(expressed_genes)\n",
    "#         print name, rbp, both, total, total_bound, total_spliced\n",
    "#         test_result = hypergeom.sf(both, total, total_bound, total_spliced)\n",
    "#         alt_cassette_events_significance[name][rbp] = test_result\n",
    "        \n",
    "#alt_cassette_events_significance = pd.DataFrame(alt_cassette_events_significance).dropna(axis=1)\n",
    "#alt_cassette_events_significance[alt_cassette_events_significance > .05 / (21 * 10)] = 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#sns.heatmap(-1 * np.log10(alt_cassette_events_significance))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Conclusions:\n",
    "----\n",
    "1. TIA1 isn't an important splicing factor and can be dropped out of this analysis\n",
    "2. TLS spliced sites are highly bound by TAF15. The opposite is not totally true\n",
    "3. TAF15 splicing is un-regulated by RBPs in Straitum and whole brain, but may be important in splinal cord\n",
    "4. ELAV, EWS, TDP43 and FUS all apparently cross-react and may cross-regulate according to binding analysis."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#What about auto/cross-regulation?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "unstacked_binding = binding_df.unstack()\n",
    "unstacked_binding = unstacked_binding.rename_axis(gene_id_to_name, axis=1)\n",
    "unstacked_binding[['Taf15', 'Tardbp', 'Fus']]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Figure Legend: Cross-binding of TAF15, TDP43 and TLS the number of times the column binds the row, and where that binding occurs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "So they all do bind each other, I don't think expression is significantly changed, but I'll have to check in another thing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "all_events[all_events.gene_name.isin(['Tardbp', 'Fus', \"Taf15\"])].gene_name"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Figure Legend: Splicing events occuring in other FET family members and the knockdown doing it"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Conclusions\n",
    "--\n",
    "1. There is some regulation of Fus for all the other RBPs\n",
    "2. Taf15 only regulates itself, badly, and Tdp-43 doesn't at all\n",
    "3. Importantly weren't many events that were even assayed in TDP-43, FUS and TAF15, only one or two each.  Could be missing things. This conclusion was drawn from Steph's analysis of the data the all_events tab of Dropbox/TAF15/Data/SplicingAnalysis_array/03262013/taf15_tdp43_fus_ewssc_summary.xlsx"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Lets make read density plots around skipped exons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "exons = pybedtools.BedTool(\"/nas3/gpratt/clipper/clipper/data/regions/mm9_exons.bed\")\n",
    "\n",
    "#Get Miso SE events\n",
    "#miso_alt_splicing_events = pybedtools.BedTool(\"/home/gpratt/data/miso/mm9/mm9/SE.mm9.gff3\")\n",
    "#miso_alt_splicing_events = miso_alt_splicing_events.filter(lambda interval: interval[2] == \"gene\").sort().saveas()\n",
    "\n",
    "#Once I get the rest of this built I'll go back and build in the proper events\n",
    "#all_array_annotated_events = pd.read_table(\"/nas3/shuelga/projects/taf15_splicing/TAF15_TDP43_FUS_EWSsc_omniviewer.txt\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Check if miso events overlap with the omnivier events and join omniver events on miso events\n",
    "It doesn't, via some simple intersect bed commands that I got rid of already, something is wrong with the mouse miso annotations so I'm falling back on using arrays "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#print len(array_alt_splicing_events), len(miso_alt_splicing_events)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Miso annotations don't work, they aren't detailed enough, the next best thing is to look at all exons that fall between the annotated event and use those as the locations of taf15 binding"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "import HTSeq\n",
    "\n",
    "location_to_gene_id = {index[-1]: item for index, item in alt_cassette_events.gene_id.iteritems()}\n",
    "\n",
    "def format_bed(array_format):\n",
    "    chrom, loc, strand = array_format.name.split(\":\")\n",
    "    start, stop = loc.split(\"-\")\n",
    "    return map(str, [chrom, start, stop, array_format.name, 0, strand, chrom, array_format.start_internal, array_format.stop_internal])\n",
    "\n",
    "# def format_bed(array_format):\n",
    "#     chrom, loc, strand = array_format.split(\":\")\n",
    "#     start, stop = loc.split(\"-\")\n",
    "#     return [chrom, start, stop, array_format, \"0\", strand]\n",
    "\n",
    "def get_bam_coverage(bamfile):\n",
    "    \"\"\"\n",
    "    \n",
    "    Given a bam file returns a properly covered htseq coverage file (this is slow)\n",
    "    \n",
    "    \"\"\"\n",
    "    bam = HTSeq.BAM_Reader(bamfile)\n",
    "    coverage = HTSeq.GenomicArray(\"auto\", typecode=\"i\", stranded=True)\n",
    "    for read in bam:\n",
    "        if read.aligned:\n",
    "            for cigop in read.cigar:\n",
    "                if cigop.type != \"M\":\n",
    "                    continue\n",
    "                coverage[cigop.ref_iv] += 1\n",
    "    return coverage\n",
    "\n",
    "def get_bam_counts(bamfile):\n",
    "    bam = HTSeq.BAM_Reader(bamfile)\n",
    "    coverage = HTSeq.GenomicArray(\"auto\", typecode=\"i\")\n",
    "    for read in bam:\n",
    "        read.iv.length = 1\n",
    "        if read.aligned:\n",
    "            coverage[read.iv] += 1\n",
    "    return coverage\n",
    "\n",
    "def bed_to_genomic_interval(bed):\n",
    "    for interval in bed:\n",
    "        yield HTSeq.GenomicInterval(interval.chrom, interval.start, interval.stop, interval.strand)\n",
    "\n",
    "def adjust_width(interval):\n",
    "    width = 250\n",
    "    interval.start -= width\n",
    "    interval.stop += width\n",
    "    return interval\n",
    "\n",
    "def is_protein_coding(interval):\n",
    "    return interval.name in protein_coding_genes.index\n",
    "\n",
    "def gff_to_bed(gff_handle):\n",
    "    for interval in gff_handle:\n",
    "        yield pybedtools.create_interval_from_list([interval.chrom, \n",
    "                                                 str(interval.start),\n",
    "                                                 str(interval.stop),\n",
    "                                                 interval.attributes['gene_id'][0], \n",
    "                                                 \"0\",\n",
    "                                                 interval.strand])\n",
    "\n",
    "def get_densities(intervals, coverage):\n",
    "    for bed_interval, interval in izip(intervals, bed_to_genomic_interval(intervals)):\n",
    "        \n",
    "        density = np.fromiter(coverage[interval], dtype=\"i\")\n",
    "        if interval.strand == \"-\":\n",
    "            density = density[::-1]\n",
    "        yield {bed_interval.name: density}\n",
    "   \n",
    "def check_same_gene(interval):\n",
    "    return location_to_gene_id[interval.name] == interval[9]\n",
    "\n",
    "def format_bed(array_format):\n",
    "    chrom, loc, strand = array_format.name.split(\":\")\n",
    "    start, stop = loc.split(\"-\")\n",
    "    return map(str, [chrom, start, stop, array_format.name, \"0\", strand, array_format.start_internal, array_format.stop_internal])\n",
    "\n",
    "def create_array_exon_overlap_tool(array_locations):\n",
    "    \"\"\"\n",
    "    Given a list of array based location returns a bedtool with all exons within those locations\n",
    "    for creation of splicing density plot\n",
    "    \"\"\"\n",
    "    \n",
    "    #to clean this up make the array id overlap only with exons from the same gene, this is already assumed for other analyzies anyway\n",
    "    bed_list = [pybedtools.interval_constructor(format_bed(array_interval)) for name, array_interval in array_locations.iterrows()]\n",
    "    return pybedtools.BedTool(bed_list).sort().saveas()\n",
    "\n",
    "def get_mapping_intervals(interval):\n",
    "    \"\"\"\n",
    "    given the crazy formatted spliced exon thing returns the interval to get coverage from\n",
    "    when plotting around regions\n",
    "    \"\"\"\n",
    "    LEN_IN_INTRON = 300\n",
    "    LEN_IN_EXON = 50\n",
    "    cassette_start = int(float(interval[6]))\n",
    "    cassette_stop  = int(float(interval[7]))\n",
    "    \n",
    "    #If there is no overlap with gencode genes...\n",
    "    if cassette_start == -1:\n",
    "        return {}\n",
    "\n",
    "    if interval.strand == \"+\":\n",
    "        result = {\n",
    "        \"upstream_exon\" : [interval.chrom, interval.start - LEN_IN_EXON, interval.start + LEN_IN_INTRON, interval.name, \"0\", interval.strand],\n",
    "        \"downstream_exon\":  [interval.chrom, interval.stop - LEN_IN_INTRON, interval.stop + LEN_IN_EXON, interval.name, \"0\", interval.strand],\n",
    "        \"upstream_cassette\": [interval.chrom, cassette_start - LEN_IN_EXON, cassette_start + LEN_IN_INTRON, interval.name, \"0\", interval.strand],\n",
    "        \"downstream_cassette\": [interval.chrom, cassette_stop - LEN_IN_INTRON, cassette_stop + LEN_IN_EXON, interval.name, \"0\", interval.strand],\n",
    "        }\n",
    "    elif interval.strand == \"-\":\n",
    "        result = {\n",
    "        \"downstream_exon\": [interval.chrom, interval.start - LEN_IN_EXON, interval.start + LEN_IN_INTRON, interval.name, \"0\", interval.strand],\n",
    "        \"upstream_exon\": [interval.chrom, interval.stop - LEN_IN_INTRON, interval.stop + LEN_IN_EXON, interval.name, \"0\", interval.strand],\n",
    "        \"downstream_cassette\": [interval.chrom, cassette_start - LEN_IN_INTRON, cassette_start + LEN_IN_EXON, interval.name, \"0\", interval.strand],\n",
    "        \"upstream_cassette\": [interval.chrom, cassette_stop - LEN_IN_EXON, cassette_stop + LEN_IN_INTRON, interval.name, \"0\", interval.strand],\n",
    "    }\n",
    "    else:\n",
    "        raise Exception\n",
    "\n",
    "    return {name: pybedtools.interval_constructor(map(str, interval)) for name, interval in result.items() }\n",
    "\n",
    "\n",
    "def get_mapping_tools(joined_files):\n",
    "    \"\"\"\n",
    "    Given a bedtool generated from joining an array file and a exon file returns back \n",
    "    all intervals to extract region information from\n",
    "    \"\"\"\n",
    "    mapping_intervals = defaultdict(list)\n",
    "    for interval in joined_files:\n",
    "        for name, interval in get_mapping_intervals(interval).items():\n",
    "            mapping_intervals[name].append(interval)\n",
    "\n",
    "    return {name: pybedtools.BedTool(intervals) for name, intervals in mapping_intervals.items() }\n",
    "\n",
    "def density_to_df(density):\n",
    "\n",
    "    \"\"\"\n",
    "    \n",
    "    Given a density generator returns a DF with properly labeled axes\n",
    "    rows being IDs and columns being bases\n",
    "    \n",
    "    \"\"\"\n",
    "    super_dict = {}\n",
    "    for d in density:\n",
    "        for k, v in d.iteritems():\n",
    "            super_dict[k] = v\n",
    "    return pd.DataFrame(super_dict).T\n",
    "\n",
    "\n",
    "\n",
    "def create_array_exon_overlap_tool(array_locations):\n",
    "    \"\"\"\n",
    "    Given a list of array based location returns a bedtool with all exons within those locations\n",
    "    for creation of splicing density plot\n",
    "    \"\"\"\n",
    "    \n",
    "    #to clean this up make the array id overlap only with exons from the same gene, this is already assumed for other analyzies anyway\n",
    "    return pybedtools.BedTool([pybedtools.interval_constructor(format_bed(array_interval)) for name, array_interval in array_locations.iterrows()])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "bam_files = {\"taf15\": \"/nas3/gpratt/projects/fet_family/analysis/mouse_clip_v1/TAF15_WholeBrain.bam\",\n",
    "             \"tdp43\": \"/nas3/gpratt/projects/fet_family/analysis/mouse_clip_v1/TDP43_WholeBrain.bam\",\n",
    "             \"tls\" : \"/nas3/gpratt/projects/fet_family/analysis/mouse_clip_v1/TLS_WholeBrain.bam\"}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "coverage = {name: get_bam_caoverage(location) for name, location in bam_files.items()}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "event_matrix.ix[]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "#Doing this shit really hacky like, getting all classes of regulated events\n",
    "event_matrix = significant_alt_cassette_events.sep_score.unstack().fillna(0).T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "plotting_order = OrderedDict()\n",
    "plotting_order['upstream_exon'] = np.arange(-50, 300)\n",
    "plotting_order['upstream_cassette'] = np.arange(-50, 300)[::-1]\n",
    "plotting_order['downstream_cassette'] = np.arange(-50, 300)\n",
    "plotting_order['downstream_exon'] = np.arange(-50, 300)[::-1]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Here is where we actually can plot the distributions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "usable_splicing_datasets = [col for col in event_matrix.columns if col[0] in coverage]\n",
    "num_cols = 4\n",
    "num_rows = len(usable_splicing_datasets)\n",
    "\n",
    "#with dataviz.Figure(os.path.join(img_dir, \"distributions_around_alt_cassete.svg\"), figsize=(num_cols * 2.5,num_rows * 2.5)) as fig:\n",
    "gs = gridspec.GridSpec(num_rows, num_cols)\n",
    "\n",
    "for y, col in enumerate(usable_splicing_datasets):\n",
    "    rbp = col[0]\n",
    "    if rbp not in coverage:\n",
    "        continue\n",
    "    rbp_of_interest = event_matrix[col] \n",
    "    activated = rbp_of_interest[rbp_of_interest > 0]\n",
    "    repressed = rbp_of_interest[rbp_of_interest < 0]\n",
    "    unchanged = rbp_of_interest[rbp_of_interest == 0]\n",
    "\n",
    "    activated_tools = get_mapping_tools(create_array_exon_overlap_tool(alt_cassette_event_locations.ix[activated.index]))\n",
    "    repressed_tools = get_mapping_tools(create_array_exon_overlap_tool(alt_cassette_event_locations.ix[repressed.index]))\n",
    "    unchanged_tools = get_mapping_tools(create_array_exon_overlap_tool(alt_cassette_event_locations.ix[unchanged.index]))\n",
    "\n",
    "    for x, (region, x_loc) in enumerate(plotting_order.items()):\n",
    "        ax = fig.add_subplot(gs[y,x])\n",
    "\n",
    "        ax.plot(x_loc, density_to_df(get_densities(activated_tools[region], coverage[rbp])).mean(), alpha=.7, label=\"Included\")\n",
    "        ax.plot(x_loc, density_to_df(get_densities(repressed_tools[region], coverage[rbp])).mean(), alpha=.7, label=\"Excluded\")\n",
    "        ax.plot(x_loc, density_to_df(get_densities(unchanged_tools[region], coverage[rbp])).mean(), alpha=.7, label=\"Unchanging\")\n",
    "        ax.set_ylim(0, 3.0)\n",
    "        ax.set_xlabel(region + \"\\nDistance from Splice Site\\n(bp)\")\n",
    "\n",
    "        if not ax.is_first_col():\n",
    "            sns.despine(ax=ax, left=True)\n",
    "            ax.set_yticklabels([])\n",
    "        else: \n",
    "            sns.despine(ax=ax)\n",
    "            ax.set_ylabel(\" \".join(col) + \"\\nMean Coverage\" )\n",
    "\n",
    "\n",
    "        if ax.is_first_row() and ax.is_last_col():\n",
    "            ax.legend(bbox_to_anchor=(1.5, 1))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Figure Legend: Mean Read Densities around annotated splciing events for upstream 5' SS, cassete exon 3' and '5 SS and downstream 3' SS.  For TAF15, FUS and TDP43"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#If we normalize out varying read depths per sample what does it look like? \n",
    "Ie. We just want to know specific binding patterns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "usable_splicing_datasets = [col for col in event_matrix.columns if col[0] in coverage]\n",
    "num_cols = 4\n",
    "num_rows = len(usable_splicing_datasets)\n",
    "\n",
    "with dataviz.Figure(os.path.join(img_dir, \"distributions_around_alt_cassete_profile.svg\"), figsize=(num_cols * 2.5,num_rows * 2.5)) as fig:\n",
    "    gs = gridspec.GridSpec(num_rows, num_cols)\n",
    "\n",
    "    for y, col in enumerate(usable_splicing_datasets):\n",
    "        rbp = col[0]\n",
    "        if rbp not in coverage:\n",
    "            continue\n",
    "        rbp_of_interest = event_matrix[col] \n",
    "        activated = rbp_of_interest[rbp_of_interest > 0]\n",
    "        repressed = rbp_of_interest[rbp_of_interest < 0]\n",
    "        unchanged = rbp_of_interest[rbp_of_interest == 0]\n",
    "\n",
    "        activated_tools = get_mapping_tools(create_array_exon_overlap_tool(alt_cassette_event_locations.ix[activated.index]))\n",
    "        repressed_tools = get_mapping_tools(create_array_exon_overlap_tool(alt_cassette_event_locations.ix[repressed.index]))\n",
    "        unchanged_tools = get_mapping_tools(create_array_exon_overlap_tool(alt_cassette_event_locations.ix[unchanged.index]))\n",
    "\n",
    "        for x, (region, x_loc) in enumerate(plotting_order.items()):\n",
    "            ax = fig.add_subplot(gs[y,x])\n",
    "\n",
    "            activated_df = density_to_df(get_densities(activated_tools[region], coverage[rbp]))\n",
    "            activated_df = activated_df.div(activated_df.sum(axis=1), axis=0).fillna(0)\n",
    "            \n",
    "            repressed_df = density_to_df(get_densities(repressed_tools[region], coverage[rbp]))\n",
    "            repressed_df = repressed_df.div(repressed_df.sum(axis=1), axis=0).fillna(0)\n",
    "\n",
    "            unchanged_df = density_to_df(get_densities(unchanged_tools[region], coverage[rbp]))\n",
    "            unchanged_df = unchanged_df.div(unchanged_df.sum(axis=1), axis=0).fillna(0)\n",
    "  \n",
    "            \n",
    "            ax.plot(x_loc, activated_df.mean(), alpha=.7, label=\"Included\")\n",
    "            ax.plot(x_loc, repressed_df.mean(), alpha=.7, label=\"Excluded\")\n",
    "            ax.plot(x_loc, unchanged_df.mean(), alpha=.7, label=\"Unchanging\")\n",
    "            ax.set_ylim(0, .006)\n",
    "            ax.set_xlabel(region + \"\\nDistance from Splice Site\\n(bp)\")\n",
    "            \n",
    "            if not ax.is_first_col():\n",
    "                sns.despine(ax=ax, left=True)\n",
    "                ax.set_yticklabels([])\n",
    "            else: \n",
    "                sns.despine(ax=ax)\n",
    "                ax.set_ylabel(\" \".join(col) + \"\\nMean Coverage\" )\n",
    "                \n",
    "                \n",
    "            if ax.is_first_row() and ax.is_last_col():\n",
    "                ax.legend(bbox_to_anchor=(1.5, 1))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Figure Legend: Densities normalized by read covered around region for annotated splciing events for upstream 5' SS, cassete exon 3' and '5 SS and downstream 3' SS.  For TAF15, FUS and TDP43"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def get_significant_sites(foreground_df, background_df):\n",
    "    targets = (foreground_df > 0).sum() \n",
    "    non_targets = foreground_df.count() - targets\n",
    "\n",
    "    unchanged_targets = (background_df > 0).sum() \n",
    "    unchanged_non_targets = background_df.count() - targets\n",
    "\n",
    "    result = []\n",
    "    for true_targets, true_nontargets, unchanged_targets, unchanged_non_targets in zip(targets, non_targets, unchanged_targets, unchanged_non_targets):\n",
    "        result.append(scipy.stats.fisher_exact([[true_targets, true_nontargets], [unchanged_targets, unchanged_non_targets]])[1])\n",
    "    \n",
    "    result = np.array(result)\n",
    "    significant = np.where(result < (.05 / 350))\n",
    "    return significant"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "usable_splicing_datasets = [col for col in event_matrix.columns if col[0] in coverage]\n",
    "num_cols = 4\n",
    "num_rows = len(usable_splicing_datasets)\n",
    "\n",
    "with dataviz.Figure(os.path.join(img_dir, \"distributions_around_alt_cassete_profile.svg\"), figsize=(num_cols * 2.5,num_rows * 2.5)) as fig:\n",
    "    gs = gridspec.GridSpec(num_rows, num_cols)\n",
    "\n",
    "    for y, col in enumerate(usable_splicing_datasets):\n",
    "        rbp = col[0]\n",
    "        if rbp not in coverage:\n",
    "            continue\n",
    "        rbp_of_interest = event_matrix[col] \n",
    "        activated = rbp_of_interest[rbp_of_interest > 0]\n",
    "        repressed = rbp_of_interest[rbp_of_interest < 0]\n",
    "        unchanged = rbp_of_interest[rbp_of_interest == 0]\n",
    "\n",
    "        activated_tools = get_mapping_tools(create_array_exon_overlap_tool(alt_cassette_event_locations.ix[activated.index]))\n",
    "        repressed_tools = get_mapping_tools(create_array_exon_overlap_tool(alt_cassette_event_locations.ix[repressed.index]))\n",
    "        unchanged_tools = get_mapping_tools(create_array_exon_overlap_tool(alt_cassette_event_locations.ix[unchanged.index]))\n",
    "\n",
    "        for x, (region, x_loc) in enumerate(plotting_order.items()):\n",
    "            ax = fig.add_subplot(gs[y,x])\n",
    "            activated_df = density_to_df(get_densities(activated_tools[region], coverage[rbp]))\n",
    "            activated_plotter = (activated_df > 0).sum() / activated_df.count()\n",
    "\n",
    "            repressed_df = density_to_df(get_densities(repressed_tools[region], coverage[rbp]))\n",
    "            repressed_plotter = (repressed_df > 0).sum() / repressed_df.count()\n",
    "\n",
    "            unchanged_df = density_to_df(get_densities(unchanged_tools[region], coverage[rbp]))\n",
    "            unchanged_plotter = (unchanged_df > 0).sum() / unchanged_df.count()\n",
    "\n",
    "\n",
    "            ax.fill_between(x_loc, 0, activated_plotter, alpha=.7, label=\"Included\", facecolor=\"blue\")\n",
    "            ax.fill_between(x_loc, 0, repressed_plotter * -1, alpha=.7, label=\"Excluded\", facecolor=\"red\")\n",
    "\n",
    "            ax.fill_between(x_loc, 0, unchanged_plotter, alpha=.7, label=\"Unchanging\")\n",
    "            ax.fill_between(x_loc, 0, unchanged_plotter * -1, alpha=.7, label=\"Unchanging\")\n",
    "\n",
    "            significant_activated = get_significant_sites(activated_df, unchanged_df)[0]\n",
    "            significant_repressed = get_significant_sites(repressed_df, unchanged_df)[0]\n",
    "            ax.scatter(x_loc[significant_activated], activated_plotter.ix[significant_activated].values, color='black')\n",
    "            ax.scatter(x_loc[significant_repressed], repressed_plotter.ix[significant_repressed].values * -1, color='black')\n",
    "            ax.set_xlabel(region + \"\\nDistance from Splice Site\\n(bp)\")\n",
    "            ax.set_xlim((x_loc[0], x_loc[-1]))\n",
    "            ax.set_ylim((-1,1))\n",
    "            \n",
    "            #This really shouldn't be here, and should be factored out, but putting in bound_exon info here\n",
    "            try:\n",
    "                cur_alt_cassette_events = alt_cassette_events.ix[col]\n",
    "                bound_alt_cassette_events = cur_alt_cassette_events.ix[activated_df[(activated_df[significant_activated] > 0).apply(any, axis=1)].index]\n",
    "                bound_alt_cassette_events.to_csv(\"/nas3/gpratt/Dropbox/TAF15/Data/SplicingAnalysis_array/splicing_event_locations/{}_{}_included_alt_cassette_events.csv\".format(\"_\".join(col), region))            \n",
    "            except KeyError:\n",
    "                pass\n",
    "            \n",
    "            if not ax.is_first_col():\n",
    "                sns.despine(ax=ax, left=True)\n",
    "                ax.set_yticklabels([])\n",
    "            else: \n",
    "                sns.despine(ax=ax)\n",
    "                ax.set_ylabel(\" \".join([rbp_name_dict[col[0]], cell_type_dict[col[1]]]) + \"\\nFraction of Sites\" )\n",
    "                \n",
    "                \n",
    "            if ax.is_first_row() and ax.is_last_col():\n",
    "                ax.legend(bbox_to_anchor=(1.5, 1))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Figure Legend: Fraction of sites covered by a binding event for annotated splciing events for upstream 5' SS, cassete exon 3' and '5 SS and downstream 3' SS.  For TAF15, FUS and TDP43"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "usable_splicing_datasets = [('taf15', 'whole brain')]\n",
    "num_cols = 4\n",
    "num_rows = len(usable_splicing_datasets)\n",
    "\n",
    "with dataviz.Figure(os.path.join(img_dir, \"distributions_around_alt_cassete_profile_taf15_only.svg\"), figsize=(num_cols * 1.3,num_rows * 1.5)) as fig:\n",
    "    gs = gridspec.GridSpec(num_rows, num_cols)\n",
    "\n",
    "    for y, col in enumerate(usable_splicing_datasets):\n",
    "        rbp = col[0]\n",
    "        if rbp not in coverage:\n",
    "            continue\n",
    "        rbp_of_interest = event_matrix[col] \n",
    "        activated = rbp_of_interest[rbp_of_interest > 0]\n",
    "        repressed = rbp_of_interest[rbp_of_interest < 0]\n",
    "        unchanged = rbp_of_interest[rbp_of_interest == 0]\n",
    "\n",
    "        activated_tools = get_mapping_tools(create_array_exon_overlap_tool(alt_cassette_event_locations.ix[activated.index].dropna()))\n",
    "        repressed_tools = get_mapping_tools(create_array_exon_overlap_tool(alt_cassette_event_locations.ix[repressed.index].dropna()))\n",
    "        unchanged_tools = get_mapping_tools(create_array_exon_overlap_tool(alt_cassette_event_locations.ix[unchanged.index].dropna()))\n",
    "\n",
    "        for x, (region, x_loc) in enumerate(plotting_order.items()):\n",
    "            ax = fig.add_subplot(gs[y,x])\n",
    "            activated_df = density_to_df(get_densities(activated_tools[region], coverage[rbp]))\n",
    "            activated_plotter = (activated_df > 0).sum() / activated_df.count()\n",
    "\n",
    "            repressed_df = density_to_df(get_densities(repressed_tools[region], coverage[rbp]))\n",
    "            repressed_plotter = (repressed_df > 0).sum() / repressed_df.count()\n",
    "\n",
    "            unchanged_df = density_to_df(get_densities(unchanged_tools[region], coverage[rbp]))\n",
    "            unchanged_plotter = (unchanged_df > 0).sum() / unchanged_df.count()\n",
    "\n",
    "            ax.vlines(0, -1, 1, linestyles=\"dashed\", alpha=.7, color=\".5\")\n",
    "            ax.fill_between(x_loc, 0, activated_plotter, alpha=.7, label=\"Included\", facecolor=\"blue\")\n",
    "            ax.fill_between(x_loc, 0, repressed_plotter * -1, alpha=.7, label=\"Excluded\", facecolor=\"red\")\n",
    "\n",
    "            ax.fill_between(x_loc, 0, unchanged_plotter, alpha=.7, label=\"Unchanging\")\n",
    "            ax.fill_between(x_loc, 0, unchanged_plotter * -1, alpha=.7, label=\"Unchanging\")\n",
    "\n",
    "            significant_activated = get_significant_sites(activated_df, unchanged_df)[0]\n",
    "            significant_repressed = get_significant_sites(repressed_df, unchanged_df)[0]\n",
    "            ax.scatter(x_loc[significant_activated], \n",
    "                       [.9] * len(x_loc[significant_activated]),\n",
    "                       color='black', s=10)\n",
    "            ax.scatter(x_loc[significant_repressed], \n",
    "                       [-.9] * len(x_loc[significant_repressed]),\n",
    "                       color='black', s=10)\n",
    "            ax.set_xlabel(region + \"\\nDistance from Splice Site\\n(bp)\", fontsize=6)\n",
    "            ax.set_xlim((x_loc[0], x_loc[-1]))\n",
    "            ax.set_ylim((-1,1))\n",
    "            [tick.set_fontsize(6) for tick in ax.get_yticklabels()]\n",
    "            [tick.set_fontsize(6) for tick in ax.get_xticklabels()]\n",
    "\n",
    "            if not ax.is_first_col():\n",
    "                sns.despine(ax=ax, left=True)\n",
    "                ax.set_yticklabels([])\n",
    "            else: \n",
    "                sns.despine(ax=ax)\n",
    "                ax.set_ylabel(\" \".join([rbp_name_dict[col[0]], cell_type_dict[col[1]]]) + \"\\nFraction of Sites\", fontsize=8)\n",
    "\n",
    "            if ax.is_first_row() and ax.is_last_col():\n",
    "                ax.legend(bbox_to_anchor=(1.5, 1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "activated_df = density_to_df(get_densities(activated_tools[\"upstream_exon\"], coverage[rbp]))\n",
    "activated_plotter = (activated_df > 0).sum() / activated_df.count()\n",
    "\n",
    "repressed_df = density_to_df(get_densities(repressed_tools[\"upstream_exon\"], coverage[rbp]))\n",
    "repressed_plotter = (repressed_df > 0).sum() / repressed_df.count()\n",
    "\n",
    "unchanged_df = density_to_df(get_densities(unchanged_tools[\"upstream_exon\"], coverage[rbp]))\n",
    "unchanged_plotter = (unchanged_df > 0).sum() / unchanged_df.count()\n",
    "x\n",
    "significant_activated = get_significant_sites(activated_df, unchanged_df)[0]\n",
    "significant_repressed = get_significant_sites(repressed_df, unchanged_df)[0]\n",
    "        \n",
    "cur_alt_cassette_events = alt_cassette_events.ix[col]\n",
    "bound_alt_cassette_events = cur_alt_cassette_events.ix[activated_df[(activated_df[significant_activated[:5]] > 0).apply(any, axis=1)].index]\n",
    "bound_alt_cassette_events.to_csv(\"/nas3/gpratt/Dropbox/TAF15/Data/SplicingAnalysis_array/splicing_event_locations/{}_{}_included_alt_cassette_events.csv\".format(\"_\".join(col), region))            "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Conclusions\n",
    "--\n",
    "1. There appear to be position specific effects\n",
    "2. Some of these effects go away when depth of coverage is normalized out\n",
    "3. Need some way to test if a region is different than background given variance, t-test maybe?  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Get TAF15 KD splicing events bound by TAF15 in mouse whole brain"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "activated_df = density_to_df(get_densities(activated_tools[\"upstream_exon\"], coverage[rbp]))\n",
    "activated_plotter = (activated_df > 0).sum() / activated_df.count()\n",
    "\n",
    "repressed_df = density_to_df(get_densities(repressed_tools[\"downstream_cassette\"], coverage[rbp]))\n",
    "repressed_plotter = (repressed_df > 0).sum() / repressed_df.count()\n",
    "\n",
    "unchanged_df = density_to_df(get_densities(unchanged_tools[\"downstream_cassette\"], coverage[rbp]))\n",
    "unchanged_plotter = (unchanged_df > 0).sum() / unchanged_df.count()\n",
    "\n",
    "significant_activated = get_significant_sites(activated_df, unchanged_df)[0]\n",
    "significant_repressed = get_significant_sites(repressed_df, unchanged_df)[0]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "taf15_whole_brain_alt_cassette = alt_cassette_events.ix['taf15', 'whole brain']\n",
    "bound_alt_cassette_events = taf15_whole_brain_alt_cassette.ix[activated_df[(activated_df[significant_activated[:-6]] > 0).apply(any, axis=1)].index]\n",
    "\n",
    "alt_cassette_exons[alt_cassette_exons.event_id.isin(bound_alt_cassette_events.old_splice_annotation)].to_csv(\"/nas3/gpratt/Dropbox/TAF15/Data/SplicingAnalysis_array/splicing_event_locations/taf15_whole_brain_included_alt_cassette_events.csv\", index=None)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Notes on how to parse \n",
    "\n",
    "(if I want to re-parse stephs list of alt-cassette exons)\n",
    "Is the file you want\n",
    "\n",
    "Original code is here:\n",
    "/nas3/shuelga/old_projects/TLS/jxnarray/get_event_exons.pl\n",
    "\n",
    "File to parse is here:\n",
    "/nas3/shuelga/projects/tdp43_eveline/ffmm9eventinfo_coords_allevents_seq.txt\n",
    "\n",
    "spans_with_positions is the important column.\n",
    "\n",
    "You want to parse the span, split em up by ;.  Then I'm not sure what the 14.16,16.16 stuff is, but I think those are the exons and spans, You want the span across the exon so make sure they are equal.  This won't always work, so build in some robustness into your parser.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "event_annotations = pd.read_table(\"/nas3/shuelga/projects/tdp43_eveline/ffmm9eventinfo_coords_allevents_seq.txt\", index_col=0)\n",
    "alt_cassette_annotations = event_annotations[event_annotations.notes2 == \"alt_cassette\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Future Ideas"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Test Hypothesis of compensatory mutations preventing observation of splicing events\n",
    "\n",
    "At the location of TAF15 specific knockdown spice events does FUS / TDP43 also bind?\n",
    "\n",
    "How many unique gene expression changes are bound by other RBPs in clip"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
